{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "112568e1-e3ee-40aa-ab1b-72bd847b6898",
   "metadata": {},
   "source": [
    "## Step 2: Micov Filter + Combine Technical Replicates\n",
    "**Goal: Run Micov, combine technical replicates and export the files**\n",
    "\n",
    "Github: https://github.com/biocore/micov\n",
    "\n",
    "Conda: https: comad_EAC_figures, terminal: comad"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "d0617143-9e81-4d0e-8a40-3442efb3b594",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "fe9e1816-7733-46ac-8be5-4862ddb2e6ba",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "from biom import load_table\n",
    "from qiime2 import Artifact\n",
    "\n",
    "from qiime2.plugins.feature_table.methods import filter_features\n",
    "from python_scripts.micov_filter import *"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "e772e973-e0e1-45ff-9225-2a47f9ea79d2",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>HCC11.B.S137.L003.aligned</th>\n",
       "      <th>CRC9.B.2.S94.L003.aligned</th>\n",
       "      <th>HCC.3.P.S115.L003.aligned</th>\n",
       "      <th>CRC7.T.S87.L003.aligned</th>\n",
       "      <th>CRC2.B.S65.L003.aligned</th>\n",
       "      <th>CRC1.T.2.S56.L003.aligned</th>\n",
       "      <th>CRC10.T.2.S60.L003.aligned</th>\n",
       "      <th>CRC11.T.S63.L003.aligned</th>\n",
       "      <th>HCC.3.P.2.S116.L003.aligned</th>\n",
       "      <th>HCC13.T.S147.L003.aligned</th>\n",
       "      <th>...</th>\n",
       "      <th>HCC16.B.S157.L003.aligned</th>\n",
       "      <th>HCC12.B.S141.L003.aligned</th>\n",
       "      <th>HCC.5.P.2.S120.L003.aligned</th>\n",
       "      <th>CRC.8.P.S49.L003.aligned</th>\n",
       "      <th>HCC10.T.S135.L003.aligned</th>\n",
       "      <th>HCC16.T.2.S160.L003.aligned</th>\n",
       "      <th>CRC10.T.S59.L003.aligned</th>\n",
       "      <th>CRC6.T.2.S84.L003.aligned</th>\n",
       "      <th>HCC.10.P.2.S100.L003.aligned</th>\n",
       "      <th>CRC5.B.2.S78.L003.aligned</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G000001985</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000002415</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000002445</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000002495</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000002655</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G907163265</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G907166805</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G910573815</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G910576415</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G913698045</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>4141 rows × 153 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            HCC11.B.S137.L003.aligned  CRC9.B.2.S94.L003.aligned  \\\n",
       "G000001985                        0.0                        0.0   \n",
       "G000002415                        0.0                        0.0   \n",
       "G000002445                        0.0                        0.0   \n",
       "G000002495                        0.0                        0.0   \n",
       "G000002655                        0.0                        0.0   \n",
       "...                               ...                        ...   \n",
       "G907163265                        0.0                        0.0   \n",
       "G907166805                        0.0                        0.0   \n",
       "G910573815                        0.0                        0.0   \n",
       "G910576415                        0.0                        0.0   \n",
       "G913698045                        0.0                        0.0   \n",
       "\n",
       "            HCC.3.P.S115.L003.aligned  CRC7.T.S87.L003.aligned  \\\n",
       "G000001985                        0.0                      0.0   \n",
       "G000002415                        0.0                      0.0   \n",
       "G000002445                        0.0                      1.0   \n",
       "G000002495                        0.0                      0.0   \n",
       "G000002655                        0.0                      0.0   \n",
       "...                               ...                      ...   \n",
       "G907163265                        0.0                      0.0   \n",
       "G907166805                        0.0                      0.0   \n",
       "G910573815                        0.0                      0.0   \n",
       "G910576415                        0.0                      0.0   \n",
       "G913698045                        0.0                      0.0   \n",
       "\n",
       "            CRC2.B.S65.L003.aligned  CRC1.T.2.S56.L003.aligned  \\\n",
       "G000001985                      0.0                        0.0   \n",
       "G000002415                      0.0                        0.0   \n",
       "G000002445                      0.0                        0.0   \n",
       "G000002495                      0.0                        0.0   \n",
       "G000002655                      0.0                        0.0   \n",
       "...                             ...                        ...   \n",
       "G907163265                      0.0                        0.0   \n",
       "G907166805                      0.0                        0.0   \n",
       "G910573815                      0.0                        0.0   \n",
       "G910576415                      0.0                        0.0   \n",
       "G913698045                      0.0                        0.0   \n",
       "\n",
       "            CRC10.T.2.S60.L003.aligned  CRC11.T.S63.L003.aligned  \\\n",
       "G000001985                         0.0                       0.0   \n",
       "G000002415                         1.0                       0.0   \n",
       "G000002445                         0.0                       1.0   \n",
       "G000002495                         0.0                       0.0   \n",
       "G000002655                         0.0                       0.0   \n",
       "...                                ...                       ...   \n",
       "G907163265                         0.0                       0.0   \n",
       "G907166805                         0.0                       0.0   \n",
       "G910573815                         0.0                       0.0   \n",
       "G910576415                         0.0                       0.0   \n",
       "G913698045                         0.0                       0.0   \n",
       "\n",
       "            HCC.3.P.2.S116.L003.aligned  HCC13.T.S147.L003.aligned  ...  \\\n",
       "G000001985                          0.0                        0.0  ...   \n",
       "G000002415                          0.0                        2.0  ...   \n",
       "G000002445                          0.0                        0.0  ...   \n",
       "G000002495                          0.0                        0.0  ...   \n",
       "G000002655                          0.0                        0.0  ...   \n",
       "...                                 ...                        ...  ...   \n",
       "G907163265                          0.0                        0.0  ...   \n",
       "G907166805                          0.0                        0.0  ...   \n",
       "G910573815                          2.0                        0.0  ...   \n",
       "G910576415                          0.0                        0.0  ...   \n",
       "G913698045                          0.0                        0.0  ...   \n",
       "\n",
       "            HCC16.B.S157.L003.aligned  HCC12.B.S141.L003.aligned  \\\n",
       "G000001985                        0.0                        0.0   \n",
       "G000002415                        0.0                        0.0   \n",
       "G000002445                        0.0                        0.0   \n",
       "G000002495                        0.0                        0.0   \n",
       "G000002655                        0.0                        0.0   \n",
       "...                               ...                        ...   \n",
       "G907163265                        0.0                        0.0   \n",
       "G907166805                        0.0                        0.0   \n",
       "G910573815                        0.0                        0.0   \n",
       "G910576415                        0.0                        0.0   \n",
       "G913698045                        0.0                        0.0   \n",
       "\n",
       "            HCC.5.P.2.S120.L003.aligned  CRC.8.P.S49.L003.aligned  \\\n",
       "G000001985                          0.0                       0.0   \n",
       "G000002415                          0.0                       0.0   \n",
       "G000002445                          0.0                       0.0   \n",
       "G000002495                          0.0                       0.0   \n",
       "G000002655                          0.0                       0.0   \n",
       "...                                 ...                       ...   \n",
       "G907163265                          0.0                       0.0   \n",
       "G907166805                          0.0                       0.0   \n",
       "G910573815                          0.0                       0.0   \n",
       "G910576415                          0.0                       0.0   \n",
       "G913698045                          0.0                       0.0   \n",
       "\n",
       "            HCC10.T.S135.L003.aligned  HCC16.T.2.S160.L003.aligned  \\\n",
       "G000001985                        0.0                          0.0   \n",
       "G000002415                        0.0                          1.0   \n",
       "G000002445                        0.0                          0.0   \n",
       "G000002495                        0.0                          0.0   \n",
       "G000002655                        0.0                          0.0   \n",
       "...                               ...                          ...   \n",
       "G907163265                        0.0                          0.0   \n",
       "G907166805                        0.0                          0.0   \n",
       "G910573815                        0.0                          0.0   \n",
       "G910576415                        0.0                          0.0   \n",
       "G913698045                        0.0                          0.0   \n",
       "\n",
       "            CRC10.T.S59.L003.aligned  CRC6.T.2.S84.L003.aligned  \\\n",
       "G000001985                       0.0                        0.0   \n",
       "G000002415                       1.0                        0.0   \n",
       "G000002445                       0.0                        1.0   \n",
       "G000002495                       0.0                        0.0   \n",
       "G000002655                       0.0                        0.0   \n",
       "...                              ...                        ...   \n",
       "G907163265                       0.0                        0.0   \n",
       "G907166805                       0.0                        0.0   \n",
       "G910573815                       0.0                        0.0   \n",
       "G910576415                       0.0                        0.0   \n",
       "G913698045                       0.0                        0.0   \n",
       "\n",
       "            HCC.10.P.2.S100.L003.aligned  CRC5.B.2.S78.L003.aligned  \n",
       "G000001985                           0.0                        0.0  \n",
       "G000002415                           0.0                        2.0  \n",
       "G000002445                           0.0                        0.0  \n",
       "G000002495                           0.0                        0.0  \n",
       "G000002655                           0.0                        0.0  \n",
       "...                                  ...                        ...  \n",
       "G907163265                           0.0                        0.0  \n",
       "G907166805                           0.0                        0.0  \n",
       "G910573815                           0.0                        0.0  \n",
       "G910576415                           0.0                        0.0  \n",
       "G913698045                           0.0                        0.0  \n",
       "\n",
       "[4141 rows x 153 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Import biom files from SCRuB into pandas df (qza was in differnt qiime2 format [like an older verison] so couldn't go q2->df)\n",
    "\n",
    "# Load the BIOM table\n",
    "biom_table = load_table('processed_data/SCRuB/pese_pangenome_align-RS210_masked_none_scrubbed.qza_extractBiom/data/feature-table.biom')\n",
    "\n",
    "# Convert the BIOM table to a pandas DataFrame\n",
    "df_rs210c = biom_table.to_dataframe(dense=True)\n",
    "\n",
    "df_rs210c"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "35523458-5dde-4c7d-95f9-9f816734f83e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>empo_1</th>\n",
       "      <th>empo_2</th>\n",
       "      <th>empo_3</th>\n",
       "      <th>empo_4</th>\n",
       "      <th>country</th>\n",
       "      <th>latitude</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>elevation</th>\n",
       "      <th>env_biome</th>\n",
       "      <th>...</th>\n",
       "      <th>run_center</th>\n",
       "      <th>run_date</th>\n",
       "      <th>run_prefix</th>\n",
       "      <th>runid</th>\n",
       "      <th>sample_plate</th>\n",
       "      <th>sample_project</th>\n",
       "      <th>sample_well</th>\n",
       "      <th>sequencing_meth</th>\n",
       "      <th>well_description</th>\n",
       "      <th>sampleid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15336.BLANK1.8D.2</td>\n",
       "      <td>Control</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Sterile water blank</td>\n",
       "      <td>Sterile water blank</td>\n",
       "      <td>USA</td>\n",
       "      <td>32.5</td>\n",
       "      <td>256318</td>\n",
       "      <td>193</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>...</td>\n",
       "      <td>IGM</td>\n",
       "      <td>4/18/21</td>\n",
       "      <td>BLANK1_8D_2_S12_L003</td>\n",
       "      <td>210418_A00953_0278_BH5G5VDSX2</td>\n",
       "      <td>HCC_Tissue_Plasma_P1.2</td>\n",
       "      <td>HCC_Tissue_Plasma</td>\n",
       "      <td>H8</td>\n",
       "      <td>sequencing by synthesis</td>\n",
       "      <td>HCC_Tissue_Plasma_P1.2.BLANK1.8D.2.H8</td>\n",
       "      <td>BLANK1.8D.2.S12.L003.aligned</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>15336.BLANK1.8E.2</td>\n",
       "      <td>Control</td>\n",
       "      <td>Negative</td>\n",
       "      <td>Sterile water blank</td>\n",
       "      <td>Sterile water blank</td>\n",
       "      <td>USA</td>\n",
       "      <td>32.5</td>\n",
       "      <td>256318</td>\n",
       "      <td>193</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>...</td>\n",
       "      <td>IGM</td>\n",
       "      <td>4/18/21</td>\n",
       "      <td>BLANK1_8E_2_S14_L003</td>\n",
       "      <td>210418_A00953_0278_BH5G5VDSX2</td>\n",
       "      <td>HCC_Tissue_Plasma_P1.2</td>\n",
       "      <td>HCC_Tissue_Plasma</td>\n",
       "      <td>J8</td>\n",
       "      <td>sequencing by synthesis</td>\n",
       "      <td>HCC_Tissue_Plasma_P1.2.BLANK1.8E.2.J8</td>\n",
       "      <td>BLANK1.8E.2.S14.L003.aligned</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2 rows × 60 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         sample_name   empo_1    empo_2               empo_3  \\\n",
       "0  15336.BLANK1.8D.2  Control  Negative  Sterile water blank   \n",
       "1  15336.BLANK1.8E.2  Control  Negative  Sterile water blank   \n",
       "\n",
       "                empo_4 country  latitude  taxon_id  elevation    env_biome  \\\n",
       "0  Sterile water blank     USA      32.5    256318        193  urban biome   \n",
       "1  Sterile water blank     USA      32.5    256318        193  urban biome   \n",
       "\n",
       "   ...  run_center run_date            run_prefix  \\\n",
       "0  ...         IGM  4/18/21  BLANK1_8D_2_S12_L003   \n",
       "1  ...         IGM  4/18/21  BLANK1_8E_2_S14_L003   \n",
       "\n",
       "                           runid            sample_plate     sample_project  \\\n",
       "0  210418_A00953_0278_BH5G5VDSX2  HCC_Tissue_Plasma_P1.2  HCC_Tissue_Plasma   \n",
       "1  210418_A00953_0278_BH5G5VDSX2  HCC_Tissue_Plasma_P1.2  HCC_Tissue_Plasma   \n",
       "\n",
       "  sample_well          sequencing_meth                       well_description  \\\n",
       "0          H8  sequencing by synthesis  HCC_Tissue_Plasma_P1.2.BLANK1.8D.2.H8   \n",
       "1          J8  sequencing by synthesis  HCC_Tissue_Plasma_P1.2.BLANK1.8E.2.J8   \n",
       "\n",
       "                       sampleid  \n",
       "0  BLANK1.8D.2.S12.L003.aligned  \n",
       "1  BLANK1.8E.2.S14.L003.aligned  \n",
       "\n",
       "[2 rows x 60 columns]"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Metadata import\n",
    "meta = pd.read_csv('qiita_downloads/pese_pangenome_align-RS210-masked/meta_pese_pangenome_align-rs210-masked.tsv', sep='\\t')\n",
    "meta[:2]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "849ad701-9ed0-4f93-a2c1-2119ad5f4f79",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "ca613ddb-8f2d-4771-98de-b051db497a1f",
   "metadata": {},
   "outputs": [],
   "source": [
    "def combo_tech_rep(df, fn, micov_filter=False):\n",
    "    \n",
    "    #Rename headers to match meta\n",
    "    mapping = dict(zip(meta['sampleid'], meta['sample_name']))\n",
    "\n",
    "    # Rename columns in df using the mapping\n",
    "    df.rename(columns=mapping, inplace=True)\n",
    "    \n",
    "    #Find cols ending with '.2'\n",
    "    rep_cols = [col for col in df.columns if col.endswith('.2')]\n",
    "\n",
    "    #Create new df with paired technical replicates\n",
    "    df_techReps = pd.DataFrame()\n",
    "    for col in rep_cols:\n",
    "        col_name = col[:-2]  #remove the '.2'\n",
    "        if col_name in df.columns:\n",
    "            df_techReps[col_name] = df[col_name] + df[col]  \n",
    "            \n",
    "    if micov_filter != False:\n",
    "        df_techReps = micov_pass_fail(df_techReps)\n",
    "            \n",
    "    #Export subset df as qza\n",
    "    qza = Artifact.import_data(\"FeatureTable[Frequency]\", df_techReps.T)\n",
    "    \n",
    "    qza.save('processed_data/TechRep_merged/'+ fn + '.qza')\n",
    "\n",
    "    return(df_techReps)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "8d1d79a0-a0bc-4911-accd-cdb3395663df",
   "metadata": {},
   "outputs": [],
   "source": [
    "def micov_pass_fail(df):\n",
    "    '''Taking micov pass_ and fail_ lists - subset dataset'''\n",
    "    \n",
    "    #Import pass and fail lists\n",
    "    pass_ = list(pd.read_csv('micov/micov_filtered_lists/PASSmicov.csv', sep='\\t')['Pass'])\n",
    "    fail_ = list(pd.read_csv('micov/micov_filtered_lists/FAILmicov.csv', sep='\\t')['Fail'])\n",
    "    \n",
    "    #Check to make sure none are missed\n",
    "    for i in df.index:\n",
    "        if i not in pass_ and i not in fail_:\n",
    "            print(i)\n",
    "    \n",
    "    subset_df = df.loc[pass_]\n",
    "    \n",
    "    return(subset_df)\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "b04710ac-097e-4880-804a-b8e5024794fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "def save_pass_fail(pass_, fail_):\n",
    "    '''Save pass and fail lists for future use'''\n",
    "    \n",
    "    #pd.DataFrame(pass_, columns=['Pass'])['Pass'].to_csv('micov/micov_filtered_lists/PASSmicov.csv', sep='\\t', index=False)\n",
    "    #pd.DataFrame(fail_, columns=['Fail'])['Fail'].to_csv('micov/micov_filtered_lists/FAILmicov.csv', sep='\\t', index=False)\n",
    "    #A bit repetative to save here and upload there but wanted to make sure we had clear copy of list\n",
    "    \n",
    "    print('already have so not re-running')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4ac636bb-ac3c-431a-b706-78a8731bfff3",
   "metadata": {},
   "source": [
    "### Determine MissMapped Microbes from Micov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ae00b5cc-528b-4b4f-a621-c4bc6b28b2f9",
   "metadata": {},
   "outputs": [],
   "source": [
    "taxa = choose_taxa('RS210')\n",
    "\n",
    "#Import dataset\n",
    "gen_len = pd.read_csv('micov/all_microbes.coverage', sep='\\t')\n",
    "pos = pd.read_csv('micov/all_microbes.covered_positions', sep='\\t')\n",
    " \n",
    "#Import biom table from prior to SCRub becuase Micov run at the SAM level while SCRuB doesn't \n",
    "micovRef_table = biom.load_table('qiita_downloads/pese_pangenome_align-RS210-masked/biom/none.biom')\n",
    "micovRef_df = micovRef_table.to_dataframe(dense=True)\n",
    "\n",
    "cov= pd.read_csv('micov/coverage_percentage.txt', sep='\\t')\n",
    "cov = cov.rename(columns={'genome_id': 'GOTU', 'percent_covered':'coverage_percent'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "1e23915a-b1c5-44b4-8192-24be64772726",
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000001985 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Eurotiales; f__Trichocomaceae; g__Talaromyces; s__Talaromyces marneffei\n",
      "Coverage Percentage: 0.0008585522888881%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 0\n",
      "Length of fail 1\n",
      "G000002415 d__Eukaryota; p__Apicomplexa; c__Aconoidasida; o__Haemosporida; f__Plasmodiidae; g__Plasmodium; s__Plasmodium vivax\n",
      "Coverage Percentage: 0.000424846014562%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 0\n",
      "Length of fail 2\n",
      "G000002445 d__Eukaryota; p__Euglenozoa; c__Kinetoplastea; o__Trypanosomatida; f__Trypanosomatidae; g__Trypanosoma; s__Trypanosoma brucei\n",
      "Coverage Percentage: 0.0018753081890681%\n",
      "Adj, read counts 66\n",
      "Number of samples per group [1, 1, 54, 10]\n",
      "Number of bins 4\n",
      "total # samples 56\n",
      "Number of total reads 66\n",
      "Number of hits in BIOM tables 26.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 98.48484848484848 %\n",
      "Percentage of reads from samples who only fall into one bin group 69.6969696969697 %\n",
      "Status: Fail - too many reads in top 3 bins\n",
      "----------------------------\n",
      "Length of pass 0\n",
      "Length of fail 3\n",
      "G000002495 d__Eukaryota; p__Ascomycota; c__Sordariomycetes; o__Magnaporthales; f__Pyriculariaceae; g__Pyricularia; s__Pyricularia oryzae\n",
      "Coverage Percentage: 0.0010590490359469%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 0\n",
      "Length of fail 4\n",
      "G000002655 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Eurotiales; f__Aspergillaceae; g__Aspergillus; s__Aspergillus fumigatus\n",
      "Coverage Percentage: 0.0008099343415495%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 0\n",
      "Length of fail 5\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000002725 d__Eukaryota; p__Euglenozoa; c__Kinetoplastea; o__Trypanosomatida; f__Trypanosomatidae; g__Leishmania; s__Leishmania major\n",
      "Coverage Percentage: 0.0026509787970698%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 0\n",
      "Length of fail 6\n",
      "G000002765 d__Eukaryota; p__Apicomplexa; c__Aconoidasida; o__Haemosporida; f__Plasmodiidae; g__Plasmodium; s__Plasmodium falciparum\n",
      "Coverage Percentage: 0.0032065637502597%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 0\n",
      "Length of fail 7\n",
      "G000002825 d__Eukaryota; p__Parabasalia; c__; o__Trichomonadida; f__Trichomonadidae; g__Trichomonas; s__Trichomonas vaginalis\n",
      "Coverage Percentage: 0.0004000778210445%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 0\n",
      "Length of fail 8\n",
      "G000002845 d__Eukaryota; p__Euglenozoa; c__Kinetoplastea; o__Trypanosomatida; f__Trypanosomatidae; g__Leishmania; s__Leishmania braziliensis\n",
      "Coverage Percentage: 0.0030338452092263%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 0\n",
      "Length of fail 9\n",
      "G000002855 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Eurotiales; f__Aspergillaceae; g__Aspergillus; s__Aspergillus niger\n",
      "Coverage Percentage: 0.0011291775931997%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 0\n",
      "Length of fail 10\n",
      "G000004695 d__Eukaryota; p__Evosea; c__Eumycetozoa; o__Dictyosteliales; f__Dictyosteliaceae; g__Dictyostelium; s__Dictyostelium discoideum\n",
      "Coverage Percentage: 0.0082296049851%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 0\n",
      "Length of fail 11\n",
      "G000005845 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Escherichia; s__Escherichia coli\n",
      "Coverage Percentage: 1.0075723039986626%\n",
      "Adj, read counts 276\n",
      "Number of samples per group [4, 2, 1, 2, 6, 1, 2, 1, 2, 1, 3, 4, 3, 2, 2, 1, 1, 4, 1, 1, 3, 2, 1, 2, 1, 2, 3, 4, 4, 3, 3, 2, 1, 4, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 1, 1, 1, 1, 3, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 3, 2, 2, 1, 2, 3, 2, 1, 1, 2, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 2, 3, 1, 1, 1, 3, 1, 1, 1, 1, 4, 1, 1, 2, 4, 1, 4, 3, 1, 2, 1, 3, 1, 1, 3, 3, 4, 1, 1, 1, 1, 1, 1, 1, 2, 1]\n",
      "Number of bins 184\n",
      "total # samples 60\n",
      "Number of total reads 272\n",
      "Number of hits in BIOM tables 50.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 8.333333333333334 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.76470588235294 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 1\n",
      "Length of fail 11\n",
      "G000006355 d__Eukaryota; p__Apicomplexa; c__Aconoidasida; o__Haemosporida; f__Plasmodiidae; g__Plasmodium; s__Plasmodium knowlesi\n",
      "Coverage Percentage: 0.0044181705831071%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 1\n",
      "Length of fail 12\n",
      "G000006405 d__Eukaryota; p__Perkinsozoa; c__; o__Perkinsida; f__Perkinsidae; g__Perkinsus; s__Perkinsus marinus\n",
      "Coverage Percentage: 0.0004588158498774%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 1\n",
      "Length of fail 13\n",
      "G000006445 d__Eukaryota; p__Ascomycota; c__Saccharomycetes; o__Saccharomycetales; f__Debaryomycetaceae; g__Debaryomyces; s__Debaryomyces hansenii\n",
      "Coverage Percentage: 0.0040961779294321%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 1\n",
      "Length of fail 14\n",
      "G000006765 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas aeruginosa\n",
      "Coverage Percentage: 0.584780291947965%\n",
      "Adj, read counts 210\n",
      "Number of samples per group [3, 1, 3, 1, 1, 1, 1, 1, 1, 1, 2, 3, 1, 1, 1, 1, 1, 2, 2, 1, 1, 3, 2, 1, 1, 1, 1, 1, 1, 3, 2, 2, 2, 1, 1, 2, 3, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 2, 2, 2, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 1, 1, 2, 1, 1, 1, 3, 1, 1, 1, 1, 2, 1, 1, 1, 1, 3, 3, 2, 1, 2, 2, 1, 1, 2, 1, 2, 1, 1, 4, 1, 3, 1, 2, 1, 1, 1, 2, 1, 1, 1, 2, 3, 1, 1, 1, 2, 3, 1]\n",
      "Number of bins 168\n",
      "total # samples 52\n",
      "Number of total reads 209\n",
      "Number of hits in BIOM tables 58.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 6.666666666666667 %\n",
      "Percentage of reads from samples who only fall into one bin group 12.440191387559809 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 2\n",
      "Length of fail 14\n",
      "G000006925 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Shigella; s__Shigella flexneri\n",
      "Coverage Percentage: 0.8626916609372024%\n",
      "Adj, read counts 242\n",
      "Number of samples per group [1, 1, 2, 1, 2, 2, 5, 1, 3, 1, 3, 1, 2, 1, 4, 1, 1, 3, 2, 1, 2, 1, 2, 1, 4, 2, 2, 2, 2, 3, 3, 3, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 2, 2, 1, 1, 2, 2, 1, 1, 1, 1, 2, 2, 2, 2, 1, 1, 1, 1, 1, 2, 2, 1, 2, 2, 2, 2, 1, 2, 1, 1, 2, 2, 1, 1, 1, 2, 3, 1, 2, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 3, 6, 1, 1, 2, 1, 1, 1, 4, 1, 5, 1, 1, 1, 2, 1]\n",
      "Number of bins 162\n",
      "total # samples 55\n",
      "Number of total reads 238\n",
      "Number of hits in BIOM tables 36.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 7.024793388429752 %\n",
      "Percentage of reads from samples who only fall into one bin group 12.184873949579831 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 3\n",
      "Length of fail 14\n",
      "G000006945 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Salmonella; s__Salmonella enterica\n",
      "Coverage Percentage: 0.0248212476342563%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 3\n",
      "Length of fail 15\n",
      "G000007145 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Xanthomonas; s__Xanthomonas campestris\n",
      "Coverage Percentage: 0.0191876266206058%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 3\n",
      "Length of fail 16\n",
      "G000008425 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__Bacillus licheniformis\n",
      "Coverage Percentage: 0.0334861206660754%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 3\n",
      "Length of fail 17\n",
      "G000008865 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Escherichia; s__Escherichia coli\n",
      "Coverage Percentage: 0.7605844517391184%\n",
      "Adj, read counts 247\n",
      "Number of samples per group [1, 2, 2, 1, 2, 2, 5, 1, 1, 1, 4, 1, 2, 2, 1, 2, 1, 1, 4, 1, 3, 2, 1, 2, 1, 2, 2, 2, 4, 4, 1, 2, 3, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 4, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 3, 1, 1, 1, 1, 2, 2, 1, 1, 2, 1, 2, 2, 2, 1, 1, 1, 2, 1, 2, 4, 1, 4, 1, 1, 1, 2, 3, 1, 2, 1, 1, 1, 3, 1, 1, 1, 1, 1, 4, 1, 4, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 4, 1, 1, 1, 1, 3, 1, 1, 1]\n",
      "Number of bins 178\n",
      "total # samples 59\n",
      "Number of total reads 242\n",
      "Number of hits in BIOM tables 42.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 5.263157894736842 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.983471074380166 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 4\n",
      "Length of fail 17\n",
      "G000009145 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Nitrosomonadaceae; g__Nitrosomonas; s__Nitrosomonas europaea\n",
      "Coverage Percentage: 0.0128374087068213%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 4\n",
      "Length of fail 18\n",
      "G000010125 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus saprophyticus\n",
      "Coverage Percentage: 0.2967874724731656%\n",
      "Adj, read counts 42\n",
      "Number of samples per group [1, 3, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1]\n",
      "Number of bins 35\n",
      "total # samples 22\n",
      "Number of total reads 42\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 23.809523809523807 %\n",
      "Percentage of reads from samples who only fall into one bin group 35.714285714285715 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 5\n",
      "Length of fail 18\n",
      "G000010425 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium adolescentis\n",
      "Coverage Percentage: 0.5777536375795889%\n",
      "Adj, read counts 70\n",
      "Number of samples per group [2, 2, 1, 3, 2, 1, 3, 1, 4, 1, 1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 4, 1, 1, 1, 1, 2, 3, 3, 1, 1]\n",
      "Number of bins 63\n",
      "total # samples 32\n",
      "Number of total reads 69\n",
      "Number of hits in BIOM tables 23.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.714285714285714 %\n",
      "Percentage of reads from samples who only fall into one bin group 26.08695652173913 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 6\n",
      "Length of fail 18\n",
      "G000010505 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Porphyromonadaceae; g__Porphyromonas; s__Porphyromonas gingivalis\n",
      "Coverage Percentage: 0.0199585032990981%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 6\n",
      "Length of fail 19\n",
      "G000011065 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides thetaiotaomicron\n",
      "Coverage Percentage: 0.103218933937181%\n",
      "Adj, read counts 41\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2]\n",
      "Number of bins 40\n",
      "total # samples 23\n",
      "Number of total reads 40\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.195121951219512 %\n",
      "Percentage of reads from samples who only fall into one bin group 35.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 7\n",
      "Length of fail 19\n",
      "G000011325 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium glutamicum\n",
      "Coverage Percentage: 0.0157732471827983%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 7\n",
      "Length of fail 20\n",
      "G000011565 d__Bacteria; p__Chlamydiae; c__Chlamydiia; o__Parachlamydiales; f__Parachlamydiaceae; g__Candidatus Protochlamydia; s__Candidatus Protochlamydia amoebophila\n",
      "Coverage Percentage: 0.0381317571113676%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 7\n",
      "Length of fail 21\n",
      "G000011645 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__Bacillus licheniformis\n",
      "Coverage Percentage: 0.0334865013166068%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 7\n",
      "Length of fail 22\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000012825 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola vulgatus\n",
      "Coverage Percentage: 0.5408672818291176%\n",
      "Adj, read counts 169\n",
      "Number of samples per group [1, 2, 2, 1, 4, 1, 1, 1, 2, 2, 1, 1, 2, 1, 1, 1, 2, 1, 2, 1, 1, 2, 1, 2, 1, 1, 1, 2, 2, 1, 2, 2, 2, 1, 1, 3, 1, 1, 2, 1, 1, 2, 1, 2, 1, 1, 1, 1, 3, 1, 1, 2, 1, 1, 1, 1, 1, 1, 4, 1, 1, 1, 4, 1, 2, 1, 1, 4, 1, 1, 3, 1, 3, 1, 1, 2, 1, 3, 2, 1, 1, 5, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 3, 1, 2]\n",
      "Number of bins 140\n",
      "total # samples 48\n",
      "Number of total reads 167\n",
      "Number of hits in BIOM tables 78.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 9.467455621301776 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.976047904191617 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 8\n",
      "Length of fail 22\n",
      "G000012845 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Tannerellaceae; g__Parabacteroides; s__Parabacteroides distasonis\n",
      "Coverage Percentage: 0.117284462521036%\n",
      "Adj, read counts 35\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 30\n",
      "total # samples 20\n",
      "Number of total reads 35\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.285714285714285 %\n",
      "Percentage of reads from samples who only fall into one bin group 37.142857142857146 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 9\n",
      "Length of fail 22\n",
      "G000013325 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium aromaticivorans\n",
      "Coverage Percentage: 0.0602831702711372%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 9\n",
      "Length of fail 23\n",
      "G000013605 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Rhodoferax; s__Rhodoferax ferrireducens\n",
      "Coverage Percentage: 0.0973479034585669%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 9\n",
      "Length of fail 24\n",
      "G000013645 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia xenovorans\n",
      "Coverage Percentage: 0.0435815684390933%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 9\n",
      "Length of fail 25\n",
      "G000013985 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingopyxis; s__Sphingopyxis alaskensis\n",
      "Coverage Percentage: 0.036102442013046%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 9\n",
      "Length of fail 26\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000014145 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Cytophagaceae; g__Cytophaga; s__Cytophaga hutchinsonii\n",
      "Coverage Percentage: 0.0102183109425252%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 9\n",
      "Length of fail 27\n",
      "G000014805 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas hydrophila\n",
      "Coverage Percentage: 0.026578434414288%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 9\n",
      "Length of fail 28\n",
      "G000014945 d__Archaea; p__Euryarchaeota; c__Methanomicrobia; o__Methanotrichales; f__Methanotrichaceae; g__Methanothrix; s__Methanothrix thermoacetophila\n",
      "Coverage Percentage: 0.0203248680080724%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 9\n",
      "Length of fail 29\n",
      "G000015325 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax citrulli\n",
      "Coverage Percentage: 0.0805563920899302%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 9\n",
      "Length of fail 30\n",
      "G000015505 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Polaromonas; s__Polaromonas naphthalenivorans\n",
      "Coverage Percentage: 0.1463391090663348%\n",
      "Adj, read counts 68\n",
      "Number of samples per group [1, 1, 15, 1, 1, 1, 1, 13, 1, 2, 1, 1, 1, 1, 1, 2, 1]\n",
      "Number of bins 19\n",
      "total # samples 22\n",
      "Number of total reads 66\n",
      "Number of hits in BIOM tables 22.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 76.47058823529413 %\n",
      "Percentage of reads from samples who only fall into one bin group 10.606060606060606 %\n",
      "Status: Fail - too many reads in top 3 bins\n",
      "----------------------------\n",
      "Length of pass 9\n",
      "Length of fail 31\n",
      "G000015725 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Methylibium; s__Methylibium petroleiphilum\n",
      "Coverage Percentage: 0.0522648196174611%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 9\n",
      "Length of fail 32\n",
      "G000016345 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Polynucleobacter; s__Polynucleobacter asymbioticus\n",
      "Coverage Percentage: 0.1334111294796456%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 9\n",
      "Length of fail 33\n",
      "G000016525 d__Archaea; p__Euryarchaeota; c__Methanobacteria; o__Methanobacteriales; f__Methanobacteriaceae; g__Methanobrevibacter; s__Methanobrevibacter smithii\n",
      "Coverage Percentage: 0.0468928748731895%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 9\n",
      "Length of fail 34\n",
      "G000016745 d__Bacteria; p__Proteobacteria; c__Deltaproteobacteria; o__Desulfuromonadales; f__Geobacteraceae; g__Geobacter; s__Geobacter uraniireducens\n",
      "Coverage Percentage: 0.0033486723292975%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 9\n",
      "Length of fail 35\n",
      "G000016765 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Rhizorhabdus; s__Rhizorhabdus wittichii\n",
      "Coverage Percentage: 0.0202864240207489%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 9\n",
      "Length of fail 36\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000017565 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Parvibaculaceae; g__Parvibaculum; s__Parvibaculum lavamentivorans\n",
      "Coverage Percentage: 0.0268727592729539%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 9\n",
      "Length of fail 37\n",
      "G000019725 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium radiotolerans\n",
      "Coverage Percentage: 0.0300176685359465%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 9\n",
      "Length of fail 38\n",
      "G000019785 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Leptothrix; s__Leptothrix cholodnii\n",
      "Coverage Percentage: 0.1449259716507281%\n",
      "Adj, read counts 58\n",
      "Number of samples per group [10, 1, 1, 1, 1, 15, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 5, 1, 1]\n",
      "Number of bins 29\n",
      "total # samples 24\n",
      "Number of total reads 57\n",
      "Number of hits in BIOM tables 27.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 60.3448275862069 %\n",
      "Percentage of reads from samples who only fall into one bin group 17.543859649122805 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 10\n",
      "Length of fail 38\n",
      "G000019945 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylorubrum; s__Methylorubrum populi\n",
      "Coverage Percentage: 0.0926819235371566%\n",
      "Adj, read counts 34\n",
      "Number of samples per group [1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 33\n",
      "total # samples 11\n",
      "Number of total reads 34\n",
      "Number of hits in BIOM tables 37.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.705882352941178 %\n",
      "Percentage of reads from samples who only fall into one bin group 14.705882352941178 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 11\n",
      "Length of fail 38\n",
      "G000019965 d__Bacteria; p__Verrucomicrobia; c__Opitutae; o__Opitutales; f__Opitutaceae; g__Opitutus; s__Opitutus terrae\n",
      "Coverage Percentage: 0.0070330275337153%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 11\n",
      "Length of fail 39\n",
      "G000020045 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia phymatum\n",
      "Coverage Percentage: 0.0045870385963569%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 11\n",
      "Length of fail 40\n",
      "G000020225 d__Bacteria; p__Verrucomicrobia; c__Verrucomicrobiae; o__Verrucomicrobiales; f__Akkermansiaceae; g__Akkermansia; s__Akkermansia muciniphila\n",
      "Coverage Percentage: 0.0902743213285377%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 11\n",
      "Length of fail 41\n",
      "G000020425 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium longum\n",
      "Coverage Percentage: 0.2554763078113549%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 11\n",
      "Length of fail 42\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000021045 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Azotobacter; s__Azotobacter vinelandii\n",
      "Coverage Percentage: 0.0021061193390587%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 11\n",
      "Length of fail 43\n",
      "G000021845 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylorubrum; s__Methylorubrum extorquens\n",
      "Coverage Percentage: 0.0729358727356388%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 11\n",
      "Length of fail 44\n",
      "G000022005 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Caulobacter; s__Caulobacter vibrioides\n",
      "Coverage Percentage: 0.0509531579703724%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 11\n",
      "Length of fail 45\n",
      "G000022025 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Pseudarthrobacter; s__Pseudarthrobacter chlorophenolicus\n",
      "Coverage Percentage: 0.0043967869325083%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 11\n",
      "Length of fail 46\n",
      "G000022085 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium nodulans\n",
      "Coverage Percentage: 0.0085981001366419%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 11\n",
      "Length of fail 47\n",
      "G000022305 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax ebreus\n",
      "Coverage Percentage: 0.0802829288413524%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 11\n",
      "Length of fail 48\n",
      "G000022905 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium aurimucosum\n",
      "Coverage Percentage: 0.1256718995078826%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 11\n",
      "Length of fail 49\n",
      "G000023065 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Tolumonas; s__Tolumonas auensis\n",
      "Coverage Percentage: 0.0672084054006404%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 11\n",
      "Length of fail 50\n",
      "G000023145 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium kroppenstedtii\n",
      "Coverage Percentage: 0.0811262365109751%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 11\n",
      "Length of fail 51\n",
      "G000023205 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Micrococcus; s__Micrococcus luteus\n",
      "Coverage Percentage: 0.7539491671054741%\n",
      "Adj, read counts 103\n",
      "Number of samples per group [2, 4, 2, 1, 1, 5, 1, 1, 1, 1, 2, 1, 2, 1, 2, 4, 2, 2, 1, 1, 3, 1, 3, 1, 1, 1, 1, 1, 1, 4, 2, 1, 2, 1, 1, 3, 2, 1, 1, 1, 1, 2, 1, 3, 1, 1, 1]\n",
      "Number of bins 72\n",
      "total # samples 23\n",
      "Number of total reads 103\n",
      "Number of hits in BIOM tables 43.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.62135922330097 %\n",
      "Percentage of reads from samples who only fall into one bin group 15.53398058252427 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 12\n",
      "Length of fail 51\n",
      "G000023405 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Dermabacteraceae; g__Brachybacterium; s__Brachybacterium faecium\n",
      "Coverage Percentage: 0.0115352952371678%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 12\n",
      "Length of fail 52\n",
      "G000023705 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Methylophilaceae; g__Methylotenera; s__Methylotenera mobilis\n",
      "Coverage Percentage: 0.0228845527306413%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 12\n",
      "Length of fail 53\n",
      "G000023865 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Saccharomonospora; s__Saccharomonospora viridis\n",
      "Coverage Percentage: 0.0028549219202065%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 12\n",
      "Length of fail 54\n",
      "G000024025 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Catenulisporales; f__Catenulisporaceae; g__Catenulispora; s__Catenulispora acidiphila\n",
      "Coverage Percentage: 0.0026462148332855%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 12\n",
      "Length of fail 55\n",
      "G000024065 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Jonesiaceae; g__Jonesia; s__Jonesia denitrificans\n",
      "Coverage Percentage: 0.0196752600152892%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 12\n",
      "Length of fail 56\n",
      "G000024105 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Anaerococcus; s__Anaerococcus prevotii\n",
      "Coverage Percentage: 0.1127759546054267%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 12\n",
      "Length of fail 57\n",
      "G000024385 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptosporangiales; f__Thermomonosporaceae; g__Thermomonospora; s__Thermomonospora curvata\n",
      "Coverage Percentage: 0.0050008724926476%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 12\n",
      "Length of fail 58\n",
      "G000024405 d__Bacteria; p__Fusobacteria; c__Fusobacteriia; o__Fusobacteriales; f__Leptotrichiaceae; g__Sebaldella; s__Sebaldella termitidis\n",
      "Coverage Percentage: 0.0093610211536789%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 12\n",
      "Length of fail 59\n",
      "G000024865 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptosporangiales; f__Streptosporangiaceae; g__Streptosporangium; s__Streptosporangium roseum\n",
      "Coverage Percentage: 0.0014561883084858%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 12\n",
      "Length of fail 60\n",
      "G000024925 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Sanguibacteraceae; g__Sanguibacter; s__Sanguibacter keddieii\n",
      "Coverage Percentage: 0.0113320761468496%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 12\n",
      "Length of fail 61\n",
      "G000024945 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Veillonellales; f__Veillonellaceae; g__Veillonella; s__Veillonella parvula\n",
      "Coverage Percentage: 0.0503249783551001%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 12\n",
      "Length of fail 62\n",
      "G000024985 d__Bacteria; p__Chloroflexi; c__Thermomicrobia; o__Sphaerobacterales; f__Sphaerobacteraceae; g__Sphaerobacter; s__Sphaerobacter thermophilus\n",
      "Coverage Percentage: 0.0275177626030851%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 12\n",
      "Length of fail 63\n",
      "G000025225 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Oscillospiraceae; g__Mageeibacillus; s__Mageeibacillus indolicus\n",
      "Coverage Percentage: 0.0083437123220606%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 12\n",
      "Length of fail 64\n",
      "G000025265 d__Bacteria; p__Actinobacteria; c__Thermoleophilia; o__Solirubrobacterales; f__Conexibacteraceae; g__Conexibacter; s__Conexibacter woesei\n",
      "Coverage Percentage: 0.0071233482441418%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 12\n",
      "Length of fail 65\n",
      "G000025345 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Geodermatophilus; s__Geodermatophilus obscurus\n",
      "Coverage Percentage: 0.0041521864643606%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 12\n",
      "Length of fail 66\n",
      "G000025565 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Enterobacter; s__Enterobacter cloacae\n",
      "Coverage Percentage: 0.0315601314273181%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 12\n",
      "Length of fail 67\n",
      "G000026225 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Escherichia; s__Escherichia fergusonii\n",
      "Coverage Percentage: 0.295464074122485%\n",
      "Adj, read counts 84\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 5, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 3, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 3, 2, 2, 1, 2, 1, 2, 1, 1, 1]\n",
      "Number of bins 76\n",
      "total # samples 36\n",
      "Number of total reads 83\n",
      "Number of hits in BIOM tables 7.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.285714285714285 %\n",
      "Percentage of reads from samples who only fall into one bin group 30.120481927710845 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 67\n",
      "G000067205 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Bordetella; s__Bordetella petrii\n",
      "Coverage Percentage: 0.0401289724751557%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 68\n",
      "G000069225 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Clavibacter; s__Clavibacter michiganensis\n",
      "Coverage Percentage: 0.0629585648620111%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 69\n",
      "G000069965 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Morganellaceae; g__Proteus; s__Proteus mirabilis\n",
      "Coverage Percentage: 0.1145389599540478%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 70\n",
      "G000083545 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylorubrum; s__Methylorubrum extorquens\n",
      "Coverage Percentage: 0.0690247426023748%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 71\n",
      "G000091305 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces scabiei\n",
      "Coverage Percentage: 0.0029757520548208%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 72\n",
      "G000091545 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Thermales; f__Thermaceae; g__Thermus; s__Thermus thermophilus\n",
      "Coverage Percentage: 0.1745194924993951%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 73\n",
      "G000092025 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Agrobacterium; s__Agrobacterium fabrum\n",
      "Coverage Percentage: 0.0026611127539908%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 74\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000092125 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Thermales; f__Thermaceae; g__Meiothermus; s__Meiothermus silvanus\n",
      "Coverage Percentage: 0.0105596649281284%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 75\n",
      "G000092285 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Caulobacter; s__Caulobacter segnis\n",
      "Coverage Percentage: 0.039371753119132%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 76\n",
      "G000092645 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptosporangiales; f__Streptosporangiaceae; g__Thermobispora; s__Thermobispora bispora\n",
      "Coverage Percentage: 0.0090215313882466%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 77\n",
      "G000092785 d__Bacteria; p__Chlamydiae; c__Chlamydiia; o__Parachlamydiales; f__Waddliaceae; g__Waddlia; s__Waddlia chondrophila\n",
      "Coverage Percentage: 0.014165601510372%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 78\n",
      "G000092985 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptosporangiales; f__Nocardiopsaceae; g__Nocardiopsis; s__Nocardiopsis dassonvillei\n",
      "Coverage Percentage: 0.0117218566931954%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 79\n",
      "G000093025 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Methylophilaceae; g__Methylotenera; s__Methylotenera versatilis\n",
      "Coverage Percentage: 0.0447404482084375%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 80\n",
      "G000142905 d__Eukaryota; p__; c__; o__; f__Apusomonadidae; g__Thecamonas; s__Thecamonas trahens\n",
      "Coverage Percentage: 0.0012899524868662%\n",
      "Adj, read counts 43\n",
      "Number of samples per group [17, 20, 6]\n",
      "Number of bins 3\n",
      "total # samples 32\n",
      "Number of total reads 43\n",
      "Number of hits in BIOM tables 10.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 100.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 48.837209302325576 %\n",
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 81\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000142945 d__Eukaryota; p__Oomycota; c__; o__Peronosporales; f__Peronosporaceae; g__Phytophthora; s__Phytophthora infestans\n",
      "Coverage Percentage: 0.001489665684862%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 82\n",
      "G000143105 d__Eukaryota; p__Ascomycota; c__Sordariomycetes; o__Ophiostomatales; f__Ophiostomataceae; g__Grosmannia; s__Grosmannia clavigera\n",
      "Coverage Percentage: 0.0012315532947473%\n",
      "Adj, read counts 107\n",
      "Number of samples per group [17, 2, 88]\n",
      "Number of bins 3\n",
      "total # samples 89\n",
      "Number of total reads 107\n",
      "Number of hits in BIOM tables 941.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 100.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 67.28971962616822 %\n",
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 83\n",
      "G000143535 d__Eukaryota; p__Ascomycota; c__Leotiomycetes; o__Helotiales; f__Sclerotiniaceae; g__Botrytis; s__Botrytis cinerea\n",
      "Coverage Percentage: 0.0010907707517493%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 84\n",
      "G000143745 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactiplantibacillus; s__Lactiplantibacillus plantarum\n",
      "Coverage Percentage: 0.0125139264125094%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 85\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000143825 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium genitalium\n",
      "Coverage Percentage: 0.0742142502424516%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 86\n",
      "G000144405 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella melaninogenica\n",
      "Coverage Percentage: 0.0418520709200069%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 87\n",
      "G000144605 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas subvibrioides\n",
      "Coverage Percentage: 0.046411551164599%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 88\n",
      "G000145235 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora aurantiaca\n",
      "Coverage Percentage: 0.0171516601027761%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 89\n",
      "G000145255 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Gallionellaceae; g__Gallionella; s__Gallionella capsiferriformans\n",
      "Coverage Percentage: 0.0206167898456618%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 90\n",
      "G000145635 d__Eukaryota; p__Ascomycota; c__Sordariomycetes; o__Magnaporthales; f__Magnaporthaceae; g__Gaeumannomyces; s__Gaeumannomyces tritici\n",
      "Coverage Percentage: 0.0006875445076569%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 91\n",
      "G000145945 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas amygdali\n",
      "Coverage Percentage: 0.0280405176611725%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 92\n",
      "G000146045 d__Eukaryota; p__Ascomycota; c__Saccharomycetes; o__Saccharomycetales; f__Saccharomycetaceae; g__Saccharomyces; s__Saccharomyces cerevisiae\n",
      "Coverage Percentage: 0.0011762359216692%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 93\n",
      "G000146065 d__Bacteria; p__Candidatus Cloacimonetes; c__; o__; f__; g__Candidatus Cloacimonas; s__Candidatus Cloacimonas acidaminovorans\n",
      "Coverage Percentage: 0.0787334989006685%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 94\n",
      "G000146185 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Lachnospira; s__Lachnospira eligens\n",
      "Coverage Percentage: 0.0415337979514937%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 95\n",
      "G000146345 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Peptoniphilus; s__Peptoniphilus duerdenii\n",
      "Coverage Percentage: 0.0122815206875392%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 96\n",
      "G000147675 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Peptostreptococcaceae; g__Peptostreptococcus; s__Peptostreptococcus stomatis\n",
      "Coverage Percentage: 0.0263382229942739%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 97\n",
      "G000148565 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus oralis\n",
      "Coverage Percentage: 0.0559624696052179%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 98\n",
      "G000148585 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus mitis\n",
      "Coverage Percentage: 0.0851310649195268%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 99\n",
      "G000149405 d__Eukaryota; p__Bacillariophyta; c__Coscinodiscophyceae; o__Thalassiosirales; f__Thalassiosiraceae; g__Thalassiosira; s__Thalassiosira pseudonana\n",
      "Coverage Percentage: 0.0314594098855916%\n",
      "Adj, read counts 119\n",
      "Number of samples per group [2, 2, 4, 39]\n",
      "Number of bins 10\n",
      "total # samples 40\n",
      "Number of total reads 118\n",
      "Number of hits in BIOM tables 173.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 97.47899159663865 %\n",
      "Percentage of reads from samples who only fall into one bin group 79.66101694915254 %\n",
      "Status: Fail -- over 75% of reads are part of a sample which only hit a single bin\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 100\n",
      "G000149645 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Eurotiales; f__Aspergillaceae; g__Aspergillus; s__Aspergillus fischeri\n",
      "Coverage Percentage: 0.0042590976092981%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 101\n",
      "G000149685 d__Eukaryota; p__Ascomycota; c__Saccharomycetes; o__Saccharomycetales; f__Debaryomycetaceae; g__Lodderomyces; s__Lodderomyces elongisporus\n",
      "Coverage Percentage: 0.0054348483246388%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 102\n",
      "G000149755 d__Eukaryota; p__Oomycota; c__; o__Peronosporales; f__Peronosporaceae; g__Phytophthora; s__Phytophthora sojae\n",
      "Coverage Percentage: 0.0006101749505967%\n",
      "Adj, read counts 94\n",
      "Number of samples per group [57, 1, 1]\n",
      "Number of bins 3\n",
      "total # samples 58\n",
      "Number of total reads 94\n",
      "Number of hits in BIOM tables 30.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 99.99999999999999 %\n",
      "Percentage of reads from samples who only fall into one bin group 97.87234042553192 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 103\n",
      "G000149925 d__Eukaryota; p__Basidiomycota; c__Pucciniomycetes; o__Pucciniales; f__Pucciniaceae; g__Puccinia; s__Puccinia graminis\n",
      "Coverage Percentage: 0.0008012873250004%\n",
      "Adj, read counts 35\n",
      "Number of samples per group [25, 2, 3, 1, 2, 2]\n",
      "Number of bins 6\n",
      "total # samples 30\n",
      "Number of total reads 35\n",
      "Number of hits in BIOM tables 10.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 85.71428571428572 %\n",
      "Percentage of reads from samples who only fall into one bin group 71.42857142857143 %\n",
      "Status: Fail - too many reads in top 3 bins\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 104\n",
      "G000150675 d__Eukaryota; p__Ascomycota; c__Sordariomycetes; o__Glomerellales; f__Plectosphaerellaceae; g__Verticillium; s__Verticillium dahliae\n",
      "Coverage Percentage: 0.0003067719555213%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 105\n",
      "G000150735 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Onygenales; f__; g__Paracoccidioides; s__Paracoccidioides brasiliensis\n",
      "Coverage Percentage: 0.0021633416417225%\n",
      "Adj, read counts 63\n",
      "Number of samples per group [3, 47, 4, 1, 8]\n",
      "Number of bins 5\n",
      "total # samples 53\n",
      "Number of total reads 63\n",
      "Number of hits in BIOM tables 26.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 93.65079365079364 %\n",
      "Percentage of reads from samples who only fall into one bin group 71.42857142857143 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:51: RuntimeWarning: More than 20 figures have been opened. Figures created through the pyplot interface (`matplotlib.pyplot.figure`) are retained until explicitly closed and may consume too much memory. (To control this warning, see the rcParam `figure.max_open_warning`). Consider using `matplotlib.pyplot.close()`.\n",
      "  plt.figure(figsize=(25, 8))\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Fail - too many reads in top 3 bins\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 106\n",
      "G000151545 d__Eukaryota; p__Oomycota; c__; o__Saprolegniales; f__Saprolegniaceae; g__Saprolegnia; s__Saprolegnia parasitica\n",
      "Coverage Percentage: 0.003735859039906%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 107\n",
      "G000153725 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Roseobacteraceae; g__Salipiger; s__Salipiger bermudensis\n",
      "Coverage Percentage: 0.0148639004735109%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 108\n",
      "G000153885 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Eubacteriaceae; g__Eubacterium; s__Eubacterium ventriosum\n",
      "Coverage Percentage: 0.0482705659253686%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 109\n",
      "G000153905 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Blautia; s__Blautia obeum\n",
      "Coverage Percentage: 0.1090287015161441%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 110\n",
      "G000153925 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Mediterraneibacter; s__[Ruminococcus] torques\n",
      "Coverage Percentage: 0.0276389550287219%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 111\n",
      "G000154065 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Dorea; s__Dorea longicatena\n",
      "Coverage Percentage: 0.0850784240636486%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 112\n",
      "G000154105 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Tannerellaceae; g__Parabacteroides; s__Parabacteroides merdae\n",
      "Coverage Percentage: 0.0897386218933834%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 13\n",
      "Length of fail 113\n",
      "G000154125 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides ovatus\n",
      "Coverage Percentage: 0.1286578124398294%\n",
      "Adj, read counts 50\n",
      "Number of samples per group [1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 3, 1, 1, 1, 1]\n",
      "Number of bins 45\n",
      "total # samples 32\n",
      "Number of total reads 50\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 42.0 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 14\n",
      "Length of fail 113\n",
      "G000154205 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides uniformis\n",
      "Coverage Percentage: 0.2692946662070365%\n",
      "Adj, read counts 76\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 2, 3, 3, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 68\n",
      "total # samples 37\n",
      "Number of total reads 76\n",
      "Number of hits in BIOM tables 23.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.526315789473683 %\n",
      "Percentage of reads from samples who only fall into one bin group 32.89473684210527 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 15\n",
      "Length of fail 113\n",
      "G000154365 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Enterocloster; s__Enterocloster bolteae\n",
      "Coverage Percentage: 0.0227462325408944%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 15\n",
      "Length of fail 114\n",
      "G000154425 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Coprococcus; s__Coprococcus eutactus\n",
      "Coverage Percentage: 0.0145645442924869%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 15\n",
      "Length of fail 115\n",
      "G000154465 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Rikenellaceae; g__Alistipes; s__Alistipes putredinis\n",
      "Coverage Percentage: 0.0896939038722834%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 15\n",
      "Length of fail 116\n",
      "G000154485 d__Bacteria; p__Firmicutes; c__Erysipelotrichia; o__Erysipelotrichales; f__Erysipelotrichaceae; g__Erysipelatoclostridium; s__Erysipelatoclostridium ramosum\n",
      "Coverage Percentage: 0.0363168248895427%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 15\n",
      "Length of fail 117\n",
      "G000154525 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides stercoris\n",
      "Coverage Percentage: 0.0614441009548522%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 15\n",
      "Length of fail 118\n",
      "G000154565 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Oscillospiraceae; g__Anaerotruncus; s__Anaerotruncus colihominis\n",
      "Coverage Percentage: 0.0408577108714842%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 15\n",
      "Length of fail 119\n",
      "G000154845 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola coprocola\n",
      "Coverage Percentage: 0.1012809000308468%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 15\n",
      "Length of fail 120\n",
      "G000154985 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus infantarius\n",
      "Coverage Percentage: 0.0482969608747877%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 15\n",
      "Length of fail 121\n",
      "G000155205 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Mediterraneibacter; s__[Ruminococcus] lactaris\n",
      "Coverage Percentage: 0.0528154515391696%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 15\n",
      "Length of fail 122\n",
      "G000155815 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides eggerthii\n",
      "Coverage Percentage: 0.0516910968636367%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 15\n",
      "Length of fail 123\n",
      "G000156075 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola dorei\n",
      "Coverage Percentage: 0.3852885137175393%\n",
      "Adj, read counts 135\n",
      "Number of samples per group [1, 1, 2, 1, 3, 2, 1, 1, 3, 1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 3, 1, 2, 2, 1, 1, 1, 2, 1, 2, 1, 1, 1, 2, 1, 1, 2, 3, 1, 1, 1, 4, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 3, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 3, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1]\n",
      "Number of bins 117\n",
      "total # samples 44\n",
      "Number of total reads 132\n",
      "Number of hits in BIOM tables 45.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.111111111111112 %\n",
      "Percentage of reads from samples who only fall into one bin group 13.636363636363635 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 16\n",
      "Length of fail 123\n",
      "G000156175 d__Bacteria; p__Actinobacteria; c__Coriobacteriia; o__Coriobacteriales; f__Coriobacteriaceae; g__Collinsella; s__Collinsella intestinalis\n",
      "Coverage Percentage: 0.0370813086441448%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 16\n",
      "Length of fail 124\n",
      "G000156195 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides finegoldii\n",
      "Coverage Percentage: 0.0582255493056363%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 16\n",
      "Length of fail 125\n",
      "G000156495 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Tannerellaceae; g__Parabacteroides; s__Parabacteroides johnsonii\n",
      "Coverage Percentage: 0.0275411514706954%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 16\n",
      "Length of fail 126\n",
      "G000156515 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Lachnoclostridium; s__[Clostridium] hylemonae\n",
      "Coverage Percentage: 0.0052154266668619%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 16\n",
      "Length of fail 127\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000156535 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Roseburia; s__Roseburia intestinalis\n",
      "Coverage Percentage: 0.0881407623836071%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 16\n",
      "Length of fail 128\n",
      "G000156615 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium pseudogenitalium\n",
      "Coverage Percentage: 0.84025667419797%\n",
      "Adj, read counts 127\n",
      "Number of samples per group [1, 2, 1, 1, 3, 1, 6, 1, 1, 1, 2, 2, 2, 2, 1, 1, 4, 1, 2, 1, 2, 1, 5, 1, 2, 2, 2, 3, 2, 2, 1, 2, 1, 3, 1, 1, 2, 3, 3, 3, 2, 2, 2, 1, 2, 6, 4, 1, 2, 2, 4, 1, 4, 1, 2]\n",
      "Number of bins 99\n",
      "total # samples 37\n",
      "Number of total reads 124\n",
      "Number of hits in BIOM tables 106.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.53543307086614 %\n",
      "Percentage of reads from samples who only fall into one bin group 10.483870967741936 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 17\n",
      "Length of fail 128\n",
      "G000156655 d__Bacteria; p__Firmicutes; c__Erysipelotrichia; o__Erysipelotrichales; f__Erysipelotrichaceae; g__Holdemanella; s__Holdemanella biformis\n",
      "Coverage Percentage: 0.0520919565969182%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 17\n",
      "Length of fail 129\n",
      "G000156675 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Blautia; s__Blautia hansenii\n",
      "Coverage Percentage: 0.0386418343298371%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 17\n",
      "Length of fail 130\n",
      "G000157915 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola coprophilus\n",
      "Coverage Percentage: 0.0476138180494214%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 17\n",
      "Length of fail 131\n",
      "G000157935 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella copri\n",
      "Coverage Percentage: 0.2419873879623443%\n",
      "Adj, read counts 49\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 3, 1, 2, 1, 1, 1, 1, 2, 1, 1, 2, 2, 1, 2, 1, 2, 1]\n",
      "Number of bins 44\n",
      "total # samples 21\n",
      "Number of total reads 47\n",
      "Number of hits in BIOM tables 25.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.285714285714285 %\n",
      "Percentage of reads from samples who only fall into one bin group 17.02127659574468 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 18\n",
      "Length of fail 131\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000157955 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Oscillospiraceae; g__Subdoligranulum; s__Subdoligranulum variabile\n",
      "Coverage Percentage: 0.0652253416935974%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 18\n",
      "Length of fail 132\n",
      "G000158015 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium breve\n",
      "Coverage Percentage: 0.0698115131999043%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 18\n",
      "Length of fail 133\n",
      "G000158035 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides cellulosilyticus\n",
      "Coverage Percentage: 0.1017835552469029%\n",
      "Adj, read counts 40\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 4]\n",
      "Number of bins 30\n",
      "total # samples 18\n",
      "Number of total reads 39\n",
      "Number of hits in BIOM tables 26.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 30.76923076923077 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 19\n",
      "Length of fail 133\n",
      "G000159035 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces urogenitalis\n",
      "Coverage Percentage: 0.0111680938652403%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 19\n",
      "Length of fail 134\n",
      "G000159115 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium accolens\n",
      "Coverage Percentage: 0.2377430652119875%\n",
      "Adj, read counts 33\n",
      "Number of samples per group [1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 1, 1, 1, 1, 1, 1, 2]\n",
      "Number of bins 30\n",
      "total # samples 17\n",
      "Number of total reads 32\n",
      "Number of hits in BIOM tables 39.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 24.242424242424242 %\n",
      "Percentage of reads from samples who only fall into one bin group 25.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 20\n",
      "Length of fail 134\n",
      "G000159135 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium striatum\n",
      "Coverage Percentage: 0.0323951458859047%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 20\n",
      "Length of fail 135\n",
      "G000159375 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Ligilactobacillus; s__Ligilactobacillus ruminis\n",
      "Coverage Percentage: 0.0063541439450698%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 20\n",
      "Length of fail 136\n",
      "G000159595 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium glucuronolyticum\n",
      "Coverage Percentage: 0.2818401321023522%\n",
      "Adj, read counts 46\n",
      "Number of samples per group [1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 37\n",
      "total # samples 22\n",
      "Number of total reads 46\n",
      "Number of hits in BIOM tables 18.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.391304347826086 %\n",
      "Percentage of reads from samples who only fall into one bin group 28.26086956521739 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 136\n",
      "G000159635 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium lipophiloflavum\n",
      "Coverage Percentage: 0.0147865542050781%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 137\n",
      "G000159995 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Alloprevotella; s__Alloprevotella tannerae\n",
      "Coverage Percentage: 0.0144248301988654%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 138\n",
      "G000160055 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Veillonellales; f__Veillonellaceae; g__Dialister; s__Dialister invisus\n",
      "Coverage Percentage: 0.0769531002763771%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 139\n",
      "G000160095 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Hungatella; s__Hungatella hathewayi\n",
      "Coverage Percentage: 0.0143073195522408%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 140\n",
      "G000160595 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Leuconostoc; s__Leuconostoc mesenteroides\n",
      "Coverage Percentage: 0.007259595052874%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 141\n",
      "G000160755 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Morganellaceae; g__Proteus; s__Proteus mirabilis\n",
      "Coverage Percentage: 0.0744508659040355%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 142\n",
      "G000160875 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactobacillus; s__Lactobacillus iners\n",
      "Coverage Percentage: 0.040457981217167%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 143\n",
      "G000162255 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactobacillus; s__Lactobacillus crispatus\n",
      "Coverage Percentage: 0.1100690164825967%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 144\n",
      "G000162295 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter baumannii\n",
      "Coverage Percentage: 0.0133687954690105%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 145\n",
      "G000162755 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Legionellales; f__Legionellaceae; g__Legionella; s__Legionella drancourtii\n",
      "Coverage Percentage: 0.0125187992641834%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 146\n",
      "G000164095 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus oralis\n",
      "Coverage Percentage: 0.0589858611570006%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 147\n",
      "G000164695 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Rothia; s__Rothia dentocariosa\n",
      "Coverage Percentage: 0.0526730579303877%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 148\n",
      "G000165735 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Veillonellales; f__Veillonellaceae; g__Megasphaera; s__Megasphaera micronuciformis\n",
      "Coverage Percentage: 0.0106439113430124%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 149\n",
      "G000166395 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Spirosomaceae; g__Leadbetterella; s__Leadbetterella byssophila\n",
      "Coverage Percentage: 0.0148042209518892%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 150\n",
      "G000166795 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Loigolactobacillus; s__Loigolactobacillus coryniformis\n",
      "Coverage Percentage: 0.045316120400616%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 151\n",
      "G000167675 d__Eukaryota; p__Ascomycota; c__Sordariomycetes; o__Hypocreales; f__Hypocreaceae; g__Trichoderma; s__Trichoderma reesei\n",
      "Coverage Percentage: 0.0010869163921454%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 152\n",
      "G000168035 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Yersinia; s__Yersinia intermedia\n",
      "Coverage Percentage: 0.0591035982550649%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 153\n",
      "G000169015 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides caccae\n",
      "Coverage Percentage: 0.0702267188350863%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 154\n",
      "G000169035 d__Bacteria; p__Actinobacteria; c__Coriobacteriia; o__Coriobacteriales; f__Coriobacteriaceae; g__Collinsella; s__Collinsella aerofaciens\n",
      "Coverage Percentage: 0.1435040643776771%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 155\n",
      "G000169235 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Dorea; s__Dorea formicigenerans\n",
      "Coverage Percentage: 0.0969139740974807%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 156\n",
      "G000169475 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Mediterraneibacter; s__[Ruminococcus] gnavus\n",
      "Coverage Percentage: 0.1103703917292436%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 157\n",
      "G000172175 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides intestinalis\n",
      "Coverage Percentage: 0.0703153086471969%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 158\n",
      "G000172555 d__Bacteria; p__Verrucomicrobia; c__Verrucomicrobiae; o__Verrucomicrobiales; f__Verrucomicrobia subdivision 3; g__Pedosphaera; s__Pedosphaera parvula\n",
      "Coverage Percentage: 0.0040721432768779%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 159\n",
      "G000172815 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__Bacillus pumilus\n",
      "Coverage Percentage: 0.0326265720609091%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 160\n",
      "G000173355 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Anaerococcus; s__Anaerococcus hydrogenalis\n",
      "Coverage Percentage: 0.0535869393869823%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 161\n",
      "G000173435 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium pseudocatenulatum\n",
      "Coverage Percentage: 0.1993920645688499%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 162\n",
      "G000173455 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium catenulatum\n",
      "Coverage Percentage: 0.1763452578861201%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 163\n",
      "G000173795 d__Bacteria; p__Firmicutes; c__Erysipelotrichia; o__Erysipelotrichales; f__Coprobacillaceae; g__Catenibacterium; s__Catenibacterium mitsuokai\n",
      "Coverage Percentage: 0.0091390868299044%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 164\n",
      "G000173915 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__; g__Gemella; s__Gemella haemolysans\n",
      "Coverage Percentage: 0.0096531543377623%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 165\n",
      "G000173975 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Anaerobutyricum; s__Anaerobutyricum hallii\n",
      "Coverage Percentage: 0.0936112450052755%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 166\n",
      "G000174195 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Roseburia; s__Roseburia inulinivorans\n",
      "Coverage Percentage: 0.0691259820405713%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 167\n",
      "G000174695 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium aurimucosum\n",
      "Coverage Percentage: 0.0550812952339125%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 168\n",
      "G000175375 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium matruchotii\n",
      "Coverage Percentage: 0.0393889909695924%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 169\n",
      "G000175575 d__Bacteria; p__Proteobacteria; c__Acidithiobacillia; o__Acidithiobacillales; f__Acidithiobacillaceae; g__Acidithiobacillus; s__Acidithiobacillus caldus\n",
      "Coverage Percentage: 0.0151651850202788%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 170\n",
      "G000175615 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Rothia; s__Rothia mucilaginosa\n",
      "Coverage Percentage: 0.0853417081996313%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 171\n",
      "G000177075 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella buccalis\n",
      "Coverage Percentage: 0.0320126364283383%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 172\n",
      "G000177435 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Veillonellales; f__Veillonellaceae; g__Veillonella; s__Veillonella parvula\n",
      "Coverage Percentage: 0.0390972870826907%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 173\n",
      "G000178435 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Aerococcaceae; g__Aerococcus; s__Aerococcus viridans\n",
      "Coverage Percentage: 0.0643657287589361%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 174\n",
      "G000178875 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Alteromonadales; f__Shewanellaceae; g__Shewanella; s__Shewanella baltica\n",
      "Coverage Percentage: 0.006835808535522%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 175\n",
      "G000178955 d__Bacteria; p__Acidobacteria; c__Acidobacteriia; o__Acidobacteriales; f__Acidobacteriaceae; g__Granulicella; s__Granulicella mallensis\n",
      "Coverage Percentage: 0.002308588735658%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 176\n",
      "G000179955 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactobacillus; s__Lactobacillus iners\n",
      "Coverage Percentage: 0.0390819890237818%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 177\n",
      "G000181695 d__Eukaryota; p__Basidiomycota; c__Malasseziomycetes; o__Malasseziales; f__Malasseziaceae; g__Malassezia; s__Malassezia globosa\n",
      "Coverage Percentage: 0.0244769817035398%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 178\n",
      "G000182645 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Brucellaceae; g__Brucella; s__Brucella intermedia\n",
      "Coverage Percentage: 0.0093536025760075%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 179\n",
      "G000182765 d__Eukaryota; p__Ascomycota; c__Saccharomycetes; o__Saccharomycetales; f__Debaryomycetaceae; g__Candida; s__Candida parapsilosis\n",
      "Coverage Percentage: 0.0011587487294051%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 180\n",
      "G000182925 d__Eukaryota; p__Ascomycota; c__Sordariomycetes; o__Sordariales; f__Sordariaceae; g__Neurospora; s__Neurospora crassa\n",
      "Coverage Percentage: 0.0034352909187792%\n",
      "Adj, read counts 50\n",
      "Number of samples per group [2, 1, 2, 2, 1, 2, 36, 3, 1]\n",
      "Number of bins 9\n",
      "total # samples 40\n",
      "Number of total reads 50\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 82.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 64.0 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Fail - too many reads in top 3 bins\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 181\n",
      "G000183135 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Paludibacteraceae; g__Paludibacter; s__Paludibacter propionicigenes\n",
      "Coverage Percentage: 0.0326685305456187%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 182\n",
      "G000184455 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Eurotiales; f__Aspergillaceae; g__Aspergillus; s__Aspergillus oryzae\n",
      "Coverage Percentage: 0.001893847774837%\n",
      "Adj, read counts 58\n",
      "Number of samples per group [3, 43, 7, 1, 3, 1]\n",
      "Number of bins 6\n",
      "total # samples 48\n",
      "Number of total reads 58\n",
      "Number of hits in BIOM tables 15.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 91.37931034482759 %\n",
      "Percentage of reads from samples who only fall into one bin group 65.51724137931035 %\n",
      "Status: Fail - too many reads in top 3 bins\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 183\n",
      "G000185305 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pasteurellales; f__Pasteurellaceae; g__Aggregatibacter; s__Aggregatibacter segnis\n",
      "Coverage Percentage: 0.0139655780805033%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 184\n",
      "G000185365 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Enterococcaceae; g__Enterococcus; s__Enterococcus italicus\n",
      "Coverage Percentage: 0.013654034891438%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 185\n",
      "G000185505 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Eubacteriaceae; g__Pseudoramibacter; s__Pseudoramibacter alactolyticus\n",
      "Coverage Percentage: 0.1349154842700137%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 186\n",
      "G000185705 d__Bacteria; p__Proteobacteria; c__Deltaproteobacteria; o__Desulfovibrionales; f__Desulfovibrionaceae; g__Bilophila; s__Bilophila wadsworthia\n",
      "Coverage Percentage: 0.0133219349188181%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 187\n",
      "G000186425 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Lautropia; s__Lautropia mirabilis\n",
      "Coverage Percentage: 0.0258877910195379%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 188\n",
      "G000186465 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus australis\n",
      "Coverage Percentage: 0.0296236760305964%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 189\n",
      "G000187265 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus equinus\n",
      "Coverage Percentage: 0.0557368805944961%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 190\n",
      "G000187405 d__Eukaryota; p__Ascomycota; c__Sordariomycetes; o__Hypocreales; f__Clavicipitaceae; g__Metarhizium; s__Metarhizium acridum\n",
      "Coverage Percentage: 0.001557303348159%\n",
      "Adj, read counts 117\n",
      "Number of samples per group [12, 69, 11, 22]\n",
      "Number of bins 4\n",
      "total # samples 74\n",
      "Number of total reads 117\n",
      "Number of hits in BIOM tables 79.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 90.5982905982906 %\n",
      "Percentage of reads from samples who only fall into one bin group 38.46153846153847 %\n",
      "Status: Fail - too many reads in top 3 bins\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 191\n",
      "G000187465 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus infantis\n",
      "Coverage Percentage: 0.0454840192134569%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 192\n",
      "G000187585 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus peroris\n",
      "Coverage Percentage: 0.0326227468962657%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 193\n",
      "G000187895 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola plebeius\n",
      "Coverage Percentage: 0.0769059608112007%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 194\n",
      "G000188175 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Acidaminococcales; f__Acidaminococcaceae; g__Phascolarctobacterium; s__Phascolarctobacterium succinatutens\n",
      "Coverage Percentage: 0.0529697427412618%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 195\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000188295 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus vestibularis\n",
      "Coverage Percentage: 0.0634751835521865%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 196\n",
      "G000189535 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Pseudarthrobacter; s__Pseudarthrobacter phenanthrenivorans\n",
      "Coverage Percentage: 0.0066587878360262%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 197\n",
      "G000190375 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Rubrivivax; s__Rubrivivax benzoatilyticus\n",
      "Coverage Percentage: 0.0614052214761495%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 198\n",
      "G000190435 d__Bacteria; p__Spirochaetes; c__Spirochaetia; o__Spirochaetales; f__Sphaerochaetaceae; g__Sphaerochaeta; s__Sphaerochaeta globosa\n",
      "Coverage Percentage: 0.0033770887444647%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 199\n",
      "G000190535 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Odoribacteraceae; g__Odoribacter; s__Odoribacter splanchnicus\n",
      "Coverage Percentage: 0.0224029025419098%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 200\n",
      "G000190575 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola salanitronis\n",
      "Coverage Percentage: 0.0112794440487355%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 201\n",
      "G000190755 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Methylococcales; f__Methylococcaceae; g__Methylobacter; s__Methylobacter tundripaludum\n",
      "Coverage Percentage: 0.017552294390134%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 202\n",
      "G000191145 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter pittii\n",
      "Coverage Percentage: 0.0182263956525904%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 203\n",
      "G000191405 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pasteurellales; f__Pasteurellaceae; g__Haemophilus; s__Haemophilus parainfluenzae\n",
      "Coverage Percentage: 0.0415071129167594%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 204\n",
      "G000191585 d__Archaea; p__Euryarchaeota; c__Methanobacteria; o__Methanobacteriales; f__Methanobacteriaceae; g__Methanobacterium; s__Methanobacterium lacus\n",
      "Coverage Percentage: 0.0274794068937704%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 205\n",
      "G000191785 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Parascardovia; s__Parascardovia denticolens\n",
      "Coverage Percentage: 0.0072431280162076%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 206\n",
      "G000192065 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Xanthomonas; s__Xanthomonas hortorum\n",
      "Coverage Percentage: 0.0196420241105845%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 207\n",
      "G000193415 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae; g__Paenibacillus; s__Paenibacillus vortex\n",
      "Coverage Percentage: 0.007092481734511%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 208\n",
      "G000194605 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Crocinitomicaceae; g__Fluviicola; s__Fluviicola taffensis\n",
      "Coverage Percentage: 0.0157761487507383%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 209\n",
      "G000194925 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Neisseriaceae; g__Neisseria; s__Neisseria bacilliformis\n",
      "Coverage Percentage: 0.0115346555691138%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 210\n",
      "G000195615 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides clarus\n",
      "Coverage Percentage: 0.0330612473616377%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 211\n",
      "G000195635 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides fluxus\n",
      "Coverage Percentage: 0.0600096569386396%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 212\n",
      "G000196015 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Cupriavidus; s__Cupriavidus metallidurans\n",
      "Coverage Percentage: 0.0997626063657134%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 213\n",
      "G000196175 d__Bacteria; p__Proteobacteria; c__Oligoflexia; o__Bdellovibrionales; f__Bdellovibrionaceae; g__Bdellovibrio; s__Bdellovibrio bacteriovorus\n",
      "Coverage Percentage: 0.0079831877238662%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 21\n",
      "Length of fail 214\n",
      "G000196555 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium longum\n",
      "Coverage Percentage: 0.3889460011974019%\n",
      "Adj, read counts 53\n",
      "Number of samples per group [4, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 2, 1, 1, 2, 1, 2, 1, 1, 1, 2, 1, 1, 2, 1, 3, 1, 1, 1, 1, 2, 1, 1]\n",
      "Number of bins 42\n",
      "total # samples 23\n",
      "Number of total reads 53\n",
      "Number of hits in BIOM tables 7.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.754716981132077 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.867924528301888 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 22\n",
      "Length of fail 214\n",
      "G000196575 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium longum\n",
      "Coverage Percentage: 0.4375078615666841%\n",
      "Adj, read counts 64\n",
      "Number of samples per group [4, 2, 1, 1, 1, 3, 2, 1, 1, 1, 1, 2, 1, 2, 1, 1, 2, 1, 1, 2, 1, 1, 1, 2, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1]\n",
      "Number of bins 48\n",
      "total # samples 24\n",
      "Number of total reads 62\n",
      "Number of hits in BIOM tables 16.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.0625 %\n",
      "Percentage of reads from samples who only fall into one bin group 14.516129032258066 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 23\n",
      "Length of fail 214\n",
      "G000196615 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Erwiniaceae; g__Erwinia; s__Erwinia billingiae\n",
      "Coverage Percentage: 0.0234536069041462%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 23\n",
      "Length of fail 215\n",
      "G000196675 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Pseudonocardia; s__Pseudonocardia dioxanivorans\n",
      "Coverage Percentage: 0.004121013210654%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 23\n",
      "Length of fail 216\n",
      "G000196855 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Leuconostoc; s__Leuconostoc gelidum\n",
      "Coverage Percentage: 0.0077274215999344%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 23\n",
      "Length of fail 217\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000197735 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Glutamicibacter; s__Glutamicibacter arilaitensis\n",
      "Coverage Percentage: 0.0339694025264456%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 23\n",
      "Length of fail 218\n",
      "G000203815 d__Eukaryota; p__Evosea; c__Eumycetozoa; o__Acytosteliales; f__Cavenderiaceae; g__Cavenderia; s__Cavenderia fasciculata\n",
      "Coverage Percentage: 0.0034816597768488%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 23\n",
      "Length of fail 219\n",
      "G000203915 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia ambifaria\n",
      "Coverage Percentage: 0.0202560174650703%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 23\n",
      "Length of fail 220\n",
      "G000204015 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Asticcacaulis; s__Asticcacaulis biprosthecium\n",
      "Coverage Percentage: 0.0469729517473956%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 23\n",
      "Length of fail 221\n",
      "G000204035 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas diminuta\n",
      "Coverage Percentage: 0.0457343856100213%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 23\n",
      "Length of fail 222\n",
      "G000204195 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax radicis\n",
      "Coverage Percentage: 0.069103777166897%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 23\n",
      "Length of fail 223\n",
      "G000204235 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Cutibacterium; s__Cutibacterium modestum\n",
      "Coverage Percentage: 1.6084856939219656%\n",
      "Adj, read counts 282\n",
      "Number of samples per group [21, 1, 1, 1, 2, 1, 4, 1, 3, 2, 7, 1, 14, 5, 1, 3, 14, 1, 1, 2, 2, 4, 3, 1, 6, 2, 6, 3, 1, 25, 4, 1, 6, 2, 2, 1, 2, 1, 3, 2, 3, 1, 1, 1, 1, 2, 8, 1, 1, 19, 5, 1]\n",
      "Number of bins 89\n",
      "total # samples 50\n",
      "Number of total reads 276\n",
      "Number of hits in BIOM tables 87.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 40.780141843971634 %\n",
      "Percentage of reads from samples who only fall into one bin group 5.072463768115942 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 24\n",
      "Length of fail 223\n",
      "G000204415 d__Archaea; p__Euryarchaeota; c__Methanomicrobia; o__Methanotrichales; f__Methanotrichaceae; g__Methanothrix; s__Methanothrix soehngenii\n",
      "Coverage Percentage: 0.1464317986959243%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 24\n",
      "Length of fail 224\n",
      "G000204645 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Alicycliphilus; s__Alicycliphilus denitrificans\n",
      "Coverage Percentage: 0.1369614206596985%\n",
      "Adj, read counts 40\n",
      "Number of samples per group [1, 2, 1, 2, 1, 1, 2, 5, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1]\n",
      "Number of bins 32\n",
      "total # samples 19\n",
      "Number of total reads 40\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 27.499999999999996 %\n",
      "Percentage of reads from samples who only fall into one bin group 27.500000000000004 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 25\n",
      "Length of fail 224\n",
      "G000205025 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Sutterellaceae; g__Parasutterella; s__Parasutterella excrementihominis\n",
      "Coverage Percentage: 0.015034218728825%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 25\n",
      "Length of fail 225\n",
      "G000205165 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Paraprevotella; s__Paraprevotella xylaniphila\n",
      "Coverage Percentage: 0.0154554939797422%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 25\n",
      "Length of fail 226\n",
      "G000208865 d__Eukaryota; p__Apicomplexa; c__Conoidasida; o__Eucoccidiorida; f__Sarcocystidae; g__Neospora; s__Neospora caninum\n",
      "Coverage Percentage: 0.0034371498555632%\n",
      "Adj, read counts 34\n",
      "Number of samples per group [2, 2, 1, 1, 1, 1, 11, 3, 2, 4, 4, 1]\n",
      "Number of bins 12\n",
      "total # samples 22\n",
      "Number of total reads 34\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 55.88235294117647 %\n",
      "Percentage of reads from samples who only fall into one bin group 50.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 26\n",
      "Length of fail 226\n",
      "G000209065 d__Eukaryota; p__Euglenozoa; c__Kinetoplastea; o__Trypanosomatida; f__Trypanosomatidae; g__Trypanosoma; s__Trypanosoma cruzi\n",
      "Coverage Percentage: 0.0010593485265047%\n",
      "Adj, read counts 126\n",
      "Number of samples per group [54, 3, 13, 3, 2, 1, 13, 37]\n",
      "Number of bins 8\n",
      "total # samples 63\n",
      "Number of total reads 126\n",
      "Number of hits in BIOM tables 54.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 82.53968253968253 %\n",
      "Percentage of reads from samples who only fall into one bin group 19.841269841269842 %\n",
      "Status: Fail - too many reads in top 3 bins\n",
      "----------------------------\n",
      "Length of pass 26\n",
      "Length of fail 227\n",
      "G000210055 d__Bacteria; p__Actinobacteria; c__Coriobacteriia; o__Eggerthellales; f__Eggerthellaceae; g__Gordonibacter; s__Gordonibacter pamelaeae\n",
      "Coverage Percentage: 0.0191517679215924%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 26\n",
      "Length of fail 228\n",
      "G000210075 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides xylanisolvens\n",
      "Coverage Percentage: 0.0884014695091902%\n",
      "Adj, read counts 34\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 33\n",
      "total # samples 23\n",
      "Number of total reads 33\n",
      "Number of hits in BIOM tables 12.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.705882352941178 %\n",
      "Percentage of reads from samples who only fall into one bin group 48.484848484848484 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 27\n",
      "Length of fail 228\n",
      "G000210375 d__Bacteria; p__Cyanobacteria; c__; o__Oscillatoriales; f__Microcoleaceae; g__Arthrospira; s__Arthrospira platensis\n",
      "Coverage Percentage: 0.0072328894656868%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 27\n",
      "Length of fail 229\n",
      "G000210575 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Rikenellaceae; g__Alistipes; s__Alistipes shahii\n",
      "Coverage Percentage: 0.0475112779497448%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 27\n",
      "Length of fail 230\n",
      "G000210695 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Anaerostipes; s__Anaerostipes hadrus\n",
      "Coverage Percentage: 0.1033136123549981%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 27\n",
      "Length of fail 231\n",
      "G000210715 d__Bacteria; p__Synergistetes; c__Synergistia; o__Synergistales; f__Synergistaceae; g__Fretibacterium; s__Fretibacterium fastidiosum\n",
      "Coverage Percentage: 0.0055345150317061%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 27\n",
      "Length of fail 232\n",
      "G000211835 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Hylemonella; s__Hylemonella gracilis\n",
      "Coverage Percentage: 0.0530794357783014%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 27\n",
      "Length of fail 233\n",
      "G000213575 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Dysgonomonadaceae; g__Dysgonomonas; s__Dysgonomonas mossii\n",
      "Coverage Percentage: 0.0033123425215017%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 27\n",
      "Length of fail 234\n",
      "G000214035 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Sterolibacteriaceae; g__Methyloversatilis; s__Methyloversatilis universalis\n",
      "Coverage Percentage: 0.0125653896262788%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 27\n",
      "Length of fail 235\n",
      "G000214075 d__Bacteria; p__Cyanobacteria; c__; o__Oscillatoriales; f__Microcoleaceae; g__Microcoleus; s__Microcoleus vaginatus\n",
      "Coverage Percentage: 0.0175977792468299%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 27\n",
      "Length of fail 236\n",
      "G000214235 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia plymuthica\n",
      "Coverage Percentage: 0.0260340114057263%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 27\n",
      "Length of fail 237\n",
      "G000214665 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Methylococcales; f__Methylococcaceae; g__Methylomonas; s__Methylomonas methanica\n",
      "Coverage Percentage: 0.0059782080459949%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 27\n",
      "Length of fail 238\n",
      "G000214725 d__Archaea; p__Euryarchaeota; c__Methanobacteria; o__Methanobacteriales; f__Methanobacteriaceae; g__Methanobacterium; s__Methanobacterium paludis\n",
      "Coverage Percentage: 0.0262709298613295%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 27\n",
      "Length of fail 239\n",
      "G000215705 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Ramlibacter; s__Ramlibacter tataouinensis\n",
      "Coverage Percentage: 0.0720113272269889%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 27\n",
      "Length of fail 240\n",
      "G000215745 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Klebsiella; s__Klebsiella aerogenes\n",
      "Coverage Percentage: 0.0059276373725226%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 27\n",
      "Length of fail 241\n",
      "G000219605 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas stutzeri\n",
      "Coverage Percentage: 0.134302858663172%\n",
      "Adj, read counts 38\n",
      "Number of samples per group [2, 1, 1, 1, 2, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 34\n",
      "total # samples 15\n",
      "Number of total reads 38\n",
      "Number of hits in BIOM tables 17.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 23.684210526315788 %\n",
      "Percentage of reads from samples who only fall into one bin group 21.052631578947366 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 28\n",
      "Length of fail 241\n",
      "G000219805 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Weissella; s__Weissella koreensis\n",
      "Coverage Percentage: 0.0209505442285413%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 28\n",
      "Length of fail 242\n",
      "G000220395 d__Eukaryota; p__Ciliophora; c__Oligohymenophorea; o__Hymenostomatida; f__; g__Ichthyophthirius; s__Ichthyophthirius multifiliis\n",
      "Coverage Percentage: 0.0002334138522398%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 28\n",
      "Length of fail 243\n",
      "G000221985 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus pseudopneumoniae\n",
      "Coverage Percentage: 0.0690054739787873%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 28\n",
      "Length of fail 244\n",
      "G000222745 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus australis\n",
      "Coverage Percentage: 0.0295922090419909%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 28\n",
      "Length of fail 245\n",
      "G000224615 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella oulorum\n",
      "Coverage Percentage: 0.005380955622083%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 28\n",
      "Length of fail 246\n",
      "G000224715 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Ornithinimicrobiaceae; g__Serinicoccus; s__Serinicoccus profundi\n",
      "Coverage Percentage: 0.0111469417215767%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 28\n",
      "Length of fail 247\n",
      "G000225345 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Roseburia; s__Roseburia hominis\n",
      "Coverage Percentage: 0.0786164178585099%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 28\n",
      "Length of fail 248\n",
      "G000225545 d__Eukaryota; p__Ascomycota; c__Orbiliomycetes; o__Orbiliales; f__Orbiliaceae; g__Orbilia; s__Orbilia oligospora\n",
      "Coverage Percentage: 0.0019612193085085%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 28\n",
      "Length of fail 249\n",
      "G000226135 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides faecis\n",
      "Coverage Percentage: 0.0352870960093532%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 28\n",
      "Length of fail 250\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000226395 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Eurotiales; f__Aspergillaceae; g__Penicillium; s__Penicillium rubens\n",
      "Coverage Percentage: 0.005542333294388%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 28\n",
      "Length of fail 251\n",
      "G000226955 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas elodea\n",
      "Coverage Percentage: 0.0146356545869207%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 28\n",
      "Length of fail 252\n",
      "G000227295 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Cutibacterium; s__Cutibacterium avidum\n",
      "Coverage Percentage: 0.4869855290358955%\n",
      "Adj, read counts 85\n",
      "Number of samples per group [2, 2, 1, 4, 2, 15, 2, 7, 1, 1, 1, 16, 2, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 2]\n",
      "Number of bins 36\n",
      "total # samples 39\n",
      "Number of total reads 83\n",
      "Number of hits in BIOM tables 15.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 61.1764705882353 %\n",
      "Percentage of reads from samples who only fall into one bin group 30.120481927710845 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 29\n",
      "Length of fail 252\n",
      "G000230695 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Rhodanobacteraceae; g__Rhodanobacter; s__Rhodanobacter denitrificans\n",
      "Coverage Percentage: 0.0104839912057536%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 29\n",
      "Length of fail 253\n",
      "Length of pass 30\n",
      "Length of fail 253\n",
      "G000231275 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Rikenellaceae; g__Alistipes; s__Alistipes indistinctus\n",
      "Coverage Percentage: 0.0113113892382112%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 30\n",
      "Length of fail 254\n",
      "G000233915 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Pseudoxanthomonas; s__Pseudoxanthomonas spadix\n",
      "Coverage Percentage: 0.017494295527311%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 30\n",
      "Length of fail 255\n",
      "G000233955 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Paraprevotella; s__Paraprevotella clara\n",
      "Coverage Percentage: 0.0388476182114613%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 30\n",
      "Length of fail 256\n",
      "G000234665 d__Eukaryota; p__Euglenozoa; c__Kinetoplastea; o__Trypanosomatida; f__Trypanosomatidae; g__Leishmania; s__Leishmania mexicana\n",
      "Coverage Percentage: 0.0011145536706003%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 30\n",
      "Length of fail 257\n",
      "G000235565 d__Archaea; p__Euryarchaeota; c__Methanomicrobia; o__Methanotrichales; f__Methanotrichaceae; g__Methanothrix; s__Methanothrix harundinacea\n",
      "Coverage Percentage: 0.0074288599150387%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 30\n",
      "Length of fail 258\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000235885 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella stercorea\n",
      "Coverage Percentage: 0.0443188349887848%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 30\n",
      "Length of fail 259\n",
      "G000238055 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Oribacterium; s__Oribacterium parvum\n",
      "Coverage Percentage: 0.0061111756404491%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 30\n",
      "Length of fail 260\n",
      "G000238575 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Carnobacterium; s__Carnobacterium maltaromaticum\n",
      "Coverage Percentage: 0.0594839345405825%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 30\n",
      "Length of fail 261\n",
      "G000239295 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Oscillospiraceae; g__Flavonifractor; s__Flavonifractor plautii\n",
      "Coverage Percentage: 0.0064837550562831%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 30\n",
      "Length of fail 262\n",
      "G000240225 d__Bacteria; p__Actinobacteria; c__Thermoleophilia; o__Solirubrobacterales; f__Patulibacteraceae; g__Patulibacter; s__Patulibacter medicamentivorans\n",
      "Coverage Percentage: 0.0029218321652541%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 30\n",
      "Length of fail 263\n",
      "G000241145 d__Archaea; p__Thaumarchaeota; c__; o__Nitrosopumilales; f__Nitrosopumilaceae; g__Nitrosarchaeum; s__Candidatus Nitrosarchaeum limnium\n",
      "Coverage Percentage: 0.0081078222228426%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 30\n",
      "Length of fail 264\n",
      "G000241325 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Gordoniaceae; g__Gordonia; s__Gordonia polyisoprenivorans\n",
      "Coverage Percentage: 0.0069818652808538%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 30\n",
      "Length of fail 265\n",
      "G000241465 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas echinoides\n",
      "Coverage Percentage: 0.0952864172776032%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 30\n",
      "Length of fail 266\n",
      "G000241525 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Comamonas; s__Comamonas testosteroni\n",
      "Coverage Percentage: 0.0355730823567675%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 30\n",
      "Length of fail 267\n",
      "G000242115 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas extremaustralis\n",
      "Coverage Percentage: 0.0331920659579621%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 30\n",
      "Length of fail 268\n",
      "G000242435 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Veillonellales; f__Veillonellaceae; g__Dialister; s__Dialister succinatiphilus\n",
      "Coverage Percentage: 0.0060396641946707%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 30\n",
      "Length of fail 269\n",
      "G000245735 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Alteromonadales; f__Alteromonadaceae; g__Alishewanella; s__Alishewanella jeotgali\n",
      "Coverage Percentage: 0.0118195383491836%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 30\n",
      "Length of fail 270\n",
      "G000245775 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Selenomonadales; f__Selenomonadaceae; g__Megamonas; s__Megamonas funiformis\n",
      "Coverage Percentage: 0.2214026171865244%\n",
      "Adj, read counts 35\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 34\n",
      "total # samples 20\n",
      "Number of total reads 35\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.142857142857142 %\n",
      "Percentage of reads from samples who only fall into one bin group 48.57142857142857 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 31\n",
      "Length of fail 270\n",
      "G000245815 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Dolosigranulum; s__Dolosigranulum pigrum\n",
      "Coverage Percentage: 0.0442498073468642%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 31\n",
      "Length of fail 271\n",
      "G000246925 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Peptoniphilus; s__Peptoniphilus rhinitidis\n",
      "Coverage Percentage: 0.0421445478473148%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 31\n",
      "Length of fail 272\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000247715 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Gordoniaceae; g__Gordonia; s__Gordonia polyisoprenivorans\n",
      "Coverage Percentage: 0.0075115680714895%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 31\n",
      "Length of fail 273\n",
      "G000248155 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter parvus\n",
      "Coverage Percentage: 0.4534917686664664%\n",
      "Adj, read counts 87\n",
      "Number of samples per group [11, 2, 3, 1, 1, 2, 3, 1, 1, 2, 1, 1, 1, 1, 10, 1, 3, 2, 1, 1, 1, 1, 1, 6, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 53\n",
      "total # samples 34\n",
      "Number of total reads 87\n",
      "Number of hits in BIOM tables 23.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 41.379310344827594 %\n",
      "Percentage of reads from samples who only fall into one bin group 22.988505747126435 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 32\n",
      "Length of fail 273\n",
      "G000252445 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Deinococcales; f__Deinococcaceae; g__Deinococcus; s__Deinococcus gobiensis\n",
      "Coverage Percentage: 0.0034270234644438%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 32\n",
      "Length of fail 274\n",
      "G000252835 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Thermales; f__Thermaceae; g__Thermus; s__Thermus parvatiensis\n",
      "Coverage Percentage: 0.2505253029083125%\n",
      "Adj, read counts 31\n",
      "Number of samples per group [2, 1, 1, 1, 3, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 3, 1, 1]\n",
      "Number of bins 29\n",
      "total # samples 9\n",
      "Number of total reads 31\n",
      "Number of hits in BIOM tables 11.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 25.806451612903224 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.129032258064516 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 33\n",
      "Length of fail 274\n",
      "G000253035 d__Bacteria; p__Chlamydiae; c__Chlamydiia; o__Parachlamydiales; f__Parachlamydiaceae; g__Parachlamydia; s__Parachlamydia acanthamoebae\n",
      "Coverage Percentage: 0.0302696636454504%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 33\n",
      "Length of fail 275\n",
      "G000253275 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium branchiophilum\n",
      "Coverage Percentage: 0.0918808120086032%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 33\n",
      "Length of fail 276\n",
      "G000255655 d__Bacteria; p__Planctomycetes; c__Planctomycetia; o__Planctomycetales; f__Planctomycetaceae; g__Schlesneria; s__Schlesneria paludicola\n",
      "Coverage Percentage: 0.0052052057573022%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 33\n",
      "Length of fail 277\n",
      "G000256565 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Herbaspirillum; s__Herbaspirillum lusitanum\n",
      "Coverage Percentage: 0.0194056545842252%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 33\n",
      "Length of fail 278\n",
      "G000257395 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Legionellales; f__Coxiellaceae; g__Rickettsiella; s__Rickettsiella massiliensis\n",
      "Coverage Percentage: 0.0120943964751489%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 33\n",
      "Length of fail 279\n",
      "G000257825 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus pseudopneumoniae\n",
      "Coverage Percentage: 0.0797306662700062%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 33\n",
      "Length of fail 280\n",
      "G000258495 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter agri\n",
      "Coverage Percentage: 0.0155949219982008%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 33\n",
      "Length of fail 281\n",
      "G000260755 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus infantis\n",
      "Coverage Percentage: 0.046656298600311%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 33\n",
      "Length of fail 282\n",
      "G000262525 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Leptothrix; s__Leptothrix ochracea\n",
      "Coverage Percentage: 0.2107559844533247%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 33\n",
      "Length of fail 283\n",
      "G000263675 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas lactis\n",
      "Coverage Percentage: 0.0964472615531601%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 35\n",
      "total # samples 20\n",
      "Number of total reads 36\n",
      "Number of hits in BIOM tables 6.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 8.108108108108109 %\n",
      "Percentage of reads from samples who only fall into one bin group 33.33333333333333 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 283\n",
      "G000264945 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium indicum\n",
      "Coverage Percentage: 0.019919960281728%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 284\n",
      "G000265365 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Rikenellaceae; g__Alistipes; s__Alistipes finegoldii\n",
      "Coverage Percentage: 0.0423111643362944%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 285\n",
      "G000269805 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces massiliensis\n",
      "Coverage Percentage: 0.0223950945547588%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 286\n",
      "G000269965 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium longum\n",
      "Coverage Percentage: 0.2625701760153385%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 287\n",
      "G000270245 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Microlunatus; s__Microlunatus phosphovorus\n",
      "Coverage Percentage: 0.011296605757081%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 288\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000271325 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Planococcaceae; g__Solibacillus; s__Solibacillus silvestris\n",
      "Coverage Percentage: 0.0093618646826541%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 289\n",
      "G000271485 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Selenomonadales; f__Sporomusaceae; g__Pelosinus; s__Pelosinus fermentans\n",
      "Coverage Percentage: 0.0929555303368618%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 290\n",
      "G000271645 d__Eukaryota; p__Basidiomycota; c__Tremellomycetes; o__Tremellales; f__Tremellaceae; g__Tremella; s__Tremella mesenterica\n",
      "Coverage Percentage: 0.0008589145854485%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 291\n",
      "G000272005 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Alteromonadales; f__Alteromonadaceae; g__Alishewanella; s__Alishewanella agri\n",
      "Coverage Percentage: 0.0325574429836705%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 292\n",
      "G000273175 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides nordii\n",
      "Coverage Percentage: 0.0278221856311614%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 293\n",
      "G000280055 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Alteromonadales; f__Alteromonadaceae; g__Alishewanella; s__Alishewanella aestuarii\n",
      "Coverage Percentage: 0.0078575739489658%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 294\n",
      "G000280495 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Kosakonia; s__Kosakonia radicincitans\n",
      "Coverage Percentage: 0.0114169179123602%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 295\n",
      "G000282995 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Melaminivora; s__Melaminivora alkalimesophila\n",
      "Coverage Percentage: 0.1263359756651981%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 296\n",
      "G000284275 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium cosmicum\n",
      "Coverage Percentage: 0.0177686428670099%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 297\n",
      "G000284375 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium japonicum\n",
      "Coverage Percentage: 0.0304320966737131%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 298\n",
      "G000284415 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Rhodospirillaceae; g__Pararhodospirillum; s__Pararhodospirillum photometricum\n",
      "Coverage Percentage: 0.0038954783815138%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 299\n",
      "G000285835 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Brevibacteriaceae; g__Brevibacterium; s__Brevibacterium senegalense\n",
      "Coverage Percentage: 0.0204805101105743%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 300\n",
      "G000285995 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces naeslundii\n",
      "Coverage Percentage: 0.0604645899035676%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 301\n",
      "G000286355 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Kocuria; s__Kocuria atrinae\n",
      "Coverage Percentage: 0.0188657439943527%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 302\n",
      "G000286555 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia multivorans\n",
      "Coverage Percentage: 0.0185051450423283%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 34\n",
      "Length of fail 303\n",
      "G000287215 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas media\n",
      "Coverage Percentage: 0.6021990414434734%\n",
      "Adj, read counts 162\n",
      "Number of samples per group [1, 1, 1, 2, 1, 3, 2, 1, 2, 2, 3, 3, 1, 1, 1, 2, 2, 1, 1, 1, 3, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 3, 1, 1, 1, 1, 6, 1, 1, 5, 3, 2, 3, 1, 2, 4, 4, 1, 2, 1, 1, 1, 2, 2, 1, 2, 1, 2, 1, 2, 2, 5, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2]\n",
      "Number of bins 117\n",
      "total # samples 29\n",
      "Number of total reads 161\n",
      "Number of hits in BIOM tables 64.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.345679012345679 %\n",
      "Percentage of reads from samples who only fall into one bin group 8.074534161490684 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 35\n",
      "Length of fail 303\n",
      "G000296405 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium otitidis\n",
      "Coverage Percentage: 0.7130903966937078%\n",
      "Adj, read counts 99\n",
      "Number of samples per group [1, 1, 2, 3, 1, 1, 1, 2, 1, 1, 1, 1, 1, 3, 2, 1, 2, 1, 1, 1, 1, 1, 4, 1, 2, 1, 1, 2, 1, 2, 3, 4, 1, 2, 1, 1, 2, 1, 2, 2, 1, 1, 2, 1, 1, 1, 2, 1, 2]\n",
      "Number of bins 80\n",
      "total # samples 16\n",
      "Number of total reads 98\n",
      "Number of hits in BIOM tables 85.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.111111111111112 %\n",
      "Percentage of reads from samples who only fall into one bin group 12.244897959183673 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 36\n",
      "Length of fail 303\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000296465 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Barnesiellaceae; g__Barnesiella; s__Barnesiella intestinihominis\n",
      "Coverage Percentage: 0.044163942594424%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 36\n",
      "Length of fail 304\n",
      "G000296815 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Spirosomaceae; g__Fibrisoma; s__Fibrisoma limi\n",
      "Coverage Percentage: 0.0308499940345315%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 36\n",
      "Length of fail 305\n",
      "G000297055 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Sulfuricellaceae; g__Sulfuricella; s__Sulfuricella denitrificans\n",
      "Coverage Percentage: 0.0235917013160066%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 36\n",
      "Length of fail 306\n",
      "G000297075 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas oleovorans\n",
      "Coverage Percentage: 0.5750756453863782%\n",
      "Adj, read counts 167\n",
      "Number of samples per group [5, 1, 2, 1, 2, 3, 2, 2, 1, 2, 1, 1, 2, 5, 1, 1, 1, 1, 1, 1, 1, 1, 2, 4, 1, 1, 1, 2, 2, 1, 1, 2, 1, 1, 1, 2, 2, 1, 4, 1, 1, 1, 1, 2, 7, 2, 5, 1, 1, 1, 3, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 3, 1, 1, 1, 2, 1, 2, 3, 1, 2, 1, 2, 2, 1, 2, 2, 1, 2, 2, 1, 1, 1, 1, 4, 1, 3, 2, 1, 5, 3]\n",
      "Number of bins 139\n",
      "total # samples 66\n",
      "Number of total reads 164\n",
      "Number of hits in BIOM tables 35.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.179640718562874 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.29268292682927 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 37\n",
      "Length of fail 306\n",
      "G000297615 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Methylococcales; f__Methylococcaceae; g__Methylococcus; s__Methylococcus capsulatus\n",
      "Coverage Percentage: 0.0399757815433901%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 37\n",
      "Length of fail 307\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000297795 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium otitidis\n",
      "Coverage Percentage: 0.7008187043343962%\n",
      "Adj, read counts 100\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 2, 1, 4, 1, 1, 1, 1, 1, 2, 1, 2, 3, 2, 2, 1, 4, 1, 2, 1, 1, 2, 1, 2, 2, 2, 1, 1, 2, 1, 1, 1, 2, 1, 1]\n",
      "Number of bins 79\n",
      "total # samples 16\n",
      "Number of total reads 98\n",
      "Number of hits in BIOM tables 85.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 12.244897959183673 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 38\n",
      "Length of fail 307\n",
      "G000298775 d__Eukaryota; p__Ascomycota; c__Leotiomycetes; o__Helotiales; f__Drepanopezizaceae; g__Drepanopeziza; s__Drepanopeziza brunnea\n",
      "Coverage Percentage: 0.001093324547213%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 38\n",
      "Length of fail 308\n",
      "G000298875 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Leuconostoc; s__Leuconostoc gelidum\n",
      "Coverage Percentage: 0.010087145543779%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 38\n",
      "Length of fail 309\n",
      "G000300095 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia phenoliruptrix\n",
      "Coverage Percentage: 0.0019735541134814%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 38\n",
      "Length of fail 310\n",
      "G000300375 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacterium; s__Mycolicibacterium hassiacum\n",
      "Coverage Percentage: 0.0442887852641458%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 38\n",
      "Length of fail 311\n",
      "G000300975 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Herbaspirillum; s__Herbaspirillum frisingense\n",
      "Coverage Percentage: 0.0151566457825339%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 38\n",
      "Length of fail 312\n",
      "G000301075 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Bergeyella; s__Bergeyella zoohelcum\n",
      "Coverage Percentage: 0.0129452390668454%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 38\n",
      "Length of fail 313\n",
      "G000306115 d__Bacteria; p__Spirochaetes; c__Spirochaetia; o__Leptospirales; f__Leptospiraceae; g__Leptospira; s__Leptospira wolffii\n",
      "Coverage Percentage: 0.0070661342451903%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 38\n",
      "Length of fail 314\n",
      "G000308295 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Afipia; s__Afipia birgiae\n",
      "Coverage Percentage: 0.0160848078052935%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 38\n",
      "Length of fail 315\n",
      "G000308495 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Nocardia; s__Nocardia brevicatena\n",
      "Coverage Percentage: 0.0009269677743218%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 38\n",
      "Length of fail 316\n",
      "G000310185 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Zoogloeaceae; g__Thauera; s__Thauera aminoaromatica\n",
      "Coverage Percentage: 0.0803355244435528%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 38\n",
      "Length of fail 317\n",
      "G000310225 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Zoogloeaceae; g__Thauera; s__Thauera phenylacetica\n",
      "Coverage Percentage: 0.1173353497512804%\n",
      "Adj, read counts 30\n",
      "Number of samples per group [3, 2, 2, 4, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 25\n",
      "total # samples 13\n",
      "Number of total reads 30\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 33.33333333333333 %\n",
      "Percentage of reads from samples who only fall into one bin group 26.666666666666668 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 39\n",
      "Length of fail 317\n",
      "G000311825 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Peptoniphilus; s__Peptoniphilus grossensis\n",
      "Coverage Percentage: 0.0246907002168404%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 39\n",
      "Length of fail 318\n",
      "G000311925 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Rikenellaceae; g__Alistipes; s__Alistipes communis\n",
      "Coverage Percentage: 0.0306599636189503%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 39\n",
      "Length of fail 319\n",
      "G000312145 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Rikenellaceae; g__Alistipes; s__Alistipes senegalensis\n",
      "Coverage Percentage: 0.0389774371414156%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 39\n",
      "Length of fail 320\n",
      "G000312425 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Reyranellaceae; g__Reyranella; s__Reyranella massiliensis\n",
      "Coverage Percentage: 0.018248493377109%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 39\n",
      "Length of fail 321\n",
      "G000312465 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Citrobacter; s__Citrobacter freundii\n",
      "Coverage Percentage: 0.0072332513764657%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 39\n",
      "Length of fail 322\n",
      "G000312505 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__; g__Fenollaria; s__Fenollaria massiliensis\n",
      "Coverage Percentage: 0.0103521061272186%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 39\n",
      "Length of fail 323\n",
      "G000312645 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactobacillus; s__Lactobacillus equicursoris\n",
      "Coverage Percentage: 0.0103270096221796%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 39\n",
      "Length of fail 324\n",
      "G000313135 d__Eukaryota; p__Discosea; c__; o__Longamoebia; f__Acanthamoebidae; g__Acanthamoeba; s__Acanthamoeba castellanii\n",
      "Coverage Percentage: 0.0054702299095515%\n",
      "Adj, read counts 167\n",
      "Number of samples per group [4, 3, 6, 45, 1, 22, 1, 17, 2, 5, 48, 1, 1, 1, 8, 1, 1]\n",
      "Number of bins 17\n",
      "total # samples 74\n",
      "Number of total reads 167\n",
      "Number of hits in BIOM tables 88.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 68.8622754491018 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.16766467065868 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 40\n",
      "Length of fail 324\n",
      "G000314675 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Afipia; s__Afipia broomeae\n",
      "Coverage Percentage: 0.1642332575957406%\n",
      "Adj, read counts 52\n",
      "Number of samples per group [1, 1, 1, 3, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 45\n",
      "total # samples 21\n",
      "Number of total reads 50\n",
      "Number of hits in BIOM tables 90.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 13.461538461538463 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 324\n",
      "G000314735 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Afipia; s__Afipia felis\n",
      "Coverage Percentage: 0.0164387211578949%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 325\n",
      "G000315425 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia timonae\n",
      "Coverage Percentage: 0.0477613594144141%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 326\n",
      "G000315445 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Alloiococcus; s__Alloiococcus otitis\n",
      "Coverage Percentage: 0.19759389307569%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 327\n",
      "G000315465 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinobaculum; s__Actinobaculum massiliense\n",
      "Coverage Percentage: 0.0161906580395229%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 328\n",
      "G000315525 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium yanoikuyae\n",
      "Coverage Percentage: 0.0594099956639921%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 329\n",
      "G000315625 d__Eukaryota; p__; c__Cryptophyceae; o__Pyrenomonadales; f__Geminigeraceae; g__Guillardia; s__Guillardia theta\n",
      "Coverage Percentage: 0.0014800586364811%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 330\n",
      "G000316515 d__Bacteria; p__Cyanobacteria; c__; o__Synechococcales; f__Prochlorococcaceae; g__Cyanobium; s__Cyanobium gracile\n",
      "Coverage Percentage: 0.0225888024164932%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 331\n",
      "G000317125 d__Bacteria; p__Cyanobacteria; c__; o__Chroococcidiopsidales; f__Chroococcidiopsidaceae; g__Chroococcidiopsis; s__Chroococcidiopsis thermalis\n",
      "Coverage Percentage: 0.0022572887629923%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 332\n",
      "G000317475 d__Bacteria; p__Cyanobacteria; c__; o__Oscillatoriales; f__Oscillatoriaceae; g__Oscillatoria; s__Oscillatoria nigro-viridis\n",
      "Coverage Percentage: 0.0264011912449588%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 333\n",
      "G000318355 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Veillonellales; f__Veillonellaceae; g__Veillonella; s__Veillonella atypica\n",
      "Coverage Percentage: 0.0299808803067159%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 334\n",
      "G000320405 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Clostridium; s__Clostridium celatum\n",
      "Coverage Percentage: 0.0073112961493934%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 335\n",
      "G000320565 d__Eukaryota; p__Ascomycota; c__Dothideomycetes; o__Mycosphaerellales; f__Mycosphaerellaceae; g__Sphaerulina; s__Sphaerulina musiva\n",
      "Coverage Percentage: 0.0021189960741679%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 336\n",
      "G000321005 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Anaerococcus; s__Anaerococcus senegalensis\n",
      "Coverage Percentage: 0.0340713985084298%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 337\n",
      "G000321105 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Promicromonosporaceae; g__Cellulosimicrobium; s__Cellulosimicrobium cellulans\n",
      "Coverage Percentage: 0.0089623825000145%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 338\n",
      "G000321205 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Rikenellaceae; g__Alistipes; s__Alistipes ihumii\n",
      "Coverage Percentage: 0.0083273131545592%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 339\n",
      "G000328665 d__Archaea; p__Euryarchaeota; c__Methanomicrobia; o__Methanosarcinales; f__Methanosarcinaceae; g__Methanomethylovorans; s__Methanomethylovorans hollandica\n",
      "Coverage Percentage: 0.0055636759022458%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 340\n",
      "G000331775 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Xanthomonas; s__Xanthomonas translucens\n",
      "Coverage Percentage: 0.0117553011156138%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 341\n",
      "G000332155 d__Bacteria; p__Cyanobacteria; c__; o__Oscillatoriales; f__Microcoleaceae; g__Kamptonema; s__Kamptonema formosum\n",
      "Coverage Percentage: 0.0025528409057943%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 342\n",
      "G000332645 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__Bacillus inaquosorum\n",
      "Coverage Percentage: 0.0095097570567923%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 343\n",
      "G000332875 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Anaerostipes; s__Anaerostipes hadrus\n",
      "Coverage Percentage: 0.0987140494103836%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 344\n",
      "G000336555 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Afipia; s__Afipia clevelandensis\n",
      "Coverage Percentage: 0.0310031848726278%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 345\n",
      "G000336955 d__Archaea; p__Euryarchaeota; c__Halobacteria; o__Haloferacales; f__Haloferacaceae; g__Haloferax; s__Haloferax larsenii\n",
      "Coverage Percentage: 0.010814762583517%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 346\n",
      "G000337315 d__Archaea; p__Euryarchaeota; c__Halobacteria; o__Haloferacales; f__Haloferacaceae; g__Haloferax; s__Haloferax volcanii\n",
      "Coverage Percentage: 0.008787541540518%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 347\n",
      "G000337615 d__Archaea; p__Euryarchaeota; c__Halobacteria; o__Natrialbales; f__Natrialbaceae; g__Natrinema; s__Natrinema pallidum\n",
      "Coverage Percentage: 0.0104897942724191%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 348\n",
      "G000337735 d__Archaea; p__Euryarchaeota; c__Halobacteria; o__Natrialbales; f__Natrialbaceae; g__Natronorubrum; s__Natronorubrum sulfidifaciens\n",
      "Coverage Percentage: 0.0135200408605679%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 349\n",
      "G000337895 d__Archaea; p__Euryarchaeota; c__Halobacteria; o__Natrialbales; f__Natrialbaceae; g__Halobiforma; s__Halobiforma nitratireducens\n",
      "Coverage Percentage: 0.0104797760631624%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 350\n",
      "G000338955 d__Eukaryota; p__Ascomycota; c__Dothideomycetes; o__Mycosphaerellales; f__Teratosphaeriaceae; g__Baudoinia; s__Baudoinia panamericana\n",
      "Coverage Percentage: 0.0024821745128675%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 351\n",
      "G000344575 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Lactococcus; s__Lactococcus lactis\n",
      "Coverage Percentage: 0.0328312831332689%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 352\n",
      "G000344785 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium callunae\n",
      "Coverage Percentage: 0.018335636384868%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 353\n",
      "G000344805 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium oligotrophicum\n",
      "Coverage Percentage: 0.0063285280485082%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 354\n",
      "G000348725 d__Bacteria; p__Proteobacteria; c__Oligoflexia; o__Bdellovibrionales; f__Bdellovibrionaceae; g__Bdellovibrio; s__Bdellovibrio exovorus\n",
      "Coverage Percentage: 0.0082772331316572%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 355\n",
      "G000348945 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Paeniglutamicibacter; s__Paeniglutamicibacter gangotriensis\n",
      "Coverage Percentage: 0.0135407890229336%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 356\n",
      "G000349305 d__Eukaryota; p__Basidiomycota; c__Malasseziomycetes; o__Malasseziales; f__Malasseziaceae; g__Malassezia; s__Malassezia sympodialis\n",
      "Coverage Percentage: 0.0032720768393145%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 357\n",
      "G000350525 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Leucobacter; s__Leucobacter salsicius\n",
      "Coverage Percentage: 0.0094790954557467%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 358\n",
      "G000354255 d__Eukaryota; p__Ascomycota; c__Dothideomycetes; o__Pleosporales; f__Pleosporaceae; g__Bipolaris; s__Bipolaris maydis\n",
      "Coverage Percentage: 0.0027236880424356%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 359\n",
      "G000364225 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Eubacteriaceae; g__Eubacterium; s__Eubacterium plexicaudatum\n",
      "Coverage Percentage: 0.0040095374760407%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 360\n",
      "G000364445 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium mesophilicum\n",
      "Coverage Percentage: 0.0164297572957199%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 361\n",
      "G000364625 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas thermotolerans\n",
      "Coverage Percentage: 0.0150523127811985%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 41\n",
      "Length of fail 362\n",
      "G000367205 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Cutibacterium; s__Cutibacterium avidum\n",
      "Coverage Percentage: 0.5281975885719624%\n",
      "Adj, read counts 89\n",
      "Number of samples per group [1, 17, 1, 1, 1, 7, 1, 1, 15, 2, 2, 1, 5, 1, 3, 1, 2, 1, 1, 1, 1, 3, 1, 1]\n",
      "Number of bins 35\n",
      "total # samples 37\n",
      "Number of total reads 88\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 64.04494382022472 %\n",
      "Percentage of reads from samples who only fall into one bin group 25.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 42\n",
      "Length of fail 362\n",
      "G000367345 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium xenophagum\n",
      "Coverage Percentage: 0.0670504058665763%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 42\n",
      "Length of fail 363\n",
      "G000367365 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces prunicolor\n",
      "Coverage Percentage: 0.0101320317639195%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 42\n",
      "Length of fail 364\n",
      "G000367505 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Anoxybacillus; s__Anoxybacillus flavithermus\n",
      "Coverage Percentage: 0.0102003889844449%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 42\n",
      "Length of fail 365\n",
      "G000367845 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas diversa\n",
      "Coverage Percentage: 0.0069505106267101%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 42\n",
      "Length of fail 366\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000367925 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter bohemicus\n",
      "Coverage Percentage: 0.93686251932091%\n",
      "Adj, read counts 212\n",
      "Number of samples per group [3, 3, 2, 1, 3, 1, 1, 1, 1, 1, 3, 1, 1, 4, 1, 2, 2, 4, 4, 3, 4, 1, 4, 1, 1, 3, 2, 1, 2, 1, 3, 1, 7, 5, 2, 1, 8, 3, 3, 1, 2, 1, 2, 3, 1, 1, 4, 1, 3, 2, 1, 2, 1, 2, 3, 4, 1, 2, 2, 2, 2, 1, 1, 1, 3, 1, 4, 3, 1, 1, 1, 1, 1, 2, 3, 1, 1, 2, 1, 1, 2, 1, 2, 1, 2, 3, 2, 6, 5, 1, 1]\n",
      "Number of bins 156\n",
      "total # samples 40\n",
      "Number of total reads 210\n",
      "Number of hits in BIOM tables 184.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.377358490566039 %\n",
      "Percentage of reads from samples who only fall into one bin group 4.285714285714286 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 43\n",
      "Length of fail 366\n",
      "G000367945 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter proteolyticus\n",
      "Coverage Percentage: 0.0806013598610791%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 43\n",
      "Length of fail 367\n",
      "G000368025 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter parvus\n",
      "Coverage Percentage: 0.440709667186441%\n",
      "Adj, read counts 90\n",
      "Number of samples per group [2, 1, 1, 1, 2, 1, 1, 1, 2, 3, 2, 1, 1, 11, 1, 1, 2, 2, 1, 5, 1, 1, 1, 1, 4, 1, 1, 4, 1, 4, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 2, 3]\n",
      "Number of bins 55\n",
      "total # samples 34\n",
      "Number of total reads 89\n",
      "Number of hits in BIOM tables 23.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 25.555555555555554 %\n",
      "Percentage of reads from samples who only fall into one bin group 17.97752808988764 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 44\n",
      "Length of fail 367\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000368045 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter johnsonii\n",
      "Coverage Percentage: 0.8934205286644168%\n",
      "Adj, read counts 209\n",
      "Number of samples per group [2, 4, 1, 2, 4, 1, 1, 2, 1, 7, 3, 4, 5, 3, 1, 1, 2, 3, 1, 3, 1, 2, 5, 1, 2, 1, 3, 4, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 2, 1, 1, 2, 1, 5, 2, 3, 1, 2, 1, 1, 2, 3, 1, 2, 1, 2, 3, 1, 3, 1, 10, 1, 1, 2, 1, 3, 5, 2, 1, 4, 2, 1, 3, 2, 4, 1, 1, 1, 5, 2, 3, 2, 2, 1]\n",
      "Number of bins 140\n",
      "total # samples 46\n",
      "Number of total reads 204\n",
      "Number of hits in BIOM tables 71.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 13.397129186602871 %\n",
      "Percentage of reads from samples who only fall into one bin group 3.431372549019608 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 45\n",
      "Length of fail 367\n",
      "G000368065 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter seifertii\n",
      "Coverage Percentage: 0.1781429746828563%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 45\n",
      "Length of fail 368\n",
      "G000368085 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter nosocomialis\n",
      "Coverage Percentage: 0.0403135395134117%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 45\n",
      "Length of fail 369\n",
      "G000368145 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter guillouiae\n",
      "Coverage Percentage: 0.2370791070747808%\n",
      "Adj, read counts 73\n",
      "Number of samples per group [1, 3, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 6, 7, 5, 2, 1]\n",
      "Number of bins 41\n",
      "total # samples 25\n",
      "Number of total reads 71\n",
      "Number of hits in BIOM tables 20.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 46.57534246575342 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.901408450704224 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 46\n",
      "Length of fail 369\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000368565 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter gerneri\n",
      "Coverage Percentage: 0.0830723813270345%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 46\n",
      "Length of fail 370\n",
      "G000368625 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter schindleri\n",
      "Coverage Percentage: 0.2153231113276451%\n",
      "Adj, read counts 47\n",
      "Number of samples per group [4, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 3, 1, 4, 1, 1, 10, 2, 1]\n",
      "Number of bins 39\n",
      "total # samples 24\n",
      "Number of total reads 47\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 42.5531914893617 %\n",
      "Percentage of reads from samples who only fall into one bin group 27.659574468085108 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 47\n",
      "Length of fail 370\n",
      "G000368665 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter junii\n",
      "Coverage Percentage: 0.3852673185246593%\n",
      "Adj, read counts 97\n",
      "Number of samples per group [1, 2, 1, 3, 2, 2, 5, 1, 1, 2, 2, 4, 1, 1, 1, 3, 1, 2, 4, 3, 2, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 2, 1, 2, 1, 2, 2, 1, 1, 1, 1, 3, 1, 1, 3, 1, 1, 1, 1, 2, 1]\n",
      "Number of bins 66\n",
      "total # samples 27\n",
      "Number of total reads 94\n",
      "Number of hits in BIOM tables 10.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.432989690721648 %\n",
      "Percentage of reads from samples who only fall into one bin group 10.638297872340425 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 48\n",
      "Length of fail 370\n",
      "G000368765 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter junii\n",
      "Coverage Percentage: 0.183449955090498%\n",
      "Adj, read counts 40\n",
      "Number of samples per group [1, 1, 2, 1, 2, 1, 1, 1, 1, 2, 4, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 1, 1]\n",
      "Number of bins 31\n",
      "total # samples 24\n",
      "Number of total reads 39\n",
      "Number of hits in BIOM tables 8.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 27.500000000000004 %\n",
      "Percentage of reads from samples who only fall into one bin group 38.46153846153847 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 49\n",
      "Length of fail 370\n",
      "G000368785 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter towneri\n",
      "Coverage Percentage: 0.1581349422688944%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 49\n",
      "Length of fail 371\n",
      "G000368805 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter johnsonii\n",
      "Coverage Percentage: 0.7132612822886722%\n",
      "Adj, read counts 158\n",
      "Number of samples per group [2, 4, 2, 5, 4, 2, 2, 1, 6, 1, 1, 2, 3, 1, 3, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 2, 1, 4, 1, 1, 1, 1, 1, 1, 1, 1, 2, 4, 4, 2, 2, 1, 2, 2, 1, 1, 1, 2, 4, 2, 5, 1, 1, 1, 2, 3, 1, 1, 2, 2, 2, 1, 1, 1, 2]\n",
      "Number of bins 115\n",
      "total # samples 46\n",
      "Number of total reads 154\n",
      "Number of hits in BIOM tables 121.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.759493670886076 %\n",
      "Percentage of reads from samples who only fall into one bin group 12.337662337662337 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 50\n",
      "Length of fail 371\n",
      "G000368825 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter ursingii\n",
      "Coverage Percentage: 0.118187096480434%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 50\n",
      "Length of fail 372\n",
      "G000368865 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter bouvetii\n",
      "Coverage Percentage: 0.0667173809252617%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 50\n",
      "Length of fail 373\n",
      "G000368885 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter radioresistens\n",
      "Coverage Percentage: 0.3360480046912756%\n",
      "Adj, read counts 62\n",
      "Number of samples per group [1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 6, 1, 2, 1, 1, 2, 1, 1, 2, 1, 2, 2, 1, 2, 1, 1, 1, 1, 2, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 47\n",
      "total # samples 22\n",
      "Number of total reads 62\n",
      "Number of hits in BIOM tables 31.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 19.35483870967742 %\n",
      "Percentage of reads from samples who only fall into one bin group 19.35483870967742 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 51\n",
      "Length of fail 373\n",
      "G000368905 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter radioresistens\n",
      "Coverage Percentage: 0.4827072158251364%\n",
      "Adj, read counts 92\n",
      "Number of samples per group [7, 1, 16, 1, 2, 1, 1, 2, 1, 1, 4, 1, 2, 3, 1, 4, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1]\n",
      "Number of bins 57\n",
      "total # samples 31\n",
      "Number of total reads 91\n",
      "Number of hits in BIOM tables 38.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 36.95652173913044 %\n",
      "Percentage of reads from samples who only fall into one bin group 13.186813186813188 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 52\n",
      "Length of fail 373\n",
      "G000368925 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter bereziniae\n",
      "Coverage Percentage: 0.1561103670652683%\n",
      "Adj, read counts 54\n",
      "Number of samples per group [3, 4, 5, 9, 3, 4, 1, 1, 1, 1, 1, 1, 1, 2, 3, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1]\n",
      "Number of bins 36\n",
      "total # samples 26\n",
      "Number of total reads 54\n",
      "Number of hits in BIOM tables 31.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 33.33333333333333 %\n",
      "Percentage of reads from samples who only fall into one bin group 25.925925925925924 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 53\n",
      "Length of fail 373\n",
      "G000369065 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter haemolyticus\n",
      "Coverage Percentage: 0.1208244410908981%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 53\n",
      "Length of fail 374\n",
      "G000369385 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter baumannii\n",
      "Coverage Percentage: 0.0366064641307528%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 53\n",
      "Length of fail 375\n",
      "G000369625 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter variabilis\n",
      "Coverage Percentage: 0.4302916894347451%\n",
      "Adj, read counts 96\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 3, 2, 1, 3, 1, 1, 3, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 9, 3, 1, 2, 2, 11, 4, 1, 1, 3, 1, 6]\n",
      "Number of bins 58\n",
      "total # samples 33\n",
      "Number of total reads 94\n",
      "Number of hits in BIOM tables 23.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 35.41666666666667 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.085106382978726 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 54\n",
      "Length of fail 375\n",
      "G000369885 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter ursingii\n",
      "Coverage Percentage: 0.0905303488893795%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 54\n",
      "Length of fail 376\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000371765 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas melonis\n",
      "Coverage Percentage: 0.1383310295126554%\n",
      "Adj, read counts 36\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1]\n",
      "Number of bins 31\n",
      "total # samples 21\n",
      "Number of total reads 33\n",
      "Number of hits in BIOM tables 14.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.666666666666664 %\n",
      "Percentage of reads from samples who only fall into one bin group 45.45454545454545 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 55\n",
      "Length of fail 376\n",
      "G000372405 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Porphyromonadaceae; g__Porphyromonas; s__Porphyromonas somerae\n",
      "Coverage Percentage: 0.0343430180644275%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 55\n",
      "Length of fail 377\n",
      "G000372725 d__Eukaryota; p__Haptophyta; c__; o__Isochrysidales; f__Noelaerhabdaceae; g__Emiliania; s__Emiliania huxleyi\n",
      "Coverage Percentage: 0.1052186088348155%\n",
      "Adj, read counts 16876\n",
      "Number of samples per group [8, 6, 8, 1, 1, 30, 71, 15, 5, 20, 1, 6, 1, 9, 53, 1, 10, 14, 11, 1, 11, 4, 1, 24, 10, 2, 11, 1, 5, 31, 2, 12, 6, 33, 13, 8, 4, 16, 6, 4, 9, 12, 1, 5, 1, 63, 3, 9, 19, 9, 1, 33, 48, 10, 10, 3, 11, 5, 4, 19, 5, 5, 19, 4, 6, 6, 17, 4, 4, 43, 8, 44, 1, 4, 5, 2, 18, 1, 1, 3, 10, 21, 12, 32, 39, 20, 42, 5, 2, 33, 2, 6, 1, 5, 24, 41, 8, 8, 44, 26, 11, 6, 1, 1, 17, 4, 2, 4, 1, 22, 2, 2, 8, 6, 72, 1, 40, 6, 5, 4, 30, 1, 2, 2, 15, 4, 6, 5, 4, 8, 7, 53, 4, 5, 3, 31, 12, 2, 30, 13, 3, 17, 1, 10, 2, 4, 1, 36, 63, 8, 11, 54, 5, 5, 16, 14, 3, 27, 1, 1, 21, 5, 2, 6, 30, 3, 1, 3, 6, 4, 16, 1, 1, 3, 1, 54, 44, 17, 2, 2, 28, 1, 2, 9, 5, 22, 4, 9, 36, 1, 2, 22, 5, 9, 51, 12, 23, 31, 33, 24, 23, 2, 12, 3, 16, 8, 19, 6, 5, 6, 1, 6, 2, 54, 6, 59, 6, 2, 1, 52, 21, 35, 29, 50, 9, 3, 12, 19, 1, 5, 11, 2, 34, 61, 3, 38, 6, 9, 2, 2, 44, 8, 2, 59, 6, 3, 1, 30, 5, 15, 13, 3, 14, 5, 65, 5, 50, 4, 6, 1, 2, 2, 4, 9, 19, 7, 8, 15, 1, 7, 4, 3, 5, 5, 13, 3, 16, 2, 10, 21, 33, 1, 34, 10, 39, 23, 31, 22, 29, 11, 50, 11, 4, 32, 83, 7, 4, 29, 48, 2, 41, 20, 1, 5, 66, 3, 1, 5, 30, 3, 73, 6, 34, 44, 11, 7, 4, 31, 55, 2, 53, 37, 30, 18, 3, 4, 75, 11, 1, 61, 3, 4, 4, 83, 23, 5, 1, 1, 20, 16, 23, 17, 19, 6, 2, 4, 1, 3, 4, 31, 2, 6, 6, 11, 5, 11, 2, 1, 32, 9, 1, 12, 3, 5, 2, 45, 1, 49, 4, 48, 2, 2, 10, 4, 31, 1, 4, 26, 13, 25, 10, 2, 1, 14, 3, 52, 14, 4, 9, 8, 52, 19, 4, 5, 2, 21, 49, 11, 19, 26, 3, 11, 21, 1, 7, 14, 6, 8, 3, 2, 20, 1, 4, 1, 5, 7, 4, 8, 37, 5, 1, 38, 3, 2, 5, 2, 2, 3, 16, 2, 3, 14, 7, 62, 1, 2, 1, 3, 7, 7, 1, 8, 1, 20, 1, 29, 37, 1, 6, 1, 4, 11, 1, 48, 17, 6, 1, 5, 43, 18, 27, 1, 2, 1, 5, 4, 2, 1, 3, 10, 1, 4, 1, 1, 84, 40, 9, 1, 1, 14, 7, 12, 52, 1, 59, 1, 9, 6, 4, 2, 57, 43, 4, 2, 2, 11, 3, 29, 3, 1, 21, 2, 12, 4, 3, 32, 25, 11, 12, 2, 5, 35, 34, 2, 32, 10, 22, 51, 1, 10, 1, 13, 2, 4, 22, 14, 1, 33, 1, 3, 18, 10, 3, 6, 15, 7, 1, 6, 5, 13, 25, 25, 6, 2, 5, 25, 13, 3, 3, 2, 11, 12, 1, 1, 1, 3, 2, 3, 3, 2, 9, 3, 2, 2, 4, 27, 9, 12, 42, 9, 7, 51, 3, 4, 39, 6, 25, 39, 15, 13, 20, 16, 9, 1, 36, 1, 10, 2, 21, 28, 70, 1, 21, 39, 18, 24, 15, 16, 21, 56, 1, 37, 27, 71, 13, 4, 10, 8, 3, 4, 26, 3, 2, 10, 9, 3, 15, 38, 3, 6, 29, 1, 46, 23, 1, 4, 1, 26, 2, 29, 69, 13, 15, 8, 7, 2, 23, 34, 14, 5, 1, 2, 3, 25, 12, 23, 2, 15, 11, 45, 28, 5, 18, 37, 1, 18, 63, 35, 32, 1, 5, 29, 1, 30, 6, 8, 1, 19, 8, 34, 4, 3, 5, 5, 87, 12, 57, 18, 4, 7, 16, 5, 1, 1, 2, 18, 12, 3, 10, 1, 8, 3, 13, 38, 4, 37, 7, 65, 11, 6, 7, 1, 18, 11, 1, 27, 3, 1, 24, 41, 17, 5, 42, 2, 31, 13, 1, 2, 6, 8, 34, 14, 5, 39, 39, 46, 34, 50, 25, 77, 77, 35, 19, 1, 2, 79, 20, 1, 20, 16, 2, 1, 1, 1, 14, 29, 8, 1, 1, 1, 2, 43, 2, 2, 11, 1, 74, 5, 4, 13, 29, 18, 3, 1, 44, 13, 2, 5, 1, 26, 3, 2, 14, 1, 27, 6, 8, 7, 59, 43, 8, 25, 2, 23, 45, 54, 9, 28, 48, 13, 22, 1, 8, 23, 3, 12, 15, 2, 3, 6, 3, 23, 55, 32, 4, 2, 2, 13, 10, 24, 12, 1, 1, 5, 9, 1, 18, 6, 4, 45, 5, 3, 4, 10, 6, 16, 28, 1, 4, 57, 5, 22, 4, 10, 8, 4, 6, 34, 1, 49, 11, 22, 3, 5, 23, 3, 30, 3, 1, 20, 2, 7, 33, 46, 45, 35, 1, 19, 40, 1, 6, 11, 9, 1, 18, 7, 75, 33, 1, 24, 5, 4, 18, 4, 16, 13, 13, 2, 45, 31, 2, 47, 3, 2, 73, 2, 24, 30, 57, 3, 18, 10, 44, 11, 69, 11, 27, 52, 51, 3, 2, 2, 1, 12, 21, 40, 7, 1, 9, 28, 4, 1, 2, 51, 20, 6, 80, 7, 4, 16, 32, 1, 63, 2, 16, 6, 6, 5, 28, 6, 47, 6, 2, 12, 9, 21, 9, 8, 8, 43, 19, 24, 52, 1, 2, 3, 11, 2, 8, 11, 11, 4, 17, 14, 5, 1, 10, 1, 42, 33, 2, 27, 8, 40, 9, 23, 42, 2, 2, 11, 13, 2, 4, 30, 2, 3, 2, 29, 16, 12, 7, 5, 2, 22, 8, 14, 7, 10, 39, 6, 3, 2, 3, 1, 66, 40, 18, 1, 5, 60, 1, 13, 12, 12, 15, 3, 7, 20, 11, 16, 68, 85, 7, 1, 8, 4, 2, 12, 3, 1, 1, 1, 32, 24, 3, 3, 53, 26, 12, 21, 8, 3, 3, 15, 3, 4, 24, 16, 37, 6, 23, 1, 14, 32, 4, 1, 1, 10, 11, 5, 19, 3, 2, 3, 7, 6, 7, 5, 2, 13, 5, 46, 4, 2]\n",
      "Number of bins 1181\n",
      "total # samples 98\n",
      "Number of total reads 16696\n",
      "Number of hits in BIOM tables 2297.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 2.014695425456269 %\n",
      "Percentage of reads from samples who only fall into one bin group 0.0 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 56\n",
      "Length of fail 377\n",
      "G000372745 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Embleya; s__Embleya scabrispora\n",
      "Coverage Percentage: 0.002325283669252%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 56\n",
      "Length of fail 378\n",
      "G000372845 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylocystaceae; g__Methylocystis; s__Methylocystis rosea\n",
      "Coverage Percentage: 0.0291917066923649%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 56\n",
      "Length of fail 379\n",
      "G000373085 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola massiliensis\n",
      "Coverage Percentage: 0.092187385042774%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 56\n",
      "Length of fail 380\n",
      "G000373185 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella paludivivens\n",
      "Coverage Percentage: 0.0517785198401403%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 56\n",
      "Length of fail 381\n",
      "G000373725 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter bouvetii\n",
      "Coverage Percentage: 0.040365463845354%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 56\n",
      "Length of fail 382\n",
      "G000373765 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Legionellales; f__Legionellaceae; g__Legionella; s__Legionella shakespearei\n",
      "Coverage Percentage: 0.005037278708362%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 56\n",
      "Length of fail 383\n",
      "G000373785 d__Bacteria; p__Firmicutes; c__Erysipelotrichia; o__Erysipelotrichales; f__Erysipelotrichaceae; g__Erysipelothrix; s__Erysipelothrix tonsillarum\n",
      "Coverage Percentage: 0.0078194504428087%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 56\n",
      "Length of fail 384\n",
      "G000374065 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Cytophagaceae; g__Spirosoma; s__Spirosoma luteum\n",
      "Coverage Percentage: 0.0030839263340702%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 56\n",
      "Length of fail 385\n",
      "G000374425 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter tjernbergiae\n",
      "Coverage Percentage: 0.2377241084465975%\n",
      "Adj, read counts 52\n",
      "Number of samples per group [1, 3, 1, 2, 1, 1, 1, 3, 1, 1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 3, 1, 1, 1, 2, 1, 6]\n",
      "Number of bins 44\n",
      "total # samples 22\n",
      "Number of total reads 50\n",
      "Number of hits in BIOM tables 33.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 25.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 12.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 57\n",
      "Length of fail 385\n",
      "G000374505 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Rikenellaceae; g__Alistipes; s__Alistipes onderdonkii\n",
      "Coverage Percentage: 0.070353172928099%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 57\n",
      "Length of fail 386\n",
      "G000374525 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Amorphaceae; g__Amorphus; s__Amorphus coralli\n",
      "Coverage Percentage: 0.0034740195448339%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 57\n",
      "Length of fail 387\n",
      "G000374585 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola barnesiae\n",
      "Coverage Percentage: 0.0227578667924609%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 57\n",
      "Length of fail 388\n",
      "G000374625 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Brachymonas; s__Brachymonas chironomi\n",
      "Coverage Percentage: 0.1218333219526395%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 57\n",
      "Length of fail 389\n",
      "G000375525 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium propinquum\n",
      "Coverage Percentage: 0.304629217321288%\n",
      "Adj, read counts 41\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 2, 2, 1, 1, 1, 4]\n",
      "Number of bins 34\n",
      "total # samples 25\n",
      "Number of total reads 41\n",
      "Number of hits in BIOM tables 46.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 21.95121951219512 %\n",
      "Percentage of reads from samples who only fall into one bin group 41.46341463414634 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 58\n",
      "Length of fail 389\n",
      "G000376105 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Asticcacaulis; s__Asticcacaulis benevestitus\n",
      "Coverage Percentage: 0.0168168187429558%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 58\n",
      "Length of fail 390\n",
      "G000376265 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Thermales; f__Thermaceae; g__Thermus; s__Thermus igniterrae\n",
      "Coverage Percentage: 0.0181373889253282%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 58\n",
      "Length of fail 391\n",
      "G000376405 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Dysgonomonadaceae; g__Dysgonomonas; s__Dysgonomonas mossii\n",
      "Coverage Percentage: 0.0033380720366138%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 58\n",
      "Length of fail 392\n",
      "G000376565 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces hokutonensis\n",
      "Coverage Percentage: 0.0147060066430581%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 58\n",
      "Length of fail 393\n",
      "Length of pass 59\n",
      "Length of fail 393\n",
      "G000377625 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella veroralis\n",
      "Coverage Percentage: 0.0055159600967717%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 59\n",
      "Length of fail 394\n",
      "G000378065 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Porphyromonadaceae; g__Porphyromonas; s__Porphyromonas gulae\n",
      "Coverage Percentage: 0.0283601978220922%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 59\n",
      "Length of fail 395\n",
      "G000378245 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Rhodocyclaceae; g__Aromatoleum; s__Aromatoleum toluclasticum\n",
      "Coverage Percentage: 0.0061077429477081%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 59\n",
      "Length of fail 396\n",
      "G000378365 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Selenomonadales; f__Selenomonadaceae; g__Megamonas; s__Megamonas rupellensis\n",
      "Coverage Percentage: 0.180747748373742%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 59\n",
      "Length of fail 397\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000378465 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Elioraeaceae; g__Elioraea; s__Elioraea tepidiphila\n",
      "Coverage Percentage: 0.0035072748079883%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 59\n",
      "Length of fail 398\n",
      "G000378485 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium rivuli\n",
      "Coverage Percentage: 0.0423860188057611%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 59\n",
      "Length of fail 399\n",
      "G000379045 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas melonis\n",
      "Coverage Percentage: 0.1346836691473818%\n",
      "Adj, read counts 34\n",
      "Number of samples per group [2, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 3, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 29\n",
      "total # samples 21\n",
      "Number of total reads 34\n",
      "Number of hits in BIOM tables 16.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.588235294117645 %\n",
      "Percentage of reads from samples who only fall into one bin group 47.05882352941176 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 60\n",
      "Length of fail 399\n",
      "G000379145 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium elkanii\n",
      "Coverage Percentage: 0.0106697177279784%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 60\n",
      "Length of fail 400\n",
      "G000379185 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Capnocytophaga; s__Capnocytophaga cynodegmi\n",
      "Coverage Percentage: 0.008089324199044%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 60\n",
      "Length of fail 401\n",
      "G000379625 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Actinomycetospora; s__Actinomycetospora chiangmaiensis\n",
      "Coverage Percentage: 0.039169530881746%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 60\n",
      "Length of fail 402\n",
      "G000379645 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes globisporus\n",
      "Coverage Percentage: 0.0068700148601606%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 60\n",
      "Length of fail 403\n",
      "G000379725 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Cytophagaceae; g__Cytophaga; s__Cytophaga aurantiaca\n",
      "Coverage Percentage: 0.0209442063964555%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 60\n",
      "Length of fail 404\n",
      "G000379745 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Enterococcaceae; g__Enterococcus; s__Enterococcus cecorum\n",
      "Coverage Percentage: 0.0274151252064141%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 60\n",
      "Length of fail 405\n",
      "G000379865 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacterium; s__Mycolicibacterium hassiacum\n",
      "Coverage Percentage: 0.0435852828863842%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 60\n",
      "Length of fail 406\n",
      "G000380105 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus orisratti\n",
      "Coverage Percentage: 0.0162950377887716%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 60\n",
      "Length of fail 407\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000380505 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Kaistiaceae; g__Kaistia; s__Kaistia granuli\n",
      "Coverage Percentage: 0.0031679515752376%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 60\n",
      "Length of fail 408\n",
      "G000380545 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium crenatum\n",
      "Coverage Percentage: 0.011082434192446%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 60\n",
      "Length of fail 409\n",
      "G000381045 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Thermales; f__Thermaceae; g__Thermus; s__Thermus scotoductus\n",
      "Coverage Percentage: 0.0511976930637983%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 60\n",
      "Length of fail 410\n",
      "G000381065 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Selenomonadales; f__Selenomonadaceae; g__Propionispira; s__Propionispira raffinosivorans\n",
      "Coverage Percentage: 0.013782890934775%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 60\n",
      "Length of fail 411\n",
      "G000381125 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Caldimonas; s__Caldimonas manganoxidans\n",
      "Coverage Percentage: 0.1205758209965488%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 60\n",
      "Length of fail 412\n",
      "G000381265 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Curvibacter; s__Curvibacter lanceolatus\n",
      "Coverage Percentage: 0.2401395504762038%\n",
      "Adj, read counts 108\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 2, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 4, 1, 1, 1, 1, 1, 1, 1, 4, 1, 1, 1, 1, 3, 1, 1, 10]\n",
      "Number of bins 86\n",
      "total # samples 33\n",
      "Number of total reads 106\n",
      "Number of hits in BIOM tables 127.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.59259259259259 %\n",
      "Percentage of reads from samples who only fall into one bin group 9.433962264150944 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 61\n",
      "Length of fail 412\n",
      "G000381365 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides salyersiae\n",
      "Coverage Percentage: 0.0064151404840515%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 61\n",
      "Length of fail 413\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000382065 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__[Pseudomonas] hibiscicola\n",
      "Coverage Percentage: 0.3002293025878758%\n",
      "Adj, read counts 81\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 2, 2, 1, 1, 1, 2, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 1, 3, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 68\n",
      "total # samples 26\n",
      "Number of total reads 80\n",
      "Number of hits in BIOM tables 24.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 13.580246913580247 %\n",
      "Percentage of reads from samples who only fall into one bin group 20.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 62\n",
      "Length of fail 413\n",
      "G000382265 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Yoonia; s__Yoonia vestfoldensis\n",
      "Coverage Percentage: 0.0040564398534738%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 62\n",
      "Length of fail 414\n",
      "G000382445 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola massiliensis\n",
      "Coverage Percentage: 0.1008503409778867%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 62\n",
      "Length of fail 415\n",
      "G000382905 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium glutamicum\n",
      "Coverage Percentage: 0.0157732519489937%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 62\n",
      "Length of fail 416\n",
      "G000383415 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Hyphomicrobiaceae; g__Hyphomicrobium; s__Hyphomicrobium zavarzinii\n",
      "Coverage Percentage: 0.0056533912286452%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 62\n",
      "Length of fail 417\n",
      "G000383855 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Methylococcales; f__Methylococcaceae; g__Methylobacter; s__Methylobacter marinus\n",
      "Coverage Percentage: 0.0260583080701377%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 62\n",
      "Length of fail 418\n",
      "G000384075 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Methylococcales; f__Methylococcaceae; g__Methylovulum; s__Methylovulum miyakonense\n",
      "Coverage Percentage: 0.0064431573775325%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 62\n",
      "Length of fail 419\n",
      "G000384115 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Sporichthyales; f__Sporichthyaceae; g__Sporichthya; s__Sporichthya polymorpha\n",
      "Coverage Percentage: 0.0054543937232291%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 62\n",
      "Length of fail 420\n",
      "G000385175 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Salinicoccus; s__Salinicoccus albus\n",
      "Coverage Percentage: 0.0123864011709126%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 62\n",
      "Length of fail 421\n",
      "G000392435 d__Bacteria; p__Candidatus Saccharibacteria; c__; o__; f__; g__Candidatus Saccharimonas; s__Candidatus Saccharimonas aalborgensis\n",
      "Coverage Percentage: 0.3123948860750004%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 62\n",
      "Length of fail 422\n",
      "G000400485 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Agrococcus; s__Agrococcus pavilionensis\n",
      "Coverage Percentage: 0.0245886010996105%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 62\n",
      "Length of fail 423\n",
      "G000400735 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter tandoii\n",
      "Coverage Percentage: 0.1622027610503542%\n",
      "Adj, read counts 45\n",
      "Number of samples per group [1, 2, 2, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 3, 6, 2, 7]\n",
      "Number of bins 31\n",
      "total # samples 25\n",
      "Number of total reads 43\n",
      "Number of hits in BIOM tables 8.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 35.55555555555555 %\n",
      "Percentage of reads from samples who only fall into one bin group 32.55813953488372 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 423\n",
      "G000402035 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Caballeronia; s__Caballeronia insecticola\n",
      "Coverage Percentage: 0.0021681175584928%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 424\n",
      "G000403725 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium ihumii\n",
      "Coverage Percentage: 0.0878300707153748%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 425\n",
      "G000407565 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Enterococcaceae; g__Enterococcus; s__Enterococcus cecorum\n",
      "Coverage Percentage: 0.037366461607829%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 426\n",
      "G000411375 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium pyruviciproducens\n",
      "Coverage Percentage: 0.1438480095394329%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 427\n",
      "G000412675 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas putida\n",
      "Coverage Percentage: 0.0262640657715877%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 428\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000413855 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter gyllenbergii\n",
      "Coverage Percentage: 0.0590648666431862%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 429\n",
      "G000413935 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter colistiniresistens\n",
      "Coverage Percentage: 0.2016478684332319%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 430\n",
      "G000414075 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter gyllenbergii\n",
      "Coverage Percentage: 0.0685196605128604%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 431\n",
      "G000417875 d__Eukaryota; p__Basidiomycota; c__Ustilaginomycetes; o__Ustilaginales; f__Ustilaginaceae; g__Pseudozyma; s__Pseudozyma flocculosa\n",
      "Coverage Percentage: 0.0027846739099653%\n",
      "Adj, read counts 47\n",
      "Number of samples per group [35, 3, 6]\n",
      "Number of bins 4\n",
      "total # samples 36\n",
      "Number of total reads 47\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 99.99999999999999 %\n",
      "Percentage of reads from samples who only fall into one bin group 63.829787234042556 %\n",
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 432\n",
      "G000420065 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Varibaculum; s__Varibaculum cambriense\n",
      "Coverage Percentage: 0.0096385970947785%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 433\n",
      "G000420525 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Chromobacteriaceae; g__Aquaspirillum; s__Aquaspirillum serpens\n",
      "Coverage Percentage: 0.0259258055449868%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 434\n",
      "G000420585 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium callunae\n",
      "Coverage Percentage: 0.0186096092120684%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 435\n",
      "G000420645 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Nevskiales; f__Sinobacteraceae; g__Nevskia; s__Nevskia ramosa\n",
      "Coverage Percentage: 0.015435635944879%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 436\n",
      "G000420765 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingosinicellaceae; g__Sandarakinorhabdus; s__Sandarakinorhabdus limnophila\n",
      "Coverage Percentage: 0.0231068896405004%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 437\n",
      "G000420785 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus hyovaginalis\n",
      "Coverage Percentage: 0.0267977978657021%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 438\n",
      "G000420805 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Yaniella; s__Yaniella halotolerans\n",
      "Coverage Percentage: 0.0054403514106723%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 439\n",
      "G000421565 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Agromyces; s__Agromyces subbeticus\n",
      "Coverage Percentage: 0.014031579416881%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 440\n",
      "G000421705 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas naejangsanensis\n",
      "Coverage Percentage: 0.0496097733149345%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 441\n",
      "G000421765 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Denitrificimonas; s__Denitrificimonas caeni\n",
      "Coverage Percentage: 0.0680829102962996%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 442\n",
      "G000422245 d__Bacteria; p__Proteobacteria; c__Deltaproteobacteria; o__Desulfovibrionales; f__Desulfovibrionaceae; g__Desulfohalovibrio; s__Desulfohalovibrio alkalitolerans\n",
      "Coverage Percentage: 0.0047144068526869%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 443\n",
      "G000422325 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Oceanospirillales; f__Oceanospirillaceae; g__Marinospirillum; s__Marinospirillum minutulum\n",
      "Coverage Percentage: 0.0520743354969273%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 444\n",
      "G000422705 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium soli\n",
      "Coverage Percentage: 0.0249297196730459%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 445\n",
      "G000422725 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium subsaxonicum\n",
      "Coverage Percentage: 0.0468858524580905%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 446\n",
      "G000422745 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium gubbeenense\n",
      "Coverage Percentage: 0.0070212997088147%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 447\n",
      "G000422885 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Ottowia; s__Ottowia thiooxydans\n",
      "Coverage Percentage: 0.0394735927621628%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 448\n",
      "G000423285 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Chromobacteriaceae; g__Laribacter; s__Laribacter hongkongensis\n",
      "Coverage Percentage: 0.0276430713766603%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 449\n",
      "G000423385 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Selenomonadales; f__Selenomonadaceae; g__Megamonas; s__Megamonas hypermegale\n",
      "Coverage Percentage: 0.0062080953563446%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 450\n",
      "G000423445 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Acidipropionibacterium; s__Acidipropionibacterium thoenii\n",
      "Coverage Percentage: 0.0223549074636128%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 451\n",
      "G000423465 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Propionicicella; s__Propionicicella superfundia\n",
      "Coverage Percentage: 0.0076150472229495%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 452\n",
      "G000423565 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Spirosomaceae; g__Runella; s__Runella zeae\n",
      "Coverage Percentage: 0.0025913162755445%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 453\n",
      "G000423665 d__Bacteria; p__Actinobacteria; c__Thermoleophilia; o__Solirubrobacterales; f__Solirubrobacteraceae; g__Solirubrobacter; s__Solirubrobacter soli\n",
      "Coverage Percentage: 0.0047987400891792%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 454\n",
      "G000423685 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Selenomonadales; f__Sporomusaceae; g__Sporomusa; s__Sporomusa ovata\n",
      "Coverage Percentage: 0.0041155319794062%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 455\n",
      "G000423805 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Thalassobaculaceae; g__Thalassobaculum; s__Thalassobaculum salexigens\n",
      "Coverage Percentage: 0.0021633238723969%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 456\n",
      "G000423825 d__Bacteria; p__Proteobacteria; c__Acidithiobacillia; o__Acidithiobacillales; f__Thermithiobacillaceae; g__Thermithiobacillus; s__Thermithiobacillus tepidarius\n",
      "Coverage Percentage: 0.0344671679950042%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 457\n",
      "G000423885 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Thermomonas; s__Thermomonas fusca\n",
      "Coverage Percentage: 0.0649966328001114%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 458\n",
      "G000423905 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Thermales; f__Thermaceae; g__Thermus; s__Thermus antranikianii\n",
      "Coverage Percentage: 0.0507436085677743%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 459\n",
      "G000424685 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Methylococcales; f__Methylococcaceae; g__Methylococcus; s__Methylococcus capsulatus\n",
      "Coverage Percentage: 0.0270614205470264%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 460\n",
      "G000425085 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Aerococcaceae; g__Aerococcus; s__Aerococcus urinaeequi\n",
      "Coverage Percentage: 0.0580236230432085%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 461\n",
      "G000425105 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Agrococcus; s__Agrococcus lahaulensis\n",
      "Coverage Percentage: 0.0112767482040284%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 462\n",
      "G000425285 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Acidipropionibacterium; s__Acidipropionibacterium jensenii\n",
      "Coverage Percentage: 0.0049843965282192%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 463\n",
      "G000425485 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium tegetincola\n",
      "Coverage Percentage: 0.0236799613492174%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 464\n",
      "G000425565 d__Bacteria; p__Proteobacteria; c__Hydrogenophilalia; o__Hydrogenophilales; f__Hydrogenophilaceae; g__Tepidiphilus; s__Tepidiphilus margaritifer\n",
      "Coverage Percentage: 0.0339548481578819%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 465\n",
      "G000425625 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas azotifigens\n",
      "Coverage Percentage: 0.0032280028047757%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 466\n",
      "G000425845 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Jeotgalicoccus; s__Jeotgalicoccus psychrophilus\n",
      "Coverage Percentage: 0.0382316762863872%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 467\n",
      "G000426485 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Dysgonomonadaceae; g__Dysgonomonas; s__Dysgonomonas capnocytophagoides\n",
      "Coverage Percentage: 0.003448490063781%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 468\n",
      "G000426525 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides alkalitolerans\n",
      "Coverage Percentage: 0.0062412663467622%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 469\n",
      "G000427405 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Spirosomaceae; g__Flectobacillus; s__Flectobacillus major\n",
      "Coverage Percentage: 0.0318751356260498%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 470\n",
      "G000427785 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia alkalitolerans\n",
      "Coverage Percentage: 0.0110090290396266%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 471\n",
      "G000427805 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Budviciaceae; g__Budvicia; s__Budvicia aquatica\n",
      "Coverage Percentage: 0.0035442790663064%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 472\n",
      "G000427825 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Thermocrispum; s__Thermocrispum municipale\n",
      "Coverage Percentage: 0.0093440540338148%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 473\n",
      "G000427905 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Thermocrispum; s__Thermocrispum agreste\n",
      "Coverage Percentage: 0.0056716009595014%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 474\n",
      "G000428125 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides graminisolvens\n",
      "Coverage Percentage: 0.1167027470137057%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 475\n",
      "G000428485 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Epilithonimonas; s__Epilithonimonas tenax\n",
      "Coverage Percentage: 0.0198182454650948%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 476\n",
      "G000428685 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces slackii\n",
      "Coverage Percentage: 0.0097415125889346%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 477\n",
      "G000428845 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Spirosomaceae; g__Dyadobacter; s__Dyadobacter alkalitolerans\n",
      "Coverage Percentage: 0.0101612538510197%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 478\n",
      "G000429065 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Silanimonas; s__Silanimonas lenta\n",
      "Coverage Percentage: 0.0632685714976407%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 479\n",
      "G000429225 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces dentalis\n",
      "Coverage Percentage: 0.0099343568268135%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 480\n",
      "G000429325 d__Bacteria; p__Proteobacteria; c__Deltaproteobacteria; o__Desulfovibrionales; f__Desulfovibrionaceae; g__Fundidesulfovibrio; s__Fundidesulfovibrio putealis\n",
      "Coverage Percentage: 0.003208463885233%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 481\n",
      "G000429605 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Azonexaceae; g__Azonexus; s__Azonexus hydrophilus\n",
      "Coverage Percentage: 0.0261859935759024%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 482\n",
      "G000429765 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Gemmobacter; s__Gemmobacter nectariphilus\n",
      "Coverage Percentage: 0.0044864297654945%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 483\n",
      "G000430205 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter haemolyticus\n",
      "Coverage Percentage: 0.0762725847580574%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 63\n",
      "Length of fail 484\n",
      "G000430225 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter junii\n",
      "Coverage Percentage: 0.2531997413986641%\n",
      "Adj, read counts 57\n",
      "Number of samples per group [1, 3, 1, 2, 2, 1, 1, 1, 4, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 3, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 2, 1, 1]\n",
      "Number of bins 40\n",
      "total # samples 28\n",
      "Number of total reads 57\n",
      "Number of hits in BIOM tables 10.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 24.561403508771928 %\n",
      "Percentage of reads from samples who only fall into one bin group 38.59649122807017 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 484\n",
      "G000430245 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter gerneri\n",
      "Coverage Percentage: 0.0801710738154763%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 485\n",
      "G000430565 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella timonensis\n",
      "Coverage Percentage: 0.034541483181527%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 486\n",
      "G000430725 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Azohydromonas; s__Azohydromonas australica\n",
      "Coverage Percentage: 0.0229248306748114%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 487\n",
      "G000430825 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium daeguense\n",
      "Coverage Percentage: 0.0098694106158796%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 488\n",
      "G000430845 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium gregarium\n",
      "Coverage Percentage: 0.0474303386607257%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 489\n",
      "G000430905 d__Archaea; p__Euryarchaeota; c__Methanomicrobia; o__Methanomicrobiales; f__Methanocorpusculaceae; g__Methanocorpusculum; s__Methanocorpusculum bavaricum\n",
      "Coverage Percentage: 0.0176887275947664%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 490\n",
      "G000442645 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium maris\n",
      "Coverage Percentage: 0.0053289722812271%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 491\n",
      "G000442985 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Agrobacterium; s__Agrobacterium fabacearum\n",
      "Coverage Percentage: 0.0027104746382144%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 492\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000443165 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Zoogloeaceae; g__Thauera; s__Thauera terpenica\n",
      "Coverage Percentage: 0.0643852450520264%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 493\n",
      "G000445065 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium quisquiliarum\n",
      "Coverage Percentage: 0.1159039477775813%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 494\n",
      "G000445105 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium lactosutens\n",
      "Coverage Percentage: 0.0859872005692945%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 495\n",
      "G000445125 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium lindaniclasticum\n",
      "Coverage Percentage: 0.0415138526254117%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 496\n",
      "G000445145 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium baderi\n",
      "Coverage Percentage: 0.0921010798526877%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 497\n",
      "G000445445 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Selenomonadales; f__Sporomusaceae; g__Sporomusa; s__Sporomusa ovata\n",
      "Coverage Percentage: 0.0039934574165562%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 498\n",
      "G000447205 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium ummariense\n",
      "Coverage Percentage: 0.0838796900361649%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 499\n",
      "G000447435 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas salmonicida\n",
      "Coverage Percentage: 0.072437484690588%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 500\n",
      "G000455225 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Veillonellales; f__Veillonellaceae; g__Megasphaera; s__Megasphaera massiliensis\n",
      "Coverage Percentage: 0.0110298401001889%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 501\n",
      "G000455445 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella timonensis\n",
      "Coverage Percentage: 0.030302578236595%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 502\n",
      "G000455525 d__Bacteria; p__Actinobacteria; c__Acidimicrobiia; o__Acidimicrobiales; f__Microthrixaceae; g__Candidatus Microthrix; s__Candidatus Microthrix parvicella\n",
      "Coverage Percentage: 0.0099489587547987%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 503\n",
      "G000455825 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium maritypicum\n",
      "Coverage Percentage: 0.0299342845006371%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 64\n",
      "Length of fail 504\n",
      "G000463665 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Cutibacterium; s__Cutibacterium granulosum\n",
      "Coverage Percentage: 0.6764204313994683%\n",
      "Adj, read counts 84\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 4, 1, 1, 1, 3, 1, 1, 2, 1, 7, 1, 2, 1, 6, 2, 1, 1, 1, 2, 3, 2, 1, 1, 1, 1, 2, 2, 1, 1, 1]\n",
      "Number of bins 57\n",
      "total # samples 32\n",
      "Number of total reads 82\n",
      "Number of hits in BIOM tables 68.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 25.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 30.48780487804878 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 65\n",
      "Length of fail 504\n",
      "G000464535 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Verrucariales; f__Verrucariaceae; g__Endocarpon; s__Endocarpon pusillum\n",
      "Coverage Percentage: 0.0020485210929433%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 65\n",
      "Length of fail 505\n",
      "G000465575 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas chengduensis\n",
      "Coverage Percentage: 0.1710509270217793%\n",
      "Adj, read counts 57\n",
      "Number of samples per group [2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 3, 2, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2]\n",
      "Number of bins 51\n",
      "total # samples 39\n",
      "Number of total reads 57\n",
      "Number of hits in BIOM tables 6.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.035087719298245 %\n",
      "Percentage of reads from samples who only fall into one bin group 52.63157894736842 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 66\n",
      "Length of fail 505\n",
      "G000467105 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas alcaligenes\n",
      "Coverage Percentage: 0.0328856710065253%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 66\n",
      "Length of fail 506\n",
      "G000467875 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella disiens\n",
      "Coverage Percentage: 0.0062136221089551%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 66\n",
      "Length of fail 507\n",
      "G000468015 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Oscillospiraceae; g__Ruminococcus; s__Ruminococcus callidus\n",
      "Coverage Percentage: 0.0183489966259129%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 66\n",
      "Length of fail 508\n",
      "G000468135 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella pleuritidis\n",
      "Coverage Percentage: 0.0196978337547095%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 66\n",
      "Length of fail 509\n",
      "G000469035 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia fonticola\n",
      "Coverage Percentage: 0.0230217982040286%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 66\n",
      "Length of fail 510\n",
      "G000469345 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Eubacteriaceae; g__Eubacterium; s__Eubacterium ramulus\n",
      "Coverage Percentage: 0.0326996709160778%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 66\n",
      "Length of fail 511\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000469665 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Holosporales; f__Holosporaceae; g__Holospora; s__Holospora obtusa\n",
      "Coverage Percentage: 0.0143644085866244%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 66\n",
      "Length of fail 512\n",
      "Length of pass 67\n",
      "Length of fail 512\n",
      "G000472965 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium murdochi\n",
      "Coverage Percentage: 0.0089015108177244%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 67\n",
      "Length of fail 513\n",
      "G000472985 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium japonicum\n",
      "Coverage Percentage: 0.031446881760272%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 67\n",
      "Length of fail 514\n",
      "G000473485 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia cepacia\n",
      "Coverage Percentage: 0.0046907092607836%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 67\n",
      "Length of fail 515\n",
      "G000474745 d__Bacteria; p__Fibrobacteres; c__Chitinivibrionia; o__Chitinivibrionales; f__Chitinivibrionaceae; g__Chitinivibrio; s__Chitinivibrio alkaliphilus\n",
      "Coverage Percentage: 0.0058227238933739%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 67\n",
      "Length of fail 516\n",
      "G000478365 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Herbaspirillum; s__Herbaspirillum huttiense\n",
      "Coverage Percentage: 0.0168035553623502%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 67\n",
      "Length of fail 517\n",
      "G000478565 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Sterolibacteriaceae; g__Methyloversatilis; s__Methyloversatilis universalis\n",
      "Coverage Percentage: 0.0126061451640268%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 67\n",
      "Length of fail 518\n",
      "G000484655 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Blautia; s__Blautia wexlerae\n",
      "Coverage Percentage: 0.2553489658400123%\n",
      "Adj, read counts 72\n",
      "Number of samples per group [1, 1, 1, 2, 2, 1, 1, 1, 1, 2, 2, 1, 2, 1, 1, 1, 2, 1, 3, 1, 1, 1, 1, 1, 1, 1, 4, 2, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 2, 1, 1, 2, 1, 1]\n",
      "Number of bins 60\n",
      "total # samples 33\n",
      "Number of total reads 71\n",
      "Number of hits in BIOM tables 51.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.499999999999998 %\n",
      "Percentage of reads from samples who only fall into one bin group 22.535211267605636 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 68\n",
      "Length of fail 518\n",
      "G000488175 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter tjernbergiae\n",
      "Coverage Percentage: 0.2402615920265664%\n",
      "Adj, read counts 51\n",
      "Number of samples per group [1, 1, 3, 1, 1, 3, 1, 2, 2, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 3, 1, 1, 1, 2, 1, 2, 1, 2, 1, 1, 1, 1, 4]\n",
      "Number of bins 40\n",
      "total # samples 22\n",
      "Number of total reads 50\n",
      "Number of hits in BIOM tables 33.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 21.568627450980394 %\n",
      "Percentage of reads from samples who only fall into one bin group 10.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 69\n",
      "Length of fail 518\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000488255 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter indicus\n",
      "Coverage Percentage: 0.3114610117999638%\n",
      "Adj, read counts 65\n",
      "Number of samples per group [2, 1, 2, 1, 2, 1, 1, 3, 1, 3, 1, 1, 1, 1, 2, 3, 3, 2, 7, 11, 1, 2]\n",
      "Number of bins 34\n",
      "total # samples 25\n",
      "Number of total reads 64\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 44.61538461538462 %\n",
      "Percentage of reads from samples who only fall into one bin group 26.5625 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 70\n",
      "Length of fail 518\n",
      "G000492155 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Enterococcaceae; g__Enterococcus; s__Enterococcus cecorum\n",
      "Coverage Percentage: 0.0265273252109583%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 70\n",
      "Length of fail 519\n",
      "G000493735 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Carnobacterium; s__Carnobacterium inhibens\n",
      "Coverage Percentage: 0.0158176208296041%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 70\n",
      "Length of fail 520\n",
      "G000494755 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes friuliensis\n",
      "Coverage Percentage: 0.0113587023818398%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 70\n",
      "Length of fail 521\n",
      "G000495775 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Asticcacaulis; s__Asticcacaulis benevestitus\n",
      "Coverage Percentage: 0.0167480982225057%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 70\n",
      "Length of fail 522\n",
      "G000495915 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas chloritidismutans\n",
      "Coverage Percentage: 0.0770680300339925%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 70\n",
      "Length of fail 523\n",
      "G000498435 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Rhodoferax; s__Rhodoferax saidenbachensis\n",
      "Coverage Percentage: 0.0948768612203004%\n",
      "Adj, read counts 30\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 11]\n",
      "Number of bins 20\n",
      "total # samples 21\n",
      "Number of total reads 30\n",
      "Number of hits in BIOM tables 11.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 46.666666666666664 %\n",
      "Percentage of reads from samples who only fall into one bin group 53.333333333333336 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 71\n",
      "Length of fail 523\n",
      "G000498515 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium saliperosum\n",
      "Coverage Percentage: 0.0181257491976335%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 71\n",
      "Length of fail 524\n",
      "G000498535 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium limnosediminis\n",
      "Coverage Percentage: 0.0094011957398164%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 71\n",
      "Length of fail 525\n",
      "G000498975 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas soli\n",
      "Coverage Percentage: 0.0440150707602283%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 71\n",
      "Length of fail 526\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000499605 d__Eukaryota; p__Apicomplexa; c__Conoidasida; o__Eucoccidiorida; f__Eimeriidae; g__Eimeria; s__Eimeria maxima\n",
      "Coverage Percentage: 0.0028927391855531%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [3, 1, 2, 1, 6, 10, 8, 6]\n",
      "Number of bins 8\n",
      "total # samples 26\n",
      "Number of total reads 37\n",
      "Number of hits in BIOM tables 7.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 64.86486486486487 %\n",
      "Percentage of reads from samples who only fall into one bin group 48.64864864864865 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 72\n",
      "Length of fail 526\n",
      "G000499745 d__Eukaryota; p__Apicomplexa; c__Conoidasida; o__Eucoccidiorida; f__Eimeriidae; g__Eimeria; s__Eimeria mitis\n",
      "Coverage Percentage: 0.0014911826110356%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 72\n",
      "Length of fail 527\n",
      "G000499785 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides neonati\n",
      "Coverage Percentage: 0.022564137665913%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 72\n",
      "Length of fail 528\n",
      "G000503215 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas canadensis\n",
      "Coverage Percentage: 0.0391278843185346%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 72\n",
      "Length of fail 529\n",
      "G000507185 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas syringae\n",
      "Coverage Percentage: 0.0295769087431228%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 72\n",
      "Length of fail 530\n",
      "G000507785 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Mediterraneibacter; s__[Ruminococcus] lactaris\n",
      "Coverage Percentage: 0.0767407003325862%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 72\n",
      "Length of fail 531\n",
      "G000510645 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillales Family X. Incertae Sedis; g__Thermicanus; s__Thermicanus aegyptius\n",
      "Coverage Percentage: 0.0122483919201076%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 72\n",
      "Length of fail 532\n",
      "G000511955 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Tissierellaceae; g__Schnuerera; s__Schnuerera ultunensis\n",
      "Coverage Percentage: 0.0079579422750762%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 72\n",
      "Length of fail 533\n",
      "G000512235 d__Bacteria; p__Proteobacteria; c__Deltaproteobacteria; o__Syntrophorhabdales; f__Syntrophorhabdaceae; g__Syntrophorhabdus; s__Syntrophorhabdus aromaticivorans\n",
      "Coverage Percentage: 0.0062773937750597%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 72\n",
      "Length of fail 534\n",
      "G000513195 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides timonensis\n",
      "Coverage Percentage: 0.0600020907445813%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 72\n",
      "Length of fail 535\n",
      "G000513835 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas thermotolerans\n",
      "Coverage Percentage: 0.0186175050891842%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 72\n",
      "Length of fail 536\n",
      "G000515315 d__Bacteria; p__Bacteroidetes; c__Chitinophagia; o__Chitinophagales; f__Chitinophagaceae; g__Asinibacterium; s__Asinibacterium sp. OR53\n",
      "Coverage Percentage: 0.0094995461477456%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 72\n",
      "Length of fail 537\n",
      "G000518625 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Roseomonas; s__Roseomonas gilardii\n",
      "Coverage Percentage: 0.0092997928032176%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 72\n",
      "Length of fail 538\n",
      "G000518645 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Curvibacter; s__Curvibacter gracilis\n",
      "Coverage Percentage: 0.2582640954830451%\n",
      "Adj, read counts 115\n",
      "Number of samples per group [1, 1, 2, 1, 2, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 4, 1, 1, 3, 1, 1, 1, 2, 1, 1, 2, 13]\n",
      "Number of bins 90\n",
      "total # samples 35\n",
      "Number of total reads 113\n",
      "Number of hits in BIOM tables 111.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 8.849557522123893 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 538\n",
      "G000518785 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Ciceribacter; s__Ciceribacter selenitireducens\n",
      "Coverage Percentage: 0.0060676112718536%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 539\n",
      "G000519045 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Azonexaceae; g__Dechloromonas; s__Dechloromonas agitata\n",
      "Coverage Percentage: 0.0116342350876399%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 540\n",
      "G000519105 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Odoribacteraceae; g__Butyricimonas; s__Butyricimonas virosa\n",
      "Coverage Percentage: 0.0156382537113835%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 541\n",
      "G000519325 d__Bacteria; p__Actinobacteria; c__Thermoleophilia; o__Solirubrobacterales; f__Patulibacteraceae; g__Patulibacter; s__Patulibacter minatonensis\n",
      "Coverage Percentage: 0.0045818334469133%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 542\n",
      "G000522465 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Aliarcobacter; s__Aliarcobacter cibarius\n",
      "Coverage Percentage: 0.0640720294940216%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 543\n",
      "G000523435 d__Eukaryota; p__Ascomycota; c__Dothideomycetes; o__Pleosporales; f__Pleosporaceae; g__Bipolaris; s__Bipolaris zeicola\n",
      "Coverage Percentage: 0.0019242573932717%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 544\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000523455 d__Eukaryota; p__Ascomycota; c__Dothideomycetes; o__Pleosporales; f__Pleosporaceae; g__Bipolaris; s__Bipolaris oryzae\n",
      "Coverage Percentage: 0.0018135772039018%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 545\n",
      "G000524495 d__Eukaryota; p__Apicomplexa; c__Aconoidasida; o__Haemosporida; f__Plasmodiidae; g__Plasmodium; s__Plasmodium inui\n",
      "Coverage Percentage: 0.0005946416512476%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 546\n",
      "G000527035 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Tolumonas; s__Tolumonas lignilytica\n",
      "Coverage Percentage: 0.0451217323221366%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 547\n",
      "G000527135 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Sterolibacteriaceae; g__Methyloversatilis; s__Methyloversatilis discipulorum\n",
      "Coverage Percentage: 0.0070126080655674%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 548\n",
      "G000527765 d__Eukaryota; p__Ascomycota; c__Dothideomycetes; o__Pleosporales; f__Pleosporaceae; g__Bipolaris; s__Bipolaris victoriae\n",
      "Coverage Percentage: 0.0022927207638139%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 549\n",
      "G000550785 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium casei\n",
      "Coverage Percentage: 0.0159629506301055%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 550\n",
      "G000550805 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium vitaeruminis\n",
      "Coverage Percentage: 0.0110513067146921%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 551\n",
      "G000559025 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Luteimonas; s__Luteimonas huabeiensis\n",
      "Coverage Percentage: 0.0585207490376652%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 552\n",
      "G000576555 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Hymenobacteraceae; g__Hymenobacter; s__Hymenobacter swuensis\n",
      "Coverage Percentage: 0.0029131038264475%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 553\n",
      "G000577555 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium jeddahense\n",
      "Coverage Percentage: 0.0210336356497919%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 554\n",
      "G000577815 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Clostridium; s__Clostridium saudiense\n",
      "Coverage Percentage: 0.0199959406325719%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 555\n",
      "G000585155 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Aliarcobacter; s__Aliarcobacter faecis\n",
      "Coverage Percentage: 0.0368706026001581%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 556\n",
      "G000585355 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella oryzae\n",
      "Coverage Percentage: 0.0061435691277743%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 557\n",
      "G000585375 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Cryptosporangiales; f__Cryptosporangiaceae; g__Cryptosporangium; s__Cryptosporangium arvum\n",
      "Coverage Percentage: 0.0052849105039553%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 558\n",
      "G000585535 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Chaetothyriales; f__Herpotrichiellaceae; g__Cladophialophora; s__Cladophialophora psammophila\n",
      "Coverage Percentage: 0.0002967843266015%\n",
      "Adj, read counts 32\n",
      "Number of samples per group [32]\n",
      "Number of bins 1\n",
      "total # samples 32\n",
      "Number of total reads 32\n",
      "Number of hits in BIOM tables 10.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 100.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 100.0 %\n",
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 73\n",
      "Length of fail 559\n",
      "G000597745 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas maltophilia\n",
      "Coverage Percentage: 0.3306492928307557%\n",
      "Adj, read counts 100\n",
      "Number of samples per group [1, 3, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 2, 2, 2, 1, 1, 3, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 3, 1]\n",
      "Number of bins 82\n",
      "total # samples 29\n",
      "Number of total reads 98\n",
      "Number of hits in BIOM tables 21.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.3265306122449 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 74\n",
      "Length of fail 559\n",
      "G000597765 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas syringae\n",
      "Coverage Percentage: 0.0332559172895703%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 74\n",
      "Length of fail 560\n",
      "G000600005 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Neisseriaceae; g__Snodgrassella; s__Snodgrassella alvi\n",
      "Coverage Percentage: 0.0478247832852975%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 74\n",
      "Length of fail 561\n",
      "G000600275 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Eurotiales; f__Aspergillaceae; g__Aspergillus; s__Aspergillus ruber\n",
      "Coverage Percentage: 0.0034488669422937%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 74\n",
      "Length of fail 562\n",
      "G000611675 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium succinicans\n",
      "Coverage Percentage: 0.0484981714656683%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 74\n",
      "Length of fail 563\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000612685 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Castellaniella; s__Castellaniella defragrans\n",
      "Coverage Percentage: 0.0218578239625502%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 74\n",
      "Length of fail 564\n",
      "G000613225 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Klebsiella; s__Klebsiella quasipneumoniae\n",
      "Coverage Percentage: 0.0510114532159228%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 74\n",
      "Length of fail 565\n",
      "G000613325 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella veroralis\n",
      "Coverage Percentage: 0.005536493248652%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 74\n",
      "Length of fail 566\n",
      "G000613605 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella paludivivens\n",
      "Coverage Percentage: 0.0518685274814816%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 74\n",
      "Length of fail 567\n",
      "G000613645 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola barnesiae\n",
      "Coverage Percentage: 0.0227008126560336%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 74\n",
      "Length of fail 568\n",
      "G000613725 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella pleuritidis\n",
      "Coverage Percentage: 0.0251457595034571%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 74\n",
      "Length of fail 569\n",
      "G000613825 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides graminisolvens\n",
      "Coverage Percentage: 0.1248796831239149%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 74\n",
      "Length of fail 570\n",
      "G000613925 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella histicola\n",
      "Coverage Percentage: 0.0228067018493083%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 74\n",
      "Length of fail 571\n",
      "G000614005 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella timonensis\n",
      "Coverage Percentage: 0.0240708522812501%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 74\n",
      "Length of fail 572\n",
      "G000614125 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides rodentium\n",
      "Coverage Percentage: 0.1496696015412593%\n",
      "Adj, read counts 49\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 45\n",
      "total # samples 32\n",
      "Number of total reads 49\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.285714285714285 %\n",
      "Percentage of reads from samples who only fall into one bin group 53.06122448979592 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 75\n",
      "Length of fail 572\n",
      "G000614145 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides faecichinchillae\n",
      "Coverage Percentage: 0.0193294539293436%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 75\n",
      "Length of fail 573\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000614185 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola sartorii\n",
      "Coverage Percentage: 0.0786428321393967%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 75\n",
      "Length of fail 574\n",
      "G000614245 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella fusca\n",
      "Coverage Percentage: 0.0049116578513952%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 75\n",
      "Length of fail 575\n",
      "G000614285 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella scopos\n",
      "Coverage Percentage: 0.0276464579740889%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 75\n",
      "Length of fail 576\n",
      "G000614355 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Porphyromonadaceae; g__Porphyromonas; s__Porphyromonas cangingivalis\n",
      "Coverage Percentage: 0.0234611022984667%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 75\n",
      "Length of fail 577\n",
      "G000614585 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Porphyromonadaceae; g__Porphyromonas; s__Porphyromonas gingivicanis\n",
      "Coverage Percentage: 0.0364002666757505%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 75\n",
      "Length of fail 578\n",
      "G000615685 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactobacillus; s__Lactobacillus kefiranofaciens\n",
      "Coverage Percentage: 0.0129402338459793%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 75\n",
      "Length of fail 579\n",
      "G000619845 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Alkanindiges; s__Alkanindiges illinoisensis\n",
      "Coverage Percentage: 0.0783426394892969%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 75\n",
      "Length of fail 580\n",
      "G000619905 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Nitrosomonadaceae; g__Nitrosospira; s__Nitrosospira briensis\n",
      "Coverage Percentage: 0.0094077685115757%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 75\n",
      "Length of fail 581\n",
      "G000620105 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Chromobacteriaceae; g__Microvirgula; s__Microvirgula aerodenitrificans\n",
      "Coverage Percentage: 0.0159883300040812%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 75\n",
      "Length of fail 582\n",
      "G000620145 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Chromobacteriaceae; g__Deefgea; s__Deefgea rivuli\n",
      "Coverage Percentage: 0.0040674288112414%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 75\n",
      "Length of fail 583\n",
      "G000620805 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__; g__Exiguobacterium; s__Exiguobacterium undae\n",
      "Coverage Percentage: 0.0046824469235399%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 75\n",
      "Length of fail 584\n",
      "G000620825 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__; g__Exiguobacterium; s__Exiguobacterium undae\n",
      "Coverage Percentage: 0.0047373187897646%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 75\n",
      "Length of fail 585\n",
      "G000620985 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Listeriaceae; g__Brochothrix; s__Brochothrix thermosphacta\n",
      "Coverage Percentage: 0.0120622540118976%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 75\n",
      "Length of fail 586\n",
      "G000621065 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter tandoii\n",
      "Coverage Percentage: 0.1807672620883322%\n",
      "Adj, read counts 46\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 3, 1, 2, 4, 4, 6, 2, 1, 2, 4]\n",
      "Number of bins 30\n",
      "total # samples 24\n",
      "Number of total reads 46\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 30.434782608695656 %\n",
      "Percentage of reads from samples who only fall into one bin group 26.08695652173913 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 76\n",
      "Length of fail 586\n",
      "G000621225 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas mosselii\n",
      "Coverage Percentage: 0.0601412600983339%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 76\n",
      "Length of fail 587\n",
      "G000621245 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas monteilii\n",
      "Coverage Percentage: 0.0996609878338797%\n",
      "Adj, read counts 41\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 5, 5, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 33\n",
      "total # samples 18\n",
      "Number of total reads 40\n",
      "Number of hits in BIOM tables 8.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 34.146341463414636 %\n",
      "Percentage of reads from samples who only fall into one bin group 37.5 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 587\n",
      "G000621265 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas fulva\n",
      "Coverage Percentage: 0.0936378253222834%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 588\n",
      "G000621325 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Thiotrichales; f__Thiotrichaceae; g__Thiothrix; s__Thiothrix lacustris\n",
      "Coverage Percentage: 0.0043552414524353%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 589\n",
      "G000622145 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus yeei\n",
      "Coverage Percentage: 0.0411261151960092%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 590\n",
      "G000622225 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Roseomonas; s__Roseomonas mucosa\n",
      "Coverage Percentage: 0.0100491153079468%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 591\n",
      "G000632105 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium resinovorum\n",
      "Coverage Percentage: 0.0283822168128641%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 592\n",
      "G000633455 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter jeongneungensis\n",
      "Coverage Percentage: 0.0180195487511513%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 593\n",
      "G000647955 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas salmonicida\n",
      "Coverage Percentage: 0.0427578194444012%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 594\n",
      "G000648235 d__Bacteria; p__Chlamydiae; c__Chlamydiia; o__Parachlamydiales; f__Parachlamydiaceae; g__Neochlamydia; s__Neochlamydia sp. S13\n",
      "Coverage Percentage: 0.0220014662912696%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 595\n",
      "G000661955 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas rhizophila\n",
      "Coverage Percentage: 0.0862555539112269%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 596\n",
      "G000685095 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia caledonica\n",
      "Coverage Percentage: 0.00207308375327%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 597\n",
      "G000685925 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Methylococcales; f__Methylococcaceae; g__Methylobacter; s__Methylobacter tundripaludum\n",
      "Coverage Percentage: 0.0255937031773489%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 598\n",
      "G000686665 d__Bacteria; p__Firmicutes; c__Erysipelotrichia; o__Erysipelotrichales; f__Erysipelotrichaceae; g__Erysipelatoclostridium; s__[Clostridium] saccharogumia\n",
      "Coverage Percentage: 0.0211374087316887%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 599\n",
      "G000686885 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium sasangense\n",
      "Coverage Percentage: 0.0463915976145984%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 600\n",
      "G000686905 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Oceanospirillales; f__Oceanospirillaceae; g__Marinospirillum; s__Marinospirillum insulare\n",
      "Coverage Percentage: 0.0402489686201791%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 601\n",
      "G000687165 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax oryzae\n",
      "Coverage Percentage: 0.0497274217335786%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 602\n",
      "G000687475 d__Eukaryota; p__Ascomycota; c__Sordariomycetes; o__Hypocreales; f__Clavicipitaceae; g__Ustilaginoidea; s__Ustilaginoidea virens\n",
      "Coverage Percentage: 0.0018730685998456%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 603\n",
      "G000687635 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium ruminantium\n",
      "Coverage Percentage: 0.1001169232294241%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 604\n",
      "G000687735 d__Bacteria; p__Tenericutes; c__Mollicutes; o__Acholeplasmatales; f__Acholeplasmataceae; g__Acholeplasma; s__Acholeplasma equifetale\n",
      "Coverage Percentage: 0.0088083612394689%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 605\n",
      "G000688255 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Simplicispira; s__Simplicispira psychrophila\n",
      "Coverage Percentage: 0.089989956998628%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 606\n",
      "G000688395 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Haematomicrobium; s__Haematomicrobium sanguinis\n",
      "Coverage Percentage: 0.0051614614524284%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 607\n",
      "G000688415 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium pseudodiphtheriticum\n",
      "Coverage Percentage: 0.0990603418996943%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 608\n",
      "G000688495 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter towneri\n",
      "Coverage Percentage: 0.1173655463804115%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 77\n",
      "Length of fail 609\n",
      "G000689195 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Sphaerotilus; s__Sphaerotilus natans\n",
      "Coverage Percentage: 0.2037173854209361%\n",
      "Adj, read counts 64\n",
      "Number of samples per group [1, 1, 3, 3, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 6, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 2, 2, 3, 1, 1, 1, 1, 1]\n",
      "Number of bins 52\n",
      "total # samples 31\n",
      "Number of total reads 63\n",
      "Number of hits in BIOM tables 20.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.3125 %\n",
      "Percentage of reads from samples who only fall into one bin group 26.984126984126984 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 78\n",
      "Length of fail 609\n",
      "G000689415 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas knackmussii\n",
      "Coverage Percentage: 0.0642229597892552%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 78\n",
      "Length of fail 610\n",
      "G000690815 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Escherichia; s__Escherichia coli\n",
      "Coverage Percentage: 0.8403674918169417%\n",
      "Adj, read counts 248\n",
      "Number of samples per group [1, 1, 1, 4, 4, 3, 1, 1, 5, 2, 4, 5, 3, 1, 1, 2, 1, 1, 2, 1, 1, 1, 2, 1, 3, 2, 2, 1, 3, 2, 2, 1, 2, 2, 1, 2, 1, 2, 2, 2, 1, 1, 4, 1, 2, 1, 2, 2, 1, 1, 1, 1, 1, 2, 2, 1, 1, 2, 2, 1, 1, 1, 1, 1, 3, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 3, 1, 1, 1, 1, 2, 4, 2, 2, 2, 1, 2, 2, 1, 1, 3, 1, 1, 2, 1, 2, 3, 1, 4, 1, 1, 1, 5, 2, 2, 1, 2, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 167\n",
      "total # samples 57\n",
      "Number of total reads 246\n",
      "Number of hits in BIOM tables 37.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 6.451612903225806 %\n",
      "Percentage of reads from samples who only fall into one bin group 10.975609756097562 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 79\n",
      "Length of fail 610\n",
      "G000695095 d__Bacteria; p__Gemmatimonadetes; c__Gemmatimonadetes; o__Gemmatimonadales; f__Gemmatimonadaceae; g__Gemmatimonas; s__Gemmatimonas phototrophica\n",
      "Coverage Percentage: 0.0124879361024748%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 79\n",
      "Length of fail 611\n",
      "G000695795 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium seoulense\n",
      "Coverage Percentage: 0.0318155698598217%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 79\n",
      "Length of fail 612\n",
      "G000696305 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Moraxella; s__Moraxella bovoculi\n",
      "Coverage Percentage: 0.0069661392048955%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 79\n",
      "Length of fail 613\n",
      "G000696675 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Rhodococcus; s__Rhodococcus erythropolis\n",
      "Coverage Percentage: 0.1303744953754812%\n",
      "Adj, read counts 51\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 49\n",
      "total # samples 28\n",
      "Number of total reads 50\n",
      "Number of hits in BIOM tables 24.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.76470588235294 %\n",
      "Percentage of reads from samples who only fall into one bin group 36.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 80\n",
      "Length of fail 613\n",
      "G000699505 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Rhodoluna; s__Rhodoluna lacicola\n",
      "Coverage Percentage: 0.2527905885840161%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 80\n",
      "Length of fail 614\n",
      "G000701445 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas bacteroides\n",
      "Coverage Percentage: 0.0419768878732745%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 80\n",
      "Length of fail 615\n",
      "G000701685 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__; g__Gemella; s__Gemella sanguinis\n",
      "Coverage Percentage: 0.0143651441469832%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 80\n",
      "Length of fail 616\n",
      "G000701705 d__Bacteria; p__Proteobacteria; c__Deltaproteobacteria; o__Desulfovibrionales; f__Desulfovibrionaceae; g__Bilophila; s__Bilophila wadsworthia\n",
      "Coverage Percentage: 0.012623042295607%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 80\n",
      "Length of fail 617\n",
      "G000701905 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Proteiniclasticum; s__Proteiniclasticum ruminis\n",
      "Coverage Percentage: 0.0144290844168774%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 80\n",
      "Length of fail 618\n",
      "G000702905 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Selenomonadales; f__Selenomonadaceae; g__Mitsuokella; s__Mitsuokella jalaludinii\n",
      "Coverage Percentage: 0.0113925508459901%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 80\n",
      "Length of fail 619\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000708125 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas jandaei\n",
      "Coverage Percentage: 0.0271493475337519%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 80\n",
      "Length of fail 620\n",
      "G000708285 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter borealis\n",
      "Coverage Percentage: 0.0213723846084952%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 80\n",
      "Length of fail 621\n",
      "G000708655 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Carnobacterium; s__Carnobacterium jeotgali\n",
      "Coverage Percentage: 0.0064207941884583%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 80\n",
      "Length of fail 622\n",
      "G000709125 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Chaetothyriales; f__Herpotrichiellaceae; g__Exophiala; s__Exophiala aquamarina\n",
      "Coverage Percentage: 0.000553284177001%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 80\n",
      "Length of fail 623\n",
      "G000709345 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Polaromonas; s__Polaromonas glacialis\n",
      "Coverage Percentage: 0.0828450872560488%\n",
      "Adj, read counts 36\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 15, 1]\n",
      "Number of bins 16\n",
      "total # samples 23\n",
      "Number of total reads 36\n",
      "Number of hits in BIOM tables 17.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 63.88888888888889 %\n",
      "Percentage of reads from samples who only fall into one bin group 58.333333333333336 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 81\n",
      "Length of fail 623\n",
      "G000711965 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces israelii\n",
      "Coverage Percentage: 0.0089130719759113%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 81\n",
      "Length of fail 624\n",
      "G000716435 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces xylophagus\n",
      "Coverage Percentage: 0.0129487908243551%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 81\n",
      "Length of fail 625\n",
      "G000716715 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Dactylosporangium; s__Dactylosporangium aurantiacum\n",
      "Coverage Percentage: 0.0062270144129619%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 81\n",
      "Length of fail 626\n",
      "G000717585 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora globosa\n",
      "Coverage Percentage: 0.0214860948549178%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 81\n",
      "Length of fail 627\n",
      "G000718325 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Cellulomonadaceae; g__Oerskovia; s__Oerskovia turbata\n",
      "Coverage Percentage: 0.0143419833523815%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 81\n",
      "Length of fail 628\n",
      "G000719095 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptacidiphilus; s__Streptacidiphilus jeojiense\n",
      "Coverage Percentage: 0.0016485477440715%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 81\n",
      "Length of fail 629\n",
      "G000719975 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Prauserella; s__Prauserella rugosa\n",
      "Coverage Percentage: 0.0092862724194071%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 81\n",
      "Length of fail 630\n",
      "G000720335 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Marmoricola; s__Marmoricola aequoreus\n",
      "Coverage Percentage: 0.16538050724053%\n",
      "Adj, read counts 35\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 31\n",
      "total # samples 6\n",
      "Number of total reads 35\n",
      "Number of hits in BIOM tables 68.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.142857142857142 %\n",
      "Percentage of reads from samples who only fall into one bin group 8.571428571428571 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 82\n",
      "Length of fail 630\n",
      "G000720925 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Intrasporangiaceae; g__Phycicoccus; s__Phycicoccus jejuensis\n",
      "Coverage Percentage: 0.0151223897890012%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 82\n",
      "Length of fail 631\n",
      "G000721705 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes subtropicus\n",
      "Coverage Percentage: 0.008400200468209%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 82\n",
      "Length of fail 632\n",
      "G000721785 d__Eukaryota; p__Ascomycota; c__Dothideomycetes; o__Dothideales; f__Saccotheciaceae; g__Aureobasidium; s__Aureobasidium pullulans\n",
      "Coverage Percentage: 0.0031190107146119%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 82\n",
      "Length of fail 633\n",
      "G000722885 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter antarcticus\n",
      "Coverage Percentage: 0.0165253512020168%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 82\n",
      "Length of fail 634\n",
      "G000723165 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Janthinobacterium; s__Janthinobacterium agaricidamnosum\n",
      "Coverage Percentage: 0.0253824129463081%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 82\n",
      "Length of fail 635\n",
      "G000723465 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Oscillospiraceae; g__Ruminococcus; s__Ruminococcus bicirculans\n",
      "Coverage Percentage: 0.0989045046016196%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 82\n",
      "Length of fail 636\n",
      "Length of pass 83\n",
      "Length of fail 636\n",
      "G000730565 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas fulva\n",
      "Coverage Percentage: 0.0968520666981329%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 83\n",
      "Length of fail 637\n",
      "G000730605 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas monteilii\n",
      "Coverage Percentage: 0.0986100191904775%\n",
      "Adj, read counts 40\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 9, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 34\n",
      "total # samples 18\n",
      "Number of total reads 40\n",
      "Number of hits in BIOM tables 8.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 37.5 %\n",
      "Percentage of reads from samples who only fall into one bin group 35.0 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 84\n",
      "Length of fail 637\n",
      "G000730685 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pasteurellales; f__Pasteurellaceae; g__Rodentibacter; s__Rodentibacter pneumotropicus\n",
      "Coverage Percentage: 0.0097342348521751%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 84\n",
      "Length of fail 638\n",
      "G000731315 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Neorhizobium; s__Neorhizobium galegae\n",
      "Coverage Percentage: 0.0091091513353814%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 84\n",
      "Length of fail 639\n",
      "G000731675 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas capeferrum\n",
      "Coverage Percentage: 0.0136350253838722%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 84\n",
      "Length of fail 640\n",
      "G000733715 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas mendocina\n",
      "Coverage Percentage: 0.0422708708469474%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 84\n",
      "Length of fail 641\n",
      "G000734905 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Citrobacter; s__Citrobacter freundii\n",
      "Coverage Percentage: 0.0073844351202918%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 84\n",
      "Length of fail 642\n",
      "G000734955 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Escherichia; s__Escherichia coli\n",
      "Coverage Percentage: 0.8204259657072221%\n",
      "Adj, read counts 237\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 5, 3, 1, 1, 2, 1, 1, 2, 1, 1, 1, 2, 1, 1, 2, 2, 1, 1, 2, 2, 1, 1, 1, 2, 1, 1, 1, 2, 2, 1, 2, 2, 1, 1, 1, 1, 2, 1, 2, 2, 1, 1, 1, 1, 1, 2, 2, 1, 2, 2, 2, 1, 1, 1, 1, 1, 3, 2, 1, 1, 1, 1, 1, 2, 1, 3, 2, 1, 1, 1, 2, 4, 2, 2, 2, 1, 2, 2, 1, 1, 4, 1, 1, 2, 1, 2, 3, 1, 4, 1, 1, 1, 6, 1, 2, 1, 3, 2, 2, 2, 1, 1, 1, 1, 1, 1, 4, 1, 2]\n",
      "Number of bins 162\n",
      "total # samples 58\n",
      "Number of total reads 236\n",
      "Number of hits in BIOM tables 36.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 7.59493670886076 %\n",
      "Percentage of reads from samples who only fall into one bin group 12.288135593220339 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 85\n",
      "Length of fail 642\n",
      "G000735105 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Epilithonimonas; s__Epilithonimonas lactis\n",
      "Coverage Percentage: 0.0182920640765103%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 85\n",
      "Length of fail 643\n",
      "G000735345 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Ewingella; s__Ewingella americana\n",
      "Coverage Percentage: 0.0083989954062218%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 85\n",
      "Length of fail 644\n",
      "G000735435 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Raoultella; s__Raoultella planticola\n",
      "Coverage Percentage: 0.0250984713743379%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 85\n",
      "Length of fail 645\n",
      "G000735445 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia marcescens\n",
      "Coverage Percentage: 0.0408938777369253%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 85\n",
      "Length of fail 646\n",
      "G000735515 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Leclercia; s__Leclercia adecarboxylata\n",
      "Coverage Percentage: 0.0121332980947118%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 85\n",
      "Length of fail 647\n",
      "G000735695 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Kaistella; s__Kaistella haifensis\n",
      "Coverage Percentage: 0.0142223183643139%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 85\n",
      "Length of fail 648\n",
      "G000735715 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium chungangense\n",
      "Coverage Percentage: 0.1493090032971361%\n",
      "Adj, read counts 52\n",
      "Number of samples per group [1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 47\n",
      "total # samples 25\n",
      "Number of total reads 52\n",
      "Number of hits in BIOM tables 39.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.384615384615385 %\n",
      "Percentage of reads from samples who only fall into one bin group 23.076923076923077 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 86\n",
      "Length of fail 648\n",
      "G000737145 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter baumannii\n",
      "Coverage Percentage: 0.0210859464695071%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 86\n",
      "Length of fail 649\n",
      "G000737325 d__Bacteria; p__Proteobacteria; c__Deltaproteobacteria; o__Myxococcales; f__Sandaracinaceae; g__Sandaracinus; s__Sandaracinus amylolyticus\n",
      "Coverage Percentage: 0.0040475108050624%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 86\n",
      "Length of fail 650\n",
      "G000737685 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium reichenbachii\n",
      "Coverage Percentage: 0.0479131514148284%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 86\n",
      "Length of fail 651\n",
      "G000737695 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium hydatis\n",
      "Coverage Percentage: 0.0484894375174224%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 86\n",
      "Length of fail 652\n",
      "G000737705 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium soli\n",
      "Coverage Percentage: 0.0031756193193743%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 86\n",
      "Length of fail 653\n",
      "G000737765 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium vrystaatense\n",
      "Coverage Percentage: 0.0071061783300918%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 86\n",
      "Length of fail 654\n",
      "G000737785 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium luteum\n",
      "Coverage Percentage: 0.0050005233598601%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 86\n",
      "Length of fail 655\n",
      "G000738005 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium crudilactis\n",
      "Coverage Percentage: 0.0049938296579311%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 86\n",
      "Length of fail 656\n",
      "G000738675 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia nematodiphila\n",
      "Coverage Percentage: 0.0462397654327131%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 86\n",
      "Length of fail 657\n",
      "G000739875 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Comamonas; s__Comamonas aquatica\n",
      "Coverage Percentage: 0.0487629840482649%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 86\n",
      "Length of fail 658\n",
      "G000739935 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Agrobacterium; s__Agrobacterium rubi\n",
      "Coverage Percentage: 0.0053829608324437%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 86\n",
      "Length of fail 659\n",
      "G000739995 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Comamonas; s__Comamonas granuli\n",
      "Coverage Percentage: 0.0309127297138063%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 86\n",
      "Length of fail 660\n",
      "G000740035 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Comamonas; s__Comamonas testosteroni\n",
      "Coverage Percentage: 0.0379640471412499%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 86\n",
      "Length of fail 661\n",
      "G000740785 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paenirhodobacter; s__Paenirhodobacter enshiensis\n",
      "Coverage Percentage: 0.0043872252137029%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 86\n",
      "Length of fail 662\n",
      "G000741245 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium longum\n",
      "Coverage Percentage: 0.4008230311177717%\n",
      "Adj, read counts 57\n",
      "Number of samples per group [4, 2, 1, 1, 1, 1, 1, 4, 1, 1, 1, 2, 2, 1, 1, 2, 1, 2, 1, 1, 1, 2, 1, 1, 2, 1, 3, 1, 1, 1, 1, 2, 1, 1]\n",
      "Number of bins 46\n",
      "total # samples 23\n",
      "Number of total reads 55\n",
      "Number of hits in BIOM tables 7.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 26.31578947368421 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.363636363636363 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 87\n",
      "Length of fail 662\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000741285 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium mongoliense\n",
      "Coverage Percentage: 0.0069542630553623%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 87\n",
      "Length of fail 663\n",
      "G000741375 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium pullorum\n",
      "Coverage Percentage: 0.0132093801619849%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 87\n",
      "Length of fail 664\n",
      "G000741415 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium adolescentis\n",
      "Coverage Percentage: 0.4755969149110175%\n",
      "Adj, read counts 64\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 3, 2, 1, 5, 1, 3, 3, 1, 3, 1, 2, 1, 1, 2, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1]\n",
      "Number of bins 55\n",
      "total # samples 29\n",
      "Number of total reads 63\n",
      "Number of hits in BIOM tables 20.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.3125 %\n",
      "Percentage of reads from samples who only fall into one bin group 25.396825396825395 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 88\n",
      "Length of fail 664\n",
      "G000741565 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium catenulatum\n",
      "Coverage Percentage: 0.1876659900641204%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 88\n",
      "Length of fail 665\n",
      "G000741605 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium catenulatum\n",
      "Coverage Percentage: 0.1216786367141136%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 88\n",
      "Length of fail 666\n",
      "G000741685 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium pseudocatenulatum\n",
      "Coverage Percentage: 0.2034196925674314%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 88\n",
      "Length of fail 667\n",
      "G000741785 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium stellenboschense\n",
      "Coverage Percentage: 0.0161001178542843%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 88\n",
      "Length of fail 668\n",
      "G000741825 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Delftia; s__Delftia acidovorans\n",
      "Coverage Percentage: 0.0420480622630001%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 88\n",
      "Length of fail 669\n",
      "G000743055 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Salmonella; s__Salmonella enterica\n",
      "Coverage Percentage: 0.0263257237090461%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 88\n",
      "Length of fail 670\n",
      "G000743575 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Devosiaceae; g__Devosia; s__Devosia riboflavina\n",
      "Coverage Percentage: 0.011949809218526%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 88\n",
      "Length of fail 671\n",
      "G000744225 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Kitasatospora; s__Kitasatospora mediocidica\n",
      "Coverage Percentage: 0.0017392544311538%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 88\n",
      "Length of fail 672\n",
      "G000744945 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Carnobacterium; s__Carnobacterium maltaromaticum\n",
      "Coverage Percentage: 0.0933508840217799%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 88\n",
      "Length of fail 673\n",
      "G000745125 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Carnobacterium; s__Carnobacterium jeotgali\n",
      "Coverage Percentage: 0.0239828720998195%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 88\n",
      "Length of fail 674\n",
      "G000745855 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Xenophilus; s__Xenophilus azovorans\n",
      "Coverage Percentage: 0.0770290524646467%\n",
      "Adj, read counts 33\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 4, 1]\n",
      "Number of bins 25\n",
      "total # samples 20\n",
      "Number of total reads 33\n",
      "Number of hits in BIOM tables 22.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 27.27272727272727 %\n",
      "Percentage of reads from samples who only fall into one bin group 42.42424242424242 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 89\n",
      "Length of fail 674\n",
      "G000746485 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pasteurellales; f__Pasteurellaceae; g__Haemophilus; s__Haemophilus parainfluenzae\n",
      "Coverage Percentage: 0.0412835534186714%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 89\n",
      "Length of fail 675\n",
      "G000746825 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Carnobacterium; s__Carnobacterium inhibens\n",
      "Coverage Percentage: 0.0143703878331201%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 89\n",
      "Length of fail 676\n",
      "G000747195 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus equinus\n",
      "Coverage Percentage: 0.0609051085381145%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 89\n",
      "Length of fail 677\n",
      "G000747315 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium ureicelerivorans\n",
      "Coverage Percentage: 0.0633509971661703%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 89\n",
      "Length of fail 678\n",
      "G000747725 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter gerneri\n",
      "Coverage Percentage: 0.0843248915000737%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 89\n",
      "Length of fail 679\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000751755 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Klebsiella; s__Klebsiella quasipneumoniae\n",
      "Coverage Percentage: 0.068593177704254%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 89\n",
      "Length of fail 680\n",
      "G000754405 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium psychrophilum\n",
      "Coverage Percentage: 0.1166828490939865%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 89\n",
      "Length of fail 681\n",
      "G000755165 d__Eukaryota; p__Euglenozoa; c__Kinetoplastea; o__Trypanosomatida; f__Trypanosomatidae; g__Leishmania; s__Leishmania panamensis\n",
      "Coverage Percentage: 0.0020397873225197%\n",
      "Adj, read counts 88\n",
      "Number of samples per group [20, 2, 60]\n",
      "Number of bins 3\n",
      "total # samples 62\n",
      "Number of total reads 88\n",
      "Number of hits in BIOM tables 55.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 100.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 51.13636363636363 %\n",
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 89\n",
      "Length of fail 682\n",
      "G000755525 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Enterobacter; s__Enterobacter cloacae\n",
      "Coverage Percentage: 0.0289081121634751%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 89\n",
      "Length of fail 683\n",
      "G000756045 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia xenovorans\n",
      "Coverage Percentage: 0.0517366243048148%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 89\n",
      "Length of fail 684\n",
      "G000757385 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium aquatile\n",
      "Coverage Percentage: 0.0662851878672325%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 89\n",
      "Length of fail 685\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000757795 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium oryzae\n",
      "Coverage Percentage: 0.0520469714373156%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 89\n",
      "Length of fail 686\n",
      "G000759055 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium tuscaniense\n",
      "Coverage Percentage: 0.0271373463712209%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 89\n",
      "Length of fail 687\n",
      "G000759735 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Citrobacter; s__Citrobacter freundii\n",
      "Coverage Percentage: 0.0073822651073365%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 89\n",
      "Length of fail 688\n",
      "G000759775 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Escherichia; s__Escherichia albertii\n",
      "Coverage Percentage: 0.1264937557261012%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 89\n",
      "Length of fail 689\n",
      "G000759995 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter tjernbergiae\n",
      "Coverage Percentage: 0.2434918215811019%\n",
      "Adj, read counts 51\n",
      "Number of samples per group [6, 1, 2, 1, 4, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 2, 2, 1, 1, 1, 3, 1, 1, 1, 1, 3, 1, 2, 1, 1]\n",
      "Number of bins 40\n",
      "total # samples 22\n",
      "Number of total reads 51\n",
      "Number of hits in BIOM tables 34.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 25.49019607843137 %\n",
      "Percentage of reads from samples who only fall into one bin group 9.803921568627452 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 90\n",
      "Length of fail 689\n",
      "G000761055 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Rhodococcus; s__Rhodococcus fascians\n",
      "Coverage Percentage: 0.0407592792291128%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 90\n",
      "Length of fail 690\n",
      "G000763375 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium profundi\n",
      "Coverage Percentage: 0.0126128591214861%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 90\n",
      "Length of fail 691\n",
      "G000764535 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas taxi\n",
      "Coverage Percentage: 0.061251490050818%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 90\n",
      "Length of fail 692\n",
      "G000764645 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas finlandensis\n",
      "Coverage Percentage: 0.0483173887361436%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 90\n",
      "Length of fail 693\n",
      "G000764655 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas aquatica\n",
      "Coverage Percentage: 0.0303988220783548%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 90\n",
      "Length of fail 694\n",
      "G000764665 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas lacus\n",
      "Coverage Percentage: 0.0250552887462874%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 90\n",
      "Length of fail 695\n",
      "G000765945 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Porphyromonadaceae; g__Porphyromonas; s__Porphyromonas gulae\n",
      "Coverage Percentage: 0.0338193752372692%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 90\n",
      "Length of fail 696\n",
      "G000767465 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium pentaromativorans\n",
      "Coverage Percentage: 0.0330726730305452%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 90\n",
      "Length of fail 697\n",
      "G000768695 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Intrasporangiaceae; g__Knoellia; s__Knoellia aerolata\n",
      "Coverage Percentage: 0.0071349158837399%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 90\n",
      "Length of fail 698\n",
      "G000769135 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Porphyromonadaceae; g__Porphyromonas; s__Porphyromonas gingivicanis\n",
      "Coverage Percentage: 0.0377046536010968%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 90\n",
      "Length of fail 699\n",
      "G000769745 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Eurotiales; f__Aspergillaceae; g__Penicillium; s__Penicillium expansum\n",
      "Coverage Percentage: 0.0018848296182002%\n",
      "Adj, read counts 57\n",
      "Number of samples per group [2, 2, 52]\n",
      "Number of bins 3\n",
      "total # samples 53\n",
      "Number of total reads 57\n",
      "Number of hits in BIOM tables 33.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 100.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 89.47368421052632 %\n",
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 90\n",
      "Length of fail 700\n",
      "G000769825 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium rivuli\n",
      "Coverage Percentage: 0.0452332984915688%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 90\n",
      "Length of fail 701\n",
      "G000769935 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium subsaxonicum\n",
      "Coverage Percentage: 0.0533979598831927%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 90\n",
      "Length of fail 702\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000770925 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium ruminantium\n",
      "Coverage Percentage: 0.0995676291388117%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 90\n",
      "Length of fail 703\n",
      "G000770965 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium pullorum\n",
      "Coverage Percentage: 0.013273674308426%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 90\n",
      "Length of fail 704\n",
      "G000771025 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium catenulatum\n",
      "Coverage Percentage: 0.1721166316523955%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 90\n",
      "Length of fail 705\n",
      "G000771045 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium porcinum\n",
      "Coverage Percentage: 0.0195323102215535%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 90\n",
      "Length of fail 706\n",
      "G000771105 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium longum\n",
      "Coverage Percentage: 0.2592479904419142%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 90\n",
      "Length of fail 707\n",
      "G000771285 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium longum\n",
      "Coverage Percentage: 0.3449911704161921%\n",
      "Adj, read counts 49\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 2, 1, 1, 1, 1, 3, 1, 1, 1, 2, 2, 2, 2, 1, 3, 2, 1]\n",
      "Number of bins 41\n",
      "total # samples 24\n",
      "Number of total reads 48\n",
      "Number of hits in BIOM tables 8.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 18.36734693877551 %\n",
      "Percentage of reads from samples who only fall into one bin group 25.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 91\n",
      "Length of fail 707\n",
      "G000771305 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium breve\n",
      "Coverage Percentage: 0.0824736423186343%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 91\n",
      "Length of fail 708\n",
      "G000771445 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium pseudocatenulatum\n",
      "Coverage Percentage: 0.1977330691059577%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 91\n",
      "Length of fail 709\n",
      "G000771525 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium mongoliense\n",
      "Coverage Percentage: 0.0070065248842995%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 91\n",
      "Length of fail 710\n",
      "G000771545 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium catenulatum\n",
      "Coverage Percentage: 0.1089558434184519%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 91\n",
      "Length of fail 711\n",
      "G000771705 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium adolescentis\n",
      "Coverage Percentage: 0.4714405345522427%\n",
      "Adj, read counts 62\n",
      "Number of samples per group [1, 1, 1, 2, 2, 1, 1, 2, 1, 3, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 3, 3, 1, 2, 3, 1, 1, 2, 2, 1, 2, 1, 1, 2, 1]\n",
      "Number of bins 54\n",
      "total # samples 29\n",
      "Number of total reads 62\n",
      "Number of hits in BIOM tables 20.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.129032258064516 %\n",
      "Percentage of reads from samples who only fall into one bin group 27.419354838709676 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 92\n",
      "Length of fail 711\n",
      "G000771745 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Thermales; f__Thermaceae; g__Thermus; s__Thermus filiformis\n",
      "Coverage Percentage: 0.0191464856981617%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 92\n",
      "Length of fail 712\n",
      "G000783715 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas caviae\n",
      "Coverage Percentage: 0.0446940619021538%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 92\n",
      "Length of fail 713\n",
      "G000783935 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Raoultella; s__Raoultella planticola\n",
      "Coverage Percentage: 0.0525586408311442%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 92\n",
      "Length of fail 714\n",
      "G000785495 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Pimelobacter; s__Pimelobacter simplex\n",
      "Coverage Percentage: 0.0053571175160003%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 92\n",
      "Length of fail 715\n",
      "G000785515 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus salivarius\n",
      "Coverage Percentage: 0.0736892069753024%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 92\n",
      "Length of fail 716\n",
      "G000786655 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Kocuria; s__Kocuria marina\n",
      "Coverage Percentage: 0.0187783570306853%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 92\n",
      "Length of fail 717\n",
      "G000787535 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Saccharomonospora; s__Saccharomonospora viridis\n",
      "Coverage Percentage: 0.0028574163195176%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 92\n",
      "Length of fail 718\n",
      "G000799205 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Enterobacter; s__Enterobacter asburiae\n",
      "Coverage Percentage: 0.0094261424981958%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 92\n",
      "Length of fail 719\n",
      "G000800395 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Deinococcales; f__Deinococcaceae; g__Deinococcus; s__Deinococcus radiopugnans\n",
      "Coverage Percentage: 0.0118648382545038%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 92\n",
      "Length of fail 720\n",
      "G000803625 d__Bacteria; p__Candidatus Saccharibacteria; c__Candidatus Saccharimonia; o__Candidatus Nanosynbacterales; f__Candidatus Nanosynbacteraceae; g__Candidatus Nanosynbacter; s__Candidatus Nanosynbacter lyticus\n",
      "Coverage Percentage: 0.0775734678885551%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 92\n",
      "Length of fail 721\n",
      "G000803645 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingopyxis; s__Sphingopyxis fribergensis\n",
      "Coverage Percentage: 0.0263350526085926%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 92\n",
      "Length of fail 722\n",
      "G000804445 d__Eukaryota; p__Ascomycota; c__Sordariomycetes; o__Hypocreales; f__Clavicipitaceae; g__Metarhizium; s__Metarhizium album\n",
      "Coverage Percentage: 0.0098508628617052%\n",
      "Adj, read counts 55\n",
      "Number of samples per group [1, 1, 4, 1, 1, 32, 3, 3, 1, 5, 2]\n",
      "Number of bins 13\n",
      "total # samples 39\n",
      "Number of total reads 55\n",
      "Number of hits in BIOM tables 21.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 74.54545454545455 %\n",
      "Percentage of reads from samples who only fall into one bin group 56.36363636363636 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 93\n",
      "Length of fail 722\n",
      "G000805675 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium minutissimum\n",
      "Coverage Percentage: 0.0379859774689102%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 93\n",
      "Length of fail 723\n",
      "G000807215 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Tepidimonas; s__Tepidimonas taiwanensis\n",
      "Coverage Percentage: 0.0300807331367785%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 93\n",
      "Length of fail 724\n",
      "G000807695 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Escherichia; s__Escherichia marmotae\n",
      "Coverage Percentage: 0.1233452309024496%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 93\n",
      "Length of fail 725\n",
      "G000807925 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium subterraneum\n",
      "Coverage Percentage: 0.0710660646343095%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 93\n",
      "Length of fail 726\n",
      "G000813265 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas mendocina\n",
      "Coverage Percentage: 0.0400381494101982%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 93\n",
      "Length of fail 727\n",
      "G000813705 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Neisseriaceae; g__Morococcus; s__Morococcus cerebrosus\n",
      "Coverage Percentage: 0.0151509659351709%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 93\n",
      "Length of fail 728\n",
      "G000815465 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rickettsiales; f__Candidatus Midichloriaceae; g__Candidatus Jidaibacter; s__Candidatus Jidaibacter acanthamoeba\n",
      "Coverage Percentage: 0.006356256829819%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 93\n",
      "Length of fail 729\n",
      "G000816495 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter harbinensis\n",
      "Coverage Percentage: 0.1924559077342722%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [1, 1, 1, 5, 1, 2, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 6, 1, 1]\n",
      "Number of bins 25\n",
      "total # samples 19\n",
      "Number of total reads 37\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 37.83783783783784 %\n",
      "Percentage of reads from samples who only fall into one bin group 21.62162162162162 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 94\n",
      "Length of fail 729\n",
      "G000818015 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas balearica\n",
      "Coverage Percentage: 0.0340368839369633%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 94\n",
      "Length of fail 730\n",
      "G000819685 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas allosaccharophila\n",
      "Coverage Percentage: 0.0342889888841527%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 94\n",
      "Length of fail 731\n",
      "G000819725 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas australiensis\n",
      "Coverage Percentage: 0.0356250493947997%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 94\n",
      "Length of fail 732\n",
      "G000819765 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas bivalvium\n",
      "Coverage Percentage: 0.0223289540546336%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 94\n",
      "Length of fail 733\n",
      "G000819785 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas caviae\n",
      "Coverage Percentage: 0.0488867807194292%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 94\n",
      "Length of fail 734\n",
      "G000819805 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas diversa\n",
      "Coverage Percentage: 0.0068910745080203%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 94\n",
      "Length of fail 735\n",
      "G000819825 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas encheleia\n",
      "Coverage Percentage: 0.0253509394937457%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 94\n",
      "Length of fail 736\n",
      "G000819865 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas eucrenophila\n",
      "Coverage Percentage: 0.0307676789166957%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 94\n",
      "Length of fail 737\n",
      "G000819885 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas fluvialis\n",
      "Coverage Percentage: 0.0423480590558277%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 94\n",
      "Length of fail 738\n",
      "G000819955 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas jandaei\n",
      "Coverage Percentage: 0.0174608851735758%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 94\n",
      "Length of fail 739\n",
      "G000819985 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas media\n",
      "Coverage Percentage: 0.2698164259214458%\n",
      "Adj, read counts 70\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 4, 1, 2, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 64\n",
      "total # samples 28\n",
      "Number of total reads 70\n",
      "Number of hits in BIOM tables 43.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.857142857142856 %\n",
      "Percentage of reads from samples who only fall into one bin group 24.285714285714285 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 95\n",
      "Length of fail 739\n",
      "G000820025 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas popoffii\n",
      "Coverage Percentage: 0.0234859387648746%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 95\n",
      "Length of fail 740\n",
      "G000820045 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas rivuli\n",
      "Coverage Percentage: 0.025594295800838%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 95\n",
      "Length of fail 741\n",
      "G000820065 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas salmonicida\n",
      "Coverage Percentage: 0.0469986739584543%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 95\n",
      "Length of fail 742\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000820085 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas sanarellii\n",
      "Coverage Percentage: 0.0519713916229374%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 95\n",
      "Length of fail 743\n",
      "G000820105 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas schubertii\n",
      "Coverage Percentage: 0.011553959047634%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 95\n",
      "Length of fail 744\n",
      "G000820145 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas sobria\n",
      "Coverage Percentage: 0.1317078970731602%\n",
      "Adj, read counts 33\n",
      "Number of samples per group [2, 1, 1, 1, 1, 3, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 28\n",
      "total # samples 9\n",
      "Number of total reads 31\n",
      "Number of hits in BIOM tables 23.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 27.27272727272727 %\n",
      "Percentage of reads from samples who only fall into one bin group 25.806451612903224 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 96\n",
      "Length of fail 744\n",
      "G000820225 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas veronii\n",
      "Coverage Percentage: 0.0490986045193738%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 96\n",
      "Length of fail 745\n",
      "G000820265 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas caviae\n",
      "Coverage Percentage: 0.0797895074726068%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 96\n",
      "Length of fail 746\n",
      "G000820285 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas veronii\n",
      "Coverage Percentage: 0.0467952581105952%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 96\n",
      "Length of fail 747\n",
      "G000820305 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas dhakensis\n",
      "Coverage Percentage: 0.0263084139611963%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 96\n",
      "Length of fail 748\n",
      "G000820325 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas hydrophila\n",
      "Coverage Percentage: 0.0252952230024725%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 96\n",
      "Length of fail 749\n",
      "G000820365 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas veronii\n",
      "Coverage Percentage: 0.0364679199029395%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 96\n",
      "Length of fail 750\n",
      "G000820725 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces polynesiensis\n",
      "Coverage Percentage: 0.0052346782527545%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 96\n",
      "Length of fail 751\n",
      "G000826065 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Tessaracoccus; s__Tessaracoccus massiliensis\n",
      "Coverage Percentage: 0.0140388886554262%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 96\n",
      "Length of fail 752\n",
      "G000827555 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Capnocytophaga; s__Capnocytophaga canis\n",
      "Coverage Percentage: 0.0083327164539369%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 96\n",
      "Length of fail 753\n",
      "G000828075 d__Bacteria; p__Cyanobacteria; c__; o__Nostocales; f__Tolypothrichaceae; g__Tolypothrix; s__Tolypothrix campylonemoides\n",
      "Coverage Percentage: 0.0139498424491488%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 96\n",
      "Length of fail 754\n",
      "G000828635 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Sterolibacteriaceae; g__Sulfuritalea; s__Sulfuritalea hydrogenivorans\n",
      "Coverage Percentage: 0.0321880962950028%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 96\n",
      "Length of fail 755\n",
      "G000828895 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Serpentinimonas; s__Serpentinimonas raichei\n",
      "Coverage Percentage: 0.0213285636086531%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 96\n",
      "Length of fail 756\n",
      "G000829435 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Planococcaceae; g__Jeotgalibacillus; s__Jeotgalibacillus soli\n",
      "Coverage Percentage: 0.004632800278921%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 96\n",
      "Length of fail 757\n",
      "G000830155 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter indicus\n",
      "Coverage Percentage: 0.2196188038093655%\n",
      "Adj, read counts 43\n",
      "Number of samples per group [4, 6, 1, 1, 2, 7, 3, 7, 2, 1, 1, 1, 1, 1]\n",
      "Number of bins 22\n",
      "total # samples 19\n",
      "Number of total reads 42\n",
      "Number of hits in BIOM tables 8.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 53.48837209302326 %\n",
      "Percentage of reads from samples who only fall into one bin group 30.952380952380953 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 757\n",
      "G000832125 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium hibernum\n",
      "Coverage Percentage: 0.0653829529704488%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 758\n",
      "G000833575 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium singulare\n",
      "Coverage Percentage: 0.0576572706277541%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 759\n",
      "G000833995 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella pectinovora\n",
      "Coverage Percentage: 0.0261062315414528%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 760\n",
      "G000834015 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella pectinovora\n",
      "Coverage Percentage: 0.0437361850469413%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 761\n",
      "G000835455 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Chaetothyriales; f__Herpotrichiellaceae; g__Fonsecaea; s__Fonsecaea pedrosoi\n",
      "Coverage Percentage: 0.0003372819342571%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 762\n",
      "G000836015 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter pittii\n",
      "Coverage Percentage: 0.0415129163053198%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 763\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000836115 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Chaetothyriales; f__Herpotrichiellaceae; g__Exophiala; s__Exophiala spinifera\n",
      "Coverage Percentage: 0.001294325757912%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 764\n",
      "G000836275 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Chaetothyriales; f__Herpotrichiellaceae; g__Exophiala; s__Exophiala mesophila\n",
      "Coverage Percentage: 0.0012024369981939%\n",
      "Adj, read counts 47\n",
      "Number of samples per group [45, 1, 1]\n",
      "Number of bins 3\n",
      "total # samples 46\n",
      "Number of total reads 47\n",
      "Number of hits in BIOM tables 29.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 100.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 95.74468085106383 %\n",
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 765\n",
      "G000837125 d__Viruses; p__; c__Naldaviricetes; o__Lefavirales; f__Baculoviridae; g__Alphabaculovirus; s__Orgyia pseudotsugata multiple nucleopolyhedrovirus\n",
      "Coverage Percentage: 0.1636425622182658%\n",
      "Adj, read counts 41\n",
      "Number of samples per group [41]\n",
      "Number of bins 1\n",
      "total # samples 41\n",
      "Number of total reads 41\n",
      "Number of hits in BIOM tables 15.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 100.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 100.0 %\n",
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 766\n",
      "G000838645 d__Viruses; p__Cossaviricota; c__Quintoviricetes; o__Piccovirales; f__Parvoviridae; g__Dependoparvovirus; s__Adeno-associated dependoparvovirus A\n",
      "Coverage Percentage: 3.2271852960034195%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 767\n",
      "G000839785 d__Viruses; p__; c__Naldaviricetes; o__Lefavirales; f__Baculoviridae; g__Betabaculovirus; s__Cydia pomonella granulovirus\n",
      "Coverage Percentage: 0.1651821862348178%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 768\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000845685 d__Viruses; p__Peploviricota; c__Herviviricetes; o__Herpesvirales; f__Herpesviridae; g__Roseolovirus; s__Human betaherpesvirus 6A\n",
      "Coverage Percentage: 3.597108760305688%\n",
      "Adj, read counts 39\n",
      "Number of samples per group [8, 1]\n",
      "Number of bins 13\n",
      "total # samples 8\n",
      "Number of total reads 39\n",
      "Number of hits in BIOM tables 24.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 100.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 51.28205128205128 %\n",
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 769\n",
      "G000846365 d__Viruses; p__Peploviricota; c__Herviviricetes; o__Herpesvirales; f__Herpesviridae; g__Roseolovirus; s__Human betaherpesvirus 6B\n",
      "Coverage Percentage: 9.63026018727562%\n",
      "Adj, read counts 119\n",
      "Number of samples per group [14]\n",
      "Number of bins 16\n",
      "total # samples 14\n",
      "Number of total reads 116\n",
      "Number of hits in BIOM tables 112.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 100.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 100.0 %\n",
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 770\n",
      "G000848125 d__Viruses; p__Peploviricota; c__Herviviricetes; o__Herpesvirales; f__Herpesviridae; g__Roseolovirus; s__Human betaherpesvirus 7\n",
      "Coverage Percentage: 0.097334726940162%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 771\n",
      "G000871465 d__Viruses; p__Peploviricota; c__Herviviricetes; o__Herpesvirales; f__Alloherpesviridae; g__Cyprinivirus; s__Cyprinid herpesvirus 3\n",
      "Coverage Percentage: 0.0775887188035751%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [19, 18]\n",
      "Number of bins 2\n",
      "total # samples 22\n",
      "Number of total reads 37\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 100.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.91891891891892 %\n",
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 772\n",
      "G000875975 d__Bacteria; p__Chlamydiae; c__Chlamydiia; o__Parachlamydiales; f__Parachlamydiaceae; g__Parachlamydia; s__Parachlamydia acanthamoebae\n",
      "Coverage Percentage: 0.0103306439223655%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 773\n",
      "G000878195 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacterium; s__Mycolicibacterium llatzerense\n",
      "Coverage Percentage: 0.0024779971087845%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 774\n",
      "G000891975 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus PAD20\n",
      "Coverage Percentage: 2.534910916970489%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 775\n",
      "G000897835 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus P1001\n",
      "Coverage Percentage: 2.2423341888423614%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 776\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000898815 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus P100D\n",
      "Coverage Percentage: 3.3721954856639327%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 777\n",
      "G000899375 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus P91\n",
      "Coverage Percentage: 4.241117272540563%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 778\n",
      "G000899395 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus P105\n",
      "Coverage Percentage: 5.0818437093349775%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 779\n",
      "G000899415 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus P1.1\n",
      "Coverage Percentage: 4.490936350006815%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 780\n",
      "G000900255 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus P101A\n",
      "Coverage Percentage: 4.226685602218164%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 781\n",
      "G000903115 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Schitoviridae; g__Gamaleyavirus; s__Escherichia virus IME11\n",
      "Coverage Percentage: 0.2080749621055532%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 782\n",
      "G000903355 d__Viruses; p__Peploviricota; c__Herviviricetes; o__Herpesvirales; f__Alloherpesviridae; g__Cyprinivirus; s__Cyprinid herpesvirus 1\n",
      "Coverage Percentage: 0.117467644876762%\n",
      "Adj, read counts 57\n",
      "Number of samples per group [27, 2, 28]\n",
      "Number of bins 3\n",
      "total # samples 34\n",
      "Number of total reads 57\n",
      "Number of hits in BIOM tables 21.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 99.99999999999999 %\n",
      "Percentage of reads from samples who only fall into one bin group 19.298245614035086 %\n",
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 783\n",
      "G000912035 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus PHL113M01\n",
      "Coverage Percentage: 4.517123287671232%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 784\n",
      "G000912675 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus PHL112N00\n",
      "Coverage Percentage: 3.522859290644434%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 785\n",
      "G000934545 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Rhodocyclaceae; g__Rugosibacter; s__Rugosibacter aromaticivorans\n",
      "Coverage Percentage: 0.0205819442508734%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 786\n",
      "G000935025 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas hengshuiensis\n",
      "Coverage Percentage: 0.0086642035846851%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 787\n",
      "G000936425 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Azospirillaceae; g__Skermanella; s__Skermanella aerolata\n",
      "Coverage Percentage: 0.0033291074768354%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 788\n",
      "G000938265 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Agreia; s__Agreia bicolorata\n",
      "Coverage Percentage: 0.0154038446874205%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 789\n",
      "G000938715 d__Eukaryota; p__Ascomycota; c__Saccharomycetes; o__Saccharomycetales; f__Saccharomycetaceae; g__Lachancea; s__Lachancea lanzarotensis\n",
      "Coverage Percentage: 0.0088978309936785%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 790\n",
      "G000939975 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas saudimassiliensis\n",
      "Coverage Percentage: 0.0121103350197099%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 791\n",
      "G000940845 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Propionibacterium; s__Propionibacterium freudenreichii\n",
      "Coverage Percentage: 0.0266498966089705%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 792\n",
      "G000949455 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Oscillospiraceae; g__Ruthenibacterium; s__Ruthenibacterium lactatiformans\n",
      "Coverage Percentage: 0.0256976411170037%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 97\n",
      "Length of fail 793\n",
      "G000949655 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter johnsonii\n",
      "Coverage Percentage: 0.7714830642963997%\n",
      "Adj, read counts 171\n",
      "Number of samples per group [9, 1, 1, 2, 1, 2, 8, 4, 1, 5, 1, 2, 1, 2, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 4, 1, 1, 1, 1, 2, 1, 2, 1, 2, 3, 3, 1, 1, 1, 2, 1, 4, 2, 1, 1, 3, 1, 4, 1, 7, 3, 1, 1, 1, 2, 1, 3, 2, 6, 2, 1, 1, 2, 1, 1, 1, 1, 4, 2, 3, 1, 2]\n",
      "Number of bins 121\n",
      "total # samples 46\n",
      "Number of total reads 168\n",
      "Number of hits in BIOM tables 69.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.789473684210526 %\n",
      "Percentage of reads from samples who only fall into one bin group 7.738095238095238 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 98\n",
      "Length of fail 793\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000952155 d__Bacteria; p__Cyanobacteria; c__; o__Chroococcidiopsidales; f__Chroococcidiopsidaceae; g__Aliterella; s__Aliterella atlantica\n",
      "Coverage Percentage: 0.0048225867177987%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 98\n",
      "Length of fail 794\n",
      "G000953015 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Methylophilaceae; g__Candidatus Methylopumilus; s__Candidatus Methylopumilus turicensis\n",
      "Coverage Percentage: 0.1190891333729917%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 98\n",
      "Length of fail 795\n",
      "G000954135 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Ralstonia; s__Ralstonia mannitolilytica\n",
      "Coverage Percentage: 0.1091803567622918%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 98\n",
      "Length of fail 796\n",
      "G000956415 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium foliorum\n",
      "Coverage Percentage: 0.0377611037031168%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 98\n",
      "Length of fail 797\n",
      "G000956465 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium trichothecenolyticum\n",
      "Coverage Percentage: 0.0165521935523791%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 98\n",
      "Length of fail 798\n",
      "G000956535 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium ginsengisoli\n",
      "Coverage Percentage: 0.0386349466539282%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 98\n",
      "Length of fail 799\n",
      "G000956575 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium ketosireducens\n",
      "Coverage Percentage: 0.0153939317325003%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 98\n",
      "Length of fail 800\n",
      "G000956595 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Priestia; s__Priestia aryabhattai\n",
      "Coverage Percentage: 0.0044340651568217%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 98\n",
      "Length of fail 801\n",
      "G000959525 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia multivorans\n",
      "Coverage Percentage: 0.0181723689525471%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 98\n",
      "Length of fail 802\n",
      "G000959545 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia ambifaria\n",
      "Coverage Percentage: 0.0254096424877912%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 98\n",
      "Length of fail 803\n",
      "G000960005 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus mitis\n",
      "Coverage Percentage: 0.109490308776246%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 98\n",
      "Length of fail 804\n",
      "G000962825 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes rectilineatus\n",
      "Coverage Percentage: 0.0078535684252074%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 98\n",
      "Length of fail 805\n",
      "G000963705 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Rhodospirillaceae; g__Elstera; s__Elstera litoralis\n",
      "Coverage Percentage: 0.0110584522268374%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 98\n",
      "Length of fail 806\n",
      "G000964545 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Comamonas; s__Comamonas thiooxydans\n",
      "Coverage Percentage: 0.018033814737058%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 98\n",
      "Length of fail 807\n",
      "G000967915 d__Bacteria; p__Tenericutes; c__Mollicutes; o__Acholeplasmatales; f__Acholeplasmataceae; g__Paracholeplasma; s__Paracholeplasma brassicae\n",
      "Coverage Percentage: 0.0183193878768255%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 98\n",
      "Length of fail 808\n",
      "G000967935 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas marginalis\n",
      "Coverage Percentage: 0.1037140401871701%\n",
      "Adj, read counts 42\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 39\n",
      "total # samples 22\n",
      "Number of total reads 41\n",
      "Number of hits in BIOM tables 18.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.285714285714285 %\n",
      "Percentage of reads from samples who only fall into one bin group 29.268292682926827 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 99\n",
      "Length of fail 808\n",
      "G000969605 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Aquincola; s__Aquincola tertiaricarbonis\n",
      "Coverage Percentage: 0.0568381714592334%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 99\n",
      "Length of fail 809\n",
      "G000970265 d__Archaea; p__Euryarchaeota; c__Methanomicrobia; o__Methanosarcinales; f__Methanosarcinaceae; g__Methanosarcina; s__Methanosarcina lacustris\n",
      "Coverage Percentage: 0.0222957199947437%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 99\n",
      "Length of fail 810\n",
      "G000970465 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Devosiaceae; g__Devosia; s__Devosia insulae\n",
      "Coverage Percentage: 0.0104892087055569%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 99\n",
      "Length of fail 811\n",
      "G000971805 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycobacteroides; s__Mycobacteroides chelonae\n",
      "Coverage Percentage: 0.0123927450787837%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 99\n",
      "Length of fail 812\n",
      "G000972175 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas syringae pv. coryli\n",
      "Coverage Percentage: 0.0421110077649549%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 99\n",
      "Length of fail 813\n",
      "G000973625 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Polynucleobacter; s__Polynucleobacter duraquae\n",
      "Coverage Percentage: 0.3039048666233692%\n",
      "Adj, read counts 40\n",
      "Number of samples per group [11, 1, 2, 1, 1, 1, 2, 1, 1, 4, 1, 1, 1, 2, 1, 1, 1, 1, 1]\n",
      "Number of bins 27\n",
      "total # samples 24\n",
      "Number of total reads 39\n",
      "Number of hits in BIOM tables 46.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 52.5 %\n",
      "Percentage of reads from samples who only fall into one bin group 38.46153846153847 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 813\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G000974685 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Gallionellaceae; g__Ferriphaselus; s__Ferriphaselus amnicola\n",
      "Coverage Percentage: 0.0339316266681976%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 814\n",
      "G000986865 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Chromatiales; f__Chromatiaceae; g__Pararheinheimera; s__Pararheinheimera mesophila\n",
      "Coverage Percentage: 0.0155386031675295%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 815\n",
      "G000987075 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia contaminans\n",
      "Coverage Percentage: 0.0153720334350363%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 816\n",
      "G001005725 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium chungbukense\n",
      "Coverage Percentage: 0.045602222111459%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 817\n",
      "G001006005 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia fonticola\n",
      "Coverage Percentage: 0.025181188735426%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 818\n",
      "G001006485 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Tannerellaceae; g__Tannerella; s__Tannerella forsythia\n",
      "Coverage Percentage: 0.0200331978707572%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 819\n",
      "G001010405 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Rhodanobacteraceae; g__Luteibacter; s__Luteibacter rhizovicinus\n",
      "Coverage Percentage: 0.0113285921077474%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 820\n",
      "G001013475 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Xanthomonas; s__Xanthomonas arboricola\n",
      "Coverage Percentage: 0.0086732196348318%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 821\n",
      "G001021565 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Delftia; s__Delftia tsuruhatensis\n",
      "Coverage Percentage: 0.0284225469041729%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 822\n",
      "G001025175 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium breve\n",
      "Coverage Percentage: 0.0820475761374627%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 823\n",
      "G001025195 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium catenulatum\n",
      "Coverage Percentage: 0.2121157475865883%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 824\n",
      "G001025215 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium pseudocatenulatum\n",
      "Coverage Percentage: 0.1997188981360146%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 825\n",
      "G001028645 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus capitis\n",
      "Coverage Percentage: 0.1230782751464575%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 826\n",
      "G001028705 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Croceicoccus; s__Croceicoccus naphthovorans\n",
      "Coverage Percentage: 0.0422921408571586%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 827\n",
      "G001039365 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Enterobacter; s__Enterobacter quasiroggenkampii\n",
      "Coverage Percentage: 0.0230890628357404%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 828\n",
      "G001041315 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus PHL009M11\n",
      "Coverage Percentage: 3.213232552621767%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 829\n",
      "G001041615 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus PHL132N00\n",
      "Coverage Percentage: 4.392649036306588%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 830\n",
      "G001041635 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus PHL095N00\n",
      "Coverage Percentage: 2.678901549527747%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 831\n",
      "G001042485 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__Bacillus paralicheniformis\n",
      "Coverage Percentage: 0.0128941792622671%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 832\n",
      "G001042615 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium catenulatum\n",
      "Coverage Percentage: 0.1137241713923381%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 833\n",
      "G001042675 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Parascardovia; s__Parascardovia denticolens\n",
      "Coverage Percentage: 0.0072453919043058%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 834\n",
      "G001042895 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas deceptionensis\n",
      "Coverage Percentage: 0.0600014287939178%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 835\n",
      "G001042905 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas lini\n",
      "Coverage Percentage: 0.0161791772295706%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 836\n",
      "G001042915 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas taetrolens\n",
      "Coverage Percentage: 0.0455065899230532%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 100\n",
      "Length of fail 837\n",
      "G001042925 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas aeruginosa\n",
      "Coverage Percentage: 0.5568992293403466%\n",
      "Adj, read counts 207\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 3, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 3, 1, 1, 2, 1, 1, 1, 1, 1, 3, 2, 2, 2, 1, 1, 2, 3, 1, 1, 1, 1, 2, 2, 1, 2, 1, 1, 1, 2, 1, 3, 1, 3, 1, 5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 4, 1, 1, 1, 2, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 3, 1, 4, 3, 2, 1, 2, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2]\n",
      "Number of bins 169\n",
      "total # samples 52\n",
      "Number of total reads 204\n",
      "Number of hits in BIOM tables 54.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 6.763285024154589 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.27450980392157 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 101\n",
      "Length of fail 837\n",
      "G001042985 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas lundensis\n",
      "Coverage Percentage: 0.0397651573126442%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 101\n",
      "Length of fail 838\n",
      "G001043005 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas psychrophila\n",
      "Coverage Percentage: 0.0738122855574148%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 101\n",
      "Length of fail 839\n",
      "G001043025 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas helleri\n",
      "Coverage Percentage: 0.0213845077168303%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 101\n",
      "Length of fail 840\n",
      "G001043885 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium platani\n",
      "Coverage Percentage: 0.0044296620663266%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 101\n",
      "Length of fail 841\n",
      "G001045685 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas aeruginosa\n",
      "Coverage Percentage: 0.5609422119501983%\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Adj, read counts 209\n",
      "Number of samples per group [3, 1, 3, 1, 2, 1, 1, 1, 1, 1, 1, 2, 3, 1, 1, 1, 1, 2, 1, 1, 1, 3, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 2, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 3, 2, 1, 1, 2, 2, 2, 3, 1, 1, 1, 1, 1, 2, 1, 1, 3, 1, 1, 2, 1, 1, 1, 1, 1, 3, 1, 2, 2, 2, 1, 1, 2, 1, 2, 1, 3, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 5, 1]\n",
      "Number of bins 165\n",
      "total # samples 51\n",
      "Number of total reads 206\n",
      "Number of hits in BIOM tables 54.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 6.220095693779904 %\n",
      "Percentage of reads from samples who only fall into one bin group 10.679611650485436 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 102\n",
      "Length of fail 841\n",
      "G001046635 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium barchaimii\n",
      "Coverage Percentage: 0.2066304668154857%\n",
      "Adj, read counts 69\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 5, 1, 1, 1, 3, 1, 1, 1, 2, 1, 1, 1, 2, 1, 4, 2, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 4, 3, 4, 3]\n",
      "Number of bins 50\n",
      "total # samples 28\n",
      "Number of total reads 69\n",
      "Number of hits in BIOM tables 57.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 21.73913043478261 %\n",
      "Percentage of reads from samples who only fall into one bin group 23.18840579710145 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 103\n",
      "Length of fail 841\n",
      "G001046645 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium cupriresistens\n",
      "Coverage Percentage: 0.1009566290235812%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 103\n",
      "Length of fail 842\n",
      "G001046655 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium baderi\n",
      "Coverage Percentage: 0.0899079593897252%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 103\n",
      "Length of fail 843\n",
      "G001046855 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Intrasporangiaceae; g__Tetrasphaera; s__Tetrasphaera japonica\n",
      "Coverage Percentage: 0.0085039268189637%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 103\n",
      "Length of fail 844\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G001050255 d__Bacteria; p__Chloroflexi; c__Anaerolineae; o__Anaerolineales; f__Anaerolineaceae; g__Levilinea; s__Levilinea saccharolytica\n",
      "Coverage Percentage: 0.0035324773358387%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 103\n",
      "Length of fail 845\n",
      "G001071155 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus parasanguinis\n",
      "Coverage Percentage: 0.0452337547565978%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 103\n",
      "Length of fail 846\n",
      "G001182725 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Anaerococcus; s__Anaerococcus jeddahensis\n",
      "Coverage Percentage: 0.0485917838385302%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 103\n",
      "Length of fail 847\n",
      "G001183605 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Lysinibacillus; s__Lysinibacillus xylanilyticus\n",
      "Coverage Percentage: 0.0125433871932974%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 103\n",
      "Length of fail 848\n",
      "G001186195 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas fulva\n",
      "Coverage Percentage: 0.1363986841354893%\n",
      "Adj, read counts 43\n",
      "Number of samples per group [1, 2, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 7, 1, 1, 1, 1]\n",
      "Number of bins 27\n",
      "total # samples 11\n",
      "Number of total reads 41\n",
      "Number of hits in BIOM tables 35.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 34.883720930232556 %\n",
      "Percentage of reads from samples who only fall into one bin group 26.82926829268293 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 104\n",
      "Length of fail 848\n",
      "G001189245 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium pachyrhizi\n",
      "Coverage Percentage: 0.0086064119145789%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 104\n",
      "Length of fail 849\n",
      "G001189915 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Herbaspirillum; s__Herbaspirillum autotrophicum\n",
      "Coverage Percentage: 0.0098306745323152%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 104\n",
      "Length of fail 850\n",
      "G001189965 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Herbaspirillum; s__Herbaspirillum rhizosphaerae\n",
      "Coverage Percentage: 0.0163654655262987%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 104\n",
      "Length of fail 851\n",
      "G001190275 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus Pirate\n",
      "Coverage Percentage: 2.0529259309780383%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 104\n",
      "Length of fail 852\n",
      "G001190715 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus Attacne\n",
      "Coverage Percentage: 2.0709239506856902%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 104\n",
      "Length of fail 853\n",
      "G001190735 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus SKKY\n",
      "Coverage Percentage: 2.831655065215922%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 104\n",
      "Length of fail 854\n",
      "G001202035 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Pseudorhodobacter; s__Pseudorhodobacter wandonensis\n",
      "Coverage Percentage: 0.0107758592976248%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 104\n",
      "Length of fail 855\n",
      "G001244495 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Clostridium; s__Clostridium phoceensis\n",
      "Coverage Percentage: 0.0187610851087361%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 104\n",
      "Length of fail 856\n",
      "G001262015 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella fusca\n",
      "Coverage Percentage: 0.0047825972323477%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 104\n",
      "Length of fail 857\n",
      "G001263755 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium riegelii\n",
      "Coverage Percentage: 0.0159143558020893%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 104\n",
      "Length of fail 858\n",
      "G001267925 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Herbaspirillum; s__Herbaspirillum hiltneri\n",
      "Coverage Percentage: 0.0305106823638589%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 104\n",
      "Length of fail 859\n",
      "G001269845 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas fluorescens\n",
      "Coverage Percentage: 0.0338891784730929%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 104\n",
      "Length of fail 860\n",
      "G001270065 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Limnohabitans; s__Limnohabitans planktonicus\n",
      "Coverage Percentage: 0.1843690859529868%\n",
      "Adj, read counts 63\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 2, 1, 1, 3, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 3, 2, 1, 2, 1, 2, 2, 1, 1, 2, 10]\n",
      "Number of bins 43\n",
      "total # samples 30\n",
      "Number of total reads 63\n",
      "Number of hits in BIOM tables 64.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 30.158730158730158 %\n",
      "Percentage of reads from samples who only fall into one bin group 25.396825396825395 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 105\n",
      "Length of fail 860\n",
      "G001275135 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella rara\n",
      "Coverage Percentage: 0.0118733195954628%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 105\n",
      "Length of fail 861\n",
      "G001275345 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Kocuria; s__Kocuria palustris\n",
      "Coverage Percentage: 0.3994118650652823%\n",
      "Adj, read counts 67\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 3, 1, 1, 2, 3, 1, 1, 2, 1, 1, 1, 2, 1, 1, 3, 1, 1, 1, 2, 1, 3, 2, 1]\n",
      "Number of bins 53\n",
      "total # samples 32\n",
      "Number of total reads 66\n",
      "Number of hits in BIOM tables 57.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 13.432835820895523 %\n",
      "Percentage of reads from samples who only fall into one bin group 30.303030303030305 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 106\n",
      "Length of fail 861\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G001278075 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces pristinaespiralis\n",
      "Coverage Percentage: 0.0011954163517955%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 106\n",
      "Length of fail 862\n",
      "G001278385 d__Eukaryota; p__Basidiomycota; c__Malasseziomycetes; o__Malasseziales; f__Malasseziaceae; g__Malassezia; s__Malassezia pachydermatis\n",
      "Coverage Percentage: 0.0044894490587367%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 106\n",
      "Length of fail 863\n",
      "G001280205 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Clavibacter; s__Clavibacter michiganensis\n",
      "Coverage Percentage: 0.0450065134512098%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 106\n",
      "Length of fail 864\n",
      "G001280255 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Thermales; f__Thermaceae; g__Thermus; s__Thermus aquaticus\n",
      "Coverage Percentage: 0.0067083234300412%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 106\n",
      "Length of fail 865\n",
      "G001280455 d__Archaea; p__Euryarchaeota; c__Halobacteria; o__Haloferacales; f__Halorubraceae; g__Halorubrum; s__Halorubrum tropicale\n",
      "Coverage Percentage: 0.0042328481349398%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 106\n",
      "Length of fail 866\n",
      "G001280875 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Sellimonas; s__Sellimonas intestinalis\n",
      "Coverage Percentage: 0.006959919641144%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 106\n",
      "Length of fail 867\n",
      "G001293125 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Lawsonellaceae; g__Lawsonella; s__Lawsonella clevelandensis\n",
      "Coverage Percentage: 0.0281099523743235%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 106\n",
      "Length of fail 868\n",
      "G001293525 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Ideonella; s__Ideonella sakaiensis\n",
      "Coverage Percentage: 0.0952724464958823%\n",
      "Adj, read counts 41\n",
      "Number of samples per group [1, 3, 3, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 8, 2, 6, 1]\n",
      "Number of bins 24\n",
      "total # samples 26\n",
      "Number of total reads 40\n",
      "Number of hits in BIOM tables 12.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 46.34146341463415 %\n",
      "Percentage of reads from samples who only fall into one bin group 42.5 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 107\n",
      "Length of fail 868\n",
      "G001294535 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Cypionkella; s__Cypionkella psychrotolerans\n",
      "Coverage Percentage: 0.0287349591678198%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 107\n",
      "Length of fail 869\n",
      "G001294575 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas versuta\n",
      "Coverage Percentage: 0.0426425320809322%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 107\n",
      "Length of fail 870\n",
      "G001298445 d__Bacteria; p__Cyanobacteria; c__; o__Nostocales; f__Nostocaceae; g__Nostoc; s__Nostoc piscinale\n",
      "Coverage Percentage: 0.0057649837424639%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 107\n",
      "Length of fail 871\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G001298675 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax caeni\n",
      "Coverage Percentage: 0.1361962523140487%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [1, 1, 4, 1, 2, 1, 1, 1, 1, 1, 4, 1, 2, 1, 9]\n",
      "Number of bins 22\n",
      "total # samples 22\n",
      "Number of total reads 37\n",
      "Number of hits in BIOM tables 14.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 59.45945945945946 %\n",
      "Percentage of reads from samples who only fall into one bin group 54.054054054054056 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 108\n",
      "Length of fail 871\n",
      "G001302325 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Chromobacteriaceae; g__Gulbenkiania; s__Gulbenkiania mobilis\n",
      "Coverage Percentage: 0.0523521467002124%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 108\n",
      "Length of fail 872\n",
      "G001305515 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Hyphomicrobiaceae; g__Prosthecomicrobium; s__Prosthecomicrobium hirschii\n",
      "Coverage Percentage: 0.0019583403944097%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 108\n",
      "Length of fail 873\n",
      "G001305955 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas viridiflava\n",
      "Coverage Percentage: 0.0283740583031975%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 108\n",
      "Length of fail 874\n",
      "G001306035 d__Bacteria; p__Chloroflexi; c__Anaerolineae; o__Anaerolineales; f__Anaerolineaceae; g__Levilinea; s__Levilinea saccharolytica\n",
      "Coverage Percentage: 0.0035213281409892%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 108\n",
      "Length of fail 875\n",
      "G001314325 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingopyxis; s__Sphingopyxis macrogoltabida\n",
      "Coverage Percentage: 0.0302157214642778%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 108\n",
      "Length of fail 876\n",
      "G001314995 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides ovatus\n",
      "Coverage Percentage: 0.1262111067318924%\n",
      "Adj, read counts 50\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 3, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 2, 1, 1, 1, 1, 1]\n",
      "Number of bins 45\n",
      "total # samples 31\n",
      "Number of total reads 49\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 40.816326530612244 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 876\n",
      "G001315785 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola coprophilus\n",
      "Coverage Percentage: 0.0680227921461641%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 877\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G001375595 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Phenylobacterium; s__Phenylobacterium immobile\n",
      "Coverage Percentage: 0.0582214518110837%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 878\n",
      "G001375615 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Paenarthrobacter; s__Paenarthrobacter nitroguajacolicus\n",
      "Coverage Percentage: 0.008536350005196%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 879\n",
      "G001399515 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Roseobacteraceae; g__Ketogulonicigenium; s__Ketogulonicigenium vulgare\n",
      "Coverage Percentage: 0.0176106310832216%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 880\n",
      "G001399775 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Thermales; f__Thermaceae; g__Thermus; s__Thermus aquaticus\n",
      "Coverage Percentage: 0.0195405950517398%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 881\n",
      "G001400235 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas syringae pv. coryli\n",
      "Coverage Percentage: 0.0410597866560869%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 882\n",
      "G001400635 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas syringae group genomosp. 3\n",
      "Coverage Percentage: 0.024775583543793%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 883\n",
      "G001400955 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas syringae group genomosp. 3\n",
      "Coverage Percentage: 0.0284711388455538%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 884\n",
      "G001401075 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas syringae\n",
      "Coverage Percentage: 0.0332420548197688%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 885\n",
      "G001401215 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas viridiflava\n",
      "Coverage Percentage: 0.0335359655031554%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 886\n",
      "G001404615 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Selenomonadales; f__Selenomonadaceae; g__Mitsuokella; s__Mitsuokella jalaludinii\n",
      "Coverage Percentage: 0.0114771798190278%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 887\n",
      "G001404635 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Dorea; s__Dorea longicatena\n",
      "Coverage Percentage: 0.0860207965987179%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 888\n",
      "G001404935 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Blautia; s__Blautia hydrogenotrophica\n",
      "Coverage Percentage: 0.024745151833024%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 889\n",
      "G001405015 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Clostridium; s__Clostridium disporicum\n",
      "Coverage Percentage: 0.0150694760806842%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 890\n",
      "G001405615 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Roseburia; s__Roseburia faecis\n",
      "Coverage Percentage: 0.0880182526485339%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 891\n",
      "G001406295 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Eubacteriaceae; g__Eubacterium; s__Eubacterium ramulus\n",
      "Coverage Percentage: 0.079974466202174%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 892\n",
      "G001411495 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia cepacia\n",
      "Coverage Percentage: 0.0046943965289353%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 893\n",
      "G001414055 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Caulobacter; s__Caulobacter henricii\n",
      "Coverage Percentage: 0.0764661229300322%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 894\n",
      "G001418245 d__Bacteria; p__Proteobacteria; c__Hydrogenophilalia; o__Hydrogenophilales; f__Hydrogenophilaceae; g__Tepidiphilus; s__Tepidiphilus thermophilus\n",
      "Coverage Percentage: 0.0647202232781346%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 895\n",
      "G001418295 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Comamonas; s__Comamonas thiooxydans\n",
      "Coverage Percentage: 0.0237407892948966%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 896\n",
      "G001418475 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces ossamyceticus\n",
      "Coverage Percentage: 0.0016203848274461%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 897\n",
      "G001418575 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptacidiphilus; s__Streptacidiphilus griseoplanus\n",
      "Coverage Percentage: 0.001827767860893%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 898\n",
      "G001420285 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium aquaticum\n",
      "Coverage Percentage: 0.0081550711693847%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 899\n",
      "G001431525 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas koreensis\n",
      "Coverage Percentage: 0.0149453455641026%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 900\n",
      "G001431535 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas chelatiphaga\n",
      "Coverage Percentage: 0.0339740809746657%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 109\n",
      "Length of fail 901\n",
      "G001431565 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas pavanii\n",
      "Coverage Percentage: 0.3432523437583259%\n",
      "Adj, read counts 93\n",
      "Number of samples per group [1, 1, 2, 1, 1, 1, 1, 2, 2, 1, 1, 1, 2, 2, 1, 2, 1, 1, 2, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 2, 1, 2, 1, 1, 2, 2, 1, 1, 1, 2, 1, 1, 2, 3, 1, 2, 1, 1, 2, 2, 1, 1, 1, 2]\n",
      "Number of bins 76\n",
      "total # samples 30\n",
      "Number of total reads 91\n",
      "Number of hits in BIOM tables 28.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 7.526881720430108 %\n",
      "Percentage of reads from samples who only fall into one bin group 24.175824175824175 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 110\n",
      "Length of fail 901\n",
      "G001431595 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas acidaminiphila\n",
      "Coverage Percentage: 0.0383014534780558%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 110\n",
      "Length of fail 902\n",
      "G001431625 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas geniculata\n",
      "Coverage Percentage: 0.3054950041261771%\n",
      "Adj, read counts 94\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 2, 2, 3, 1, 1, 1, 1, 1, 1, 3, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 4, 2, 1, 1, 1, 2, 3, 1, 1, 2, 2, 2]\n",
      "Number of bins 69\n",
      "total # samples 24\n",
      "Number of total reads 91\n",
      "Number of hits in BIOM tables 28.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 13.82978723404255 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.483516483516482 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 111\n",
      "Length of fail 902\n",
      "G001431665 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas maltophilia\n",
      "Coverage Percentage: 0.2558852949200049%\n",
      "Adj, read counts 72\n",
      "Number of samples per group [1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2]\n",
      "Number of bins 62\n",
      "total # samples 21\n",
      "Number of total reads 70\n",
      "Number of hits in BIOM tables 24.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.11111111111111 %\n",
      "Percentage of reads from samples who only fall into one bin group 21.428571428571427 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 112\n",
      "Length of fail 902\n",
      "G001431675 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas maltophilia\n",
      "Coverage Percentage: 0.3624762656440797%\n",
      "Adj, read counts 102\n",
      "Number of samples per group [1, 2, 1, 1, 2, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2]\n",
      "Number of bins 79\n",
      "total # samples 30\n",
      "Number of total reads 101\n",
      "Number of hits in BIOM tables 35.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 8.823529411764707 %\n",
      "Percentage of reads from samples who only fall into one bin group 20.792079207920793 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 902\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G001433765 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Loigolactobacillus; s__Loigolactobacillus coryniformis\n",
      "Coverage Percentage: 0.0496121100798223%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 903\n",
      "G001434005 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactobacillus; s__Lactobacillus crispatus\n",
      "Coverage Percentage: 0.0980072823439808%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 904\n",
      "G001434065 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Latilactobacillus; s__Latilactobacillus sakei\n",
      "Coverage Percentage: 0.0220818357017235%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 905\n",
      "G001434175 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactiplantibacillus; s__Lactiplantibacillus plantarum\n",
      "Coverage Percentage: 0.0125239692259283%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 906\n",
      "G001434815 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactobacillus; s__Lactobacillus equicursoris\n",
      "Coverage Percentage: 0.0108451535780739%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 907\n",
      "G001435015 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactobacillus; s__Lactobacillus iners\n",
      "Coverage Percentage: 0.040747259017794%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 908\n",
      "G001435275 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactobacillus; s__Lactobacillus kefiranofaciens\n",
      "Coverage Percentage: 0.0129211601254768%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 909\n",
      "G001435495 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Latilactobacillus; s__Latilactobacillus curvatus\n",
      "Coverage Percentage: 0.0130161851044315%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 910\n",
      "G001435575 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lentilactobacillus; s__Lentilactobacillus sunkii\n",
      "Coverage Percentage: 0.0069395763295875%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 911\n",
      "G001435585 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Schleiferilactobacillus; s__Schleiferilactobacillus perolens\n",
      "Coverage Percentage: 0.0110534646420504%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 912\n",
      "G001435655 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactiplantibacillus; s__Lactiplantibacillus paraplantarum\n",
      "Coverage Percentage: 0.0065867377799106%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 913\n",
      "G001435715 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Latilactobacillus; s__Latilactobacillus sakei\n",
      "Coverage Percentage: 0.0178187261724872%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 914\n",
      "G001435835 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lapidilactobacillus; s__Lapidilactobacillus concavus\n",
      "Coverage Percentage: 0.0286314583938593%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 915\n",
      "G001436045 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Limosilactobacillus; s__Limosilactobacillus frumenti\n",
      "Coverage Percentage: 0.0174155949155689%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 916\n",
      "G001436455 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactobacillus; s__Lactobacillus jensenii\n",
      "Coverage Percentage: 0.0079995683953795%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 917\n",
      "G001436855 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactiplantibacillus; s__Lactiplantibacillus plantarum\n",
      "Coverage Percentage: 0.0116806815474287%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 918\n",
      "G001437035 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Carnobacterium; s__Carnobacterium maltaromaticum\n",
      "Coverage Percentage: 0.0450119617191438%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 919\n",
      "G001437315 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Carnobacterium; s__Carnobacterium maltaromaticum\n",
      "Coverage Percentage: 0.0601579972604154%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 920\n",
      "G001437565 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Lactococcus; s__Lactococcus lactis\n",
      "Coverage Percentage: 0.0286689972840131%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 921\n",
      "G001438765 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Carnobacterium; s__Carnobacterium maltaromaticum\n",
      "Coverage Percentage: 0.0455219544469436%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 922\n",
      "G001438845 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactiplantibacillus; s__Lactiplantibacillus xiangfangensis\n",
      "Coverage Percentage: 0.0050474358691517%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 923\n",
      "G001439685 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas libanensis\n",
      "Coverage Percentage: 0.0403213219987404%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 113\n",
      "Length of fail 924\n",
      "G001439695 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas veronii\n",
      "Coverage Percentage: 0.1903211644649442%\n",
      "Adj, read counts 81\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 2, 2, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 2, 1]\n",
      "Number of bins 75\n",
      "total # samples 34\n",
      "Number of total reads 81\n",
      "Number of hits in BIOM tables 58.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 8.641975308641975 %\n",
      "Percentage of reads from samples who only fall into one bin group 20.98765432098765 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 114\n",
      "Length of fail 924\n",
      "G001439725 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas synxantha\n",
      "Coverage Percentage: 0.0408728630241002%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 114\n",
      "Length of fail 925\n",
      "G001439735 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas paralactis\n",
      "Coverage Percentage: 0.0444153956527434%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 114\n",
      "Length of fail 926\n",
      "G001439785 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas poae\n",
      "Coverage Percentage: 0.0243840561917405%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 114\n",
      "Length of fail 927\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G001439815 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas orientalis\n",
      "Coverage Percentage: 0.0498271479434911%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 114\n",
      "Length of fail 928\n",
      "G001439845 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas lactis\n",
      "Coverage Percentage: 0.0902446525551176%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 35\n",
      "total # samples 20\n",
      "Number of total reads 37\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 13.513513513513514 %\n",
      "Percentage of reads from samples who only fall into one bin group 27.027027027027028 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 115\n",
      "Length of fail 928\n",
      "G001440385 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium valentinum\n",
      "Coverage Percentage: 0.0194726517424417%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 115\n",
      "Length of fail 929\n",
      "G001440405 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium valentinum\n",
      "Coverage Percentage: 0.0114123287952942%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 115\n",
      "Length of fail 930\n",
      "G001440415 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium retamae\n",
      "Coverage Percentage: 0.0089010560264812%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 115\n",
      "Length of fail 931\n",
      "G001440475 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium lablabi\n",
      "Coverage Percentage: 0.0080952647123216%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 115\n",
      "Length of fail 932\n",
      "G001441535 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Tsuneonella; s__Tsuneonella troitsensis\n",
      "Coverage Percentage: 0.0678053229245731%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 115\n",
      "Length of fail 933\n",
      "G001442625 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter ginsenosidimutans\n",
      "Coverage Percentage: 0.014372812345218%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 115\n",
      "Length of fail 934\n",
      "G001443605 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Salinivirgaceae; g__Salinivirga; s__Salinivirga cyanobacteriivorans\n",
      "Coverage Percentage: 0.0031377366653464%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 115\n",
      "Length of fail 935\n",
      "G001447195 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Aquabacterium; s__Aquabacterium parvum\n",
      "Coverage Percentage: 0.0667917528253497%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 115\n",
      "Length of fail 936\n",
      "G001456965 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Rhodococcus; s__Rhodococcus qingshengii\n",
      "Coverage Percentage: 0.1434979928462226%\n",
      "Adj, read counts 64\n",
      "Number of samples per group [1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 64\n",
      "total # samples 31\n",
      "Number of total reads 62\n",
      "Number of hits in BIOM tables 29.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 9.375 %\n",
      "Percentage of reads from samples who only fall into one bin group 24.193548387096776 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 116\n",
      "Length of fail 936\n",
      "G001457435 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinobaculum; s__Actinobaculum massiliense\n",
      "Coverage Percentage: 0.0086188321482439%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 116\n",
      "Length of fail 937\n",
      "G001457475 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Achromobacter; s__Achromobacter xylosoxidans\n",
      "Coverage Percentage: 0.1411675190828602%\n",
      "Adj, read counts 55\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 45\n",
      "total # samples 15\n",
      "Number of total reads 54\n",
      "Number of hits in BIOM tables 19.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.545454545454545 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.51851851851852 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 117\n",
      "Length of fail 937\n",
      "G001457615 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas aeruginosa\n",
      "Coverage Percentage: 0.5585423032117093%\n",
      "Adj, read counts 211\n",
      "Number of samples per group [3, 1, 3, 1, 2, 1, 1, 1, 1, 1, 1, 2, 3, 1, 1, 1, 1, 2, 1, 1, 4, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 2, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 2, 1, 1, 2, 2, 2, 3, 1, 1, 1, 1, 1, 2, 1, 1, 3, 1, 1, 2, 1, 1, 1, 1, 1, 3, 1, 2, 2, 2, 1, 1, 2, 1, 2, 1, 3, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 3, 1, 1, 1, 5, 1]\n",
      "Number of bins 168\n",
      "total # samples 51\n",
      "Number of total reads 205\n",
      "Number of hits in BIOM tables 54.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 6.6350710900473935 %\n",
      "Percentage of reads from samples who only fall into one bin group 10.731707317073171 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 937\n",
      "G001457635 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus pneumoniae\n",
      "Coverage Percentage: 0.0413080634097722%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 938\n",
      "G001457835 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__; g__Fenollaria; s__Fenollaria timonensis\n",
      "Coverage Percentage: 0.0173610911503275%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 939\n",
      "G001458655 d__Archaea; p__Euryarchaeota; c__Methanobacteria; o__Methanobacteriales; f__Methanobacteriaceae; g__Methanobacterium; s__Methanobacterium formicicum\n",
      "Coverage Percentage: 0.0194827841940902%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 940\n",
      "G001458775 d__Bacteria; p__Nitrospirae; c__Nitrospira; o__Nitrospirales; f__Nitrospiraceae; g__Nitrospira; s__Candidatus Nitrospira nitrificans\n",
      "Coverage Percentage: 0.0073340435860753%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 941\n",
      "G001463885 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Aurantimonadaceae; g__Aureimonas; s__Aureimonas altamirensis\n",
      "Coverage Percentage: 0.02743442956443%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 942\n",
      "G001463905 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Aurantimonadaceae; g__Aureimonas; s__Aureimonas frigidaquae\n",
      "Coverage Percentage: 0.0060749294978814%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 943\n",
      "G001465545 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Pandoraea; s__Pandoraea norimbergensis\n",
      "Coverage Percentage: 0.0024483694021925%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 944\n",
      "G001467115 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas savastanoi\n",
      "Coverage Percentage: 0.0187043968797237%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 945\n",
      "G001467895 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Legionellales; f__Legionellaceae; g__Legionella; s__Legionella nautarum\n",
      "Coverage Percentage: 0.0073425683482152%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 946\n",
      "G001468025 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Legionellales; f__Legionellaceae; g__Legionella; s__Legionella shakespearei\n",
      "Coverage Percentage: 0.0093485859693685%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 947\n",
      "G001468135 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Legionellales; f__Legionellaceae; g__Legionella; s__Legionella santicrucis\n",
      "Coverage Percentage: 0.0076512585387215%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 948\n",
      "G001481395 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas schubertii\n",
      "Coverage Percentage: 0.0115179888287581%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 949\n",
      "G001483145 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Elizabethkingia; s__Elizabethkingia miricola\n",
      "Coverage Percentage: 0.0169368435369261%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 950\n",
      "G001483865 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Roseateles; s__Roseateles depolymerans\n",
      "Coverage Percentage: 0.0571657677021156%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 951\n",
      "G001487165 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Blautia; s__Blautia massiliensis\n",
      "Coverage Percentage: 0.1221545140275938%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 952\n",
      "G001499615 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Acetobacter; s__Acetobacter senegalensis\n",
      "Coverage Percentage: 0.0138541520515858%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 953\n",
      "G001499655 d__Bacteria; p__Chlamydiae; c__Chlamydiia; o__Parachlamydiales; f__Parachlamydiaceae; g__Candidatus Protochlamydia; s__Candidatus Protochlamydia naegleriophila\n",
      "Coverage Percentage: 0.040654766127848%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 954\n",
      "G001500315 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus auricularis\n",
      "Coverage Percentage: 0.0552940663746704%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 955\n",
      "G001501155 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Myoviridae; g__Peduovirus; s__Escherichia virus pro147\n",
      "Coverage Percentage: 0.9242540168324408%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 956\n",
      "G001507325 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium greenlandense\n",
      "Coverage Percentage: 0.0157545812761755%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 957\n",
      "G001509405 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Aeromicrobium; s__Aeromicrobium erythreum\n",
      "Coverage Percentage: 0.0041606518611753%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 118\n",
      "Length of fail 958\n",
      "G001509495 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes awajinensis\n",
      "Coverage Percentage: 0.2369696724609643%\n",
      "Adj, read counts 162\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 4, 2, 3, 1, 1, 1, 1, 6, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 141\n",
      "total # samples 40\n",
      "Number of total reads 157\n",
      "Number of hits in BIOM tables 180.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.11111111111111 %\n",
      "Percentage of reads from samples who only fall into one bin group 8.9171974522293 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 119\n",
      "Length of fail 958\n",
      "G001513985 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces griseorubiginosus\n",
      "Coverage Percentage: 0.0044158250613349%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 119\n",
      "Length of fail 959\n",
      "G001514035 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces yokosukanensis\n",
      "Coverage Percentage: 0.0044679709467728%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 119\n",
      "Length of fail 960\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G001514115 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces olivochromogenes\n",
      "Coverage Percentage: 0.0075141895659523%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 119\n",
      "Length of fail 961\n",
      "G001514195 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces griseorubiginosus\n",
      "Coverage Percentage: 0.0029187271309368%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 119\n",
      "Length of fail 962\n",
      "G001515305 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Chromobacteriaceae; g__Aquitalea; s__Aquitalea pelogenes\n",
      "Coverage Percentage: 0.0068495348346633%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 119\n",
      "Length of fail 963\n",
      "G001515505 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Leclercia; s__Leclercia adecarboxylata\n",
      "Coverage Percentage: 0.0182613284593552%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 119\n",
      "Length of fail 964\n",
      "G001517325 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Comamonas; s__Comamonas thiooxydans\n",
      "Coverage Percentage: 0.0269290008069211%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 119\n",
      "Length of fail 965\n",
      "G001517365 d__Bacteria; p__Proteobacteria; c__Hydrogenophilalia; o__Hydrogenophilales; f__Hydrogenophilaceae; g__Tepidiphilus; s__Tepidiphilus thermophilus\n",
      "Coverage Percentage: 0.0643663191180354%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 119\n",
      "Length of fail 966\n",
      "G001517405 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Erwiniaceae; g__Erwinia; s__Erwinia gerundensis\n",
      "Coverage Percentage: 0.0033026160289344%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 119\n",
      "Length of fail 967\n",
      "G001518795 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Aerococcaceae; g__Aerococcus; s__Aerococcus urinaeequi\n",
      "Coverage Percentage: 0.0562714425349798%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 119\n",
      "Length of fail 968\n",
      "G001519075 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium fuchskuhlense\n",
      "Coverage Percentage: 0.0318254022650211%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 119\n",
      "Length of fail 969\n",
      "G001528845 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia cepacia\n",
      "Coverage Percentage: 0.0047595272422851%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 119\n",
      "Length of fail 970\n",
      "G001535545 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Thermales; f__Thermaceae; g__Thermus; s__Thermus parvatiensis\n",
      "Coverage Percentage: 0.3057360729877914%\n",
      "Adj, read counts 40\n",
      "Number of samples per group [1, 3, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 2, 3, 2]\n",
      "Number of bins 35\n",
      "total # samples 9\n",
      "Number of total reads 40\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 27.500000000000004 %\n",
      "Percentage of reads from samples who only fall into one bin group 12.5 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 120\n",
      "Length of fail 970\n",
      "G001541305 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Agrobacterium; s__Agrobacterium tumefaciens\n",
      "Coverage Percentage: 0.0027465931331533%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 120\n",
      "Length of fail 971\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G001542325 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora rifamycinica\n",
      "Coverage Percentage: 0.0132141550729228%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 120\n",
      "Length of fail 972\n",
      "G001542415 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium macuxiense\n",
      "Coverage Percentage: 0.0111291145775011%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 120\n",
      "Length of fail 973\n",
      "G001543205 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Aerococcaceae; g__Aerococcus; s__Aerococcus urinaeequi\n",
      "Coverage Percentage: 0.057417056938747%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 120\n",
      "Length of fail 974\n",
      "G001543285 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Aerococcaceae; g__Aerococcus; s__Aerococcus viridans\n",
      "Coverage Percentage: 0.1746461279426077%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 120\n",
      "Length of fail 975\n",
      "G001544075 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Comamonas; s__Comamonas terrae\n",
      "Coverage Percentage: 0.0570642076204458%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 120\n",
      "Length of fail 976\n",
      "G001544295 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Enterococcaceae; g__Enterococcus; s__Enterococcus pseudoavium\n",
      "Coverage Percentage: 0.0110499905965311%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 120\n",
      "Length of fail 977\n",
      "G001544535 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Caballeronia; s__Caballeronia choica\n",
      "Coverage Percentage: 0.001542496230844%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 120\n",
      "Length of fail 978\n",
      "G001544975 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Caballeronia; s__Caballeronia arationis\n",
      "Coverage Percentage: 0.0016080844689498%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 120\n",
      "Length of fail 979\n",
      "G001546165 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Anaerococcus; s__Anaerococcus tetradius\n",
      "Coverage Percentage: 0.0224600107689058%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 120\n",
      "Length of fail 980\n",
      "G001551835 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Azonexaceae; g__Dechloromonas; s__Dechloromonas denitrificans\n",
      "Coverage Percentage: 0.0323636351627593%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 120\n",
      "Length of fail 981\n",
      "G001552235 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptosporangiales; f__Thermomonosporaceae; g__Actinomadura; s__Actinomadura rubrobrunea\n",
      "Coverage Percentage: 0.0064648378953044%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 120\n",
      "Length of fail 982\n",
      "G001552375 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Cellulomonadaceae; g__Cellulomonas; s__Cellulomonas iranensis\n",
      "Coverage Percentage: 0.0097958065386127%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 120\n",
      "Length of fail 983\n",
      "G001552395 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium minutissimum\n",
      "Coverage Percentage: 0.0379840669600548%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 120\n",
      "Length of fail 984\n",
      "G001552455 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium hydrocarbonoxydans\n",
      "Coverage Percentage: 0.0281683177708168%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 120\n",
      "Length of fail 985\n",
      "G001552495 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium paraoxydans\n",
      "Coverage Percentage: 0.0378395407650498%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 120\n",
      "Length of fail 986\n",
      "G001552595 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Rhodococcus; s__Rhodococcus erythropolis\n",
      "Coverage Percentage: 0.1202534228979815%\n",
      "Adj, read counts 47\n",
      "Number of samples per group [1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1]\n",
      "Number of bins 44\n",
      "total # samples 27\n",
      "Number of total reads 45\n",
      "Number of hits in BIOM tables 20.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.893617021276595 %\n",
      "Percentage of reads from samples who only fall into one bin group 40.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 986\n",
      "G001552735 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Sanguibacteraceae; g__Sanguibacter; s__Sanguibacter suarezii\n",
      "Coverage Percentage: 0.0119108343894448%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 987\n",
      "G001553115 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Peptoniphilus; s__Peptoniphilus coxii\n",
      "Coverage Percentage: 0.0377116176814701%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 988\n",
      "G001553955 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Moraxella; s__Moraxella osloensis\n",
      "Coverage Percentage: 0.1353607499876378%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 989\n",
      "G001558355 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Salmonella; s__Salmonella enterica\n",
      "Coverage Percentage: 0.0182976962473363%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 990\n",
      "G001558775 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus lugdunensis\n",
      "Coverage Percentage: 0.0293954224794883%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 991\n",
      "G001565895 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Alteromonadales; f__Alteromonadaceae; g__Paraglaciecola; s__Paraglaciecola hydrolytica\n",
      "Coverage Percentage: 0.0020299396954707%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 992\n",
      "G001570485 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacterium; s__Mycolicibacterium novocastrense\n",
      "Coverage Percentage: 0.001781535587377%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 993\n",
      "G001570805 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Brevibacteriaceae; g__Brevibacterium; s__Brevibacterium epidermidis\n",
      "Coverage Percentage: 0.0158218406740644%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 994\n",
      "G001570825 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Promicromonosporaceae; g__Cellulosimicrobium; s__Cellulosimicrobium funkei\n",
      "Coverage Percentage: 0.0053284132031218%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 995\n",
      "G001570885 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Micrococcus; s__Micrococcus lylae\n",
      "Coverage Percentage: 0.0542803335543238%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 996\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G001570945 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Intrasporangiaceae; g__Janibacter; s__Janibacter anophelis\n",
      "Coverage Percentage: 0.0222889010031515%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 997\n",
      "G001570985 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Intrasporangiaceae; g__Janibacter; s__Janibacter limosus\n",
      "Coverage Percentage: 0.008076304212192%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 998\n",
      "G001571005 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Herbiconiux; s__Herbiconiux solani\n",
      "Coverage Percentage: 0.0187267425415747%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 999\n",
      "G001571085 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Azohydromonas; s__Azohydromonas lata\n",
      "Coverage Percentage: 0.021571414246118%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 1000\n",
      "G001571145 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Hydrogenophaga; s__Hydrogenophaga flava\n",
      "Coverage Percentage: 0.0412920451689956%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 1001\n",
      "G001571185 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Janthinobacterium; s__Janthinobacterium agaricidamnosum\n",
      "Coverage Percentage: 0.0283466642481136%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 1002\n",
      "G001571225 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Hydrogenophaga; s__Hydrogenophaga palleronii\n",
      "Coverage Percentage: 0.0360451418952312%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 1003\n",
      "G001571305 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Erwiniaceae; g__Erwinia; s__Erwinia persicina\n",
      "Coverage Percentage: 0.0051965794687424%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 1004\n",
      "G001571325 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Delftia; s__Delftia tsuruhatensis\n",
      "Coverage Percentage: 0.0581941333294607%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 1005\n",
      "G001575205 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Tsukamurellaceae; g__Tsukamurella; s__Tsukamurella pseudospumae\n",
      "Coverage Percentage: 0.0072058767568372%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 1006\n",
      "G001580995 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Acetobacter; s__Acetobacter senegalensis\n",
      "Coverage Percentage: 0.0151587407562838%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 1007\n",
      "G001584145 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Collimonas; s__Collimonas fungivorans\n",
      "Coverage Percentage: 0.0254833760705205%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 1008\n",
      "G001584165 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Collimonas; s__Collimonas arenae\n",
      "Coverage Percentage: 0.0293607920597579%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 1009\n",
      "G001584185 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Collimonas; s__Collimonas pratensis\n",
      "Coverage Percentage: 0.0405233834058315%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 1010\n",
      "G001584205 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Collimonas; s__Collimonas arenae\n",
      "Coverage Percentage: 0.0274770951656662%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 1011\n",
      "G001584225 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Collimonas; s__Collimonas pratensis\n",
      "Coverage Percentage: 0.0324209227556746%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 121\n",
      "Length of fail 1012\n",
      "G001586155 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas citronellolis\n",
      "Coverage Percentage: 0.6349903703460749%\n",
      "Adj, read counts 259\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 3, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 2, 1, 4, 2, 3, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 3, 1, 1, 1, 2, 2, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 3, 3, 1, 2, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 158\n",
      "total # samples 14\n",
      "Number of total reads 252\n",
      "Number of hits in BIOM tables 119.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 6.563706563706565 %\n",
      "Percentage of reads from samples who only fall into one bin group 2.7777777777777777 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 122\n",
      "Length of fail 1012\n",
      "G001586235 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium simulans\n",
      "Coverage Percentage: 0.1423402914224101%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 122\n",
      "Length of fail 1013\n",
      "G001590925 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia plymuthica\n",
      "Coverage Percentage: 0.0288205420115187%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 122\n",
      "Length of fail 1014\n",
      "G001590965 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium lentum\n",
      "Coverage Percentage: 0.0278301467399489%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 122\n",
      "Length of fail 1015\n",
      "G001591005 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas sanguinis\n",
      "Coverage Percentage: 0.0443567555412658%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 122\n",
      "Length of fail 1016\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G001591205 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas maltophilia\n",
      "Coverage Percentage: 0.3384876834335394%\n",
      "Adj, read counts 102\n",
      "Number of samples per group [1, 3, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 2, 2, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 3, 3, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 2, 2, 3, 1]\n",
      "Number of bins 82\n",
      "total # samples 28\n",
      "Number of total reads 100\n",
      "Number of hits in BIOM tables 22.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 8.823529411764705 %\n",
      "Percentage of reads from samples who only fall into one bin group 15.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 123\n",
      "Length of fail 1016\n",
      "G001591285 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium cloacae\n",
      "Coverage Percentage: 0.0362367695618588%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 123\n",
      "Length of fail 1017\n",
      "G001591305 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium amiense\n",
      "Coverage Percentage: 0.0298701812886849%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 123\n",
      "Length of fail 1018\n",
      "G001591345 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Variovorax; s__Variovorax boronicumulans\n",
      "Coverage Percentage: 0.2213395978858621%\n",
      "Adj, read counts 96\n",
      "Number of samples per group [1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 3, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 2, 1, 2, 9, 2, 1, 1]\n",
      "Number of bins 83\n",
      "total # samples 34\n",
      "Number of total reads 95\n",
      "Number of hits in BIOM tables 66.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.583333333333334 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.842105263157894 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 124\n",
      "Length of fail 1018\n",
      "G001591365 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Variovorax; s__Variovorax paradoxus\n",
      "Coverage Percentage: 0.088767944148818%\n",
      "Adj, read counts 39\n",
      "Number of samples per group [2, 1, 1, 1, 1, 2, 1, 3, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 9]\n",
      "Number of bins 29\n",
      "total # samples 19\n",
      "Number of total reads 39\n",
      "Number of hits in BIOM tables 10.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 35.8974358974359 %\n",
      "Percentage of reads from samples who only fall into one bin group 28.205128205128204 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 125\n",
      "Length of fail 1018\n",
      "G001591385 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Variovorax; s__Variovorax soli\n",
      "Coverage Percentage: 0.0340755925094354%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 125\n",
      "Length of fail 1019\n",
      "G001591425 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Sutcliffiella; s__Sutcliffiella cohnii\n",
      "Coverage Percentage: 0.0029175599568772%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 125\n",
      "Length of fail 1020\n",
      "G001591765 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Lactococcus; s__Lactococcus raffinolactis\n",
      "Coverage Percentage: 0.062527160808489%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 125\n",
      "Length of fail 1021\n",
      "G001592045 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptosporangiales; f__Nocardiopsaceae; g__Thermobifida; s__Thermobifida fusca\n",
      "Coverage Percentage: 0.0364283035849236%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 125\n",
      "Length of fail 1022\n",
      "G001592125 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium hominis\n",
      "Coverage Percentage: 0.0719525484216961%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 125\n",
      "Length of fail 1023\n",
      "G001592165 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Caldimonas; s__Caldimonas taiwanensis\n",
      "Coverage Percentage: 0.1276246143948625%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 125\n",
      "Length of fail 1024\n",
      "G001592205 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas vesicularis\n",
      "Coverage Percentage: 0.0663838852076758%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 125\n",
      "Length of fail 1025\n",
      "G001592265 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Curvibacter; s__Curvibacter delicatus\n",
      "Coverage Percentage: 0.1039676138354895%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 125\n",
      "Length of fail 1026\n",
      "G001592285 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Hydrogenophaga; s__Hydrogenophaga pseudoflava\n",
      "Coverage Percentage: 0.0354760307986149%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 125\n",
      "Length of fail 1027\n",
      "G001592305 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Hydrogenophaga; s__Hydrogenophaga taeniospiralis\n",
      "Coverage Percentage: 0.050091246426758%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 125\n",
      "Length of fail 1028\n",
      "G001592345 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas adhaesiva\n",
      "Coverage Percentage: 0.0674132222180884%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 125\n",
      "Length of fail 1029\n",
      "G001595745 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter lactucae\n",
      "Coverage Percentage: 0.0404855502157736%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 125\n",
      "Length of fail 1030\n",
      "G001595965 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Polynucleobacter; s__Polynucleobacter yangtzensis\n",
      "Coverage Percentage: 0.3249706382747866%\n",
      "Adj, read counts 43\n",
      "Number of samples per group [1, 1, 14, 1, 1, 3, 3, 1, 1, 1, 1, 1, 3, 1, 1, 2, 1, 3]\n",
      "Number of bins 30\n",
      "total # samples 24\n",
      "Number of total reads 43\n",
      "Number of hits in BIOM tables 49.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 53.48837209302325 %\n",
      "Percentage of reads from samples who only fall into one bin group 27.906976744186046 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 126\n",
      "Length of fail 1030\n",
      "G001595985 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Polynucleobacter; s__Polynucleobacter sinensis\n",
      "Coverage Percentage: 0.0894574367640734%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 126\n",
      "Length of fail 1031\n",
      "G001596795 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Alteromonadales; f__Pseudoalteromonadaceae; g__Pseudoalteromonas; s__Pseudoalteromonas telluritireducens\n",
      "Coverage Percentage: 0.0074416916235593%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 126\n",
      "Length of fail 1032\n",
      "G001597285 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas alcaligenes\n",
      "Coverage Percentage: 0.0309102524677706%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 126\n",
      "Length of fail 1033\n",
      "G001598055 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Cupriavidus; s__Cupriavidus nantongensis\n",
      "Coverage Percentage: 0.0361804031690782%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 126\n",
      "Length of fail 1034\n",
      "G001598235 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Polaromonas; s__Polaromonas jejuensis\n",
      "Coverage Percentage: 0.0521590822805142%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 126\n",
      "Length of fail 1035\n",
      "G001598255 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Mitsuaria; s__Mitsuaria chitosanitabida\n",
      "Coverage Percentage: 0.0396700428295581%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 126\n",
      "Length of fail 1036\n",
      "G001598355 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas asaccharolytica\n",
      "Coverage Percentage: 0.0065472792694276%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 126\n",
      "Length of fail 1037\n",
      "G001598375 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium capsulatum\n",
      "Coverage Percentage: 0.0454335462040354%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 126\n",
      "Length of fail 1038\n",
      "G001598435 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas pituitosa\n",
      "Coverage Percentage: 0.0228679291730703%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 126\n",
      "Length of fail 1039\n",
      "G001598455 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas pruni\n",
      "Coverage Percentage: 0.0235638638673004%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 126\n",
      "Length of fail 1040\n",
      "G001598475 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Erwiniaceae; g__Pantoea; s__Pantoea agglomerans\n",
      "Coverage Percentage: 0.0602899074050192%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 126\n",
      "Length of fail 1041\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G001598575 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium subterraneum\n",
      "Coverage Percentage: 0.0782797425806498%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 126\n",
      "Length of fail 1042\n",
      "G001598595 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Achromobacter; s__Achromobacter xylosoxidans\n",
      "Coverage Percentage: 0.141875664283232%\n",
      "Adj, read counts 56\n",
      "Number of samples per group [1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 47\n",
      "total # samples 14\n",
      "Number of total reads 54\n",
      "Number of hits in BIOM tables 19.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.285714285714285 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.666666666666664 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1042\n",
      "G001598795 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Delftia; s__Delftia acidovorans\n",
      "Coverage Percentage: 0.0522301158787845%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1043\n",
      "G001598815 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingopyxis; s__Sphingopyxis terrae\n",
      "Coverage Percentage: 0.046349595928483%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1044\n",
      "G001598855 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Pluralibacter; s__Pluralibacter gergoviae\n",
      "Coverage Percentage: 0.0276618824731876%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1045\n",
      "G001601555 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Enterococcaceae; g__Enterococcus; s__Enterococcus pernyi\n",
      "Coverage Percentage: 0.0073383401301834%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1046\n",
      "G001602025 d__Eukaryota; p__Apicomplexa; c__Aconoidasida; o__Haemosporida; f__Plasmodiidae; g__Plasmodium; s__Plasmodium gaboni\n",
      "Coverage Percentage: 0.0119350008579966%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1047\n",
      "G001602525 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium chilense\n",
      "Coverage Percentage: 0.0337220312207274%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1048\n",
      "G001605895 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter pragensis\n",
      "Coverage Percentage: 0.0327417169389569%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1049\n",
      "G001606025 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter alimentarius\n",
      "Coverage Percentage: 0.0482166698910631%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1050\n",
      "G001610975 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingopyxis; s__Sphingopyxis terrae\n",
      "Coverage Percentage: 0.006928693451757%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1051\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G001611955 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus haemolyticus\n",
      "Coverage Percentage: 0.0309354940265592%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1052\n",
      "G001612705 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas yamanorum\n",
      "Coverage Percentage: 0.0675530328497039%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1053\n",
      "G001612865 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Nocardia; s__Nocardia salmonicida\n",
      "Coverage Percentage: 0.0034924807023778%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1054\n",
      "G001613085 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Nocardia; s__Nocardia salmonicida\n",
      "Coverage Percentage: 0.0050029569534923%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1055\n",
      "G001613285 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Nocardia; s__Nocardia salmonicida\n",
      "Coverage Percentage: 0.006856684181656%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1056\n",
      "G001618265 d__Bacteria; p__Chlamydiae; c__Chlamydiia; o__Parachlamydiales; f__Waddliaceae; g__Waddlia; s__Waddlia chondrophila\n",
      "Coverage Percentage: 0.0071374618429399%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1057\n",
      "G001618385 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Paludibacteraceae; g__Paludibacter; s__Paludibacter jiangxiensis\n",
      "Coverage Percentage: 0.0163713279111495%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1058\n",
      "G001618865 d__Bacteria; p__Acidobacteria; c__Vicinamibacteria; o__; f__Vicinamibacteraceae; g__Luteitalea; s__Luteitalea pratensis\n",
      "Coverage Percentage: 0.0030747372369662%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1059\n",
      "G001619695 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Neisseriaceae; g__Crenobacter; s__Crenobacter luteus\n",
      "Coverage Percentage: 0.0127962408420346%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1060\n",
      "G001619955 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas hankookensis\n",
      "Coverage Percentage: 0.0277131051262689%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1061\n",
      "G001619985 d__Eukaryota; p__Ascomycota; c__Xylonomycetes; o__Xylonales; f__Xylonaceae; g__Xylona; s__Xylona heveae\n",
      "Coverage Percentage: 0.0016476100635365%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1062\n",
      "G001620265 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Frigidibacter; s__Frigidibacter mobilis\n",
      "Coverage Percentage: 0.0360288399134586%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1063\n",
      "G001620305 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Rhodococcus; s__Rhodococcus fascians\n",
      "Coverage Percentage: 0.0427832756286179%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1064\n",
      "G001624335 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Cellulomonadaceae; g__Oerskovia; s__Oerskovia enterophila\n",
      "Coverage Percentage: 0.0039455522604928%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1065\n",
      "G001632305 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae; g__Paenibacillus; s__Paenibacillus glucanolyticus\n",
      "Coverage Percentage: 0.0101736558269189%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1066\n",
      "G001632805 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycobacteroides; s__Mycobacteroides chelonae\n",
      "Coverage Percentage: 0.015070130782094%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1067\n",
      "G001634625 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Croceicoccus; s__Croceicoccus bisphenolivorans\n",
      "Coverage Percentage: 0.0125693743448248%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1068\n",
      "G001639375 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus oralis\n",
      "Coverage Percentage: 0.058975508187096%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1069\n",
      "G001640105 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Hydrogenophaga; s__Hydrogenophaga crassostreae\n",
      "Coverage Percentage: 0.0306041829910961%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1070\n",
      "G001641265 d__Eukaryota; p__Ascomycota; c__Leotiomycetes; o__; f__Pseudeurotiaceae; g__Pseudogymnoascus; s__Pseudogymnoascus destructans\n",
      "Coverage Percentage: 0.0034952695630335%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1071\n",
      "G001641335 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium stylosanthis\n",
      "Coverage Percentage: 0.0191089959176958%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1072\n",
      "G001641635 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium centrolobii\n",
      "Coverage Percentage: 0.0171944084456086%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1073\n",
      "G001641695 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium neotropicale\n",
      "Coverage Percentage: 0.0184754122095982%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 127\n",
      "Length of fail 1074\n",
      "G001642025 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Cutibacterium; s__Cutibacterium namnetense\n",
      "Coverage Percentage: 2.1884047331546124%\n",
      "Adj, read counts 339\n",
      "Number of samples per group [1, 11, 3, 6, 3, 14, 9, 1, 2, 3, 5, 1, 2, 3, 2, 2, 2, 7, 1, 12, 10, 10, 2, 4, 1, 26, 1, 7, 1, 18, 5, 5, 5, 5, 2, 7, 7, 3, 13, 6, 1, 2, 5, 1, 3, 8, 2, 5]\n",
      "Number of bins 137\n",
      "total # samples 54\n",
      "Number of total reads 333\n",
      "Number of hits in BIOM tables 104.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 30.973451327433626 %\n",
      "Percentage of reads from samples who only fall into one bin group 3.6036036036036037 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 128\n",
      "Length of fail 1074\n",
      "G001642055 d__Eukaryota; p__Ascomycota; c__Dothideomycetes; o__Pleosporales; f__Pleosporaceae; g__Alternaria; s__Alternaria alternata\n",
      "Coverage Percentage: 0.0060650342621393%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 128\n",
      "Length of fail 1075\n",
      "G001642795 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas viridiflava\n",
      "Coverage Percentage: 0.0306595191979267%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 128\n",
      "Length of fail 1076\n",
      "G001642805 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia surfactantfaciens\n",
      "Coverage Percentage: 0.0241712788150172%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 128\n",
      "Length of fail 1077\n",
      "G001643015 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium crudilactis\n",
      "Coverage Percentage: 0.0093866711755842%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 128\n",
      "Length of fail 1078\n",
      "G001643305 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas salmonicida\n",
      "Coverage Percentage: 0.0439276753444612%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 128\n",
      "Length of fail 1079\n",
      "G001645105 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas marginalis\n",
      "Coverage Percentage: 0.1072094541416963%\n",
      "Adj, read counts 43\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 41\n",
      "total # samples 21\n",
      "Number of total reads 43\n",
      "Number of hits in BIOM tables 17.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.627906976744185 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.27906976744186 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 129\n",
      "Length of fail 1079\n",
      "G001645135 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia ginsengiterrae\n",
      "Coverage Percentage: 0.0167757799206166%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 129\n",
      "Length of fail 1080\n",
      "G001646725 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Rhodococcus; s__Rhodococcus kroppenstedtii\n",
      "Coverage Percentage: 0.0174256490734894%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 129\n",
      "Length of fail 1081\n",
      "G001646745 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Rhodococcus; s__Rhodococcus qingshengii\n",
      "Coverage Percentage: 0.1419371376799872%\n",
      "Adj, read counts 62\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 61\n",
      "total # samples 30\n",
      "Number of total reads 60\n",
      "Number of hits in BIOM tables 30.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 9.67741935483871 %\n",
      "Percentage of reads from samples who only fall into one bin group 28.333333333333332 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1081\n",
      "G001646785 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Rhodococcus; s__Rhodococcus phenolicus\n",
      "Coverage Percentage: 0.0039118858836099%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1082\n",
      "G001646965 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Chaetothyriales; f__Herpotrichiellaceae; g__Fonsecaea; s__Fonsecaea nubica\n",
      "Coverage Percentage: 0.0011511400414534%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1083\n",
      "G001648155 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium glaciei\n",
      "Coverage Percentage: 0.0030995416783669%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1084\n",
      "G001650025 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Devosiaceae; g__Devosia; s__Devosia elaeis\n",
      "Coverage Percentage: 0.0069567045626706%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1085\n",
      "G001650715 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Rhodospirillaceae; g__Magnetospirillum; s__Magnetospirillum marisnigri\n",
      "Coverage Percentage: 0.0032666878532647%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1086\n",
      "G001652465 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium chocolatum\n",
      "Coverage Percentage: 0.0172942130659673%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1087\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G001653265 d__Eukaryota; p__Ascomycota; c__Sordariomycetes; o__Hypocreales; f__Ophiocordycipitaceae; g__Purpureocillium; s__Purpureocillium lilacinum\n",
      "Coverage Percentage: 0.0007836453526288%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1088\n",
      "G001653715 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium platani\n",
      "Coverage Percentage: 0.0043262505212135%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1089\n",
      "G001654915 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Buttiauxella; s__Buttiauxella ferragutiae\n",
      "Coverage Percentage: 0.0059345385031679%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1090\n",
      "G001656035 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Yersinia; s__Yersinia entomophaga\n",
      "Coverage Percentage: 0.0129578337121667%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1091\n",
      "G001659725 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Polynucleobacter; s__Polynucleobacter wuianus\n",
      "Coverage Percentage: 0.0849832340667635%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1092\n",
      "G001661075 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas silesiensis\n",
      "Coverage Percentage: 0.0203266955754191%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1093\n",
      "G001661255 d__Eukaryota; p__Ascomycota; c__Saccharomycetes; o__Saccharomycetales; f__Phaffomycetaceae; g__Wickerhamomyces; s__Wickerhamomyces anomalus\n",
      "Coverage Percentage: 0.0014561940770224%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1094\n",
      "G001661965 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Croceicoccus; s__Croceicoccus mobilis\n",
      "Coverage Percentage: 0.058161364657032%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1095\n",
      "G001662105 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium aegyptiacum\n",
      "Coverage Percentage: 0.0083485404127143%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1096\n",
      "G001663175 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Erythrobacter; s__Erythrobacter neustonensis\n",
      "Coverage Percentage: 0.0457875013388394%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1097\n",
      "G001675285 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Elizabethkingia; s__Elizabethkingia miricola\n",
      "Coverage Percentage: 0.020351854219249%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1098\n",
      "G001675355 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Tepidimonas; s__Tepidimonas fonticaldi\n",
      "Coverage Percentage: 0.0549842613021279%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1099\n",
      "G001676785 d__Bacteria; p__Spirochaetes; c__Spirochaetia; o__Brachyspirales; f__Brachyspiraceae; g__Brachyspira; s__Brachyspira hyodysenteriae\n",
      "Coverage Percentage: 0.0077096613116508%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1100\n",
      "G001676875 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Chromobacteriaceae; g__Chromobacterium; s__Chromobacterium subtsugae\n",
      "Coverage Percentage: 0.003141188301133%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1101\n",
      "G001677885 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Janthinobacterium; s__Janthinobacterium psychrotolerans\n",
      "Coverage Percentage: 0.0784222215455131%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1102\n",
      "G001678755 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter gandensis\n",
      "Coverage Percentage: 0.0757018712369402%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1103\n",
      "G001678905 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Dermabacteraceae; g__Dermabacter; s__Dermabacter vaginalis\n",
      "Coverage Percentage: 0.0210674685680893%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1104\n",
      "G001679005 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Moraxella; s__Moraxella nonliquefaciens\n",
      "Coverage Percentage: 0.0132421350706546%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1105\n",
      "G001679175 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Moraxella; s__Moraxella osloensis\n",
      "Coverage Percentage: 0.111421449611161%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1106\n",
      "G001679535 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus pneumoniae\n",
      "Coverage Percentage: 0.0500210499139667%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1107\n",
      "G001679665 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomonadales; f__Hyphomonadaceae; g__Candidatus Viadribacter; s__Candidatus Viadribacter manganicus\n",
      "Coverage Percentage: 0.0033755554597064%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1108\n",
      "G001679725 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Cryobacterium; s__Cryobacterium arcticum\n",
      "Coverage Percentage: 0.017971768695523%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1109\n",
      "G001680005 d__Eukaryota; p__Apicomplexa; c__Aconoidasida; o__Haemosporida; f__Plasmodiidae; g__Plasmodium; s__Plasmodium coatneyi\n",
      "Coverage Percentage: 0.0023766704869176%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1110\n",
      "G001682385 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Zoogloeaceae; g__Azoarcus; s__Azoarcus olearius\n",
      "Coverage Percentage: 0.015701686819411%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1111\n",
      "G001683355 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella scopos\n",
      "Coverage Percentage: 0.0266663841624076%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 130\n",
      "Length of fail 1112\n",
      "G001686925 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium piscis\n",
      "Coverage Percentage: 0.1505870320401723%\n",
      "Adj, read counts 53\n",
      "Number of samples per group [1, 1, 2, 1, 1, 1, 1, 3, 1, 2, 1, 1, 1, 5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1]\n",
      "Number of bins 44\n",
      "total # samples 23\n",
      "Number of total reads 51\n",
      "Number of hits in BIOM tables 35.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.754716981132077 %\n",
      "Percentage of reads from samples who only fall into one bin group 21.568627450980394 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 131\n",
      "Length of fail 1112\n",
      "G001693385 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium icense\n",
      "Coverage Percentage: 0.0176984281560965%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 131\n",
      "Length of fail 1113\n",
      "G001693515 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium paxllaeri\n",
      "Coverage Percentage: 0.016102234987829%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 131\n",
      "Length of fail 1114\n",
      "G001693815 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Tessaracoccus; s__Tessaracoccus lapidicaptus\n",
      "Coverage Percentage: 0.0096809182302374%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 131\n",
      "Length of fail 1115\n",
      "G001695395 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Aliarcobacter; s__Aliarcobacter thereius\n",
      "Coverage Percentage: 0.0382861626161547%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 131\n",
      "Length of fail 1116\n",
      "G001695625 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas defluvii\n",
      "Coverage Percentage: 0.0668492441722554%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 131\n",
      "Length of fail 1117\n",
      "G001696605 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter wuhouensis\n",
      "Coverage Percentage: 0.1724490107663225%\n",
      "Adj, read counts 49\n",
      "Number of samples per group [1, 3, 1, 1, 2, 1, 2, 2, 1, 1, 1, 2, 1, 1, 2, 1, 1, 2, 5, 2, 4, 8]\n",
      "Number of bins 29\n",
      "total # samples 19\n",
      "Number of total reads 48\n",
      "Number of hits in BIOM tables 6.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 40.816326530612244 %\n",
      "Percentage of reads from samples who only fall into one bin group 22.916666666666664 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 132\n",
      "Length of fail 1117\n",
      "G001696715 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Zoogloeaceae; g__Thauera; s__Thauera phenolivorans\n",
      "Coverage Percentage: 0.0434972316247909%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 132\n",
      "Length of fail 1118\n",
      "G001698205 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Tsuneonella; s__Tsuneonella dongtanensis\n",
      "Coverage Percentage: 0.0138561453001251%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 132\n",
      "Length of fail 1119\n",
      "G001700315 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Planococcaceae; g__Caryophanon; s__Caryophanon tenue\n",
      "Coverage Percentage: 0.0107128472174426%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 132\n",
      "Length of fail 1120\n",
      "G001700325 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Planococcaceae; g__Caryophanon; s__Caryophanon latum\n",
      "Coverage Percentage: 0.0388585712237805%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 132\n",
      "Length of fail 1121\n",
      "G001700755 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Cutibacterium; s__Cutibacterium granulosum\n",
      "Coverage Percentage: 0.650672506224687%\n",
      "Adj, read counts 85\n",
      "Number of samples per group [3, 1, 1, 1, 2, 1, 1, 2, 2, 7, 1, 2, 1, 1, 1, 1, 6, 1, 3, 3, 1, 1, 2, 1, 2, 2, 1, 1, 2, 1, 2, 4, 1, 1, 1, 1, 1, 1, 2, 1]\n",
      "Number of bins 62\n",
      "total # samples 31\n",
      "Number of total reads 81\n",
      "Number of hits in BIOM tables 59.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 23.52941176470588 %\n",
      "Percentage of reads from samples who only fall into one bin group 28.39506172839506 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 133\n",
      "Length of fail 1121\n",
      "G001702265 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas aylmerensis\n",
      "Coverage Percentage: 0.0404007086015141%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 133\n",
      "Length of fail 1122\n",
      "G001703515 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Listeriaceae; g__Brochothrix; s__Brochothrix thermosphacta\n",
      "Coverage Percentage: 0.0117127127534764%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 133\n",
      "Length of fail 1123\n",
      "G001704615 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter defluvii\n",
      "Coverage Percentage: 0.2314535513851015%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 133\n",
      "Length of fail 1124\n",
      "G001705425 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Phyllobacteriaceae; g__Mesorhizobium; s__Mesorhizobium hungaricum\n",
      "Coverage Percentage: 0.1554395843819305%\n",
      "Adj, read counts 56\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1]\n",
      "Number of bins 46\n",
      "total # samples 18\n",
      "Number of total reads 56\n",
      "Number of hits in BIOM tables 64.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.285714285714285 %\n",
      "Percentage of reads from samples who only fall into one bin group 17.857142857142858 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 134\n",
      "Length of fail 1124\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G001707755 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter celticus\n",
      "Coverage Percentage: 0.73766629326566%\n",
      "Adj, read counts 130\n",
      "Number of samples per group [1, 1, 3, 1, 2, 2, 3, 2, 1, 1, 1, 4, 1, 4, 3, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 2, 2, 1, 2, 1, 7, 1, 1, 1, 1, 1, 3, 1, 2, 3, 3, 1, 1, 3, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 2, 2, 1, 1, 2, 1, 1, 1, 2, 3, 1, 2, 2, 1, 1, 3]\n",
      "Number of bins 106\n",
      "total # samples 37\n",
      "Number of total reads 129\n",
      "Number of hits in BIOM tables 274.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 13.846153846153847 %\n",
      "Percentage of reads from samples who only fall into one bin group 10.077519379844961 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 135\n",
      "Length of fail 1124\n",
      "G001708485 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas lurida\n",
      "Coverage Percentage: 0.0456075835240845%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 135\n",
      "Length of fail 1125\n",
      "G001715655 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Listeriaceae; g__Brochothrix; s__Brochothrix thermosphacta\n",
      "Coverage Percentage: 0.0121037135956766%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 135\n",
      "Length of fail 1126\n",
      "G001717565 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Cutibacterium; s__Cutibacterium modestum\n",
      "Coverage Percentage: 1.6890209861216123%\n",
      "Adj, read counts 300\n",
      "Number of samples per group [3, 1, 25, 4, 1, 6, 2, 2, 1, 2, 1, 4, 1, 7, 1, 5, 19, 1, 1, 3, 4, 4, 1, 2, 1, 1, 1, 1, 2, 2, 1, 1, 14, 3, 1, 5, 15, 6, 1, 2, 3, 6, 1, 1, 1, 3, 8, 1, 1, 3, 3, 29, 1]\n",
      "Number of bins 94\n",
      "total # samples 51\n",
      "Number of total reads 299\n",
      "Number of hits in BIOM tables 100.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 42.666666666666664 %\n",
      "Percentage of reads from samples who only fall into one bin group 4.682274247491638 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 136\n",
      "Length of fail 1126\n",
      "G001717955 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas panacis\n",
      "Coverage Percentage: 0.0446180236901979%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 136\n",
      "Length of fail 1127\n",
      "G001720465 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Porphyrobacter; s__Porphyrobacter sp. LM 6\n",
      "Coverage Percentage: 0.0354786445076742%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 136\n",
      "Length of fail 1128\n",
      "G001721185 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Morganellaceae; g__Xenorhabdus; s__Xenorhabdus hominickii\n",
      "Coverage Percentage: 0.0095297078138518%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 136\n",
      "Length of fail 1129\n",
      "G001721295 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas turrisvirgatae\n",
      "Coverage Percentage: 0.0609579914097998%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 136\n",
      "Length of fail 1130\n",
      "G001722415 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycobacterium; s__Mycobacterium koreense\n",
      "Coverage Percentage: 0.1020132787542512%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 136\n",
      "Length of fail 1131\n",
      "G001729805 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Enterobacter; s__Enterobacter roggenkampii\n",
      "Coverage Percentage: 0.0226529826325092%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 136\n",
      "Length of fail 1132\n",
      "G001730605 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas poae\n",
      "Coverage Percentage: 0.023924545817089%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 136\n",
      "Length of fail 1133\n",
      "G001730645 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas salomonii\n",
      "Coverage Percentage: 0.0421463438676789%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 136\n",
      "Length of fail 1134\n",
      "G001741865 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Boseaceae; g__Bosea; s__Bosea vaviloviae\n",
      "Coverage Percentage: 0.013375897883212%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 136\n",
      "Length of fail 1135\n",
      "G001742225 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium resinovorum\n",
      "Coverage Percentage: 0.0462113888452038%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 136\n",
      "Length of fail 1136\n",
      "G001746835 d__Bacteria; p__Verrucomicrobia; c__Opitutae; o__Opitutales; f__Opitutaceae; g__Lacunisphaera; s__Lacunisphaera limnophila\n",
      "Coverage Percentage: 0.0496751350944589%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 136\n",
      "Length of fail 1137\n",
      "G001747495 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Cloacibacterium; s__Cloacibacterium normanense\n",
      "Coverage Percentage: 0.0377431363373694%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 136\n",
      "Length of fail 1138\n",
      "G001748245 d__Bacteria; p__Proteobacteria; c__Oligoflexia; o__Oligoflexales; f__Oligoflexaceae; g__Oligoflexus; s__Oligoflexus tunisiensis\n",
      "Coverage Percentage: 0.0079798031710205%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 136\n",
      "Length of fail 1139\n",
      "G001748265 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas humi\n",
      "Coverage Percentage: 0.6096546234072376%\n",
      "Adj, read counts 248\n",
      "Number of samples per group [1, 1, 1, 1, 1, 3, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 4, 1, 1, 1, 2, 1, 1, 1, 3, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 2, 1, 1, 2, 1, 3, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 2, 1, 1, 3, 2, 1, 1, 1, 1, 1, 2, 2, 1, 1, 3, 2, 1, 2, 1, 2, 2, 1, 1, 1, 1, 2, 1]\n",
      "Number of bins 151\n",
      "total # samples 13\n",
      "Number of total reads 247\n",
      "Number of hits in BIOM tables 111.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 6.854838709677419 %\n",
      "Percentage of reads from samples who only fall into one bin group 2.0242914979757085 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 137\n",
      "Length of fail 1139\n",
      "G001752395 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Chromatiales; f__Chromatiaceae; g__Rheinheimera; s__Rheinheimera salexigens\n",
      "Coverage Percentage: 0.0043871104950432%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 137\n",
      "Length of fail 1140\n",
      "G001758385 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia contaminans\n",
      "Coverage Percentage: 0.013754897767495%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 137\n",
      "Length of fail 1141\n",
      "G001758635 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Janthinobacterium; s__Janthinobacterium lividum\n",
      "Coverage Percentage: 0.071110033129558%\n",
      "Adj, read counts 33\n",
      "Number of samples per group [1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 32\n",
      "total # samples 21\n",
      "Number of total reads 32\n",
      "Number of hits in BIOM tables 8.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.151515151515152 %\n",
      "Percentage of reads from samples who only fall into one bin group 46.875 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 138\n",
      "Length of fail 1141\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G001758785 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Duganella; s__Duganella phyllosphaerae\n",
      "Coverage Percentage: 0.0896322410969887%\n",
      "Adj, read counts 35\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 2, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1]\n",
      "Number of bins 31\n",
      "total # samples 22\n",
      "Number of total reads 35\n",
      "Number of hits in BIOM tables 45.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 22.857142857142858 %\n",
      "Percentage of reads from samples who only fall into one bin group 37.142857142857146 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 139\n",
      "Length of fail 1141\n",
      "G001761385 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Hydrogenophaga; s__Hydrogenophaga crassostreae\n",
      "Coverage Percentage: 0.0348696191768342%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 139\n",
      "Length of fail 1142\n",
      "G001761545 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Weissella; s__Weissella soli\n",
      "Coverage Percentage: 0.0305373625224574%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 139\n",
      "Length of fail 1143\n",
      "G001766745 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lentilactobacillus; s__Lentilactobacillus sunkii\n",
      "Coverage Percentage: 0.0066895207405549%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 139\n",
      "Length of fail 1144\n",
      "G001853345 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas haloaromaticamans\n",
      "Coverage Percentage: 0.0324738559058449%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 139\n",
      "Length of fail 1145\n",
      "G001853485 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium tructae\n",
      "Coverage Percentage: 0.0362278827097113%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 139\n",
      "Length of fail 1146\n",
      "G001854325 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Cupriavidus; s__Cupriavidus malaysiensis\n",
      "Coverage Percentage: 0.0033964447274535%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 139\n",
      "Length of fail 1147\n",
      "G001866515 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia timonae\n",
      "Coverage Percentage: 0.0922057185973272%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 139\n",
      "Length of fail 1148\n",
      "G001870435 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas costantinii\n",
      "Coverage Percentage: 0.0595156305097484%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 139\n",
      "Length of fail 1149\n",
      "G001879695 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Nioella; s__Nioella sediminis\n",
      "Coverage Percentage: 0.0184978485210427%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 139\n",
      "Length of fail 1150\n",
      "G001880325 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Thermales; f__Thermaceae; g__Thermus; s__Thermus brockianus\n",
      "Coverage Percentage: 0.0148640483889674%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 139\n",
      "Length of fail 1151\n",
      "G001881485 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Legionellales; f__Coxiellaceae; g__Rickettsiella; s__Rickettsiella grylli\n",
      "Coverage Percentage: 0.0095131442439859%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 139\n",
      "Length of fail 1152\n",
      "G001881495 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Legionellales; f__Coxiellaceae; g__Rickettsiella; s__Candidatus Rickettsiella isopodorum\n",
      "Coverage Percentage: 0.1692506174500868%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 139\n",
      "Length of fail 1153\n",
      "G001881565 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Eisenbergiella; s__Eisenbergiella tayi\n",
      "Coverage Percentage: 0.0166237506543285%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 139\n",
      "Length of fail 1154\n",
      "G001885765 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Enterococcaceae; g__Enterococcus; s__Enterococcus aquimarinus\n",
      "Coverage Percentage: 0.200292042827643%\n",
      "Adj, read counts 30\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2]\n",
      "Number of bins 28\n",
      "total # samples 19\n",
      "Number of total reads 30\n",
      "Number of hits in BIOM tables 52.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 26.666666666666668 %\n",
      "Percentage of reads from samples who only fall into one bin group 33.33333333333333 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 140\n",
      "Length of fail 1154\n",
      "G001885885 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Enterococcaceae; g__Enterococcus; s__Enterococcus cecorum\n",
      "Coverage Percentage: 0.0276971860608086%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 140\n",
      "Length of fail 1155\n",
      "G001885995 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Enterococcaceae; g__Enterococcus; s__Enterococcus italicus\n",
      "Coverage Percentage: 0.014196070097613%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 140\n",
      "Length of fail 1156\n",
      "G001886915 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Leuconostoc; s__Leuconostoc mesenteroides\n",
      "Coverage Percentage: 0.0063508800757266%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 140\n",
      "Length of fail 1157\n",
      "G001887595 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Rhodanobacteraceae; g__Luteibacter; s__Luteibacter rhizovicinus\n",
      "Coverage Percentage: 0.0107649041461878%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 140\n",
      "Length of fail 1158\n",
      "G001889125 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Intrasporangiaceae; g__Janibacter; s__Janibacter indicus\n",
      "Coverage Percentage: 0.0140833102776705%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 140\n",
      "Length of fail 1159\n",
      "G001890385 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Tardibacter; s__Tardibacter chloracetimidivorans\n",
      "Coverage Percentage: 0.0300417945246849%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 140\n",
      "Length of fail 1160\n",
      "G001890805 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Eurotiales; f__Aspergillaceae; g__Aspergillus; s__Aspergillus glaucus\n",
      "Coverage Percentage: 0.0079228484590559%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 140\n",
      "Length of fail 1161\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G001894785 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Rhodococcus; s__Rhodococcus fascians\n",
      "Coverage Percentage: 0.0407685477408526%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 140\n",
      "Length of fail 1162\n",
      "G001905915 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Shigella; s__Shigella boydii\n",
      "Coverage Percentage: 0.9022378278609432%\n",
      "Adj, read counts 227\n",
      "Number of samples per group [1, 2, 1, 3, 1, 3, 2, 2, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 2, 2, 1, 2, 2, 1, 1, 2, 1, 3, 2, 1, 2, 1, 4, 1, 1, 2, 2, 1, 1, 3, 2, 1, 1, 2, 2, 1, 1, 2, 3, 1, 1, 2, 4, 2, 1, 3, 1, 1, 2, 1, 1, 1, 1, 1, 2, 2, 1, 4, 4, 2, 1, 2, 2, 2, 2, 4, 1, 1, 1, 1, 3, 6, 2, 1, 2, 1, 1, 2, 2, 2, 2, 1, 2, 1, 2, 2]\n",
      "Number of bins 155\n",
      "total # samples 52\n",
      "Number of total reads 225\n",
      "Number of hits in BIOM tables 38.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 8.370044052863436 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.555555555555555 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 141\n",
      "Length of fail 1162\n",
      "G001912155 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium glutamicum\n",
      "Coverage Percentage: 0.0113662857582277%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 141\n",
      "Length of fail 1163\n",
      "G001922385 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas koreensis\n",
      "Coverage Percentage: 0.040296659776256%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 141\n",
      "Length of fail 1164\n",
      "G001932995 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Shigella; s__Shigella dysenteriae\n",
      "Coverage Percentage: 0.902861994907571%\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Adj, read counts 231\n",
      "Number of samples per group [1, 1, 5, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 1, 1, 1, 1, 2, 2, 1, 2, 2, 2, 1, 3, 3, 1, 2, 5, 1, 1, 1, 2, 1, 4, 1, 1, 2, 1, 2, 2, 1, 2, 1, 3, 1, 2, 1, 1, 2, 1, 1, 3, 3, 2, 3, 1, 1, 1, 1, 2, 1, 3, 1, 2, 1, 2, 3, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 4, 2, 4, 1, 2, 1, 3, 1, 1, 5, 2, 1, 1, 3, 1, 1, 3, 1]\n",
      "Number of bins 155\n",
      "total # samples 53\n",
      "Number of total reads 227\n",
      "Number of hits in BIOM tables 40.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 8.658008658008658 %\n",
      "Percentage of reads from samples who only fall into one bin group 12.77533039647577 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 142\n",
      "Length of fail 1164\n",
      "G001936175 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium phyllosphaerae\n",
      "Coverage Percentage: 0.0522426007063199%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 142\n",
      "Length of fail 1165\n",
      "G001938565 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Rhodoferax; s__Rhodoferax antarcticus\n",
      "Coverage Percentage: 0.0351830378831489%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 142\n",
      "Length of fail 1166\n",
      "G001941945 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Roseomonas; s__Roseomonas gilardii\n",
      "Coverage Percentage: 0.01104841107036%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 142\n",
      "Length of fail 1167\n",
      "G001942545 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycobacteroides; s__Mycobacteroides chelonae\n",
      "Coverage Percentage: 0.0150533290993362%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 142\n",
      "Length of fail 1168\n",
      "G001942625 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycobacterium; s__Mycobacterium syngnathidarum\n",
      "Coverage Percentage: 0.0049477422301932%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 142\n",
      "Length of fail 1169\n",
      "G001953355 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Polynucleobacter; s__Polynucleobacter sphagniphilus\n",
      "Coverage Percentage: 0.0889483727392084%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 142\n",
      "Length of fail 1170\n",
      "G001955695 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Rhodoferax; s__Rhodoferax koreense\n",
      "Coverage Percentage: 0.1708538843943557%\n",
      "Adj, read counts 68\n",
      "Number of samples per group [9, 1, 1, 1, 1, 10, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 13, 1]\n",
      "Number of bins 24\n",
      "total # samples 27\n",
      "Number of total reads 68\n",
      "Number of hits in BIOM tables 14.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 76.47058823529413 %\n",
      "Percentage of reads from samples who only fall into one bin group 22.058823529411764 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Fail - too many reads in top 3 bins\n",
      "----------------------------\n",
      "Length of pass 142\n",
      "Length of fail 1171\n",
      "G001955715 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Rhodoferax; s__Rhodoferax saidenbachensis\n",
      "Coverage Percentage: 0.1372379600244322%\n",
      "Adj, read counts 47\n",
      "Number of samples per group [1, 1, 1, 2, 17, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 18\n",
      "total # samples 25\n",
      "Number of total reads 45\n",
      "Number of hits in BIOM tables 11.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 72.3404255319149 %\n",
      "Percentage of reads from samples who only fall into one bin group 62.22222222222222 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 143\n",
      "Length of fail 1171\n",
      "G001955735 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Rhodoferax; s__Rhodoferax antarcticus\n",
      "Coverage Percentage: 0.087659253808924%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 143\n",
      "Length of fail 1172\n",
      "G001956585 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces naeslundii\n",
      "Coverage Percentage: 0.0604369017595918%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 143\n",
      "Length of fail 1173\n",
      "G001956985 d__Bacteria; p__Planctomycetes; c__Planctomycetia; o__Isosphaerales; f__Isosphaeraceae; g__Paludisphaera; s__Paludisphaera borealis\n",
      "Coverage Percentage: 0.0037245476177532%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 143\n",
      "Length of fail 1174\n",
      "G001969855 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__Bacillus haynesii\n",
      "Coverage Percentage: 0.0256163272351101%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 143\n",
      "Length of fail 1175\n",
      "G001974985 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium aurum\n",
      "Coverage Percentage: 0.0893750810244527%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 143\n",
      "Length of fail 1176\n",
      "G001982605 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Alkanindiges; s__Alkanindiges hydrocarboniclasticus\n",
      "Coverage Percentage: 0.1363874008349939%\n",
      "Adj, read counts 30\n",
      "Number of samples per group [3, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 2, 6, 1]\n",
      "Number of bins 19\n",
      "total # samples 22\n",
      "Number of total reads 30\n",
      "Number of hits in BIOM tables 16.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 40.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 53.333333333333336 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 144\n",
      "Length of fail 1176\n",
      "G001982615 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Roseomonas; s__Roseomonas deserti\n",
      "Coverage Percentage: 0.0021559358023003%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 144\n",
      "Length of fail 1177\n",
      "G001996285 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Anoxybacillus; s__Anoxybacillus karvacharensis\n",
      "Coverage Percentage: 0.0103924146389829%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 144\n",
      "Length of fail 1178\n",
      "G001997165 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus saprophyticus\n",
      "Coverage Percentage: 0.2961571184930399%\n",
      "Adj, read counts 42\n",
      "Number of samples per group [1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 2, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 37\n",
      "total # samples 22\n",
      "Number of total reads 42\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 23.809523809523807 %\n",
      "Percentage of reads from samples who only fall into one bin group 35.714285714285715 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 145\n",
      "Length of fail 1178\n",
      "G001997185 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas maltophilia\n",
      "Coverage Percentage: 0.3227745778593268%\n",
      "Adj, read counts 99\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 3, 2, 1, 1, 2, 1, 2, 1, 1, 2, 2, 1, 1, 1, 2, 1, 1, 2, 1, 2, 2, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 2, 1, 3, 1, 1, 1, 1]\n",
      "Number of bins 80\n",
      "total # samples 27\n",
      "Number of total reads 97\n",
      "Number of hits in BIOM tables 20.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.121212121212121 %\n",
      "Percentage of reads from samples who only fall into one bin group 14.432989690721648 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 146\n",
      "Length of fail 1178\n",
      "G001997345 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Tessaracoccus; s__Tessaracoccus aquimaris\n",
      "Coverage Percentage: 0.010623430153107%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 146\n",
      "Length of fail 1179\n",
      "G001998865 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Tessaracoccus; s__Tessaracoccus flavescens\n",
      "Coverage Percentage: 0.0273639553981222%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 146\n",
      "Length of fail 1180\n",
      "G001999385 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium jeikeium\n",
      "Coverage Percentage: 0.0397269496929429%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 146\n",
      "Length of fail 1181\n",
      "G001999465 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Fictibacillus; s__Fictibacillus arsenicus\n",
      "Coverage Percentage: 0.0102263526460559%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 146\n",
      "Length of fail 1182\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G001999685 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas hydrophila\n",
      "Coverage Percentage: 0.0323536015567364%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 146\n",
      "Length of fail 1183\n",
      "G002008165 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium laguerreae\n",
      "Coverage Percentage: 0.0162755927551765%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 146\n",
      "Length of fail 1184\n",
      "G002008215 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Agrobacterium; s__Agrobacterium tumefaciens\n",
      "Coverage Percentage: 0.0027495338538631%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 146\n",
      "Length of fail 1185\n",
      "G002008275 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium pusense\n",
      "Coverage Percentage: 0.0712838548518071%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 146\n",
      "Length of fail 1186\n",
      "G002008365 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium leguminosarum\n",
      "Coverage Percentage: 0.0156289360342592%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 146\n",
      "Length of fail 1187\n",
      "G002013685 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycobacteroides; s__Mycobacteroides salmoniphilum\n",
      "Coverage Percentage: 0.0057472373407658%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 146\n",
      "Length of fail 1188\n",
      "G002014755 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus mitis\n",
      "Coverage Percentage: 0.0779160487040234%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 146\n",
      "Length of fail 1189\n",
      "G002014885 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus oralis\n",
      "Coverage Percentage: 0.0511936281206913%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 146\n",
      "Length of fail 1190\n",
      "G002017865 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Rhodoferax; s__Rhodoferax fermentans\n",
      "Coverage Percentage: 0.1859544411619153%\n",
      "Adj, read counts 63\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 18]\n",
      "Number of bins 10\n",
      "total # samples 21\n",
      "Number of total reads 63\n",
      "Number of hits in BIOM tables 6.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 93.65079365079364 %\n",
      "Percentage of reads from samples who only fall into one bin group 82.53968253968253 %\n",
      "Status: Fail -- over 75% of reads are part of a sample which only hit a single bin\n",
      "----------------------------\n",
      "Length of pass 146\n",
      "Length of fail 1191\n",
      "G002019595 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Priestia; s__Priestia abyssalis\n",
      "Coverage Percentage: 0.0024419834100999%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 146\n",
      "Length of fail 1192\n",
      "G002019635 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Metabacillus; s__Metabacillus halosaccharovorans\n",
      "Coverage Percentage: 0.0025928677246678%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 146\n",
      "Length of fail 1193\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G002023205 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Elizabethkingia; s__Elizabethkingia bruuniana\n",
      "Coverage Percentage: 0.0470094296994725%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 146\n",
      "Length of fail 1194\n",
      "G002024805 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Elizabethkingia; s__Elizabethkingia bruuniana\n",
      "Coverage Percentage: 0.0465092989275794%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 146\n",
      "Length of fail 1195\n",
      "G002028325 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas veronii\n",
      "Coverage Percentage: 0.157541236009934%\n",
      "Adj, read counts 67\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 2, 1, 1, 1, 2, 1, 2, 3, 1, 1, 2, 1]\n",
      "Number of bins 63\n",
      "total # samples 29\n",
      "Number of total reads 66\n",
      "Number of hits in BIOM tables 46.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.940298507462686 %\n",
      "Percentage of reads from samples who only fall into one bin group 24.242424242424242 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 147\n",
      "Length of fail 1195\n",
      "G002028545 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Nitrobacter; s__Nitrobacter vulgaris\n",
      "Coverage Percentage: 0.0038693611428504%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 147\n",
      "Length of fail 1196\n",
      "G002029255 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Clostridium; s__Clostridium chromiireducens\n",
      "Coverage Percentage: 0.0026965312702239%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 147\n",
      "Length of fail 1197\n",
      "G002042885 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Citrobacter; s__Citrobacter portucalensis\n",
      "Coverage Percentage: 0.0165951947962675%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 147\n",
      "Length of fail 1198\n",
      "G002067135 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Cytophagaceae; g__Spirosoma; s__Spirosoma rigui\n",
      "Coverage Percentage: 0.0078065967973393%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 147\n",
      "Length of fail 1199\n",
      "G002068095 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium sacchari\n",
      "Coverage Percentage: 0.0076665274354787%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 147\n",
      "Length of fail 1200\n",
      "G002072235 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Eurotiales; f__Aspergillaceae; g__Penicillium; s__Penicillium solitum\n",
      "Coverage Percentage: 0.0012821502176826%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 147\n",
      "Length of fail 1201\n",
      "G002074095 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__Bacillus licheniformis\n",
      "Coverage Percentage: 0.0277951385450638%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 147\n",
      "Length of fail 1202\n",
      "G002075885 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Phyllobacteriaceae; g__Pseudaminobacter; s__Pseudaminobacter manganicus\n",
      "Coverage Percentage: 0.0025598060492758%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 147\n",
      "Length of fail 1203\n",
      "G002076835 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus pneumoniae\n",
      "Coverage Percentage: 0.0421002403473478%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 147\n",
      "Length of fail 1204\n",
      "G002080435 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium herbicidovorans\n",
      "Coverage Percentage: 0.1227104387346205%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 147\n",
      "Length of fail 1205\n",
      "G002086155 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycobacterium; s__Mycobacterium angelicum\n",
      "Coverage Percentage: 0.0035514178623731%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 147\n",
      "Length of fail 1206\n",
      "G002086275 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycobacterium; s__Mycobacterium intermedium\n",
      "Coverage Percentage: 0.0079992639505112%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 147\n",
      "Length of fail 1207\n",
      "G002086405 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacter; s__Mycolicibacter minnesotensis\n",
      "Coverage Percentage: 0.0135115714912525%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 147\n",
      "Length of fail 1208\n",
      "G002087075 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus pseudopneumoniae\n",
      "Coverage Percentage: 0.0697229478641598%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 147\n",
      "Length of fail 1209\n",
      "G002087235 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas floridensis\n",
      "Coverage Percentage: 0.0157297893894141%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 147\n",
      "Length of fail 1210\n",
      "G002087975 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus epidermidis\n",
      "Coverage Percentage: 0.3364196890538308%\n",
      "Adj, read counts 48\n",
      "Number of samples per group [1, 1, 1, 1, 2, 2, 1, 3, 1, 1, 1, 1, 1, 4, 1, 1, 2, 1, 1, 3, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 43\n",
      "total # samples 19\n",
      "Number of total reads 45\n",
      "Number of hits in BIOM tables 19.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 25.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 22.22222222222222 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 148\n",
      "Length of fail 1210\n",
      "G002088015 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactobacillus; s__Lactobacillus crispatus\n",
      "Coverage Percentage: 0.0974372883188182%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 148\n",
      "Length of fail 1211\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G002091475 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas azotifigens\n",
      "Coverage Percentage: 0.0111478303292455%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 148\n",
      "Length of fail 1212\n",
      "G002091495 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas alcaliphila\n",
      "Coverage Percentage: 0.0627459606935878%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 148\n",
      "Length of fail 1213\n",
      "G002091555 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas citronellolis\n",
      "Coverage Percentage: 0.5305674309630596%\n",
      "Adj, read counts 229\n",
      "Number of samples per group [1, 1, 1, 1, 2, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 3, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 2, 3, 2, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 3, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 2, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 143\n",
      "total # samples 14\n",
      "Number of total reads 223\n",
      "Number of hits in BIOM tables 78.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 7.423580786026202 %\n",
      "Percentage of reads from samples who only fall into one bin group 3.1390134529147984 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 149\n",
      "Length of fail 1213\n",
      "G002091575 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas flavescens\n",
      "Coverage Percentage: 0.0025808524675342%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 149\n",
      "Length of fail 1214\n",
      "G002091595 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas fluorescens\n",
      "Coverage Percentage: 0.0369011515704197%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 149\n",
      "Length of fail 1215\n",
      "G002091615 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas fragi\n",
      "Coverage Percentage: 0.1948652606143902%\n",
      "Adj, read counts 63\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 2, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 2, 1, 3, 2, 2, 1, 1, 1, 1, 1, 1, 3]\n",
      "Number of bins 49\n",
      "total # samples 20\n",
      "Number of total reads 61\n",
      "Number of hits in BIOM tables 26.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.698412698412698 %\n",
      "Percentage of reads from samples who only fall into one bin group 24.59016393442623 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 150\n",
      "Length of fail 1215\n",
      "G002091695 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas mandelii\n",
      "Coverage Percentage: 0.0183373085826834%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 150\n",
      "Length of fail 1216\n",
      "G002091775 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas oleovorans\n",
      "Coverage Percentage: 0.5008420316355726%\n",
      "Adj, read counts 145\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 1, 1, 4, 1, 2, 2, 1, 1, 3, 5, 2, 3, 1, 1, 1, 1, 2, 1, 1, 1, 1, 3, 1, 2, 1, 2, 2, 2, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 3, 2, 2, 2, 2, 1, 3, 3, 2, 1, 1, 2, 1, 1, 8, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 3, 1, 2, 1, 1]\n",
      "Number of bins 117\n",
      "total # samples 67\n",
      "Number of total reads 143\n",
      "Number of hits in BIOM tables 35.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.724137931034482 %\n",
      "Percentage of reads from samples who only fall into one bin group 25.874125874125873 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 151\n",
      "Length of fail 1216\n",
      "G002091795 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas synxantha\n",
      "Coverage Percentage: 0.0460561844090805%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 151\n",
      "Length of fail 1217\n",
      "G002091815 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas oleovorans\n",
      "Coverage Percentage: 0.4540131375498721%\n",
      "Adj, read counts 140\n",
      "Number of samples per group [1, 2, 1, 1, 1, 2, 2, 1, 1, 1, 6, 1, 2, 1, 1, 1, 2, 2, 2, 2, 1, 2, 2, 1, 1, 1, 1, 2, 1, 5, 2, 1, 1, 2, 2, 1, 1, 3, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 4, 2, 1, 3, 1, 2, 2, 3, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 13]\n",
      "Number of bins 113\n",
      "total # samples 66\n",
      "Number of total reads 135\n",
      "Number of hits in BIOM tables 20.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 29.629629629629626 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 152\n",
      "Length of fail 1217\n",
      "G002095385 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Erwiniaceae; g__Pantoea; s__Pantoea eucrina\n",
      "Coverage Percentage: 0.0054948779520846%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 152\n",
      "Length of fail 1218\n",
      "G002102055 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycobacteroides; s__Mycobacteroides chelonae\n",
      "Coverage Percentage: 0.0150878048537746%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 152\n",
      "Length of fail 1219\n",
      "G002102115 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacterium; s__Mycobacterium doricum\n",
      "Coverage Percentage: 0.010218630637731%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 152\n",
      "Length of fail 1220\n",
      "G002102255 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycobacterium; s__Mycobacterium nebraskense\n",
      "Coverage Percentage: 0.0018719533402258%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 152\n",
      "Length of fail 1221\n",
      "G002102565 d__Eukaryota; p__Basidiomycota; c__Tremellomycetes; o__Tremellales; f__Cuniculitremaceae; g__Kockovaella; s__Kockovaella imperatae\n",
      "Coverage Percentage: 0.0047920433530494%\n",
      "Adj, read counts 125\n",
      "Number of samples per group [4, 56, 45, 1]\n",
      "Number of bins 4\n",
      "total # samples 67\n",
      "Number of total reads 125\n",
      "Number of hits in BIOM tables 72.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 99.2 %\n",
      "Percentage of reads from samples who only fall into one bin group 28.000000000000004 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Fail - too many reads in top 3 bins\n",
      "----------------------------\n",
      "Length of pass 152\n",
      "Length of fail 1222\n",
      "G002105155 d__Eukaryota; p__Mucoromycota; c__Mortierellomycetes; o__Mortierellales; f__Mortierellaceae; g__Lobosporangium; s__Lobosporangium transversale\n",
      "Coverage Percentage: 0.0010170278756118%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 152\n",
      "Length of fail 1223\n",
      "G002116905 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Rhizobacter; s__Rhizobacter gummiphilus\n",
      "Coverage Percentage: 0.0994202025102468%\n",
      "Adj, read counts 47\n",
      "Number of samples per group [3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 12, 1]\n",
      "Number of bins 19\n",
      "total # samples 18\n",
      "Number of total reads 46\n",
      "Number of hits in BIOM tables 12.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 74.46808510638299 %\n",
      "Percentage of reads from samples who only fall into one bin group 45.65217391304348 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 153\n",
      "Length of fail 1223\n",
      "G002119215 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Pseudonocardia; s__Pseudonocardia autotrophica\n",
      "Coverage Percentage: 0.022994363865207%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 153\n",
      "Length of fail 1224\n",
      "G002119765 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__; g__Pseudorhodoplanes; s__Pseudorhodoplanes sinuspersici\n",
      "Coverage Percentage: 0.0046721029403268%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 153\n",
      "Length of fail 1225\n",
      "G002135195 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter terrae\n",
      "Coverage Percentage: 0.4900882591654775%\n",
      "Adj, read counts 106\n",
      "Number of samples per group [1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 3, 1, 4, 3, 2, 4, 2, 2, 1, 1, 1, 1, 1, 3, 1, 1, 2, 2, 1, 3, 1, 6, 3, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 4, 1, 4, 1, 1]\n",
      "Number of bins 80\n",
      "total # samples 32\n",
      "Number of total reads 102\n",
      "Number of hits in BIOM tables 53.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.094339622641511 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.76470588235294 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 154\n",
      "Length of fail 1225\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G002149965 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium panipatense\n",
      "Coverage Percentage: 0.018894629323628%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 154\n",
      "Length of fail 1226\n",
      "G002154655 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium kefirresidentii\n",
      "Coverage Percentage: 0.461109323278004%\n",
      "Adj, read counts 70\n",
      "Number of samples per group [1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 2, 2, 3, 1, 1, 1, 1, 1, 1, 1, 4, 2, 1, 1, 1, 1, 2, 1, 1, 1, 3, 2, 1, 2, 4, 2, 1, 1, 1, 1, 3]\n",
      "Number of bins 56\n",
      "total # samples 36\n",
      "Number of total reads 69\n",
      "Number of hits in BIOM tables 63.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.142857142857142 %\n",
      "Percentage of reads from samples who only fall into one bin group 30.434782608695656 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 155\n",
      "Length of fail 1226\n",
      "G002155425 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Erythrobacter; s__Erythrobacter donghaensis\n",
      "Coverage Percentage: 0.0422834109369333%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 155\n",
      "Length of fail 1227\n",
      "G002155655 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Erythrobacter; s__Erythrobacter sanguineus\n",
      "Coverage Percentage: 0.0412993087085239%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 155\n",
      "Length of fail 1228\n",
      "G002155685 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Erythrobacter; s__Erythrobacter colymbi\n",
      "Coverage Percentage: 0.0800190245288819%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 155\n",
      "Length of fail 1229\n",
      "G002155905 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces tricolor\n",
      "Coverage Percentage: 0.0013891279393608%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 155\n",
      "Length of fail 1230\n",
      "G002157145 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax carolinensis\n",
      "Coverage Percentage: 0.2808162275249386%\n",
      "Adj, read counts 87\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 9, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 10, 2, 2, 1, 7, 1, 9, 1, 1, 1, 2, 1]\n",
      "Number of bins 51\n",
      "total # samples 31\n",
      "Number of total reads 86\n",
      "Number of hits in BIOM tables 30.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 37.93103448275862 %\n",
      "Percentage of reads from samples who only fall into one bin group 13.953488372093023 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 156\n",
      "Length of fail 1230\n",
      "G002157165 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax carolinensis\n",
      "Coverage Percentage: 0.2242706644557487%\n",
      "Adj, read counts 66\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 2, 1, 8, 1, 1, 2, 1, 1, 4, 1, 1, 7, 1, 1, 6, 1, 1, 1, 3]\n",
      "Number of bins 37\n",
      "total # samples 26\n",
      "Number of total reads 64\n",
      "Number of hits in BIOM tables 17.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 37.878787878787875 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.75 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 157\n",
      "Length of fail 1230\n",
      "G002157705 d__Eukaryota; p__Apicomplexa; c__Aconoidasida; o__Haemosporida; f__Plasmodiidae; g__Plasmodium; s__Plasmodium gonderi\n",
      "Coverage Percentage: 0.0296900964449652%\n",
      "Adj, read counts 114\n",
      "Number of samples per group [2, 1, 2, 2, 2, 1, 1, 4, 1, 1, 1, 2, 1, 2, 2, 3, 5, 1, 1, 1, 3, 1, 1, 1, 2, 2, 2, 2, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 2, 3, 2, 2, 3, 1, 1, 1, 2, 6, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 5]\n",
      "Number of bins 71\n",
      "total # samples 41\n",
      "Number of total reads 110\n",
      "Number of hits in BIOM tables 10.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.035087719298245 %\n",
      "Percentage of reads from samples who only fall into one bin group 22.727272727272727 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 158\n",
      "Length of fail 1230\n",
      "G002158995 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas caspiana\n",
      "Coverage Percentage: 0.0181826461548945%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 158\n",
      "Length of fail 1231\n",
      "G002159215 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Rikenellaceae; g__Alistipes; s__Alistipes onderdonkii\n",
      "Coverage Percentage: 0.0626352743953198%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 158\n",
      "Length of fail 1232\n",
      "G002159895 d__Bacteria; p__Actinobacteria; c__Coriobacteriia; o__Eggerthellales; f__Eggerthellaceae; g__Gordonibacter; s__Gordonibacter urolithinfaciens\n",
      "Coverage Percentage: 0.0110693375396815%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 158\n",
      "Length of fail 1233\n",
      "G002160635 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactobacillus; s__Lactobacillus gallinarum\n",
      "Coverage Percentage: 0.0122177368893048%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 158\n",
      "Length of fail 1234\n",
      "G002162115 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium silvaticum\n",
      "Coverage Percentage: 0.0255790963581664%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 158\n",
      "Length of fail 1235\n",
      "G002165375 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter chinensis\n",
      "Coverage Percentage: 0.1131258948330523%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 158\n",
      "Length of fail 1236\n",
      "G002174105 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter indicus\n",
      "Coverage Percentage: 0.1164085635289448%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 158\n",
      "Length of fail 1237\n",
      "G002174125 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter populi\n",
      "Coverage Percentage: 0.0194471160496479%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 158\n",
      "Length of fail 1238\n",
      "G002174535 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Planococcaceae; g__Solibacillus; s__Solibacillus kalamii\n",
      "Coverage Percentage: 0.0097907457765925%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 158\n",
      "Length of fail 1239\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G002192535 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Polynucleobacter; s__Polynucleobacter hirudinilacicola\n",
      "Coverage Percentage: 0.2238687061822817%\n",
      "Adj, read counts 31\n",
      "Number of samples per group [1, 1, 4, 5, 1, 1, 2, 1, 2, 1, 1, 2, 2, 1, 1, 2, 1, 1]\n",
      "Number of bins 24\n",
      "total # samples 19\n",
      "Number of total reads 30\n",
      "Number of hits in BIOM tables 30.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 35.48387096774193 %\n",
      "Percentage of reads from samples who only fall into one bin group 30.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 159\n",
      "Length of fail 1239\n",
      "G002196895 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Haematobacter; s__Haematobacter missouriensis\n",
      "Coverage Percentage: 0.0131321246095784%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 159\n",
      "Length of fail 1240\n",
      "G002196965 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus yeei\n",
      "Coverage Percentage: 0.0375363100785333%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 159\n",
      "Length of fail 1241\n",
      "G002197665 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas mucosissima\n",
      "Coverage Percentage: 0.0272517565807274%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 159\n",
      "Length of fail 1242\n",
      "G002197685 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas dokdonensis\n",
      "Coverage Percentage: 0.0104258254647311%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 159\n",
      "Length of fail 1243\n",
      "G002197815 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas oleovorans\n",
      "Coverage Percentage: 0.460554325328078%\n",
      "Adj, read counts 140\n",
      "Number of samples per group [2, 2, 1, 1, 3, 1, 1, 3, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 4, 2, 1, 2, 1, 2, 2, 2, 1, 1, 1, 1, 1, 2, 2, 2, 1, 1, 1, 1, 2, 1, 1, 5, 2, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 10, 1, 2, 2, 1, 1, 1, 5, 2, 3, 3, 1, 1, 2, 1, 5, 1, 3]\n",
      "Number of bins 114\n",
      "total # samples 67\n",
      "Number of total reads 137\n",
      "Number of hits in BIOM tables 22.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 27.73722627737226 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 160\n",
      "Length of fail 1243\n",
      "G002198095 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Deinococcales; f__Deinococcaceae; g__Deinococcus; s__Deinococcus indicus\n",
      "Coverage Percentage: 0.0116820129227232%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 160\n",
      "Length of fail 1244\n",
      "G002198115 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Robiginitalea; s__Robiginitalea sediminis\n",
      "Coverage Percentage: 0.0111726363061629%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 160\n",
      "Length of fail 1245\n",
      "G002205635 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingopyxis; s__Sphingopyxis witflariensis\n",
      "Coverage Percentage: 0.0120719646960035%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 160\n",
      "Length of fail 1246\n",
      "G002205645 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Roseateles; s__Roseateles aquatilis\n",
      "Coverage Percentage: 0.0437160112709647%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 160\n",
      "Length of fail 1247\n",
      "G002205675 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingopyxis; s__Sphingopyxis bauzanensis\n",
      "Coverage Percentage: 0.0284134604659572%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 160\n",
      "Length of fail 1248\n",
      "G002205845 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Pelomonas; s__Pelomonas puraquae\n",
      "Coverage Percentage: 0.0696834207264438%\n",
      "Adj, read counts 31\n",
      "Number of samples per group [1, 2, 6, 2, 1, 2, 1, 1, 1, 1, 2, 1, 3, 1, 1, 3]\n",
      "Number of bins 20\n",
      "total # samples 22\n",
      "Number of total reads 31\n",
      "Number of hits in BIOM tables 23.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 45.16129032258065 %\n",
      "Percentage of reads from samples who only fall into one bin group 58.06451612903226 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 161\n",
      "Length of fail 1248\n",
      "G002206625 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Polynucleobacter; s__Polynucleobacter campilacus\n",
      "Coverage Percentage: 0.1865411048334693%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 161\n",
      "Length of fail 1249\n",
      "G002206635 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Polynucleobacter; s__Polynucleobacter aenigmaticus\n",
      "Coverage Percentage: 0.1863466872428369%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 161\n",
      "Length of fail 1250\n",
      "G002207865 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Alteromonadales; f__Colwelliaceae; g__Colwellia; s__Colwellia marinimaniae\n",
      "Coverage Percentage: 0.0206849256217809%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 161\n",
      "Length of fail 1251\n",
      "G002208825 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas vesicularis\n",
      "Coverage Percentage: 0.0807047738667181%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 161\n",
      "Length of fail 1252\n",
      "G002209535 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Achromobacter; s__Achromobacter marplatensis\n",
      "Coverage Percentage: 0.0069702470647636%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 161\n",
      "Length of fail 1253\n",
      "G002210435 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora wenchangensis\n",
      "Coverage Percentage: 0.0140220342832086%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 161\n",
      "Length of fail 1254\n",
      "G002213425 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Herbaspirillum; s__Herbaspirillum aquaticum\n",
      "Coverage Percentage: 0.0166964382958796%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 161\n",
      "Length of fail 1255\n",
      "G002214645 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Diaphorobacter; s__Diaphorobacter polyhydroxybutyrativorans\n",
      "Coverage Percentage: 0.0756658744690092%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 161\n",
      "Length of fail 1256\n",
      "G002215055 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Chromobacteriaceae; g__Laribacter; s__Laribacter hongkongensis\n",
      "Coverage Percentage: 0.0883107416700542%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 161\n",
      "Length of fail 1257\n",
      "G002217195 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium aquidurense\n",
      "Coverage Percentage: 0.0631836380572969%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 161\n",
      "Length of fail 1258\n",
      "G002217205 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium branchiophilum\n",
      "Coverage Percentage: 0.0371806833380053%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 161\n",
      "Length of fail 1259\n",
      "G002217235 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium aquatile\n",
      "Coverage Percentage: 0.0576640940417686%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 161\n",
      "Length of fail 1260\n",
      "G002217275 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium frigidimaris\n",
      "Coverage Percentage: 0.0677411711994925%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 161\n",
      "Length of fail 1261\n",
      "G002217285 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium hercynium\n",
      "Coverage Percentage: 0.0944857529281176%\n",
      "Adj, read counts 30\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 25\n",
      "total # samples 15\n",
      "Number of total reads 30\n",
      "Number of hits in BIOM tables 41.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 26.666666666666668 %\n",
      "Percentage of reads from samples who only fall into one bin group 36.666666666666664 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 162\n",
      "Length of fail 1261\n",
      "G002217315 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium hibernum\n",
      "Coverage Percentage: 0.0705388466832607%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 162\n",
      "Length of fail 1262\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G002217335 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium hydatis\n",
      "Coverage Percentage: 0.0600949221992641%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 162\n",
      "Length of fail 1263\n",
      "G002217365 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium pectinovorum\n",
      "Coverage Percentage: 0.1694958767948431%\n",
      "Adj, read counts 58\n",
      "Number of samples per group [1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 4, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 56\n",
      "total # samples 29\n",
      "Number of total reads 58\n",
      "Number of hits in BIOM tables 41.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.517241379310342 %\n",
      "Percentage of reads from samples who only fall into one bin group 25.862068965517242 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 163\n",
      "Length of fail 1263\n",
      "G002217395 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium plurextorum\n",
      "Coverage Percentage: 0.0207957853874948%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 163\n",
      "Length of fail 1264\n",
      "G002217435 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium reichenbachii\n",
      "Coverage Percentage: 0.0378746597912717%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 163\n",
      "Length of fail 1265\n",
      "G002217445 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium tructae\n",
      "Coverage Percentage: 0.0376588481769626%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 163\n",
      "Length of fail 1266\n",
      "G002217735 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Parageobacillus; s__Parageobacillus galactosidasius\n",
      "Coverage Percentage: 0.0068512462680471%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 163\n",
      "Length of fail 1267\n",
      "G002220155 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Azotobacter; s__Azotobacter chroococcum\n",
      "Coverage Percentage: 0.0130397173865221%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 163\n",
      "Length of fail 1268\n",
      "G002222045 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium piscis\n",
      "Coverage Percentage: 0.1485136900857931%\n",
      "Adj, read counts 52\n",
      "Number of samples per group [2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 4, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1]\n",
      "Number of bins 45\n",
      "total # samples 22\n",
      "Number of total reads 49\n",
      "Number of hits in BIOM tables 34.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.30769230769231 %\n",
      "Percentage of reads from samples who only fall into one bin group 22.448979591836736 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 164\n",
      "Length of fail 1268\n",
      "G002222055 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium araucananum\n",
      "Coverage Percentage: 0.0454445369005063%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 164\n",
      "Length of fail 1269\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G002222595 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Blautia; s__Blautia hansenii\n",
      "Coverage Percentage: 0.0408356433848051%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 164\n",
      "Length of fail 1270\n",
      "G002222615 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides caccae\n",
      "Coverage Percentage: 0.072634939637521%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 164\n",
      "Length of fail 1271\n",
      "G002222655 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Neisseriaceae; g__Vitreoscilla; s__Vitreoscilla filiformis\n",
      "Coverage Percentage: 0.2317304059493224%\n",
      "Adj, read counts 61\n",
      "Number of samples per group [1, 1, 1, 2, 1, 6, 1, 1, 1, 2, 2, 1, 4, 1, 1, 1, 2, 1, 1, 1, 3, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 2]\n",
      "Number of bins 45\n",
      "total # samples 26\n",
      "Number of total reads 59\n",
      "Number of hits in BIOM tables 70.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 27.86885245901639 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.94915254237288 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 165\n",
      "Length of fail 1271\n",
      "G002223275 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia aenigmatica\n",
      "Coverage Percentage: 0.0234577092238971%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 165\n",
      "Length of fail 1272\n",
      "G002224225 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Polynucleobacter; s__Polynucleobacter cosmopolitanus\n",
      "Coverage Percentage: 0.0265754094133253%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 165\n",
      "Length of fail 1273\n",
      "G002233675 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae; g__Paenibacillus; s__Paenibacillus herberti\n",
      "Coverage Percentage: 0.0112659366832195%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 165\n",
      "Length of fail 1274\n",
      "G002233755 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter piscicola\n",
      "Coverage Percentage: 0.0803393372591706%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 165\n",
      "Length of fail 1275\n",
      "G002234575 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Enterocloster; s__Enterocloster bolteae\n",
      "Coverage Percentage: 0.0199574965800108%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 165\n",
      "Length of fail 1276\n",
      "G002237805 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Kocuria; s__Kocuria indica\n",
      "Coverage Percentage: 0.0147709666051035%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 165\n",
      "Length of fail 1277\n",
      "G002240565 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Clavibacter; s__Clavibacter michiganensis\n",
      "Coverage Percentage: 0.0583076607409353%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 165\n",
      "Length of fail 1278\n",
      "G002240575 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Clavibacter; s__Clavibacter michiganensis\n",
      "Coverage Percentage: 0.0810533814208784%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 165\n",
      "Length of fail 1279\n",
      "G002243135 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Finegoldia; s__Finegoldia magna\n",
      "Coverage Percentage: 0.1484544914414436%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 165\n",
      "Length of fail 1280\n",
      "G002243625 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Aeribacillus; s__Aeribacillus pallidus\n",
      "Coverage Percentage: 0.0258469522970898%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 165\n",
      "Length of fail 1281\n",
      "G002245355 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Thermoactinomycetaceae; g__Paludifilum; s__Paludifilum halophilum\n",
      "Coverage Percentage: 0.0059567660032076%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 165\n",
      "Length of fail 1282\n",
      "G002245625 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax kalamii\n",
      "Coverage Percentage: 0.0939361281034243%\n",
      "Adj, read counts 34\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 4, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 29\n",
      "total # samples 20\n",
      "Number of total reads 33\n",
      "Number of hits in BIOM tables 16.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 29.411764705882355 %\n",
      "Percentage of reads from samples who only fall into one bin group 39.39393939393939 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 166\n",
      "Length of fail 1282\n",
      "G002250055 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Sutcliffiella; s__Sutcliffiella cohnii\n",
      "Coverage Percentage: 0.0029188790443059%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 166\n",
      "Length of fail 1283\n",
      "G002250115 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Alkalihalobacillus; s__Alkalihalobacillus clausii\n",
      "Coverage Percentage: 0.0066851741211154%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 166\n",
      "Length of fail 1284\n",
      "G002251755 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingosinicellaceae; g__Sandarakinorhabdus; s__Sandarakinorhabdus cyanobacteriorum\n",
      "Coverage Percentage: 0.0213276007156343%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 166\n",
      "Length of fail 1285\n",
      "G002251835 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium cyanobacteriorum\n",
      "Coverage Percentage: 0.0266305838693868%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 166\n",
      "Length of fail 1286\n",
      "G002252525 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Brucellaceae; g__Brucella; s__Brucella pseudogrignonensis\n",
      "Coverage Percentage: 0.0128510961027017%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 166\n",
      "Length of fail 1287\n",
      "G002252535 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Brucellaceae; g__Brucella; s__Brucella lupini\n",
      "Coverage Percentage: 0.0216507855438654%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 166\n",
      "Length of fail 1288\n",
      "G002257585 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Mucilaginibacter; s__Mucilaginibacter xinganensis\n",
      "Coverage Percentage: 0.0023767580228315%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 166\n",
      "Length of fail 1289\n",
      "G002268635 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Cellvibrionales; f__Cellvibrionaceae; g__Cellvibrio; s__Cellvibrio mixtus\n",
      "Coverage Percentage: 0.0283621626128045%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 166\n",
      "Length of fail 1290\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G002272085 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Alkalihalobacillus; s__Alkalihalobacillus clausii\n",
      "Coverage Percentage: 0.0126808653386533%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 166\n",
      "Length of fail 1291\n",
      "G002273005 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium hadale\n",
      "Coverage Percentage: 0.1367274749010872%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 166\n",
      "Length of fail 1292\n",
      "G002278035 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Brucellaceae; g__Brucella; s__[Ochrobactrum] quorumnocens\n",
      "Coverage Percentage: 0.0048889785595262%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 166\n",
      "Length of fail 1293\n",
      "G002278075 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia aromaticivorans\n",
      "Coverage Percentage: 0.0096319473507429%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 166\n",
      "Length of fail 1294\n",
      "G002278135 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium ottawaense\n",
      "Coverage Percentage: 0.0092954858332148%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 166\n",
      "Length of fail 1295\n",
      "G002285125 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium hadale\n",
      "Coverage Percentage: 0.1612692148947522%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 166\n",
      "Length of fail 1296\n",
      "G002285515 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Xanthomonas; s__Xanthomonas hortorum\n",
      "Coverage Percentage: 0.020720162775161%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 166\n",
      "Length of fail 1297\n",
      "G002286955 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Oceanospirillales; f__Halomonadaceae; g__Halomonas; s__Halomonas humidisoli\n",
      "Coverage Percentage: 0.0056969798608062%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 166\n",
      "Length of fail 1298\n",
      "G002287065 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus salipaludis\n",
      "Coverage Percentage: 0.0139196931577639%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 166\n",
      "Length of fail 1299\n",
      "G002287885 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Candidatus Nanopelagicales; f__Candidatus Nanopelagicaceae; g__Candidatus Nanopelagicus; s__Candidatus Nanopelagicus limnes\n",
      "Coverage Percentage: 0.1653329111838385%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 166\n",
      "Length of fail 1300\n",
      "G002287965 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Candidatus Nanopelagicales; f__Candidatus Nanopelagicaceae; g__Candidatus Planktophila; s__Candidatus Planktophila dulcis\n",
      "Coverage Percentage: 0.5840167545910635%\n",
      "Adj, read counts 57\n",
      "Number of samples per group [2, 1, 2, 4, 2, 1, 1, 1, 4, 10, 3, 15, 2, 1, 1, 1]\n",
      "Number of bins 26\n",
      "total # samples 31\n",
      "Number of total reads 56\n",
      "Number of hits in BIOM tables 38.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 61.403508771929815 %\n",
      "Percentage of reads from samples who only fall into one bin group 30.357142857142854 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 167\n",
      "Length of fail 1300\n",
      "G002288005 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Candidatus Nanopelagicales; f__Candidatus Nanopelagicaceae; g__Candidatus Nanopelagicus; s__Candidatus Nanopelagicus hibericus\n",
      "Coverage Percentage: 0.0947601480841934%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 167\n",
      "Length of fail 1301\n",
      "G002288065 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Candidatus Nanopelagicales; f__Candidatus Nanopelagicaceae; g__Candidatus Planktophila; s__Candidatus Planktophila sulfonica\n",
      "Coverage Percentage: 4.352475877835572%\n",
      "Adj, read counts 384\n",
      "Number of samples per group [9, 23, 15, 20, 30, 21, 3, 3, 41, 16]\n",
      "Number of bins 121\n",
      "total # samples 49\n",
      "Number of total reads 377\n",
      "Number of hits in BIOM tables 690.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 67.44791666666667 %\n",
      "Percentage of reads from samples who only fall into one bin group 4.244031830238726 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 168\n",
      "Length of fail 1301\n",
      "G002288145 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Candidatus Nanopelagicales; f__Candidatus Nanopelagicaceae; g__Candidatus Planktophila; s__Candidatus Planktophila versatilis\n",
      "Coverage Percentage: 0.280267241372812%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 168\n",
      "Length of fail 1302\n",
      "G002288185 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Candidatus Nanopelagicales; f__Candidatus Nanopelagicaceae; g__Candidatus Planktophila; s__Candidatus Planktophila vernalis\n",
      "Coverage Percentage: 0.2665681332312808%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 168\n",
      "Length of fail 1303\n",
      "G002288285 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium xenophagum\n",
      "Coverage Percentage: 0.0629622637041234%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 168\n",
      "Length of fail 1304\n",
      "G002288305 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Candidatus Nanopelagicales; f__Candidatus Nanopelagicaceae; g__Candidatus Nanopelagicus; s__Candidatus Nanopelagicus abundans\n",
      "Coverage Percentage: 0.171448785269864%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 168\n",
      "Length of fail 1305\n",
      "G002288325 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Candidatus Nanopelagicales; f__Candidatus Nanopelagicaceae; g__Candidatus Planktophila; s__Candidatus Planktophila lacus\n",
      "Coverage Percentage: 0.4029427821249382%\n",
      "Adj, read counts 38\n",
      "Number of samples per group [4, 2, 1, 1, 1, 1, 1, 2, 3, 1, 9, 1, 1, 1, 6]\n",
      "Number of bins 18\n",
      "total # samples 19\n",
      "Number of total reads 37\n",
      "Number of hits in BIOM tables 56.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 55.263157894736835 %\n",
      "Percentage of reads from samples who only fall into one bin group 27.027027027027028 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 169\n",
      "Length of fail 1305\n",
      "G002288365 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Candidatus Nanopelagicales; f__Candidatus Nanopelagicaceae; g__Candidatus Planktophila; s__Candidatus Planktophila limnetica\n",
      "Coverage Percentage: 0.2122226712512784%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 169\n",
      "Length of fail 1306\n",
      "G002303785 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Rahnella; s__Rahnella victoriana\n",
      "Coverage Percentage: 0.0095626438553272%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 169\n",
      "Length of fail 1307\n",
      "G002307495 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas abyssi\n",
      "Coverage Percentage: 0.0126725595610299%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 169\n",
      "Length of fail 1308\n",
      "G002310295 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Caulobacter; s__Caulobacter vibrioides\n",
      "Coverage Percentage: 0.0466568341349546%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 169\n",
      "Length of fail 1309\n",
      "G002310475 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Lactococcus; s__Lactococcus raffinolactis\n",
      "Coverage Percentage: 0.0957563992103914%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 169\n",
      "Length of fail 1310\n",
      "G002310495 d__Bacteria; p__Verrucomicrobia; c__Opitutae; o__Opitutales; f__Opitutaceae; g__Nibricoccus; s__Nibricoccus aquaticus\n",
      "Coverage Percentage: 0.0245008558387822%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 169\n",
      "Length of fail 1311\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G002327145 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Janthinobacterium; s__Janthinobacterium svalbardensis\n",
      "Coverage Percentage: 0.1423316700876208%\n",
      "Adj, read counts 64\n",
      "Number of samples per group [4, 1, 1, 1, 6, 1, 1, 1, 3, 2, 1, 4, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 4, 3]\n",
      "Number of bins 39\n",
      "total # samples 17\n",
      "Number of total reads 61\n",
      "Number of hits in BIOM tables 16.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 34.375 %\n",
      "Percentage of reads from samples who only fall into one bin group 6.557377049180328 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 170\n",
      "Length of fail 1311\n",
      "G002332425 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Dermabacteraceae; g__Brachybacterium; s__Brachybacterium alimentarium\n",
      "Coverage Percentage: 0.0240710374215526%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 170\n",
      "Length of fail 1312\n",
      "G002335465 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces olivochromogenes\n",
      "Coverage Percentage: 0.0088299151340364%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 170\n",
      "Length of fail 1313\n",
      "G002355275 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Rhizorhabdus; s__Rhizorhabdus dicambivorans\n",
      "Coverage Percentage: 0.0513951491136051%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 170\n",
      "Length of fail 1314\n",
      "G002355855 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium cloacae\n",
      "Coverage Percentage: 0.0732680312580947%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 170\n",
      "Length of fail 1315\n",
      "G002355975 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Lysobacter; s__Lysobacter enzymogenes\n",
      "Coverage Percentage: 0.102509472570801%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1]\n",
      "Number of bins 32\n",
      "total # samples 23\n",
      "Number of total reads 37\n",
      "Number of hits in BIOM tables 45.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 18.91891891891892 %\n",
      "Percentage of reads from samples who only fall into one bin group 48.64864864864865 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 171\n",
      "Length of fail 1315\n",
      "G002356335 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycobacteroides; s__[Mycobacterium] stephanolepidis\n",
      "Coverage Percentage: 0.0061067357295096%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 171\n",
      "Length of fail 1316\n",
      "G002362315 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia acidicola\n",
      "Coverage Percentage: 0.0045878372112643%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 171\n",
      "Length of fail 1317\n",
      "G002367995 d__Bacteria; p__Cyanobacteria; c__; o__Nostocales; f__Calotrichaceae; g__Calothrix; s__Calothrix brevissima\n",
      "Coverage Percentage: 0.0071400486517082%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 171\n",
      "Length of fail 1318\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G002370355 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Latilactobacillus; s__Latilactobacillus sakei\n",
      "Coverage Percentage: 0.0217172476116173%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 171\n",
      "Length of fail 1319\n",
      "G002370525 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter guillouiae\n",
      "Coverage Percentage: 0.1069180955757421%\n",
      "Adj, read counts 31\n",
      "Number of samples per group [1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 28\n",
      "total # samples 17\n",
      "Number of total reads 30\n",
      "Number of hits in BIOM tables 20.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 19.35483870967742 %\n",
      "Percentage of reads from samples who only fall into one bin group 33.33333333333333 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1319\n",
      "G002374835 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas ginsenosidimutans\n",
      "Coverage Percentage: 0.0329825192647896%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1320\n",
      "G002374855 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas adhaesiva\n",
      "Coverage Percentage: 0.0500407212351602%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1321\n",
      "G002407065 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Dermabacteraceae; g__Brachybacterium; s__Brachybacterium ginsengisoli\n",
      "Coverage Percentage: 0.0164927466064023%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1322\n",
      "G002412335 d__Bacteria; p__Cyanobacteria; c__; o__Oscillatoriales; f__Microcoleaceae; g__Tychonema; s__Tychonema bourrellyi\n",
      "Coverage Percentage: 0.088082658134515%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1323\n",
      "G002441695 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Lactococcus; s__Lactococcus piscium\n",
      "Coverage Percentage: 0.0089342740777647%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1324\n",
      "G002441745 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Lactococcus; s__Lactococcus raffinolactis\n",
      "Coverage Percentage: 0.0615435427332822%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1325\n",
      "G002441825 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Lactococcus; s__Lactococcus cremoris\n",
      "Coverage Percentage: 0.0088416991372645%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1326\n",
      "G002443115 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Dermabacteraceae; g__Dermabacter; s__Dermabacter jinjuensis\n",
      "Coverage Percentage: 0.0513176248318941%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1327\n",
      "G002504085 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium yanoikuyae\n",
      "Coverage Percentage: 0.0273468346807026%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1328\n",
      "G002531755 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium acidisoli\n",
      "Coverage Percentage: 0.0110299535928373%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1329\n",
      "G002554425 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Comamonas; s__Comamonas terrigena\n",
      "Coverage Percentage: 0.1521795667935183%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1330\n",
      "G002563875 d__Eukaryota; p__Apicomplexa; c__Conoidasida; o__Eucoccidiorida; f__Sarcocystidae; g__Besnoitia; s__Besnoitia besnoiti\n",
      "Coverage Percentage: 0.0023211702313962%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1331\n",
      "G002563895 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas lurida\n",
      "Coverage Percentage: 0.0524110343289106%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1332\n",
      "G002573965 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Azospirillaceae; g__Azospirillum; s__Azospirillum palustre\n",
      "Coverage Percentage: 0.0056011685437874%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1333\n",
      "G002591785 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Budviciaceae; g__Budvicia; s__Budvicia aquatica\n",
      "Coverage Percentage: 0.003380145262163%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1334\n",
      "G002622385 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Herelleviridae; g__Sepunavirus; s__Staphylococcus virus SEP1\n",
      "Coverage Percentage: 0.2151106283231376%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1335\n",
      "G002623085 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus PHL082M03\n",
      "Coverage Percentage: 2.356651181716456%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1336\n",
      "G002623105 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus PHL117M00\n",
      "Coverage Percentage: 2.679883780550333%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1337\n",
      "G002623125 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus PHL117M01\n",
      "Coverage Percentage: 3.089524845353817%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1338\n",
      "G002623185 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus PHL163M00\n",
      "Coverage Percentage: 2.163067249863313%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1339\n",
      "G002623205 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Pahexavirus; s__Propionibacterium virus PHL194M00\n",
      "Coverage Percentage: 2.163067249863313%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1340\n",
      "G002631185 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Roseomonas; s__Roseomonas rhizosphaerae\n",
      "Coverage Percentage: 0.0381658485457995%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1341\n",
      "G002706425 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Loigolactobacillus; s__Loigolactobacillus coryniformis\n",
      "Coverage Percentage: 0.0496017628954419%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 172\n",
      "Length of fail 1342\n",
      "G002735225 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas salmonicida\n",
      "Coverage Percentage: 0.1258615953361186%\n",
      "Adj, read counts 41\n",
      "Number of samples per group [1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 2, 1, 2, 2, 1, 1, 1, 1, 1, 6]\n",
      "Number of bins 25\n",
      "total # samples 17\n",
      "Number of total reads 41\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 26.82926829268293 %\n",
      "Percentage of reads from samples who only fall into one bin group 36.58536585365854 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 173\n",
      "Length of fail 1342\n",
      "G002736145 d__Bacteria; p__Actinobacteria; c__Coriobacteriia; o__Coriobacteriales; f__Coriobacteriaceae; g__Collinsella; s__Collinsella aerofaciens\n",
      "Coverage Percentage: 0.1311163228115086%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 173\n",
      "Length of fail 1343\n",
      "G002741095 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas toyotomiensis\n",
      "Coverage Percentage: 0.1250704525804364%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 173\n",
      "Length of fail 1344\n",
      "G002741105 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas sediminis\n",
      "Coverage Percentage: 0.0858102198088552%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 173\n",
      "Length of fail 1345\n",
      "G002749495 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus yeei\n",
      "Coverage Percentage: 0.0236129553096143%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 173\n",
      "Length of fail 1346\n",
      "G002750975 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas indicatrix\n",
      "Coverage Percentage: 0.0977721678153854%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 173\n",
      "Length of fail 1347\n",
      "G002752675 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia violaceinigra\n",
      "Coverage Percentage: 0.0861345544980338%\n",
      "Adj, read counts 45\n",
      "Number of samples per group [5, 1, 1, 1, 2, 5, 5, 4, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 21\n",
      "total # samples 20\n",
      "Number of total reads 45\n",
      "Number of hits in BIOM tables 11.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 48.88888888888889 %\n",
      "Percentage of reads from samples who only fall into one bin group 31.11111111111111 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 174\n",
      "Length of fail 1347\n",
      "G002759055 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium frigidaeris\n",
      "Coverage Percentage: 0.0047287862216858%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 174\n",
      "Length of fail 1348\n",
      "G002761875 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter junii\n",
      "Coverage Percentage: 0.1390182712769273%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 174\n",
      "Length of fail 1349\n",
      "G002762215 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Rhizobacter; s__Rhizobacter gummiphilus\n",
      "Coverage Percentage: 0.1004829558775261%\n",
      "Adj, read counts 48\n",
      "Number of samples per group [3, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 10, 1]\n",
      "Number of bins 20\n",
      "total # samples 18\n",
      "Number of total reads 46\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 74.99999999999999 %\n",
      "Percentage of reads from samples who only fall into one bin group 47.82608695652174 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 175\n",
      "Length of fail 1349\n",
      "G002776695 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium nitroreducens\n",
      "Coverage Percentage: 0.0055246954496388%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 175\n",
      "Length of fail 1350\n",
      "G002797815 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Mucilaginibacter; s__Mucilaginibacter auburnensis\n",
      "Coverage Percentage: 0.0119862950089711%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 175\n",
      "Length of fail 1351\n",
      "G002802985 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Geodermatophilus; s__Geodermatophilus chilensis\n",
      "Coverage Percentage: 0.0082261890331129%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 175\n",
      "Length of fail 1352\n",
      "G002803515 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas lactitubi\n",
      "Coverage Percentage: 0.1192641586159938%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 175\n",
      "Length of fail 1353\n",
      "G002803605 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter pseudolwoffii\n",
      "Coverage Percentage: 1.65292342841394%\n",
      "Adj, read counts 307\n",
      "Number of samples per group [10, 3, 5, 1, 4, 2, 3, 1, 1, 2, 7, 4, 5, 3, 9, 2, 2, 1, 4, 3, 4, 2, 2, 1, 6, 1, 3, 5, 1, 1, 2, 5, 4, 4, 5, 2, 1, 1, 1, 3, 1, 3, 1, 1, 5, 1, 5, 4, 6, 4, 3, 3, 1, 4, 3, 2, 1, 3, 2, 1, 1, 1, 1, 2, 13, 1, 3, 2, 4, 3, 1, 4, 3, 4, 4, 3, 1, 1, 2, 10, 1, 1, 3, 3, 2, 3]\n",
      "Number of bins 181\n",
      "total # samples 46\n",
      "Number of total reads 303\n",
      "Number of hits in BIOM tables 285.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.983713355048861 %\n",
      "Percentage of reads from samples who only fall into one bin group 2.31023102310231 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1353\n",
      "G002806685 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas qingdaonensis\n",
      "Coverage Percentage: 0.0116766810424218%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1354\n",
      "G002806995 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Confluentibacter; s__Confluentibacter lentus\n",
      "Coverage Percentage: 0.0040904875562591%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1355\n",
      "G002811175 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter baumannii\n",
      "Coverage Percentage: 0.0360734501667643%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1356\n",
      "G002812985 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas lusitana\n",
      "Coverage Percentage: 0.0225154815966781%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1357\n",
      "G002813245 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium kunmingense\n",
      "Coverage Percentage: 0.0202797866875356%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1358\n",
      "G002813445 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas tolaasii\n",
      "Coverage Percentage: 0.0323283619398607%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1359\n",
      "G002813455 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas baetica\n",
      "Coverage Percentage: 0.0105029706022577%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1360\n",
      "G002827565 d__Viruses; p__Nucleocytoviricota; c__Megaviricetes; o__Algavirales; f__Phycodnaviridae; g__Chlorovirus; s__Paramecium bursaria Chlorella virus A1\n",
      "Coverage Percentage: 0.7538457769939252%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1361\n",
      "G002831565 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Cytophagaceae; g__Spirosoma; s__Spirosoma pollinicola\n",
      "Coverage Percentage: 0.0051507492407193%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1362\n",
      "G002833945 d__Viruses; p__; c__; o__; f__Polydnaviridae; g__Bracovirus; s__Diolcogaster facetosa bracovirus\n",
      "Coverage Percentage: 0.0600341420384954%\n",
      "Adj, read counts 71\n",
      "Number of samples per group [47]\n",
      "Number of bins 3\n",
      "total # samples 47\n",
      "Number of total reads 71\n",
      "Number of hits in BIOM tables 30.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 100.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 100.0 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1363\n",
      "G002834225 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Oscillospiraceae; g__Ruminococcus; s__Ruminococcus bromii\n",
      "Coverage Percentage: 0.1567075620455585%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1364\n",
      "G002835605 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas pharmacofabricae\n",
      "Coverage Percentage: 0.0203811655933361%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1365\n",
      "G002835805 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Niallia; s__Niallia nealsonii\n",
      "Coverage Percentage: 0.0084375707125954%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1366\n",
      "G002836395 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides alpinus\n",
      "Coverage Percentage: 0.0234620149108823%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1367\n",
      "G002837135 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Macromonas; s__Macromonas bipunctata\n",
      "Coverage Percentage: 0.0698427150939101%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1368\n",
      "G002843425 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Raineyaceae; g__Raineya; s__Raineya orbicola\n",
      "Coverage Percentage: 0.0049149278075223%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1369\n",
      "G002844395 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Oscillospiraceae; g__Acetivibrio; s__Acetivibrio saccincola\n",
      "Coverage Percentage: 0.0147591361688445%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1370\n",
      "G002846365 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Nocardia; s__Nocardia fluminea\n",
      "Coverage Percentage: 0.0082777337277687%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1371\n",
      "G002846995 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Pectobacteriaceae; g__Dickeya; s__Dickeya solani\n",
      "Coverage Percentage: 0.0307307882195286%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1372\n",
      "G002847405 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium glutamicum\n",
      "Coverage Percentage: 0.0157388959375557%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1373\n",
      "G002847555 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces oris\n",
      "Coverage Percentage: 0.1426722762761408%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1374\n",
      "G002848365 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Legionellales; f__Legionellaceae; g__Legionella; s__Legionella sainthelensi\n",
      "Coverage Percentage: 0.0165531222538348%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1375\n",
      "G002849795 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella jejuni\n",
      "Coverage Percentage: 0.0181956368294%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1376\n",
      "G002858845 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Caulobacter; s__Caulobacter flavus\n",
      "Coverage Percentage: 0.0280613097620616%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1377\n",
      "G002858865 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Caulobacter; s__Caulobacter vibrioides\n",
      "Coverage Percentage: 0.0370817027635573%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1378\n",
      "G002858925 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Caulobacter; s__Caulobacter zeae\n",
      "Coverage Percentage: 0.0336086785653376%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1379\n",
      "G002861525 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces urogenitalis\n",
      "Coverage Percentage: 0.0441866162676746%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1380\n",
      "G002861965 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Gardnerella; s__Gardnerella vaginalis\n",
      "Coverage Percentage: 0.0138447868111696%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1381\n",
      "G002865645 d__Eukaryota; p__Ascomycota; c__Leotiomycetes; o__Helotiales; f__Hyaloscyphaceae; g__Hyaloscypha; s__Hyaloscypha bicolor\n",
      "Coverage Percentage: 0.0005741061358631%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1382\n",
      "G002866825 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Cellvibrionales; f__Halieaceae; g__Pseudohalioglobus; s__Pseudohalioglobus lutimaris\n",
      "Coverage Percentage: 0.0060003595929784%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1383\n",
      "G002869765 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Deinococcales; f__Deinococcaceae; g__Deinococcus; s__Deinococcus planocerae\n",
      "Coverage Percentage: 0.0023628903894529%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1384\n",
      "G002871895 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Anaerococcus; s__Anaerococcus hydrogenalis\n",
      "Coverage Percentage: 0.0293943506258743%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1385\n",
      "G002871975 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces graevenitzii\n",
      "Coverage Percentage: 0.0200166868968016%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1386\n",
      "G002874535 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Mesobacillus; s__Mesobacillus jeotgali\n",
      "Coverage Percentage: 0.0430312908069089%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1387\n",
      "G002879885 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Trinickia; s__Trinickia dabaoshanensis\n",
      "Coverage Percentage: 0.0051865684789119%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1388\n",
      "G002885975 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Paucibacter; s__Paucibacter aquatile\n",
      "Coverage Percentage: 0.2040334904590355%\n",
      "Adj, read counts 100\n",
      "Number of samples per group [8, 1, 15, 1, 1, 1, 1, 1, 1, 1, 9, 1, 1]\n",
      "Number of bins 18\n",
      "total # samples 21\n",
      "Number of total reads 99\n",
      "Number of hits in BIOM tables 10.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 90.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 12.121212121212121 %\n",
      "Status: Fail - too many reads in top 3 bins\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1389\n",
      "G002890855 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas kunmingensis\n",
      "Coverage Percentage: 0.0252942719385721%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1390\n",
      "G002892575 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Porphyromonadaceae; g__Porphyromonas; s__Porphyromonas gingivalis\n",
      "Coverage Percentage: 0.0387835658526459%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 176\n",
      "Length of fail 1391\n",
      "G002893965 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Rhodococcus; s__Rhodococcus qingshengii\n",
      "Coverage Percentage: 0.1450431957980159%\n",
      "Adj, read counts 55\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 55\n",
      "total # samples 27\n",
      "Number of total reads 55\n",
      "Number of hits in BIOM tables 27.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.909090909090908 %\n",
      "Percentage of reads from samples who only fall into one bin group 25.454545454545453 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 177\n",
      "Length of fail 1391\n",
      "G002894165 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella timonensis\n",
      "Coverage Percentage: 0.0363839061854972%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 177\n",
      "Length of fail 1392\n",
      "G002894965 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Vibrionales; f__Vibrionaceae; g__Vibrio; s__Vibrio agarivorans\n",
      "Coverage Percentage: 0.0728477773463985%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 177\n",
      "Length of fail 1393\n",
      "G002895945 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Lysobacter; s__Lysobacter silvestris\n",
      "Coverage Percentage: 0.0106075006972148%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 177\n",
      "Length of fail 1394\n",
      "G002899365 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia marcescens\n",
      "Coverage Percentage: 0.0241888428863857%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 177\n",
      "Length of fail 1395\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G002899945 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium viscerum\n",
      "Coverage Percentage: 0.0161370699225034%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 177\n",
      "Length of fail 1396\n",
      "G002900365 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Escherichia; s__Escherichia marmotae\n",
      "Coverage Percentage: 0.1057118074738002%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 177\n",
      "Length of fail 1397\n",
      "G002900975 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Tabrizicola; s__Tabrizicola aquatica\n",
      "Coverage Percentage: 0.016818893585491%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 177\n",
      "Length of fail 1398\n",
      "G002901705 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus lugdunensis\n",
      "Coverage Percentage: 0.0074606312805221%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 177\n",
      "Length of fail 1399\n",
      "G002901805 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus haemolyticus\n",
      "Coverage Percentage: 0.0331013544071886%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 177\n",
      "Length of fail 1400\n",
      "G002901875 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus epidermidis\n",
      "Coverage Percentage: 0.2827789270360799%\n",
      "Adj, read counts 38\n",
      "Number of samples per group [1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 3]\n",
      "Number of bins 34\n",
      "total # samples 17\n",
      "Number of total reads 38\n",
      "Number of hits in BIOM tables 16.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 18.421052631578945 %\n",
      "Percentage of reads from samples who only fall into one bin group 23.684210526315788 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 178\n",
      "Length of fail 1400\n",
      "G002901925 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus capitis\n",
      "Coverage Percentage: 0.078360616609457%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 178\n",
      "Length of fail 1401\n",
      "G002902025 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus saprophyticus\n",
      "Coverage Percentage: 0.2895496276254422%\n",
      "Adj, read counts 42\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 3, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 36\n",
      "total # samples 22\n",
      "Number of total reads 42\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.666666666666664 %\n",
      "Percentage of reads from samples who only fall into one bin group 35.714285714285715 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 179\n",
      "Length of fail 1401\n",
      "G002902325 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus capitis\n",
      "Coverage Percentage: 0.1130173582018047%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 179\n",
      "Length of fail 1402\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G002902365 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus cohnii\n",
      "Coverage Percentage: 0.0264091678080959%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 179\n",
      "Length of fail 1403\n",
      "G002902455 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus auricularis\n",
      "Coverage Percentage: 0.0552716342161365%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 179\n",
      "Length of fail 1404\n",
      "G002902465 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus hominis\n",
      "Coverage Percentage: 0.0856251210995634%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 179\n",
      "Length of fail 1405\n",
      "G002902545 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus saprophyticus\n",
      "Coverage Percentage: 0.2872904008109823%\n",
      "Adj, read counts 43\n",
      "Number of samples per group [1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 1]\n",
      "Number of bins 36\n",
      "total # samples 22\n",
      "Number of total reads 43\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 23.255813953488374 %\n",
      "Percentage of reads from samples who only fall into one bin group 32.55813953488372 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 180\n",
      "Length of fail 1405\n",
      "G002902625 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus devriesei\n",
      "Coverage Percentage: 0.0236312054475175%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 180\n",
      "Length of fail 1406\n",
      "G002902755 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Mammaliicoccus; s__Mammaliicoccus lentus\n",
      "Coverage Percentage: 0.0074175911510099%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 180\n",
      "Length of fail 1407\n",
      "G002911015 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces populi\n",
      "Coverage Percentage: 0.0047535547676747%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 180\n",
      "Length of fail 1408\n",
      "G002915175 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Agrobacterium; s__Agrobacterium rosae\n",
      "Coverage Percentage: 0.0082893658415545%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 180\n",
      "Length of fail 1409\n",
      "G002915195 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Agrobacterium; s__Agrobacterium rosae\n",
      "Coverage Percentage: 0.0148311876893677%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 180\n",
      "Length of fail 1410\n",
      "G002917105 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Boseaceae; g__Bosea; s__Bosea psychrotolerans\n",
      "Coverage Percentage: 0.0070848279114068%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 180\n",
      "Length of fail 1411\n",
      "G002917885 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium croceum\n",
      "Coverage Percentage: 0.0049198572223504%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 180\n",
      "Length of fail 1412\n",
      "G002920895 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium alvei\n",
      "Coverage Percentage: 0.0972404936825063%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 180\n",
      "Length of fail 1413\n",
      "G002925905 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Klebsiella; s__Klebsiella michiganensis\n",
      "Coverage Percentage: 0.0246866578543855%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 180\n",
      "Length of fail 1414\n",
      "G002926165 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora chalcea\n",
      "Coverage Percentage: 0.0281881552943278%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 180\n",
      "Length of fail 1415\n",
      "G002930615 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Zhizhongheella; s__Zhizhongheella caldifontis\n",
      "Coverage Percentage: 0.7541771192415472%\n",
      "Adj, read counts 165\n",
      "Number of samples per group [2, 1, 1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 3, 4, 1, 1, 3, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 2, 1, 2, 2, 2, 1, 1, 1, 1, 1, 1, 3, 1, 1, 2, 1, 2, 2, 1, 5, 1, 1, 2, 1, 1, 1, 1, 3, 1, 1, 1]\n",
      "Number of bins 114\n",
      "total # samples 11\n",
      "Number of total reads 162\n",
      "Number of hits in BIOM tables 270.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.545454545454545 %\n",
      "Percentage of reads from samples who only fall into one bin group 4.320987654320987 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 181\n",
      "Length of fail 1415\n",
      "G002933415 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Schlegelella; s__Schlegelella thermodepolymerans\n",
      "Coverage Percentage: 0.0453823640323235%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 181\n",
      "Length of fail 1416\n",
      "G002934265 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Actinokineospora; s__Actinokineospora auranticolor\n",
      "Coverage Percentage: 0.0015415464556882%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 181\n",
      "Length of fail 1417\n",
      "G002938425 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Blastococcus; s__Blastococcus atacamensis\n",
      "Coverage Percentage: 0.0131381058700389%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 181\n",
      "Length of fail 1418\n",
      "G002939185 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Enterobacter; s__Enterobacter sichuanensis\n",
      "Coverage Percentage: 0.0061616796167353%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 181\n",
      "Length of fail 1419\n",
      "G002939985 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Xanthomonas; s__Xanthomonas hortorum\n",
      "Coverage Percentage: 0.0185181525326587%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 181\n",
      "Length of fail 1420\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G002943655 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium shigense\n",
      "Coverage Percentage: 0.0086911816612385%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 181\n",
      "Length of fail 1421\n",
      "G002943685 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas fluorescens\n",
      "Coverage Percentage: 0.0336700650158094%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 181\n",
      "Length of fail 1422\n",
      "G002949675 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Shigella; s__Shigella dysenteriae\n",
      "Coverage Percentage: 0.7705615773273177%\n",
      "Adj, read counts 204\n",
      "Number of samples per group [2, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 2, 1, 2, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 3, 1, 1, 4, 1, 1, 2, 1, 2, 1, 1, 4, 5, 1, 1, 2, 1, 2, 1, 1, 1, 2, 2, 1, 1, 1, 1, 3, 1, 4, 1, 1, 2, 1, 5, 1, 2, 1, 1, 2, 1, 2, 1, 1, 1, 3, 1, 1, 3, 2, 2, 1, 2, 2, 2, 1, 1, 2, 2, 1, 1, 1, 2, 1, 1, 2, 2, 1, 2, 1, 2, 2, 1, 2, 4, 1]\n",
      "Number of bins 143\n",
      "total # samples 50\n",
      "Number of total reads 201\n",
      "Number of hits in BIOM tables 27.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 9.313725490196079 %\n",
      "Percentage of reads from samples who only fall into one bin group 12.935323383084576 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 182\n",
      "Length of fail 1422\n",
      "G002954265 d__Bacteria; p__Bacteroidetes; c__Chitinophagia; o__Chitinophagales; f__Chitinophagaceae; g__Flavipsychrobacter; s__Flavipsychrobacter stenotrophus\n",
      "Coverage Percentage: 0.0131498396603584%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 182\n",
      "Length of fail 1423\n",
      "G002954725 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Vibrionales; f__Vibrionaceae; g__Photobacterium; s__Photobacterium phosphoreum\n",
      "Coverage Percentage: 0.005010783073311%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 182\n",
      "Length of fail 1424\n",
      "G002957165 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Siphoviridae; g__Nickievirus; s__Pseudomonas virus nickie\n",
      "Coverage Percentage: 0.1345511249721541%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 182\n",
      "Length of fail 1425\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G002959615 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Mediterraneibacter; s__[Ruminococcus] gnavus\n",
      "Coverage Percentage: 0.1218591033175307%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 182\n",
      "Length of fail 1426\n",
      "G002959635 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides ovatus\n",
      "Coverage Percentage: 0.1266041507246652%\n",
      "Adj, read counts 52\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 47\n",
      "total # samples 33\n",
      "Number of total reads 52\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.384615384615385 %\n",
      "Percentage of reads from samples who only fall into one bin group 42.30769230769231 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 183\n",
      "Length of fail 1426\n",
      "G002959675 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Enterocloster; s__Enterocloster bolteae\n",
      "Coverage Percentage: 0.0260673935435942%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 183\n",
      "Length of fail 1427\n",
      "G002959695 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides thetaiotaomicron\n",
      "Coverage Percentage: 0.1058667979073488%\n",
      "Adj, read counts 44\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 39\n",
      "total # samples 26\n",
      "Number of total reads 44\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 18.181818181818183 %\n",
      "Percentage of reads from samples who only fall into one bin group 40.909090909090914 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 184\n",
      "Length of fail 1427\n",
      "G002959715 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides caccae\n",
      "Coverage Percentage: 0.073288030917719%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 184\n",
      "Length of fail 1428\n",
      "G002959775 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Veillonellales; f__Veillonellaceae; g__Veillonella; s__Veillonella rogosae\n",
      "Coverage Percentage: 0.0251441675131869%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 184\n",
      "Length of fail 1429\n",
      "G002959895 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Veillonellales; f__Veillonellaceae; g__Veillonella; s__Veillonella infantium\n",
      "Coverage Percentage: 0.0254745815614484%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 184\n",
      "Length of fail 1430\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G002959915 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Veillonellales; f__Veillonellaceae; g__Veillonella; s__Veillonella atypica\n",
      "Coverage Percentage: 0.029397185918797%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 184\n",
      "Length of fail 1431\n",
      "G002964975 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Apibacter; s__Apibacter adventoris\n",
      "Coverage Percentage: 0.0109532799088919%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 184\n",
      "Length of fail 1432\n",
      "G002967035 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycobacterium; s__Mycobacterium virginiense\n",
      "Coverage Percentage: 0.0059769610821514%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 184\n",
      "Length of fail 1433\n",
      "G002967995 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas frederiksbergensis\n",
      "Coverage Percentage: 0.0092564709091647%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 184\n",
      "Length of fail 1434\n",
      "G002973535 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Defluviimonas; s__Defluviimonas denitrificans\n",
      "Coverage Percentage: 0.0119267115743867%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 184\n",
      "Length of fail 1435\n",
      "G002975235 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Brucellaceae; g__Brucella; s__Brucella pseudogrignonensis\n",
      "Coverage Percentage: 0.0116995906952471%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 184\n",
      "Length of fail 1436\n",
      "G002980495 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Phyllobacteriaceae; g__Phyllobacterium; s__Phyllobacterium phragmitis\n",
      "Coverage Percentage: 0.0040158016464075%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 184\n",
      "Length of fail 1437\n",
      "G002980595 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Malikia; s__Malikia granosa\n",
      "Coverage Percentage: 0.4238582676344329%\n",
      "Adj, read counts 101\n",
      "Number of samples per group [1, 1, 1, 2, 4, 1, 4, 1, 1, 3, 3, 1, 1, 1, 1, 1, 3, 2, 2, 1, 1, 1, 3, 1, 2, 2, 3, 1, 2, 3, 1, 2, 3, 1, 2, 2, 3, 3, 2, 1, 1, 1, 1, 2, 2, 1, 3, 2, 1, 1, 2, 1, 1, 3]\n",
      "Number of bins 85\n",
      "total # samples 35\n",
      "Number of total reads 101\n",
      "Number of hits in BIOM tables 89.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 13.861386138613863 %\n",
      "Percentage of reads from samples who only fall into one bin group 15.841584158415841 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 185\n",
      "Length of fail 1437\n",
      "G002980625 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Malikia; s__Malikia spinosa\n",
      "Coverage Percentage: 0.9949463128397932%\n",
      "Adj, read counts 231\n",
      "Number of samples per group [2, 3, 1, 4, 2, 4, 1, 1, 1, 7, 4, 1, 3, 1, 2, 3, 1, 1, 1, 2, 1, 5, 1, 9, 4, 5, 1, 2, 2, 5, 2, 1, 2, 1, 2, 2, 4, 2, 4, 1, 1, 1, 1, 5, 4, 2, 2, 1, 3, 2, 1, 3, 2, 3, 3, 1, 3, 1, 2, 4, 5, 3, 1, 5, 1, 2, 2, 3, 2, 6, 2, 5, 1, 1, 3, 3, 1, 2, 2, 1, 3, 3, 3, 1]\n",
      "Number of bins 169\n",
      "total # samples 43\n",
      "Number of total reads 227\n",
      "Number of hits in BIOM tables 402.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.255411255411255 %\n",
      "Percentage of reads from samples who only fall into one bin group 4.845814977973569 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 186\n",
      "Length of fail 1437\n",
      "G002990485 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Myoviridae; g__Eneladusvirus; s__Yersinia virus Yen9-04\n",
      "Coverage Percentage: 0.0603874958377777%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 186\n",
      "Length of fail 1438\n",
      "G002992955 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Aliarcobacter; s__Aliarcobacter cryaerophilus\n",
      "Coverage Percentage: 0.7466408673907848%\n",
      "Adj, read counts 96\n",
      "Number of samples per group [1, 3, 1, 1, 3, 1, 1, 2, 1, 1, 1, 2, 1, 1, 3, 2, 2, 3, 3, 4, 1, 1, 5, 1, 1, 3, 2, 3, 1, 7, 1, 2, 1, 3, 2, 2, 3, 1, 5, 4, 1]\n",
      "Number of bins 74\n",
      "total # samples 30\n",
      "Number of total reads 94\n",
      "Number of hits in BIOM tables 59.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 19.791666666666664 %\n",
      "Percentage of reads from samples who only fall into one bin group 13.829787234042554 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 187\n",
      "Length of fail 1438\n",
      "G002993885 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingopyxis; s__Sphingopyxis lindanitolerans\n",
      "Coverage Percentage: 0.0245819299719125%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 187\n",
      "Length of fail 1439\n",
      "G002993925 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__Bacillus paralicheniformis\n",
      "Coverage Percentage: 0.0202201090240861%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 187\n",
      "Length of fail 1440\n",
      "G002995745 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Clostridium; s__Clostridium vincentii\n",
      "Coverage Percentage: 0.0129140293396483%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 187\n",
      "Length of fail 1441\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G002998365 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__Bacillus pumilus\n",
      "Coverage Percentage: 0.0379105807836079%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 187\n",
      "Length of fail 1442\n",
      "G002999335 d__Eukaryota; p__Apicomplexa; c__Conoidasida; o__Eucoccidiorida; f__Eimeriidae; g__Cyclospora; s__Cyclospora cayetanensis\n",
      "Coverage Percentage: 0.0067104844627272%\n",
      "Adj, read counts 315\n",
      "Number of samples per group [2, 8, 20, 32, 5, 28, 6, 4, 46, 3, 2, 2, 32, 1, 35, 1, 3, 1, 6, 17, 15]\n",
      "Number of bins 24\n",
      "total # samples 75\n",
      "Number of total reads 269\n",
      "Number of hits in BIOM tables 106.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 35.873015873015866 %\n",
      "Percentage of reads from samples who only fall into one bin group 5.947955390334572 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 188\n",
      "Length of fail 1442\n",
      "G003001815 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes italicus\n",
      "Coverage Percentage: 0.0121927275759154%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 188\n",
      "Length of fail 1443\n",
      "G003002115 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia insulsa\n",
      "Coverage Percentage: 0.0080850934259564%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 188\n",
      "Length of fail 1444\n",
      "G003002895 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Intrasporangiaceae; g__Knoellia; s__Knoellia remsis\n",
      "Coverage Percentage: 0.0276127015069764%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 188\n",
      "Length of fail 1445\n",
      "G003003155 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium granuli\n",
      "Coverage Percentage: 0.059002422204701%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 188\n",
      "Length of fail 1446\n",
      "G003003775 d__Bacteria; p__Cyanobacteria; c__; o__Synechococcales; f__Merismopediaceae; g__Merismopedia; s__Merismopedia glauca\n",
      "Coverage Percentage: 0.0022840752355188%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 188\n",
      "Length of fail 1447\n",
      "G003003835 d__Bacteria; p__Cyanobacteria; c__; o__Chroococcidiopsidales; f__Chroococcidiopsidaceae; g__Chroococcidiopsis; s__Chroococcidiopsis cubana\n",
      "Coverage Percentage: 0.0019895890470883%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 188\n",
      "Length of fail 1448\n",
      "G003003885 d__Bacteria; p__Cyanobacteria; c__; o__Chroococcales; f__Entophysalidaceae; g__Chlorogloea; s__Chlorogloea sp. CCALA 695\n",
      "Coverage Percentage: 0.0077445721544577%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 188\n",
      "Length of fail 1449\n",
      "G003003925 d__Bacteria; p__Cyanobacteria; c__; o__Chroococcales; f__Aphanothecaceae; g__Aphanothece; s__Aphanothece minutissima\n",
      "Coverage Percentage: 0.0187624137907469%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 188\n",
      "Length of fail 1450\n",
      "G003004785 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Alsobacteraceae; g__Alsobacter; s__Alsobacter soli\n",
      "Coverage Percentage: 0.0055850095262929%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 188\n",
      "Length of fail 1451\n",
      "G003008515 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Phreatobacteraceae; g__Phreatobacter; s__Phreatobacter cathodiphilus\n",
      "Coverage Percentage: 0.0067194058443256%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 188\n",
      "Length of fail 1452\n",
      "G003008535 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Ottowia; s__Ottowia oryzae\n",
      "Coverage Percentage: 0.2020501864773003%\n",
      "Adj, read counts 46\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 4, 1, 1, 1, 2, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1]\n",
      "Number of bins 40\n",
      "total # samples 21\n",
      "Number of total reads 46\n",
      "Number of hits in BIOM tables 91.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 19.565217391304348 %\n",
      "Percentage of reads from samples who only fall into one bin group 26.08695652173913 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 189\n",
      "Length of fail 1452\n",
      "G003008595 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Simplicispira; s__Simplicispira suum\n",
      "Coverage Percentage: 0.1237697641682922%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [4, 1, 1, 1, 1, 1, 1, 1, 1, 6, 1, 7, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 20\n",
      "total # samples 15\n",
      "Number of total reads 36\n",
      "Number of hits in BIOM tables 19.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 54.054054054054056 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.666666666666664 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 190\n",
      "Length of fail 1452\n",
      "G003008695 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Trichococcus; s__Trichococcus alkaliphilus\n",
      "Coverage Percentage: 0.1206318179994004%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 190\n",
      "Length of fail 1453\n",
      "G003011895 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia glaciei\n",
      "Coverage Percentage: 0.0276229666780293%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 190\n",
      "Length of fail 1454\n",
      "G003014535 d__Bacteria; p__Bacteroidetes; c__Chitinophagia; o__Chitinophagales; f__Chitinophagaceae; g__Taibaiella; s__Taibaiella chishuiensis\n",
      "Coverage Percentage: 0.003783232746704%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 190\n",
      "Length of fail 1455\n",
      "G003014735 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Saccharothrix; s__Saccharothrix carnea\n",
      "Coverage Percentage: 0.0014575708415868%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 190\n",
      "Length of fail 1456\n",
      "G003015125 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Caulobacter; s__Caulobacter segnis\n",
      "Coverage Percentage: 0.0389005012816114%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 190\n",
      "Length of fail 1457\n",
      "G003015185 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Rhodanobacteraceae; g__Ahniella; s__Ahniella affigens\n",
      "Coverage Percentage: 0.0118118822961449%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 190\n",
      "Length of fail 1458\n",
      "G003019925 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Pluralibacter; s__Pluralibacter gergoviae\n",
      "Coverage Percentage: 0.0127032097516272%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 190\n",
      "Length of fail 1459\n",
      "G003019965 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia multivorans\n",
      "Coverage Percentage: 0.0181720441841629%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 190\n",
      "Length of fail 1460\n",
      "G003024515 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter sichuanensis\n",
      "Coverage Percentage: 0.1656729997324056%\n",
      "Adj, read counts 54\n",
      "Number of samples per group [3, 1, 1, 4, 1, 3, 2, 1, 1, 1, 1, 2, 6, 3, 1, 8, 4, 1, 1, 3, 1, 1]\n",
      "Number of bins 30\n",
      "total # samples 23\n",
      "Number of total reads 54\n",
      "Number of hits in BIOM tables 26.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 35.18518518518518 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.666666666666664 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 191\n",
      "Length of fail 1460\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G003024525 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter cumulans\n",
      "Coverage Percentage: 0.1307985258084569%\n",
      "Adj, read counts 32\n",
      "Number of samples per group [2, 6, 1, 3, 1, 1, 1, 1, 1, 1, 6, 1, 2]\n",
      "Number of bins 20\n",
      "total # samples 20\n",
      "Number of total reads 31\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 59.375 %\n",
      "Percentage of reads from samples who only fall into one bin group 51.61290322580645 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 192\n",
      "Length of fail 1460\n",
      "G003024675 d__Bacteria; p__Firmicutes; c__Erysipelotrichia; o__Erysipelotrichales; f__Erysipelotrichaceae; g__Faecalibacillus; s__Faecalibacillus faecis\n",
      "Coverage Percentage: 0.0260377816917989%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 192\n",
      "Length of fail 1461\n",
      "G003024685 d__Bacteria; p__Firmicutes; c__Erysipelotrichia; o__Erysipelotrichales; f__Erysipelotrichaceae; g__Faecalibacillus; s__Faecalibacillus intestinalis\n",
      "Coverage Percentage: 0.045943697501543%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 192\n",
      "Length of fail 1462\n",
      "G003024715 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Clostridium; s__Clostridium fessum\n",
      "Coverage Percentage: 0.0892406036509017%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 192\n",
      "Length of fail 1463\n",
      "G003025115 d__Eukaryota; p__Ascomycota; c__Sordariomycetes; o__Hypocreales; f__Hypocreaceae; g__Trichoderma; s__Trichoderma citrinoviride\n",
      "Coverage Percentage: 0.0026726217977408%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 192\n",
      "Length of fail 1464\n",
      "G003026275 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Vibrionales; f__Vibrionaceae; g__Photobacterium; s__Photobacterium phosphoreum\n",
      "Coverage Percentage: 0.0050556005188642%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 192\n",
      "Length of fail 1465\n",
      "G003028855 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia armeniaca\n",
      "Coverage Percentage: 0.1165603233203802%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 192\n",
      "Length of fail 1466\n",
      "G003029845 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax cattleyae\n",
      "Coverage Percentage: 0.0367187302545606%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 192\n",
      "Length of fail 1467\n",
      "Length of pass 193\n",
      "Length of fail 1467\n",
      "G003030465 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Zoogloeaceae; g__Thauera; s__Thauera aromatica\n",
      "Coverage Percentage: 0.0617757580394696%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 193\n",
      "Length of fail 1468\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G003030905 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium adolescentis\n",
      "Coverage Percentage: 0.5213854229192475%\n",
      "Adj, read counts 67\n",
      "Number of samples per group [2, 2, 4, 2, 1, 3, 1, 4, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 2, 2, 1, 1, 1, 1, 2, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1]\n",
      "Number of bins 59\n",
      "total # samples 30\n",
      "Number of total reads 67\n",
      "Number of hits in BIOM tables 27.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.417910447761194 %\n",
      "Percentage of reads from samples who only fall into one bin group 19.402985074626866 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 194\n",
      "Length of fail 1468\n",
      "G003031005 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas baetica\n",
      "Coverage Percentage: 0.0144234786477892%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 194\n",
      "Length of fail 1469\n",
      "G003031105 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Achromobacter; s__Achromobacter xylosoxidans\n",
      "Coverage Percentage: 0.1346038004997805%\n",
      "Adj, read counts 48\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1]\n",
      "Number of bins 40\n",
      "total # samples 10\n",
      "Number of total reads 47\n",
      "Number of hits in BIOM tables 20.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.416666666666666 %\n",
      "Percentage of reads from samples who only fall into one bin group 14.893617021276595 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 195\n",
      "Length of fail 1469\n",
      "G003031675 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas palleroniana\n",
      "Coverage Percentage: 0.060367970218468%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 195\n",
      "Length of fail 1470\n",
      "G003034225 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas fennica\n",
      "Coverage Percentage: 0.0217302530495752%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 195\n",
      "Length of fail 1471\n",
      "G003044275 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Rathayibacter; s__Rathayibacter caricis\n",
      "Coverage Percentage: 0.1590303228009725%\n",
      "Adj, read counts 38\n",
      "Number of samples per group [2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 30\n",
      "total # samples 12\n",
      "Number of total reads 37\n",
      "Number of hits in BIOM tables 100.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 21.052631578947366 %\n",
      "Percentage of reads from samples who only fall into one bin group 21.62162162162162 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 196\n",
      "Length of fail 1471\n",
      "G003046185 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Phreatobacteraceae; g__Phreatobacter; s__Phreatobacter oligotrophus\n",
      "Coverage Percentage: 0.0140111048247772%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 196\n",
      "Length of fail 1472\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G003046295 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas aerolata\n",
      "Coverage Percentage: 0.5240411520440427%\n",
      "Adj, read counts 126\n",
      "Number of samples per group [1, 1, 1, 2, 5, 4, 1, 1, 1, 2, 1, 2, 1, 2, 1, 2, 2, 2, 4, 2, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 2, 2, 1, 2, 1, 4, 1, 1, 1, 1, 1, 2, 2, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 2, 2, 1, 1, 1, 1, 7]\n",
      "Number of bins 95\n",
      "total # samples 40\n",
      "Number of total reads 124\n",
      "Number of hits in BIOM tables 204.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 13.492063492063492 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.29032258064516 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 197\n",
      "Length of fail 1472\n",
      "G003047265 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides terrigena\n",
      "Coverage Percentage: 0.0175300788330573%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 197\n",
      "Length of fail 1473\n",
      "G003047285 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides allogilvus\n",
      "Coverage Percentage: 0.0248883617027875%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 197\n",
      "Length of fail 1474\n",
      "G003049645 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Planococcaceae; g__Metasolibacillus; s__Metasolibacillus fluoroglycofenilyticus\n",
      "Coverage Percentage: 0.0364627216575236%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 197\n",
      "Length of fail 1475\n",
      "G003050615 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas oleivorans\n",
      "Coverage Percentage: 0.006504504885864%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 197\n",
      "Length of fail 1476\n",
      "G003050705 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas aurantiaca\n",
      "Coverage Percentage: 0.4385612958945658%\n",
      "Adj, read counts 125\n",
      "Number of samples per group [1, 1, 2, 2, 1, 2, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 3, 1, 3, 2, 1, 1, 2, 1, 1, 2, 2, 1, 1, 1, 8, 2, 4, 3, 1, 1, 2, 1, 3, 3, 1, 1, 1, 2, 5, 2, 1, 1, 2, 1, 2, 2, 1, 1, 2, 1, 1, 3, 1, 1, 2, 7, 5]\n",
      "Number of bins 89\n",
      "total # samples 42\n",
      "Number of total reads 125\n",
      "Number of hits in BIOM tables 149.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.6 %\n",
      "Percentage of reads from samples who only fall into one bin group 13.600000000000001 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 198\n",
      "Length of fail 1476\n",
      "G003051055 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Agitococcus; s__Agitococcus lubricus\n",
      "Coverage Percentage: 0.0173698517095675%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 198\n",
      "Length of fail 1477\n",
      "G003051085 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Trichococcus; s__Trichococcus patagoniensis\n",
      "Coverage Percentage: 0.0804820676094474%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 198\n",
      "Length of fail 1478\n",
      "G003051865 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Chromobacteriaceae; g__Microvirgula; s__Microvirgula aerodenitrificans\n",
      "Coverage Percentage: 0.0339452771119426%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 198\n",
      "Length of fail 1479\n",
      "G003052295 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Trichococcus; s__Trichococcus paludicola\n",
      "Coverage Percentage: 0.0827261068298384%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 198\n",
      "Length of fail 1480\n",
      "G003052605 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas indoloxydans\n",
      "Coverage Percentage: 0.4472857329306288%\n",
      "Adj, read counts 149\n",
      "Number of samples per group [2, 1, 1, 1, 1, 3, 3, 3, 2, 3, 4, 1, 3, 2, 2, 1, 2, 1, 1, 1, 3, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 3, 2, 10, 2, 1, 2, 1, 1, 3, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 2, 1, 1, 3, 1, 1, 1, 3, 1, 1, 1, 1, 2, 4, 1, 1, 2, 1, 1, 2, 2, 1, 1, 1, 1, 1, 2]\n",
      "Number of bins 125\n",
      "total # samples 64\n",
      "Number of total reads 147\n",
      "Number of hits in BIOM tables 25.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.751677852348994 %\n",
      "Percentage of reads from samples who only fall into one bin group 23.809523809523807 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 199\n",
      "Length of fail 1480\n",
      "G003053745 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas faeni\n",
      "Coverage Percentage: 0.6863513007396108%\n",
      "Adj, read counts 200\n",
      "Number of samples per group [1, 6, 1, 4, 3, 2, 1, 5, 3, 6, 1, 3, 6, 1, 1, 3, 6, 2, 1, 1, 1, 1, 1, 3, 3, 3, 1, 1, 1, 2, 2, 1, 2, 3, 2, 1, 1, 1, 1, 5, 2, 1, 5, 3, 1, 2, 5, 1, 2, 2, 5, 1, 3, 2, 3, 1, 3, 2, 2, 2, 2, 2, 2, 1, 2, 1, 1, 3, 2, 1, 6, 1, 3, 1, 2, 2, 2, 6]\n",
      "Number of bins 139\n",
      "total # samples 45\n",
      "Number of total reads 192\n",
      "Number of hits in BIOM tables 269.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.000000000000002 %\n",
      "Percentage of reads from samples who only fall into one bin group 4.6875 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 200\n",
      "Length of fail 1480\n",
      "G003054045 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Sphingobacterium; s__Sphingobacterium faecium\n",
      "Coverage Percentage: 0.0156874915757603%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 200\n",
      "Length of fail 1481\n",
      "G003058325 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium currus\n",
      "Coverage Percentage: 0.0052290349590037%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 200\n",
      "Length of fail 1482\n",
      "G003058385 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium leguminosarum\n",
      "Coverage Percentage: 0.013623501176141%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 200\n",
      "Length of fail 1483\n",
      "G003060865 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Gemmobacter; s__Gemmobacter aquarius\n",
      "Coverage Percentage: 0.0375595872153821%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 200\n",
      "Length of fail 1484\n",
      "G003062295 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium algeriense\n",
      "Coverage Percentage: 0.0173982650194364%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 200\n",
      "Length of fail 1485\n",
      "G003063245 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Arcobacter; s__Arcobacter caeni\n",
      "Coverage Percentage: 0.1128159965302666%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 200\n",
      "Length of fail 1486\n",
      "G003063455 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Limnohabitans; s__Limnohabitans parvus\n",
      "Coverage Percentage: 0.2461037497519115%\n",
      "Adj, read counts 53\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 1, 1, 6, 1, 2, 1, 2, 2, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 6, 1]\n",
      "Number of bins 37\n",
      "total # samples 25\n",
      "Number of total reads 53\n",
      "Number of hits in BIOM tables 41.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 35.84905660377359 %\n",
      "Percentage of reads from samples who only fall into one bin group 20.754716981132077 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 201\n",
      "Length of fail 1486\n",
      "G003063475 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Limnohabitans; s__Limnohabitans curvus\n",
      "Coverage Percentage: 0.8779648061592852%\n",
      "Adj, read counts 173\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 1, 1, 5, 8, 2, 2, 2, 3, 1, 2, 2, 1, 1, 1, 1, 1, 14, 1, 2, 7, 2, 2, 2, 2, 3, 2, 1, 1, 13, 2, 1, 5, 4, 1, 7, 2, 2, 4, 1, 1, 1, 7, 1, 3, 1, 1, 2, 1, 2, 1, 2, 1, 2, 1]\n",
      "Number of bins 105\n",
      "total # samples 43\n",
      "Number of total reads 167\n",
      "Number of hits in BIOM tables 291.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 28.90173410404624 %\n",
      "Percentage of reads from samples who only fall into one bin group 5.389221556886228 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1486\n",
      "G003064105 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Xanthomonas; s__Xanthomonas hortorum\n",
      "Coverage Percentage: 0.0386073937135378%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1487\n",
      "G003064145 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Xanthomonas; s__Xanthomonas oryzae\n",
      "Coverage Percentage: 0.0479586890837334%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1488\n",
      "G003065365 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Polynucleobacter; s__Polynucleobacter difficilis\n",
      "Coverage Percentage: 0.1368559414571274%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1489\n",
      "G003065385 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Polynucleobacter; s__Polynucleobacter acidiphobus\n",
      "Coverage Percentage: 0.0437427129230554%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1490\n",
      "G003065605 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Aeromicrobium; s__Aeromicrobium chenweiae\n",
      "Coverage Percentage: 0.0040616062280831%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1491\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G003067705 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Planctomonas; s__Planctomonas deserti\n",
      "Coverage Percentage: 0.0232280887085363%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1492\n",
      "G003070885 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Agromyces; s__Agromyces badenianii\n",
      "Coverage Percentage: 0.0197899642601109%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1493\n",
      "G003072125 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Pelagivirga; s__Pelagivirga sediminicola\n",
      "Coverage Percentage: 0.0182749868397502%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1494\n",
      "G003072645 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Dongshaea; s__Dongshaea marina\n",
      "Coverage Percentage: 0.0293417357631941%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1495\n",
      "G003073115 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Amorphaceae; g__Acuticoccus; s__Acuticoccus kandeliae\n",
      "Coverage Percentage: 0.0155963161223639%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1496\n",
      "G003075095 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Blastococcus; s__Blastococcus litoris\n",
      "Coverage Percentage: 0.015754550151481%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1497\n",
      "G003076135 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides gansuensis\n",
      "Coverage Percentage: 0.0115377248991025%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1498\n",
      "G003076475 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium kingsejongi\n",
      "Coverage Percentage: 0.0510410262371234%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1499\n",
      "G003094615 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Caulobacter; s__Caulobacter radicis\n",
      "Coverage Percentage: 0.0364363131835735%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1500\n",
      "G003096035 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium album\n",
      "Coverage Percentage: 0.0420981708256914%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1501\n",
      "G003096215 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Caldibacillus; s__Caldibacillus thermoamylovorans\n",
      "Coverage Percentage: 0.0541560740705697%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1502\n",
      "G003096415 d__Bacteria; p__Lentisphaerae; c__Lentisphaeria; o__Victivallales; f__Victivallaceae; g__Victivallis; s__Victivallis vadensis\n",
      "Coverage Percentage: 0.0019801462462963%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1503\n",
      "G003096695 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Porphyromonadaceae; g__Porphyromonas; s__Porphyromonas loveana\n",
      "Coverage Percentage: 0.0132685954752332%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1504\n",
      "G003096815 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella colorans\n",
      "Coverage Percentage: 0.0216823015235475%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1505\n",
      "G003096875 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia unamae\n",
      "Coverage Percentage: 0.0016081371741009%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1506\n",
      "G003096955 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Williamsia; s__Williamsia marianensis\n",
      "Coverage Percentage: 0.0027108780354428%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1507\n",
      "G003097315 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Oceanospirillales; f__; g__Pelagibaculum; s__Pelagibaculum spongiae\n",
      "Coverage Percentage: 0.0023064025735441%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1508\n",
      "G003097635 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium psychrotolerans\n",
      "Coverage Percentage: 0.0612964366009136%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 202\n",
      "Length of fail 1509\n",
      "G003100395 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Aquabacterium; s__Aquabacterium olei\n",
      "Coverage Percentage: 0.130018175291002%\n",
      "Adj, read counts 43\n",
      "Number of samples per group [8, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 18\n",
      "total # samples 19\n",
      "Number of total reads 43\n",
      "Number of hits in BIOM tables 10.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 72.09302325581395 %\n",
      "Percentage of reads from samples who only fall into one bin group 60.46511627906976 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1509\n",
      "G003111625 d__Archaea; p__Euryarchaeota; c__Methanobacteria; o__Methanobacteriales; f__Methanobacteriaceae; g__Methanobrevibacter; s__Methanobrevibacter thaueri\n",
      "Coverage Percentage: 0.0182267469713073%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1510\n",
      "G003112735 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__; f__; g__Candidatus Phycosocius; s__Candidatus Phycosocius bacilliformis\n",
      "Coverage Percentage: 0.0041104862705258%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1511\n",
      "G003116135 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Roseomonas; s__Roseomonas aestuarii\n",
      "Coverage Percentage: 0.0029049919497093%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1512\n",
      "G003116815 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Caulobacter; s__Caulobacter endophyticus\n",
      "Coverage Percentage: 0.0493656777795071%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1513\n",
      "G003123745 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Ancrocorticia; s__Ancrocorticia populi\n",
      "Coverage Percentage: 0.005694080079883%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1514\n",
      "G003124025 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Kocuria; s__Kocuria rosea\n",
      "Coverage Percentage: 0.017661941972809%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1515\n",
      "G003130545 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Alteromonadales; f__Shewanellaceae; g__Shewanella; s__Shewanella xiamenensis\n",
      "Coverage Percentage: 0.0219262842123367%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1516\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G003131185 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas ovata\n",
      "Coverage Percentage: 0.006685101622851%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1517\n",
      "G003131205 d__Bacteria; p__Acidobacteria; c__Acidobacteriia; o__Acidobacteriales; f__Acidobacteriaceae; g__Occallatibacter; s__Occallatibacter savannae\n",
      "Coverage Percentage: 0.0017837476697805%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1518\n",
      "G003143535 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Cytophagaceae; g__Aquirufa; s__Aquirufa nivalisilvae\n",
      "Coverage Percentage: 0.0384752258373816%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1519\n",
      "G003143695 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus halitosis\n",
      "Coverage Percentage: 0.0134501779889166%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1520\n",
      "G003144135 d__Eukaryota; p__Basidiomycota; c__Exobasidiomycetes; o__Microstromatales; f__; g__Pseudomicrostroma; s__Pseudomicrostroma glucosiphilum\n",
      "Coverage Percentage: 0.0006400982446062%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1521\n",
      "G003144195 d__Eukaryota; p__Basidiomycota; c__Exobasidiomycetes; o__Ceraceosorales; f__Ceraceosoraceae; g__Ceraceosorus; s__Ceraceosorus guamensis\n",
      "Coverage Percentage: 0.0007286849307128%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1522\n",
      "G003148415 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__Bacillus inaquosorum\n",
      "Coverage Percentage: 0.0094931660697235%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1523\n",
      "G003148495 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Phyllobacteriaceae; g__Mesorhizobium; s__Mesorhizobium loti\n",
      "Coverage Percentage: 0.0036093691171644%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1524\n",
      "G003148525 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium araucananum\n",
      "Coverage Percentage: 0.0542313954891848%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1525\n",
      "G003148585 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter immobilis\n",
      "Coverage Percentage: 0.0220113164061994%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1526\n",
      "G003148625 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Spirosomaceae; g__Arcicella; s__Arcicella aurantiaca\n",
      "Coverage Percentage: 0.0229168769486717%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1527\n",
      "G003148685 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes xinjiangensis\n",
      "Coverage Percentage: 0.021038762994816%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1528\n",
      "G003149245 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Faecalicatena; s__Faecalicatena orotica\n",
      "Coverage Percentage: 0.0052813111362041%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1529\n",
      "G003150835 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Phenylobacterium; s__Phenylobacterium parvum\n",
      "Coverage Percentage: 0.0106769051823647%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1530\n",
      "G003160695 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides silvaticus\n",
      "Coverage Percentage: 0.0056748644073169%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1531\n",
      "G003173575 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter paludis\n",
      "Coverage Percentage: 0.0080896008977464%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1532\n",
      "G003173735 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium radiodurans\n",
      "Coverage Percentage: 0.0105059935610173%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1533\n",
      "G003182075 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Finegoldia; s__Finegoldia magna\n",
      "Coverage Percentage: 0.1563565126219697%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1534\n",
      "G003182375 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Melaminivora; s__Melaminivora alkalimesophila\n",
      "Coverage Percentage: 0.0602608818028753%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1535\n",
      "G003184545 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Eurotiales; f__Aspergillaceae; g__Aspergillus; s__Aspergillus heteromorphus\n",
      "Coverage Percentage: 0.0016118322188243%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1536\n",
      "G003184835 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Eurotiales; f__Aspergillaceae; g__Aspergillus; s__Aspergillus costaricensis\n",
      "Coverage Percentage: 0.0006874225330286%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1537\n",
      "G003194085 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Aurantimicrobium; s__Aurantimicrobium photophilum\n",
      "Coverage Percentage: 0.2735599346247818%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1538\n",
      "G003194385 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Oceanospirillales; f__Balneatrichaceae; g__Pokkaliibacter; s__Pokkaliibacter plantistimulans\n",
      "Coverage Percentage: 0.0069881853080392%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1539\n",
      "G003201355 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Dysgonomonadaceae; g__Dysgonomonas; s__Dysgonomonas alginatilytica\n",
      "Coverage Percentage: 0.0029479848571193%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1540\n",
      "G003201475 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Chelatococcaceae; g__Chelatococcus; s__Chelatococcus asaccharovorans\n",
      "Coverage Percentage: 0.0026463081038637%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 203\n",
      "Length of fail 1541\n",
      "G003201595 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Sphaerotilus; s__Sphaerotilus hippei\n",
      "Coverage Percentage: 0.8984305342684367%\n",
      "Adj, read counts 249\n",
      "Number of samples per group [1, 1, 1, 1, 2, 3, 1, 2, 2, 1, 1, 4, 3, 3, 1, 2, 1, 1, 3, 2, 2, 4, 1, 1, 2, 2, 3, 1, 5, 1, 5, 2, 1, 4, 1, 2, 2, 2, 1, 1, 3, 1, 3, 2, 2, 1, 1, 1, 1, 5, 1, 1, 2, 3, 2, 5, 2, 1, 7, 1, 1, 6, 2, 2, 2, 2, 1, 2, 2, 1, 2, 2, 1, 4, 2, 4, 3, 5, 1, 1, 3, 1, 2, 1, 1, 1, 5, 2, 2, 4, 1, 3, 1, 3, 1, 1, 5, 2, 1, 3, 12, 4]\n",
      "Number of bins 173\n",
      "total # samples 43\n",
      "Number of total reads 244\n",
      "Number of hits in BIOM tables 469.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.048192771084338 %\n",
      "Percentage of reads from samples who only fall into one bin group 2.8688524590163933 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 204\n",
      "Length of fail 1541\n",
      "G003201815 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Undibacterium; s__Undibacterium pigrum\n",
      "Coverage Percentage: 0.0482991143081639%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 204\n",
      "Length of fail 1542\n",
      "G003201855 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Neisseriaceae; g__Rivicola; s__Rivicola pingtungensis\n",
      "Coverage Percentage: 0.0933611440297785%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 204\n",
      "Length of fail 1543\n",
      "G003201955 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Blastomonas; s__Blastomonas natatoria\n",
      "Coverage Percentage: 0.0713878877809091%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 204\n",
      "Length of fail 1544\n",
      "G003202165 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Oceanospirillales; f__Halomonadaceae; g__Halomonas; s__Halomonas heilongjiangensis\n",
      "Coverage Percentage: 0.0132927716950445%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 204\n",
      "Length of fail 1545\n",
      "G003202405 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium hydrophilum\n",
      "Coverage Percentage: 0.0726090517609579%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 204\n",
      "Length of fail 1546\n",
      "G003202435 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium cheongpyeongense\n",
      "Coverage Percentage: 0.0420289896678733%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 204\n",
      "Length of fail 1547\n",
      "G003205195 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium wuzhouense\n",
      "Coverage Percentage: 0.0061200601792407%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 204\n",
      "Length of fail 1548\n",
      "G003205615 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora arborensis\n",
      "Coverage Percentage: 0.0041351305907102%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 204\n",
      "Length of fail 1549\n",
      "G003206695 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Oceanospirillales; f__Halomonadaceae; g__Salinicola; s__Salinicola halophyticus\n",
      "Coverage Percentage: 0.0055387484172257%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 204\n",
      "Length of fail 1550\n",
      "G003217575 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Xylophilus; s__Xylophilus ampelinus\n",
      "Coverage Percentage: 0.0506053758500644%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 204\n",
      "Length of fail 1551\n",
      "G003226595 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Modestobacter; s__Modestobacter versicolor\n",
      "Coverage Percentage: 0.0152410142531053%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 204\n",
      "Length of fail 1552\n",
      "G003231275 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas huaxiensis\n",
      "Coverage Percentage: 0.0123524837639806%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 204\n",
      "Length of fail 1553\n",
      "G003234965 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Aestuariivirgaceae; g__Aestuariivirga; s__Aestuariivirga litoralis\n",
      "Coverage Percentage: 0.0059313172368568%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 204\n",
      "Length of fail 1554\n",
      "G003244875 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas sepilia\n",
      "Coverage Percentage: 0.291441792779258%\n",
      "Adj, read counts 82\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 2, 2, 1, 4, 3, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1]\n",
      "Number of bins 67\n",
      "total # samples 29\n",
      "Number of total reads 82\n",
      "Number of hits in BIOM tables 31.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.634146341463413 %\n",
      "Percentage of reads from samples who only fall into one bin group 28.04878048780488 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1554\n",
      "G003254155 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Psychrobacillus; s__Psychrobacillus insolitus\n",
      "Coverage Percentage: 0.0045911350958224%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1555\n",
      "G003254525 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Phenylobacterium; s__Phenylobacterium kunshanense\n",
      "Coverage Percentage: 0.0106556752341321%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1556\n",
      "G003254745 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium aquariorum\n",
      "Coverage Percentage: 0.0322636771275874%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1557\n",
      "G003258865 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Hyphomicrobiaceae; g__Rhodoplanes; s__Rhodoplanes roseus\n",
      "Coverage Percentage: 0.0043766111580976%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1558\n",
      "G003259615 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter cryoconitis\n",
      "Coverage Percentage: 0.0049819867472575%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1559\n",
      "G003259835 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium aquaticum\n",
      "Coverage Percentage: 0.0762918138848266%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1560\n",
      "G003259905 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Roseobacteraceae; g__Salipiger; s__Salipiger aestuarii\n",
      "Coverage Percentage: 0.0032394834246788%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1561\n",
      "G003261575 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Klebsiella; s__Klebsiella huaxiensis\n",
      "Coverage Percentage: 0.0225842580896637%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1562\n",
      "G003263515 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Rahnella; s__Rahnella inusitata\n",
      "Coverage Percentage: 0.0085813005696553%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1563\n",
      "G003263915 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Dolosigranulum; s__Dolosigranulum pigrum\n",
      "Coverage Percentage: 0.0356006297026686%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1564\n",
      "G003265465 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas bivalvium\n",
      "Coverage Percentage: 0.0190276450540807%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1565\n",
      "G003269035 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia bryophila\n",
      "Coverage Percentage: 0.0033555803738216%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1566\n",
      "G003269065 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax anthurii\n",
      "Coverage Percentage: 0.066981966225006%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1567\n",
      "G003269275 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Acidaminococcales; f__Acidaminococcaceae; g__Phascolarctobacterium; s__Phascolarctobacterium faecium\n",
      "Coverage Percentage: 0.0186836767232024%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1568\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G003269405 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactiplantibacillus; s__Lactiplantibacillus plantarum\n",
      "Coverage Percentage: 0.0124410096529236%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1569\n",
      "G003285015 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae; g__Paenibacillus; s__Paenibacillus contaminans\n",
      "Coverage Percentage: 0.0071521469777928%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1570\n",
      "G003285265 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus mutanolyticus\n",
      "Coverage Percentage: 0.0042033684291399%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1571\n",
      "G003286075 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus endophyticus\n",
      "Coverage Percentage: 0.0075485234370735%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1572\n",
      "G003287395 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas carotinifaciens\n",
      "Coverage Percentage: 0.0116421103389665%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1573\n",
      "G003287895 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Blautia; s__Blautia argi\n",
      "Coverage Percentage: 0.0363859641143428%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1574\n",
      "G003290025 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Pseudogemmobacter; s__Pseudogemmobacter bohemicus\n",
      "Coverage Percentage: 0.0456308993406254%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1575\n",
      "G003290185 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium lathyri\n",
      "Coverage Percentage: 0.0048572143628651%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 205\n",
      "Length of fail 1576\n",
      "G003290485 d__Eukaryota; p__Basidiomycota; c__Malasseziomycetes; o__Malasseziales; f__Malasseziaceae; g__Malassezia; s__Malassezia restricta\n",
      "Coverage Percentage: 0.2061926452076696%\n",
      "Adj, read counts 90\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 4, 2, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 2, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 2]\n",
      "Number of bins 76\n",
      "total # samples 35\n",
      "Number of total reads 89\n",
      "Number of hits in BIOM tables 175.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.11111111111111 %\n",
      "Percentage of reads from samples who only fall into one bin group 22.47191011235955 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 206\n",
      "Length of fail 1576\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G003293845 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium tibetense\n",
      "Coverage Percentage: 0.0573972326647212%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 206\n",
      "Length of fail 1577\n",
      "G003294055 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia reimsis\n",
      "Coverage Percentage: 0.0045202503860674%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 206\n",
      "Length of fail 1578\n",
      "G003308535 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Myoviridae; g__Phapecoctavirus; s__Klebsiella virus ZCKP1\n",
      "Coverage Percentage: 0.0920987245320523%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 206\n",
      "Length of fail 1579\n",
      "G003312465 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Oscillospiraceae; g__Faecalibacterium; s__Faecalibacterium prausnitzii\n",
      "Coverage Percentage: 0.1165962119021709%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 206\n",
      "Length of fail 1580\n",
      "G003314435 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium petrolei\n",
      "Coverage Percentage: 0.0443453879860373%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 206\n",
      "Length of fail 1581\n",
      "G003314795 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas paucimobilis\n",
      "Coverage Percentage: 0.0113094353895451%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 206\n",
      "Length of fail 1582\n",
      "G003315095 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Achromobacter; s__Achromobacter marplatensis\n",
      "Coverage Percentage: 0.0033123979131312%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 206\n",
      "Length of fail 1583\n",
      "G003315175 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Eoetvoesia; s__Eoetvoesia caeni\n",
      "Coverage Percentage: 0.0566544704227058%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 206\n",
      "Length of fail 1584\n",
      "G003315195 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Sutterellaceae; g__Sutterella; s__Sutterella wadsworthensis\n",
      "Coverage Percentage: 0.0138277727068375%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 206\n",
      "Length of fail 1585\n",
      "G003316845 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Cyclobacteriaceae; g__Aquiflexum; s__Aquiflexum aquatile\n",
      "Coverage Percentage: 0.1584035537831452%\n",
      "Adj, read counts 48\n",
      "Number of samples per group [1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 44\n",
      "total # samples 21\n",
      "Number of total reads 48\n",
      "Number of hits in BIOM tables 94.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.583333333333332 %\n",
      "Percentage of reads from samples who only fall into one bin group 25.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1585\n",
      "G003323815 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter haemolyticus\n",
      "Coverage Percentage: 0.0383300311580435%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1586\n",
      "G003324185 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Oscillospiraceae; g__Faecalibacterium; s__Faecalibacterium prausnitzii\n",
      "Coverage Percentage: 0.0972411732779013%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1587\n",
      "G003334855 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Spirosomaceae; g__Larkinella; s__Larkinella punicea\n",
      "Coverage Percentage: 0.0018107973977467%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1588\n",
      "G003336665 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Oceanospirillales; f__Halomonadaceae; g__Halomonas; s__Halomonas rituensis\n",
      "Coverage Percentage: 0.0033719921439282%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1589\n",
      "G003337425 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Extensimonas; s__Extensimonas vulgaris\n",
      "Coverage Percentage: 0.043339808988114%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1590\n",
      "G003337555 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Pseudorhodoferax; s__Pseudorhodoferax soli\n",
      "Coverage Percentage: 0.035073719126276%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1591\n",
      "G003337565 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus lutimaris\n",
      "Coverage Percentage: 0.0388309368973511%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1592\n",
      "G003339505 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Spirosomaceae; g__Runella; s__Runella aurantiaca\n",
      "Coverage Percentage: 0.0017745158725592%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1593\n",
      "G003340315 d__Bacteria; p__Actinobacteria; c__Coriobacteriia; o__Eggerthellales; f__Eggerthellaceae; g__Slackia; s__Slackia isoflavoniconvertens\n",
      "Coverage Percentage: 0.0334394045628606%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1594\n",
      "G003340385 d__Bacteria; p__Actinobacteria; c__Coriobacteriia; o__Eggerthellales; f__Eggerthellaceae; g__Gordonibacter; s__Gordonibacter pamelaeae\n",
      "Coverage Percentage: 0.0131760791811772%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1595\n",
      "G003343305 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Enterococcaceae; g__Enterococcus; s__Enterococcus gilvus\n",
      "Coverage Percentage: 0.0038272573721594%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1596\n",
      "G003344785 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Roseobacteraceae; g__Thalassococcus; s__Thalassococcus profundi\n",
      "Coverage Percentage: 0.0078532495012873%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1597\n",
      "G003349825 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Schlegelella; s__Schlegelella thermodepolymerans\n",
      "Coverage Percentage: 0.0679587430213252%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1598\n",
      "G003350205 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas bullata\n",
      "Coverage Percentage: 0.0400657594675888%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1599\n",
      "G003350455 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Legionellales; f__Coxiellaceae; g__Aquicella; s__Aquicella lusitana\n",
      "Coverage Percentage: 0.0090489870271509%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1600\n",
      "G003350475 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Pseudacidovorax; s__Pseudacidovorax intermedius\n",
      "Coverage Percentage: 0.0262893979091979%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1601\n",
      "G003350925 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Rhodanobacteraceae; g__Dyella; s__Dyella tabacisoli\n",
      "Coverage Percentage: 0.0051065514491959%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1602\n",
      "G003351345 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Boseaceae; g__Bosea; s__Bosea caraganae\n",
      "Coverage Percentage: 0.0158883893287034%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1603\n",
      "G003355515 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Aliarcobacter; s__Aliarcobacter trophiarum\n",
      "Coverage Percentage: 0.0542907152958348%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1604\n",
      "G003367705 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium foliorum\n",
      "Coverage Percentage: 0.0456260380063775%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1605\n",
      "G003383615 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Spirosomaceae; g__Dyadobacter; s__Dyadobacter luteus\n",
      "Coverage Percentage: 0.0044023487825563%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 207\n",
      "Length of fail 1606\n",
      "G003384605 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Cutibacterium; s__Cutibacterium namnetense\n",
      "Coverage Percentage: 2.2343382153794877%\n",
      "Adj, read counts 348\n",
      "Number of samples per group [4, 1, 3, 6, 3, 13, 1, 9, 1, 1, 2, 3, 6, 1, 1, 2, 4, 2, 2, 1, 2, 14, 4, 2, 6, 16, 3, 23, 6, 2, 13, 1, 4, 2, 1, 6, 1, 3, 10, 3, 5, 2, 6, 8, 8, 1, 12, 11, 3, 6, 2, 5, 1, 1, 5, 1, 4, 1]\n",
      "Number of bins 131\n",
      "total # samples 53\n",
      "Number of total reads 338\n",
      "Number of hits in BIOM tables 116.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 24.42528735632184 %\n",
      "Percentage of reads from samples who only fall into one bin group 2.6627218934911245 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1606\n",
      "G003385075 d__Bacteria; p__Chloroflexi; c__Anaerolineae; o__Anaerolineales; f__Anaerolineaceae; g__Pelolinea; s__Pelolinea submarina\n",
      "Coverage Percentage: 0.0042901505530316%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1607\n",
      "G003385115 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium aquicola\n",
      "Coverage Percentage: 0.0284647047416072%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1608\n",
      "G003385415 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium piscium\n",
      "Coverage Percentage: 0.0176988734493531%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1609\n",
      "G003385455 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium rhizosphaerae\n",
      "Coverage Percentage: 0.0080656946081306%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1610\n",
      "G003385515 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium pennae\n",
      "Coverage Percentage: 0.0024395713333889%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1611\n",
      "G003385595 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium flavum\n",
      "Coverage Percentage: 0.0026165078386773%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1612\n",
      "G003386075 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Citricoccus; s__Citricoccus muralis\n",
      "Coverage Percentage: 0.004288906416914%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1613\n",
      "G003386795 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Thermasporomyces; s__Thermasporomyces composti\n",
      "Coverage Percentage: 0.0034533908867987%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1614\n",
      "G003387155 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Roseateles; s__Roseateles depolymerans\n",
      "Coverage Percentage: 0.0228211986485466%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1615\n",
      "G003390495 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycobacteroides; s__Mycobacteroides chelonae\n",
      "Coverage Percentage: 0.0200719221029264%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1616\n",
      "G003402235 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Cellvibrionales; f__Halieaceae; g__Parahaliea; s__Parahaliea mediterranea\n",
      "Coverage Percentage: 0.0055013876267897%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1617\n",
      "G003402615 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Faecalimonas; s__Faecalimonas umbilicata\n",
      "Coverage Percentage: 0.0270613102638983%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1618\n",
      "G003403015 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Aurantimonadaceae; g__Fulvimarina; s__Fulvimarina endophytica\n",
      "Coverage Percentage: 0.0248591922294784%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1619\n",
      "G003412465 d__Bacteria; p__Bacteroidetes; c__Chitinophagia; o__Chitinophagales; f__Chitinophagaceae; g__Chitinophaga; s__Chitinophaga silvisoli\n",
      "Coverage Percentage: 0.0033026537540882%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1620\n",
      "G003416765 d__Bacteria; p__Actinobacteria; c__Coriobacteriia; o__Coriobacteriales; f__Atopobiaceae; g__Paratractidigestivibacter; s__Paratractidigestivibacter faecalis\n",
      "Coverage Percentage: 0.0112898619885526%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1621\n",
      "G003417535 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Hydrogenophaga; s__Hydrogenophaga borbori\n",
      "Coverage Percentage: 0.0483228375693809%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1622\n",
      "G003425685 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium terrae\n",
      "Coverage Percentage: 0.0021663557491853%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1623\n",
      "G003426815 d__Bacteria; p__Actinobacteria; c__Acidimicrobiia; o__Acidimicrobiales; f__Ilumatobacteraceae; g__Desertimonas; s__Desertimonas flava\n",
      "Coverage Percentage: 0.0118450183019193%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1624\n",
      "G003428485 d__Bacteria; p__Actinobacteria; c__Coriobacteriia; o__Eggerthellales; f__Eggerthellaceae; g__Adlercreutzia; s__Adlercreutzia equolifaciens\n",
      "Coverage Percentage: 0.0148905252727799%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1625\n",
      "G003429605 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Citrobacter; s__Citrobacter gillenii\n",
      "Coverage Percentage: 0.0076045995249544%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1626\n",
      "G003431825 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Blastomonas; s__Blastomonas fulva\n",
      "Coverage Percentage: 0.0636394283122697%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1627\n",
      "G003433695 d__Bacteria; p__Firmicutes; c__Erysipelotrichia; o__Erysipelotrichales; f__Erysipelotrichaceae; g__Absiella; s__Absiella sp. AM22-9\n",
      "Coverage Percentage: 0.0207970037591044%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1628\n",
      "G003433955 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Anaerococcus; s__Anaerococcus nagyae\n",
      "Coverage Percentage: 0.0584546390357559%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1629\n",
      "G003435485 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Eisenbergiella; s__Eisenbergiella massiliensis\n",
      "Coverage Percentage: 0.0103237026086397%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1630\n",
      "G003457595 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Blautia; s__Blautia obeum\n",
      "Coverage Percentage: 0.1152493504847601%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1631\n",
      "G003457795 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Roseburia; s__Roseburia inulinivorans\n",
      "Coverage Percentage: 0.0503105457722096%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1632\n",
      "G003464225 d__Bacteria; p__Firmicutes; c__Erysipelotrichia; o__Erysipelotrichales; f__Erysipelotrichaceae; g__Holdemanella; s__Holdemanella biformis\n",
      "Coverage Percentage: 0.0234820584162602%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1633\n",
      "G003468475 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Anaerobutyricum; s__Anaerobutyricum hallii\n",
      "Coverage Percentage: 0.106334243678971%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1634\n",
      "G003470375 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Selenomonadales; f__Selenomonadaceae; g__Megamonas; s__Megamonas funiformis\n",
      "Coverage Percentage: 0.1949385282718616%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1635\n",
      "G003470435 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Mediterraneibacter; s__[Ruminococcus] torques\n",
      "Coverage Percentage: 0.0339398949788204%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1636\n",
      "G003472935 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides intestinalis\n",
      "Coverage Percentage: 0.0627259100701617%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1637\n",
      "G003516165 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia marcescens\n",
      "Coverage Percentage: 0.0311536173033909%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1638\n",
      "G003544815 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Arcobacter; s__Arcobacter suis\n",
      "Coverage Percentage: 0.1389020975126067%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1639\n",
      "G003544835 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Aliarcobacter; s__Aliarcobacter skirrowii\n",
      "Coverage Percentage: 0.1994064510626719%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1640\n",
      "G003545715 d__Bacteria; p__Bacteroidetes; c__Chitinophagia; o__Chitinophagales; f__Chitinophagaceae; g__Taibaiella; s__Taibaiella koreensis\n",
      "Coverage Percentage: 0.0074677197721336%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1641\n",
      "G003546465 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia cepacia\n",
      "Coverage Percentage: 0.0096882586825765%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1642\n",
      "G003550065 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Hephaestia; s__Hephaestia caeni\n",
      "Coverage Percentage: 0.0133255516236952%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 208\n",
      "Length of fail 1643\n",
      "G003568725 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Simplicispira; s__Simplicispira metamorpha\n",
      "Coverage Percentage: 1.4565372285468237%\n",
      "Adj, read counts 340\n",
      "Number of samples per group [3, 2, 2, 6, 2, 6, 2, 2, 6, 6, 12, 1, 1, 1, 7, 7, 1, 1, 3, 4, 2, 8, 2, 1, 2, 2, 3, 1, 2, 1, 3, 3, 5, 1, 4, 1, 4, 1, 6, 2, 5, 5, 9, 1, 5, 1, 7, 3, 1, 3, 5, 4, 5, 2, 1, 6, 1, 6, 1, 5, 9, 4, 7, 1, 2, 4, 3, 2, 2, 1, 3, 1, 1, 5, 4, 3, 6, 7, 4, 4, 4, 7]\n",
      "Number of bins 203\n",
      "total # samples 47\n",
      "Number of total reads 333\n",
      "Number of hits in BIOM tables 371.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.470588235294118 %\n",
      "Percentage of reads from samples who only fall into one bin group 3.003003003003003 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 209\n",
      "Length of fail 1643\n",
      "G003570715 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Gemmobacter; s__Gemmobacter lutimaris\n",
      "Coverage Percentage: 0.0115586433884023%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 209\n",
      "Length of fail 1644\n",
      "G003570885 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Simplicispira; s__Simplicispira hankyongi\n",
      "Coverage Percentage: 0.0966993886398841%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 209\n",
      "Length of fail 1645\n",
      "G003572535 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Shigella; s__Shigella boydii\n",
      "Coverage Percentage: 0.9544853382429196%\n",
      "Adj, read counts 227\n",
      "Number of samples per group [1, 9, 1, 2, 2, 2, 1, 1, 1, 1, 1, 2, 2, 1, 4, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 4, 1, 5, 5, 2, 1, 4, 2, 2, 1, 2, 2, 3, 1, 3, 2, 2, 2, 1, 1, 1, 2, 2, 2, 1, 1, 3, 1, 1, 1, 2, 4, 1, 1, 1, 3, 1, 1, 1, 3, 2, 1, 2, 1, 3, 1, 1, 3, 2, 2, 2, 1, 1, 2, 1, 3, 1, 1, 1, 2, 1, 1, 1, 1, 1, 4, 1]\n",
      "Number of bins 149\n",
      "total # samples 56\n",
      "Number of total reads 225\n",
      "Number of hits in BIOM tables 35.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 9.251101321585903 %\n",
      "Percentage of reads from samples who only fall into one bin group 14.222222222222221 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 210\n",
      "Length of fail 1645\n",
      "G003574215 d__Bacteria; p__Proteobacteria; c__Hydrogenophilalia; o__Hydrogenophilales; f__Hydrogenophilaceae; g__Hydrogenophilus; s__Hydrogenophilus thermoluteolus\n",
      "Coverage Percentage: 0.0243358965396714%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 210\n",
      "Length of fail 1646\n",
      "G003574295 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Mediterraneibacter; s__Mediterraneibacter butyricigenes\n",
      "Coverage Percentage: 0.0780001641068091%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 210\n",
      "Length of fail 1647\n",
      "G003574395 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Rhodobacter; s__Rhodobacter thermarum\n",
      "Coverage Percentage: 0.0092193205223989%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 210\n",
      "Length of fail 1648\n",
      "G003574675 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Calidifontimicrobium; s__Calidifontimicrobium sediminis\n",
      "Coverage Percentage: 0.0390150639131151%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 210\n",
      "Length of fail 1649\n",
      "G003575975 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Galactobacter; s__Galactobacter valiniphilus\n",
      "Coverage Percentage: 0.004125068568294%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 210\n",
      "Length of fail 1650\n",
      "G003576155 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Clavibacter; s__Clavibacter michiganensis\n",
      "Coverage Percentage: 0.0613802581772057%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 210\n",
      "Length of fail 1651\n",
      "G003576365 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Clavibacter; s__Clavibacter michiganensis\n",
      "Coverage Percentage: 0.1481389634840718%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 210\n",
      "Length of fail 1652\n",
      "G003576535 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Nakamurellales; f__Nakamurellaceae; g__Nakamurella; s__Nakamurella silvestris\n",
      "Coverage Percentage: 0.0077271747995488%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 210\n",
      "Length of fail 1653\n",
      "G003581425 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Amnibacterium; s__Amnibacterium setariae\n",
      "Coverage Percentage: 0.0075514257838408%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 210\n",
      "Length of fail 1654\n",
      "G003581445 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax cavernicola\n",
      "Coverage Percentage: 0.1718180627375057%\n",
      "Adj, read counts 87\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 5, 1, 2, 1, 1, 1, 3, 1, 2, 1, 1, 1, 1, 1, 2, 2, 1, 1, 4, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 2, 1, 5, 1, 1]\n",
      "Number of bins 73\n",
      "total # samples 30\n",
      "Number of total reads 85\n",
      "Number of hits in BIOM tables 33.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 18.39080459770115 %\n",
      "Percentage of reads from samples who only fall into one bin group 15.294117647058824 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1654\n",
      "G003581645 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Pelagerythrobacter; s__Pelagerythrobacter aerophilus\n",
      "Coverage Percentage: 0.0152458009040732%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1655\n",
      "G003586285 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Aeromicrobium; s__Aeromicrobium endophyticum\n",
      "Coverage Percentage: 0.0194821036403241%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1656\n",
      "G003589925 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Fulvivirgaceae; g__Chryseolinea; s__Chryseolinea soli\n",
      "Coverage Percentage: 0.0033246822835325%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1657\n",
      "G003594815 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus aestuarii\n",
      "Coverage Percentage: 0.0115114815036872%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1658\n",
      "G003595215 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Nesterenkonia; s__Nesterenkonia natronophila\n",
      "Coverage Percentage: 0.0059809346814597%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1659\n",
      "G003595235 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces griseorubiginosus\n",
      "Coverage Percentage: 0.0030276340994859%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1660\n",
      "G003595305 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Ralstonia; s__Ralstonia pseudosolanacearum\n",
      "Coverage Percentage: 0.0051970874970981%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1661\n",
      "G003596335 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Alteromonadales; f__; g__Motilimonas; s__Motilimonas pumilua\n",
      "Coverage Percentage: 0.003321355192103%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1662\n",
      "G003597125 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Oceanospirillales; f__Alcanivoracaceae; g__Alcanivorax; s__Alcanivorax profundi\n",
      "Coverage Percentage: 0.0305192875483444%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1663\n",
      "G003601985 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Phyllobacteriaceae; g__Mesorhizobium; s__Mesorhizobium jarvisii\n",
      "Coverage Percentage: 0.0171337587488407%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1664\n",
      "G003602055 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Rahnella; s__Rahnella inusitata\n",
      "Coverage Percentage: 0.0116266260732485%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1665\n",
      "G003602185 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Rahnella; s__Rahnella variigena\n",
      "Coverage Percentage: 0.0230716920216113%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1666\n",
      "G003604195 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Simplicispira; s__Simplicispira lacusdiani\n",
      "Coverage Percentage: 0.1064503585327785%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1667\n",
      "G003605645 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Vibrionales; f__Vibrionaceae; g__Vibrio; s__Vibrio sinensis\n",
      "Coverage Percentage: 0.0198062418117868%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1668\n",
      "G003609635 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Lacrimispora; s__Lacrimispora algidixylanolytica\n",
      "Coverage Percentage: 0.0065077609358935%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1669\n",
      "G003610395 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Dietziaceae; g__Dietzia; s__Dietzia kunjamensis\n",
      "Coverage Percentage: 0.0237259302412512%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1670\n",
      "G003610435 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__; g__Pseudorhodoplanes; s__Pseudorhodoplanes sinuspersici\n",
      "Coverage Percentage: 0.0046783257026878%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1671\n",
      "G003610915 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Rahnella; s__Rahnella variigena\n",
      "Coverage Percentage: 0.0225126601890336%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1672\n",
      "G003611455 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter guerrae\n",
      "Coverage Percentage: 0.0483192411621229%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 211\n",
      "Length of fail 1673\n",
      "G003611465 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter tianfuensis\n",
      "Coverage Percentage: 0.1389298278908335%\n",
      "Adj, read counts 32\n",
      "Number of samples per group [2, 1, 1, 1, 1, 2, 1, 1, 11, 4, 3, 1]\n",
      "Number of bins 19\n",
      "total # samples 22\n",
      "Number of total reads 31\n",
      "Number of hits in BIOM tables 10.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 62.5 %\n",
      "Percentage of reads from samples who only fall into one bin group 61.29032258064516 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 212\n",
      "Length of fail 1673\n",
      "G003611475 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter rongchengensis\n",
      "Coverage Percentage: 0.1790570893605417%\n",
      "Adj, read counts 52\n",
      "Number of samples per group [1, 3, 1, 5, 1, 1, 2, 2, 1, 5, 2, 1, 1, 3, 2, 1, 2, 1, 3, 1, 3, 2, 1, 1]\n",
      "Number of bins 31\n",
      "total # samples 24\n",
      "Number of total reads 51\n",
      "Number of hits in BIOM tables 29.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 26.923076923076927 %\n",
      "Percentage of reads from samples who only fall into one bin group 25.49019607843137 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 213\n",
      "Length of fail 1673\n",
      "G003612775 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora tulbaghiae\n",
      "Coverage Percentage: 0.0219825158069367%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 213\n",
      "Length of fail 1674\n",
      "G003626515 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus chosunense\n",
      "Coverage Percentage: 0.0861093251983327%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 213\n",
      "Length of fail 1675\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G003626915 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Tsuneonella; s__Tsuneonella amylolytica\n",
      "Coverage Percentage: 0.0425305116043693%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 213\n",
      "Length of fail 1676\n",
      "G003627135 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus koreensis\n",
      "Coverage Percentage: 0.0350318819982902%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 213\n",
      "Length of fail 1677\n",
      "G003627155 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus gwangjuense\n",
      "Coverage Percentage: 0.0886193580571878%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 213\n",
      "Length of fail 1678\n",
      "G003627755 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium jaguaris\n",
      "Coverage Percentage: 0.0083607164448688%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 213\n",
      "Length of fail 1679\n",
      "G003633755 d__Bacteria; p__Actinobacteria; c__Thermoleophilia; o__Solirubrobacterales; f__Solirubrobacteraceae; g__Solirubrobacter; s__Solirubrobacter pauli\n",
      "Coverage Percentage: 0.0063484882819939%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 213\n",
      "Length of fail 1680\n",
      "G003634555 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Frondihabitans; s__Frondihabitans australicus\n",
      "Coverage Percentage: 0.0224439477269062%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 213\n",
      "Length of fail 1681\n",
      "G003634965 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Azonexaceae; g__Azonexus; s__Azonexus fungiphilus\n",
      "Coverage Percentage: 0.0150855037895631%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 213\n",
      "Length of fail 1682\n",
      "G003635045 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Intrasporangiaceae; g__Terracoccus; s__Terracoccus luteus\n",
      "Coverage Percentage: 0.0049458258100032%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 213\n",
      "Length of fail 1683\n",
      "G003640395 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas urumqiensis\n",
      "Coverage Percentage: 0.0226009108584248%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 213\n",
      "Length of fail 1684\n",
      "G003641145 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactiplantibacillus; s__Lactiplantibacillus paraplantarum\n",
      "Coverage Percentage: 0.0598459501123447%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 213\n",
      "Length of fail 1685\n",
      "G003660105 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Aliarcobacter; s__Aliarcobacter cryaerophilus\n",
      "Coverage Percentage: 0.7588836821036256%\n",
      "Adj, read counts 98\n",
      "Number of samples per group [1, 1, 1, 1, 5, 3, 3, 2, 2, 1, 1, 2, 1, 1, 4, 1, 1, 1, 2, 1, 1, 1, 1, 2, 4, 5, 3, 2, 1, 2, 1, 1, 4, 1, 5, 4, 2, 1, 3, 2, 2, 3, 2, 1, 1, 1]\n",
      "Number of bins 76\n",
      "total # samples 30\n",
      "Number of total reads 97\n",
      "Number of hits in BIOM tables 60.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.3265306122449 %\n",
      "Percentage of reads from samples who only fall into one bin group 13.402061855670103 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 214\n",
      "Length of fail 1685\n",
      "G003660455 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides ferulae\n",
      "Coverage Percentage: 0.0116754306025027%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 214\n",
      "Length of fail 1686\n",
      "G003663795 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Actinokineospora; s__Actinokineospora cianjurensis\n",
      "Coverage Percentage: 0.0032430811424153%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 214\n",
      "Length of fail 1687\n",
      "G003664035 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter alluvionis\n",
      "Coverage Percentage: 0.0347798147892053%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 214\n",
      "Length of fail 1688\n",
      "G003664645 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter chengduensis\n",
      "Coverage Percentage: 0.1293378533472206%\n",
      "Adj, read counts 31\n",
      "Number of samples per group [1, 1, 4, 2, 1, 1, 3, 3, 4, 1, 1, 2, 1, 1, 3, 1]\n",
      "Number of bins 24\n",
      "total # samples 20\n",
      "Number of total reads 31\n",
      "Number of hits in BIOM tables 6.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 35.483870967741936 %\n",
      "Percentage of reads from samples who only fall into one bin group 41.935483870967744 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 215\n",
      "Length of fail 1688\n",
      "G003667205 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Kocuria; s__Kocuria indica\n",
      "Coverage Percentage: 0.0142674423679254%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 215\n",
      "Length of fail 1689\n",
      "G003667225 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Kocuria; s__Kocuria rhizophila\n",
      "Coverage Percentage: 0.0371846695665287%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 215\n",
      "Length of fail 1690\n",
      "G003667725 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Agrobacterium; s__Agrobacterium fabrum\n",
      "Coverage Percentage: 0.0027191558587724%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 215\n",
      "Length of fail 1691\n",
      "G003687725 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Faecalibacter; s__Faecalibacter macacae\n",
      "Coverage Percentage: 0.0104280255604027%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 215\n",
      "Length of fail 1692\n",
      "G003688495 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium weaverense\n",
      "Coverage Percentage: 0.0378442658433135%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 215\n",
      "Length of fail 1693\n",
      "G003688935 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium macginleyi\n",
      "Coverage Percentage: 0.1027612488859267%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 215\n",
      "Length of fail 1694\n",
      "G003693265 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium gottingense\n",
      "Coverage Percentage: 0.1471898391039161%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 215\n",
      "Length of fail 1695\n",
      "G003693415 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Clavibacter; s__Clavibacter michiganensis\n",
      "Coverage Percentage: 0.0599830832519638%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 215\n",
      "Length of fail 1696\n",
      "G003696365 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas zhaodongensis\n",
      "Coverage Percentage: 0.0029772563319173%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 215\n",
      "Length of fail 1697\n",
      "G003697165 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Escherichia; s__Escherichia coli\n",
      "Coverage Percentage: 0.8472341005320114%\n",
      "Adj, read counts 254\n",
      "Number of samples per group [3, 1, 1, 2, 1, 1, 2, 1, 1, 1, 2, 1, 1, 2, 2, 2, 1, 3, 2, 2, 1, 2, 1, 1, 1, 2, 1, 2, 2, 2, 1, 1, 2, 1, 2, 1, 2, 2, 1, 1, 1, 1, 1, 2, 2, 1, 1, 2, 2, 1, 1, 1, 1, 1, 3, 2, 1, 1, 1, 1, 1, 2, 1, 2, 4, 1, 1, 1, 2, 4, 2, 2, 2, 1, 2, 2, 1, 1, 4, 1, 1, 2, 1, 2, 3, 1, 4, 1, 1, 1, 1, 6, 1, 2, 1, 3, 2, 2, 2, 1, 1, 1, 1, 1, 1, 4, 4, 4, 1, 1, 1, 4, 1, 1, 3, 5, 1, 1]\n",
      "Number of bins 175\n",
      "total # samples 58\n",
      "Number of total reads 249\n",
      "Number of hits in BIOM tables 37.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 6.299212598425196 %\n",
      "Percentage of reads from samples who only fall into one bin group 12.048192771084338 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 216\n",
      "Length of fail 1697\n",
      "G003700725 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas marginalis\n",
      "Coverage Percentage: 0.0989776314505338%\n",
      "Adj, read counts 40\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 38\n",
      "total # samples 21\n",
      "Number of total reads 39\n",
      "Number of hits in BIOM tables 17.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.5 %\n",
      "Percentage of reads from samples who only fall into one bin group 28.205128205128204 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1697\n",
      "G003702705 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Ornithinimicrobiaceae; g__Ornithinimicrobium; s__Ornithinimicrobium cerasi\n",
      "Coverage Percentage: 0.0136314245255651%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1698\n",
      "G003704105 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas lutea\n",
      "Coverage Percentage: 0.0542930845527439%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1699\n",
      "G003710065 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium nematophagum\n",
      "Coverage Percentage: 0.0062767038134759%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1700\n",
      "G003710245 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Erwiniaceae; g__Pantoea; s__[Curtobacterium] plantarum\n",
      "Coverage Percentage: 0.0518031448382611%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1701\n",
      "G003710905 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae; g__Brevibacillus; s__Brevibacillus parabrevis\n",
      "Coverage Percentage: 0.0205922196776673%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1702\n",
      "G003711105 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes teichomyceticus\n",
      "Coverage Percentage: 0.0314945351551413%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1703\n",
      "G003719485 d__Eukaryota; p__Euglenozoa; c__Kinetoplastea; o__Trypanosomatida; f__Trypanosomatidae; g__Trypanosoma; s__Trypanosoma conorhini\n",
      "Coverage Percentage: 0.002241737663996%\n",
      "Adj, read counts 100\n",
      "Number of samples per group [2, 69]\n",
      "Number of bins 2\n",
      "total # samples 69\n",
      "Number of total reads 100\n",
      "Number of hits in BIOM tables 54.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 100.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 96.0 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1704\n",
      "G003721155 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Intrasporangiaceae; g__Janibacter; s__Janibacter melonis\n",
      "Coverage Percentage: 0.0439229862574771%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1705\n",
      "G003721415 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora aurantiaca\n",
      "Coverage Percentage: 0.0215681361429566%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1706\n",
      "G003722335 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Caulobacter; s__Caulobacter flavus\n",
      "Coverage Percentage: 0.0342981218906835%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1707\n",
      "G003722355 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylocystaceae; g__Methylocystis; s__Methylocystis hirsuta\n",
      "Coverage Percentage: 0.0171607482655895%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1708\n",
      "G003725415 d__Bacteria; p__Firmicutes; c__Erysipelotrichia; o__Erysipelotrichales; f__Erysipelotrichaceae; g__Absicoccus; s__Absicoccus porci\n",
      "Coverage Percentage: 0.0115642205715334%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1709\n",
      "G003725535 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Marmoricola; s__Marmoricola pocheonensis\n",
      "Coverage Percentage: 0.0113315713213769%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1710\n",
      "G003725695 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Marmoricola; s__Marmoricola ginsengisoli\n",
      "Coverage Percentage: 0.0033870680843303%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1711\n",
      "G003725955 d__Bacteria; p__Actinobacteria; c__Coriobacteriia; o__Eggerthellales; f__Eggerthellaceae; g__Slackia; s__Slackia isoflavoniconvertens\n",
      "Coverage Percentage: 0.0161441405640159%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1712\n",
      "G003729975 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Kaistella; s__Kaistella haifensis\n",
      "Coverage Percentage: 0.0102894362286933%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1713\n",
      "G003729985 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium cucumeris\n",
      "Coverage Percentage: 0.0085883459080542%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1714\n",
      "G003730015 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Epilithonimonas; s__Epilithonimonas hominis\n",
      "Coverage Percentage: 0.0297686055495524%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1715\n",
      "G003730315 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Tsuneonella; s__Tsuneonella troitsensis\n",
      "Coverage Percentage: 0.0628540513016358%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1716\n",
      "G003751265 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Kineosporiales; f__Kineosporiaceae; g__Pseudokineococcus; s__Pseudokineococcus lusitanus\n",
      "Coverage Percentage: 0.0039329174511881%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1717\n",
      "G003751345 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Stellaceae; g__Stella; s__Stella humosa\n",
      "Coverage Percentage: 0.01026096657962%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1718\n",
      "G003752415 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Luteococcus; s__Luteococcus japonicus\n",
      "Coverage Percentage: 0.0177244935993436%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1719\n",
      "G003752465 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Agrococcus; s__Agrococcus jenensis\n",
      "Coverage Percentage: 0.0101956931636526%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1720\n",
      "G003752505 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Marmoricola; s__Marmoricola aurantiacus\n",
      "Coverage Percentage: 0.0148768179767133%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 217\n",
      "Length of fail 1721\n",
      "G003755125 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Plantibacter; s__Plantibacter flavus\n",
      "Coverage Percentage: 0.2303501224589053%\n",
      "Adj, read counts 60\n",
      "Number of samples per group [1, 4, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 3, 1, 1, 1, 1, 2]\n",
      "Number of bins 50\n",
      "total # samples 29\n",
      "Number of total reads 60\n",
      "Number of hits in BIOM tables 50.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.666666666666664 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.333333333333332 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 218\n",
      "Length of fail 1721\n",
      "G003788965 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium mongoliense\n",
      "Coverage Percentage: 0.0070792610001636%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 218\n",
      "Length of fail 1722\n",
      "G003812345 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Citrobacter; s__Citrobacter freundii\n",
      "Coverage Percentage: 0.0100545237419056%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 218\n",
      "Length of fail 1723\n",
      "G003814555 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Vulcaniibacterium; s__Vulcaniibacterium tengchongense\n",
      "Coverage Percentage: 0.0441430189195919%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 218\n",
      "Length of fail 1724\n",
      "G003815635 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium shandongense\n",
      "Coverage Percentage: 0.0069745914954675%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 218\n",
      "Length of fail 1725\n",
      "G003815695 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium carnipullorum\n",
      "Coverage Percentage: 0.0113938737595423%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 218\n",
      "Length of fail 1726\n",
      "G003843835 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Tibeticola; s__Tibeticola sediminis\n",
      "Coverage Percentage: 0.0534529078418452%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 218\n",
      "Length of fail 1727\n",
      "G003851555 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas synxantha\n",
      "Coverage Percentage: 0.0321989293436934%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 218\n",
      "Length of fail 1728\n",
      "G003852045 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas orientalis\n",
      "Coverage Percentage: 0.031757754551072%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 218\n",
      "Length of fail 1729\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G003852895 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Albitalea; s__Albitalea terrae\n",
      "Coverage Percentage: 0.0427528045515198%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 218\n",
      "Length of fail 1730\n",
      "G003856525 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Vibrionales; f__Vibrionaceae; g__Vibrio; s__Vibrio viridaestus\n",
      "Coverage Percentage: 0.0237233488633993%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 218\n",
      "Length of fail 1731\n",
      "G003858535 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium viscosum\n",
      "Coverage Percentage: 0.0657932356973684%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 218\n",
      "Length of fail 1732\n",
      "G003860525 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Epilithonimonas; s__Epilithonimonas vandammei\n",
      "Coverage Percentage: 0.0307350374237995%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 218\n",
      "Length of fail 1733\n",
      "G003860545 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium taklimakanense\n",
      "Coverage Percentage: 0.0173705001640547%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 218\n",
      "Length of fail 1734\n",
      "G003860565 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Cloacibacterium; s__Cloacibacterium normanense\n",
      "Coverage Percentage: 0.0373247638574527%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 218\n",
      "Length of fail 1735\n",
      "G003862465 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Chromatiales; f__Chromatiaceae; g__Pararheinheimera; s__Pararheinheimera mesophila\n",
      "Coverage Percentage: 0.0152714506908956%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 218\n",
      "Length of fail 1736\n",
      "G003863335 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Falsigemmobacter; s__Falsigemmobacter faecalis\n",
      "Coverage Percentage: 0.0260241222869734%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 218\n",
      "Length of fail 1737\n",
      "G003863415 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Variovorax; s__Variovorax beijingensis\n",
      "Coverage Percentage: 0.1124387640962766%\n",
      "Adj, read counts 54\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 14, 1, 1, 1, 1]\n",
      "Number of bins 28\n",
      "total # samples 18\n",
      "Number of total reads 54\n",
      "Number of hits in BIOM tables 10.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 59.25925925925925 %\n",
      "Percentage of reads from samples who only fall into one bin group 29.629629629629626 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 219\n",
      "Length of fail 1737\n",
      "G003865015 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Sterolibacteriaceae; g__Sulfurisoma; s__Sulfurisoma sediminicola\n",
      "Coverage Percentage: 0.0142225142400419%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 219\n",
      "Length of fail 1738\n",
      "G003865035 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Anaerotignum; s__Anaerotignum faecicola\n",
      "Coverage Percentage: 0.033202190630556%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 219\n",
      "Length of fail 1739\n",
      "G003865365 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium macacae\n",
      "Coverage Percentage: 0.0456185017008251%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 219\n",
      "Length of fail 1740\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G003865375 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Gulosibacter; s__Gulosibacter macacae\n",
      "Coverage Percentage: 0.011764038892196%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 219\n",
      "Length of fail 1741\n",
      "G003865405 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium tangerinum\n",
      "Coverage Percentage: 0.0349273254089492%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 219\n",
      "Length of fail 1742\n",
      "G003932995 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Brevibacteriaceae; g__Brevibacterium; s__Brevibacterium aurantiacum\n",
      "Coverage Percentage: 0.0292425607272161%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 219\n",
      "Length of fail 1743\n",
      "G003933235 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingorhabdus; s__Sphingorhabdus wooponensis\n",
      "Coverage Percentage: 0.0514679381374666%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 219\n",
      "Length of fail 1744\n",
      "G003933245 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Maribacter; s__Maribacter algicola\n",
      "Coverage Percentage: 0.0204888448812157%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 219\n",
      "Length of fail 1745\n",
      "G003934165 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Phyllobacteriaceae; g__Pseudaminobacter; s__Pseudaminobacter arsenicus\n",
      "Coverage Percentage: 0.0063108334996653%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 219\n",
      "Length of fail 1746\n",
      "G003934285 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas fluminis\n",
      "Coverage Percentage: 0.0371084350123385%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 219\n",
      "Length of fail 1747\n",
      "G003940805 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Tabrizicola; s__Tabrizicola piscis\n",
      "Coverage Percentage: 0.0598518245689688%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 219\n",
      "Length of fail 1748\n",
      "G003940825 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas oryziphila\n",
      "Coverage Percentage: 0.0199682491614327%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 219\n",
      "Length of fail 1749\n",
      "G003944765 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Hymenobacteraceae; g__Hymenobacter; s__Hymenobacter perfusus\n",
      "Coverage Percentage: 0.0029529214366451%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 219\n",
      "Length of fail 1750\n",
      "G003945365 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Acidaminococcales; f__Acidaminococcaceae; g__Phascolarctobacterium; s__Phascolarctobacterium faecium\n",
      "Coverage Percentage: 0.0217571019214291%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 219\n",
      "Length of fail 1751\n",
      "G003945385 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Pseudonocardia; s__Pseudonocardia autotrophica\n",
      "Coverage Percentage: 0.0374238335409117%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 219\n",
      "Length of fail 1752\n",
      "G003946445 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Loigolactobacillus; s__Loigolactobacillus jiayinensis\n",
      "Coverage Percentage: 0.0088134759362509%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 219\n",
      "Length of fail 1753\n",
      "G003946725 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Agrilactobacillus; s__Agrilactobacillus yilanensis\n",
      "Coverage Percentage: 0.0092694079112627%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 219\n",
      "Length of fail 1754\n",
      "G003951285 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Variovorax; s__Variovorax beijingensis\n",
      "Coverage Percentage: 0.1105568898770026%\n",
      "Adj, read counts 53\n",
      "Number of samples per group [1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 14, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 29\n",
      "total # samples 21\n",
      "Number of total reads 51\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 56.60377358490566 %\n",
      "Percentage of reads from samples who only fall into one bin group 39.21568627450981 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 220\n",
      "Length of fail 1754\n",
      "G003952165 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Variovorax; s__Variovorax guangxiensis\n",
      "Coverage Percentage: 0.2061190541764263%\n",
      "Adj, read counts 95\n",
      "Number of samples per group [1, 3, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 3, 2, 1, 6, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 2, 1, 1]\n",
      "Number of bins 82\n",
      "total # samples 24\n",
      "Number of total reads 94\n",
      "Number of hits in BIOM tables 103.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 13.684210526315791 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.702127659574469 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 221\n",
      "Length of fail 1754\n",
      "G003952825 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium pseudocatenulatum\n",
      "Coverage Percentage: 0.1824032621858743%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 221\n",
      "Length of fail 1755\n",
      "G003953935 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia atriviolacea\n",
      "Coverage Percentage: 0.0459278922577045%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 221\n",
      "Length of fail 1756\n",
      "G003955735 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Undibacterium; s__Undibacterium parvum\n",
      "Coverage Percentage: 0.0365352673265552%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 221\n",
      "Length of fail 1757\n",
      "G003955755 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Jeotgalibaca; s__Jeotgalibaca ciconiae\n",
      "Coverage Percentage: 0.0178269249605079%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 221\n",
      "Length of fail 1758\n",
      "G003957875 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia marcescens\n",
      "Coverage Percentage: 0.0233116087857281%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 221\n",
      "Length of fail 1759\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G003958625 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Tsuneonella; s__Tsuneonella rigui\n",
      "Coverage Percentage: 0.0192067004145853%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 221\n",
      "Length of fail 1760\n",
      "G003958705 d__Viruses; p__Uroviricota; c__Caudoviricetes; o__Caudovirales; f__Myoviridae; g__Xuanwuvirus; s__Escherichia virus 500465-2\n",
      "Coverage Percentage: 0.9140159196150236%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 221\n",
      "Length of fail 1761\n",
      "G003959475 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Thermales; f__Thermaceae; g__Thermus; s__Thermus scotoductus\n",
      "Coverage Percentage: 0.0956292089140722%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 221\n",
      "Length of fail 1762\n",
      "G003963555 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus periodonticum\n",
      "Coverage Percentage: 0.0154952030528176%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 221\n",
      "Length of fail 1763\n",
      "G003964805 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Enterobacter; s__Enterobacter quasiroggenkampii\n",
      "Coverage Percentage: 0.0186473586110495%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 221\n",
      "Length of fail 1764\n",
      "G003964905 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Enterobacter; s__Enterobacter quasimori\n",
      "Coverage Percentage: 0.0121157544698858%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 221\n",
      "Length of fail 1765\n",
      "G003965815 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Variovorax; s__Variovorax gossypii\n",
      "Coverage Percentage: 0.3549160732354812%\n",
      "Adj, read counts 132\n",
      "Number of samples per group [1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 4, 2, 1, 1, 2, 2, 7, 3, 3, 1, 1, 1, 2, 1, 1, 1, 1, 3, 2, 1, 1, 3, 1, 1, 2, 2, 1, 2, 1, 1, 1, 1, 1, 3, 3, 1, 1, 2, 1, 2, 1, 2, 1, 1, 2, 1, 1, 2, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 2]\n",
      "Number of bins 103\n",
      "total # samples 29\n",
      "Number of total reads 131\n",
      "Number of hits in BIOM tables 245.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.363636363636365 %\n",
      "Percentage of reads from samples who only fall into one bin group 9.923664122137405 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 222\n",
      "Length of fail 1765\n",
      "G003966125 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Azospirillaceae; g__Azospirillum; s__Azospirillum griseum\n",
      "Coverage Percentage: 0.0044998425055123%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 222\n",
      "Length of fail 1766\n",
      "G003966755 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Legionellales; f__Coxiellaceae; g__Rickettsiella; s__Candidatus Rickettsiella viridis\n",
      "Coverage Percentage: 0.0455772645412047%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 222\n",
      "Length of fail 1767\n",
      "G003966915 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Mycoavidus; s__Mycoavidus cysteinexigens\n",
      "Coverage Percentage: 0.015026812126208%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 222\n",
      "Length of fail 1768\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G003966975 d__Bacteria; p__Chloroflexi; c__Anaerolineae; o__Anaerolineales; f__Anaerolineaceae; g__Pelolinea; s__Pelolinea submarina\n",
      "Coverage Percentage: 0.0042899104431477%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 222\n",
      "Length of fail 1769\n",
      "G003967075 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium amiense\n",
      "Coverage Percentage: 0.044323028578299%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 222\n",
      "Length of fail 1770\n",
      "G003967275 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Thiobacillaceae; g__Sulfuritortus; s__Sulfuritortus calidifontis\n",
      "Coverage Percentage: 0.0618237794398074%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 222\n",
      "Length of fail 1771\n",
      "G003970355 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium bomense\n",
      "Coverage Percentage: 0.0294509014014134%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 222\n",
      "Length of fail 1772\n",
      "G003970385 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Vibrionales; f__Vibrionaceae; g__Vibrio; s__Vibrio aquaticus\n",
      "Coverage Percentage: 0.019771605416208%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 222\n",
      "Length of fail 1773\n",
      "G003970495 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus pasteuri\n",
      "Coverage Percentage: 0.0209754007640401%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 222\n",
      "Length of fail 1774\n",
      "G003977605 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella koreensis\n",
      "Coverage Percentage: 0.0188738129560552%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 222\n",
      "Length of fail 1775\n",
      "G003987395 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Alteromonadales; f__Idiomarinaceae; g__Pseudidiomarina; s__Pseudidiomarina aquimaris\n",
      "Coverage Percentage: 0.0071798381571605%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 222\n",
      "Length of fail 1776\n",
      "G003987645 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Enterococcaceae; g__Vagococcus; s__Vagococcus elongatus\n",
      "Coverage Percentage: 0.0052425276621451%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 222\n",
      "Length of fail 1777\n",
      "G003989665 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Azospirillaceae; g__Azospirillum; s__Azospirillum doebereinerae\n",
      "Coverage Percentage: 0.0075831771712949%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 222\n",
      "Length of fail 1778\n",
      "G003990335 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Chromatiales; f__Chromatiaceae; g__Rheinheimera; s__Rheinheimera mangrovi\n",
      "Coverage Percentage: 0.0307668773823264%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 222\n",
      "Length of fail 1779\n",
      "G003990875 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Rhodococcus; s__Rhodococcus erythropolis\n",
      "Coverage Percentage: 0.1236742264235013%\n",
      "Adj, read counts 48\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1]\n",
      "Number of bins 45\n",
      "total # samples 28\n",
      "Number of total reads 47\n",
      "Number of hits in BIOM tables 22.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.5 %\n",
      "Percentage of reads from samples who only fall into one bin group 42.5531914893617 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 223\n",
      "Length of fail 1779\n",
      "G003991565 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Odoribacteraceae; g__Butyricimonas; s__Butyricimonas faecalis\n",
      "Coverage Percentage: 0.0539743053370259%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 223\n",
      "Length of fail 1780\n",
      "G003991585 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Saezia; s__Saezia sanguinis\n",
      "Coverage Percentage: 0.0251010750534068%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 223\n",
      "Length of fail 1781\n",
      "G003991855 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium oxydans\n",
      "Coverage Percentage: 0.0702860843279437%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 223\n",
      "Length of fail 1782\n",
      "G003991875 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium lemovicicum\n",
      "Coverage Percentage: 0.05757039436142%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 223\n",
      "Length of fail 1783\n",
      "G003993775 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Frigidibacter; s__Frigidibacter oleivorans\n",
      "Coverage Percentage: 0.0132503864079164%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 223\n",
      "Length of fail 1784\n",
      "G003994235 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Methylococcales; f__Methylococcaceae; g__Methylobacter; s__Candidatus Methylobacter oryzae\n",
      "Coverage Percentage: 0.0078445084400406%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 223\n",
      "Length of fail 1785\n",
      "G003994255 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Dermabacteraceae; g__Brachybacterium; s__Brachybacterium paraconglomeratum\n",
      "Coverage Percentage: 0.0167265086359519%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 223\n",
      "Length of fail 1786\n",
      "G003994395 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces griseoviridis\n",
      "Coverage Percentage: 0.0016846259695245%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 223\n",
      "Length of fail 1787\n",
      "G003996965 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium cupreum\n",
      "Coverage Percentage: 0.041906685085163%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 223\n",
      "Length of fail 1788\n",
      "G004000885 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae; g__Paenibacillus; s__Paenibacillus glucanolyticus\n",
      "Coverage Percentage: 0.0097540293895408%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 223\n",
      "Length of fail 1789\n",
      "G004000905 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae; g__Paenibacillus; s__Paenibacillus humicus\n",
      "Coverage Percentage: 0.0080879049933618%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 223\n",
      "Length of fail 1790\n",
      "G004001245 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Chromatiales; f__Thioalkalispiraceae; g__Sulfurivermis; s__Sulfurivermis fontis\n",
      "Coverage Percentage: 0.0475630996600963%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 223\n",
      "Length of fail 1791\n",
      "G004004565 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Inhella; s__Inhella crocodyli\n",
      "Coverage Percentage: 0.0334769198042969%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 223\n",
      "Length of fail 1792\n",
      "G004005485 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium algorifonticola\n",
      "Coverage Percentage: 0.0400046819523741%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 223\n",
      "Length of fail 1793\n",
      "G004005865 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas crocodyli\n",
      "Coverage Percentage: 0.0118967834250982%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 223\n",
      "Length of fail 1794\n",
      "G004005905 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium umbonatum\n",
      "Coverage Percentage: 0.039556981367146%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 223\n",
      "Length of fail 1795\n",
      "G004005945 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Chromatiales; f__Chromatiaceae; g__Rheinheimera; s__Rheinheimera riviphila\n",
      "Coverage Percentage: 0.005370398489886%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 223\n",
      "Length of fail 1796\n",
      "G004011135 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Rathayibacter; s__Rathayibacter festucae\n",
      "Coverage Percentage: 0.0988128240145439%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 223\n",
      "Length of fail 1797\n",
      "G004011695 d__Eukaryota; p__Ascomycota; c__Dothideomycetes; o__Pleosporales; f__Didymellaceae; g__Ascochyta; s__Ascochyta rabiei\n",
      "Coverage Percentage: 0.0053106154020799%\n",
      "Adj, read counts 49\n",
      "Number of samples per group [1, 1, 21, 1, 4, 2, 3, 1, 1, 2, 7, 1, 1, 1, 1, 1]\n",
      "Number of bins 16\n",
      "total # samples 37\n",
      "Number of total reads 49\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 65.3061224489796 %\n",
      "Percentage of reads from samples who only fall into one bin group 55.10204081632652 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 224\n",
      "Length of fail 1797\n",
      "G004011805 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Piscinibacter; s__Piscinibacter defluvii\n",
      "Coverage Percentage: 0.0693829768203296%\n",
      "Adj, read counts 35\n",
      "Number of samples per group [1, 8, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 3, 1]\n",
      "Number of bins 23\n",
      "total # samples 18\n",
      "Number of total reads 34\n",
      "Number of hits in BIOM tables 18.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 42.85714285714286 %\n",
      "Percentage of reads from samples who only fall into one bin group 32.35294117647059 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1797\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G004014855 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Apibacter; s__Apibacter raozihei\n",
      "Coverage Percentage: 0.0091708350440078%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1798\n",
      "G004016505 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Aquincola; s__Aquincola rivuli\n",
      "Coverage Percentage: 0.079842619162472%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1799\n",
      "G004016515 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Rubrivivax; s__Rubrivivax albus\n",
      "Coverage Percentage: 0.0521666675138337%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1800\n",
      "G004016525 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium sufflavum\n",
      "Coverage Percentage: 0.0220177591460248%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1801\n",
      "G004023405 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Arcobacter; s__Arcobacter venerupis\n",
      "Coverage Percentage: 0.0327560348804541%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1802\n",
      "G004023465 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Arcobacter; s__Arcobacter suis\n",
      "Coverage Percentage: 0.0924474686122552%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1803\n",
      "G004025965 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Intrasporangiaceae; g__Phycicoccus; s__Phycicoccus flavus\n",
      "Coverage Percentage: 0.0106501993628178%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1804\n",
      "G004028155 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium cerinum\n",
      "Coverage Percentage: 0.0150823325070485%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1805\n",
      "G004101845 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Latilactobacillus; s__Latilactobacillus curvatus\n",
      "Coverage Percentage: 0.0123660281726297%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1806\n",
      "G004114935 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium zhanjiangense\n",
      "Coverage Percentage: 0.0128237762659946%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1807\n",
      "G004114975 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium guangdongense\n",
      "Coverage Percentage: 0.0186062805559272%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1808\n",
      "G004115735 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Aliarcobacter; s__Aliarcobacter skirrowii\n",
      "Coverage Percentage: 0.1923533590445825%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1809\n",
      "G004116585 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Aliarcobacter; s__Aliarcobacter trophiarum\n",
      "Coverage Percentage: 0.0517364287636635%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1810\n",
      "G004118375 d__Bacteria; p__Verrucomicrobia; c__Opitutae; o__Opitutales; f__Opitutaceae; g__Oleiharenicola; s__Oleiharenicola lentus\n",
      "Coverage Percentage: 0.0084227749288148%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1811\n",
      "G004121315 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Cellulomonadaceae; g__Oerskovia; s__Oerskovia turbata\n",
      "Coverage Percentage: 0.014358218676664%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1812\n",
      "G004122105 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium stagni\n",
      "Coverage Percentage: 0.0193363816693617%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1813\n",
      "G004122115 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium fluviale\n",
      "Coverage Percentage: 0.0339096563365453%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1814\n",
      "G004123145 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Blautia; s__Blautia faecicola\n",
      "Coverage Percentage: 0.0700797652167377%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1815\n",
      "G004135605 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas desiccabilis\n",
      "Coverage Percentage: 0.0144992183451983%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1816\n",
      "G004135645 d__Bacteria; p__Actinobacteria; c__Coriobacteriia; o__Coriobacteriales; f__Coriobacteriaceae; g__Senegalimassilia; s__Senegalimassilia faecalis\n",
      "Coverage Percentage: 0.007532583883691%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 225\n",
      "Length of fail 1817\n",
      "Length of pass 226\n",
      "Length of fail 1817\n",
      "G004136555 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Kocuria; s__Kocuria carniphila\n",
      "Coverage Percentage: 0.006454774810139%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1818\n",
      "G004137085 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Lichenibacteriaceae; g__Lichenibacterium; s__Lichenibacterium ramalinae\n",
      "Coverage Percentage: 0.0195078540653971%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1819\n",
      "G004137235 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides ganghwensis\n",
      "Coverage Percentage: 0.0323677582424329%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1820\n",
      "G004137245 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides glacieisoli\n",
      "Coverage Percentage: 0.0539979685230288%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1821\n",
      "G004137255 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides oleivorans\n",
      "Coverage Percentage: 0.0279454342499331%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1822\n",
      "G004137685 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Lichenibacteriaceae; g__Lichenibacterium; s__Lichenibacterium minor\n",
      "Coverage Percentage: 0.0287155613850022%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1823\n",
      "G004138065 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium striatum\n",
      "Coverage Percentage: 0.0480803929643419%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1824\n",
      "G004138325 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Cytophagaceae; g__Spirosoma; s__Spirosoma sordidisoli\n",
      "Coverage Percentage: 0.0091203614920422%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1825\n",
      "G004138385 d__Bacteria; p__Candidatus Saccharibacteria; c__Candidatus Saccharimonia; o__Candidatus Saccharimonales; f__Candidatus Sacchiramonaceae; g__Candidatus Nanosynsacchari; s__Candidatus Nanosynsacchari sp. TM7_ANC_38.39_G1_1\n",
      "Coverage Percentage: 0.0199121754381337%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1826\n",
      "G004151455 d__Bacteria; p__Candidatus Saccharibacteria; c__Candidatus Nanosyncoccalia; o__Candidatus Nanogingivales; f__Candidatus Nanogingivalaceae; g__Candidatus Nanogingivalis; s__Candidatus Nanogingivalis gingivitcus\n",
      "Coverage Percentage: 0.0632505007331308%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1827\n",
      "G004152865 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium cupriresistens\n",
      "Coverage Percentage: 0.0671689058581956%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1828\n",
      "G004154835 d__Eukaryota; p__Ascomycota; c__Dothideomycetes; o__Pleosporales; f__Pleosporaceae; g__Alternaria; s__Alternaria arborescens\n",
      "Coverage Percentage: 0.007514006877599%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1829\n",
      "G004166975 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Tannerellaceae; g__Parabacteroides; s__Parabacteroides merdae\n",
      "Coverage Percentage: 0.0721616333382385%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1830\n",
      "G004167665 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Hymenobacteraceae; g__Hymenobacter; s__Hymenobacter persicinus\n",
      "Coverage Percentage: 0.0048961355383093%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1831\n",
      "G004168035 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides iriomotensis\n",
      "Coverage Percentage: 0.0117510134188512%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1832\n",
      "G004168285 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Spirosomaceae; g__Emticicia; s__Emticicia agri\n",
      "Coverage Percentage: 0.0061614192663973%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1833\n",
      "G004208535 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas populi\n",
      "Coverage Percentage: 0.0572423145327886%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1834\n",
      "G004208635 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Leucobacter; s__Leucobacter triazinivorans\n",
      "Coverage Percentage: 0.0263020597531055%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1835\n",
      "G004209115 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter wuhouensis\n",
      "Coverage Percentage: 0.0789438187770024%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1836\n",
      "G004209755 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia lutea\n",
      "Coverage Percentage: 0.0360828567922313%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1837\n",
      "G004210305 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Holosporales; f__Candidatus Paracaedibacteraceae; g__Candidatus Finniella; s__Candidatus Finniella inopinata\n",
      "Coverage Percentage: 0.0084230114441392%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1838\n",
      "G004214795 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Malaciobacter; s__Malaciobacter pacificus\n",
      "Coverage Percentage: 0.1218965752705708%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1839\n",
      "G004214815 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Aliarcobacter; s__Aliarcobacter thereius\n",
      "Coverage Percentage: 0.0539403053658332%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 226\n",
      "Length of fail 1840\n",
      "G004216565 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Leptothrix; s__Leptothrix mobilis\n",
      "Coverage Percentage: 0.1153206677636281%\n",
      "Adj, read counts 41\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 2, 1, 2, 1, 1, 7]\n",
      "Number of bins 25\n",
      "total # samples 21\n",
      "Number of total reads 41\n",
      "Number of hits in BIOM tables 17.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 46.34146341463414 %\n",
      "Percentage of reads from samples who only fall into one bin group 46.34146341463415 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 227\n",
      "Length of fail 1840\n",
      "G004216575 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microcella; s__Microcella putealis\n",
      "Coverage Percentage: 0.0296699988760768%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 227\n",
      "Length of fail 1841\n",
      "G004216755 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Kerstersia; s__Kerstersia gyiorum\n",
      "Coverage Percentage: 0.0067037624929761%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 227\n",
      "Length of fail 1842\n",
      "G004216855 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microcella; s__Microcella alkaliphila\n",
      "Coverage Percentage: 0.022468716775873%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 227\n",
      "Length of fail 1843\n",
      "G004217045 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Cupriavidus; s__Cupriavidus agavae\n",
      "Coverage Percentage: 0.002503720182851%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 227\n",
      "Length of fail 1844\n",
      "G004217095 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Advenella; s__Advenella incenata\n",
      "Coverage Percentage: 0.0051460941552167%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 227\n",
      "Length of fail 1845\n",
      "G004217185 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Pseudonocardia; s__Pseudonocardia sediminis\n",
      "Coverage Percentage: 0.0179671636472858%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 227\n",
      "Length of fail 1846\n",
      "G004217215 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Rivibacter; s__Rivibacter subsaxonicus\n",
      "Coverage Percentage: 0.052532705477454%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 227\n",
      "Length of fail 1847\n",
      "G004217445 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Fluviicoccus; s__Fluviicoccus keumensis\n",
      "Coverage Percentage: 0.0076845542089633%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 227\n",
      "Length of fail 1848\n",
      "G004283755 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Pseudoxanthomonas; s__Pseudoxanthomonas winnipegensis\n",
      "Coverage Percentage: 0.0286254741789764%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 227\n",
      "Length of fail 1849\n",
      "G004295685 d__Bacteria; p__Proteobacteria; c__Oligoflexia; o__Silvanigrellales; f__Silvanigrellaceae; g__Fluviispira; s__Fluviispira sanaruensis\n",
      "Coverage Percentage: 0.0041161181462203%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 227\n",
      "Length of fail 1850\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G004299785 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Aliarcobacter; s__[Arcobacter] porcinus\n",
      "Coverage Percentage: 0.0713500368394113%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 227\n",
      "Length of fail 1851\n",
      "G004306555 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium leguminosarum\n",
      "Coverage Percentage: 0.0186884498536246%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 227\n",
      "Length of fail 1852\n",
      "G004307015 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Thermales; f__Thermaceae; g__Thermus; s__Thermus thermamylovorans\n",
      "Coverage Percentage: 0.0102137425461883%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 227\n",
      "Length of fail 1853\n",
      "G004307415 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Cytophagaceae; g__Aquirufa; s__Aquirufa antheringensis\n",
      "Coverage Percentage: 0.5256534614541208%\n",
      "Adj, read counts 86\n",
      "Number of samples per group [2, 1, 2, 2, 2, 1, 1, 4, 1, 6, 1, 2, 1, 2, 1, 1, 1, 4, 2, 1, 1, 1, 3, 2, 1, 1, 1, 2, 1, 1, 1, 1, 4, 2, 2, 2, 1, 1, 3, 3, 2, 1, 1, 3]\n",
      "Number of bins 65\n",
      "total # samples 33\n",
      "Number of total reads 83\n",
      "Number of hits in BIOM tables 94.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.27906976744186 %\n",
      "Percentage of reads from samples who only fall into one bin group 21.686746987951807 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 228\n",
      "Length of fail 1853\n",
      "G004307425 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Cytophagaceae; g__Aquirufa; s__Aquirufa antheringensis\n",
      "Coverage Percentage: 0.5022490017124949%\n",
      "Adj, read counts 80\n",
      "Number of samples per group [3, 4, 2, 1, 1, 1, 2, 2, 2, 1, 2, 1, 1, 6, 1, 3, 1, 2, 2, 1, 3, 1, 3, 2, 3, 1, 2, 1, 3, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 2, 1]\n",
      "Number of bins 60\n",
      "total # samples 29\n",
      "Number of total reads 79\n",
      "Number of hits in BIOM tables 83.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.5 %\n",
      "Percentage of reads from samples who only fall into one bin group 12.658227848101266 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 229\n",
      "Length of fail 1853\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G004307455 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Cytophagaceae; g__Aquirufa; s__Aquirufa nivalisilvae\n",
      "Coverage Percentage: 0.0470518689455077%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 229\n",
      "Length of fail 1854\n",
      "G004310385 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus sediminis\n",
      "Coverage Percentage: 0.0070599997494838%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 229\n",
      "Length of fail 1855\n",
      "G004310665 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter kyonggii\n",
      "Coverage Percentage: 0.0167269402159763%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 229\n",
      "Length of fail 1856\n",
      "G004310865 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Aquabacterium; s__Aquabacterium lacunae\n",
      "Coverage Percentage: 0.0420467290774552%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 229\n",
      "Length of fail 1857\n",
      "G004322755 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia albidiflava\n",
      "Coverage Percentage: 0.0380521750448457%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 229\n",
      "Length of fail 1858\n",
      "G004324755 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Brevilactibacter; s__Brevilactibacter sinopodophylli\n",
      "Coverage Percentage: 0.0265660991861734%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 229\n",
      "Length of fail 1859\n",
      "G004327275 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas daroniae\n",
      "Coverage Percentage: 0.0054213541034624%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 229\n",
      "Length of fail 1860\n",
      "G004327895 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Azotobacter; s__Azotobacter chroococcum\n",
      "Coverage Percentage: 0.0158427469601191%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 229\n",
      "Length of fail 1861\n",
      "G004328535 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Kocuria; s__Kocuria indica\n",
      "Coverage Percentage: 0.0179411698487031%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 229\n",
      "Length of fail 1862\n",
      "G004328555 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Haematobacter; s__Haematobacter massiliensis\n",
      "Coverage Percentage: 0.0139141790445376%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 229\n",
      "Length of fail 1863\n",
      "G004330295 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Rahnella; s__Rahnella victoriana\n",
      "Coverage Percentage: 0.0100161934736968%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 229\n",
      "Length of fail 1864\n",
      "G004331155 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter terrestris\n",
      "Coverage Percentage: 0.9535482438599344%\n",
      "Adj, read counts 194\n",
      "Number of samples per group [2, 1, 3, 1, 4, 2, 1, 4, 3, 2, 3, 2, 1, 2, 1, 1, 5, 1, 2, 1, 2, 1, 4, 1, 4, 1, 1, 7, 1, 8, 4, 4, 1, 1, 2, 4, 5, 1, 3, 1, 1, 3, 1, 5, 2, 9, 1, 4, 6, 3, 3, 1, 6, 3, 4, 1, 1, 2, 1, 2, 2, 2, 4, 4, 3, 1, 3, 2]\n",
      "Number of bins 128\n",
      "total # samples 38\n",
      "Number of total reads 194\n",
      "Number of hits in BIOM tables 256.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.463917525773196 %\n",
      "Percentage of reads from samples who only fall into one bin group 5.154639175257731 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 230\n",
      "Length of fail 1864\n",
      "G004335085 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter nototheniae\n",
      "Coverage Percentage: 0.0123243752785565%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 230\n",
      "Length of fail 1865\n",
      "G004337635 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Myroides; s__Myroides odoratimimus\n",
      "Coverage Percentage: 0.0268406952679625%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 230\n",
      "Length of fail 1866\n",
      "G004339465 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Xanthobacteraceae; g__Ancylobacter; s__Ancylobacter aquaticus\n",
      "Coverage Percentage: 0.0098713026332372%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 230\n",
      "Length of fail 1867\n",
      "G004339565 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Pseudonocardia; s__Pseudonocardia endophytica\n",
      "Coverage Percentage: 0.0090116112364485%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 230\n",
      "Length of fail 1868\n",
      "G004340665 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter psychrotolerans\n",
      "Coverage Percentage: 0.0087401951951258%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 230\n",
      "Length of fail 1869\n",
      "G004340685 d__Bacteria; p__Firmicutes; c__; o__; f__; g__Hydrogenispora; s__Hydrogenispora ethanolica\n",
      "Coverage Percentage: 0.0025227675595481%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 230\n",
      "Length of fail 1870\n",
      "G004340905 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Rubrivivax; s__Rubrivivax gelatinosus\n",
      "Coverage Percentage: 0.040858469139149%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 230\n",
      "Length of fail 1871\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G004341365 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Simplicispira; s__Simplicispira metamorpha\n",
      "Coverage Percentage: 1.5007403045499432%\n",
      "Adj, read counts 352\n",
      "Number of samples per group [7, 2, 7, 2, 2, 3, 2, 2, 6, 4, 4, 1, 1, 4, 4, 3, 1, 2, 1, 2, 1, 8, 2, 1, 1, 5, 1, 12, 3, 1, 1, 3, 1, 2, 12, 5, 1, 3, 11, 1, 9, 3, 4, 3, 10, 2, 2, 2, 4, 1, 4, 2, 7, 1, 7, 3, 3, 3, 6, 2, 12, 2, 2, 3, 3, 1, 14, 5, 2, 2, 4, 7, 1, 5, 2, 1, 3, 3, 9, 1, 1, 1]\n",
      "Number of bins 205\n",
      "total # samples 47\n",
      "Number of total reads 346\n",
      "Number of hits in BIOM tables 372.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.931818181818182 %\n",
      "Percentage of reads from samples who only fall into one bin group 2.312138728323699 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 231\n",
      "Length of fail 1871\n",
      "G004341385 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Neisseriaceae; g__Uruburuella; s__Uruburuella suis\n",
      "Coverage Percentage: 0.061180893399955%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 231\n",
      "Length of fail 1872\n",
      "G004342465 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Schlegelella; s__Schlegelella thermodepolymerans\n",
      "Coverage Percentage: 0.1232586585303889%\n",
      "Adj, read counts 30\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 3]\n",
      "Number of bins 19\n",
      "total # samples 9\n",
      "Number of total reads 30\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 63.33333333333333 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.666666666666664 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 232\n",
      "Length of fail 1872\n",
      "G004342485 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Pelomonas; s__Pelomonas saccharophila\n",
      "Coverage Percentage: 0.1060598397133214%\n",
      "Adj, read counts 42\n",
      "Number of samples per group [1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 3, 5, 2]\n",
      "Number of bins 36\n",
      "total # samples 24\n",
      "Number of total reads 42\n",
      "Number of hits in BIOM tables 46.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 26.190476190476186 %\n",
      "Percentage of reads from samples who only fall into one bin group 33.33333333333333 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1872\n",
      "G004342585 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paucimonas; s__Paucimonas lemoignei\n",
      "Coverage Percentage: 0.0099386130535024%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1873\n",
      "G004342625 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Tepidimonas; s__Tepidimonas ignava\n",
      "Coverage Percentage: 0.084658936723629%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1874\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G004345285 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Raoultella; s__Raoultella planticola\n",
      "Coverage Percentage: 0.0429518808728298%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1875\n",
      "G004346085 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Thiobacillaceae; g__Sulfuritortus; s__Sulfuritortus calidifontis\n",
      "Coverage Percentage: 0.0627280043618712%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1876\n",
      "G004346095 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Faecalimonas; s__Faecalimonas umbilicata\n",
      "Coverage Percentage: 0.0286847490911899%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1877\n",
      "G004346505 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Chromobacteriaceae; g__Iodobacter; s__Iodobacter fluviatilis\n",
      "Coverage Percentage: 0.0026002276912558%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1878\n",
      "G004348265 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium segetis\n",
      "Coverage Percentage: 0.0026982118003344%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1879\n",
      "G004349085 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium caseinilyticum\n",
      "Coverage Percentage: 0.0593677426581587%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1880\n",
      "G004349135 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium sandaracinum\n",
      "Coverage Percentage: 0.0356530120840637%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1881\n",
      "G004349195 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium rhamnosiphilum\n",
      "Coverage Percentage: 0.0357630216709087%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1882\n",
      "G004349265 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Spirosomaceae; g__Dyadobacter; s__Dyadobacter psychrotolerans\n",
      "Coverage Percentage: 0.0089579246280221%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1883\n",
      "G004349315 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium ranwuense\n",
      "Coverage Percentage: 0.0351995614375554%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1884\n",
      "G004349355 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium cellulosilyticum\n",
      "Coverage Percentage: 0.0357887600107087%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1885\n",
      "G004352825 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Luteimonas; s__Luteimonas arsenica\n",
      "Coverage Percentage: 0.0179720755083433%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1886\n",
      "G004352845 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Luteimonas; s__Luteimonas terricola\n",
      "Coverage Percentage: 0.0121709885737443%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1887\n",
      "G004353185 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Clostridium; s__Clostridium cuniculi\n",
      "Coverage Percentage: 0.0251497856251526%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1888\n",
      "G004353865 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Hydrogenophaga; s__Hydrogenophaga pseudoflava\n",
      "Coverage Percentage: 0.0581737479038392%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1889\n",
      "G004353895 d__Bacteria; p__Candidatus Cloacimonetes; c__; o__; f__; g__Candidatus Syntrophosphaera; s__Candidatus Syntrophosphaera thermopropionivorans\n",
      "Coverage Percentage: 0.0251336614957435%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1890\n",
      "G004354475 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Latilactobacillus; s__Latilactobacillus sakei\n",
      "Coverage Percentage: 0.0217895634187308%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1891\n",
      "G004354505 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus thermophilus\n",
      "Coverage Percentage: 0.0895297401036888%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 233\n",
      "Length of fail 1892\n",
      "G004355225 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium nackdongense\n",
      "Coverage Percentage: 0.1256512869712728%\n",
      "Adj, read counts 35\n",
      "Number of samples per group [1, 8, 1, 7, 1, 1, 1, 2, 1, 1, 9]\n",
      "Number of bins 12\n",
      "total # samples 15\n",
      "Number of total reads 34\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 74.28571428571429 %\n",
      "Percentage of reads from samples who only fall into one bin group 23.52941176470588 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 234\n",
      "Length of fail 1892\n",
      "G004355245 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycobacteroides; s__Mycobacteroides chelonae\n",
      "Coverage Percentage: 0.0150687376970118%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 234\n",
      "Length of fail 1893\n",
      "G004361835 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Stakelama; s__Stakelama pacifica\n",
      "Coverage Percentage: 0.0278979952867203%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 234\n",
      "Length of fail 1894\n",
      "G004361965 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Pseudonocardia; s__Pseudonocardia autotrophica\n",
      "Coverage Percentage: 0.0357579428998357%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 234\n",
      "Length of fail 1895\n",
      "G004362405 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Kinneretia; s__Kinneretia asaccharophila\n",
      "Coverage Percentage: 0.0432426728330488%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 234\n",
      "Length of fail 1896\n",
      "G004362525 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Paucibacter; s__Paucibacter toxinivorans\n",
      "Coverage Percentage: 0.057166318503219%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 234\n",
      "Length of fail 1897\n",
      "G004362665 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium dankookense\n",
      "Coverage Percentage: 0.0289837295740201%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 234\n",
      "Length of fail 1898\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G004362855 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Aquabacterium; s__Aquabacterium commune\n",
      "Coverage Percentage: 0.1224658486740634%\n",
      "Adj, read counts 35\n",
      "Number of samples per group [1, 1, 1, 1, 1, 6, 1, 3, 1, 2, 3, 1, 5, 2, 4]\n",
      "Number of bins 21\n",
      "total # samples 19\n",
      "Number of total reads 35\n",
      "Number of hits in BIOM tables 26.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 45.714285714285715 %\n",
      "Percentage of reads from samples who only fall into one bin group 34.285714285714285 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 235\n",
      "Length of fail 1898\n",
      "G004363095 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Actinomycetospora; s__Actinomycetospora succinea\n",
      "Coverage Percentage: 0.0023354143782956%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 235\n",
      "Length of fail 1899\n",
      "G004363235 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Rhodospirillaceae; g__Dongia; s__Dongia mobilis\n",
      "Coverage Percentage: 0.0101669989238732%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 235\n",
      "Length of fail 1900\n",
      "G004363315 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Tepidicella; s__Tepidicella xavieri\n",
      "Coverage Percentage: 0.1070101116853107%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 235\n",
      "Length of fail 1901\n",
      "G004363655 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Rhodanobacteraceae; g__Tahibacter; s__Tahibacter aquaticus\n",
      "Coverage Percentage: 0.0110715560858899%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 235\n",
      "Length of fail 1902\n",
      "G004363695 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium cheniae\n",
      "Coverage Percentage: 0.1406407754043331%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 235\n",
      "Length of fail 1903\n",
      "G004364345 d__Bacteria; p__Verrucomicrobia; c__Verrucomicrobiae; o__Verrucomicrobiales; f__Verrucomicrobiaceae; g__Prosthecobacter; s__Prosthecobacter fusiformis\n",
      "Coverage Percentage: 0.009813455139128%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 235\n",
      "Length of fail 1904\n",
      "G004365655 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Epilithonimonas; s__Epilithonimonas xixisoli\n",
      "Coverage Percentage: 0.036523817629911%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 235\n",
      "Length of fail 1905\n",
      "G004367745 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Herbaspirillum; s__Herbaspirillum huttiense\n",
      "Coverage Percentage: 0.030324459417317%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 235\n",
      "Length of fail 1906\n",
      "G004369705 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium circumlabens\n",
      "Coverage Percentage: 0.0380249448033857%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 235\n",
      "Length of fail 1907\n",
      "G004378255 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Mucilaginibacter; s__Mucilaginibacter phyllosphaerae\n",
      "Coverage Percentage: 0.0034160832366008%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 235\n",
      "Length of fail 1908\n",
      "G004402005 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter alluvionis\n",
      "Coverage Percentage: 0.0336180532749407%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 235\n",
      "Length of fail 1909\n",
      "G004405045 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Dysgonomonadaceae; g__Dysgonomonas; s__Dysgonomonas capnocytophagoides\n",
      "Coverage Percentage: 0.0032116090948515%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 235\n",
      "Length of fail 1910\n",
      "G004421005 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia plicata\n",
      "Coverage Percentage: 0.0632513432418045%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 235\n",
      "Length of fail 1911\n",
      "G004462915 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Micrococcus; s__Micrococcus aloeverae\n",
      "Coverage Percentage: 0.7924293995205328%\n",
      "Adj, read counts 111\n",
      "Number of samples per group [1, 1, 1, 2, 2, 1, 2, 2, 3, 2, 1, 1, 1, 4, 1, 1, 3, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 1, 2, 1, 4, 3, 3, 1, 1, 1, 2, 1, 1, 1]\n",
      "Number of bins 81\n",
      "total # samples 23\n",
      "Number of total reads 110\n",
      "Number of hits in BIOM tables 55.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 13.513513513513514 %\n",
      "Percentage of reads from samples who only fall into one bin group 15.454545454545453 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 236\n",
      "Length of fail 1911\n",
      "G004519295 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Promicromonosporaceae; g__Cellulosimicrobium; s__Cellulosimicrobium funkei\n",
      "Coverage Percentage: 0.0053218667281231%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 236\n",
      "Length of fail 1912\n",
      "G004519315 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Mucilaginibacter; s__Mucilaginibacter psychrotolerans\n",
      "Coverage Percentage: 0.0435944486726338%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 236\n",
      "Length of fail 1913\n",
      "G004522235 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Amycolatopsis; s__Amycolatopsis nivea\n",
      "Coverage Percentage: 0.0084497859718303%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 236\n",
      "Length of fail 1914\n",
      "G004525795 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces viscosus\n",
      "Coverage Percentage: 0.0302531439306801%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 236\n",
      "Length of fail 1915\n",
      "G004526345 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Ornithinimicrobiaceae; g__Ornithinimicrobium; s__Ornithinimicrobium flavum\n",
      "Coverage Percentage: 0.0040399546124565%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 236\n",
      "Length of fail 1916\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G004535825 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella hominis\n",
      "Coverage Percentage: 0.089501696808073%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 236\n",
      "Length of fail 1917\n",
      "G004551575 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas indicatrix\n",
      "Coverage Percentage: 0.1032682203960662%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 236\n",
      "Length of fail 1918\n",
      "G004563945 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Kocuria; s__Kocuria rhizophila\n",
      "Coverage Percentage: 0.0466962058392815%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 236\n",
      "Length of fail 1919\n",
      "G004563965 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Micrococcus; s__Micrococcus flavus\n",
      "Coverage Percentage: 0.0876740731215264%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 236\n",
      "Length of fail 1920\n",
      "G004570865 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium frederickii\n",
      "Coverage Percentage: 0.0132699109529917%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 236\n",
      "Length of fail 1921\n",
      "G004571025 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium niftali\n",
      "Coverage Percentage: 0.0096215296740946%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 236\n",
      "Length of fail 1922\n",
      "G004614165 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Duganella; s__Duganella callida\n",
      "Coverage Percentage: 0.0352780141995149%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 236\n",
      "Length of fail 1923\n",
      "G004614185 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia arenosa\n",
      "Coverage Percentage: 0.0028833710226382%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 236\n",
      "Length of fail 1924\n",
      "G004614235 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas intermedia\n",
      "Coverage Percentage: 0.1489664907639878%\n",
      "Adj, read counts 33\n",
      "Number of samples per group [1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 4, 1, 1, 1, 1]\n",
      "Number of bins 29\n",
      "total # samples 18\n",
      "Number of total reads 33\n",
      "Number of hits in BIOM tables 30.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 27.27272727272727 %\n",
      "Percentage of reads from samples who only fall into one bin group 36.36363636363637 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 237\n",
      "Length of fail 1924\n",
      "G004681125 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingosinicellaceae; g__Polymorphobacter; s__Polymorphobacter arshaanensis\n",
      "Coverage Percentage: 0.0554941341349698%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 237\n",
      "Length of fail 1925\n",
      "G004681155 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Thermales; f__Thermaceae; g__Thermus; s__Thermus tengchongensis\n",
      "Coverage Percentage: 0.015611516628109%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 237\n",
      "Length of fail 1926\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G004682045 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas nabeulensis\n",
      "Coverage Percentage: 0.035528010673773%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 237\n",
      "Length of fail 1927\n",
      "G004682055 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas kairouanensis\n",
      "Coverage Percentage: 0.0368710664150524%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 237\n",
      "Length of fail 1928\n",
      "G004683865 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus liaowanqingii\n",
      "Coverage Percentage: 0.0210946135036386%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 237\n",
      "Length of fail 1929\n",
      "G004683965 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides seonyuensis\n",
      "Coverage Percentage: 0.0214374030347512%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 237\n",
      "Length of fail 1930\n",
      "G004684145 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia marcescens\n",
      "Coverage Percentage: 0.0397830902267959%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 237\n",
      "Length of fail 1931\n",
      "G004723625 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia contaminans\n",
      "Coverage Percentage: 0.0039992347531702%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 237\n",
      "Length of fail 1932\n",
      "G004745455 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas citronellolis\n",
      "Coverage Percentage: 0.5431025675432006%\n",
      "Adj, read counts 239\n",
      "Number of samples per group [1, 1, 3, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 3, 2, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 4, 1, 2, 3, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 2, 1, 2, 1, 3, 2, 1, 2, 2, 1, 1, 1, 1]\n",
      "Number of bins 151\n",
      "total # samples 14\n",
      "Number of total reads 234\n",
      "Number of hits in BIOM tables 78.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 5.857740585774058 %\n",
      "Percentage of reads from samples who only fall into one bin group 2.9914529914529915 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1932\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G004768745 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia nematodiphila\n",
      "Coverage Percentage: 0.0303810972883776%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1933\n",
      "G004770635 d__Bacteria; p__Spirochaetes; c__Spirochaetia; o__Leptospirales; f__Leptospiraceae; g__Leptospira; s__Leptospira wolffii\n",
      "Coverage Percentage: 0.0074507846922071%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1934\n",
      "G004770745 d__Bacteria; p__Spirochaetes; c__Spirochaetia; o__Leptospirales; f__Leptospiraceae; g__Leptospira; s__Leptospira ognonensis\n",
      "Coverage Percentage: 0.0037850938991555%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1935\n",
      "G004770765 d__Bacteria; p__Spirochaetes; c__Spirochaetia; o__Leptospirales; f__Leptospiraceae; g__Leptospira; s__Leptospira noumeaensis\n",
      "Coverage Percentage: 0.0073493981767651%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1936\n",
      "G004771075 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Weissella; s__Weissella confusa\n",
      "Coverage Percentage: 0.0047366042066458%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1937\n",
      "G004785795 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides eburneiflavus\n",
      "Coverage Percentage: 0.0283711725677571%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1938\n",
      "G004785935 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus rubneri\n",
      "Coverage Percentage: 0.0397658122340748%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1939\n",
      "G004786095 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Beutenbergiaceae; g__Serinibacter; s__Serinibacter arcticus\n",
      "Coverage Percentage: 0.029179871067445%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1940\n",
      "G004792695 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas naasensis\n",
      "Coverage Percentage: 0.0104379635878668%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1941\n",
      "G004795735 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium endometrii\n",
      "Coverage Percentage: 0.0299144833332727%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1942\n",
      "G004797125 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium sangjuense\n",
      "Coverage Percentage: 0.0766051461535463%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1943\n",
      "G004799035 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Rhodanobacteraceae; g__Rhodanobacter; s__Rhodanobacter lindaniclasticus\n",
      "Coverage Percentage: 0.0102395866357641%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1944\n",
      "G004801285 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Phyllobacteriaceae; g__Mesorhizobium; s__Mesorhizobium composti\n",
      "Coverage Percentage: 0.0032443535357007%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1945\n",
      "G004801885 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Naasia; s__Naasia lichenicola\n",
      "Coverage Percentage: 0.0187298232978523%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1946\n",
      "G004920405 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas leptonychotis\n",
      "Coverage Percentage: 0.0034357685347766%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1947\n",
      "G004923205 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus aeridis\n",
      "Coverage Percentage: 0.0090899595978385%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1948\n",
      "G004924335 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycobacteroides; s__Mycobacteroides salmoniphilum\n",
      "Coverage Percentage: 0.0057361812984997%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1949\n",
      "G005048205 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Chromobacteriaceae; g__Chitiniphilus; s__Chitiniphilus eburneus\n",
      "Coverage Percentage: 0.014977665995744%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1950\n",
      "G005048225 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus gahaiensis\n",
      "Coverage Percentage: 0.0268186229282494%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1951\n",
      "G005048265 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus hibiscisoli\n",
      "Coverage Percentage: 0.038955714194826%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1952\n",
      "G005049235 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Rhodococcus; s__Rhodococcus oryzae\n",
      "Coverage Percentage: 0.0039877453232039%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1953\n",
      "G005121165 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Oscillospiraceae; g__Dysosmobacter; s__Dysosmobacter welbionis\n",
      "Coverage Percentage: 0.0420848234296977%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1954\n",
      "G005155025 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Chromobacteriaceae; g__Aquitalea; s__Aquitalea aquatilis\n",
      "Coverage Percentage: 0.014656329710995%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1955\n",
      "G005250675 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides dongxiaopingii\n",
      "Coverage Percentage: 0.0137059910611523%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1956\n",
      "G005280585 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Spirosomaceae; g__Dyadobacter; s__Dyadobacter frigoris\n",
      "Coverage Percentage: 0.0147233716064998%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1957\n",
      "G005298075 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Nakamurellales; f__Nakamurellaceae; g__Nakamurella; s__Nakamurella flava\n",
      "Coverage Percentage: 0.0031175686763216%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 238\n",
      "Length of fail 1958\n",
      "G005403045 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Aquabacterium; s__Aquabacterium pictum\n",
      "Coverage Percentage: 0.1817454930995035%\n",
      "Adj, read counts 77\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 6, 1, 1, 1, 2, 10, 1, 3, 1, 1, 2, 3, 2, 1]\n",
      "Number of bins 58\n",
      "total # samples 32\n",
      "Number of total reads 76\n",
      "Number of hits in BIOM tables 147.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 25.97402597402597 %\n",
      "Percentage of reads from samples who only fall into one bin group 19.736842105263158 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1958\n",
      "G005405165 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Loigolactobacillus; s__Loigolactobacillus binensis\n",
      "Coverage Percentage: 0.0111659263680893%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1959\n",
      "G005405205 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Enterococcaceae; g__Enterococcus; s__Enterococcus pingfangensis\n",
      "Coverage Percentage: 0.0174941013161213%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1960\n",
      "G005405365 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Enterococcaceae; g__Enterococcus; s__Enterococcus xiangfangensis\n",
      "Coverage Percentage: 0.018998070414237%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1961\n",
      "G005405545 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Weissella; s__Weissella hellenica\n",
      "Coverage Percentage: 0.0232486776625933%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1962\n",
      "G005434915 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Aurantiacibacter; s__Aurantiacibacter suaedae\n",
      "Coverage Percentage: 0.0046337667183848%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1963\n",
      "G005473895 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Carideicomes; s__Carideicomes alvinocaridis\n",
      "Coverage Percentage: 0.0408123621654357%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1964\n",
      "G005498985 d__Eukaryota; p__Basidiomycota; c__Ustilaginomycetes; o__Ustilaginales; f__Ustilaginaceae; g__Sporisorium; s__Sporisorium graminicola\n",
      "Coverage Percentage: 0.0031482753890522%\n",
      "Adj, read counts 108\n",
      "Number of samples per group [80, 13, 13, 2]\n",
      "Number of bins 4\n",
      "total # samples 81\n",
      "Number of total reads 108\n",
      "Number of hits in BIOM tables 276.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 98.14814814814814 %\n",
      "Percentage of reads from samples who only fall into one bin group 52.77777777777778 %\n",
      "Status: Fail - too many reads in top 3 bins\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1965\n",
      "G005503535 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Phyllobacteriaceae; g__Mesorhizobium; s__Mesorhizobium comanense\n",
      "Coverage Percentage: 0.0055943622949757%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1966\n",
      "G005670655 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacterium; s__Mycolicibacterium phocaicum\n",
      "Coverage Percentage: 0.0078467061935108%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1967\n",
      "G005670695 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacterium; s__Mycolicibacterium aubagnense\n",
      "Coverage Percentage: 0.0060226247244406%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1968\n",
      "G005818875 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Aliarcobacter; s__Aliarcobacter cibarius\n",
      "Coverage Percentage: 0.0826825268891915%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1969\n",
      "G005818985 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Aliarcobacter; s__Aliarcobacter thereius\n",
      "Coverage Percentage: 0.0385126224988527%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1970\n",
      "G005845345 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Rhodoluna; s__Rhodoluna limnophila\n",
      "Coverage Percentage: 0.1610668890990575%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1971\n",
      "G005845365 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Rhodoluna; s__Rhodoluna limnophila\n",
      "Coverage Percentage: 0.1459207952504604%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1972\n",
      "G005860805 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Spirosomaceae; g__Dyadobacter; s__Dyadobacter luticola\n",
      "Coverage Percentage: 0.0031830382755671%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1973\n",
      "G005862305 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium indicum\n",
      "Coverage Percentage: 0.0204828020791836%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1974\n",
      "G005862345 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Thermomonas; s__Thermomonas fusca\n",
      "Coverage Percentage: 0.0258371850834696%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1975\n",
      "G005863185 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas edaphica\n",
      "Coverage Percentage: 0.0430154241443637%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1976\n",
      "G005864045 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Aerococcaceae; g__Ruoffia; s__Ruoffia tabacinasalis\n",
      "Coverage Percentage: 0.0232802000114219%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1977\n",
      "G005869865 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces tibetensis\n",
      "Coverage Percentage: 0.0028014639205353%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1978\n",
      "G005870035 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Cytophagaceae; g__Spirosoma; s__Spirosoma lacussanchae\n",
      "Coverage Percentage: 0.0063648093952799%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1979\n",
      "G005876985 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Rhodoferax; s__Rhodoferax bucti\n",
      "Coverage Percentage: 0.0993564695080439%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1980\n",
      "G005877885 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Marinilabiliales; f__Marinilabiliaceae; g__Labilibacter; s__Labilibacter sediminis\n",
      "Coverage Percentage: 0.0047337560065407%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1981\n",
      "G005877905 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas nicosulfuronedens\n",
      "Coverage Percentage: 0.0131137626393992%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1982\n",
      "G005887635 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paraoceanicella; s__Paraoceanicella profunda\n",
      "Coverage Percentage: 0.0051383298745607%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1983\n",
      "G005925365 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter xixiisoli\n",
      "Coverage Percentage: 0.0030325785292235%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1984\n",
      "G005930475 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Modestobacter; s__Modestobacter altitudinis\n",
      "Coverage Percentage: 0.0161046174487157%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1985\n",
      "G005930495 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Modestobacter; s__Modestobacter excelsi\n",
      "Coverage Percentage: 0.0225689169543712%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1986\n",
      "G005938105 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Brucellaceae; g__Brucella; s__Brucella haematophila\n",
      "Coverage Percentage: 0.0027432888436074%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1987\n",
      "G006094315 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia cepacia\n",
      "Coverage Percentage: 0.0047116620282842%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1988\n",
      "G006094335 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__Bacillus licheniformis\n",
      "Coverage Percentage: 0.0333553107245344%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 239\n",
      "Length of fail 1989\n",
      "G006094355 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus saprophyticus\n",
      "Coverage Percentage: 0.3096831575833326%\n",
      "Adj, read counts 45\n",
      "Number of samples per group [1, 3, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 38\n",
      "total # samples 23\n",
      "Number of total reads 45\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.555555555555555 %\n",
      "Percentage of reads from samples who only fall into one bin group 31.11111111111111 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 240\n",
      "Length of fail 1989\n",
      "G006094375 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus epidermidis\n",
      "Coverage Percentage: 0.3640161888452401%\n",
      "Adj, read counts 51\n",
      "Number of samples per group [2, 1, 1, 2, 3, 1, 1, 1, 1, 1, 1, 3, 1, 1, 5, 2, 2, 1, 1, 1, 1, 1, 2, 3, 1, 2, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 43\n",
      "total # samples 19\n",
      "Number of total reads 51\n",
      "Number of hits in BIOM tables 19.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 21.568627450980394 %\n",
      "Percentage of reads from samples who only fall into one bin group 17.647058823529413 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 241\n",
      "Length of fail 1989\n",
      "G006094395 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus haemolyticus\n",
      "Coverage Percentage: 0.0532256741367934%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 241\n",
      "Length of fail 1990\n",
      "G006165035 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Aliarcobacter; s__Aliarcobacter cryaerophilus\n",
      "Coverage Percentage: 0.7521701596931901%\n",
      "Adj, read counts 98\n",
      "Number of samples per group [1, 1, 4, 1, 5, 4, 2, 1, 3, 2, 2, 3, 2, 1, 1, 1, 1, 1, 5, 6, 2, 2, 1, 1, 2, 1, 1, 3, 1, 1, 1, 2, 1, 2, 1, 2, 4, 1, 2, 2, 3, 2, 1, 1, 1, 1, 1]\n",
      "Number of bins 74\n",
      "total # samples 30\n",
      "Number of total reads 96\n",
      "Number of hits in BIOM tables 60.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.408163265306122 %\n",
      "Percentage of reads from samples who only fall into one bin group 13.541666666666666 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 242\n",
      "Length of fail 1990\n",
      "G006175945 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium mulctrae\n",
      "Coverage Percentage: 0.0202106882088503%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 242\n",
      "Length of fail 1991\n",
      "G006239215 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus haeundaensis\n",
      "Coverage Percentage: 0.2787199915858115%\n",
      "Adj, read counts 68\n",
      "Number of samples per group [1, 2, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 3, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 59\n",
      "total # samples 17\n",
      "Number of total reads 67\n",
      "Number of hits in BIOM tables 77.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.294117647058822 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.940298507462686 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 243\n",
      "Length of fail 1991\n",
      "G006265165 d__Bacteria; p__Chlorobi; c__Chlorobia; o__Chlorobiales; f__Chlorobiaceae; g__Chlorobaculum; s__Chlorobaculum thiosulfatiphilum\n",
      "Coverage Percentage: 0.0100593309282735%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 243\n",
      "Length of fail 1992\n",
      "G006335125 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Deinococcales; f__Deinococcaceae; g__Deinococcus; s__Deinococcus radiopugnans\n",
      "Coverage Percentage: 0.009680067977642%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 243\n",
      "Length of fail 1993\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G006335145 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium smilacinae\n",
      "Coverage Percentage: 0.001731425590903%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 243\n",
      "Length of fail 1994\n",
      "G006337125 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Ornithinimicrobiaceae; g__Serinicoccus; s__Serinicoccus chungangensis\n",
      "Coverage Percentage: 0.0062365704406194%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 243\n",
      "Length of fail 1995\n",
      "G006346315 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides litoris\n",
      "Coverage Percentage: 0.0251363232320803%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 243\n",
      "Length of fail 1996\n",
      "G006352475 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter pullicarnis\n",
      "Coverage Percentage: 0.4082808134198863%\n",
      "Adj, read counts 114\n",
      "Number of samples per group [6, 5, 1, 3, 2, 4, 5, 1, 2, 1, 1, 2, 2, 1, 3, 1, 1, 2, 1, 2, 5, 1, 1, 1, 2, 2, 1, 3, 1, 2, 1, 4, 1, 2, 2, 1, 2, 1, 1, 1, 14]\n",
      "Number of bins 53\n",
      "total # samples 29\n",
      "Number of total reads 111\n",
      "Number of hits in BIOM tables 27.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 28.947368421052634 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.216216216216218 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 244\n",
      "Length of fail 1996\n",
      "G006363815 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptosporangiales; f__Thermomonosporaceae; g__Thermomonospora; s__Thermomonospora catenispora\n",
      "Coverage Percentage: 0.0040942507745425%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 244\n",
      "Length of fail 1997\n",
      "G006363895 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Methylophilaceae; g__Candidatus Methylopumilus; s__Candidatus Methylopumilus universalis\n",
      "Coverage Percentage: 0.0856167080966468%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 244\n",
      "Length of fail 1998\n",
      "G006363915 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Methylophilaceae; g__Methylophilus; s__Methylophilus medardicus\n",
      "Coverage Percentage: 0.0352659157613288%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 244\n",
      "Length of fail 1999\n",
      "G006364575 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Methylophilaceae; g__Candidatus Methylopumilus; s__Candidatus Methylopumilus planktonicus\n",
      "Coverage Percentage: 0.0462562010541504%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 244\n",
      "Length of fail 2000\n",
      "G006364615 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Methylophilaceae; g__Candidatus Methylopumilus; s__Candidatus Methylopumilus rimovensis\n",
      "Coverage Percentage: 0.162629614946347%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 244\n",
      "Length of fail 2001\n",
      "G006365335 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Salmonella; s__Salmonella enterica\n",
      "Coverage Percentage: 0.0222005912219265%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 244\n",
      "Length of fail 2002\n",
      "G006370295 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Trichococcus; s__Trichococcus shcherbakoviae\n",
      "Coverage Percentage: 0.184255471998336%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 3, 1, 2, 2, 1, 1, 1, 1]\n",
      "Number of bins 33\n",
      "total # samples 18\n",
      "Number of total reads 36\n",
      "Number of hits in BIOM tables 19.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 18.91891891891892 %\n",
      "Percentage of reads from samples who only fall into one bin group 25.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 245\n",
      "Length of fail 2002\n",
      "G006385165 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus symci\n",
      "Coverage Percentage: 0.1103438700463435%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 245\n",
      "Length of fail 2003\n",
      "G006385365 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Tannerellaceae; g__Tannerella; s__Tannerella forsythia\n",
      "Coverage Percentage: 0.0199515220464318%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 245\n",
      "Length of fail 2004\n",
      "G006385685 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Amaricoccus; s__Amaricoccus solimangrovi\n",
      "Coverage Percentage: 0.0030994742388532%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 245\n",
      "Length of fail 2005\n",
      "G006385785 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus shenyangsis\n",
      "Coverage Percentage: 0.114902026871754%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 245\n",
      "Length of fail 2006\n",
      "G006385805 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus xiaochunlingii\n",
      "Coverage Percentage: 0.0392671700314598%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 245\n",
      "Length of fail 2007\n",
      "G006386545 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Pseudorivibacter; s__Pseudorivibacter rhizosphaerae\n",
      "Coverage Percentage: 0.0507132635174803%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 245\n",
      "Length of fail 2008\n",
      "G006438825 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Roseomonas; s__Roseomonas nepalensis\n",
      "Coverage Percentage: 0.0064083642153271%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 245\n",
      "Length of fail 2009\n",
      "G006438845 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Variovorax; s__Variovorax ginsengisoli\n",
      "Coverage Percentage: 0.0921647857190686%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 245\n",
      "Length of fail 2010\n",
      "G006438965 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Intrasporangiaceae; g__Pedococcus; s__Pedococcus bigeumensis\n",
      "Coverage Percentage: 0.0069601234640676%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 245\n",
      "Length of fail 2011\n",
      "G006494355 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Erwiniaceae; g__Pantoea; s__Pantoea eucalypti\n",
      "Coverage Percentage: 0.0139121311475409%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 245\n",
      "Length of fail 2012\n",
      "G006494775 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides humi\n",
      "Coverage Percentage: 0.0161881022110664%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 245\n",
      "Length of fail 2013\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G006503695 d__Bacteria; p__Proteobacteria; c__Hydrogenophilalia; o__Hydrogenophilales; f__Hydrogenophilaceae; g__Tepidiphilus; s__Tepidiphilus succinatimandens\n",
      "Coverage Percentage: 0.0818436937324605%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 245\n",
      "Length of fail 2014\n",
      "G006517255 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Janthinobacterium; s__Janthinobacterium tructae\n",
      "Coverage Percentage: 0.1653846702046285%\n",
      "Adj, read counts 73\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 7, 4, 1, 1, 1, 3, 4, 1, 1, 1, 5, 1, 1, 1, 6, 3, 1, 2, 1, 1]\n",
      "Number of bins 41\n",
      "total # samples 22\n",
      "Number of total reads 72\n",
      "Number of hits in BIOM tables 19.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 31.506849315068493 %\n",
      "Percentage of reads from samples who only fall into one bin group 12.5 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 246\n",
      "Length of fail 2014\n",
      "G006519715 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Ideonella; s__Ideonella azotifigens\n",
      "Coverage Percentage: 0.2250434989122685%\n",
      "Adj, read counts 93\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 2, 3, 1, 1, 1, 2, 1, 1, 1, 1, 3, 1, 1, 1, 4, 1, 1, 1, 12, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 72\n",
      "total # samples 30\n",
      "Number of total reads 93\n",
      "Number of hits in BIOM tables 161.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 24.731182795698928 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.827956989247312 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 247\n",
      "Length of fail 2014\n",
      "G006538485 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium striatum\n",
      "Coverage Percentage: 0.0807140075403706%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 247\n",
      "Length of fail 2015\n",
      "G006538965 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Pimelobacter; s__Pimelobacter simplex\n",
      "Coverage Percentage: 0.0080664992942258%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 247\n",
      "Length of fail 2016\n",
      "G006538985 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Paenarthrobacter; s__Paenarthrobacter aurescens\n",
      "Coverage Percentage: 0.0060958254489355%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 247\n",
      "Length of fail 2017\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G006539065 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae; g__Brevibacillus; s__Brevibacillus parabrevis\n",
      "Coverage Percentage: 0.0208293943878657%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 247\n",
      "Length of fail 2018\n",
      "G006539145 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium testaceum\n",
      "Coverage Percentage: 0.0188251334468039%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 247\n",
      "Length of fail 2019\n",
      "G006539445 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium lacticum\n",
      "Coverage Percentage: 0.1698827625300231%\n",
      "Adj, read counts 31\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 4, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1]\n",
      "Number of bins 27\n",
      "total # samples 23\n",
      "Number of total reads 31\n",
      "Number of hits in BIOM tables 23.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 25.806451612903224 %\n",
      "Percentage of reads from samples who only fall into one bin group 70.96774193548387 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 248\n",
      "Length of fail 2019\n",
      "G006539525 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Glutamicibacter; s__Glutamicibacter nicotianae\n",
      "Coverage Percentage: 0.0084933404617396%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 248\n",
      "Length of fail 2020\n",
      "G006539565 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Pseudonocardia; s__Pseudonocardia hydrocarbonoxydans\n",
      "Coverage Percentage: 0.0142673246895392%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 248\n",
      "Length of fail 2021\n",
      "G006539585 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Pseudonocardia; s__Pseudonocardia saturnea\n",
      "Coverage Percentage: 0.0247634220858348%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 248\n",
      "Length of fail 2022\n",
      "G006539645 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium japonicum\n",
      "Coverage Percentage: 0.0291316056272899%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 248\n",
      "Length of fail 2023\n",
      "G006539665 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium elkanii\n",
      "Coverage Percentage: 0.0092920414150926%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 248\n",
      "Length of fail 2024\n",
      "G006539725 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Raoultella; s__Raoultella terrigena\n",
      "Coverage Percentage: 0.0229499204094584%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 248\n",
      "Length of fail 2025\n",
      "G006539765 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium liquefaciens\n",
      "Coverage Percentage: 0.082351861152062%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 248\n",
      "Length of fail 2026\n",
      "G006539785 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium aquatile\n",
      "Coverage Percentage: 0.0544212132911954%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 248\n",
      "Length of fail 2027\n",
      "G006539865 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Zoogloeaceae; g__Zoogloea; s__Zoogloea ramigera\n",
      "Coverage Percentage: 0.3083528146223659%\n",
      "Adj, read counts 84\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 7, 1, 2, 1, 2, 1, 4, 1, 1, 2, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 3, 1, 3, 1, 1]\n",
      "Number of bins 69\n",
      "total # samples 31\n",
      "Number of total reads 84\n",
      "Number of hits in BIOM tables 167.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.857142857142854 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.904761904761903 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2027\n",
      "G006539885 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Kocuria; s__Kocuria varians\n",
      "Coverage Percentage: 0.0196518493831639%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2028\n",
      "G006539945 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Promicromonosporaceae; g__Cellulosimicrobium; s__Cellulosimicrobium cellulans\n",
      "Coverage Percentage: 0.006004669283049%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2029\n",
      "G006540065 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Weissella; s__Weissella hellenica\n",
      "Coverage Percentage: 0.0233262064955193%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2030\n",
      "G006540085 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium oxydans\n",
      "Coverage Percentage: 0.0479859515218585%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2031\n",
      "G006540285 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Latilactobacillus; s__Latilactobacillus curvatus\n",
      "Coverage Percentage: 0.0130919914023732%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2032\n",
      "G006542275 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae; g__Saccharibacillus; s__Saccharibacillus brassicae\n",
      "Coverage Percentage: 0.0026870871050426%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2033\n",
      "G006542415 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Kytococcaceae; g__Kytococcus; s__Kytococcus schroeteri\n",
      "Coverage Percentage: 0.0096419200500373%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2034\n",
      "G006542645 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Rikenellaceae; g__Alistipes; s__Alistipes onderdonkii\n",
      "Coverage Percentage: 0.0706772816588034%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2035\n",
      "G006542665 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Rikenellaceae; g__Alistipes; s__Alistipes communis\n",
      "Coverage Percentage: 0.0239296263010219%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2036\n",
      "G006542685 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Rikenellaceae; g__Alistipes; s__Alistipes dispar\n",
      "Coverage Percentage: 0.0099582227239216%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2037\n",
      "G006546825 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces oris\n",
      "Coverage Percentage: 0.1146935898431016%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2038\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G006546835 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces johnsonii\n",
      "Coverage Percentage: 0.0418618177822552%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2039\n",
      "G006547045 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Lysobacter; s__Lysobacter alkalisoli\n",
      "Coverage Percentage: 0.0134298101859665%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2040\n",
      "G006711645 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Raoultella; s__Raoultella electrica\n",
      "Coverage Percentage: 0.0668245034011183%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2041\n",
      "G006715055 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Intrasporangiaceae; g__Lapillicoccus; s__Lapillicoccus jejuensis\n",
      "Coverage Percentage: 0.0098928233790346%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2042\n",
      "G006716085 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Leucobacter; s__Leucobacter komagatae\n",
      "Coverage Percentage: 0.0116460134256948%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2043\n",
      "G006716095 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Dermacoccaceae; g__Barrientosiimonas; s__Barrientosiimonas humi\n",
      "Coverage Percentage: 0.0320114374369783%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2044\n",
      "G006716345 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium saperdae\n",
      "Coverage Percentage: 0.0158001965525917%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2045\n",
      "G006716545 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microcella; s__Microcella putealis\n",
      "Coverage Percentage: 0.0355606214395222%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2046\n",
      "G006716765 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptosporangiales; f__Thermomonosporaceae; g__Actinomadura; s__Actinomadura hallensis\n",
      "Coverage Percentage: 0.0015071227966349%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 249\n",
      "Length of fail 2047\n",
      "G006716815 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium lacticum\n",
      "Coverage Percentage: 0.2280483980895764%\n",
      "Adj, read counts 45\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1]\n",
      "Number of bins 34\n",
      "total # samples 23\n",
      "Number of total reads 45\n",
      "Number of hits in BIOM tables 23.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.555555555555555 %\n",
      "Percentage of reads from samples who only fall into one bin group 42.22222222222222 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 250\n",
      "Length of fail 2047\n",
      "G006716885 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Ornithinimicrobiaceae; g__Ornithinimicrobium; s__Ornithinimicrobium humiphilum\n",
      "Coverage Percentage: 0.0069517949738231%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 250\n",
      "Length of fail 2048\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G006716905 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax temperans\n",
      "Coverage Percentage: 0.5631094009795562%\n",
      "Adj, read counts 180\n",
      "Number of samples per group [1, 4, 2, 1, 2, 2, 1, 1, 2, 6, 1, 1, 2, 1, 2, 1, 2, 1, 1, 1, 1, 3, 1, 1, 1, 1, 9, 1, 1, 1, 9, 2, 1, 8, 1, 1, 2, 2, 3, 1, 2, 1, 3, 1, 1, 1, 3, 1, 6, 2, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 4, 1, 1, 3, 2, 12, 3, 9]\n",
      "Number of bins 98\n",
      "total # samples 42\n",
      "Number of total reads 178\n",
      "Number of hits in BIOM tables 107.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.555555555555554 %\n",
      "Percentage of reads from samples who only fall into one bin group 8.98876404494382 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 251\n",
      "Length of fail 2048\n",
      "G006738645 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Stellaceae; g__Stella; s__Stella humosa\n",
      "Coverage Percentage: 0.0102526295937524%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 251\n",
      "Length of fail 2049\n",
      "G006742205 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus epidermidis\n",
      "Coverage Percentage: 0.3622900289691936%\n",
      "Adj, read counts 50\n",
      "Number of samples per group [2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 4, 2, 2, 1, 1, 1, 1, 1, 2, 3, 1, 2, 1, 1, 2, 1, 1]\n",
      "Number of bins 42\n",
      "total # samples 19\n",
      "Number of total reads 49\n",
      "Number of hits in BIOM tables 19.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 20.408163265306122 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 252\n",
      "Length of fail 2049\n",
      "G006757745 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter radioresistens\n",
      "Coverage Percentage: 0.4698176930303095%\n",
      "Adj, read counts 97\n",
      "Number of samples per group [3, 3, 1, 4, 1, 1, 3, 1, 4, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 5, 4, 1, 1, 1, 1, 3, 7, 6, 1, 6, 1]\n",
      "Number of bins 61\n",
      "total # samples 33\n",
      "Number of total reads 96\n",
      "Number of hits in BIOM tables 36.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 22.68041237113402 %\n",
      "Percentage of reads from samples who only fall into one bin group 15.625 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 253\n",
      "Length of fail 2049\n",
      "G006770285 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Lactococcus; s__Lactococcus paracarnosus\n",
      "Coverage Percentage: 0.0217205117564477%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 253\n",
      "Length of fail 2050\n",
      "G006864685 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycobacterium; s__Mycobacterium hodleri\n",
      "Coverage Percentage: 0.0490363667313714%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 253\n",
      "Length of fail 2051\n",
      "G006970865 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Rhodoferax; s__Rhodoferax sediminis\n",
      "Coverage Percentage: 0.0405926203482304%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 253\n",
      "Length of fail 2052\n",
      "G007035645 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Enterobacter; s__Enterobacter asburiae\n",
      "Coverage Percentage: 0.0114657192810428%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 253\n",
      "Length of fail 2053\n",
      "G007095465 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactobacillus; s__Lactobacillus mulieris\n",
      "Coverage Percentage: 0.007760537486404%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 253\n",
      "Length of fail 2054\n",
      "G007097145 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium zepuense\n",
      "Coverage Percentage: 0.0282232566780155%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 253\n",
      "Length of fail 2055\n",
      "G007097265 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium gawalongense\n",
      "Coverage Percentage: 0.0471633412121127%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 253\n",
      "Length of fail 2056\n",
      "G007097285 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium gawalongense\n",
      "Coverage Percentage: 0.0683184573289595%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 253\n",
      "Length of fail 2057\n",
      "G007097365 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium franklandianum\n",
      "Coverage Percentage: 0.0517022670842927%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 253\n",
      "Length of fail 2058\n",
      "G007197755 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Rhodospirillaceae; g__Ferrovibrio; s__Ferrovibrio terrae\n",
      "Coverage Percentage: 0.0102705179389368%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 253\n",
      "Length of fail 2059\n",
      "G007197895 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides zhangwenhongi\n",
      "Coverage Percentage: 0.0595693376094138%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 253\n",
      "Length of fail 2060\n",
      "G007280415 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingorhabdus; s__Sphingorhabdus contaminans\n",
      "Coverage Percentage: 0.0342040584968379%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 253\n",
      "Length of fail 2061\n",
      "G007290395 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium cosmicum\n",
      "Coverage Percentage: 0.0192219860090228%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 253\n",
      "Length of fail 2062\n",
      "G007341375 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides koreensis\n",
      "Coverage Percentage: 0.1365015997347093%\n",
      "Adj, read counts 60\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 2, 1, 1, 1, 1, 6, 1, 1, 1]\n",
      "Number of bins 51\n",
      "total # samples 34\n",
      "Number of total reads 60\n",
      "Number of hits in BIOM tables 7.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 36.666666666666664 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 254\n",
      "Length of fail 2062\n",
      "G007341385 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium daemonense\n",
      "Coverage Percentage: 0.0285370530273583%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 254\n",
      "Length of fail 2063\n",
      "G007341395 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides kribbi\n",
      "Coverage Percentage: 0.1359712962796576%\n",
      "Adj, read counts 64\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 2, 2, 2, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 4, 1, 1, 1, 1]\n",
      "Number of bins 55\n",
      "total # samples 36\n",
      "Number of total reads 60\n",
      "Number of hits in BIOM tables 7.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 18.75 %\n",
      "Percentage of reads from samples who only fall into one bin group 40.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 255\n",
      "Length of fail 2063\n",
      "G007556585 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Tepidimonas; s__Tepidimonas aquatica\n",
      "Coverage Percentage: 0.0979877554560599%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 255\n",
      "Length of fail 2064\n",
      "G007556615 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Tepidimonas; s__Tepidimonas ignava\n",
      "Coverage Percentage: 0.0815369437818417%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 255\n",
      "Length of fail 2065\n",
      "G007556675 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Tepidimonas; s__Tepidimonas taiwanensis\n",
      "Coverage Percentage: 0.0277435297377486%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 255\n",
      "Length of fail 2066\n",
      "G007556755 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Tepidimonas; s__Tepidimonas fonticaldi\n",
      "Coverage Percentage: 0.1960969240180043%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 6, 4, 1]\n",
      "Number of bins 29\n",
      "total # samples 18\n",
      "Number of total reads 36\n",
      "Number of hits in BIOM tables 12.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 35.13513513513514 %\n",
      "Percentage of reads from samples who only fall into one bin group 27.77777777777778 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 256\n",
      "Length of fail 2066\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G007556815 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia megapolitana\n",
      "Coverage Percentage: 0.0023336213214327%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 256\n",
      "Length of fail 2067\n",
      "G007558865 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Mucilaginibacter; s__Mucilaginibacter corticis\n",
      "Coverage Percentage: 0.0111512974300288%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 256\n",
      "Length of fail 2068\n",
      "G007559235 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium godavarianum\n",
      "Coverage Percentage: 0.1679341100697624%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 256\n",
      "Length of fail 2069\n",
      "G007559335 d__Bacteria; p__Verrucomicrobia; c__Opitutae; o__Opitutales; f__Opitutaceae; g__Rariglobus; s__Rariglobus hedericola\n",
      "Coverage Percentage: 0.0563987504988239%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 256\n",
      "Length of fail 2070\n",
      "G007632255 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Klebsiella; s__Klebsiella aerogenes\n",
      "Coverage Percentage: 0.0059627145553291%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 256\n",
      "Length of fail 2071\n",
      "G007641235 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium sanguinis\n",
      "Coverage Percentage: 0.014939364378715%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 256\n",
      "Length of fail 2072\n",
      "G007655005 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces tibetensis\n",
      "Coverage Percentage: 0.004068792836968%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 256\n",
      "Length of fail 2073\n",
      "G007655255 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Extensimonas; s__Extensimonas perlucida\n",
      "Coverage Percentage: 0.0926727854180657%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 256\n",
      "Length of fail 2074\n",
      "G007665635 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas oryzihabitans\n",
      "Coverage Percentage: 0.0050735585779436%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 256\n",
      "Length of fail 2075\n",
      "G007681555 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Kocuria; s__Kocuria salsicia\n",
      "Coverage Percentage: 0.033662951092541%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 256\n",
      "Length of fail 2076\n",
      "G007743815 d__Bacteria; p__Planctomycetes; c__Planctomycetia; o__Planctomycetales; f__Planctomycetaceae; g__Alienimonas; s__Alienimonas californiensis\n",
      "Coverage Percentage: 0.0019725252798291%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 256\n",
      "Length of fail 2077\n",
      "G007746075 d__Bacteria; p__Planctomycetes; c__Planctomycetia; o__Pirellulales; f__Lacipirellulaceae; g__Lacipirellula; s__Lacipirellula limnantheis\n",
      "Coverage Percentage: 0.0743961788295586%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 256\n",
      "Length of fail 2078\n",
      "G007747215 d__Bacteria; p__Planctomycetes; c__Planctomycetia; o__Gemmatales; f__Gemmataceae; g__Urbifossiella; s__Urbifossiella limnaea\n",
      "Coverage Percentage: 0.0186382578398374%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 256\n",
      "Length of fail 2079\n",
      "G007747655 d__Bacteria; p__Planctomycetes; c__Planctomycetia; o__Pirellulales; f__Pirellulaceae; g__Anatilimnocola; s__Anatilimnocola aggregata\n",
      "Coverage Percentage: 0.0012544767055887%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 256\n",
      "Length of fail 2080\n",
      "G007752345 d__Bacteria; p__Planctomycetes; c__Planctomycetia; o__Planctomycetales; f__Planctomycetaceae; g__Planctopirus; s__Planctopirus ephydatiae\n",
      "Coverage Percentage: 0.0057920304346266%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 256\n",
      "Length of fail 2081\n",
      "G007785815 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas solaris\n",
      "Coverage Percentage: 0.1223481768817916%\n",
      "Adj, read counts 34\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 31\n",
      "total # samples 20\n",
      "Number of total reads 34\n",
      "Number of hits in BIOM tables 63.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.705882352941178 %\n",
      "Percentage of reads from samples who only fall into one bin group 35.294117647058826 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 257\n",
      "Length of fail 2081\n",
      "G007795095 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Luteimonas; s__Luteimonas granuli\n",
      "Coverage Percentage: 0.028621335954842%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 257\n",
      "Length of fail 2082\n",
      "G007814115 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus saprophyticus\n",
      "Coverage Percentage: 0.3084442006230345%\n",
      "Adj, read counts 46\n",
      "Number of samples per group [1, 3, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 2]\n",
      "Number of bins 41\n",
      "total # samples 23\n",
      "Number of total reads 46\n",
      "Number of hits in BIOM tables 11.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 21.73913043478261 %\n",
      "Percentage of reads from samples who only fall into one bin group 30.434782608695656 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2082\n",
      "G007827045 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptosporangiales; f__Treboniaceae; g__Trebonia; s__Trebonia kvetii\n",
      "Coverage Percentage: 0.0032625771810166%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2083\n",
      "G007828095 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium sacchari\n",
      "Coverage Percentage: 0.007021686578638%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2084\n",
      "G007828955 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Kitasatospora; s__Kitasatospora atroaurantiaca\n",
      "Coverage Percentage: 0.0033665046255773%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2085\n",
      "G007829565 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Prauserella; s__Prauserella rugosa\n",
      "Coverage Percentage: 0.0091855747461466%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2086\n",
      "G007829915 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes teichomyceticus\n",
      "Coverage Percentage: 0.0303515819748165%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2087\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G007829925 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora palomenae\n",
      "Coverage Percentage: 0.0052439188571923%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2088\n",
      "G007830205 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium daqingense\n",
      "Coverage Percentage: 0.0094407633915327%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2089\n",
      "G007830315 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium taihuense\n",
      "Coverage Percentage: 0.0378807072464208%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2090\n",
      "G007830355 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium tiangeerense\n",
      "Coverage Percentage: 0.0455528171953195%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2091\n",
      "G007830415 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium cheniae\n",
      "Coverage Percentage: 0.1379264179111614%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2092\n",
      "G007830455 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia lurida\n",
      "Coverage Percentage: 0.0153961714853573%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2093\n",
      "G007830535 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Extensimonas; s__Extensimonas vulgaris\n",
      "Coverage Percentage: 0.0402818958054539%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2094\n",
      "G007830615 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Mucilaginibacter; s__Mucilaginibacter frigoritolerans\n",
      "Coverage Percentage: 0.0080794137421811%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2095\n",
      "G007844305 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Sterolibacteriaceae; g__Denitratisoma; s__Denitratisoma oestradiolicum\n",
      "Coverage Percentage: 0.0118427828368891%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2096\n",
      "G007845635 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Apibacter; s__Apibacter muscae\n",
      "Coverage Percentage: 0.0060228349204766%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2097\n",
      "G007858165 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas fluorescens\n",
      "Coverage Percentage: 0.0402512013791448%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2098\n",
      "G007858175 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas panacis\n",
      "Coverage Percentage: 0.0728430518166219%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2099\n",
      "G007858185 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas grimontii\n",
      "Coverage Percentage: 0.0594917115647541%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2100\n",
      "G007858235 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas extremaustralis\n",
      "Coverage Percentage: 0.0350272987289467%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2101\n",
      "G007858255 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas rhodesiae\n",
      "Coverage Percentage: 0.0609495244484786%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2102\n",
      "G007858265 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas mandelii\n",
      "Coverage Percentage: 0.0242324195155375%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2103\n",
      "G007858285 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas brenneri\n",
      "Coverage Percentage: 0.0303257412105448%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2104\n",
      "G007858365 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas saxonica\n",
      "Coverage Percentage: 0.0173226098846577%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2105\n",
      "G007858445 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Tsukamurellaceae; g__Tsukamurella; s__Tsukamurella sputi\n",
      "Coverage Percentage: 0.0110270755198923%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2106\n",
      "G007858475 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Tsukamurellaceae; g__Tsukamurella; s__Tsukamurella conjunctivitidis\n",
      "Coverage Percentage: 0.0067869551181608%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2107\n",
      "G007859215 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium canis\n",
      "Coverage Percentage: 0.0045144262871673%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2108\n",
      "G007859635 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas panacisoli\n",
      "Coverage Percentage: 0.0236554242181588%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2109\n",
      "G007859655 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Devosiaceae; g__Devosia; s__Devosia ginsengisoli\n",
      "Coverage Percentage: 0.0101108984772049%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2110\n",
      "G007923255 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Vulcaniibacterium; s__Vulcaniibacterium thermophilum\n",
      "Coverage Percentage: 0.0205385320084247%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2111\n",
      "G007923295 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Agrococcus; s__Agrococcus sediminis\n",
      "Coverage Percentage: 0.0110364102869576%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2112\n",
      "G007954425 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium ginsenosidimutans\n",
      "Coverage Percentage: 0.0242074698801361%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2113\n",
      "G007954525 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Leuconostoc; s__Leuconostoc carnosum\n",
      "Coverage Percentage: 0.0177503698483518%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2114\n",
      "G007971525 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Mucilaginibacter; s__Mucilaginibacter ginsenosidivorax\n",
      "Coverage Percentage: 0.0115984137568964%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2115\n",
      "G007988805 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Frigoribacterium; s__Frigoribacterium faeni\n",
      "Coverage Percentage: 0.015001686580148%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2116\n",
      "G007989025 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lapidilactobacillus; s__Lapidilactobacillus concavus\n",
      "Coverage Percentage: 0.021607257458535%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2117\n",
      "G007989145 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactiplantibacillus; s__Lactiplantibacillus plantarum\n",
      "Coverage Percentage: 0.0125900288850339%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2118\n",
      "G007989165 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Latilactobacillus; s__Latilactobacillus sakei\n",
      "Coverage Percentage: 0.022150277560824%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2119\n",
      "G007989805 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Cellulomonadaceae; g__Cellulomonas; s__Cellulomonas xylanilytica\n",
      "Coverage Percentage: 0.0059405232115818%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2120\n",
      "G007990265 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Cellulomonadaceae; g__Cellulomonas; s__Cellulomonas terrae\n",
      "Coverage Percentage: 0.0092568747955483%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2121\n",
      "G007990815 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium branchiophilum\n",
      "Coverage Percentage: 0.0293309504259539%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2122\n",
      "G007990875 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Sphingobacterium; s__Sphingobacterium faecium\n",
      "Coverage Percentage: 0.0157277669277425%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2123\n",
      "G007990915 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas aquatilis\n",
      "Coverage Percentage: 0.0804379972132437%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2124\n",
      "G007991055 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium radiotolerans\n",
      "Coverage Percentage: 0.0260355726280621%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2125\n",
      "G007991415 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium rhizosphaerae\n",
      "Coverage Percentage: 0.0110784932402919%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2126\n",
      "G007991435 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium lathyri\n",
      "Coverage Percentage: 0.0048882010425952%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2127\n",
      "G007991615 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium sediminis\n",
      "Coverage Percentage: 0.0546165984668917%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2128\n",
      "G007991775 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Azospirillaceae; g__Skermanella; s__Skermanella aerolata\n",
      "Coverage Percentage: 0.0033244019295577%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2129\n",
      "G007992175 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium haplocladii\n",
      "Coverage Percentage: 0.0094203042262459%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2130\n",
      "G007992215 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium gnaphalii\n",
      "Coverage Percentage: 0.013408226961748%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2131\n",
      "G007992375 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Sulfuricellaceae; g__Sulfuriferula; s__Sulfuriferula plumbiphila\n",
      "Coverage Percentage: 0.0086821602868538%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2132\n",
      "G007992455 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium saccharophilum\n",
      "Coverage Percentage: 0.02021451482487%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2133\n",
      "G007992495 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Reyranellaceae; g__Reyranella; s__Reyranella soli\n",
      "Coverage Percentage: 0.0158303441719776%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2134\n",
      "G007992535 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Cellulomonadaceae; g__Cellulomonas; s__Cellulomonas soli\n",
      "Coverage Percentage: 0.0267031215476467%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2135\n",
      "G007992695 d__Bacteria; p__Bacteroidetes; c__Chitinophagia; o__Chitinophagales; f__Chitinophagaceae; g__Chitinophaga; s__Chitinophaga terrae (ex Kim and Jung 2007)\n",
      "Coverage Percentage: 0.0034522873447776%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2136\n",
      "G007993735 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Arenimonas; s__Arenimonas daejeonensis\n",
      "Coverage Percentage: 0.0398675434668341%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 258\n",
      "Length of fail 2137\n",
      "G007993815 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax delafieldii\n",
      "Coverage Percentage: 0.1647374623621488%\n",
      "Adj, read counts 56\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 5]\n",
      "Number of bins 46\n",
      "total # samples 20\n",
      "Number of total reads 55\n",
      "Number of hits in BIOM tables 66.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 21.428571428571423 %\n",
      "Percentage of reads from samples who only fall into one bin group 12.727272727272727 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 259\n",
      "Length of fail 2137\n",
      "G007995055 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingosinicellaceae; g__Sphingosinicella; s__Sphingosinicella ginsenosidimutans\n",
      "Coverage Percentage: 0.0132478676028493%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 259\n",
      "Length of fail 2138\n",
      "G007995065 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas ginsenosidivorax\n",
      "Coverage Percentage: 0.1687290720819881%\n",
      "Adj, read counts 46\n",
      "Number of samples per group [6, 1, 1, 6, 1, 1, 1, 5, 3, 2, 1, 2, 1, 1, 1, 1, 1, 1, 6, 2]\n",
      "Number of bins 25\n",
      "total # samples 26\n",
      "Number of total reads 45\n",
      "Number of hits in BIOM tables 34.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 43.47826086956522 %\n",
      "Percentage of reads from samples who only fall into one bin group 37.77777777777778 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 260\n",
      "Length of fail 2138\n",
      "G007996985 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Extensimonas; s__Extensimonas vulgaris\n",
      "Coverage Percentage: 0.0403097259810718%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 260\n",
      "Length of fail 2139\n",
      "G008000935 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium alkalisoli\n",
      "Coverage Percentage: 0.0276728581446138%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 260\n",
      "Length of fail 2140\n",
      "G008000975 d__Bacteria; p__Verrucomicrobia; c__Verrucomicrobiae; o__Verrucomicrobiales; f__Akkermansiaceae; g__Akkermansia; s__Akkermansia muciniphila\n",
      "Coverage Percentage: 0.090276320614945%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 260\n",
      "Length of fail 2141\n",
      "G008001015 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Ornithinimicrobiaceae; g__Serinicoccus; s__Serinicoccus profundi\n",
      "Coverage Percentage: 0.006674893280828%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 260\n",
      "Length of fail 2142\n",
      "G008017415 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium hatanonis\n",
      "Coverage Percentage: 0.0078700044512068%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 260\n",
      "Length of fail 2143\n",
      "G008017435 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium saccharophilum\n",
      "Coverage Percentage: 0.0252395416365519%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 260\n",
      "Length of fail 2144\n",
      "G008017875 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Chromatiales; f__Chromatiaceae; g__Rheinheimera; s__Rheinheimera tangshanensis\n",
      "Coverage Percentage: 0.0329446627207853%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 260\n",
      "Length of fail 2145\n",
      "G008033445 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Vulcaniibacterium; s__Vulcaniibacterium gelatinicum\n",
      "Coverage Percentage: 0.1194619449004752%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 260\n",
      "Length of fail 2146\n",
      "G008033455 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Vulcaniibacterium; s__Vulcaniibacterium tengchongense\n",
      "Coverage Percentage: 0.0842926587539642%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 260\n",
      "Length of fail 2147\n",
      "G008040105 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Lacisediminihabitans; s__Lacisediminihabitans profunda\n",
      "Coverage Percentage: 0.0112687127796655%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 260\n",
      "Length of fail 2148\n",
      "G008087345 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Marmoricola; s__Marmoricola caldifontis\n",
      "Coverage Percentage: 0.005304902936919%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 260\n",
      "Length of fail 2149\n",
      "G008107625 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Zoogloeaceae; g__Zoogloea; s__Zoogloea oleivorans\n",
      "Coverage Percentage: 1.526756529655156%\n",
      "Adj, read counts 532\n",
      "Number of samples per group [8, 3, 1, 10, 6, 4, 3, 4, 2, 3, 5, 4, 1, 5, 2, 2, 4, 4, 2, 4, 2, 6, 1, 1, 9, 3, 2, 2, 1, 5, 3, 1, 1, 2, 3, 9, 3, 1, 1, 2, 2, 1, 1, 1, 2, 5, 1, 3, 4, 5, 7, 3, 2, 1, 1, 6, 1, 11, 10, 15, 3, 4, 4, 1, 2, 1, 6, 1, 11, 5, 3, 3, 12, 1, 2, 2, 2, 2, 5, 2, 5, 2, 2, 4, 2, 7, 1, 6, 2, 7, 2, 4, 2, 2, 3, 10, 1, 3, 8, 3, 5, 2, 2, 2, 9, 1, 6, 7, 3, 3, 1, 1, 8, 4, 1, 4, 5, 5, 8, 3, 2, 2, 3, 1]\n",
      "Number of bins 315\n",
      "total # samples 49\n",
      "Number of total reads 523\n",
      "Number of hits in BIOM tables 971.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 9.022556390977442 %\n",
      "Percentage of reads from samples who only fall into one bin group 0.7648183556405354 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 261\n",
      "Length of fail 2149\n",
      "G008123425 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium rifense\n",
      "Coverage Percentage: 0.0175175818665706%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 261\n",
      "Length of fail 2150\n",
      "G008123965 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium hipponense\n",
      "Coverage Percentage: 0.0118303324289449%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 261\n",
      "Length of fail 2151\n",
      "G008124555 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Elizabethkingia; s__Elizabethkingia miricola\n",
      "Coverage Percentage: 0.0203447859269948%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 261\n",
      "Length of fail 2152\n",
      "G008254045 d__Bacteria; p__Planctomycetes; c__Planctomycetia; o__Gemmatales; f__Gemmataceae; g__Limnoglobus; s__Limnoglobus roseus\n",
      "Coverage Percentage: 0.005982022582615%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 261\n",
      "Length of fail 2153\n",
      "G008271655 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Gemmobacter; s__Gemmobacter caeruleus\n",
      "Coverage Percentage: 0.0120438425917347%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 261\n",
      "Length of fail 2154\n",
      "G008274695 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas montanisoli\n",
      "Coverage Percentage: 0.0275736142618992%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 261\n",
      "Length of fail 2155\n",
      "G008274785 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Oceanospirillales; f__Halomonadaceae; g__Halomonas; s__Halomonas eurihalina\n",
      "Coverage Percentage: 0.00392704537948%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 261\n",
      "Length of fail 2156\n",
      "G008274825 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Sphingobacterium; s__Sphingobacterium hotanense\n",
      "Coverage Percentage: 0.0100439752405988%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 261\n",
      "Length of fail 2157\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G008326345 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Cutibacterium; s__Cutibacterium modestum\n",
      "Coverage Percentage: 1.6168655083160153%\n",
      "Adj, read counts 281\n",
      "Number of samples per group [2, 1, 2, 3, 1, 2, 1, 2, 2, 6, 1, 4, 25, 4, 4, 2, 2, 1, 1, 1, 1, 2, 1, 4, 4, 3, 1, 1, 18, 1, 5, 1, 2, 2, 1, 1, 14, 3, 1, 5, 15, 6, 1, 2, 3, 1, 1, 1, 1, 2, 1, 28]\n",
      "Number of bins 88\n",
      "total # samples 50\n",
      "Number of total reads 278\n",
      "Number of hits in BIOM tables 91.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 45.907473309608534 %\n",
      "Percentage of reads from samples who only fall into one bin group 5.0359712230215825 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 262\n",
      "Length of fail 2157\n",
      "G008329925 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Sphaerotilus; s__Sphaerotilus natans\n",
      "Coverage Percentage: 0.359327762346278%\n",
      "Adj, read counts 108\n",
      "Number of samples per group [4, 1, 1, 2, 1, 3, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 2, 3, 1, 1, 1, 4, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 7, 1, 1, 4, 3, 8]\n",
      "Number of bins 67\n",
      "total # samples 33\n",
      "Number of total reads 106\n",
      "Number of hits in BIOM tables 38.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 29.629629629629626 %\n",
      "Percentage of reads from samples who only fall into one bin group 15.09433962264151 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 263\n",
      "Length of fail 2157\n",
      "G008360935 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Siccirubricoccus; s__Siccirubricoccus phaeus\n",
      "Coverage Percentage: 0.0023044755509642%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 263\n",
      "Length of fail 2158\n",
      "G008364325 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia liquefaciens\n",
      "Coverage Percentage: 0.0071962791685288%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 263\n",
      "Length of fail 2159\n",
      "G008364685 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Xanthobacteraceae; g__Xanthobacter; s__Xanthobacter oligotrophicus\n",
      "Coverage Percentage: 0.0056641700410191%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 263\n",
      "Length of fail 2160\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G008369745 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium nitrogenifigens\n",
      "Coverage Percentage: 0.0317617049886978%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 263\n",
      "Length of fail 2161\n",
      "G008386465 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Arenimonas; s__Arenimonas fontis\n",
      "Coverage Percentage: 0.046645233701752%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 263\n",
      "Length of fail 2162\n",
      "G008386485 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium viscerum\n",
      "Coverage Percentage: 0.0123987628401898%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 263\n",
      "Length of fail 2163\n",
      "G008571395 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides finegoldii\n",
      "Coverage Percentage: 0.07240890761507%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 263\n",
      "Length of fail 2164\n",
      "G008572545 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides caccae\n",
      "Coverage Percentage: 0.0957489136180955%\n",
      "Adj, read counts 31\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 4, 1, 1, 1, 2, 2, 1, 1, 1]\n",
      "Number of bins 22\n",
      "total # samples 21\n",
      "Number of total reads 31\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 32.25806451612903 %\n",
      "Percentage of reads from samples who only fall into one bin group 54.83870967741935 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 264\n",
      "Length of fail 2164\n",
      "G008629655 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium baculatum\n",
      "Coverage Percentage: 0.0443240271818143%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 264\n",
      "Length of fail 2165\n",
      "G008629695 d__Bacteria; p__Bacteroidetes; c__Chitinophagia; o__Chitinophagales; f__Chitinophagaceae; g__Taibaiella; s__Taibaiella lutea\n",
      "Coverage Percentage: 0.0107150694584491%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 264\n",
      "Length of fail 2166\n",
      "G008630065 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Blastochloridaceae; g__Blastochloris; s__Blastochloris sulfoviridis\n",
      "Coverage Percentage: 0.0039249997530629%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 264\n",
      "Length of fail 2167\n",
      "G008632125 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Cupriavidus; s__Cupriavidus cauae\n",
      "Coverage Percentage: 0.0460409879547525%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 264\n",
      "Length of fail 2168\n",
      "G008632635 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter baumannii\n",
      "Coverage Percentage: 0.0186420450976697%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 264\n",
      "Length of fail 2169\n",
      "G008692035 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas panacis\n",
      "Coverage Percentage: 0.0669900027755264%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 264\n",
      "Length of fail 2170\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G008692995 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__; g__Gemella; s__Gemella haemolysans\n",
      "Coverage Percentage: 0.0097076068807517%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 264\n",
      "Length of fail 2171\n",
      "G008693005 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Helicobacteraceae; g__Helicobacter; s__Helicobacter canis\n",
      "Coverage Percentage: 0.0094791001381462%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 264\n",
      "Length of fail 2172\n",
      "G008693045 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Ewingella; s__Ewingella americana\n",
      "Coverage Percentage: 0.0083283427758549%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 264\n",
      "Length of fail 2173\n",
      "G008693065 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium tuscaniense\n",
      "Coverage Percentage: 0.06794622503825%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 264\n",
      "Length of fail 2174\n",
      "G008693165 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces johnsonii\n",
      "Coverage Percentage: 0.0417532555329019%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 264\n",
      "Length of fail 2175\n",
      "G008693185 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter qingfengensis\n",
      "Coverage Percentage: 0.0084665305679805%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 264\n",
      "Length of fail 2176\n",
      "G008693605 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Citrobacter; s__Citrobacter portucalensis\n",
      "Coverage Percentage: 0.0255205543924566%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 264\n",
      "Length of fail 2177\n",
      "G008693705 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas veronii\n",
      "Coverage Percentage: 0.0452444052793907%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 264\n",
      "Length of fail 2178\n",
      "G008693785 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Roseomonas; s__Roseomonas mucosa\n",
      "Coverage Percentage: 0.003022191693416%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 264\n",
      "Length of fail 2179\n",
      "G008694755 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactobacillus; s__Lactobacillus crispatus\n",
      "Coverage Percentage: 0.098146065067098%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 264\n",
      "Length of fail 2180\n",
      "G008710035 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas cyclobalanopsidis\n",
      "Coverage Percentage: 0.1547781840481319%\n",
      "Adj, read counts 42\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1]\n",
      "Number of bins 32\n",
      "total # samples 12\n",
      "Number of total reads 42\n",
      "Number of hits in BIOM tables 7.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 23.809523809523807 %\n",
      "Percentage of reads from samples who only fall into one bin group 21.428571428571427 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 265\n",
      "Length of fail 2180\n",
      "G008710145 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Niallia; s__Niallia endozanthoxylicus\n",
      "Coverage Percentage: 0.0042643537276195%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 265\n",
      "Length of fail 2181\n",
      "G008710155 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium limneticum\n",
      "Coverage Percentage: 0.0444183521792373%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 265\n",
      "Length of fail 2182\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G008710165 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium limneticum\n",
      "Coverage Percentage: 0.0410318292532692%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 265\n",
      "Length of fail 2183\n",
      "G008710205 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium limneticum\n",
      "Coverage Percentage: 0.0408823717414557%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 265\n",
      "Length of fail 2184\n",
      "G008710745 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium rhizomatis\n",
      "Coverage Percentage: 0.0107819046793942%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 265\n",
      "Length of fail 2185\n",
      "G008727715 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Phyllobacteriaceae; g__Mesorhizobium; s__Mesorhizobium terrae\n",
      "Coverage Percentage: 0.1666608573548377%\n",
      "Adj, read counts 57\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 46\n",
      "total # samples 18\n",
      "Number of total reads 57\n",
      "Number of hits in BIOM tables 65.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.035087719298245 %\n",
      "Percentage of reads from samples who only fall into one bin group 17.543859649122805 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 266\n",
      "Length of fail 2185\n",
      "G008727755 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium caowuchunii\n",
      "Coverage Percentage: 0.0147116561371052%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 266\n",
      "Length of fail 2186\n",
      "Length of pass 267\n",
      "Length of fail 2186\n",
      "G008801385 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Agrobacterium; s__Agrobacterium tumefaciens\n",
      "Coverage Percentage: 0.0027450867038362%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 267\n",
      "Length of fail 2187\n",
      "G008801395 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas tolaasii\n",
      "Coverage Percentage: 0.0370606020895063%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 267\n",
      "Length of fail 2188\n",
      "G008801405 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Ralstonia; s__Ralstonia insidiosa\n",
      "Coverage Percentage: 0.0266591615711155%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 267\n",
      "Length of fail 2189\n",
      "G008801485 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas psychrophila\n",
      "Coverage Percentage: 0.075351472589028%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 267\n",
      "Length of fail 2190\n",
      "G008801495 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas palleroniana\n",
      "Coverage Percentage: 0.0606035886090822%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 267\n",
      "Length of fail 2191\n",
      "G008801525 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas lini\n",
      "Coverage Percentage: 0.0161627159514165%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 267\n",
      "Length of fail 2192\n",
      "G008801675 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas aeruginosa\n",
      "Coverage Percentage: 0.5424773542357668%\n",
      "Adj, read counts 203\n",
      "Number of samples per group [1, 1, 1, 3, 1, 2, 1, 1, 1, 1, 1, 1, 2, 3, 1, 1, 1, 1, 2, 1, 1, 1, 3, 1, 1, 1, 2, 1, 1, 3, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 3, 2, 1, 1, 2, 2, 2, 3, 1, 1, 1, 1, 1, 2, 3, 1, 1, 2, 1, 1, 1, 1, 1, 3, 3, 2, 2, 1, 1, 2, 1, 2, 1, 4, 1, 1, 2, 1, 2, 1, 1, 2, 1, 1, 1, 1, 2, 3, 1, 1, 1, 5, 1, 2, 2]\n",
      "Number of bins 166\n",
      "total # samples 51\n",
      "Number of total reads 200\n",
      "Number of hits in BIOM tables 58.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 6.896551724137931 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.5 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 268\n",
      "Length of fail 2192\n",
      "G008801725 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Kerstersia; s__Kerstersia gyiorum\n",
      "Coverage Percentage: 0.0066961951316404%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 268\n",
      "Length of fail 2193\n",
      "G008801825 d__Bacteria; p__Fusobacteria; c__Fusobacteriia; o__Fusobacteriales; f__Fusobacteriaceae; g__Fusobacterium; s__Fusobacterium naviforme\n",
      "Coverage Percentage: 0.0350588517399195%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 268\n",
      "Length of fail 2194\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G008801835 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Cupriavidus; s__Cupriavidus pauculus\n",
      "Coverage Percentage: 0.0320981867952422%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 268\n",
      "Length of fail 2195\n",
      "G008801845 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Ideonella; s__Ideonella dechloratans\n",
      "Coverage Percentage: 0.0608765694325187%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 268\n",
      "Length of fail 2196\n",
      "G008801915 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Cupriavidus; s__Cupriavidus gilardii\n",
      "Coverage Percentage: 0.0214518612811852%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 268\n",
      "Length of fail 2197\n",
      "G008801925 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Cupriavidus; s__Cupriavidus basilensis\n",
      "Coverage Percentage: 0.0494031478012915%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 268\n",
      "Length of fail 2198\n",
      "G008802125 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia stagnalis\n",
      "Coverage Percentage: 0.0103149033017771%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 268\n",
      "Length of fail 2199\n",
      "G008802155 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Aliarcobacter; s__Aliarcobacter skirrowii\n",
      "Coverage Percentage: 0.1944463780459342%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 268\n",
      "Length of fail 2200\n",
      "G008802205 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter gandensis\n",
      "Coverage Percentage: 0.0924017030557262%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 268\n",
      "Length of fail 2201\n",
      "G008802215 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter guillouiae\n",
      "Coverage Percentage: 0.1669069689858367%\n",
      "Adj, read counts 49\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 6, 6, 3]\n",
      "Number of bins 31\n",
      "total # samples 23\n",
      "Number of total reads 48\n",
      "Number of hits in BIOM tables 20.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 38.775510204081634 %\n",
      "Percentage of reads from samples who only fall into one bin group 22.916666666666664 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 269\n",
      "Length of fail 2201\n",
      "G008802275 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter bohemicus\n",
      "Coverage Percentage: 1.2071489105533704%\n",
      "Adj, read counts 289\n",
      "Number of samples per group [2, 4, 1, 1, 1, 2, 1, 2, 1, 3, 3, 1, 2, 3, 4, 9, 2, 2, 4, 3, 3, 3, 3, 3, 2, 1, 7, 3, 2, 1, 2, 5, 3, 3, 2, 3, 3, 2, 1, 3, 1, 3, 2, 2, 4, 1, 2, 1, 1, 1, 5, 1, 8, 3, 1, 1, 2, 4, 5, 1, 1, 1, 4, 2, 1, 1, 1, 10, 2, 2, 2, 1, 1, 3, 3, 9, 2, 4, 2, 1, 2, 2, 7, 5, 3, 18]\n",
      "Number of bins 178\n",
      "total # samples 47\n",
      "Number of total reads 284\n",
      "Number of hits in BIOM tables 221.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.99307958477509 %\n",
      "Percentage of reads from samples who only fall into one bin group 3.5211267605633805 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 270\n",
      "Length of fail 2201\n",
      "G008802955 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Citrobacter; s__Citrobacter cronae\n",
      "Coverage Percentage: 0.0573234441661422%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 270\n",
      "Length of fail 2202\n",
      "G008806385 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium soli\n",
      "Coverage Percentage: 0.0036238721449789%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 270\n",
      "Length of fail 2203\n",
      "G008806775 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium luteum\n",
      "Coverage Percentage: 0.0678841097004869%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 270\n",
      "Length of fail 2204\n",
      "G008830365 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia proteamaculans\n",
      "Coverage Percentage: 0.0083386443729822%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 270\n",
      "Length of fail 2205\n",
      "G008830385 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Odoribacteraceae; g__Butyricimonas; s__Butyricimonas faecihominis\n",
      "Coverage Percentage: 0.031236616223312%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 270\n",
      "Length of fail 2206\n",
      "G008831035 d__Bacteria; p__Actinobacteria; c__Coriobacteriia; o__Eggerthellales; f__Eggerthellaceae; g__Ellagibacter; s__Ellagibacter isourolithinifaciens\n",
      "Coverage Percentage: 0.0160326855722471%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 270\n",
      "Length of fail 2207\n",
      "G008831075 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Pseudoclavibacter; s__Pseudoclavibacter terrae\n",
      "Coverage Percentage: 0.0318158981359716%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 270\n",
      "Length of fail 2208\n",
      "G008831135 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Pseudoclavibacter; s__Pseudoclavibacter chungangensis\n",
      "Coverage Percentage: 0.0120735059288996%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 270\n",
      "Length of fail 2209\n",
      "G008867985 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter tandoii\n",
      "Coverage Percentage: 0.0376554054810156%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 270\n",
      "Length of fail 2210\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G008868005 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium algeriense\n",
      "Coverage Percentage: 0.0199810647760799%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 270\n",
      "Length of fail 2211\n",
      "G008868025 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium oxydans\n",
      "Coverage Percentage: 0.0453410062160311%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 270\n",
      "Length of fail 2212\n",
      "G008868275 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Micrococcus; s__Micrococcus luteus\n",
      "Coverage Percentage: 0.7362337384663068%\n",
      "Adj, read counts 100\n",
      "Number of samples per group [3, 1, 3, 1, 1, 1, 1, 2, 1, 1, 1, 3, 1, 3, 1, 1, 1, 1, 1, 1, 3, 2, 1, 1, 2, 2, 1, 2, 1, 1, 2, 1, 1, 2, 2, 1, 1, 2, 2, 2, 2, 3, 1, 1, 2, 2, 1, 2]\n",
      "Number of bins 74\n",
      "total # samples 23\n",
      "Number of total reads 97\n",
      "Number of hits in BIOM tables 41.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.556701030927837 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 271\n",
      "Length of fail 2212\n",
      "G008868685 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces triticiradicis\n",
      "Coverage Percentage: 0.0076381869443231%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 271\n",
      "Length of fail 2213\n",
      "G008876665 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Limosilactobacillus; s__Limosilactobacillus frumenti\n",
      "Coverage Percentage: 0.0172283174209094%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 271\n",
      "Length of fail 2214\n",
      "G008930655 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Roseobacteraceae; g__Salipiger; s__Salipiger aestuarii\n",
      "Coverage Percentage: 0.0032631264122206%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 271\n",
      "Length of fail 2215\n",
      "G008932115 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium betae\n",
      "Coverage Percentage: 0.0144889992778413%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 271\n",
      "Length of fail 2216\n",
      "G008932435 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Brucellaceae; g__Brucella; s__Brucella pseudintermedia\n",
      "Coverage Percentage: 0.0032988163391965%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 271\n",
      "Length of fail 2217\n",
      "G008973825 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Gemmobacter; s__Gemmobacter serpentinus\n",
      "Coverage Percentage: 0.0150797797742395%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 271\n",
      "Length of fail 2218\n",
      "G009035845 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter baumannii\n",
      "Coverage Percentage: 0.0467596324842917%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 271\n",
      "Length of fail 2219\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G009036245 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Enterobacter; s__Enterobacter sichuanensis\n",
      "Coverage Percentage: 0.0064099992592418%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 271\n",
      "Length of fail 2220\n",
      "G009173485 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Klebsiella; s__Klebsiella michiganensis\n",
      "Coverage Percentage: 0.0164729726396096%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 271\n",
      "Length of fail 2221\n",
      "G009177095 d__Bacteria; p__Planctomycetes; c__Planctomycetia; o__Pirellulales; f__Lacipirellulaceae; g__Lacipirellula; s__Lacipirellula parvula\n",
      "Coverage Percentage: 0.0034237383235354%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 271\n",
      "Length of fail 2222\n",
      "G009184865 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces phaeolivaceus\n",
      "Coverage Percentage: 0.0029089593983957%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 271\n",
      "Length of fail 2223\n",
      "G009192735 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Arthrobacter; s__Arthrobacter gandavensis\n",
      "Coverage Percentage: 0.0040566305626546%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 271\n",
      "Length of fail 2224\n",
      "G009192745 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Arthrobacter; s__Arthrobacter citreus\n",
      "Coverage Percentage: 0.0104953082033073%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 271\n",
      "Length of fail 2225\n",
      "G009193255 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Arthrobacter; s__Arthrobacter koreensis\n",
      "Coverage Percentage: 0.0107857275818088%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 271\n",
      "Length of fail 2226\n",
      "G009193295 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides luhongzhouii\n",
      "Coverage Percentage: 0.0591657627452913%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 271\n",
      "Length of fail 2227\n",
      "G009193325 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides zhangwenhongi\n",
      "Coverage Percentage: 0.0522870215418018%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 271\n",
      "Length of fail 2228\n",
      "G009193445 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Eurotiales; f__Aspergillaceae; g__Aspergillus; s__Aspergillus pseudotamarii\n",
      "Coverage Percentage: 0.0006379270258019%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 271\n",
      "Length of fail 2229\n",
      "G009208555 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Janthinobacterium; s__Janthinobacterium violaceinigrum\n",
      "Coverage Percentage: 0.0905719509495648%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2]\n",
      "Number of bins 34\n",
      "total # samples 25\n",
      "Number of total reads 37\n",
      "Number of hits in BIOM tables 18.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.216216216216218 %\n",
      "Percentage of reads from samples who only fall into one bin group 40.54054054054054 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 272\n",
      "Length of fail 2229\n",
      "G009295725 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Clostridium; s__Clostridium tarantellae\n",
      "Coverage Percentage: 0.0037805517402305%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 272\n",
      "Length of fail 2230\n",
      "G009363855 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Microvirga; s__Microvirga thermotolerans\n",
      "Coverage Percentage: 0.007454782818635%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 272\n",
      "Length of fail 2231\n",
      "G009371875 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium tibiigranuli\n",
      "Coverage Percentage: 0.0162805511295088%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 272\n",
      "Length of fail 2232\n",
      "G009371885 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium tibiigranuli\n",
      "Coverage Percentage: 0.0243660647171394%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 272\n",
      "Length of fail 2233\n",
      "G009372215 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter portensis\n",
      "Coverage Percentage: 0.1812243315822453%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [3, 2, 1, 2, 2, 1, 2, 1, 1, 1, 1, 3, 1, 3, 3, 2]\n",
      "Number of bins 23\n",
      "total # samples 18\n",
      "Number of total reads 35\n",
      "Number of hits in BIOM tables 14.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 29.72972972972973 %\n",
      "Percentage of reads from samples who only fall into one bin group 40.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 273\n",
      "Length of fail 2233\n",
      "G009380135 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Fertoeibacter; s__Fertoeibacter niger\n",
      "Coverage Percentage: 0.0067249169264772%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 273\n",
      "Length of fail 2234\n",
      "G009380165 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Rugamonas; s__Rugamonas rivuli\n",
      "Coverage Percentage: 0.1212668147825285%\n",
      "Adj, read counts 53\n",
      "Number of samples per group [1, 2, 2, 1, 1, 1, 1, 5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 3, 1, 2, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1]\n",
      "Number of bins 43\n",
      "total # samples 25\n",
      "Number of total reads 53\n",
      "Number of hits in BIOM tables 36.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 22.641509433962266 %\n",
      "Percentage of reads from samples who only fall into one bin group 24.528301886792452 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 274\n",
      "Length of fail 2234\n",
      "G009380215 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Rugamonas; s__Rugamonas aquatica\n",
      "Coverage Percentage: 0.1126940080270786%\n",
      "Adj, read counts 57\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 1, 1, 2, 2, 1, 2, 1, 1, 1, 1, 1, 2, 2, 1, 1, 2, 6, 2, 2, 1, 2, 1, 3, 2, 1, 2, 1]\n",
      "Number of bins 40\n",
      "total # samples 27\n",
      "Number of total reads 55\n",
      "Number of hits in BIOM tables 31.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 22.807017543859647 %\n",
      "Percentage of reads from samples who only fall into one bin group 30.909090909090907 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 275\n",
      "Length of fail 2234\n",
      "G009467805 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas nematodicola\n",
      "Coverage Percentage: 0.0842586071529879%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 275\n",
      "Length of fail 2235\n",
      "G009469755 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Rhodocyclaceae; g__Rhodocyclus; s__Rhodocyclus tenuis\n",
      "Coverage Percentage: 0.0179504432877533%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 275\n",
      "Length of fail 2236\n",
      "G009469765 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Azonexaceae; g__Dechloromonas; s__Dechloromonas hortensis\n",
      "Coverage Percentage: 0.0399068623078524%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 275\n",
      "Length of fail 2237\n",
      "G009495405 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella copri\n",
      "Coverage Percentage: 0.194924578568986%\n",
      "Adj, read counts 38\n",
      "Number of samples per group [1, 3, 1, 1, 2, 1, 2, 1, 1, 2, 2, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 3, 1, 1]\n",
      "Number of bins 34\n",
      "total # samples 21\n",
      "Number of total reads 38\n",
      "Number of hits in BIOM tables 25.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 21.052631578947366 %\n",
      "Percentage of reads from samples who only fall into one bin group 23.684210526315788 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 276\n",
      "Length of fail 2237\n",
      "G009557235 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter wanghuae\n",
      "Coverage Percentage: 0.18133764969208%\n",
      "Adj, read counts 33\n",
      "Number of samples per group [2, 1, 1, 3, 1, 1, 1, 4, 1, 2, 2, 2, 2, 5, 2, 2]\n",
      "Number of bins 19\n",
      "total # samples 17\n",
      "Number of total reads 32\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 39.39393939393939 %\n",
      "Percentage of reads from samples who only fall into one bin group 25.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 277\n",
      "Length of fail 2237\n",
      "G009602465 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Cumulibacter; s__Cumulibacter manganitolerans\n",
      "Coverage Percentage: 0.0065768950350584%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 277\n",
      "Length of fail 2238\n",
      "G009617585 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Endobacterium; s__Endobacterium cerealis\n",
      "Coverage Percentage: 0.0083556491301049%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 277\n",
      "Length of fail 2239\n",
      "G009646115 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Erwiniaceae; g__Pantoea; s__Pantoea eucalypti\n",
      "Coverage Percentage: 0.0138152342651149%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 277\n",
      "Length of fail 2240\n",
      "G009649175 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Rhodovulum; s__Rhodovulum strictum\n",
      "Coverage Percentage: 0.0039456318067048%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 277\n",
      "Length of fail 2241\n",
      "G009650215 d__Bacteria; p__Actinobacteria; c__Acidimicrobiia; o__Acidimicrobiales; f__Iamiaceae; g__Actinomarinicola; s__Actinomarinicola tropica\n",
      "Coverage Percentage: 0.0112772221106761%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 277\n",
      "Length of fail 2242\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G009659375 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Aerococcaceae; g__Fundicoccus; s__Fundicoccus ignavus\n",
      "Coverage Percentage: 0.0118005906278712%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 277\n",
      "Length of fail 2243\n",
      "G009659415 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Aerococcaceae; g__Fundicoccus; s__Fundicoccus ignavus\n",
      "Coverage Percentage: 0.0130065406412261%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 277\n",
      "Length of fail 2244\n",
      "G009674485 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia rivuli\n",
      "Coverage Percentage: 0.0120915861774567%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 277\n",
      "Length of fail 2245\n",
      "G009674495 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Duganella; s__Duganella alba\n",
      "Coverage Percentage: 0.1197908192911098%\n",
      "Adj, read counts 54\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 4, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 3, 1, 1, 1]\n",
      "Number of bins 49\n",
      "total # samples 29\n",
      "Number of total reads 54\n",
      "Number of hits in BIOM tables 18.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.666666666666664 %\n",
      "Percentage of reads from samples who only fall into one bin group 35.18518518518518 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 278\n",
      "Length of fail 2245\n",
      "G009674535 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Duganella; s__Duganella guangzhouensis\n",
      "Coverage Percentage: 0.0429550819568402%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 278\n",
      "Length of fail 2246\n",
      "G009674565 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Duganella; s__Duganella aquatilis\n",
      "Coverage Percentage: 0.1090177634106531%\n",
      "Adj, read counts 48\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 2, 1, 1, 1, 1, 2, 2]\n",
      "Number of bins 44\n",
      "total # samples 28\n",
      "Number of total reads 48\n",
      "Number of hits in BIOM tables 15.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.666666666666664 %\n",
      "Percentage of reads from samples who only fall into one bin group 37.5 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 279\n",
      "Length of fail 2246\n",
      "G009676365 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Latilactobacillus; s__Latilactobacillus sakei\n",
      "Coverage Percentage: 0.02091064366628%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 279\n",
      "Length of fail 2247\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G009695545 d__Bacteria; p__Lentisphaerae; c__Lentisphaeria; o__Victivallales; f__Victivallaceae; g__Victivallis; s__Victivallis lenta\n",
      "Coverage Percentage: 0.0037320092813448%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 279\n",
      "Length of fail 2248\n",
      "G009695765 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Roseburia; s__Roseburia porci\n",
      "Coverage Percentage: 0.010123343894685%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 279\n",
      "Length of fail 2249\n",
      "G009695995 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Waltera; s__Waltera intestinalis\n",
      "Coverage Percentage: 0.0285373049544193%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 279\n",
      "Length of fail 2250\n",
      "G009696065 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Oliverpabstia; s__Oliverpabstia intestinalis\n",
      "Coverage Percentage: 0.0787893212496252%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 279\n",
      "Length of fail 2251\n",
      "G009696075 d__Bacteria; p__Firmicutes; c__Erysipelotrichia; o__Erysipelotrichales; f__Erysipelotrichaceae; g__Holdemanella; s__Holdemanella porci\n",
      "Coverage Percentage: 0.0270167262481968%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 279\n",
      "Length of fail 2252\n",
      "G009696275 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Eisenbergiella; s__Eisenbergiella porci\n",
      "Coverage Percentage: 0.0410063028570016%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 279\n",
      "Length of fail 2253\n",
      "G009696315 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Cutibacterium; s__Cutibacterium porci\n",
      "Coverage Percentage: 0.3852324986245804%\n",
      "Adj, read counts 68\n",
      "Number of samples per group [1, 2, 9, 1, 2, 1, 1, 1, 2, 1, 1, 2, 9, 2, 2, 1, 1, 2, 2, 6, 1, 1, 2, 1, 7]\n",
      "Number of bins 34\n",
      "total # samples 33\n",
      "Number of total reads 68\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 47.05882352941176 %\n",
      "Percentage of reads from samples who only fall into one bin group 27.941176470588236 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2253\n",
      "G009696615 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Scrofimicrobium; s__Scrofimicrobium canadense\n",
      "Coverage Percentage: 0.0124870600912788%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2254\n",
      "G009697165 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Anaerobutyricum; s__Anaerobutyricum soehngenii\n",
      "Coverage Percentage: 0.0400238453245796%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2255\n",
      "G009707925 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Blautia; s__Blautia luti\n",
      "Coverage Percentage: 0.1069842714540759%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2256\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G009708035 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Hyphomicrobiaceae; g__Hyphomicrobium; s__Hyphomicrobium album\n",
      "Coverage Percentage: 0.0028145884726226%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2257\n",
      "G009717905 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Oscillospiraceae; g__Ruthenibacterium; s__Ruthenibacterium lactatiformans\n",
      "Coverage Percentage: 0.0320608666499616%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2258\n",
      "G009718305 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Mediterraneibacter; s__Mediterraneibacter faecis\n",
      "Coverage Percentage: 0.1192211454626103%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2259\n",
      "G009720745 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia eburnea\n",
      "Coverage Percentage: 0.0166598973770174%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2260\n",
      "G009720775 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Pseudoduganella; s__Pseudoduganella danionis\n",
      "Coverage Percentage: 0.0601426623618546%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2261\n",
      "G009720825 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Duganella; s__Duganella radicis\n",
      "Coverage Percentage: 0.0418283579690024%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2262\n",
      "G009727155 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia dura\n",
      "Coverage Percentage: 0.0558658655390315%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2263\n",
      "G009730055 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Yersinia; s__Yersinia intermedia\n",
      "Coverage Percentage: 0.0657143262912084%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2264\n",
      "G009731575 d__Bacteria; p__Verrucomicrobia; c__Verrucomicrobiae; o__Verrucomicrobiales; f__Akkermansiaceae; g__Akkermansia; s__Akkermansia muciniphila\n",
      "Coverage Percentage: 0.0835510698532106%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2265\n",
      "G009733795 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Cellulomonadaceae; g__Actinotalea; s__Actinotalea caeni\n",
      "Coverage Percentage: 0.0075456593594534%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2266\n",
      "G009734405 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium comes\n",
      "Coverage Percentage: 0.0081737602899329%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2267\n",
      "G009734425 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium occultum\n",
      "Coverage Percentage: 0.0046922417666694%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2268\n",
      "G009735685 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces lycii\n",
      "Coverage Percentage: 0.0025967044036974%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2269\n",
      "G009738105 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus mutans\n",
      "Coverage Percentage: 0.0074454950317345%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2270\n",
      "G009739655 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Clavibacter; s__Clavibacter michiganensis\n",
      "Coverage Percentage: 0.0816796109091157%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2271\n",
      "G009749395 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Agromyces; s__Agromyces salentinus\n",
      "Coverage Percentage: 0.0107095160788135%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2272\n",
      "G009749405 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Agromyces; s__Agromyces lapidis\n",
      "Coverage Percentage: 0.0238304910343695%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2273\n",
      "G009754945 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Cytophagaceae; g__Spirosoma; s__Spirosoma arboris\n",
      "Coverage Percentage: 0.0028562998611162%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2274\n",
      "G009758125 d__Bacteria; p__Bacteroidetes; c__Chitinophagia; o__Chitinophagales; f__Chitinophagaceae; g__Chitinophaga; s__Chitinophaga oryziterrae\n",
      "Coverage Percentage: 0.0015535271477636%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2275\n",
      "G009759665 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium cajani\n",
      "Coverage Percentage: 0.0097692162810251%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2276\n",
      "G009759685 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter baumannii\n",
      "Coverage Percentage: 0.0423766191326801%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2277\n",
      "G009759865 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium pachyrhizi\n",
      "Coverage Percentage: 0.0107242656556331%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2278\n",
      "G009760915 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Azohydromonas; s__Azohydromonas aeria\n",
      "Coverage Percentage: 0.0187055527362976%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2279\n",
      "G009763245 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas xionganensis\n",
      "Coverage Percentage: 0.0228853356275924%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 280\n",
      "Length of fail 2280\n",
      "G009768975 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingorhabdus; s__Sphingorhabdus lacus\n",
      "Coverage Percentage: 0.3069612284000286%\n",
      "Adj, read counts 61\n",
      "Number of samples per group [1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 2, 1, 3, 1, 3, 1, 2, 1, 1, 1, 1]\n",
      "Number of bins 53\n",
      "total # samples 33\n",
      "Number of total reads 60\n",
      "Number of hits in BIOM tables 131.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.754098360655737 %\n",
      "Percentage of reads from samples who only fall into one bin group 38.333333333333336 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 281\n",
      "Length of fail 2280\n",
      "G009769755 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingorhabdus; s__Sphingorhabdus profundilacus\n",
      "Coverage Percentage: 0.0675555518865507%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 281\n",
      "Length of fail 2281\n",
      "G009789595 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia flava\n",
      "Coverage Percentage: 0.0760470445740563%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 281\n",
      "Length of fail 2282\n",
      "G009792295 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Myroides; s__Myroides fluvii\n",
      "Coverage Percentage: 0.004008596980438%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 281\n",
      "Length of fail 2283\n",
      "G009792355 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Terricaulis; s__Terricaulis silvestris\n",
      "Coverage Percentage: 0.0066239508088853%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 281\n",
      "Length of fail 2284\n",
      "G009800965 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Zoogloeaceae; g__Azoarcus; s__Azoarcus halotolerans\n",
      "Coverage Percentage: 0.0052912916405036%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 281\n",
      "Length of fail 2285\n",
      "G009811375 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Variovorax; s__Variovorax boronicumulans\n",
      "Coverage Percentage: 0.2307699540564513%\n",
      "Adj, read counts 103\n",
      "Number of samples per group [2, 2, 2, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 3, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 10]\n",
      "Number of bins 83\n",
      "total # samples 37\n",
      "Number of total reads 103\n",
      "Number of hits in BIOM tables 67.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 22.330097087378643 %\n",
      "Percentage of reads from samples who only fall into one bin group 21.35922330097087 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 282\n",
      "Length of fail 2285\n",
      "G009826815 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pasteurellales; f__Pasteurellaceae; g__Pasteurella; s__Pasteurella canis\n",
      "Coverage Percentage: 0.0086896578290457%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 282\n",
      "Length of fail 2286\n",
      "G009826855 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Shinella; s__Shinella zoogloeoides\n",
      "Coverage Percentage: 0.0150502049930796%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 282\n",
      "Length of fail 2287\n",
      "G009827305 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Croceibacterium; s__Croceibacterium xixiisoli\n",
      "Coverage Percentage: 0.0065474042377687%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 282\n",
      "Length of fail 2288\n",
      "G009827495 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Tsuneonella; s__Tsuneonella aeria\n",
      "Coverage Percentage: 0.0446459409641051%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 282\n",
      "Length of fail 2289\n",
      "G009827655 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Alteraurantiacibacter; s__Alteraurantiacibacter buctensis\n",
      "Coverage Percentage: 0.0205122007842339%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 282\n",
      "Length of fail 2290\n",
      "G009828015 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Qipengyuania; s__Qipengyuania citrea\n",
      "Coverage Percentage: 0.0099611416523818%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 282\n",
      "Length of fail 2291\n",
      "G009828055 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Erythrobacter; s__Erythrobacter ramosus\n",
      "Coverage Percentage: 0.5139282771645379%\n",
      "Adj, read counts 101\n",
      "Number of samples per group [1, 3, 1, 1, 1, 1, 3, 1, 1, 2, 1, 2, 1, 2, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 3, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 2, 1, 1, 1, 3, 1, 1, 2, 2, 2, 1, 1, 1, 2, 2, 1, 4, 2]\n",
      "Number of bins 82\n",
      "total # samples 32\n",
      "Number of total reads 99\n",
      "Number of hits in BIOM tables 120.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 9.900990099009901 %\n",
      "Percentage of reads from samples who only fall into one bin group 6.0606060606060606 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 283\n",
      "Length of fail 2291\n",
      "G009829075 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Sphingobacterium; s__Sphingobacterium composti Ten et al. 2007 non Yoo et al. 2007\n",
      "Coverage Percentage: 0.0039836482305824%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 283\n",
      "Length of fail 2292\n",
      "G009830285 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Sporofaciens; s__Sporofaciens musculi\n",
      "Coverage Percentage: 0.018720392632042%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 283\n",
      "Length of fail 2293\n",
      "G009831375 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Mediterraneibacter; s__[Ruminococcus] gnavus\n",
      "Coverage Percentage: 0.1100532487544344%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 283\n",
      "Length of fail 2294\n",
      "G009834025 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Sporichthyales; f__Sporichthyaceae; g__Epidermidibacterium; s__Epidermidibacterium keratini\n",
      "Coverage Percentage: 0.013013809069754%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 283\n",
      "Length of fail 2295\n",
      "G009834205 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Rathayibacter; s__Rathayibacter festucae\n",
      "Coverage Percentage: 0.1011019744539126%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 283\n",
      "Length of fail 2296\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G009834985 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Deinococcales; f__Deinococcaceae; g__Deinococcus; s__Deinococcus xianganensis\n",
      "Coverage Percentage: 0.014390004378945%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 283\n",
      "Length of fail 2297\n",
      "G009857475 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia guangdongensis\n",
      "Coverage Percentage: 0.0183415949117271%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 283\n",
      "Length of fail 2298\n",
      "G009857505 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Duganella; s__Duganella lactea\n",
      "Coverage Percentage: 0.0744921301662257%\n",
      "Adj, read counts 31\n",
      "Number of samples per group [1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1]\n",
      "Number of bins 27\n",
      "total # samples 23\n",
      "Number of total reads 31\n",
      "Number of hits in BIOM tables 22.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 25.806451612903224 %\n",
      "Percentage of reads from samples who only fall into one bin group 48.38709677419355 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 284\n",
      "Length of fail 2298\n",
      "G009857535 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Duganella; s__Duganella pernnla\n",
      "Coverage Percentage: 0.2996181097044341%\n",
      "Adj, read counts 131\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 3, 1, 1, 1, 1, 3, 1, 2, 2, 1, 1, 3, 2, 1, 2, 3, 1, 1, 1, 3, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 2, 1, 1, 1, 2, 3, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 5, 1, 2, 1, 2, 1, 1, 3]\n",
      "Number of bins 113\n",
      "total # samples 42\n",
      "Number of total reads 129\n",
      "Number of hits in BIOM tables 120.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 8.396946564885496 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.627906976744185 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 285\n",
      "Length of fail 2298\n",
      "G009857595 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__'Massilia aquatica' Lu et al. 2020\n",
      "Coverage Percentage: 0.0541259544921236%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 285\n",
      "Length of fail 2299\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G009857605 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Duganella; s__Duganella levis\n",
      "Coverage Percentage: 0.2702007892086919%\n",
      "Adj, read counts 123\n",
      "Number of samples per group [1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 4, 2, 1, 1, 1, 2, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 3, 2, 1, 1, 1, 1, 1, 1, 1, 3, 2, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1]\n",
      "Number of bins 105\n",
      "total # samples 40\n",
      "Number of total reads 121\n",
      "Number of hits in BIOM tables 131.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 8.130081300813007 %\n",
      "Percentage of reads from samples who only fall into one bin group 9.917355371900827 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 286\n",
      "Length of fail 2299\n",
      "G009857835 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Duganella; s__Duganella fentianensis\n",
      "Coverage Percentage: 0.0466217846960459%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 286\n",
      "Length of fail 2300\n",
      "G009858245 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Intrasporangiaceae; g__Ornithinibacter; s__Ornithinibacter aureus\n",
      "Coverage Percentage: 0.0339318397551279%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 286\n",
      "Length of fail 2301\n",
      "G009858335 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Methylophilaceae; g__Methylovorus; s__Methylovorus glucosotrophus\n",
      "Coverage Percentage: 0.0231803669857922%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 286\n",
      "Length of fail 2302\n",
      "G009858895 d__Viruses; p__Pisuviricota; c__Pisoniviricetes; o__Nidovirales; f__Coronaviridae; g__Betacoronavirus; s__Severe acute respiratory syndrome-related coronavirus\n",
      "Coverage Percentage: 5.297127378523895%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 286\n",
      "Length of fail 2303\n",
      "G009861505 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas izuensis\n",
      "Coverage Percentage: 0.0043961623540605%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 286\n",
      "Length of fail 2304\n",
      "G009900765 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Rhodovarius; s__Rhodovarius lipocyclicus\n",
      "Coverage Percentage: 0.0049914534981806%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 286\n",
      "Length of fail 2305\n",
      "G009906855 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Xylophilus; s__Xylophilus rhododendri\n",
      "Coverage Percentage: 0.054499763340609%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 286\n",
      "Length of fail 2306\n",
      "G009908265 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Rhodobacter; s__Rhodobacter amnigenus\n",
      "Coverage Percentage: 0.0178111840720711%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 286\n",
      "Length of fail 2307\n",
      "G009909155 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium undicola\n",
      "Coverage Percentage: 0.0342159134208583%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 286\n",
      "Length of fail 2308\n",
      "G009909205 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Aquabacterium; s__Aquabacterium fontiphilum\n",
      "Coverage Percentage: 0.0249790769451327%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 286\n",
      "Length of fail 2309\n",
      "G009910365 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Cellvibrionales; f__Halieaceae; g__Aequoribacter; s__Aequoribacter fuscus\n",
      "Coverage Percentage: 0.0090457040659147%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 286\n",
      "Length of fail 2310\n",
      "G009911755 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas knackmussii\n",
      "Coverage Percentage: 0.0589930335395738%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 286\n",
      "Length of fail 2311\n",
      "G009915155 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium ichthyis\n",
      "Coverage Percentage: 0.0147116636235771%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 286\n",
      "Length of fail 2312\n",
      "G009925085 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Rhodobacter; s__Rhodobacter tardus\n",
      "Coverage Percentage: 0.0309371047137174%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 286\n",
      "Length of fail 2313\n",
      "G009932335 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas asiatica\n",
      "Coverage Percentage: 0.1001495038763118%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 2, 1, 1, 6, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 24\n",
      "total # samples 15\n",
      "Number of total reads 37\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 32.432432432432435 %\n",
      "Percentage of reads from samples who only fall into one bin group 32.432432432432435 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 287\n",
      "Length of fail 2313\n",
      "G009932375 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas juntendi\n",
      "Coverage Percentage: 0.0401657677972915%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 287\n",
      "Length of fail 2314\n",
      "G009932725 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas yangonensis\n",
      "Coverage Percentage: 0.0513358249465943%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 287\n",
      "Length of fail 2315\n",
      "G009935415 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Priestia; s__Priestia megaterium\n",
      "Coverage Percentage: 0.0040352387040098%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 287\n",
      "Length of fail 2316\n",
      "G009938015 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Sulfuricellaceae; g__Sulfuriferula; s__Sulfuriferula plumbiphila\n",
      "Coverage Percentage: 0.0085148926631166%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 287\n",
      "Length of fail 2317\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G009938305 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Fluviibacteraceae; g__Fluviibacter; s__Fluviibacter phosphoraccumulans\n",
      "Coverage Percentage: 0.7094780253989231%\n",
      "Adj, read counts 101\n",
      "Number of samples per group [3, 2, 1, 5, 2, 1, 1, 1, 1, 2, 2, 1, 2, 3, 1, 2, 1, 4, 1, 1, 2, 1, 1, 1, 4, 1, 2, 1, 1, 1, 1, 1, 1, 5, 1, 2, 2, 2, 3, 1, 6, 1, 1, 3, 1, 1, 5, 1, 1]\n",
      "Number of bins 73\n",
      "total # samples 36\n",
      "Number of total reads 100\n",
      "Number of hits in BIOM tables 93.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 18.81188118811881 %\n",
      "Percentage of reads from samples who only fall into one bin group 17.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2317\n",
      "G009939195 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter kanungonis\n",
      "Coverage Percentage: 0.0604569243701263%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2318\n",
      "G009996845 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Caldibacillus; s__Caldibacillus pasinlerensis\n",
      "Coverage Percentage: 0.0206386519671861%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2319\n",
      "G010093135 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Pseudoxanthomonas; s__Pseudoxanthomonas taiwanensis\n",
      "Coverage Percentage: 0.0875773153745128%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2320\n",
      "G010093235 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Pseudoxanthomonas; s__Pseudoxanthomonas suwonensis\n",
      "Coverage Percentage: 0.0112849011812987%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2321\n",
      "G010093385 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Pseudoxanthomonas; s__Pseudoxanthomonas mexicana\n",
      "Coverage Percentage: 0.0169129249025805%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2322\n",
      "G010093505 d__Eukaryota; p__Ascomycota; c__Dothideomycetes; o__Mycosphaerellales; f__Teratosphaeriaceae; g__Neohortaea; s__Neohortaea acidophila\n",
      "Coverage Percentage: 0.0004893994129556%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2323\n",
      "G010093625 d__Eukaryota; p__Ascomycota; c__Dothideomycetes; o__Pleosporales; f__Didymellaceae; g__Macroventuria; s__Macroventuria anomochaeta\n",
      "Coverage Percentage: 0.0009058844272647%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2324\n",
      "G010093815 d__Eukaryota; p__Ascomycota; c__Dothideomycetes; o__Acrospermales; f__Acrospermaceae; g__Pseudovirgaria; s__Pseudovirgaria hyperparasitica\n",
      "Coverage Percentage: 0.0026070346428349%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2325\n",
      "G010094085 d__Eukaryota; p__Ascomycota; c__Dothideomycetes; o__Pleosporales; f__Sporormiaceae; g__Westerdykella; s__Westerdykella ornata\n",
      "Coverage Percentage: 0.0011600495537589%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2326\n",
      "G010094145 d__Eukaryota; p__Ascomycota; c__Dothideomycetes; o__Pleosporales; f__Didymellaceae; g__Didymella; s__Didymella exigua\n",
      "Coverage Percentage: 0.0012909706171307%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2327\n",
      "G010119545 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Hymenobacteraceae; g__Pontibacter; s__Pontibacter fetidus\n",
      "Coverage Percentage: 0.0037073610512995%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2328\n",
      "G010120595 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus thermophilus\n",
      "Coverage Percentage: 0.088428306952301%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2329\n",
      "G010211765 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Pseudoxanthomonas; s__Pseudoxanthomonas kaohsiungensis\n",
      "Coverage Percentage: 0.0169647669136602%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2330\n",
      "G010450875 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas insulae\n",
      "Coverage Percentage: 0.0501261570632057%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2331\n",
      "G010499455 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Ideonella; s__Ideonella livida\n",
      "Coverage Percentage: 0.032885120614596%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2332\n",
      "G010508875 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Oscillospiraceae; g__Flavonifractor; s__Flavonifractor plautii\n",
      "Coverage Percentage: 0.0062227253931357%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2333\n",
      "G010509075 d__Bacteria; p__Actinobacteria; c__Coriobacteriia; o__Coriobacteriales; f__Coriobacteriaceae; g__Collinsella; s__Collinsella aerofaciens\n",
      "Coverage Percentage: 0.1364764874348774%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2334\n",
      "G010634845 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas antarctica\n",
      "Coverage Percentage: 0.0198152160447352%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2335\n",
      "G010645075 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium limi\n",
      "Coverage Percentage: 0.0303398449956685%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 288\n",
      "Length of fail 2336\n",
      "G010646905 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter portensis\n",
      "Coverage Percentage: 0.3967481477566165%\n",
      "Adj, read counts 81\n",
      "Number of samples per group [4, 1, 3, 2, 2, 1, 3, 1, 3, 3, 2, 3, 1, 8, 3, 3, 1, 3, 2, 1, 1, 2, 1, 2, 2, 3, 3, 2]\n",
      "Number of bins 37\n",
      "total # samples 21\n",
      "Number of total reads 80\n",
      "Number of hits in BIOM tables 22.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 23.456790123456788 %\n",
      "Percentage of reads from samples who only fall into one bin group 8.75 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 289\n",
      "Length of fail 2336\n",
      "G010667645 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium platyrrhinorum\n",
      "Coverage Percentage: 0.0057730694339677%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 289\n",
      "Length of fail 2337\n",
      "G010669145 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium oryzihabitans\n",
      "Coverage Percentage: 0.0364573492537252%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 289\n",
      "Length of fail 2338\n",
      "G010669225 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Selenomonadales; f__Selenomonadaceae; g__Megamonas; s__Megamonas funiformis\n",
      "Coverage Percentage: 0.2599671595843328%\n",
      "Adj, read counts 41\n",
      "Number of samples per group [1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 40\n",
      "total # samples 20\n",
      "Number of total reads 41\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 19.51219512195122 %\n",
      "Percentage of reads from samples who only fall into one bin group 41.46341463414634 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 290\n",
      "Length of fail 2338\n",
      "G010669285 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Cellvibrionales; f__Halieaceae; g__Kineobactrum; s__Kineobactrum salinum\n",
      "Coverage Percentage: 0.0032476589433322%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 290\n",
      "Length of fail 2339\n",
      "G010728155 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacterium; s__Mycobacterium doricum\n",
      "Coverage Percentage: 0.013699316870659%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 290\n",
      "Length of fail 2340\n",
      "G010729665 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacterium; s__Mycolicibacterium madagascariense\n",
      "Coverage Percentage: 0.0017507253305546%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 290\n",
      "Length of fail 2341\n",
      "G010730955 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacterium; s__Mycolicibacterium aubagnense\n",
      "Coverage Percentage: 0.0198198403403507%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 290\n",
      "Length of fail 2342\n",
      "G010731115 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacterium; s__Mycolicibacterium phocaicum\n",
      "Coverage Percentage: 0.0232180806489171%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 290\n",
      "Length of fail 2343\n",
      "G010731735 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacterium; s__Mycolicibacterium sediminis\n",
      "Coverage Percentage: 0.0032156702008641%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 290\n",
      "Length of fail 2344\n",
      "G010731755 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacter; s__Mycolicibacter minnesotensis\n",
      "Coverage Percentage: 0.0133994721176161%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 290\n",
      "Length of fail 2345\n",
      "G010731875 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacterium; s__Mycolicibacterium sarraceniae\n",
      "Coverage Percentage: 0.0031274238829556%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 290\n",
      "Length of fail 2346\n",
      "G010731895 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacterium; s__Mycolicibacterium helvum\n",
      "Coverage Percentage: 0.0070772281824912%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 290\n",
      "Length of fail 2347\n",
      "G010811875 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium uaiense\n",
      "Coverage Percentage: 0.0046104974726773%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 290\n",
      "Length of fail 2348\n",
      "G010915705 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Tabrizicola; s__Tabrizicola algicola\n",
      "Coverage Percentage: 0.0270935680937949%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 290\n",
      "Length of fail 2349\n",
      "G010918895 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter schindleri\n",
      "Coverage Percentage: 0.2434868904812386%\n",
      "Adj, read counts 54\n",
      "Number of samples per group [3, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 2, 10, 2, 7]\n",
      "Number of bins 43\n",
      "total # samples 27\n",
      "Number of total reads 54\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 44.44444444444444 %\n",
      "Percentage of reads from samples who only fall into one bin group 31.48148148148148 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 291\n",
      "Length of fail 2349\n",
      "G010974825 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas rivipollensis\n",
      "Coverage Percentage: 0.1503984665842714%\n",
      "Adj, read counts 43\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1]\n",
      "Number of bins 40\n",
      "total # samples 21\n",
      "Number of total reads 42\n",
      "Number of hits in BIOM tables 26.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 18.6046511627907 %\n",
      "Percentage of reads from samples who only fall into one bin group 35.714285714285715 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 292\n",
      "Length of fail 2349\n",
      "G010977555 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Caulobacter; s__Caulobacter rhizosphaerae\n",
      "Coverage Percentage: 0.0551481992578719%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 292\n",
      "Length of fail 2350\n",
      "G011008935 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Tabrizicola; s__Tabrizicola oligotrophica\n",
      "Coverage Percentage: 0.0108523621801928%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 292\n",
      "Length of fail 2351\n",
      "G011008945 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces harenosi\n",
      "Coverage Percentage: 0.0080776420963824%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 292\n",
      "Length of fail 2352\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G011038635 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium qintianiae\n",
      "Coverage Percentage: 0.0214908956550957%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 292\n",
      "Length of fail 2353\n",
      "G011038645 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium qintianiae\n",
      "Coverage Percentage: 0.032052279025232%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 292\n",
      "Length of fail 2354\n",
      "G011040435 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas psychrophila\n",
      "Coverage Percentage: 0.0723947397044969%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 292\n",
      "Length of fail 2355\n",
      "G011045025 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces ureilyticus\n",
      "Coverage Percentage: 0.0043178242740836%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 292\n",
      "Length of fail 2356\n",
      "G011045195 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Caulobacter; s__Caulobacter soli\n",
      "Coverage Percentage: 0.0766949783989315%\n",
      "Adj, read counts 32\n",
      "Number of samples per group [2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1]\n",
      "Number of bins 27\n",
      "total # samples 16\n",
      "Number of total reads 30\n",
      "Number of hits in BIOM tables 21.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 21.875 %\n",
      "Percentage of reads from samples who only fall into one bin group 36.666666666666664 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 293\n",
      "Length of fail 2356\n",
      "G011058205 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter lwoffii\n",
      "Coverage Percentage: 0.7889899174973233%\n",
      "Adj, read counts 175\n",
      "Number of samples per group [1, 4, 1, 1, 2, 2, 1, 7, 2, 3, 1, 8, 5, 8, 4, 3, 2, 5, 1, 1, 2, 2, 1, 1, 7, 2, 3, 3, 4, 1, 7, 1, 2, 9, 1, 2, 4, 1, 3, 1, 3, 1, 5, 1, 2, 2, 3, 1, 1, 4, 7, 3]\n",
      "Number of bins 100\n",
      "total # samples 43\n",
      "Number of total reads 173\n",
      "Number of hits in BIOM tables 71.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.142857142857142 %\n",
      "Percentage of reads from samples who only fall into one bin group 5.202312138728324 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 294\n",
      "Length of fail 2356\n",
      "G011058215 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Devosiaceae; g__Devosia; s__Devosia aurantiaca\n",
      "Coverage Percentage: 0.034524159600722%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 294\n",
      "Length of fail 2357\n",
      "G011064845 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Citrobacter; s__Citrobacter freundii\n",
      "Coverage Percentage: 0.0073016654856928%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 294\n",
      "Length of fail 2358\n",
      "G011083025 d__Bacteria; p__Proteobacteria; c__Deltaproteobacteria; o__Myxococcales; f__Nannocystaceae; g__Pseudenhygromyxa; s__Pseudenhygromyxa sp. WMMC2535\n",
      "Coverage Percentage: 0.0996609780374008%\n",
      "Adj, read counts 82\n",
      "Number of samples per group [24]\n",
      "Number of bins 8\n",
      "total # samples 24\n",
      "Number of total reads 82\n",
      "Number of hits in BIOM tables 6.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 100.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 100.0 %\n",
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 294\n",
      "Length of fail 2359\n",
      "G011089895 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Mucilaginibacter; s__Mucilaginibacter inviolabilis\n",
      "Coverage Percentage: 0.0270689073660022%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 294\n",
      "Length of fail 2360\n",
      "G011090165 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Duganella; s__Duganella aceris\n",
      "Coverage Percentage: 0.1164723456426258%\n",
      "Adj, read counts 54\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 1, 2, 1, 1, 7, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 46\n",
      "total # samples 26\n",
      "Number of total reads 54\n",
      "Number of hits in BIOM tables 40.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 24.074074074074073 %\n",
      "Percentage of reads from samples who only fall into one bin group 20.37037037037037 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 295\n",
      "Length of fail 2360\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G011100465 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Jeotgalibaca; s__Jeotgalibaca arthritidis\n",
      "Coverage Percentage: 0.0342830957507699%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 295\n",
      "Length of fail 2361\n",
      "G011250515 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Cytophagaceae; g__Aquirufa; s__Aquirufa ecclesiirivi\n",
      "Coverage Percentage: 0.0166298023360546%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 295\n",
      "Length of fail 2362\n",
      "G011250645 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Motilibacterales; f__Motilibacteraceae; g__Motilibacter; s__Motilibacter aurantiacus\n",
      "Coverage Percentage: 0.006578119084866%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 295\n",
      "Length of fail 2363\n",
      "G011299095 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Jeotgalibaca; s__Jeotgalibaca porci\n",
      "Coverage Percentage: 0.0763536069528998%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 295\n",
      "Length of fail 2364\n",
      "G011300215 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides piscis\n",
      "Coverage Percentage: 0.0199489096533698%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 295\n",
      "Length of fail 2365\n",
      "G011305415 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium bernardetii\n",
      "Coverage Percentage: 0.1107101926003078%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 295\n",
      "Length of fail 2366\n",
      "G011326735 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Intrasporangiaceae; g__Phycicoccus; s__Phycicoccus endophyticus\n",
      "Coverage Percentage: 0.0187878175636826%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 295\n",
      "Length of fail 2367\n",
      "G011355085 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas akapageensis\n",
      "Coverage Percentage: 0.002368364051747%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 295\n",
      "Length of fail 2368\n",
      "G011365745 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium turcicum\n",
      "Coverage Percentage: 0.0342005014700034%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 295\n",
      "Length of fail 2369\n",
      "G011392075 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium celericrescens\n",
      "Coverage Percentage: 0.0677042026241092%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 295\n",
      "Length of fail 2370\n",
      "G011405655 d__Bacteria; p__Actinobacteria; c__Coriobacteriia; o__Eggerthellales; f__Eggerthellaceae; g__Adlercreutzia; s__Adlercreutzia hattorii\n",
      "Coverage Percentage: 0.0123435396182923%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 295\n",
      "Length of fail 2371\n",
      "G011578045 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter shaoyimingii\n",
      "Coverage Percentage: 0.061830692554162%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 295\n",
      "Length of fail 2372\n",
      "G011578285 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter lanii\n",
      "Coverage Percentage: 0.2337597492252242%\n",
      "Adj, read counts 54\n",
      "Number of samples per group [2, 1, 1, 2, 1, 1, 2, 3, 1, 6, 2, 6, 1, 2, 2, 1, 1, 1, 1, 2, 2, 1, 1, 1, 3]\n",
      "Number of bins 32\n",
      "total # samples 20\n",
      "Number of total reads 53\n",
      "Number of hits in BIOM tables 12.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 37.03703703703704 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.867924528301888 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 296\n",
      "Length of fail 2372\n",
      "G011612265 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium matruchotii\n",
      "Coverage Percentage: 0.0392459201685655%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 296\n",
      "Length of fail 2373\n",
      "G011682045 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia aquatica\n",
      "Coverage Percentage: 0.0502335572483099%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 296\n",
      "Length of fail 2374\n",
      "G011682065 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia rubra\n",
      "Coverage Percentage: 0.0661007202298567%\n",
      "Adj, read counts 34\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 2, 2, 1, 2]\n",
      "Number of bins 30\n",
      "total # samples 21\n",
      "Number of total reads 34\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.588235294117645 %\n",
      "Percentage of reads from samples who only fall into one bin group 44.11764705882353 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 297\n",
      "Length of fail 2374\n",
      "G011682145 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia mucilaginosa\n",
      "Coverage Percentage: 0.0565940277219957%\n",
      "Adj, read counts 31\n",
      "Number of samples per group [1, 1, 2, 4, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 26\n",
      "total # samples 22\n",
      "Number of total reads 30\n",
      "Number of hits in BIOM tables 8.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 25.806451612903224 %\n",
      "Percentage of reads from samples who only fall into one bin group 56.666666666666664 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2374\n",
      "G011682175 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia frigida\n",
      "Coverage Percentage: 0.0319722532653891%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2375\n",
      "G011758645 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Brooklawnia; s__Brooklawnia cerclae\n",
      "Coverage Percentage: 0.0079732731659227%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2376\n",
      "G011758655 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Modestobacter; s__Modestobacter marinus\n",
      "Coverage Percentage: 0.0203993131251453%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2377\n",
      "G011758805 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacterium; s__Mycolicibacterium fluoranthenivorans\n",
      "Coverage Percentage: 0.0136836728038797%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2378\n",
      "G011759365 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Rhodanobacteraceae; g__Luteibacter; s__Luteibacter anthropi\n",
      "Coverage Percentage: 0.0052665973264227%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2379\n",
      "G011759585 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Frigoribacterium; s__Frigoribacterium endophyticum\n",
      "Coverage Percentage: 0.0488853978865263%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2380\n",
      "G011759705 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium ulmi\n",
      "Coverage Percentage: 0.0086025205385177%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2381\n",
      "G011761195 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium glutamicum\n",
      "Coverage Percentage: 0.0110671078395692%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2382\n",
      "G011761305 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas vulcanisoli\n",
      "Coverage Percentage: 0.0065182265527456%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2383\n",
      "G011761945 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas leidyi\n",
      "Coverage Percentage: 0.0263365760112012%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2384\n",
      "G011761985 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas terrae\n",
      "Coverage Percentage: 0.0080517978611435%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2385\n",
      "G011762035 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas insulae\n",
      "Coverage Percentage: 0.0295221401330075%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2386\n",
      "G011762125 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingopyxis; s__Sphingopyxis panaciterrae\n",
      "Coverage Percentage: 0.008259855039544%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2387\n",
      "G011762145 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas naasensis\n",
      "Coverage Percentage: 0.0102647616499278%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2388\n",
      "G011762225 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Beijerinckiaceae; g__Pseudochelatococcus; s__Pseudochelatococcus lubricantis\n",
      "Coverage Percentage: 0.0032632053926306%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2389\n",
      "G011764425 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Phytohabitans; s__Phytohabitans houttuyneae\n",
      "Coverage Percentage: 0.0013312295791806%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2390\n",
      "G011764505 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Legionellales; f__Legionellaceae; g__Legionella; s__Legionella antarctica\n",
      "Coverage Percentage: 0.0292576575998484%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2391\n",
      "G011764545 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Phytohabitans; s__Phytohabitans flavus\n",
      "Coverage Percentage: 0.0015613608571413%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2392\n",
      "G011764565 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Phytohabitans; s__Phytohabitans suffuscus\n",
      "Coverage Percentage: 0.0014878725088358%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2393\n",
      "G011764585 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Sulfuricellaceae; g__Sulfurimicrobium; s__Sulfurimicrobium lacus\n",
      "Coverage Percentage: 0.0185818033182762%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2394\n",
      "G011800055 d__Eukaryota; p__Ascomycota; c__Sordariomycetes; o__Glomerellales; f__Glomerellaceae; g__Colletotrichum; s__Colletotrichum gloeosporioides\n",
      "Coverage Percentage: 0.0038493487969573%\n",
      "Adj, read counts 145\n",
      "Number of samples per group [3, 1, 2, 1, 54, 3, 16, 5, 2, 4, 1, 2]\n",
      "Number of bins 13\n",
      "total # samples 63\n",
      "Number of total reads 121\n",
      "Number of hits in BIOM tables 44.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 86.20689655172413 %\n",
      "Percentage of reads from samples who only fall into one bin group 45.45454545454545 %\n",
      "Status: Fail - too many reads in top 3 bins\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2395\n",
      "G011839805 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Kineosporiales; f__Kineosporiaceae; g__Kineococcus; s__Kineococcus rubinsiae\n",
      "Coverage Percentage: 0.0088479317037982%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2396\n",
      "G011926545 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingopyxis; s__Sphingopyxis italica\n",
      "Coverage Percentage: 0.0149891662461785%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2397\n",
      "G011927635 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas trueperi\n",
      "Coverage Percentage: 0.0293681624879616%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2398\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G011927695 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas jejuensis\n",
      "Coverage Percentage: 0.0717336625317695%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2399\n",
      "G011927945 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas alba\n",
      "Coverage Percentage: 0.0318642029373892%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2400\n",
      "G012029655 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Raoultella; s__Raoultella terrigena\n",
      "Coverage Percentage: 0.0199162125829738%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2401\n",
      "G012030425 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Sphingobacterium; s__Sphingobacterium kitahiroshimense\n",
      "Coverage Percentage: 0.0650946829486977%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2402\n",
      "G012163135 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Roseomonas; s__Roseomonas oryzicola\n",
      "Coverage Percentage: 0.0028454627306567%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2403\n",
      "G012163155 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Kocuria; s__Kocuria subflava\n",
      "Coverage Percentage: 0.0172487316575569%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2404\n",
      "G012184415 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Ramlibacter; s__Ramlibacter lithotrophicus\n",
      "Coverage Percentage: 0.0489517729108828%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2405\n",
      "G012221945 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Glutamicibacter; s__Glutamicibacter mishrai\n",
      "Coverage Percentage: 0.0042288070255327%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2406\n",
      "G012271765 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Erwiniaceae; g__Erwinia; s__Erwinia rhapontici\n",
      "Coverage Percentage: 0.0057624240342692%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2407\n",
      "G012271785 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Paeniglutamicibacter; s__Paeniglutamicibacter terrestris\n",
      "Coverage Percentage: 0.0104361456480466%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 298\n",
      "Length of fail 2408\n",
      "G012273055 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Tannerellaceae; g__Parabacteroides; s__Parabacteroides distasonis\n",
      "Coverage Percentage: 0.141801025583252%\n",
      "Adj, read counts 46\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 40\n",
      "total # samples 27\n",
      "Number of total reads 45\n",
      "Number of hits in BIOM tables 19.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 19.565217391304348 %\n",
      "Percentage of reads from samples who only fall into one bin group 42.22222222222222 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 299\n",
      "Length of fail 2408\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G012274265 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingopyxis; s__Sphingopyxis microcysteis\n",
      "Coverage Percentage: 0.008402155456045%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 299\n",
      "Length of fail 2409\n",
      "G012277075 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus alactolyticus\n",
      "Coverage Percentage: 0.0847693596440157%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 299\n",
      "Length of fail 2410\n",
      "G012349115 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium ruizarguesonis\n",
      "Coverage Percentage: 0.0160420782285395%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 299\n",
      "Length of fail 2411\n",
      "G012396185 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Sanguibacteraceae; g__Sanguibacter; s__Sanguibacter hominis\n",
      "Coverage Percentage: 0.0050625577919469%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 299\n",
      "Length of fail 2412\n",
      "G012396285 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Gordoniaceae; g__Gordonia; s__Gordonia polyisoprenivorans\n",
      "Coverage Percentage: 0.0069810758776158%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 299\n",
      "Length of fail 2413\n",
      "G012396315 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium mucifaciens\n",
      "Coverage Percentage: 0.046474523178711%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 299\n",
      "Length of fail 2414\n",
      "G012396345 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptosporangiales; f__Nocardiopsaceae; g__Nocardiopsis; s__Nocardiopsis dassonvillei\n",
      "Coverage Percentage: 0.0118242123486708%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 299\n",
      "Length of fail 2415\n",
      "G012396435 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Weissella; s__Weissella hellenica\n",
      "Coverage Percentage: 0.0231967164003533%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 299\n",
      "Length of fail 2416\n",
      "G012396465 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Weissella; s__Weissella soli\n",
      "Coverage Percentage: 0.0309837335398915%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 299\n",
      "Length of fail 2417\n",
      "G012396615 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Rothia; s__Rothia terrae\n",
      "Coverage Percentage: 0.0258134976806109%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 299\n",
      "Length of fail 2418\n",
      "G012396675 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Alteromonadales; f__Shewanellaceae; g__Shewanella; s__Shewanella algae\n",
      "Coverage Percentage: 0.0033337483783121%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 299\n",
      "Length of fail 2419\n",
      "G012530715 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Alicycliphilus; s__Alicycliphilus denitrificans\n",
      "Coverage Percentage: 0.1297469860397777%\n",
      "Adj, read counts 46\n",
      "Number of samples per group [1, 1, 1, 1, 6, 1, 1, 2, 5, 1, 1, 1, 2, 1, 2, 2, 3, 1, 3, 6]\n",
      "Number of bins 27\n",
      "total # samples 18\n",
      "Number of total reads 43\n",
      "Number of hits in BIOM tables 6.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 39.13043478260869 %\n",
      "Percentage of reads from samples who only fall into one bin group 20.930232558139537 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 300\n",
      "Length of fail 2419\n",
      "G012584515 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Polaromonas; s__Polaromonas vacuolata\n",
      "Coverage Percentage: 0.3097700020272633%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 300\n",
      "Length of fail 2420\n",
      "G012689545 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus sanguinis\n",
      "Coverage Percentage: 0.0191177272608713%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 300\n",
      "Length of fail 2421\n",
      "G012767755 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas tohonis\n",
      "Coverage Percentage: 0.0154847699889042%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 300\n",
      "Length of fail 2422\n",
      "G012844455 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Comamonas; s__Comamonas suwonensis\n",
      "Coverage Percentage: 0.0445976328330016%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 300\n",
      "Length of fail 2423\n",
      "G012863495 d__Bacteria; p__Cyanobacteria; c__; o__Pseudanabaenales; f__Pseudanabaenaceae; g__Pseudanabaena; s__Pseudanabaena yagii\n",
      "Coverage Percentage: 0.0153924407849031%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 300\n",
      "Length of fail 2424\n",
      "G012871055 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium elkanii\n",
      "Coverage Percentage: 0.0090155300276735%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 300\n",
      "Length of fail 2425\n",
      "G012910805 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Rhodocyclaceae; g__Aromatoleum; s__Aromatoleum evansii\n",
      "Coverage Percentage: 0.0465387655134124%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 300\n",
      "Length of fail 2426\n",
      "G012911005 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Rhodocyclaceae; g__Aromatoleum; s__Aromatoleum toluolicum\n",
      "Coverage Percentage: 0.0305937542965865%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 300\n",
      "Length of fail 2427\n",
      "G012911495 d__Bacteria; p__Proteobacteria; c__Hydrogenophilalia; o__Hydrogenophilales; f__Hydrogenophilaceae; g__Tepidiphilus; s__Tepidiphilus baoligensis\n",
      "Coverage Percentage: 0.0361556344603171%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 300\n",
      "Length of fail 2428\n",
      "G012911605 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium solisilvae\n",
      "Coverage Percentage: 0.0526969687046169%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 300\n",
      "Length of fail 2429\n",
      "G012913485 d__Bacteria; p__Verrucomicrobia; c__Verrucomicrobiae; o__Verrucomicrobiales; f__Verrucomicrobiaceae; g__Luteolibacter; s__Luteolibacter luteus\n",
      "Coverage Percentage: 0.0127727683953987%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 300\n",
      "Length of fail 2430\n",
      "G012927045 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Azohydromonas; s__Azohydromonas caseinilytica\n",
      "Coverage Percentage: 0.0313698199796006%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 300\n",
      "Length of fail 2431\n",
      "G012927105 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium psychrophilum\n",
      "Coverage Percentage: 0.0477152545707592%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 300\n",
      "Length of fail 2432\n",
      "G012927165 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Zoogloeaceae; g__Zoogloea; s__Zoogloea dura\n",
      "Coverage Percentage: 0.1139418140565617%\n",
      "Adj, read counts 39\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 4, 1, 5, 4, 2, 5, 1, 4, 2, 2]\n",
      "Number of bins 27\n",
      "total # samples 16\n",
      "Number of total reads 38\n",
      "Number of hits in BIOM tables 55.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 38.46153846153846 %\n",
      "Percentage of reads from samples who only fall into one bin group 15.789473684210526 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 301\n",
      "Length of fail 2432\n",
      "G012927325 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium antibioticum\n",
      "Coverage Percentage: 0.0086524421874351%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 301\n",
      "Length of fail 2433\n",
      "G012927405 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium olei\n",
      "Coverage Percentage: 0.0472853410959567%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 301\n",
      "Length of fail 2434\n",
      "G012931585 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas salmonicida\n",
      "Coverage Percentage: 0.0441610004922404%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 301\n",
      "Length of fail 2435\n",
      "G012932915 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Antrihabitans; s__Antrihabitans stalactiti\n",
      "Coverage Percentage: 0.0033428743305561%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 301\n",
      "Length of fail 2436\n",
      "G012933545 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Rouxiella; s__Rouxiella aceris\n",
      "Coverage Percentage: 0.0235381618967438%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 301\n",
      "Length of fail 2437\n",
      "G012933655 d__Bacteria; p__Proteobacteria; c__Deltaproteobacteria; o__Myxococcales; f__Myxococcaceae; g__Pyxidicoccus; s__Pyxidicoccus fallax\n",
      "Coverage Percentage: 0.0011144432060922%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 301\n",
      "Length of fail 2438\n",
      "G012958305 d__Bacteria; p__Planctomycetes; c__Planctomycetia; o__Isosphaerales; f__Isosphaeraceae; g__Tautonia; s__Tautonia rosea\n",
      "Coverage Percentage: 0.0043344869000897%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 301\n",
      "Length of fail 2439\n",
      "G012971845 d__Eukaryota; p__Ascomycota; c__Dothideomycetes; o__Botryosphaeriales; f__Botryosphaeriaceae; g__Lasiodiplodia; s__Lasiodiplodia theobromae\n",
      "Coverage Percentage: 0.0006407253157025%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 301\n",
      "Length of fail 2440\n",
      "G012985915 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas psychrotolerans\n",
      "Coverage Percentage: 0.0659926260558113%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 301\n",
      "Length of fail 2441\n",
      "G012986145 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas peli\n",
      "Coverage Percentage: 0.0144078868329203%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 301\n",
      "Length of fail 2442\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G012986205 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas nitroreducens\n",
      "Coverage Percentage: 0.0230668151978205%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 301\n",
      "Length of fail 2443\n",
      "G012986985 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas chengduensis\n",
      "Coverage Percentage: 0.2523227962006913%\n",
      "Adj, read counts 88\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 1, 2, 3, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 1, 2, 4, 1, 1, 2, 2, 1, 1, 1, 4, 1, 1, 1, 4, 1, 1, 2, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1]\n",
      "Number of bins 67\n",
      "total # samples 48\n",
      "Number of total reads 87\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 19.318181818181817 %\n",
      "Percentage of reads from samples who only fall into one bin group 40.229885057471265 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 302\n",
      "Length of fail 2443\n",
      "G012987025 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas aeruginosa\n",
      "Coverage Percentage: 0.5496991793691389%\n",
      "Adj, read counts 204\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 2, 1, 1, 1, 1, 1, 3, 2, 2, 2, 1, 1, 2, 3, 1, 1, 1, 1, 2, 2, 1, 2, 1, 1, 1, 2, 1, 3, 1, 3, 1, 5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 4, 1, 1, 1, 2, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 3, 1, 4, 3, 2, 1, 2, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2]\n",
      "Number of bins 168\n",
      "total # samples 51\n",
      "Number of total reads 201\n",
      "Number of hits in BIOM tables 54.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 6.862745098039216 %\n",
      "Percentage of reads from samples who only fall into one bin group 10.945273631840797 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 303\n",
      "Length of fail 2443\n",
      "G012999515 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Salinisphaerales; f__Salinisphaeraceae; g__Salifodinibacter; s__Salifodinibacter halophilus\n",
      "Coverage Percentage: 0.0112112822601622%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 303\n",
      "Length of fail 2444\n",
      "G013004105 d__Bacteria; p__Gemmatimonadetes; c__Gemmatimonadetes; o__Gemmatimonadales; f__Gemmatimonadaceae; g__Gemmatimonas; s__Gemmatimonas groenlandica\n",
      "Coverage Percentage: 0.0028190269645721%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 303\n",
      "Length of fail 2445\n",
      "G013004375 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter terrae\n",
      "Coverage Percentage: 0.4815699596145974%\n",
      "Adj, read counts 102\n",
      "Number of samples per group [1, 2, 1, 1, 1, 2, 2, 1, 3, 3, 1, 1, 2, 2, 1, 1, 1, 1, 4, 1, 1, 1, 2, 1, 2, 2, 1, 1, 1, 1, 1, 4, 1, 2, 1, 2, 1, 1, 5, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 3, 3, 1, 5, 2, 1, 1, 1, 1, 1, 1, 2]\n",
      "Number of bins 76\n",
      "total # samples 35\n",
      "Number of total reads 101\n",
      "Number of hits in BIOM tables 69.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.686274509803921 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.881188118811881 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 304\n",
      "Length of fail 2445\n",
      "G013004565 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium ulmi\n",
      "Coverage Percentage: 0.0086222279323069%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 304\n",
      "Length of fail 2446\n",
      "G013004615 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Cupriavidus; s__Cupriavidus gilardii\n",
      "Coverage Percentage: 0.0216525211047019%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 304\n",
      "Length of fail 2447\n",
      "G013004645 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas maltophilia\n",
      "Coverage Percentage: 0.3706077635035845%\n",
      "Adj, read counts 104\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 3, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 2, 2, 1, 1, 2, 2, 1, 1, 1, 1, 1]\n",
      "Number of bins 82\n",
      "total # samples 30\n",
      "Number of total reads 103\n",
      "Number of hits in BIOM tables 35.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 9.615384615384617 %\n",
      "Percentage of reads from samples who only fall into one bin group 20.388349514563107 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 305\n",
      "Length of fail 2447\n",
      "G013009555 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola dorei\n",
      "Coverage Percentage: 0.3970144456800903%\n",
      "Adj, read counts 137\n",
      "Number of samples per group [1, 1, 1, 1, 2, 1, 1, 3, 2, 1, 1, 2, 1, 1, 1, 2, 1, 2, 1, 1, 2, 1, 3, 1, 1, 2, 1, 1, 2, 1, 2, 1, 1, 1, 3, 1, 2, 2, 1, 1, 1, 2, 2, 1, 1, 1, 4, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 2, 2, 1, 4, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 2, 1, 2, 1, 1, 1, 3]\n",
      "Number of bins 116\n",
      "total # samples 44\n",
      "Number of total reads 136\n",
      "Number of hits in BIOM tables 45.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 9.48905109489051 %\n",
      "Percentage of reads from samples who only fall into one bin group 13.970588235294118 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2447\n",
      "G013036055 d__Eukaryota; p__Ascomycota; c__Dothideomycetes; o__Pleosporales; f__Pleosporaceae; g__Alternaria; s__Alternaria burnsii\n",
      "Coverage Percentage: 0.0079795025693391%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2448\n",
      "G013087515 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium sophorae\n",
      "Coverage Percentage: 0.016607476429979%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2449\n",
      "G013087605 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium indigoferae\n",
      "Coverage Percentage: 0.013238845572666%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2450\n",
      "G013087625 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium changzhiense\n",
      "Coverage Percentage: 0.0154769659502197%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2451\n",
      "G013106755 d__Bacteria; p__Bacteroidetes; c__Chitinophagia; o__Chitinophagales; f__Chitinophagaceae; g__Limnovirga; s__Limnovirga soli\n",
      "Coverage Percentage: 0.0099060403859962%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2452\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G013114825 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium australiense\n",
      "Coverage Percentage: 0.0161620920685272%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2453\n",
      "G013114835 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium archetypum\n",
      "Coverage Percentage: 0.0140355416963882%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2454\n",
      "G013155125 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Pseudonocardia; s__Pseudonocardia broussonetiae\n",
      "Coverage Percentage: 0.0120892021195211%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2455\n",
      "G013170785 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Phyllobacteriaceae; g__Mesorhizobium; s__Mesorhizobium jarvisii\n",
      "Coverage Percentage: 0.0139978672693896%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2456\n",
      "G013177675 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Pseudoarcobacter; s__Pseudoarcobacter acticola\n",
      "Coverage Percentage: 0.0563749577270624%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2457\n",
      "G013177735 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia elongata\n",
      "Coverage Percentage: 0.0101334868064799%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2458\n",
      "G013179805 d__Bacteria; p__Cyanobacteria; c__; o__Oscillatoriales; f__Microcoleaceae; g__Microcoleus; s__Microcoleus asticus\n",
      "Coverage Percentage: 0.0198107071045221%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2459\n",
      "G013185915 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas bentonitica\n",
      "Coverage Percentage: 0.1005444779466699%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2460\n",
      "G013200955 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas campi\n",
      "Coverage Percentage: 0.0218845258822269%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2461\n",
      "G013201075 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Lichenicola; s__Lichenicola cladoniae\n",
      "Coverage Percentage: 0.0074734667184915%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2462\n",
      "G013201665 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Arcobacter; s__Arcobacter venerupis\n",
      "Coverage Percentage: 0.0570256007553555%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2463\n",
      "G013201705 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Aliarcobacter; s__Aliarcobacter faecis\n",
      "Coverage Percentage: 0.0376069101982917%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2464\n",
      "G013204795 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter panaciterrae\n",
      "Coverage Percentage: 0.0083643732925079%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2465\n",
      "G013260645 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Erythrobacter; s__Erythrobacter mangrovi\n",
      "Coverage Percentage: 0.009808110849961%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2466\n",
      "G013266735 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter foliorum\n",
      "Coverage Percentage: 0.0113988943565932%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2467\n",
      "G013266755 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Aquabacterium; s__Aquabacterium terrae\n",
      "Coverage Percentage: 0.0256697708351107%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2468\n",
      "G013285305 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas rhodesiae\n",
      "Coverage Percentage: 0.0188954894286485%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2469\n",
      "G013285525 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium pusense\n",
      "Coverage Percentage: 0.0724144210538651%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2470\n",
      "G013300945 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Tyzzerella; s__Tyzzerella nexilis\n",
      "Coverage Percentage: 0.0380709517197661%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2471\n",
      "G013302405 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Blautia; s__Blautia caecimuris\n",
      "Coverage Percentage: 0.0299081248251897%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 306\n",
      "Length of fail 2472\n",
      "G013302415 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Blautia; s__Blautia faecis\n",
      "Coverage Percentage: 0.150805351771903%\n",
      "Adj, read counts 38\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 35\n",
      "total # samples 22\n",
      "Number of total reads 37\n",
      "Number of hits in BIOM tables 53.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 18.421052631578945 %\n",
      "Percentage of reads from samples who only fall into one bin group 35.13513513513514 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 307\n",
      "Length of fail 2472\n",
      "G013304385 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Blautia; s__Blautia glucerasea\n",
      "Coverage Percentage: 0.096321943580573%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 307\n",
      "Length of fail 2473\n",
      "G013304825 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Blautia; s__Blautia schinkii\n",
      "Coverage Percentage: 0.0625351331017111%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 307\n",
      "Length of fail 2474\n",
      "G013305055 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Blautia; s__Blautia massiliensis\n",
      "Coverage Percentage: 0.1387456543855053%\n",
      "Adj, read counts 31\n",
      "Number of samples per group [1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1]\n",
      "Number of bins 29\n",
      "total # samples 20\n",
      "Number of total reads 31\n",
      "Number of hits in BIOM tables 16.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 25.806451612903224 %\n",
      "Percentage of reads from samples who only fall into one bin group 45.16129032258064 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 308\n",
      "Length of fail 2474\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G013314975 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Pseudarthrobacter; s__Pseudarthrobacter oxydans\n",
      "Coverage Percentage: 0.0049236871126859%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 308\n",
      "Length of fail 2475\n",
      "G013328205 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas hominis\n",
      "Coverage Percentage: 0.0176463827865717%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 308\n",
      "Length of fail 2476\n",
      "G013340325 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Eurotiales; f__Aspergillaceae; g__Aspergillus; s__Aspergillus tubingensis\n",
      "Coverage Percentage: 0.001335330279821%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 308\n",
      "Length of fail 2477\n",
      "G013340935 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Azospirillaceae; g__Azospirillum; s__Azospirillum melinis\n",
      "Coverage Percentage: 0.0056814234009928%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 308\n",
      "Length of fail 2478\n",
      "G013343135 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Achromobacter; s__Achromobacter xylosoxidans\n",
      "Coverage Percentage: 0.1413073916195225%\n",
      "Adj, read counts 55\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1]\n",
      "Number of bins 46\n",
      "total # samples 15\n",
      "Number of total reads 54\n",
      "Number of hits in BIOM tables 19.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.545454545454545 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.51851851851852 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 309\n",
      "Length of fail 2478\n",
      "G013347325 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Cupriavidus; s__Cupriavidus gilardii\n",
      "Coverage Percentage: 0.0338191508754264%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 309\n",
      "Length of fail 2479\n",
      "G013350005 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylocystaceae; g__Methylocystis; s__Methylocystis silviterrae\n",
      "Coverage Percentage: 0.0194037094808132%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 309\n",
      "Length of fail 2480\n",
      "G013358205 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola sartorii\n",
      "Coverage Percentage: 0.0929744690509571%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 309\n",
      "Length of fail 2481\n",
      "G013359755 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Tardiphaga; s__Tardiphaga robiniae\n",
      "Coverage Percentage: 0.0023892647014146%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 309\n",
      "Length of fail 2482\n",
      "G013359795 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas zeae\n",
      "Coverage Percentage: 0.0495479470009808%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 309\n",
      "Length of fail 2483\n",
      "G013359815 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Curtobacterium; s__Curtobacterium flaccumfaciens\n",
      "Coverage Percentage: 0.0462853402416554%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 309\n",
      "Length of fail 2484\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G013359865 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Curtobacterium; s__Curtobacterium pusillum\n",
      "Coverage Percentage: 0.0121056856832122%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 309\n",
      "Length of fail 2485\n",
      "G013361095 d__Bacteria; p__Cyanobacteria; c__; o__Pseudanabaenales; f__Pseudanabaenaceae; g__Pseudanabaena; s__Pseudanabaena biceps\n",
      "Coverage Percentage: 0.1912127936260965%\n",
      "Adj, read counts 56\n",
      "Number of samples per group [1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 3, 2, 2, 2, 1, 1]\n",
      "Number of bins 29\n",
      "total # samples 6\n",
      "Number of total reads 55\n",
      "Number of hits in BIOM tables 165.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 21.428571428571427 %\n",
      "Percentage of reads from samples who only fall into one bin group 5.454545454545454 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 310\n",
      "Length of fail 2485\n",
      "G013363755 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax antarcticus\n",
      "Coverage Percentage: 0.211283342344083%\n",
      "Adj, read counts 84\n",
      "Number of samples per group [1, 8, 2, 9, 3, 7, 5, 1, 13, 1, 2]\n",
      "Number of bins 22\n",
      "total # samples 24\n",
      "Number of total reads 83\n",
      "Number of hits in BIOM tables 14.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 66.66666666666667 %\n",
      "Percentage of reads from samples who only fall into one bin group 13.253012048192772 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 311\n",
      "Length of fail 2485\n",
      "G013366375 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Schlegelella; s__Schlegelella koreensis\n",
      "Coverage Percentage: 0.094396922201263%\n",
      "Adj, read counts 35\n",
      "Number of samples per group [1, 1, 1, 1, 2, 1, 2, 11, 2, 4, 6, 1, 1]\n",
      "Number of bins 14\n",
      "total # samples 21\n",
      "Number of total reads 35\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 62.857142857142854 %\n",
      "Percentage of reads from samples who only fall into one bin group 40.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 312\n",
      "Length of fail 2485\n",
      "G013372045 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Janthinobacterium; s__Janthinobacterium lividum\n",
      "Coverage Percentage: 0.1889045559730945%\n",
      "Adj, read counts 85\n",
      "Number of samples per group [5, 1, 1, 1, 1, 1, 1, 5, 1, 1, 8, 1, 2, 7, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 6, 5]\n",
      "Number of bins 42\n",
      "total # samples 25\n",
      "Number of total reads 81\n",
      "Number of hits in BIOM tables 14.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 34.11764705882353 %\n",
      "Percentage of reads from samples who only fall into one bin group 13.580246913580247 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 313\n",
      "Length of fail 2485\n",
      "G013372265 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Aliarcobacter; s__Aliarcobacter cibarius\n",
      "Coverage Percentage: 0.0917977334305091%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 313\n",
      "Length of fail 2486\n",
      "G013374815 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Shigella; s__Shigella sonnei\n",
      "Coverage Percentage: 1.0746048416322085%\n",
      "Adj, read counts 303\n",
      "Number of samples per group [1, 4, 4, 2, 1, 2, 1, 2, 4, 1, 4, 1, 1, 2, 1, 2, 4, 3, 1, 1, 2, 1, 3, 7, 3, 1, 3, 2, 3, 1, 1, 1, 1, 1, 1, 1, 5, 4, 2, 1, 4, 1, 2, 5, 4, 1, 5, 2, 1, 1, 3, 3, 1, 1, 4, 1, 3, 1, 1, 2, 3, 2, 1, 3, 2, 1, 2, 1, 1, 2, 2, 2, 1, 1, 2, 1, 1, 2, 2, 2, 1, 1, 1, 1, 2, 2, 1, 1, 1, 3, 3, 1, 1, 1, 2, 5, 1, 1, 1, 1, 1, 2, 1, 2, 3, 1, 2, 2, 4]\n",
      "Number of bins 189\n",
      "total # samples 61\n",
      "Number of total reads 299\n",
      "Number of hits in BIOM tables 56.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 6.9306930693069315 %\n",
      "Percentage of reads from samples who only fall into one bin group 10.702341137123746 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 314\n",
      "Length of fail 2486\n",
      "G013375155 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia ureilytica\n",
      "Coverage Percentage: 0.0413977519686482%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 314\n",
      "Length of fail 2487\n",
      "G013377995 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pasteurellales; f__Pasteurellaceae; g__Mannheimia; s__Mannheimia pernigra\n",
      "Coverage Percentage: 0.0072371708470163%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 314\n",
      "Length of fail 2488\n",
      "G013385825 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas reactans\n",
      "Coverage Percentage: 0.0550519111776606%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 314\n",
      "Length of fail 2489\n",
      "G013385965 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas edaphica\n",
      "Coverage Percentage: 0.0494462160852941%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 314\n",
      "Length of fail 2490\n",
      "G013386885 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas salomonii\n",
      "Coverage Percentage: 0.0411893584286304%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 314\n",
      "Length of fail 2491\n",
      "G013387005 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas putida\n",
      "Coverage Percentage: 0.0248170991291185%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 314\n",
      "Length of fail 2492\n",
      "G013387465 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Hydrogenophaga; s__Hydrogenophaga aromaticivorans\n",
      "Coverage Percentage: 0.0835319717468104%\n",
      "Adj, read counts 33\n",
      "Number of samples per group [1, 2, 1, 3, 1, 5, 1, 1, 1, 6, 1, 1, 1, 3]\n",
      "Number of bins 16\n",
      "total # samples 19\n",
      "Number of total reads 33\n",
      "Number of hits in BIOM tables 15.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 57.57575757575758 %\n",
      "Percentage of reads from samples who only fall into one bin group 42.42424242424242 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 315\n",
      "Length of fail 2492\n",
      "G013388375 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Xanthomonas; s__Xanthomonas campestris\n",
      "Coverage Percentage: 0.022763883490195%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 315\n",
      "Length of fail 2493\n",
      "G013391375 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__Bacillus paramycoides\n",
      "Coverage Percentage: 0.0082334636029449%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 315\n",
      "Length of fail 2494\n",
      "G013393325 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas hunanensis\n",
      "Coverage Percentage: 0.1520511574486069%\n",
      "Adj, read counts 50\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1]\n",
      "Number of bins 38\n",
      "total # samples 7\n",
      "Number of total reads 50\n",
      "Number of hits in BIOM tables 106.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 8.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 8.0 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 316\n",
      "Length of fail 2494\n",
      "G013403565 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Chromobacteriaceae; g__Chitinibacter; s__Chitinibacter bivalviorum\n",
      "Coverage Percentage: 0.0162322743292427%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 316\n",
      "Length of fail 2495\n",
      "G013408295 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Psychromicrobium; s__Psychromicrobium silvestre\n",
      "Coverage Percentage: 0.0127621492343816%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 316\n",
      "Length of fail 2496\n",
      "G013408445 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium tuberculostearicum\n",
      "Coverage Percentage: 0.8900721188730346%\n",
      "Adj, read counts 127\n",
      "Number of samples per group [1, 3, 1, 2, 1, 1, 2, 1, 4, 1, 1, 3, 3, 1, 1, 1, 1, 2, 3, 3, 2, 1, 2, 2, 1, 4, 2, 2, 2, 1, 1, 3, 5, 1, 2, 1, 2, 1, 6, 2, 5, 1, 1, 4, 2, 2, 4, 3, 6, 1, 4, 1, 1, 1]\n",
      "Number of bins 99\n",
      "total # samples 38\n",
      "Number of total reads 123\n",
      "Number of hits in BIOM tables 94.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.53543307086614 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.38211382113821 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 317\n",
      "Length of fail 2496\n",
      "G013408525 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides aromaticivorans\n",
      "Coverage Percentage: 0.0078178825217635%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 317\n",
      "Length of fail 2497\n",
      "G013408685 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Schumannella; s__Schumannella luteola\n",
      "Coverage Percentage: 0.014355150143152%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 317\n",
      "Length of fail 2498\n",
      "G013408715 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Pseudonocardia; s__Pseudonocardia antarctica\n",
      "Coverage Percentage: 0.868672600281329%\n",
      "Adj, read counts 289\n",
      "Number of samples per group [2, 2, 1, 1, 1, 1, 1, 1, 3, 2, 1, 1, 1, 1, 2, 1, 1, 3, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 2, 3, 1, 2, 2, 1, 3, 2, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 2, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 2, 1, 2, 3, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 2, 1, 2, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 3, 1, 1, 1, 3, 1, 2, 1, 1, 1, 1, 1, 1, 1, 3, 1]\n",
      "Number of bins 207\n",
      "total # samples 9\n",
      "Number of total reads 283\n",
      "Number of hits in BIOM tables 382.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 4.1522491349480966 %\n",
      "Percentage of reads from samples who only fall into one bin group 2.1201413427561837 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 318\n",
      "Length of fail 2498\n",
      "G013409125 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas sediminis\n",
      "Coverage Percentage: 0.1145719603317134%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 318\n",
      "Length of fail 2499\n",
      "G013409305 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Cellulomonadaceae; g__Cellulomonas; s__Cellulomonas soli\n",
      "Coverage Percentage: 0.0265159601889514%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 318\n",
      "Length of fail 2500\n",
      "G013409505 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Actinomycetospora; s__Actinomycetospora corticicola\n",
      "Coverage Percentage: 0.0399741311423941%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 318\n",
      "Length of fail 2501\n",
      "G013409575 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Porphyromonadaceae; g__Macellibacteroides; s__Macellibacteroides fermentans\n",
      "Coverage Percentage: 0.0760892899560867%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 318\n",
      "Length of fail 2502\n",
      "G013409595 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Leucobacter; s__Leucobacter aridicollis\n",
      "Coverage Percentage: 0.0206245228655157%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 318\n",
      "Length of fail 2503\n",
      "G013409785 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium immunditiarum\n",
      "Coverage Percentage: 0.0112366251525838%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 318\n",
      "Length of fail 2504\n",
      "G013409865 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Herbiconiux; s__Herbiconiux flava\n",
      "Coverage Percentage: 0.0483480728707558%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 318\n",
      "Length of fail 2505\n",
      "G013410065 d__Bacteria; p__Acidobacteria; c__Acidobacteriia; o__Acidobacteriales; f__Acidobacteriaceae; g__Granulicella; s__Granulicella arctica\n",
      "Coverage Percentage: 0.0031588289384964%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 318\n",
      "Length of fail 2506\n",
      "G013410185 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora purpureochromogenes\n",
      "Coverage Percentage: 0.0022884782998455%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 318\n",
      "Length of fail 2507\n",
      "G013410345 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Saccharopolyspora; s__Saccharopolyspora hordei\n",
      "Coverage Percentage: 0.0037690258001918%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 318\n",
      "Length of fail 2508\n",
      "G013410545 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Pseudoclavibacter; s__Pseudoclavibacter chungangensis\n",
      "Coverage Percentage: 0.039146731737234%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 318\n",
      "Length of fail 2509\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G013410645 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora jinlongensis\n",
      "Coverage Percentage: 0.0132740134375211%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 318\n",
      "Length of fail 2510\n",
      "G013410775 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Sphaerotilus; s__Sphaerotilus montanus\n",
      "Coverage Percentage: 4.212220837981718%\n",
      "Adj, read counts 1346\n",
      "Number of samples per group [27, 28, 18, 20, 19, 20, 7, 13, 24, 10, 11, 17, 35, 11, 21, 1, 10, 27, 32, 18, 9, 26, 27, 12, 17, 8, 27, 24, 10, 31, 5, 3, 11, 12, 16, 8, 14, 15, 11, 33, 3, 2, 2, 6, 5]\n",
      "Number of bins 454\n",
      "total # samples 51\n",
      "Number of total reads 1329\n",
      "Number of hits in BIOM tables 1412.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 21.47102526002972 %\n",
      "Percentage of reads from samples who only fall into one bin group 0.3009781790820165 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 319\n",
      "Length of fail 2510\n",
      "G013415235 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__; g__Gemella; s__Gemella palaticanis\n",
      "Coverage Percentage: 0.0210196755225878%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 319\n",
      "Length of fail 2511\n",
      "G013415325 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Sanguibacteraceae; g__Sanguibacter; s__Sanguibacter inulinus\n",
      "Coverage Percentage: 0.0305073513329835%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 319\n",
      "Length of fail 2512\n",
      "G013416545 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Eoetvoesia; s__Eoetvoesia caeni\n",
      "Coverage Percentage: 0.0543529926778754%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 319\n",
      "Length of fail 2513\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G013426955 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Sphaerotilus; s__Sphaerotilus montanus\n",
      "Coverage Percentage: 4.1476431563645%\n",
      "Adj, read counts 1278\n",
      "Number of samples per group [30, 9, 35, 5, 5, 3, 2, 9, 29, 20, 12, 18, 4, 10, 15, 12, 1, 19, 34, 19, 20, 20, 10, 19, 4, 20, 6, 20, 9, 28, 33, 1, 7, 21, 2, 24, 14, 19, 8, 27, 24, 11, 4, 11, 11, 11, 15, 2, 10, 8, 14]\n",
      "Number of bins 437\n",
      "total # samples 51\n",
      "Number of total reads 1267\n",
      "Number of hits in BIOM tables 1407.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.970266040688575 %\n",
      "Percentage of reads from samples who only fall into one bin group 0.15785319652722968 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 320\n",
      "Length of fail 2513\n",
      "G013426975 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Sphaerotilus; s__Sphaerotilus natans\n",
      "Coverage Percentage: 0.2883219050946651%\n",
      "Adj, read counts 86\n",
      "Number of samples per group [2, 1, 4, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 4, 1, 6, 1, 1, 3, 2, 2, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 5, 1, 1, 1, 1, 1]\n",
      "Number of bins 64\n",
      "total # samples 33\n",
      "Number of total reads 85\n",
      "Number of hits in BIOM tables 41.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 23.25581395348837 %\n",
      "Percentage of reads from samples who only fall into one bin group 20.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 321\n",
      "Length of fail 2513\n",
      "G013432195 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Tepidicella; s__Tepidicella baoligensis\n",
      "Coverage Percentage: 0.1374032899585008%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 321\n",
      "Length of fail 2514\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G013449735 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Lactococcus; s__Lactococcus laudensis\n",
      "Coverage Percentage: 0.0541816010597834%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 321\n",
      "Length of fail 2515\n",
      "G013461495 d__Eukaryota; p__Ascomycota; c__Leotiomycetes; o__Helotiales; f__Sclerotiniaceae; g__Botrytis; s__Botrytis fragariae\n",
      "Coverage Percentage: 0.001707839553626%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 321\n",
      "Length of fail 2516\n",
      "G013488205 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Empedobacter; s__Empedobacter falsenii\n",
      "Coverage Percentage: 0.0114383727079363%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 321\n",
      "Length of fail 2517\n",
      "G013601005 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium wankanglinii\n",
      "Coverage Percentage: 0.0494687309537805%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 321\n",
      "Length of fail 2518\n",
      "G013607875 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas ursincola\n",
      "Coverage Percentage: 0.0757501935123268%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 321\n",
      "Length of fail 2519\n",
      "G013626205 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Hyphomicrobiaceae; g__Hyphomicrobium; s__Hyphomicrobium methylovorum\n",
      "Coverage Percentage: 0.0031345551326163%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 321\n",
      "Length of fail 2520\n",
      "G013694245 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora ferruginea\n",
      "Coverage Percentage: 0.010775166371729%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 321\n",
      "Length of fail 2521\n",
      "G013752735 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium japonicum\n",
      "Coverage Percentage: 0.0349546207732547%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 321\n",
      "Length of fail 2522\n",
      "G013778505 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas sivasensis\n",
      "Coverage Percentage: 0.0372828846693685%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 321\n",
      "Length of fail 2523\n",
      "G013778535 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas sivasensis\n",
      "Coverage Percentage: 0.0547979316165794%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 321\n",
      "Length of fail 2524\n",
      "G013868055 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Thermoactinomycetaceae; g__Thermoactinomyces; s__Thermoactinomyces mirandus\n",
      "Coverage Percentage: 0.0398915631991503%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 321\n",
      "Length of fail 2525\n",
      "G013892435 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Escherichia; s__Escherichia fergusonii\n",
      "Coverage Percentage: 0.3223545109566674%\n",
      "Adj, read counts 93\n",
      "Number of samples per group [2, 2, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 2, 3, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 3, 2, 1, 3, 1, 1, 2, 2, 1, 1, 1, 1, 2, 1, 1, 5, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1]\n",
      "Number of bins 74\n",
      "total # samples 39\n",
      "Number of total reads 92\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.827956989247312 %\n",
      "Percentage of reads from samples who only fall into one bin group 30.434782608695656 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 322\n",
      "Length of fail 2525\n",
      "G013912435 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces himalayensis\n",
      "Coverage Percentage: 0.0016644486877475%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 322\n",
      "Length of fail 2526\n",
      "G013978595 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium haemomassiliense\n",
      "Coverage Percentage: 0.2995126809992134%\n",
      "Adj, read counts 45\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 3, 1, 1]\n",
      "Number of bins 39\n",
      "total # samples 25\n",
      "Number of total reads 45\n",
      "Number of hits in BIOM tables 48.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 35.55555555555556 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 323\n",
      "Length of fail 2526\n",
      "G014042035 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__Bacillus haynesii\n",
      "Coverage Percentage: 0.0264888216936367%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 323\n",
      "Length of fail 2527\n",
      "G014049385 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Aerococcaceae; g__Ruoffia; s__Ruoffia halotolerans\n",
      "Coverage Percentage: 0.0221289068452526%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 323\n",
      "Length of fail 2528\n",
      "G014055025 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Neisseriaceae; g__Neisseria; s__Neisseria bacilliformis\n",
      "Coverage Percentage: 0.011433330444913%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 323\n",
      "Length of fail 2529\n",
      "G014066305 d__Eukaryota; p__Ascomycota; c__Lecanoromycetes; o__Lecanorales; f__Parmeliaceae; g__Letharia; s__Letharia columbiana\n",
      "Coverage Percentage: 0.0132872232567464%\n",
      "Adj, read counts 75\n",
      "Number of samples per group [2, 1, 3, 1, 2, 2, 2, 3, 1, 2, 1, 1, 1, 1, 3, 1, 1, 2, 2, 5, 14, 1, 1, 2, 2, 1, 1, 1, 4, 1, 1, 1, 1, 1, 2, 2]\n",
      "Number of bins 38\n",
      "total # samples 34\n",
      "Number of total reads 74\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 30.66666666666667 %\n",
      "Percentage of reads from samples who only fall into one bin group 24.324324324324326 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 324\n",
      "Length of fail 2529\n",
      "G014066315 d__Eukaryota; p__Ascomycota; c__Lecanoromycetes; o__Lecanorales; f__Parmeliaceae; g__Letharia; s__Letharia lupina\n",
      "Coverage Percentage: 0.0072300049592553%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 324\n",
      "Length of fail 2530\n",
      "G014070455 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Gordoniaceae; g__Gordonia; s__Gordonia rubripertincta\n",
      "Coverage Percentage: 0.0143448934174419%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 324\n",
      "Length of fail 2531\n",
      "G014116795 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter baumannii\n",
      "Coverage Percentage: 0.0115164216579597%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 324\n",
      "Length of fail 2532\n",
      "G014117215 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas tumulicola\n",
      "Coverage Percentage: 0.0255756893962822%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 324\n",
      "Length of fail 2533\n",
      "G014131695 d__Bacteria; p__Firmicutes; c__Erysipelotrichia; o__Erysipelotrichales; f__Erysipelotrichaceae; g__Erysipelatoclostridium; s__Erysipelatoclostridium ramosum\n",
      "Coverage Percentage: 0.0562260669712194%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 324\n",
      "Length of fail 2534\n",
      "G014131735 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactiplantibacillus; s__Lactiplantibacillus plantarum\n",
      "Coverage Percentage: 0.0123685685750978%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 324\n",
      "Length of fail 2535\n",
      "G014131755 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides thetaiotaomicron\n",
      "Coverage Percentage: 0.1054532897286306%\n",
      "Adj, read counts 43\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2]\n",
      "Number of bins 40\n",
      "total # samples 23\n",
      "Number of total reads 41\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 13.953488372093023 %\n",
      "Percentage of reads from samples who only fall into one bin group 31.70731707317073 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 325\n",
      "Length of fail 2535\n",
      "G014137855 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Microlunatus; s__Microlunatus kandeliicorticis\n",
      "Coverage Percentage: 0.0070186039485457%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 325\n",
      "Length of fail 2536\n",
      "G014137965 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Frigoribacterium; s__Frigoribacterium faeni\n",
      "Coverage Percentage: 0.0208574522036249%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 325\n",
      "Length of fail 2537\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G014138325 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Curtobacterium; s__Curtobacterium pusillum\n",
      "Coverage Percentage: 0.0164303913059758%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 325\n",
      "Length of fail 2538\n",
      "G014138435 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium fujisawaense\n",
      "Coverage Percentage: 0.0670613097768692%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 325\n",
      "Length of fail 2539\n",
      "G014138565 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium leguminosarum\n",
      "Coverage Percentage: 0.0188691054143834%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 325\n",
      "Length of fail 2540\n",
      "G014138625 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Phyllobacteriaceae; g__Aminobacter; s__Aminobacter ciceronei\n",
      "Coverage Percentage: 0.0017117582147424%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 325\n",
      "Length of fail 2541\n",
      "G014138635 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Phyllobacteriaceae; g__Aminobacter; s__Aminobacter ciceronei\n",
      "Coverage Percentage: 0.0017113864738504%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 325\n",
      "Length of fail 2542\n",
      "G014138885 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Micrococcus; s__Micrococcus yunnanensis\n",
      "Coverage Percentage: 0.8220609938463938%\n",
      "Adj, read counts 118\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 3, 1, 1, 2, 1, 1, 2, 3, 2, 2, 1, 2, 2, 3, 2, 2, 1, 1, 1, 1, 7, 1, 1, 2, 5, 1, 4, 1, 3, 1, 2, 1, 1, 2, 3, 1, 1, 2, 1, 2, 2, 2]\n",
      "Number of bins 83\n",
      "total # samples 23\n",
      "Number of total reads 117\n",
      "Number of hits in BIOM tables 57.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.10169491525424 %\n",
      "Percentage of reads from samples who only fall into one bin group 12.82051282051282 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2542\n",
      "G014144975 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Dietziaceae; g__Dietzia; s__Dietzia kunjamensis\n",
      "Coverage Percentage: 0.0253472574267464%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2543\n",
      "G014145335 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Aquariibacter; s__Aquariibacter albus\n",
      "Coverage Percentage: 0.0880802843191747%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2544\n",
      "G014155795 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Erwiniaceae; g__Pantoea; s__Pantoea hericii\n",
      "Coverage Percentage: 0.0165278913776009%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2545\n",
      "G014156615 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Erwiniaceae; g__Pantoea; s__Pantoea pleuroti\n",
      "Coverage Percentage: 0.0615004135513827%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2546\n",
      "G014163495 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Rikenellaceae; g__Alistipes; s__Alistipes indistinctus\n",
      "Coverage Percentage: 0.0171533512835132%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2547\n",
      "G014164785 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas guryensis\n",
      "Coverage Percentage: 0.0065701216031655%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2548\n",
      "G014172405 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter gandavensis\n",
      "Coverage Percentage: 0.0123697816438168%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2549\n",
      "G014174215 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Gluconacetobacter; s__Gluconacetobacter tumulicola\n",
      "Coverage Percentage: 0.0290085271577505%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2550\n",
      "G014180685 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides pelophilus\n",
      "Coverage Percentage: 0.0097469871481778%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2551\n",
      "G014180735 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus cohnii\n",
      "Coverage Percentage: 0.0172956875322146%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2552\n",
      "G014191305 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Pseudoclavibacter; s__Pseudoclavibacter helvolus\n",
      "Coverage Percentage: 0.0464964022626859%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2553\n",
      "G014191495 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides soli\n",
      "Coverage Percentage: 0.0248008749445979%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2554\n",
      "G014191875 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia violacea\n",
      "Coverage Percentage: 0.0257680973669165%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2555\n",
      "G014192115 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter luti\n",
      "Coverage Percentage: 0.0724107837818713%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2556\n",
      "G014192415 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium proteolyticum\n",
      "Coverage Percentage: 0.0605907628089877%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2557\n",
      "G014192445 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium laguerreae\n",
      "Coverage Percentage: 0.0150194063302902%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2558\n",
      "G014192995 d__Bacteria; p__Actinobacteria; c__Coriobacteriia; o__Eggerthellales; f__Eggerthellaceae; g__Slackia; s__Slackia isoflavoniconvertens\n",
      "Coverage Percentage: 0.015698519422605%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2559\n",
      "G014193375 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Oceanospirillales; f__Halomonadaceae; g__Halomonas; s__Halomonas campaniensis\n",
      "Coverage Percentage: 0.0066076955255227%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2560\n",
      "G014193425 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Microlunatus; s__Microlunatus antarcticus\n",
      "Coverage Percentage: 0.0057437728120459%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2561\n",
      "G014195485 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Modestobacter; s__Modestobacter versicolor\n",
      "Coverage Percentage: 0.0150428874680991%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2562\n",
      "G014195535 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Spirosomaceae; g__Runella; s__Runella defluvii\n",
      "Coverage Percentage: 0.0037786690876841%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2563\n",
      "G014195655 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Hyphomicrobiaceae; g__Angulomicrobium; s__Angulomicrobium tetraedrale\n",
      "Coverage Percentage: 0.0023055740233359%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 326\n",
      "Length of fail 2564\n",
      "G014195675 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Erythrobacter; s__Erythrobacter ramosus\n",
      "Coverage Percentage: 0.4938072910091594%\n",
      "Adj, read counts 98\n",
      "Number of samples per group [3, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 2, 1, 2, 1, 1, 3, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 3, 1, 1, 2, 2, 1, 1, 1, 2, 4, 1, 2, 2]\n",
      "Number of bins 81\n",
      "total # samples 32\n",
      "Number of total reads 98\n",
      "Number of hits in BIOM tables 118.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.224489795918366 %\n",
      "Percentage of reads from samples who only fall into one bin group 6.122448979591836 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2564\n",
      "G014196055 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium hassiacum\n",
      "Coverage Percentage: 0.0820862465525363%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2565\n",
      "G014196115 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas aquatilis\n",
      "Coverage Percentage: 0.0852559387332137%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2566\n",
      "G014196125 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas mediterranea\n",
      "Coverage Percentage: 0.0761763184046913%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2567\n",
      "G014196135 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas desiccabilis\n",
      "Coverage Percentage: 0.0186592303303155%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2568\n",
      "G014196295 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Phenylobacterium; s__Phenylobacterium haematophilum\n",
      "Coverage Percentage: 0.0101652255765207%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2569\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G014196335 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas lenta\n",
      "Coverage Percentage: 0.0480909299494301%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2570\n",
      "G014196495 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium jiangsuense\n",
      "Coverage Percentage: 0.0115748543549352%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2571\n",
      "G014196505 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium metallidurans\n",
      "Coverage Percentage: 0.0056554808461603%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2572\n",
      "G014196515 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium skierniewicense\n",
      "Coverage Percentage: 0.0062921700402698%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2573\n",
      "G014196525 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium sediminicola\n",
      "Coverage Percentage: 0.0109234075597796%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2574\n",
      "G014196535 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium borbori\n",
      "Coverage Percentage: 0.010813552764741%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2575\n",
      "G014196595 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingorhabdus; s__Sphingorhabdus rigui\n",
      "Coverage Percentage: 0.1017877123878637%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2576\n",
      "G014196695 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Mucilaginibacter; s__Mucilaginibacter phyllosphaerae\n",
      "Coverage Percentage: 0.0034219202774157%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2577\n",
      "G014196745 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas kyeonggiensis\n",
      "Coverage Percentage: 0.0189745548863012%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2578\n",
      "G014196835 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Aurantimonadaceae; g__Aureimonas; s__Aureimonas pseudogalii\n",
      "Coverage Percentage: 0.0021386355505187%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2579\n",
      "G014196855 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Croceicoccus; s__Croceicoccus naphthovorans\n",
      "Coverage Percentage: 0.0142557584705416%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2580\n",
      "G014196975 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Odoribacteraceae; g__Butyricimonas; s__Butyricimonas faecihominis\n",
      "Coverage Percentage: 0.0340823880198528%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2581\n",
      "G014197035 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium scionense\n",
      "Coverage Percentage: 0.0494633586207555%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2582\n",
      "G014197135 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas zeae\n",
      "Coverage Percentage: 0.0553575461798477%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2583\n",
      "G014197145 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Gellertiella; s__Gellertiella hungarica\n",
      "Coverage Percentage: 0.0126500624125595%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2584\n",
      "G014197755 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Rhodocyclaceae; g__Rhodocyclus; s__Rhodocyclus tenuis\n",
      "Coverage Percentage: 0.014041888953316%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2585\n",
      "G014199215 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium boeckii\n",
      "Coverage Percentage: 0.0122526465964677%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2586\n",
      "G014199255 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas xinjiangensis\n",
      "Coverage Percentage: 0.0063304447417044%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2587\n",
      "G014199295 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingopyxis; s__Sphingopyxis panaciterrulae\n",
      "Coverage Percentage: 0.0068116249627559%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2588\n",
      "G014199415 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas endophytica\n",
      "Coverage Percentage: 0.0242638486956309%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2589\n",
      "G014199435 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium subterraneum\n",
      "Coverage Percentage: 0.0307548515578279%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2590\n",
      "G014199525 d__Bacteria; p__Actinobacteria; c__Thermoleophilia; o__Solirubrobacterales; f__Conexibacteraceae; g__Conexibacter; s__Conexibacter arvalis\n",
      "Coverage Percentage: 0.0087099622933619%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2591\n",
      "G014199625 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas abaci\n",
      "Coverage Percentage: 0.0285476259363791%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2592\n",
      "G014199635 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium taihuense\n",
      "Coverage Percentage: 0.0387816821413625%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2593\n",
      "G014199915 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Hyphomicrobiaceae; g__Prosthecomicrobium; s__Prosthecomicrobium pneumaticum\n",
      "Coverage Percentage: 0.0083168018093234%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2594\n",
      "G014199935 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylorubrum; s__Methylorubrum rhodinum\n",
      "Coverage Percentage: 0.0247263183085348%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2595\n",
      "G014199945 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas variabilis\n",
      "Coverage Percentage: 0.0259111500202706%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2596\n",
      "G014199955 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas aurantiaca\n",
      "Coverage Percentage: 0.0533594242168478%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2597\n",
      "G014199985 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylorubrum; s__Methylorubrum rhodesianum\n",
      "Coverage Percentage: 0.0666212198968341%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2598\n",
      "G014200505 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia aurea\n",
      "Coverage Percentage: 0.0837120816459962%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2599\n",
      "G014200545 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Mucilaginibacter; s__Mucilaginibacter lappiensis\n",
      "Coverage Percentage: 0.0015643962308676%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2600\n",
      "G014201135 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Neomicrococcus; s__Neomicrococcus aestuarii\n",
      "Coverage Percentage: 0.0183007874527863%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2601\n",
      "G014201415 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Chelatococcaceae; g__Chelatococcus; s__Chelatococcus composti\n",
      "Coverage Percentage: 0.0191325228166083%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2602\n",
      "G014201615 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Phyllobacteriaceae; g__Aquamicrobium; s__Aquamicrobium lusatiense\n",
      "Coverage Percentage: 0.0057633185051273%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2603\n",
      "G014201625 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Deinococcales; f__Deinococcaceae; g__Deinococcus; s__Deinococcus radiopugnans\n",
      "Coverage Percentage: 0.0097326991980395%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2604\n",
      "G014201885 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Deinococcales; f__Deinococcaceae; g__Deinococcus; s__Deinococcus budaensis\n",
      "Coverage Percentage: 0.0039423189137579%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2605\n",
      "G014201905 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Amaricoccus; s__Amaricoccus macauensis\n",
      "Coverage Percentage: 0.0031265445078104%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2606\n",
      "G014202075 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas basaltis\n",
      "Coverage Percentage: 0.0325125764385399%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2607\n",
      "G014202125 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium leguminosarum\n",
      "Coverage Percentage: 0.0144725790531158%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2608\n",
      "G014202215 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Quisquiliibacterium; s__Quisquiliibacterium transsilvanicum\n",
      "Coverage Percentage: 0.0106562238802221%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2609\n",
      "G014202415 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Tolumonas; s__Tolumonas osonensis\n",
      "Coverage Percentage: 0.031102930614292%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2610\n",
      "G014202505 d__Bacteria; p__Armatimonadetes; c__Armatimonadia; o__Armatimonadales; f__Armatimonadaceae; g__Armatimonas; s__Armatimonas rosea\n",
      "Coverage Percentage: 0.0023868672351731%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2611\n",
      "G014202935 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Rehaibacterium; s__Rehaibacterium terrae\n",
      "Coverage Percentage: 0.0077013160271293%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2612\n",
      "G014202975 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Nevskiales; f__Steroidobacteraceae; g__Povalibacter; s__Povalibacter uvarum\n",
      "Coverage Percentage: 0.006168933881683%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2613\n",
      "G014203045 d__Bacteria; p__Verrucomicrobia; c__Verrucomicrobiae; o__Verrucomicrobiales; f__Verrucomicrobiaceae; g__Prosthecobacter; s__Prosthecobacter dejongeii\n",
      "Coverage Percentage: 0.0095391331262741%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2614\n",
      "G014203095 d__Bacteria; p__Verrucomicrobia; c__Verrucomicrobiae; o__Verrucomicrobiales; f__Verrucomicrobiaceae; g__Prosthecobacter; s__Prosthecobacter vanneervenii\n",
      "Coverage Percentage: 0.0170634865250232%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2615\n",
      "G014203115 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Afipia; s__Afipia massiliensis\n",
      "Coverage Percentage: 0.028611599600564%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2616\n",
      "G014203425 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora echinospora\n",
      "Coverage Percentage: 0.0297227031774102%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2617\n",
      "G014203795 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptosporangiales; f__Streptosporangiaceae; g__Streptosporangium; s__Streptosporangium album\n",
      "Coverage Percentage: 0.0016735537281678%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 327\n",
      "Length of fail 2618\n",
      "G014204325 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Spirosomaceae; g__Arcicella; s__Arcicella rosea\n",
      "Coverage Percentage: 0.346527326007078%\n",
      "Adj, read counts 119\n",
      "Number of samples per group [2, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 3, 1, 1, 1, 1, 2, 1, 1, 1, 4, 2, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 1, 1, 3, 2, 2, 1, 1, 1, 1, 1, 3, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 5]\n",
      "Number of bins 96\n",
      "total # samples 39\n",
      "Number of total reads 116\n",
      "Number of hits in BIOM tables 208.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.92436974789916 %\n",
      "Percentage of reads from samples who only fall into one bin group 7.758620689655173 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 328\n",
      "Length of fail 2618\n",
      "G014204535 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium chloroacetimidivorans\n",
      "Coverage Percentage: 0.0204523611644326%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 328\n",
      "Length of fail 2619\n",
      "G014204755 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Dermabacteraceae; g__Brachybacterium; s__Brachybacterium aquaticum\n",
      "Coverage Percentage: 0.0279763989359111%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 328\n",
      "Length of fail 2620\n",
      "G014204815 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Micrococcus; s__Micrococcus flavus\n",
      "Coverage Percentage: 0.0948066996734435%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 328\n",
      "Length of fail 2621\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G014204835 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium marinum\n",
      "Coverage Percentage: 0.0300543955753784%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 328\n",
      "Length of fail 2622\n",
      "G014205075 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium ginsengiterrae\n",
      "Coverage Percentage: 0.0302866499738942%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 328\n",
      "Length of fail 2623\n",
      "G014205115 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Micrococcus; s__Micrococcus endophyticus\n",
      "Coverage Percentage: 0.1472158100410699%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 328\n",
      "Length of fail 2624\n",
      "G014205175 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Micrococcus; s__Micrococcus cohnii\n",
      "Coverage Percentage: 0.0599843117953765%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 328\n",
      "Length of fail 2625\n",
      "G014205335 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes digitatis\n",
      "Coverage Percentage: 0.0112723182864498%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 328\n",
      "Length of fail 2626\n",
      "G014205735 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium wenxiniae\n",
      "Coverage Percentage: 0.0537311644348581%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 328\n",
      "Length of fail 2627\n",
      "G014207075 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium soli\n",
      "Coverage Percentage: 0.0704052097351165%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 328\n",
      "Length of fail 2628\n",
      "G014207505 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Leuconostoc; s__Leuconostoc carnosum\n",
      "Coverage Percentage: 0.0165846586415956%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 328\n",
      "Length of fail 2629\n",
      "G014207855 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Comamonas; s__Comamonas odontotermitis\n",
      "Coverage Percentage: 0.1087026834891294%\n",
      "Adj, read counts 30\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 3]\n",
      "Number of bins 29\n",
      "total # samples 19\n",
      "Number of total reads 30\n",
      "Number of hits in BIOM tables 49.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 23.333333333333332 %\n",
      "Percentage of reads from samples who only fall into one bin group 40.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2629\n",
      "G014218725 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas huaxiensis\n",
      "Coverage Percentage: 0.1061811994809762%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2630\n",
      "G014230305 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium aerophilum\n",
      "Coverage Percentage: 0.0301604205087996%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2631\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G014230315 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium flavum\n",
      "Coverage Percentage: 0.0294728596370892%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2632\n",
      "G014230345 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium aerophilum\n",
      "Coverage Percentage: 0.0286437311655631%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2633\n",
      "G014230355 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium piscinae\n",
      "Coverage Percentage: 0.0340049036977949%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2634\n",
      "G014230465 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas cremoris\n",
      "Coverage Percentage: 0.0407436921344046%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2635\n",
      "G014235765 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas baltica\n",
      "Coverage Percentage: 0.0122628304227294%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2636\n",
      "G014235925 d__Eukaryota; p__Ascomycota; c__Sordariomycetes; o__Glomerellales; f__Glomerellaceae; g__Colletotrichum; s__Colletotrichum truncatum\n",
      "Coverage Percentage: 0.0005935298191519%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2637\n",
      "G014236655 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas kielensis\n",
      "Coverage Percentage: 0.0119846596356663%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2638\n",
      "G014268275 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas tritici\n",
      "Coverage Percentage: 0.0578667814066707%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2639\n",
      "G014268375 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas salmasensis\n",
      "Coverage Percentage: 0.0811727547027725%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2640\n",
      "G014268485 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas zeae\n",
      "Coverage Percentage: 0.031987384814875%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2641\n",
      "G014268495 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas monsensis\n",
      "Coverage Percentage: 0.0188859970841141%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2642\n",
      "G014268585 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas iranensis\n",
      "Coverage Percentage: 0.0177656625143203%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2643\n",
      "G014269185 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas oryzicola\n",
      "Coverage Percentage: 0.0207583027021406%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2644\n",
      "G014269205 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas kermanshahensis\n",
      "Coverage Percentage: 0.0116635130175866%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2645\n",
      "G014269225 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas anuradhapurensis\n",
      "Coverage Percentage: 0.0244362059286166%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2646\n",
      "G014269245 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas kurunegalensis\n",
      "Coverage Percentage: 0.0630704342429218%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2647\n",
      "G014284125 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Undibacterium; s__Undibacterium umbellatum\n",
      "Coverage Percentage: 0.0168401344981918%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2648\n",
      "G014284155 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Undibacterium; s__Undibacterium nitidum\n",
      "Coverage Percentage: 0.0165319559287698%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2649\n",
      "G014284175 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Undibacterium; s__Undibacterium flavidum\n",
      "Coverage Percentage: 0.0067058362226082%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2650\n",
      "G014284275 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Undibacterium; s__Undibacterium amnicola\n",
      "Coverage Percentage: 0.0446495371760915%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 329\n",
      "Length of fail 2651\n",
      "G014284305 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Undibacterium; s__Undibacterium seohonense\n",
      "Coverage Percentage: 0.1640176145475207%\n",
      "Adj, read counts 52\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 5, 2, 2, 1, 1, 1, 1]\n",
      "Number of bins 47\n",
      "total # samples 28\n",
      "Number of total reads 52\n",
      "Number of hits in BIOM tables 83.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.30769230769231 %\n",
      "Percentage of reads from samples who only fall into one bin group 34.61538461538461 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 330\n",
      "Length of fail 2651\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G014284335 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Undibacterium; s__Undibacterium hunanense\n",
      "Coverage Percentage: 0.0093954535241151%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 330\n",
      "Length of fail 2652\n",
      "G014284475 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Eubacteriaceae; g__Acetobacterium; s__Acetobacterium fimetarium\n",
      "Coverage Percentage: 0.0175361304490515%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 330\n",
      "Length of fail 2653\n",
      "G014284495 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Eubacteriaceae; g__Acetobacterium; s__Acetobacterium malicum\n",
      "Coverage Percentage: 0.0115480037767611%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 330\n",
      "Length of fail 2654\n",
      "G014295435 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacterium; s__Mycolicibacterium fluoranthenivorans\n",
      "Coverage Percentage: 0.0088654512521712%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 330\n",
      "Length of fail 2655\n",
      "G014297355 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Blautia; s__Blautia intestinalis\n",
      "Coverage Percentage: 0.0903303460576804%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 330\n",
      "Length of fail 2656\n",
      "G014305095 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium kayseriense\n",
      "Coverage Percentage: 0.054574816375573%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 330\n",
      "Length of fail 2657\n",
      "G014305215 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium turcicum\n",
      "Coverage Percentage: 0.0357805459846276%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 330\n",
      "Length of fail 2658\n",
      "G014334155 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Salmonella; s__Salmonella enterica\n",
      "Coverage Percentage: 0.0244160226751973%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 330\n",
      "Length of fail 2659\n",
      "G014335175 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium amycolatum\n",
      "Coverage Percentage: 0.0586155950930955%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 330\n",
      "Length of fail 2660\n",
      "G014357575 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas foliumensis\n",
      "Coverage Percentage: 0.0349705891815374%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 330\n",
      "Length of fail 2661\n",
      "G014358015 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas triticumensis\n",
      "Coverage Percentage: 0.0222793668652074%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 330\n",
      "Length of fail 2662\n",
      "G014358075 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas albertensis\n",
      "Coverage Percentage: 0.3923641975430304%\n",
      "Adj, read counts 102\n",
      "Number of samples per group [1, 1, 1, 2, 3, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 2, 1, 1, 2, 3, 4, 3, 1, 1, 1, 1, 1, 2, 2, 1, 3, 1, 1, 4, 2, 1, 3, 2, 1, 2, 1, 1, 1, 5, 2, 10, 1, 1, 1]\n",
      "Number of bins 79\n",
      "total # samples 37\n",
      "Number of total reads 100\n",
      "Number of hits in BIOM tables 97.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.58823529411765 %\n",
      "Percentage of reads from samples who only fall into one bin group 15.0 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 331\n",
      "Length of fail 2662\n",
      "G014385295 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium faecale\n",
      "Coverage Percentage: 0.5214046969382256%\n",
      "Adj, read counts 69\n",
      "Number of samples per group [3, 2, 1, 2, 3, 3, 1, 3, 1, 4, 1, 1, 2, 2, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 57\n",
      "total # samples 30\n",
      "Number of total reads 69\n",
      "Number of hits in BIOM tables 29.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.942028985507244 %\n",
      "Percentage of reads from samples who only fall into one bin group 24.637681159420293 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 332\n",
      "Length of fail 2662\n",
      "G014385405 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Blautia; s__Blautia stercoris\n",
      "Coverage Percentage: 0.0838900462446069%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 332\n",
      "Length of fail 2663\n",
      "G014385475 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium macrobrachii\n",
      "Coverage Percentage: 0.0473665864573728%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 332\n",
      "Length of fail 2664\n",
      "G014395225 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter roseus\n",
      "Coverage Percentage: 0.0550732999396816%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 332\n",
      "Length of fail 2665\n",
      "G014396585 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas rhizophila\n",
      "Coverage Percentage: 0.0118084851394906%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 332\n",
      "Length of fail 2666\n",
      "G014396975 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Thermomonas; s__Thermomonas carbonis\n",
      "Coverage Percentage: 0.0701259431852568%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 332\n",
      "Length of fail 2667\n",
      "G014397415 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Pseudoxanthomonas; s__Pseudoxanthomonas mexicana\n",
      "Coverage Percentage: 0.0430316462844051%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 332\n",
      "Length of fail 2668\n",
      "G014450905 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas cerasi\n",
      "Coverage Percentage: 0.0253178193439354%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 332\n",
      "Length of fail 2669\n",
      "G014486675 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium macrobrachii\n",
      "Coverage Percentage: 0.0486726098282275%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 332\n",
      "Length of fail 2670\n",
      "G014486695 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium buctense\n",
      "Coverage Percentage: 0.0206767740946037%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 332\n",
      "Length of fail 2671\n",
      "G014489535 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Diaphorobacter; s__Diaphorobacter aerolatus\n",
      "Coverage Percentage: 0.2326709832627654%\n",
      "Adj, read counts 72\n",
      "Number of samples per group [1, 5, 1, 1, 1, 5, 1, 1, 2, 2, 1, 3, 1, 15, 4, 3, 2, 1, 1, 6]\n",
      "Number of bins 30\n",
      "total # samples 30\n",
      "Number of total reads 72\n",
      "Number of hits in BIOM tables 36.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 48.61111111111111 %\n",
      "Percentage of reads from samples who only fall into one bin group 27.77777777777778 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 333\n",
      "Length of fail 2671\n",
      "G014489575 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Tessaracoccus; s__Tessaracoccus defluvii\n",
      "Coverage Percentage: 0.0300404264964324%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 333\n",
      "Length of fail 2672\n",
      "G014489595 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax monticola\n",
      "Coverage Percentage: 0.0872437761619148%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 333\n",
      "Length of fail 2673\n",
      "G014490555 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium lujinxingii\n",
      "Coverage Percentage: 0.0129142833684556%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 333\n",
      "Length of fail 2674\n",
      "G014490665 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas alpina\n",
      "Coverage Percentage: 0.0143339246478087%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 333\n",
      "Length of fail 2675\n",
      "G014529705 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium campsiandrae\n",
      "Coverage Percentage: 0.0493599751847645%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 333\n",
      "Length of fail 2676\n",
      "G014530645 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium campsiandrae\n",
      "Coverage Percentage: 0.0203601275014957%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 333\n",
      "Length of fail 2677\n",
      "G014623485 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium manosquense\n",
      "Coverage Percentage: 0.0255072235370197%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 333\n",
      "Length of fail 2678\n",
      "G014635185 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium murale\n",
      "Coverage Percentage: 0.0161625076002267%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 333\n",
      "Length of fail 2679\n",
      "G014635205 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Jeotgalicoccus; s__Jeotgalicoccus aerolatus\n",
      "Coverage Percentage: 0.0143118330425156%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 333\n",
      "Length of fail 2680\n",
      "G014635305 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Williamsia; s__Williamsia phyllosphaerae\n",
      "Coverage Percentage: 0.0062790144525447%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 333\n",
      "Length of fail 2681\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G014635525 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Mucilaginibacter; s__Mucilaginibacter phyllosphaerae\n",
      "Coverage Percentage: 0.0034154890392206%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 333\n",
      "Length of fail 2682\n",
      "G014635765 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus capitis\n",
      "Coverage Percentage: 0.1071084264940637%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 333\n",
      "Length of fail 2683\n",
      "G014635805 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Agromyces; s__Agromyces flavus\n",
      "Coverage Percentage: 0.0635351303077969%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 333\n",
      "Length of fail 2684\n",
      "G014635825 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Mucilaginibacter; s__Mucilaginibacter galii\n",
      "Coverage Percentage: 0.002243505565952%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 333\n",
      "Length of fail 2685\n",
      "G014636095 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter modestus\n",
      "Coverage Percentage: 0.1775763228561987%\n",
      "Adj, read counts 39\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 3, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 7]\n",
      "Number of bins 28\n",
      "total # samples 24\n",
      "Number of total reads 37\n",
      "Number of hits in BIOM tables 15.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 38.46153846153847 %\n",
      "Percentage of reads from samples who only fall into one bin group 40.54054054054054 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2685\n",
      "G014636175 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas psychrolutea\n",
      "Coverage Percentage: 0.0321549323826564%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2686\n",
      "G014637085 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Comamonas; s__Comamonas phosphati\n",
      "Coverage Percentage: 0.0629998215696646%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2687\n",
      "G014638745 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Arenimonas; s__Arenimonas maotaiensis\n",
      "Coverage Percentage: 0.0277160708328378%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2688\n",
      "G014639315 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Brevibacteriaceae; g__Sediminivirga; s__Sediminivirga luteola\n",
      "Coverage Percentage: 0.0117257661675148%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2689\n",
      "G014639655 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Conyzicola; s__Conyzicola nivalis\n",
      "Coverage Percentage: 0.0271560917767997%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2690\n",
      "G014640515 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium guangdongense\n",
      "Coverage Percentage: 0.0169237354613527%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2691\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G014640675 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium endophyticum\n",
      "Coverage Percentage: 0.0232090823188656%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2692\n",
      "G014640975 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium faecale\n",
      "Coverage Percentage: 0.0116339963963455%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2693\n",
      "G014641535 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Chelatococcaceae; g__Chelatococcus; s__Chelatococcus composti\n",
      "Coverage Percentage: 0.0190765196176015%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2694\n",
      "G014641735 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas metalli\n",
      "Coverage Percentage: 0.0200198019699277%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2695\n",
      "G014642175 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter quisquiliarum\n",
      "Coverage Percentage: 0.0230501005858699%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2696\n",
      "G014642595 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Croceicoccus; s__Croceicoccus mobilis\n",
      "Coverage Percentage: 0.0287312214873592%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2697\n",
      "G014643375 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Aurantimonadaceae; g__Aureimonas; s__Aureimonas glaciei\n",
      "Coverage Percentage: 0.0053514844344455%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2698\n",
      "G014643495 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter psychrotolerans\n",
      "Coverage Percentage: 0.00581513735911%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2699\n",
      "G014644295 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Spirosomaceae; g__Emticicia; s__Emticicia aquatilis\n",
      "Coverage Percentage: 0.0319344796354342%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2700\n",
      "G014644395 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium limi\n",
      "Coverage Percentage: 0.0312005913712087%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2701\n",
      "G014644515 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Pseudarthrobacter; s__Pseudarthrobacter scleromae\n",
      "Coverage Percentage: 0.0054996032429089%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2702\n",
      "G014644535 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Roseomonas; s__Roseomonas lacus\n",
      "Coverage Percentage: 0.0047157194134457%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2703\n",
      "G014644555 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Glutamicibacter; s__Glutamicibacter ardleyensis\n",
      "Coverage Percentage: 0.0474494642884358%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2704\n",
      "G014644875 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Micrococcus; s__Micrococcus flavus\n",
      "Coverage Percentage: 0.0842646190703273%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2705\n",
      "G014645135 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Chromatiales; f__Chromatiaceae; g__Rheinheimera; s__Rheinheimera tangshanensis\n",
      "Coverage Percentage: 0.0280218280969942%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2706\n",
      "G014645195 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium indicum\n",
      "Coverage Percentage: 0.024175553197096%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2707\n",
      "G014645235 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Agrococcus; s__Agrococcus terreus\n",
      "Coverage Percentage: 0.0225731162310596%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2708\n",
      "G014645395 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Stakelama; s__Stakelama pacifica\n",
      "Coverage Percentage: 0.0278586308877106%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2709\n",
      "G014645535 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Chromobacteriaceae; g__Silvimonas; s__Silvimonas iriomotensis\n",
      "Coverage Percentage: 0.0161036104992053%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2710\n",
      "G014645635 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingopyxis; s__Sphingopyxis bauzanensis\n",
      "Coverage Percentage: 0.0285952095225356%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2711\n",
      "G014645675 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Luteimonas; s__Luteimonas terricola\n",
      "Coverage Percentage: 0.0121747891704921%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2712\n",
      "G014645755 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Modestobacter; s__Modestobacter marinus\n",
      "Coverage Percentage: 0.0174149757833999%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2713\n",
      "G014646015 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium nanhaiense\n",
      "Coverage Percentage: 0.0170348955661969%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2714\n",
      "G014646095 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces lacrimifluminis\n",
      "Coverage Percentage: 0.0078223671169846%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2715\n",
      "G014646175 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Intrasporangiaceae; g__Phycicoccus; s__Phycicoccus endophyticus\n",
      "Coverage Percentage: 0.0183547559263264%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2716\n",
      "G014646715 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas brenneri\n",
      "Coverage Percentage: 0.028528225823289%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2717\n",
      "G014646775 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium saperdae\n",
      "Coverage Percentage: 0.0156693104607351%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 334\n",
      "Length of fail 2718\n",
      "G014646835 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium lacticum\n",
      "Coverage Percentage: 0.1654533306361283%\n",
      "Adj, read counts 31\n",
      "Number of samples per group [1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1]\n",
      "Number of bins 28\n",
      "total # samples 22\n",
      "Number of total reads 30\n",
      "Number of hits in BIOM tables 22.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 29.032258064516125 %\n",
      "Percentage of reads from samples who only fall into one bin group 63.33333333333333 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2718\n",
      "G014647135 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Alteromonadales; f__Shewanellaceae; g__Shewanella; s__Shewanella xiamenensis\n",
      "Coverage Percentage: 0.0065623456517832%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2719\n",
      "G014647335 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Odoribacteraceae; g__Butyricimonas; s__Butyricimonas faecihominis\n",
      "Coverage Percentage: 0.0313048447427778%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2720\n",
      "G014647515 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Dactylosporangium; s__Dactylosporangium sucinum\n",
      "Coverage Percentage: 0.0024942881214979%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2721\n",
      "G014647535 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Thermales; f__Thermaceae; g__Thermus; s__Thermus composti\n",
      "Coverage Percentage: 0.032807674017571%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2722\n",
      "G014647595 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Paenarthrobacter; s__Paenarthrobacter histidinolovorans\n",
      "Coverage Percentage: 0.0211034708332625%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2723\n",
      "G014647675 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces flaveus\n",
      "Coverage Percentage: 0.0051956639362836%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2724\n",
      "G014647755 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Porphyromonadaceae; g__Porphyromonas; s__Porphyromonas pasteri\n",
      "Coverage Percentage: 0.0201665470671979%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2725\n",
      "G014647795 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes campanulatus\n",
      "Coverage Percentage: 0.0175684186178334%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2726\n",
      "G014647975 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces fumigatiscleroticus\n",
      "Coverage Percentage: 0.015633498374397%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2727\n",
      "G014648015 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Butyricicoccus; s__Butyricicoccus faecihominis\n",
      "Coverage Percentage: 0.0556952107124348%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2728\n",
      "G014648115 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Deinococcales; f__Deinococcaceae; g__Deinococcus; s__Deinococcus seoulensis\n",
      "Coverage Percentage: 0.0083219999358465%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2729\n",
      "G014648155 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Deinococcales; f__Deinococcaceae; g__Deinococcus; s__Deinococcus knuensis\n",
      "Coverage Percentage: 0.0086227153000205%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2730\n",
      "G014648255 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides albus\n",
      "Coverage Percentage: 0.008009492574776%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2731\n",
      "G014648375 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes ianthinogenes\n",
      "Coverage Percentage: 0.0305008084444429%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2732\n",
      "G014648495 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptosporangiales; f__Thermomonosporaceae; g__Actinomadura; s__Actinomadura cremea\n",
      "Coverage Percentage: 0.0142898587299173%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2733\n",
      "G014648575 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Agromyces; s__Agromyces mediolanus\n",
      "Coverage Percentage: 0.0078201854523316%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2734\n",
      "G014648735 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Paenarthrobacter; s__Paenarthrobacter nicotinovorans\n",
      "Coverage Percentage: 0.072503363656671%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2735\n",
      "G014648755 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium liquefaciens\n",
      "Coverage Percentage: 0.0823585181212681%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2736\n",
      "G014649515 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces pilosus\n",
      "Coverage Percentage: 0.0117555086155226%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2737\n",
      "G014649895 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces rubiginosus\n",
      "Coverage Percentage: 0.005635639235375%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2738\n",
      "G014650275 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces mirabilis\n",
      "Coverage Percentage: 0.0104009633382875%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2739\n",
      "G014650775 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces cirratus\n",
      "Coverage Percentage: 0.0041537526679608%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2740\n",
      "G014650855 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces chryseus\n",
      "Coverage Percentage: 0.027337420657408%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2741\n",
      "G014650875 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces minutiscleroticus\n",
      "Coverage Percentage: 0.0013765074727018%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2742\n",
      "G014650935 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces subrutilus\n",
      "Coverage Percentage: 0.0091082838131101%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2743\n",
      "G014651095 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces geysiriensis\n",
      "Coverage Percentage: 0.0655812530725399%\n",
      "Adj, read counts 57\n",
      "Number of samples per group [1, 11, 1, 4]\n",
      "Number of bins 6\n",
      "total # samples 12\n",
      "Number of total reads 56\n",
      "Number of hits in BIOM tables 85.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 98.24561403508771 %\n",
      "Percentage of reads from samples who only fall into one bin group 33.92857142857143 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Fail - too many reads in top 3 bins\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2744\n",
      "G014651435 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia plicata\n",
      "Coverage Percentage: 0.0204927812552974%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2745\n",
      "G014651895 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Rhodanobacteraceae; g__Rhodanobacter; s__Rhodanobacter panaciterrae\n",
      "Coverage Percentage: 0.0161400656883318%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2746\n",
      "G014652195 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Gemmobacter; s__Gemmobacter nanjingensis\n",
      "Coverage Percentage: 0.0164083766310661%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 335\n",
      "Length of fail 2747\n",
      "G014652235 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Pseudorhodoferax; s__Pseudorhodoferax aquiterrae\n",
      "Coverage Percentage: 0.0825154509198182%\n",
      "Adj, read counts 36\n",
      "Number of samples per group [5, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 3, 2, 3]\n",
      "Number of bins 22\n",
      "total # samples 21\n",
      "Number of total reads 36\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 38.88888888888889 %\n",
      "Percentage of reads from samples who only fall into one bin group 44.44444444444444 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2747\n",
      "G014652395 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Advenella; s__Advenella faeciporci\n",
      "Coverage Percentage: 0.0145878355524916%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2748\n",
      "G014652535 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Cellvibrionales; f__Cellvibrionaceae; g__Cellvibrio; s__Cellvibrio zantedeschiae\n",
      "Coverage Percentage: 0.0293614653946799%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2749\n",
      "G014652615 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium arvoryzae\n",
      "Coverage Percentage: 0.0306755840914229%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2750\n",
      "G014652775 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Thermomonas; s__Thermomonas carbonis\n",
      "Coverage Percentage: 0.0703720173476015%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2751\n",
      "G014652935 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Luteimonas; s__Luteimonas padinae\n",
      "Coverage Percentage: 0.0456372489750861%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2752\n",
      "G014653115 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides flavus Wang et al. 2016\n",
      "Coverage Percentage: 0.0201108944852388%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2753\n",
      "G014653215 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Pseudolysinimonas; s__Pseudolysinimonas yzui\n",
      "Coverage Percentage: 0.0257284029223827%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2754\n",
      "G014653575 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas glacialis\n",
      "Coverage Percentage: 0.0838606678484684%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2755\n",
      "G014654935 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces capoamus\n",
      "Coverage Percentage: 0.0268508857819446%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2756\n",
      "G014655475 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces viridodiastaticus\n",
      "Coverage Percentage: 0.0092358032510027%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2757\n",
      "G014656335 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Vulcaniibacterium; s__Vulcaniibacterium thermophilum\n",
      "Coverage Percentage: 0.0206296684009297%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2758\n",
      "G014656355 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Chelatococcaceae; g__Camelimonas; s__Camelimonas fluminis\n",
      "Coverage Percentage: 0.0028976374084605%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2759\n",
      "G014672755 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter baumannii\n",
      "Coverage Percentage: 0.0247183266295757%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2760\n",
      "G014692505 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Sphingobacterium; s__Sphingobacterium arenae\n",
      "Coverage Percentage: 0.011122468764733%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2761\n",
      "G014698445 d__Bacteria; p__Cyanobacteria; c__; o__Chroococcales; f__Microcystaceae; g__Microcystis; s__Microcystis elabens\n",
      "Coverage Percentage: 0.0108677253823605%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2762\n",
      "G014699135 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Undibacterium; s__Undibacterium aquatile\n",
      "Coverage Percentage: 0.0744441189402564%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2763\n",
      "G014705925 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Rothia; s__Rothia terrae\n",
      "Coverage Percentage: 0.024096847523344%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2764\n",
      "G014705945 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Rothia; s__Rothia amarae\n",
      "Coverage Percentage: 0.0258057763069961%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2765\n",
      "G014725695 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium hominis\n",
      "Coverage Percentage: 0.0923276287192152%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2766\n",
      "G014773275 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Mucilaginibacter; s__Mucilaginibacter rigui\n",
      "Coverage Percentage: 0.0096406111807731%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2767\n",
      "G014773485 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Erwiniaceae; g__Erwinia; s__Erwinia aphidicola\n",
      "Coverage Percentage: 0.0029500697759881%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2768\n",
      "G014775655 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Clavibacter; s__Clavibacter zhangzhiyongii\n",
      "Coverage Percentage: 0.0610327608109643%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2769\n",
      "G014779665 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides hwasunensis\n",
      "Coverage Percentage: 0.0124245277830772%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2770\n",
      "G014779675 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides cavernae\n",
      "Coverage Percentage: 0.0377438344518295%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2771\n",
      "G014779795 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium helvum\n",
      "Coverage Percentage: 0.0103652669742687%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 336\n",
      "Length of fail 2772\n",
      "G014836715 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Escherichia; s__Escherichia whittamii\n",
      "Coverage Percentage: 0.1828375705389149%\n",
      "Adj, read counts 53\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1]\n",
      "Number of bins 50\n",
      "total # samples 22\n",
      "Number of total reads 53\n",
      "Number of hits in BIOM tables 11.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.320754716981133 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.867924528301888 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 337\n",
      "Length of fail 2772\n",
      "G014837235 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Limnohabitans; s__Limnohabitans radicicola\n",
      "Coverage Percentage: 0.1013113021643995%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 337\n",
      "Length of fail 2773\n",
      "G014844295 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium columnare\n",
      "Coverage Percentage: 0.0265733041358119%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 337\n",
      "Length of fail 2774\n",
      "G014845095 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium bullatum\n",
      "Coverage Percentage: 0.0205775331907609%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 337\n",
      "Length of fail 2775\n",
      "G014845115 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylorubrum; s__Methylorubrum zatmanii\n",
      "Coverage Percentage: 0.0165748652098735%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 337\n",
      "Length of fail 2776\n",
      "G014858745 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Alteromonadales; f__Pseudoalteromonadaceae; g__Pseudoalteromonas; s__Pseudoalteromonas peptidolytica\n",
      "Coverage Percentage: 0.0099727071887787%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 337\n",
      "Length of fail 2777\n",
      "G014863585 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas gozinkensis\n",
      "Coverage Percentage: 0.0115029617460246%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 337\n",
      "Length of fail 2778\n",
      "G014863605 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas allokribbensis\n",
      "Coverage Percentage: 0.0115003335096717%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 337\n",
      "Length of fail 2779\n",
      "G014873535 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptosporangiales; f__Streptosporangiaceae; g__Nonomuraea; s__Nonomuraea africana\n",
      "Coverage Percentage: 0.0015575755212668%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 337\n",
      "Length of fail 2780\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G014874165 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Micrococcus; s__Micrococcus yunnanensis\n",
      "Coverage Percentage: 0.7428596187666303%\n",
      "Adj, read counts 102\n",
      "Number of samples per group [1, 2, 1, 1, 1, 2, 2, 2, 2, 3, 1, 1, 1, 2, 1, 2, 1, 2, 1, 1, 1, 1, 2, 2, 2, 1, 1, 1, 1, 1, 1, 3, 1, 2, 1, 1, 1, 2, 1, 2, 1, 1, 1, 3, 3, 2, 1, 1, 4, 2, 1]\n",
      "Number of bins 69\n",
      "total # samples 22\n",
      "Number of total reads 102\n",
      "Number of hits in BIOM tables 49.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.745098039215685 %\n",
      "Percentage of reads from samples who only fall into one bin group 15.686274509803921 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2780\n",
      "G014883995 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus cohnii\n",
      "Coverage Percentage: 0.0175729198589511%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2781\n",
      "G014892695 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas simiae\n",
      "Coverage Percentage: 0.0051835674871549%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2782\n",
      "G014897585 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Dermabacteraceae; g__Brachybacterium; s__Brachybacterium tyrofermentans\n",
      "Coverage Percentage: 0.0136829637776602%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2783\n",
      "G014898295 d__Eukaryota; p__Ascomycota; c__Leotiomycetes; o__Helotiales; f__Sclerotiniaceae; g__Botrytis; s__Botrytis byssoidea\n",
      "Coverage Percentage: 0.00130516606598%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2784\n",
      "G014898435 d__Eukaryota; p__Ascomycota; c__Leotiomycetes; o__Helotiales; f__Sclerotiniaceae; g__Botrytis; s__Botrytis sinoallii\n",
      "Coverage Percentage: 0.0015451073449867%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2785\n",
      "G014898465 d__Eukaryota; p__Ascomycota; c__Leotiomycetes; o__Helotiales; f__Sclerotiniaceae; g__Botrytis; s__Botrytis porri\n",
      "Coverage Percentage: 0.0008099923044319%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2786\n",
      "G014898535 d__Eukaryota; p__Ascomycota; c__Leotiomycetes; o__Helotiales; f__Sclerotiniaceae; g__Botrytis; s__Botrytis deweyae\n",
      "Coverage Percentage: 0.0040389087553715%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2787\n",
      "G014899185 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Pseudopuniceibacterium; s__Pseudopuniceibacterium antarcticum\n",
      "Coverage Percentage: 0.0048101359184072%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2788\n",
      "G014904065 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Gulosibacter; s__Gulosibacter sediminis\n",
      "Coverage Percentage: 0.0548290805983754%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2789\n",
      "G014904705 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Thalassobaculaceae; g__Nisaea; s__Nisaea sediminum\n",
      "Coverage Percentage: 0.0060122678134835%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2790\n",
      "G014925295 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Alteromonadales; f__Pseudoalteromonadaceae; g__Pseudoalteromonas; s__Pseudoalteromonas carrageenovora\n",
      "Coverage Percentage: 0.0119718201771215%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2791\n",
      "G014926585 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Devosiaceae; g__Devosia; s__Devosia beringensis\n",
      "Coverage Percentage: 0.0074590646777474%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2792\n",
      "G014982805 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__; g__Gemmiger; s__Gemmiger gallinarum\n",
      "Coverage Percentage: 0.0122476501935771%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2793\n",
      "G015024225 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides lijunqiniae\n",
      "Coverage Percentage: 0.0114108343065798%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2794\n",
      "G015024265 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides lijunqiniae\n",
      "Coverage Percentage: 0.0083276263984082%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2795\n",
      "G015034585 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptosporangiales; f__Nocardiopsaceae; g__Thermobifida; s__Thermobifida fusca\n",
      "Coverage Percentage: 0.0353104434092977%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2796\n",
      "G015070375 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides kribbensis\n",
      "Coverage Percentage: 0.0243070074601386%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2797\n",
      "G015097455 d__Bacteria; p__Firmicutes; c__Erysipelotrichia; o__Erysipelotrichales; f__Erysipelotrichaceae; g__Faecalibacillus; s__Faecalibacillus intestinalis\n",
      "Coverage Percentage: 0.0458887895464152%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2798\n",
      "G015099135 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Kluyvera; s__Kluyvera ascorbata\n",
      "Coverage Percentage: 0.0451278324044645%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 338\n",
      "Length of fail 2799\n",
      "G015099595 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Rhodococcus; s__Rhodococcus qingshengii\n",
      "Coverage Percentage: 0.1409429230224169%\n",
      "Adj, read counts 56\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1]\n",
      "Number of bins 55\n",
      "total # samples 29\n",
      "Number of total reads 56\n",
      "Number of hits in BIOM tables 21.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.714285714285714 %\n",
      "Percentage of reads from samples who only fall into one bin group 28.57142857142857 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2799\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G015139575 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Klebsiella; s__Klebsiella michiganensis\n",
      "Coverage Percentage: 0.0181897597445555%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2800\n",
      "G015140235 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Oscillospiraceae; g__Solibaculum; s__Solibaculum mannosilyticum\n",
      "Coverage Percentage: 0.0059414339867336%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2801\n",
      "G015158295 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Devosiaceae; g__Devosia; s__Devosia faecipullorum\n",
      "Coverage Percentage: 0.0177621251500517%\n",
      "Adj, read counts 39\n",
      "Number of samples per group [2, 1, 36]\n",
      "Number of bins 4\n",
      "total # samples 38\n",
      "Number of total reads 39\n",
      "Number of hits in BIOM tables 84.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 100.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 94.87179487179486 %\n",
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2802\n",
      "G015163715 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas cyclaminis\n",
      "Coverage Percentage: 0.0666457800945935%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2803\n",
      "G015182315 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Cellvibrionales; f__Cellvibrionaceae; g__Cellvibrio; s__Cellvibrio polysaccharolyticus\n",
      "Coverage Percentage: 0.0345613848204896%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2804\n",
      "G015207785 d__Bacteria; p__Cyanobacteria; c__; o__Nostocales; f__Aphanizomenonaceae; g__Dolichospermum; s__Dolichospermum flos-aquae\n",
      "Coverage Percentage: 0.0065122680765013%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2805\n",
      "G015207825 d__Bacteria; p__Cyanobacteria; c__; o__Chroococcales; f__Geminocystaceae; g__Cyanobacterium; s__Cyanobacterium stanieri\n",
      "Coverage Percentage: 0.0080765582592199%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2806\n",
      "G015207985 d__Bacteria; p__Cyanobacteria; c__; o__Synechococcales; f__Merismopediaceae; g__Synechocystis; s__Synechocystis salina\n",
      "Coverage Percentage: 0.006520746295637%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2807\n",
      "G015210005 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Sphingobacterium; s__Sphingobacterium hungaricum\n",
      "Coverage Percentage: 0.0091217819524311%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2808\n",
      "G015220045 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Marivivens; s__Marivivens aquimaris\n",
      "Coverage Percentage: 0.0037267240170827%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2809\n",
      "G015221875 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Spirosomaceae; g__Dyadobacter; s__Dyadobacter subterraneus\n",
      "Coverage Percentage: 0.0116327218450575%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2810\n",
      "G015223195 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Chromobacteriaceae; g__Chitinilyticum; s__Chitinilyticum piscinae\n",
      "Coverage Percentage: 0.0381692880537252%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2811\n",
      "G015234135 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Faecalibacter; s__Faecalibacter rhinopitheci\n",
      "Coverage Percentage: 0.0178144782934905%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2812\n",
      "G015234745 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Sanguibacteraceae; g__Sanguibacter; s__Sanguibacter inulinus\n",
      "Coverage Percentage: 0.0305073513329835%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2813\n",
      "G015277675 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium soyangense\n",
      "Coverage Percentage: 0.0378757112136875%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2814\n",
      "G015278215 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium ginsengiterrae\n",
      "Coverage Percentage: 0.0235847656268039%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2815\n",
      "G015319355 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Arcticiflavibacter; s__Arcticiflavibacter luteus\n",
      "Coverage Percentage: 0.0165301435695445%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2816\n",
      "G015326295 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Brucellaceae; g__Brucella; s__Brucella anthropi\n",
      "Coverage Percentage: 0.0045897173454474%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2817\n",
      "G015351335 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium afermentans\n",
      "Coverage Percentage: 0.1215524076891755%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2818\n",
      "G015352455 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides islandensis\n",
      "Coverage Percentage: 0.0060831587950189%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2819\n",
      "G015352955 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Noviherbaspirillum; s__Noviherbaspirillum soli\n",
      "Coverage Percentage: 0.0072829152601647%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2820\n",
      "G015352975 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Aestuariimicrobium; s__Aestuariimicrobium ganziense\n",
      "Coverage Percentage: 0.0125967122898221%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2821\n",
      "G015356815 d__Bacteria; p__Chlamydiae; c__Chlamydiia; o__Parachlamydiales; f__Rhabdochlamydiaceae; g__Candidatus Rhabdochlamydia; s__Candidatus Rhabdochlamydia porcellionis\n",
      "Coverage Percentage: 0.0478876518647081%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2822\n",
      "G015356825 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Catenulisporales; f__Catenulisporaceae; g__Catenulispora; s__Catenulispora rubra\n",
      "Coverage Percentage: 0.0045258087756836%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2823\n",
      "G015377145 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas lundensis\n",
      "Coverage Percentage: 0.0390340313520691%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2824\n",
      "G015461835 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas neuropathica\n",
      "Coverage Percentage: 0.0291489239832189%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2825\n",
      "G015461845 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas mucoides\n",
      "Coverage Percentage: 0.0257901682936295%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2826\n",
      "G015476235 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Schlegelella; s__Schlegelella thermodepolymerans\n",
      "Coverage Percentage: 0.0711934505135543%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2827\n",
      "G015476605 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Rouxiella; s__Rouxiella silvae\n",
      "Coverage Percentage: 0.0028124108898784%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2828\n",
      "G015547255 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola massiliensis\n",
      "Coverage Percentage: 0.0986121967236434%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2829\n",
      "G015547475 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Clostridium; s__Clostridium celatum\n",
      "Coverage Percentage: 0.0082978750296058%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2830\n",
      "G015548275 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Rikenellaceae; g__Alistipes; s__Alistipes finegoldii\n",
      "Coverage Percentage: 0.040538571943726%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2831\n",
      "G015552745 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Oscillospiraceae; g__Agathobaculum; s__Agathobaculum butyriciproducens\n",
      "Coverage Percentage: 0.0438557523276901%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2832\n",
      "G015555825 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides clarus\n",
      "Coverage Percentage: 0.0409956241404195%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 339\n",
      "Length of fail 2833\n",
      "G015557635 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Fusicatenibacter; s__Fusicatenibacter saccharivorans\n",
      "Coverage Percentage: 0.3924051583259715%\n",
      "Adj, read counts 88\n",
      "Number of samples per group [3, 1, 1, 2, 3, 2, 3, 1, 1, 1, 2, 2, 1, 1, 2, 2, 2, 1, 1, 2, 3, 1, 1, 1, 2, 2, 3, 1, 2, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 3, 3, 1, 1, 2, 1, 1, 1, 2, 1]\n",
      "Number of bins 68\n",
      "total # samples 31\n",
      "Number of total reads 84\n",
      "Number of hits in BIOM tables 153.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.227272727272727 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.666666666666664 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 340\n",
      "Length of fail 2833\n",
      "G015558405 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Rikenellaceae; g__Alistipes; s__Alistipes shahii\n",
      "Coverage Percentage: 0.0506339630394523%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 340\n",
      "Length of fail 2834\n",
      "G015668545 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Clostridium; s__Clostridium paraputrificum\n",
      "Coverage Percentage: 0.0130847216954795%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 340\n",
      "Length of fail 2835\n",
      "G015668785 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides xylanisolvens\n",
      "Coverage Percentage: 0.128565424142279%\n",
      "Adj, read counts 50\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 1]\n",
      "Number of bins 46\n",
      "total # samples 26\n",
      "Number of total reads 49\n",
      "Number of hits in BIOM tables 21.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 18.000000000000004 %\n",
      "Percentage of reads from samples who only fall into one bin group 26.53061224489796 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 341\n",
      "Length of fail 2835\n",
      "G015669135 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Blautia; s__Blautia luti\n",
      "Coverage Percentage: 0.1591197584846705%\n",
      "Adj, read counts 39\n",
      "Number of samples per group [1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 2, 1, 4, 2, 2, 1, 1, 1]\n",
      "Number of bins 32\n",
      "total # samples 20\n",
      "Number of total reads 39\n",
      "Number of hits in BIOM tables 12.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 25.64102564102564 %\n",
      "Percentage of reads from samples who only fall into one bin group 25.64102564102564 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 342\n",
      "Length of fail 2835\n",
      "G015680405 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas pudica\n",
      "Coverage Percentage: 0.0443008796112126%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 342\n",
      "Length of fail 2836\n",
      "G015694345 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium jiangmenense\n",
      "Coverage Percentage: 0.0336891479046697%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 342\n",
      "Length of fail 2837\n",
      "G015707995 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium paulum\n",
      "Coverage Percentage: 0.3451412213965805%\n",
      "Adj, read counts 69\n",
      "Number of samples per group [3, 1, 1, 2, 2, 1, 2, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 3, 2, 1, 2, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 2, 2, 2, 1, 1, 1, 1, 1, 1, 1, 2]\n",
      "Number of bins 55\n",
      "total # samples 31\n",
      "Number of total reads 67\n",
      "Number of hits in BIOM tables 123.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 13.043478260869565 %\n",
      "Percentage of reads from samples who only fall into one bin group 20.8955223880597 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 343\n",
      "Length of fail 2837\n",
      "G015751765 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora ureilytica\n",
      "Coverage Percentage: 0.0106393922440944%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 343\n",
      "Length of fail 2838\n",
      "G015752125 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium aquatimens\n",
      "Coverage Percentage: 0.0142163297713309%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 343\n",
      "Length of fail 2839\n",
      "G015767265 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Aureivirga; s__Aureivirga marina\n",
      "Coverage Percentage: 0.0028007075096388%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 343\n",
      "Length of fail 2840\n",
      "G015965555 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Citrobacter; s__Citrobacter rodentium\n",
      "Coverage Percentage: 0.0199255971345238%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 343\n",
      "Length of fail 2841\n",
      "G015999335 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Xinfangfangia; s__Xinfangfangia soli\n",
      "Coverage Percentage: 0.0052933926933251%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 343\n",
      "Length of fail 2842\n",
      "G016009075 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas juntendi\n",
      "Coverage Percentage: 0.0225684763453073%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 343\n",
      "Length of fail 2843\n",
      "G016009095 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas monteilii\n",
      "Coverage Percentage: 0.0687421677693166%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 343\n",
      "Length of fail 2844\n",
      "G016026595 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Brevibacteriaceae; g__Brevibacterium; s__Brevibacterium casei\n",
      "Coverage Percentage: 0.3288167794318044%\n",
      "Adj, read counts 67\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 2, 1, 1, 1, 1]\n",
      "Number of bins 56\n",
      "total # samples 5\n",
      "Number of total reads 65\n",
      "Number of hits in BIOM tables 60.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.417910447761194 %\n",
      "Percentage of reads from samples who only fall into one bin group 1.5384615384615385 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 344\n",
      "Length of fail 2844\n",
      "G016026615 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas allosaccharophila\n",
      "Coverage Percentage: 0.0670996432942361%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 344\n",
      "Length of fail 2845\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G016027055 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter johnsonii\n",
      "Coverage Percentage: 1.034273527545843%\n",
      "Adj, read counts 241\n",
      "Number of samples per group [2, 3, 7, 1, 2, 1, 1, 5, 1, 2, 2, 3, 2, 2, 4, 2, 1, 3, 2, 3, 1, 3, 1, 12, 1, 1, 6, 5, 2, 1, 5, 2, 1, 1, 2, 2, 4, 1, 1, 1, 6, 2, 1, 2, 1, 2, 5, 1, 1, 1, 1, 1, 3, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 6, 1, 3, 4, 3, 2, 3, 3, 1, 1, 1, 4, 3, 1, 3, 1, 2, 5, 4, 4, 3, 2, 6, 1, 4]\n",
      "Number of bins 140\n",
      "total # samples 47\n",
      "Number of total reads 238\n",
      "Number of hits in BIOM tables 74.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.448132780082988 %\n",
      "Percentage of reads from samples who only fall into one bin group 3.7815126050420167 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 345\n",
      "Length of fail 2845\n",
      "G016027135 d__Bacteria; p__Firmicutes; c__Erysipelotrichia; o__Erysipelotrichales; f__Erysipelotrichaceae; g__Erysipelatoclostridium; s__Erysipelatoclostridium ramosum\n",
      "Coverage Percentage: 0.049593038111433%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 345\n",
      "Length of fail 2846\n",
      "G016027415 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Brevibacteriaceae; g__Brevibacterium; s__Brevibacterium casei\n",
      "Coverage Percentage: 0.3534259540303841%\n",
      "Adj, read counts 75\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 3, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 57\n",
      "total # samples 6\n",
      "Number of total reads 71\n",
      "Number of hits in BIOM tables 76.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.666666666666668 %\n",
      "Percentage of reads from samples who only fall into one bin group 2.8169014084507045 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 346\n",
      "Length of fail 2846\n",
      "G016027615 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium minutissimum\n",
      "Coverage Percentage: 0.048739577464214%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 346\n",
      "Length of fail 2847\n",
      "G016027675 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia plymuthica\n",
      "Coverage Percentage: 0.0341361561529112%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 346\n",
      "Length of fail 2848\n",
      "G016027815 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Delftia; s__Delftia lacustris\n",
      "Coverage Percentage: 0.170004156970734%\n",
      "Adj, read counts 84\n",
      "Number of samples per group [1, 1, 1, 5, 1, 1, 6, 1, 5, 1, 1, 1, 7, 1, 1, 3, 1, 1, 1, 5, 1, 9, 7, 1, 1, 6]\n",
      "Number of bins 36\n",
      "total # samples 31\n",
      "Number of total reads 82\n",
      "Number of hits in BIOM tables 8.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 35.714285714285715 %\n",
      "Percentage of reads from samples who only fall into one bin group 19.51219512195122 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 347\n",
      "Length of fail 2848\n",
      "G016028295 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus auricularis\n",
      "Coverage Percentage: 0.0549496407746185%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 347\n",
      "Length of fail 2849\n",
      "G016028495 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Salmonella; s__Salmonella enterica\n",
      "Coverage Percentage: 0.0216956411484601%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 347\n",
      "Length of fail 2850\n",
      "G016028655 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas stutzeri\n",
      "Coverage Percentage: 0.1288094933610855%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 4, 1, 1, 1, 1]\n",
      "Number of bins 32\n",
      "total # samples 15\n",
      "Number of total reads 37\n",
      "Number of hits in BIOM tables 17.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 27.027027027027028 %\n",
      "Percentage of reads from samples who only fall into one bin group 21.62162162162162 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 348\n",
      "Length of fail 2850\n",
      "G016028715 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Neisseriaceae; g__Neisseria; s__Neisseria cinerea\n",
      "Coverage Percentage: 0.0207867200683506%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 348\n",
      "Length of fail 2851\n",
      "G016028755 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Moraxella; s__Moraxella osloensis\n",
      "Coverage Percentage: 0.1193434019150777%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 348\n",
      "Length of fail 2852\n",
      "G016028775 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Moraxella; s__Moraxella nonliquefaciens\n",
      "Coverage Percentage: 0.0128852618630808%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 348\n",
      "Length of fail 2853\n",
      "G016031625 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium agreste\n",
      "Coverage Percentage: 0.0066126729020406%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 348\n",
      "Length of fail 2854\n",
      "G016031635 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium diversitatis\n",
      "Coverage Percentage: 0.0094640866304633%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 348\n",
      "Length of fail 2855\n",
      "G016093545 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia rhizosphaerae\n",
      "Coverage Percentage: 0.0169226485015958%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 348\n",
      "Length of fail 2856\n",
      "G016103195 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides thetaiotaomicron\n",
      "Coverage Percentage: 0.1067654087304856%\n",
      "Adj, read counts 45\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 42\n",
      "total # samples 23\n",
      "Number of total reads 44\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.111111111111112 %\n",
      "Percentage of reads from samples who only fall into one bin group 27.27272727272727 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 349\n",
      "Length of fail 2856\n",
      "G016103505 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces javensis\n",
      "Coverage Percentage: 0.0045580361546797%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 349\n",
      "Length of fail 2857\n",
      "G016106035 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Arcobacter; s__Arcobacter vandammei\n",
      "Coverage Percentage: 0.1058069729331631%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 349\n",
      "Length of fail 2858\n",
      "G016107525 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter aquimaris\n",
      "Coverage Percentage: 0.025844979131492%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 349\n",
      "Length of fail 2859\n",
      "G016107535 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter cibarius\n",
      "Coverage Percentage: 0.0426881383379242%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 349\n",
      "Length of fail 2860\n",
      "G016107545 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter namhaensis\n",
      "Coverage Percentage: 0.0138560821824225%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 349\n",
      "Length of fail 2861\n",
      "G016107555 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter celer\n",
      "Coverage Percentage: 0.0428070853837474%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 349\n",
      "Length of fail 2862\n",
      "G016107575 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter faecalis\n",
      "Coverage Percentage: 0.4510440305504735%\n",
      "Adj, read counts 82\n",
      "Number of samples per group [1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 4, 2, 1, 1, 1, 1, 1, 1, 2, 3, 2, 1, 1, 4, 2, 2, 1, 1, 1, 3, 2, 1, 1, 2, 1, 1, 1, 2, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 67\n",
      "total # samples 31\n",
      "Number of total reads 82\n",
      "Number of hits in BIOM tables 79.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.634146341463413 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.29268292682927 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 350\n",
      "Length of fail 2862\n",
      "G016107625 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Oceanospirillales; f__Halomonadaceae; g__Halomonas; s__Halomonas alkaliphila\n",
      "Coverage Percentage: 0.0056392794612071%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 350\n",
      "Length of fail 2863\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G016110345 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Fluviibacteraceae; g__Fluviibacter; s__Fluviibacter phosphoraccumulans\n",
      "Coverage Percentage: 0.7077856420626896%\n",
      "Adj, read counts 96\n",
      "Number of samples per group [1, 1, 2, 1, 1, 1, 4, 1, 2, 1, 1, 1, 1, 1, 4, 1, 1, 2, 2, 2, 3, 7, 1, 2, 2, 1, 1, 1, 4, 1, 1, 3, 2, 1, 5, 2, 1, 1, 1, 1, 2, 1, 2, 3, 1, 2, 1, 3, 2]\n",
      "Number of bins 68\n",
      "total # samples 35\n",
      "Number of total reads 96\n",
      "Number of hits in BIOM tables 88.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 18.75 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.666666666666664 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 351\n",
      "Length of fail 2863\n",
      "G016117655 d__Bacteria; p__Spirochaetes; c__Spirochaetia; o__Spirochaetales; f__Treponemataceae; g__Treponema; s__Treponema peruense\n",
      "Coverage Percentage: 0.0055148414976118%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 351\n",
      "Length of fail 2864\n",
      "G016117815 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides uniformis\n",
      "Coverage Percentage: 0.2732113895672119%\n",
      "Adj, read counts 78\n",
      "Number of samples per group [1, 3, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1]\n",
      "Number of bins 69\n",
      "total # samples 37\n",
      "Number of total reads 77\n",
      "Number of hits in BIOM tables 23.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.256410256410255 %\n",
      "Percentage of reads from samples who only fall into one bin group 32.467532467532465 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 352\n",
      "Length of fail 2864\n",
      "G016127115 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pasteurellales; f__Pasteurellaceae; g__Aggregatibacter; s__Aggregatibacter segnis\n",
      "Coverage Percentage: 0.0136242027901979%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 352\n",
      "Length of fail 2865\n",
      "G016127195 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas jandaei\n",
      "Coverage Percentage: 0.0238844458643763%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 352\n",
      "Length of fail 2866\n",
      "G016127215 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pasteurellales; f__Pasteurellaceae; g__Haemophilus; s__Haemophilus parainfluenzae\n",
      "Coverage Percentage: 0.0411821614938385%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 352\n",
      "Length of fail 2867\n",
      "G016127255 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Leuconostoc; s__Leuconostoc pseudomesenteroides\n",
      "Coverage Percentage: 0.0192466201374501%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 352\n",
      "Length of fail 2868\n",
      "G016127275 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus infantarius\n",
      "Coverage Percentage: 0.0552395363963196%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 352\n",
      "Length of fail 2869\n",
      "G016127395 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium glucuronolyticum\n",
      "Coverage Percentage: 0.2736367650343264%\n",
      "Adj, read counts 43\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2]\n",
      "Number of bins 36\n",
      "total # samples 18\n",
      "Number of total reads 43\n",
      "Number of hits in BIOM tables 14.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 18.6046511627907 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.6046511627907 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 353\n",
      "Length of fail 2869\n",
      "G016127415 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Delftia; s__Delftia acidovorans\n",
      "Coverage Percentage: 0.0883961382893434%\n",
      "Adj, read counts 47\n",
      "Number of samples per group [1, 1, 1, 1, 6, 1, 1, 1, 1, 1, 1, 1, 8, 1, 1, 1, 8]\n",
      "Number of bins 21\n",
      "total # samples 19\n",
      "Number of total reads 45\n",
      "Number of hits in BIOM tables 6.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 68.08510638297872 %\n",
      "Percentage of reads from samples who only fall into one bin group 22.22222222222222 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 354\n",
      "Length of fail 2869\n",
      "G016127555 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus oralis\n",
      "Coverage Percentage: 0.0624369999798089%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 354\n",
      "Length of fail 2870\n",
      "G016127855 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces naeslundii\n",
      "Coverage Percentage: 0.059832690539043%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 354\n",
      "Length of fail 2871\n",
      "G016127955 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces oris\n",
      "Coverage Percentage: 0.1139349852809163%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 354\n",
      "Length of fail 2872\n",
      "G016135805 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Caulobacter; s__Caulobacter hibisci\n",
      "Coverage Percentage: 0.0189697486400522%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 354\n",
      "Length of fail 2873\n",
      "G016307665 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas carnis\n",
      "Coverage Percentage: 0.0928888602483424%\n",
      "Adj, read counts 34\n",
      "Number of samples per group [1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 31\n",
      "total # samples 18\n",
      "Number of total reads 34\n",
      "Number of hits in BIOM tables 7.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.705882352941178 %\n",
      "Percentage of reads from samples who only fall into one bin group 26.47058823529412 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 355\n",
      "Length of fail 2873\n",
      "G016403165 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus lutetiensis\n",
      "Coverage Percentage: 0.0288667785486859%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 355\n",
      "Length of fail 2874\n",
      "G016403285 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium striatum\n",
      "Coverage Percentage: 0.0953429297205757%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 355\n",
      "Length of fail 2875\n",
      "G016415705 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Klebsiella; s__Klebsiella quasipneumoniae\n",
      "Coverage Percentage: 0.0460939504147806%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 355\n",
      "Length of fail 2876\n",
      "G016446475 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus binzhouensis\n",
      "Coverage Percentage: 0.0151864900984084%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 355\n",
      "Length of fail 2877\n",
      "G016461705 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus vicugnae\n",
      "Coverage Percentage: 0.0273159151119926%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 355\n",
      "Length of fail 2878\n",
      "G016464365 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingosinicellaceae; g__Sandaracinobacteroides; s__Sandaracinobacteroides hominis\n",
      "Coverage Percentage: 0.012830651264598%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 355\n",
      "Length of fail 2879\n",
      "G016502565 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter bereziniae\n",
      "Coverage Percentage: 0.0535452654941592%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 355\n",
      "Length of fail 2880\n",
      "G016522065 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium agrisoli\n",
      "Coverage Percentage: 0.0346344904087757%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 355\n",
      "Length of fail 2881\n",
      "G016583515 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Chromatiales; f__Chromatiaceae; g__Marichromatium; s__Marichromatium gracile\n",
      "Coverage Percentage: 0.0139785076131094%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 355\n",
      "Length of fail 2882\n",
      "G016583655 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Rhodoferax; s__Rhodoferax fermentans\n",
      "Coverage Percentage: 0.1096321018317292%\n",
      "Adj, read counts 35\n",
      "Number of samples per group [1, 1, 1, 15, 1, 1, 1]\n",
      "Number of bins 8\n",
      "total # samples 18\n",
      "Number of total reads 35\n",
      "Number of hits in BIOM tables 6.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 88.57142857142858 %\n",
      "Percentage of reads from samples who only fall into one bin group 77.14285714285715 %\n",
      "Status: Fail -- over 75% of reads are part of a sample which only hit a single bin\n",
      "----------------------------\n",
      "Length of pass 355\n",
      "Length of fail 2883\n",
      "G016584445 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Rhodopseudomonas; s__Rhodopseudomonas palustris\n",
      "Coverage Percentage: 0.0081542020068865%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 355\n",
      "Length of fail 2884\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G016586375 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Butyricicoccus; s__Butyricicoccus faecihominis\n",
      "Coverage Percentage: 0.0600687424047695%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 355\n",
      "Length of fail 2885\n",
      "G016595435 d__Bacteria; p__Verrucomicrobia; c__Verrucomicrobiae; o__Verrucomicrobiales; f__Verrucomicrobiaceae; g__Luteolibacter; s__Luteolibacter pohnpeiensis\n",
      "Coverage Percentage: 0.009501665466475%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 355\n",
      "Length of fail 2886\n",
      "G016595465 d__Bacteria; p__Verrucomicrobia; c__Verrucomicrobiae; o__Verrucomicrobiales; f__Verrucomicrobiaceae; g__Luteolibacter; s__Luteolibacter yonseiensis\n",
      "Coverage Percentage: 0.081030513985926%\n",
      "Adj, read counts 32\n",
      "Number of samples per group [1, 1, 1, 1, 1, 4, 5, 5]\n",
      "Number of bins 10\n",
      "total # samples 9\n",
      "Number of total reads 32\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 84.375 %\n",
      "Percentage of reads from samples who only fall into one bin group 9.375 %\n",
      "Status: Fail - too many reads in top 3 bins\n",
      "----------------------------\n",
      "Length of pass 355\n",
      "Length of fail 2887\n",
      "G016598775 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces weissii\n",
      "Coverage Percentage: 0.0113486528447094%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 355\n",
      "Length of fail 2888\n",
      "G016599755 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium kefirresidentii\n",
      "Coverage Percentage: 0.5499309217644397%\n",
      "Adj, read counts 79\n",
      "Number of samples per group [1, 1, 1, 1, 3, 3, 1, 1, 2, 3, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 4, 1, 7, 3, 1, 6]\n",
      "Number of bins 56\n",
      "total # samples 35\n",
      "Number of total reads 78\n",
      "Number of hits in BIOM tables 75.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 24.050632911392405 %\n",
      "Percentage of reads from samples who only fall into one bin group 21.794871794871796 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 356\n",
      "Length of fail 2888\n",
      "G016599875 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Comamonas; s__Comamonas testosteroni\n",
      "Coverage Percentage: 0.0797511850345737%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 356\n",
      "Length of fail 2889\n",
      "G016622365 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas balearica\n",
      "Coverage Percentage: 0.020525254520013%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 356\n",
      "Length of fail 2890\n",
      "G016628615 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium macginleyi\n",
      "Coverage Percentage: 0.1159949078651199%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 356\n",
      "Length of fail 2891\n",
      "G016634425 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Lacisediminihabitans; s__Lacisediminihabitans changchengi\n",
      "Coverage Percentage: 0.0085566561994146%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 356\n",
      "Length of fail 2892\n",
      "G016647625 d__Eukaryota; p__Basidiomycota; c__Agaricomycetes; o__Boletales; f__Suillaceae; g__Suillus; s__Suillus subalutaceus\n",
      "Coverage Percentage: 0.0005723824806832%\n",
      "Adj, read counts 34\n",
      "Number of samples per group [1, 1, 32]\n",
      "Number of bins 3\n",
      "total # samples 34\n",
      "Number of total reads 34\n",
      "Number of hits in BIOM tables 10.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 100.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 100.0 %\n",
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 356\n",
      "Length of fail 2893\n",
      "G016647785 d__Eukaryota; p__Basidiomycota; c__Agaricomycetes; o__Boletales; f__Suillaceae; g__Suillus; s__Suillus fuscotomentosus\n",
      "Coverage Percentage: 0.0017672760300523%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 356\n",
      "Length of fail 2894\n",
      "G016654005 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas nasdae\n",
      "Coverage Percentage: 0.0309920356064645%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 356\n",
      "Length of fail 2895\n",
      "G016654015 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Ramlibacter; s__Ramlibacter ginsenosidimutans\n",
      "Coverage Percentage: 0.0276447308636247%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 356\n",
      "Length of fail 2896\n",
      "G016694755 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas syringae\n",
      "Coverage Percentage: 0.0312531880411777%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 356\n",
      "Length of fail 2897\n",
      "G016698665 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Devosiaceae; g__Devosia; s__Devosia rhizoryzae\n",
      "Coverage Percentage: 0.0173334415136775%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 356\n",
      "Length of fail 2898\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G016722785 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Ramlibacter; s__Ramlibacter monticola\n",
      "Coverage Percentage: 0.0246185990827738%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 356\n",
      "Length of fail 2899\n",
      "G016724805 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Sutterellaceae; g__Sutterella; s__Sutterella wadsworthensis\n",
      "Coverage Percentage: 0.0252765803066693%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 356\n",
      "Length of fail 2900\n",
      "G016725165 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus lutetiensis\n",
      "Coverage Percentage: 0.0317253045824386%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 356\n",
      "Length of fail 2901\n",
      "G016726205 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter ursingii\n",
      "Coverage Percentage: 0.1545434615077994%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 356\n",
      "Length of fail 2902\n",
      "G016726285 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Shigella; s__Shigella boydii\n",
      "Coverage Percentage: 0.8808829581066204%\n",
      "Adj, read counts 230\n",
      "Number of samples per group [2, 2, 1, 1, 1, 1, 2, 3, 3, 4, 4, 1, 2, 3, 1, 1, 1, 1, 1, 2, 2, 2, 2, 1, 1, 1, 2, 2, 3, 2, 2, 2, 3, 1, 2, 1, 2, 3, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 3, 2, 5, 2, 2, 1, 2, 1, 1, 1, 2, 2, 3, 1, 1, 1, 1, 3, 1, 2, 3, 1, 2, 6, 1, 1, 4, 1, 3, 1, 2, 1, 1, 4, 1, 2, 1, 1, 1, 3, 2, 2, 1, 1, 1, 2, 1, 1, 1]\n",
      "Number of bins 152\n",
      "total # samples 56\n",
      "Number of total reads 229\n",
      "Number of hits in BIOM tables 35.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.869565217391305 %\n",
      "Percentage of reads from samples who only fall into one bin group 13.537117903930133 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 357\n",
      "Length of fail 2902\n",
      "G016726305 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides caccae\n",
      "Coverage Percentage: 0.0701387307339998%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 357\n",
      "Length of fail 2903\n",
      "G016726365 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Rothia; s__Rothia aeria\n",
      "Coverage Percentage: 0.0220982330314789%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 357\n",
      "Length of fail 2904\n",
      "G016728365 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium tuberculostearicum\n",
      "Coverage Percentage: 0.880150286273163%\n",
      "Adj, read counts 123\n",
      "Number of samples per group [1, 5, 3, 2, 1, 2, 2, 1, 4, 2, 2, 2, 1, 1, 8, 1, 2, 1, 2, 1, 2, 3, 2, 5, 1, 1, 6, 2, 4, 3, 6, 1, 4, 1, 1, 1, 1, 3, 1, 2, 1, 1, 2, 1, 3, 1, 1, 1, 3, 1, 2, 1, 1, 1]\n",
      "Number of bins 96\n",
      "total # samples 38\n",
      "Number of total reads 121\n",
      "Number of hits in BIOM tables 94.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.88617886178862 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.570247933884298 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 358\n",
      "Length of fail 2904\n",
      "G016728505 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium riegelii\n",
      "Coverage Percentage: 0.0161954119350659%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 358\n",
      "Length of fail 2905\n",
      "G016728645 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium glucuronolyticum\n",
      "Coverage Percentage: 0.2965411352250011%\n",
      "Adj, read counts 52\n",
      "Number of samples per group [2, 1, 1, 2, 1, 1, 2, 3, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 3, 1, 1, 2, 1, 2, 1, 1, 1]\n",
      "Number of bins 40\n",
      "total # samples 21\n",
      "Number of total reads 49\n",
      "Number of hits in BIOM tables 15.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 21.153846153846153 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.367346938775512 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 359\n",
      "Length of fail 2905\n",
      "G016728665 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium propinquum\n",
      "Coverage Percentage: 0.3301847042510323%\n",
      "Adj, read counts 47\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 1, 4, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1]\n",
      "Number of bins 39\n",
      "total # samples 25\n",
      "Number of total reads 46\n",
      "Number of hits in BIOM tables 47.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 23.404255319148938 %\n",
      "Percentage of reads from samples who only fall into one bin group 32.608695652173914 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 360\n",
      "Length of fail 2905\n",
      "G016728685 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium aurimucosum\n",
      "Coverage Percentage: 0.1038705657535316%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 360\n",
      "Length of fail 2906\n",
      "G016728705 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium aurimucosum\n",
      "Coverage Percentage: 0.0824382434554371%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 360\n",
      "Length of fail 2907\n",
      "G016728725 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium amycolatum\n",
      "Coverage Percentage: 0.0588017923238621%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 360\n",
      "Length of fail 2908\n",
      "G016728785 d__Bacteria; p__Firmicutes; c__Erysipelotrichia; o__Erysipelotrichales; f__Erysipelotrichaceae; g__Erysipelatoclostridium; s__Erysipelatoclostridium ramosum\n",
      "Coverage Percentage: 0.0435418271031083%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 360\n",
      "Length of fail 2909\n",
      "G016742975 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia nevei\n",
      "Coverage Percentage: 0.0425659808073105%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 360\n",
      "Length of fail 2910\n",
      "G016743815 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas aliaeris\n",
      "Coverage Percentage: 0.0742600914537075%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 360\n",
      "Length of fail 2911\n",
      "G016758215 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Leucobacter; s__Leucobacter aridicollis\n",
      "Coverage Percentage: 0.0083141279785855%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 360\n",
      "Length of fail 2912\n",
      "G016758255 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Leucobacter; s__Leucobacter chromiireducens\n",
      "Coverage Percentage: 0.0046859934414711%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 360\n",
      "Length of fail 2913\n",
      "G016758755 d__Eukaryota; p__Basidiomycota; c__Agaricomycetes; o__Boletales; f__Suillaceae; g__Suillus; s__Suillus discolor\n",
      "Coverage Percentage: 0.0003678290328016%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 360\n",
      "Length of fail 2914\n",
      "G016766915 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola vulgatus\n",
      "Coverage Percentage: 0.5632549706662756%\n",
      "Adj, read counts 176\n",
      "Number of samples per group [1, 2, 1, 1, 1, 2, 2, 1, 2, 2, 2, 1, 1, 3, 1, 1, 2, 1, 3, 1, 2, 1, 1, 2, 2, 2, 1, 2, 1, 2, 2, 1, 1, 2, 1, 1, 1, 1, 1, 4, 1, 2, 2, 1, 2, 1, 3, 1, 1, 1, 1, 1, 3, 2, 1, 1, 1, 3, 1, 1, 1, 1, 5, 1, 1, 2, 3, 2, 2, 1, 1, 3, 1, 3, 1, 1, 1, 4, 1, 1, 2, 1, 5, 1, 1, 1, 4, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1]\n",
      "Number of bins 143\n",
      "total # samples 48\n",
      "Number of total reads 175\n",
      "Number of hits in BIOM tables 78.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 9.090909090909092 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.428571428571429 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 361\n",
      "Length of fail 2914\n",
      "G016772315 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Thiotrichales; f__Thiotrichaceae; g__Thiothrix; s__Thiothrix subterranea\n",
      "Coverage Percentage: 0.0066460369755087%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 361\n",
      "Length of fail 2915\n",
      "G016785885 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium godavarianum\n",
      "Coverage Percentage: 0.176077203081351%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 361\n",
      "Length of fail 2916\n",
      "G016801025 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Devosiaceae; g__Paradevosia; s__Paradevosia shaoguanensis\n",
      "Coverage Percentage: 0.0277997081786655%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 361\n",
      "Length of fail 2917\n",
      "G016808375 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Erwiniaceae; g__Pantoea; s__Pantoea eucrina\n",
      "Coverage Percentage: 0.0236575924786673%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 361\n",
      "Length of fail 2918\n",
      "G016812215 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas fragi\n",
      "Coverage Percentage: 0.2075480644325746%\n",
      "Adj, read counts 66\n",
      "Number of samples per group [1, 1, 1, 1, 3, 1, 1, 2, 1, 2, 1, 2, 1, 5, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 50\n",
      "total # samples 20\n",
      "Number of total reads 65\n",
      "Number of hits in BIOM tables 31.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.666666666666668 %\n",
      "Percentage of reads from samples who only fall into one bin group 23.076923076923077 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 362\n",
      "Length of fail 2918\n",
      "G016858125 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Rhodocyclaceae; g__Azospira; s__Azospira restricta\n",
      "Coverage Percentage: 0.0205020460538844%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 362\n",
      "Length of fail 2919\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G016861625 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Eurotiales; f__Aspergillaceae; g__Aspergillus; s__Aspergillus luchuensis\n",
      "Coverage Percentage: 0.0017807393356174%\n",
      "Adj, read counts 50\n",
      "Number of samples per group [44, 2, 3, 1]\n",
      "Number of bins 4\n",
      "total # samples 48\n",
      "Number of total reads 50\n",
      "Number of hits in BIOM tables 10.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 98.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 92.0 %\n",
      "Status: Fail -- over 75% of reads are part of a sample which only hit a single bin\n",
      "----------------------------\n",
      "Length of pass 362\n",
      "Length of fail 2920\n",
      "G016861735 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Eurotiales; f__Aspergillaceae; g__Aspergillus; s__Aspergillus chevalieri\n",
      "Coverage Percentage: 0.0010202884870748%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 362\n",
      "Length of fail 2921\n",
      "G016861865 d__Eukaryota; p__Ascomycota; c__Eurotiomycetes; o__Eurotiales; f__Aspergillaceae; g__Aspergillus; s__Aspergillus puulaauensis\n",
      "Coverage Percentage: 0.0003584020304553%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 362\n",
      "Length of fail 2922\n",
      "G016862035 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes campanulatus\n",
      "Coverage Percentage: 0.0145666379836627%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 362\n",
      "Length of fail 2923\n",
      "G016862075 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes couchii\n",
      "Coverage Percentage: 0.0374400166291569%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 362\n",
      "Length of fail 2924\n",
      "G016862095 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes cyaneus\n",
      "Coverage Percentage: 0.017342200212758%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 362\n",
      "Length of fail 2925\n",
      "G016862115 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes deccanensis\n",
      "Coverage Percentage: 0.0042856864931984%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 362\n",
      "Length of fail 2926\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G016862135 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes derwentensis\n",
      "Coverage Percentage: 0.1825073344624165%\n",
      "Adj, read counts 108\n",
      "Number of samples per group [3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 4, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 97\n",
      "total # samples 38\n",
      "Number of total reads 104\n",
      "Number of hits in BIOM tables 99.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.11111111111111 %\n",
      "Percentage of reads from samples who only fall into one bin group 17.307692307692307 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 363\n",
      "Length of fail 2926\n",
      "G016862155 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes digitatis\n",
      "Coverage Percentage: 0.0085099862815052%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 363\n",
      "Length of fail 2927\n",
      "G016862175 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes durhamensis\n",
      "Coverage Percentage: 0.0175789303962821%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 363\n",
      "Length of fail 2928\n",
      "G016862195 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes ferrugineus\n",
      "Coverage Percentage: 0.014481536041547%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 363\n",
      "Length of fail 2929\n",
      "G016862215 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes humidus\n",
      "Coverage Percentage: 0.0185605730741289%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 363\n",
      "Length of fail 2930\n",
      "G016862235 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes italicus\n",
      "Coverage Percentage: 0.0149683341378887%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 363\n",
      "Length of fail 2931\n",
      "G016862275 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes nipponensis\n",
      "Coverage Percentage: 0.0073475705726895%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 363\n",
      "Length of fail 2932\n",
      "G016862295 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes octamycinicus\n",
      "Coverage Percentage: 0.0290156650083798%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 363\n",
      "Length of fail 2933\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G016862315 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes palleronii\n",
      "Coverage Percentage: 0.2628289657979566%\n",
      "Adj, read counts 172\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 6, 1, 2, 1, 1, 1, 1, 1, 1, 2, 2, 2, 1, 1, 1, 3, 1, 1, 2, 2, 3, 1, 1, 2, 4, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 2, 1, 2, 1, 2, 1, 1, 2, 3, 1, 1, 1, 1, 1, 2, 1, 2, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2]\n",
      "Number of bins 144\n",
      "total # samples 41\n",
      "Number of total reads 169\n",
      "Number of hits in BIOM tables 218.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.465116279069768 %\n",
      "Percentage of reads from samples who only fall into one bin group 7.6923076923076925 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 364\n",
      "Length of fail 2933\n",
      "G016862335 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes philippinensis\n",
      "Coverage Percentage: 0.0254534339171734%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 364\n",
      "Length of fail 2934\n",
      "G016862355 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes regularis\n",
      "Coverage Percentage: 0.0166871846834108%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 364\n",
      "Length of fail 2935\n",
      "G016862375 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes rishiriensis\n",
      "Coverage Percentage: 0.0088527244809448%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 364\n",
      "Length of fail 2936\n",
      "G016862395 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes siamensis\n",
      "Coverage Percentage: 0.0345739850456573%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 364\n",
      "Length of fail 2937\n",
      "G016862415 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes teichomyceticus\n",
      "Coverage Percentage: 0.0305591581488912%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 364\n",
      "Length of fail 2938\n",
      "G016862435 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes tereljensis\n",
      "Coverage Percentage: 0.0103481432216373%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 364\n",
      "Length of fail 2939\n",
      "G016862475 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes xinjiangensis\n",
      "Coverage Percentage: 0.020196179525102%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 364\n",
      "Length of fail 2940\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G016862675 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Cellulomonadaceae; g__Cellulomonas; s__Cellulomonas chitinilytica\n",
      "Coverage Percentage: 0.0030810013621698%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 364\n",
      "Length of fail 2941\n",
      "G016862775 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Cellulomonadaceae; g__Cellulomonas; s__Cellulomonas phragmiteti\n",
      "Coverage Percentage: 0.0072715470144039%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 364\n",
      "Length of fail 2942\n",
      "G016862795 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Dactylosporangium; s__Dactylosporangium siamense\n",
      "Coverage Percentage: 0.0023737361034257%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 364\n",
      "Length of fail 2943\n",
      "G016865555 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas arcuscaelestis\n",
      "Coverage Percentage: 0.0071096393409676%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 364\n",
      "Length of fail 2944\n",
      "G016881005 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas arcuscaelestis\n",
      "Coverage Percentage: 0.006888859155604%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 364\n",
      "Length of fail 2945\n",
      "G016882605 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium longum\n",
      "Coverage Percentage: 0.3414240594737561%\n",
      "Adj, read counts 46\n",
      "Number of samples per group [1, 1, 2, 1, 2, 1, 2, 1, 1, 2, 1, 3, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 4, 2]\n",
      "Number of bins 37\n",
      "total # samples 23\n",
      "Number of total reads 46\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 19.565217391304348 %\n",
      "Percentage of reads from samples who only fall into one bin group 26.08695652173913 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 365\n",
      "Length of fail 2945\n",
      "G016888945 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola coprophilus\n",
      "Coverage Percentage: 0.0430764877493337%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 365\n",
      "Length of fail 2946\n",
      "G016889065 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Odoribacteraceae; g__Butyricimonas; s__Butyricimonas virosa\n",
      "Coverage Percentage: 0.0153330162847852%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 365\n",
      "Length of fail 2947\n",
      "G016889225 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium glucuronolyticum\n",
      "Coverage Percentage: 0.3038023644657849%\n",
      "Adj, read counts 49\n",
      "Number of samples per group [2, 3, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1]\n",
      "Number of bins 36\n",
      "total # samples 21\n",
      "Number of total reads 46\n",
      "Number of hits in BIOM tables 17.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.40816326530612 %\n",
      "Percentage of reads from samples who only fall into one bin group 26.08695652173913 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 366\n",
      "Length of fail 2947\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G016889365 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium kroppenstedtii\n",
      "Coverage Percentage: 0.0872974452010313%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 366\n",
      "Length of fail 2948\n",
      "G016889425 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium amycolatum\n",
      "Coverage Percentage: 0.0580619432447535%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 366\n",
      "Length of fail 2949\n",
      "G016889465 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium macginleyi\n",
      "Coverage Percentage: 0.1057934973493475%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 366\n",
      "Length of fail 2950\n",
      "G016894325 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Glutamicibacter; s__Glutamicibacter protophormiae\n",
      "Coverage Percentage: 0.0153026311374481%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 366\n",
      "Length of fail 2951\n",
      "G016899755 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Sutterellaceae; g__Sutterella; s__Sutterella massiliensis\n",
      "Coverage Percentage: 0.0378952515484081%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 366\n",
      "Length of fail 2952\n",
      "G016900115 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Limosilactobacillus; s__Limosilactobacillus alvi\n",
      "Coverage Percentage: 0.0156454715425968%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 366\n",
      "Length of fail 2953\n",
      "G016901255 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora humida\n",
      "Coverage Percentage: 0.0149830422904511%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 366\n",
      "Length of fail 2954\n",
      "G016901995 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides caecigallinarum\n",
      "Coverage Percentage: 0.0118244895784092%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 366\n",
      "Length of fail 2955\n",
      "G016904155 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Coprococcus; s__Coprococcus comes\n",
      "Coverage Percentage: 0.1302968485357687%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 366\n",
      "Length of fail 2956\n",
      "G016904755 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Escherichia; s__Escherichia albertii\n",
      "Coverage Percentage: 0.121008056481077%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 366\n",
      "Length of fail 2957\n",
      "G016907295 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium dextranolyticum\n",
      "Coverage Percentage: 0.0730541933861435%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 366\n",
      "Length of fail 2958\n",
      "G016907315 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium esteraromaticum\n",
      "Coverage Percentage: 0.0125010610468481%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 366\n",
      "Length of fail 2959\n",
      "G016907335 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Curtobacterium; s__Curtobacterium herbarum\n",
      "Coverage Percentage: 0.0235223445207158%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 366\n",
      "Length of fail 2960\n",
      "G016907365 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Cellulomonadaceae; g__Oerskovia; s__Oerskovia paurometabola\n",
      "Coverage Percentage: 0.0139607052375065%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 366\n",
      "Length of fail 2961\n",
      "G016907455 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Dermabacteraceae; g__Brachybacterium; s__Brachybacterium muris\n",
      "Coverage Percentage: 0.0069642580021792%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 366\n",
      "Length of fail 2962\n",
      "G016907475 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides cavernae\n",
      "Coverage Percentage: 0.0408042032654897%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 366\n",
      "Length of fail 2963\n",
      "G016907545 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Paenarthrobacter; s__Paenarthrobacter ilicis\n",
      "Coverage Percentage: 0.0121318200272185%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 366\n",
      "Length of fail 2964\n",
      "G016907555 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium laevaniformans\n",
      "Coverage Percentage: 0.4645129278434591%\n",
      "Adj, read counts 86\n",
      "Number of samples per group [1, 2, 1, 2, 1, 1, 2, 3, 1, 1, 1, 2, 1, 2, 1, 2, 1, 2, 3, 1, 1, 1, 1, 1, 2, 1, 2, 3, 3, 1, 2, 3, 1, 1, 1, 1, 3, 1, 2, 1, 1, 1, 1, 4, 3, 1, 1]\n",
      "Number of bins 66\n",
      "total # samples 36\n",
      "Number of total reads 85\n",
      "Number of hits in BIOM tables 197.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.790697674418606 %\n",
      "Percentage of reads from samples who only fall into one bin group 24.705882352941178 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 367\n",
      "Length of fail 2964\n",
      "G016907795 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Kocuria; s__Kocuria palustris\n",
      "Coverage Percentage: 0.3396872282571188%\n",
      "Adj, read counts 60\n",
      "Number of samples per group [1, 2, 2, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1]\n",
      "Number of bins 48\n",
      "total # samples 30\n",
      "Number of total reads 57\n",
      "Number of hits in BIOM tables 53.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 13.333333333333334 %\n",
      "Percentage of reads from samples who only fall into one bin group 36.84210526315789 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 368\n",
      "Length of fail 2964\n",
      "G016907815 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium aurum\n",
      "Coverage Percentage: 0.0855105106425127%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 368\n",
      "Length of fail 2965\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G016907895 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Alkalihalobacillus; s__Alkalihalobacillus xiaoxiensis\n",
      "Coverage Percentage: 0.0084777812435374%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 368\n",
      "Length of fail 2966\n",
      "G016908115 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Peptoniphilus; s__Peptoniphilus gorbachii\n",
      "Coverage Percentage: 0.0753119011564799%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 368\n",
      "Length of fail 2967\n",
      "G016908715 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Leuconostoc; s__Leuconostoc rapi\n",
      "Coverage Percentage: 0.0156853212919926%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 368\n",
      "Length of fail 2968\n",
      "G016909025 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Planococcaceae; g__Solibacillus; s__Solibacillus kalamii\n",
      "Coverage Percentage: 0.009794589342804%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 368\n",
      "Length of fail 2969\n",
      "G016909125 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Enterococcaceae; g__Enterococcus; s__Enterococcus lemanii\n",
      "Coverage Percentage: 0.0084871085964122%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 368\n",
      "Length of fail 2970\n",
      "G016918095 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Nakamurellales; f__Nakamurellaceae; g__Nakamurella; s__Nakamurella flavida\n",
      "Coverage Percentage: 0.0089723742475947%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 368\n",
      "Length of fail 2971\n",
      "G016918735 d__Bacteria; p__Spirochaetes; c__Spirochaetia; o__Leptospirales; f__Leptospiraceae; g__Leptospira; s__Leptospira abararensis\n",
      "Coverage Percentage: 0.0067016749400439%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 368\n",
      "Length of fail 2972\n",
      "G016937555 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas lactucae\n",
      "Coverage Percentage: 0.049292374036397%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 368\n",
      "Length of fail 2973\n",
      "G016937615 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas lactucae\n",
      "Coverage Percentage: 0.0432423938976487%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 368\n",
      "Length of fail 2974\n",
      "G017090145 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter baumannii\n",
      "Coverage Percentage: 0.0181774472156375%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 368\n",
      "Length of fail 2975\n",
      "G017132755 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium longum\n",
      "Coverage Percentage: 0.3659744604184348%\n",
      "Adj, read counts 50\n",
      "Number of samples per group [1, 2, 1, 2, 1, 1, 2, 1, 2, 1, 3, 2, 2, 2, 2, 1, 1, 5, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 38\n",
      "total # samples 23\n",
      "Number of total reads 50\n",
      "Number of hits in BIOM tables 6.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 24.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 20.0 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 369\n",
      "Length of fail 2975\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G017132775 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium longum\n",
      "Coverage Percentage: 0.393040258754277%\n",
      "Adj, read counts 58\n",
      "Number of samples per group [1, 2, 3, 1, 1, 2, 1, 1, 2, 1, 3, 1, 1, 1, 1, 2, 1, 1, 2, 2, 2, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 2, 2, 1]\n",
      "Number of bins 47\n",
      "total # samples 23\n",
      "Number of total reads 54\n",
      "Number of hits in BIOM tables 7.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.689655172413794 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.51851851851852 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 370\n",
      "Length of fail 2975\n",
      "G017161365 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptosporangiales; f__Streptosporangiaceae; g__Thermobispora; s__Thermobispora bispora\n",
      "Coverage Percentage: 0.0086166493725802%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 370\n",
      "Length of fail 2976\n",
      "G017167395 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Pseudoclavibacter; s__Pseudoclavibacter alba\n",
      "Coverage Percentage: 0.0289771080846131%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 370\n",
      "Length of fail 2977\n",
      "G017167925 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium organophilum\n",
      "Coverage Percentage: 0.0210112179507328%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 370\n",
      "Length of fail 2978\n",
      "G017167965 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas atacamensis\n",
      "Coverage Percentage: 0.0138045094508309%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 370\n",
      "Length of fail 2979\n",
      "G017167985 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas toyotomiensis\n",
      "Coverage Percentage: 0.1484110845698695%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 370\n",
      "Length of fail 2980\n",
      "G017254865 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Alteromonadales; f__Alteromonadaceae; g__Bowmanella; s__Bowmanella dokdonensis\n",
      "Coverage Percentage: 0.020189655420296%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 370\n",
      "Length of fail 2981\n",
      "G017298775 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia marcescens\n",
      "Coverage Percentage: 0.0182539461078401%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 370\n",
      "Length of fail 2982\n",
      "G017298915 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia marcescens\n",
      "Coverage Percentage: 0.0192424408018807%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 370\n",
      "Length of fail 2983\n",
      "G017298935 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia marcescens\n",
      "Coverage Percentage: 0.010464084066126%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 370\n",
      "Length of fail 2984\n",
      "G017298975 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia marcescens\n",
      "Coverage Percentage: 0.0182539461078401%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 370\n",
      "Length of fail 2985\n",
      "G017299015 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia marcescens\n",
      "Coverage Percentage: 0.0221682656626354%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 370\n",
      "Length of fail 2986\n",
      "G017299175 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia marcescens\n",
      "Coverage Percentage: 0.0192422692778277%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 370\n",
      "Length of fail 2987\n",
      "G017299535 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia marcescens\n",
      "Coverage Percentage: 0.0133901598021589%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 370\n",
      "Length of fail 2988\n",
      "G017299575 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia marcescens\n",
      "Coverage Percentage: 0.019242220804193%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 370\n",
      "Length of fail 2989\n",
      "G017301315 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia marcescens\n",
      "Coverage Percentage: 0.0133901286653781%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 370\n",
      "Length of fail 2990\n",
      "G017301775 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Xanthomonas; s__Xanthomonas translucens\n",
      "Coverage Percentage: 0.0150342852681426%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 370\n",
      "Length of fail 2991\n",
      "G017309445 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter indicus\n",
      "Coverage Percentage: 0.171959230198722%\n",
      "Adj, read counts 34\n",
      "Number of samples per group [1, 5, 6, 6, 1, 2, 1, 2, 3, 4, 1]\n",
      "Number of bins 19\n",
      "total # samples 17\n",
      "Number of total reads 34\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 52.94117647058824 %\n",
      "Percentage of reads from samples who only fall into one bin group 23.52941176470588 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 371\n",
      "Length of fail 2991\n",
      "G017309605 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia ureilytica\n",
      "Coverage Percentage: 0.0479331428680049%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 371\n",
      "Length of fail 2992\n",
      "G017309995 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides aromaticivorans\n",
      "Coverage Percentage: 0.0107093801974794%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 371\n",
      "Length of fail 2993\n",
      "G017329565 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Alicyclobacillaceae; g__Alicyclobacillus; s__Alicyclobacillus mali\n",
      "Coverage Percentage: 0.0107054767375662%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 371\n",
      "Length of fail 2994\n",
      "G017349355 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Thiotrichales; f__Thiotrichaceae; g__Thiothrix; s__Thiothrix fructosivorans\n",
      "Coverage Percentage: 0.0133404451759365%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 371\n",
      "Length of fail 2995\n",
      "G017355985 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Azospirillaceae; g__Arenibaculum; s__Arenibaculum pallidiluteum\n",
      "Coverage Percentage: 0.0115837662797944%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 371\n",
      "Length of fail 2996\n",
      "G017356265 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus fontiphilus\n",
      "Coverage Percentage: 0.0126765532427401%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 371\n",
      "Length of fail 2997\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G017357285 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium ruizarguesonis\n",
      "Coverage Percentage: 0.0130703274507472%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 371\n",
      "Length of fail 2998\n",
      "G017368815 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Comamonas; s__Comamonas denitrificans\n",
      "Coverage Percentage: 0.383478440030609%\n",
      "Adj, read counts 71\n",
      "Number of samples per group [1, 2, 6, 1, 1, 2, 1, 2, 2, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 7, 1, 2, 1, 1, 1, 1, 5, 1, 1, 2]\n",
      "Number of bins 56\n",
      "total # samples 30\n",
      "Number of total reads 71\n",
      "Number of hits in BIOM tables 100.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 28.169014084507044 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.901408450704224 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 372\n",
      "Length of fail 2998\n",
      "G017487405 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas goettingensis\n",
      "Coverage Percentage: 0.0932256449570948%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 372\n",
      "Length of fail 2999\n",
      "G017493175 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium quebecense\n",
      "Coverage Percentage: 0.0136331848931629%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 372\n",
      "Length of fail 3000\n",
      "G017498585 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter towneri\n",
      "Coverage Percentage: 0.1989731684450198%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 4, 3, 5, 3]\n",
      "Number of bins 26\n",
      "total # samples 15\n",
      "Number of total reads 36\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 37.83783783783784 %\n",
      "Percentage of reads from samples who only fall into one bin group 22.22222222222222 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 373\n",
      "Length of fail 3000\n",
      "G017504145 d__Bacteria; p__Verrucomicrobia; c__Verrucomicrobiae; o__Verrucomicrobiales; f__Akkermansiaceae; g__Akkermansia; s__Akkermansia muciniphila\n",
      "Coverage Percentage: 0.090276049520073%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 373\n",
      "Length of fail 3001\n",
      "G017582745 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia oculi\n",
      "Coverage Percentage: 0.0152568821439509%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 373\n",
      "Length of fail 3002\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G017615005 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Pseudonocardia; s__Pseudonocardia alni\n",
      "Coverage Percentage: 0.963905023734264%\n",
      "Adj, read counts 301\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 3, 1, 1, 3, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 3, 3, 1, 2, 2, 2, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 2, 1, 1, 2, 1, 1, 2, 1, 2, 3, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 2, 2, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 2, 2, 1, 3, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 3, 1, 2, 1, 2, 1, 1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 1, 2, 1]\n",
      "Number of bins 206\n",
      "total # samples 9\n",
      "Number of total reads 298\n",
      "Number of hits in BIOM tables 434.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 4.318936877076411 %\n",
      "Percentage of reads from samples who only fall into one bin group 1.6778523489932886 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3002\n",
      "G017811175 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Melaminivora; s__Melaminivora jejuensis\n",
      "Coverage Percentage: 0.0783520815500822%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3003\n",
      "G017826695 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas alliivorans\n",
      "Coverage Percentage: 0.0133385637397565%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3004\n",
      "G017831975 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium terrae\n",
      "Coverage Percentage: 0.0269280233701358%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3005\n",
      "G017848315 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas palmensis\n",
      "Coverage Percentage: 0.0095099531304836%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3006\n",
      "G017873055 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Agrobacterium; s__Agrobacterium rubi\n",
      "Coverage Percentage: 0.0053853928870864%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3007\n",
      "G017873755 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Jeotgalicoccus; s__Jeotgalicoccus aerolatus\n",
      "Coverage Percentage: 0.0143693542593002%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3008\n",
      "G017874535 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Anaerococcus; s__Anaerococcus nagyae\n",
      "Coverage Percentage: 0.0423581259669012%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3009\n",
      "G017875275 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Xanthobacteraceae; g__Xanthobacter; s__Xanthobacter flavus\n",
      "Coverage Percentage: 0.0205826457665027%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3010\n",
      "G017876055 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Azospirillaceae; g__Azospirillum; s__Azospirillum melinis\n",
      "Coverage Percentage: 0.0037972624001363%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3011\n",
      "G017876095 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Azospirillaceae; g__Azospirillum; s__Azospirillum agricola\n",
      "Coverage Percentage: 0.0069833623693036%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3012\n",
      "G017876315 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Pseudonocardia; s__Pseudonocardia parietis\n",
      "Coverage Percentage: 0.020232531350578%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3013\n",
      "G017876395 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Nesterenkonia; s__Nesterenkonia lacusekhoensis\n",
      "Coverage Percentage: 0.0319761516974888%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3014\n",
      "G017876435 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium phyllosphaerae\n",
      "Coverage Percentage: 0.046531586499135%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3015\n",
      "G017876445 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Paenarthrobacter; s__Paenarthrobacter nicotinovorans\n",
      "Coverage Percentage: 0.0717577966193016%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3016\n",
      "G017876495 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Microlunatus; s__Microlunatus capsulatus\n",
      "Coverage Percentage: 0.0265763072221114%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3017\n",
      "G017876515 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Dermabacteraceae; g__Brachybacterium; s__Brachybacterium fresconis\n",
      "Coverage Percentage: 0.0097861140010991%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3018\n",
      "G017876545 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Dermabacteraceae; g__Brachybacterium; s__Brachybacterium sacelli\n",
      "Coverage Percentage: 0.0062023451436664%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3019\n",
      "G017876595 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Aeromicrobium; s__Aeromicrobium fastidiosum\n",
      "Coverage Percentage: 0.0134750120039686%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3020\n",
      "G017876775 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacterium; s__Mycolicibacterium lutetiense\n",
      "Coverage Percentage: 0.0098340054550362%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3021\n",
      "G017893965 d__Bacteria; p__Spirochaetes; c__Spirochaetia; o__Spirochaetales; f__Treponemataceae; g__Treponema; s__Treponema parvum\n",
      "Coverage Percentage: 0.018921959893721%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3022\n",
      "G017901175 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Thiotrichales; f__Thiotrichaceae; g__Thiothrix; s__Thiothrix unzii\n",
      "Coverage Percentage: 0.0535100785816848%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3023\n",
      "G017905455 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas chengduensis\n",
      "Coverage Percentage: 0.1851117501499769%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3024\n",
      "G017909455 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Curtobacterium; s__Curtobacterium flaccumfaciens\n",
      "Coverage Percentage: 0.0556848453308513%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3025\n",
      "G017939625 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Pseudoxanthomonas; s__Pseudoxanthomonas helianthi\n",
      "Coverage Percentage: 0.0035947276213575%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3026\n",
      "G017939745 d__Bacteria; p__Planctomycetes; c__Planctomycetia; o__Gemmatales; f__Gemmataceae; g__Gemmata; s__Gemmata palustris\n",
      "Coverage Percentage: 0.0032725346246624%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3027\n",
      "G017948305 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Faecalibacter; s__Faecalibacter bovis\n",
      "Coverage Percentage: 0.0063134096468194%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3028\n",
      "G017948595 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium geliluteum\n",
      "Coverage Percentage: 0.0509908770818708%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3029\n",
      "G017948675 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium flabelliforme\n",
      "Coverage Percentage: 0.0516192618307906%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3030\n",
      "G017973435 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Mammaliicoccus; s__Mammaliicoccus lentus\n",
      "Coverage Percentage: 0.006667816776862%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3031\n",
      "G017980685 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas iridis\n",
      "Coverage Percentage: 0.0050889093119289%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3032\n",
      "G018064205 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Comamonas; s__Comamonas aquatica\n",
      "Coverage Percentage: 0.1098965883684967%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3033\n",
      "G018069865 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Ideonella; s__Ideonella paludis\n",
      "Coverage Percentage: 0.0590300451265323%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3034\n",
      "G018069925 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium erciyesense\n",
      "Coverage Percentage: 0.0419611127353287%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3035\n",
      "G018094625 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactobacillus; s__Lactobacillus jensenii\n",
      "Coverage Percentage: 0.0079740971653011%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3036\n",
      "G018127785 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella scopos\n",
      "Coverage Percentage: 0.0266615421629138%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3037\n",
      "G018128145 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella fusca\n",
      "Coverage Percentage: 0.0047826236246891%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3038\n",
      "G018128265 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Promicromonosporaceae; g__Cellulosimicrobium; s__Cellulosimicrobium cellulans\n",
      "Coverage Percentage: 0.00528149698932%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3039\n",
      "G018128325 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Arachnia; s__Arachnia rubra\n",
      "Coverage Percentage: 0.0160689402759998%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3040\n",
      "G018129025 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Roseomonas; s__Roseomonas terrae\n",
      "Coverage Percentage: 0.0026158831923268%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3041\n",
      "G018129445 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium viridifuturi\n",
      "Coverage Percentage: 0.0063088845350274%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3042\n",
      "G018129525 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium denitrificans\n",
      "Coverage Percentage: 0.0178399765867958%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3043\n",
      "G018130265 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium ottawaense\n",
      "Coverage Percentage: 0.0087524798692963%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3044\n",
      "G018130825 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium liaoningense\n",
      "Coverage Percentage: 0.0112166666295253%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3045\n",
      "G018137965 d__Bacteria; p__Verrucomicrobia; c__Verrucomicrobiae; o__Verrucomicrobiales; f__Verrucomicrobiaceae; g__Luteolibacter; s__Luteolibacter ambystomatis\n",
      "Coverage Percentage: 0.0111800327246134%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3046\n",
      "G018139565 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Undibacterium; s__Undibacterium rivi\n",
      "Coverage Percentage: 0.0274901844471409%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3047\n",
      "G018139645 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Undibacterium; s__Undibacterium baiyunense\n",
      "Coverage Percentage: 0.0165482890543717%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3048\n",
      "G018219815 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Roseobacteraceae; g__Thalassobius; s__Thalassobius aquimarinus\n",
      "Coverage Percentage: 0.0034836774714499%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3049\n",
      "G018274385 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Catenulisporales; f__Catenulisporaceae; g__Catenulispora; s__Catenulispora pinistramenti\n",
      "Coverage Percentage: 0.0024829080097392%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3050\n",
      "G018286635 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium phenoxybenzoativorans\n",
      "Coverage Percentage: 0.0115350572548883%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 374\n",
      "Length of fail 3051\n",
      "G018289355 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola vulgatus\n",
      "Coverage Percentage: 0.5157130981213712%\n",
      "Adj, read counts 169\n",
      "Number of samples per group [3, 1, 2, 1, 1, 1, 1, 1, 3, 1, 2, 1, 2, 1, 1, 1, 3, 1, 1, 1, 4, 1, 2, 2, 1, 1, 3, 1, 1, 1, 2, 1, 1, 2, 1, 3, 1, 1, 4, 1, 1, 1, 2, 1, 1, 2, 1, 2, 2, 2, 1, 3, 1, 1, 1, 2, 1, 3, 3, 1, 1, 1, 1, 2, 1, 2, 3, 1, 1, 1, 1, 2, 1, 1, 1, 3, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 2, 2, 2, 2, 3, 1, 1, 1, 2]\n",
      "Number of bins 141\n",
      "total # samples 47\n",
      "Number of total reads 166\n",
      "Number of hits in BIOM tables 70.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 7.6923076923076925 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.44578313253012 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 375\n",
      "Length of fail 3051\n",
      "G018292125 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides cellulosilyticus\n",
      "Coverage Percentage: 0.1020889099096658%\n",
      "Adj, read counts 43\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1]\n",
      "Number of bins 37\n",
      "total # samples 23\n",
      "Number of total reads 42\n",
      "Number of hits in BIOM tables 25.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 13.953488372093023 %\n",
      "Percentage of reads from samples who only fall into one bin group 40.476190476190474 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 376\n",
      "Length of fail 3051\n",
      "G018292165 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides uniformis\n",
      "Coverage Percentage: 0.2727745178480222%\n",
      "Adj, read counts 84\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 2, 2, 1, 1, 1, 2, 1, 2, 3, 1, 2, 1, 1, 2, 1, 1, 2, 1, 1]\n",
      "Number of bins 74\n",
      "total # samples 42\n",
      "Number of total reads 82\n",
      "Number of hits in BIOM tables 34.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.714285714285714 %\n",
      "Percentage of reads from samples who only fall into one bin group 37.80487804878049 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 377\n",
      "Length of fail 3051\n",
      "G018304965 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Collimonas; s__Collimonas antrihumi\n",
      "Coverage Percentage: 0.0191783698470302%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 377\n",
      "Length of fail 3052\n",
      "G018324105 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter bouvetii\n",
      "Coverage Percentage: 0.0691035803538726%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 377\n",
      "Length of fail 3053\n",
      "G018324205 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes ianthinogenes\n",
      "Coverage Percentage: 0.0273798051810673%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 377\n",
      "Length of fail 3054\n",
      "G018324255 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Polymorphospora; s__Polymorphospora rubra\n",
      "Coverage Percentage: 0.0036101894369204%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 377\n",
      "Length of fail 3055\n",
      "G018332655 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes auranticolor\n",
      "Coverage Percentage: 0.0081869714014617%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 377\n",
      "Length of fail 3056\n",
      "G018332695 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes toevensis\n",
      "Coverage Percentage: 0.0137971790932285%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 377\n",
      "Length of fail 3057\n",
      "G018332715 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes consettensis\n",
      "Coverage Percentage: 0.0205861685266326%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 377\n",
      "Length of fail 3058\n",
      "G018359905 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas veronii\n",
      "Coverage Percentage: 0.0470544825270873%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 377\n",
      "Length of fail 3059\n",
      "G018359945 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas sobria\n",
      "Coverage Percentage: 0.1315374316730197%\n",
      "Adj, read counts 31\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1]\n",
      "Number of bins 27\n",
      "total # samples 9\n",
      "Number of total reads 31\n",
      "Number of hits in BIOM tables 23.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 29.032258064516125 %\n",
      "Percentage of reads from samples who only fall into one bin group 25.806451612903224 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 378\n",
      "Length of fail 3059\n",
      "G018359985 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas media\n",
      "Coverage Percentage: 0.2778607427862173%\n",
      "Adj, read counts 75\n",
      "Number of samples per group [1, 1, 1, 1, 2, 1, 1, 4, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 4, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 2]\n",
      "Number of bins 65\n",
      "total # samples 28\n",
      "Number of total reads 73\n",
      "Number of hits in BIOM tables 43.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 13.333333333333334 %\n",
      "Percentage of reads from samples who only fall into one bin group 20.54794520547945 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 379\n",
      "Length of fail 3059\n",
      "G018360005 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas caviae\n",
      "Coverage Percentage: 0.0454804678590918%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 379\n",
      "Length of fail 3060\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G018360025 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas hydrophila\n",
      "Coverage Percentage: 0.0346697974155941%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 379\n",
      "Length of fail 3061\n",
      "G018380615 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium psychroterrae\n",
      "Coverage Percentage: 0.0757785461703003%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 379\n",
      "Length of fail 3062\n",
      "G018383905 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium branchiicola\n",
      "Coverage Percentage: 0.0356091241887088%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 379\n",
      "Length of fail 3063\n",
      "G018390605 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Xanthobacteraceae; g__Ancylobacter; s__Ancylobacter defluvii\n",
      "Coverage Percentage: 0.0050368256334281%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 379\n",
      "Length of fail 3064\n",
      "G018390645 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Xanthobacteraceae; g__Ancylobacter; s__Ancylobacter dichloromethanicus\n",
      "Coverage Percentage: 0.0077202694295252%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 379\n",
      "Length of fail 3065\n",
      "G018390695 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Xanthobacteraceae; g__Ancylobacter; s__Ancylobacter sonchi\n",
      "Coverage Percentage: 0.001997980661614%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 379\n",
      "Length of fail 3066\n",
      "G018398275 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Chelatococcaceae; g__Chelatococcus; s__Chelatococcus asaccharovorans\n",
      "Coverage Percentage: 0.0046987601574728%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 379\n",
      "Length of fail 3067\n",
      "G018398355 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Chelatococcaceae; g__Chelatococcus; s__Chelatococcus composti\n",
      "Coverage Percentage: 0.0270629638269059%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 379\n",
      "Length of fail 3068\n",
      "G018398425 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas lalucatii\n",
      "Coverage Percentage: 0.0065893573587411%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 379\n",
      "Length of fail 3069\n",
      "G018398475 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas qingdaonensis\n",
      "Coverage Percentage: 0.0192962135474573%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 379\n",
      "Length of fail 3070\n",
      "G018409485 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter variabilis\n",
      "Coverage Percentage: 0.1774937211244416%\n",
      "Adj, read counts 39\n",
      "Number of samples per group [1, 4, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 5, 1, 2, 1, 1, 1]\n",
      "Number of bins 30\n",
      "total # samples 25\n",
      "Number of total reads 37\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 33.33333333333333 %\n",
      "Percentage of reads from samples who only fall into one bin group 56.75675675675676 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 380\n",
      "Length of fail 3070\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G018459925 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides aquaticus\n",
      "Coverage Percentage: 0.033749460248768%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 380\n",
      "Length of fail 3071\n",
      "G018499185 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus infantarius\n",
      "Coverage Percentage: 0.0505357056496201%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 380\n",
      "Length of fail 3072\n",
      "G018551565 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactiplantibacillus; s__Lactiplantibacillus argentoratensis\n",
      "Coverage Percentage: 0.0031489338654611%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 380\n",
      "Length of fail 3073\n",
      "G018597265 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Enterobacter; s__Enterobacter dykesii\n",
      "Coverage Percentage: 0.0159447752066051%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 380\n",
      "Length of fail 3074\n",
      "G018597345 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Enterobacter; s__Enterobacter quasimori\n",
      "Coverage Percentage: 0.0123046210424535%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 380\n",
      "Length of fail 3075\n",
      "G018604145 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Paenibacillaceae; g__Paenibacillus; s__Paenibacillus agaridevorans\n",
      "Coverage Percentage: 0.0016699887646252%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 380\n",
      "Length of fail 3076\n",
      "G018682675 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingopyxis; s__Sphingopyxis soli\n",
      "Coverage Percentage: 0.0213159563823273%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 380\n",
      "Length of fail 3077\n",
      "G018685455 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Polynucleobacter; s__Polynucleobacter paneuropaeus\n",
      "Coverage Percentage: 0.1159473560373981%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 380\n",
      "Length of fail 3078\n",
      "G018717645 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium flavum\n",
      "Coverage Percentage: 0.049255901583887%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 380\n",
      "Length of fail 3079\n",
      "G018729305 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Intrasporangiaceae; g__Phycicoccus; s__Phycicoccus mangrovi\n",
      "Coverage Percentage: 0.0101026017552153%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 380\n",
      "Length of fail 3080\n",
      "G018729355 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Gemmobacter; s__Gemmobacter fulva\n",
      "Coverage Percentage: 0.0377754312877139%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 380\n",
      "Length of fail 3081\n",
      "G018785545 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Blautia; s__Blautia wexlerae\n",
      "Coverage Percentage: 0.2839010205276976%\n",
      "Adj, read counts 79\n",
      "Number of samples per group [3, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 3, 1, 1, 1, 1, 1, 1, 2, 3, 1, 2, 1, 2, 2, 2, 1, 1, 1, 1, 2, 1, 1, 1, 2, 2, 2]\n",
      "Number of bins 67\n",
      "total # samples 32\n",
      "Number of total reads 78\n",
      "Number of hits in BIOM tables 67.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.39240506329114 %\n",
      "Percentage of reads from samples who only fall into one bin group 12.82051282051282 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 381\n",
      "Length of fail 3081\n",
      "G018853545 d__Bacteria; p__Proteobacteria; c__Acidithiobacillia; o__Acidithiobacillales; f__Acidithiobacillaceae; g__Acidithiobacillus; s__Acidithiobacillus caldus\n",
      "Coverage Percentage: 0.0210680746465603%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 381\n",
      "Length of fail 3082\n",
      "G018853615 d__Bacteria; p__Proteobacteria; c__Acidithiobacillia; o__Acidithiobacillales; f__Acidithiobacillaceae; g__Acidithiobacillus; s__Acidithiobacillus caldus\n",
      "Coverage Percentage: 0.0199553119950257%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 381\n",
      "Length of fail 3083\n",
      "G018854775 d__Bacteria; p__Proteobacteria; c__Acidithiobacillia; o__Acidithiobacillales; f__Acidithiobacillaceae; g__Acidithiobacillus; s__Acidithiobacillus ferriphilus\n",
      "Coverage Percentage: 0.0104413467542706%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 381\n",
      "Length of fail 3084\n",
      "G018866205 d__Bacteria; p__Candidatus Saccharibacteria; c__; o__; f__; g__Candidatus Minimicrobia; s__Candidatus Minimicrobia vallesae\n",
      "Coverage Percentage: 0.0213586657481968%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 381\n",
      "Length of fail 3085\n",
      "G018866245 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Novisyntrophococcus; s__Novisyntrophococcus fermenticellae\n",
      "Coverage Percentage: 0.0176591612307733%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 381\n",
      "Length of fail 3086\n",
      "G018885085 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Peptostreptococcaceae; g__Clostridioides; s__Clostridioides difficile\n",
      "Coverage Percentage: 0.0320807516954701%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 381\n",
      "Length of fail 3087\n",
      "G018917425 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Caldibacillus; s__Caldibacillus hisashii\n",
      "Coverage Percentage: 0.0317838000300556%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 381\n",
      "Length of fail 3088\n",
      "G018986875 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus parasuis\n",
      "Coverage Percentage: 0.0381970273778789%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 381\n",
      "Length of fail 3089\n",
      "G018987235 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactobacillus; s__Lactobacillus crispatus\n",
      "Coverage Percentage: 0.090170600632668%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 381\n",
      "Length of fail 3090\n",
      "G019024325 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Roseomonas; s__Roseomonas tokyonensis\n",
      "Coverage Percentage: 0.0044122379909263%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 381\n",
      "Length of fail 3091\n",
      "G019042115 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium ruminantium\n",
      "Coverage Percentage: 0.1611178421953499%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 381\n",
      "Length of fail 3092\n",
      "G019047025 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Enterobacter; s__Enterobacter roggenkampii\n",
      "Coverage Percentage: 0.0225732899817705%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 381\n",
      "Length of fail 3093\n",
      "G019047205 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter pittii\n",
      "Coverage Percentage: 0.039771930714935%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 381\n",
      "Length of fail 3094\n",
      "G019047825 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus mitis\n",
      "Coverage Percentage: 0.0689725656135642%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 381\n",
      "Length of fail 3095\n",
      "G019048125 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Klebsiella; s__Klebsiella aerogenes\n",
      "Coverage Percentage: 0.0059260504634436%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 381\n",
      "Length of fail 3096\n",
      "G019048165 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium coyleae\n",
      "Coverage Percentage: 0.0442185095494054%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 381\n",
      "Length of fail 3097\n",
      "G019048305 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter lwoffii\n",
      "Coverage Percentage: 0.6972025439874769%\n",
      "Adj, read counts 159\n",
      "Number of samples per group [6, 1, 2, 3, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 5, 1, 2, 2, 1, 2, 1, 3, 5, 5, 5, 1, 1, 1, 4, 4, 3, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 4, 1, 4, 1, 2, 1, 15, 8, 2, 6, 3]\n",
      "Number of bins 100\n",
      "total # samples 36\n",
      "Number of total reads 157\n",
      "Number of hits in BIOM tables 33.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 28.930817610062892 %\n",
      "Percentage of reads from samples who only fall into one bin group 4.45859872611465 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 382\n",
      "Length of fail 3097\n",
      "G019048385 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Erwiniaceae; g__Pantoea; s__Pantoea agglomerans\n",
      "Coverage Percentage: 0.0597818697040829%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 382\n",
      "Length of fail 3098\n",
      "G019048465 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Jonesiaceae; g__Jonesia; s__Jonesia denitrificans\n",
      "Coverage Percentage: 0.0196752600152892%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 382\n",
      "Length of fail 3099\n",
      "G019048865 d__Bacteria; p__Actinobacteria; c__Acidimicrobiia; o__Acidimicrobiales; f__Iamiaceae; g__Aquihabitans; s__Aquihabitans sp. G128\n",
      "Coverage Percentage: 0.0054262402856632%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 382\n",
      "Length of fail 3100\n",
      "G019049775 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas shirazica\n",
      "Coverage Percentage: 0.0806188465986842%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 382\n",
      "Length of fail 3101\n",
      "G019052655 d__Bacteria; p__Firmicutes; c__Erysipelotrichia; o__Erysipelotrichales; f__Erysipelotrichaceae; g__Holdemanella; s__Holdemanella porci\n",
      "Coverage Percentage: 0.0676193541618134%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 382\n",
      "Length of fail 3102\n",
      "G019130295 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides eggerthii\n",
      "Coverage Percentage: 0.0336895966207558%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 382\n",
      "Length of fail 3103\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G019132845 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Advenella; s__Advenella alkanexedens\n",
      "Coverage Percentage: 0.0100946625307519%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 382\n",
      "Length of fail 3104\n",
      "G019139675 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas maumuensis\n",
      "Coverage Percentage: 0.007866866909978%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 382\n",
      "Length of fail 3105\n",
      "G019141545 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus marcusii\n",
      "Coverage Percentage: 0.3397732899201735%\n",
      "Adj, read counts 79\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 1, 3, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 2, 2, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 70\n",
      "total # samples 16\n",
      "Number of total reads 78\n",
      "Number of hits in BIOM tables 80.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.126582278481013 %\n",
      "Percentage of reads from samples who only fall into one bin group 8.974358974358974 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 383\n",
      "Length of fail 3105\n",
      "G019145205 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas khavaziana\n",
      "Coverage Percentage: 0.0172026354746114%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 383\n",
      "Length of fail 3106\n",
      "G019145235 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas farris\n",
      "Coverage Percentage: 0.0153522035961149%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 383\n",
      "Length of fail 3107\n",
      "G019145375 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas triticicola\n",
      "Coverage Percentage: 0.0199277139356049%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 383\n",
      "Length of fail 3108\n",
      "G019249635 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella copri\n",
      "Coverage Percentage: 0.1885343300623819%\n",
      "Adj, read counts 39\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 1, 3, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1]\n",
      "Number of bins 35\n",
      "total # samples 21\n",
      "Number of total reads 39\n",
      "Number of hits in BIOM tables 21.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.948717948717945 %\n",
      "Percentage of reads from samples who only fall into one bin group 28.205128205128204 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 384\n",
      "Length of fail 3108\n",
      "G019263585 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Pseudonocardia; s__Pseudonocardia oceani\n",
      "Coverage Percentage: 0.008853909756085%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 384\n",
      "Length of fail 3109\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G019278975 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Carnobacterium; s__Carnobacterium maltaromaticum\n",
      "Coverage Percentage: 0.0463079733022468%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 384\n",
      "Length of fail 3110\n",
      "G019285595 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Legionellales; f__Coxiellaceae; g__Rickettsiella; s__Rickettsiella endosymbiont of Dermanyssus gallinae\n",
      "Coverage Percentage: 0.0733302801110808%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 384\n",
      "Length of fail 3111\n",
      "G019321695 d__Eukaryota; p__Ascomycota; c__Sordariomycetes; o__Diaporthales; f__Diaporthaceae; g__Diaporthe; s__Diaporthe batatas\n",
      "Coverage Percentage: 0.0004613667110372%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 384\n",
      "Length of fail 3112\n",
      "G019331655 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter baumannii\n",
      "Coverage Percentage: 0.028314623374951%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 384\n",
      "Length of fail 3113\n",
      "G019355975 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter septicus\n",
      "Coverage Percentage: 0.3665741186393523%\n",
      "Adj, read counts 105\n",
      "Number of samples per group [5, 1, 5, 6, 1, 1, 4, 1, 2, 1, 10, 3, 1, 2, 3, 1, 2, 5, 17, 3, 2, 1, 5, 2]\n",
      "Number of bins 44\n",
      "total # samples 34\n",
      "Number of total reads 103\n",
      "Number of hits in BIOM tables 8.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 48.57142857142857 %\n",
      "Percentage of reads from samples who only fall into one bin group 24.271844660194176 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 385\n",
      "Length of fail 3113\n",
      "G019355995 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter haemolyticus\n",
      "Coverage Percentage: 0.1419102249291227%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 385\n",
      "Length of fail 3114\n",
      "G019395145 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas nasdae\n",
      "Coverage Percentage: 0.0700008001585673%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 385\n",
      "Length of fail 3115\n",
      "G019400025 d__Bacteria; p__Proteobacteria; c__Acidithiobacillia; o__Acidithiobacillales; f__Acidithiobacillaceae; g__Acidithiobacillus; s__Acidithiobacillus ferriphilus\n",
      "Coverage Percentage: 0.067526483096725%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 385\n",
      "Length of fail 3116\n",
      "G019402665 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium canariense\n",
      "Coverage Percentage: 0.0127852870419539%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 385\n",
      "Length of fail 3117\n",
      "G019429605 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes hulinensis\n",
      "Coverage Percentage: 0.0154310446238144%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 385\n",
      "Length of fail 3118\n",
      "G019443745 d__Bacteria; p__Actinobacteria; c__Acidimicrobiia; o__Acidimicrobiales; f__Iamiaceae; g__Iamia; s__Iamia sp. SCSIO 61187\n",
      "Coverage Percentage: 0.0051050983550534%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 385\n",
      "Length of fail 3119\n",
      "G019511525 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium cauense\n",
      "Coverage Percentage: 0.0021606053815958%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 385\n",
      "Length of fail 3120\n",
      "G019552185 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Riemerella; s__Riemerella anatipestifer\n",
      "Coverage Percentage: 0.0117672421123854%\n",
      "Adj, read counts 57\n",
      "Number of samples per group [37, 20]\n",
      "Number of bins 2\n",
      "total # samples 41\n",
      "Number of total reads 57\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 100.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 43.859649122807014 %\n",
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 385\n",
      "Length of fail 3121\n",
      "G019586025 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Lactococcus; s__Lactococcus laudensis\n",
      "Coverage Percentage: 0.044225917523908%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 385\n",
      "Length of fail 3122\n",
      "G019633635 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium glucuronolyticum\n",
      "Coverage Percentage: 0.2373250626889705%\n",
      "Adj, read counts 39\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 2, 1, 2, 1, 1]\n",
      "Number of bins 32\n",
      "total # samples 18\n",
      "Number of total reads 38\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.948717948717945 %\n",
      "Percentage of reads from samples who only fall into one bin group 26.31578947368421 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 386\n",
      "Length of fail 3122\n",
      "G019655335 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Alteromonadales; f__Shewanellaceae; g__Shewanella; s__Shewanella hafniensis\n",
      "Coverage Percentage: 0.0272965841251756%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 386\n",
      "Length of fail 3123\n",
      "G019655915 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Leuconostoc; s__Leuconostoc carnosum\n",
      "Coverage Percentage: 0.017473939528597%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 386\n",
      "Length of fail 3124\n",
      "G019656095 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Leuconostoc; s__Leuconostoc rapi\n",
      "Coverage Percentage: 0.0156986116085514%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 386\n",
      "Length of fail 3125\n",
      "G019693255 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas mandelii\n",
      "Coverage Percentage: 0.0131708191538378%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 386\n",
      "Length of fail 3126\n",
      "G019702405 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas chloritidismutans\n",
      "Coverage Percentage: 0.0230952336074213%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 386\n",
      "Length of fail 3127\n",
      "G019703875 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Methylophilaceae; g__Methyloradius; s__Methyloradius palustris\n",
      "Coverage Percentage: 0.0269960180873321%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 386\n",
      "Length of fail 3128\n",
      "G019704495 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas pavanii\n",
      "Coverage Percentage: 0.341612255572596%\n",
      "Adj, read counts 98\n",
      "Number of samples per group [2, 1, 3, 2, 1, 2, 2, 3, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 2, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 3, 1, 1]\n",
      "Number of bins 83\n",
      "total # samples 30\n",
      "Number of total reads 97\n",
      "Number of hits in BIOM tables 22.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.204081632653061 %\n",
      "Percentage of reads from samples who only fall into one bin group 19.587628865979383 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 387\n",
      "Length of fail 3128\n",
      "G019704535 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas stutzeri\n",
      "Coverage Percentage: 0.1132722321779375%\n",
      "Adj, read counts 32\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 29\n",
      "total # samples 11\n",
      "Number of total reads 32\n",
      "Number of hits in BIOM tables 17.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 21.875 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.75 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 388\n",
      "Length of fail 3128\n",
      "G019734885 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Coprococcus; s__Coprococcus catus\n",
      "Coverage Percentage: 0.0826585156185987%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 388\n",
      "Length of fail 3129\n",
      "G019737615 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium xanthum\n",
      "Coverage Percentage: 0.0037159233220452%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 388\n",
      "Length of fail 3130\n",
      "G019740335 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Aquisediminimonas; s__Aquisediminimonas sediminicola\n",
      "Coverage Percentage: 0.015045070447795%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 388\n",
      "Length of fail 3131\n",
      "G019778565 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium laguerreae\n",
      "Coverage Percentage: 0.0160786147763693%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 388\n",
      "Length of fail 3132\n",
      "G019779905 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Modestobacter; s__Modestobacter lapidis\n",
      "Coverage Percentage: 0.012488137303348%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 388\n",
      "Length of fail 3133\n",
      "G019793415 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Ottowia; s__Ottowia caeni\n",
      "Coverage Percentage: 0.0376956394902682%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 388\n",
      "Length of fail 3134\n",
      "G019795205 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Cupriavidus; s__Cupriavidus pauculus\n",
      "Coverage Percentage: 0.0391827613595546%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 388\n",
      "Length of fail 3135\n",
      "G019795235 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Cupriavidus; s__Cupriavidus respiraculi\n",
      "Coverage Percentage: 0.029859649110481%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 388\n",
      "Length of fail 3136\n",
      "G019823065 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas mosselii\n",
      "Coverage Percentage: 0.0675199550224131%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 388\n",
      "Length of fail 3137\n",
      "G019840805 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Escherichia; s__Escherichia ruysiae\n",
      "Coverage Percentage: 0.2630536490218989%\n",
      "Adj, read counts 77\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 3, 1, 2, 2, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 66\n",
      "total # samples 27\n",
      "Number of total reads 74\n",
      "Number of hits in BIOM tables 6.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.38961038961039 %\n",
      "Percentage of reads from samples who only fall into one bin group 24.324324324324326 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 389\n",
      "Length of fail 3137\n",
      "G019844095 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Erwiniaceae; g__Erwinia; s__Erwinia persicina\n",
      "Coverage Percentage: 0.0053092425584719%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 389\n",
      "Length of fail 3138\n",
      "G019890915 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Micrococcus; s__Micrococcus luteus\n",
      "Coverage Percentage: 0.7087865950882623%\n",
      "Adj, read counts 100\n",
      "Number of samples per group [5, 2, 1, 1, 4, 1, 1, 1, 1, 2, 1, 2, 1, 3, 2, 3, 1, 1, 2, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 3, 1, 2, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1]\n",
      "Number of bins 68\n",
      "total # samples 21\n",
      "Number of total reads 99\n",
      "Number of hits in BIOM tables 45.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.000000000000002 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.161616161616163 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 390\n",
      "Length of fail 3138\n",
      "G019898765 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas yunnanensis\n",
      "Coverage Percentage: 0.0134964180506493%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 390\n",
      "Length of fail 3139\n",
      "G019931045 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Cupriavidus; s__Cupriavidus pauculus\n",
      "Coverage Percentage: 0.0344260785565478%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 390\n",
      "Length of fail 3140\n",
      "G019931675 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides rotundus\n",
      "Coverage Percentage: 0.0173654890011639%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 390\n",
      "Length of fail 3141\n",
      "G019972875 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus thermophilus\n",
      "Coverage Percentage: 0.0920698581906267%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 390\n",
      "Length of fail 3142\n",
      "G019972895 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus salivarius\n",
      "Coverage Percentage: 0.0674630256936301%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 390\n",
      "Length of fail 3143\n",
      "G019973715 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Thermales; f__Thermaceae; g__Thermus; s__Thermus thermophilus\n",
      "Coverage Percentage: 0.1745183378935877%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 390\n",
      "Length of fail 3144\n",
      "G019973735 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Arachnia; s__Arachnia rubra\n",
      "Coverage Percentage: 0.016068872453551%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 390\n",
      "Length of fail 3145\n",
      "G020026925 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium hibisci\n",
      "Coverage Percentage: 0.0487455498465687%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 390\n",
      "Length of fail 3146\n",
      "G020091585 d__Bacteria; p__Cyanobacteria; c__; o__Pseudanabaenales; f__Leptolyngbyaceae; g__Kovacikia; s__Kovacikia sp. CCNU0001\n",
      "Coverage Percentage: 0.0038531703885957%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 390\n",
      "Length of fail 3147\n",
      "G020092905 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingosinicellaceae; g__Pacificimonas; s__Pacificimonas aurantium\n",
      "Coverage Percentage: 0.0107790313502602%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 390\n",
      "Length of fail 3148\n",
      "G020097395 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Leclercia; s__Leclercia adecarboxylata\n",
      "Coverage Percentage: 0.0179548727606585%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 390\n",
      "Length of fail 3149\n",
      "G020097475 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Escherichia; s__Escherichia fergusonii\n",
      "Coverage Percentage: 0.3052982943416499%\n",
      "Adj, read counts 83\n",
      "Number of samples per group [1, 1, 2, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 5, 1, 1, 1, 1, 1, 2, 1, 2, 2, 2, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 72\n",
      "total # samples 37\n",
      "Number of total reads 83\n",
      "Number of hits in BIOM tables 7.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.048192771084338 %\n",
      "Percentage of reads from samples who only fall into one bin group 32.53012048192771 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 391\n",
      "Length of fail 3149\n",
      "G020099175 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Klebsiella; s__Klebsiella quasipneumoniae\n",
      "Coverage Percentage: 0.0626672156261411%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 391\n",
      "Length of fail 3150\n",
      "G020115515 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Klebsiella; s__Klebsiella quasipneumoniae\n",
      "Coverage Percentage: 0.0556666438855978%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 391\n",
      "Length of fail 3151\n",
      "G020149745 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides ovatus\n",
      "Coverage Percentage: 0.1247634653782937%\n",
      "Adj, read counts 51\n",
      "Number of samples per group [2, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 47\n",
      "total # samples 32\n",
      "Number of total reads 49\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.686274509803921 %\n",
      "Percentage of reads from samples who only fall into one bin group 42.857142857142854 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 392\n",
      "Length of fail 3151\n",
      "G020162115 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Tepidimonas; s__Tepidimonas taiwanensis\n",
      "Coverage Percentage: 0.0271986395878428%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 392\n",
      "Length of fail 3152\n",
      "G020179495 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides thetaiotaomicron\n",
      "Coverage Percentage: 0.1071669498151066%\n",
      "Adj, read counts 42\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 39\n",
      "total # samples 23\n",
      "Number of total reads 41\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.904761904761903 %\n",
      "Percentage of reads from samples who only fall into one bin group 31.70731707317073 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 393\n",
      "Length of fail 3152\n",
      "G020312235 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptomycetales; f__Streptomycetaceae; g__Streptomyces; s__Streptomyces pseudogriseolus\n",
      "Coverage Percentage: 0.0035431978949331%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 393\n",
      "Length of fail 3153\n",
      "G020320395 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Oceanospirillales; f__Oceanospirillaceae; g__Thalassolituus; s__Thalassolituus marinus\n",
      "Coverage Percentage: 0.0360675891272334%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 393\n",
      "Length of fail 3154\n",
      "G020341515 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Peptostreptococcaceae; g__Clostridioides; s__Clostridioides difficile\n",
      "Coverage Percentage: 0.034011473358453%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 393\n",
      "Length of fail 3155\n",
      "G020341535 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas jandaei\n",
      "Coverage Percentage: 0.0172695993452951%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 393\n",
      "Length of fail 3156\n",
      "G020510525 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Azonexaceae; g__Quatrionicoccus; s__Quatrionicoccus australiensis\n",
      "Coverage Percentage: 0.0808417903148082%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 393\n",
      "Length of fail 3157\n",
      "G020510565 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Azonexaceae; g__Ferribacterium; s__Ferribacterium limneticum\n",
      "Coverage Percentage: 0.0469391821931826%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 393\n",
      "Length of fail 3158\n",
      "G020523825 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium brachiatum\n",
      "Coverage Percentage: 0.0198503000565318%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 393\n",
      "Length of fail 3159\n",
      "G020525685 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Klebsiella; s__Klebsiella quasipneumoniae\n",
      "Coverage Percentage: 0.0560527423741583%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 393\n",
      "Length of fail 3160\n",
      "G020525925 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Klebsiella; s__Klebsiella quasipneumoniae\n",
      "Coverage Percentage: 0.0857233646556811%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 393\n",
      "Length of fail 3161\n",
      "G020532675 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Methylophilaceae; g__Methylobacillus; s__Methylobacillus gramineus\n",
      "Coverage Percentage: 0.0098932155145274%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 393\n",
      "Length of fail 3162\n",
      "G020567375 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Intrasporangiaceae; g__Janibacter; s__Janibacter melonis\n",
      "Coverage Percentage: 0.0522287165115228%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 393\n",
      "Length of fail 3163\n",
      "G020687245 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Oscillospiraceae; g__Faecalibacterium; s__Faecalibacterium longum\n",
      "Coverage Percentage: 0.0518894087576366%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 393\n",
      "Length of fail 3164\n",
      "G020731325 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Finegoldia; s__Finegoldia magna\n",
      "Coverage Percentage: 0.1826374239980146%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 393\n",
      "Length of fail 3165\n",
      "G020731405 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Xanthomonas; s__Xanthomonas campestris\n",
      "Coverage Percentage: 0.0216189664080748%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 393\n",
      "Length of fail 3166\n",
      "G020735725 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Rikenellaceae; g__Alistipes; s__Alistipes senegalensis\n",
      "Coverage Percentage: 0.0389233299878159%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 393\n",
      "Length of fail 3167\n",
      "G020735785 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella melaninogenica\n",
      "Coverage Percentage: 0.041852797461814%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 393\n",
      "Length of fail 3168\n",
      "G020735945 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Tannerellaceae; g__Parabacteroides; s__Parabacteroides distasonis\n",
      "Coverage Percentage: 0.1225668974615061%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2]\n",
      "Number of bins 32\n",
      "total # samples 22\n",
      "Number of total reads 37\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.216216216216218 %\n",
      "Percentage of reads from samples who only fall into one bin group 40.54054054054054 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 394\n",
      "Length of fail 3168\n",
      "G020736445 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Dermabacteraceae; g__Dermabacter; s__Dermabacter jinjuensis\n",
      "Coverage Percentage: 0.0458233182568439%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 394\n",
      "Length of fail 3169\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G020736705 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter sanguinis\n",
      "Coverage Percentage: 0.1891198652425252%\n",
      "Adj, read counts 42\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 3, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1]\n",
      "Number of bins 40\n",
      "total # samples 10\n",
      "Number of total reads 41\n",
      "Number of hits in BIOM tables 10.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 19.047619047619047 %\n",
      "Percentage of reads from samples who only fall into one bin group 9.75609756097561 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 395\n",
      "Length of fail 3169\n",
      "G020748465 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Peptostreptococcaceae; g__Terrisporobacter; s__Terrisporobacter mayombei\n",
      "Coverage Percentage: 0.016922297371091%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 395\n",
      "Length of fail 3170\n",
      "G020782235 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium naphthae\n",
      "Coverage Percentage: 0.0402219580161149%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 395\n",
      "Length of fail 3171\n",
      "G020813135 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Xanthomonas; s__Xanthomonas campestris\n",
      "Coverage Percentage: 0.0195131528466865%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 395\n",
      "Length of fail 3172\n",
      "G020858175 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Janthinobacterium; s__Janthinobacterium lividum\n",
      "Coverage Percentage: 0.0731575132362651%\n",
      "Adj, read counts 33\n",
      "Number of samples per group [1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 30\n",
      "total # samples 22\n",
      "Number of total reads 33\n",
      "Number of hits in BIOM tables 8.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 18.181818181818183 %\n",
      "Percentage of reads from samples who only fall into one bin group 45.45454545454545 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 396\n",
      "Length of fail 3172\n",
      "G020880995 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Reyranellaceae; g__Reyranella; s__Reyranella aquatilis\n",
      "Coverage Percentage: 0.0956530255203537%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1]\n",
      "Number of bins 36\n",
      "total # samples 18\n",
      "Number of total reads 36\n",
      "Number of hits in BIOM tables 67.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.216216216216218 %\n",
      "Percentage of reads from samples who only fall into one bin group 22.22222222222222 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 397\n",
      "Length of fail 3172\n",
      "G020889785 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium acaciae\n",
      "Coverage Percentage: 0.010933374000295%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 397\n",
      "Length of fail 3173\n",
      "G020991065 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides okcheonensis\n",
      "Coverage Percentage: 0.0191433228827786%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 397\n",
      "Length of fail 3174\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G021021555 d__Bacteria; p__Spirochaetes; c__Spirochaetia; o__Spirochaetales; f__Treponemataceae; g__Treponema; s__Treponema zuelzerae\n",
      "Coverage Percentage: 0.0047220767859391%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 397\n",
      "Length of fail 3175\n",
      "G021026375 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Actinomycetospora; s__Actinomycetospora endophytica\n",
      "Coverage Percentage: 0.0085499218047352%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 397\n",
      "Length of fail 3176\n",
      "G021083185 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides furvisabuli\n",
      "Coverage Percentage: 0.0406832844046435%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 397\n",
      "Length of fail 3177\n",
      "G021129235 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Kineosporiales; f__Kineosporiaceae; g__Kineosporia; s__Kineosporia mesophila\n",
      "Coverage Percentage: 0.072324144741296%\n",
      "Adj, read counts 34\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 31\n",
      "total # samples 21\n",
      "Number of total reads 33\n",
      "Number of hits in BIOM tables 75.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.58823529411765 %\n",
      "Percentage of reads from samples who only fall into one bin group 45.45454545454545 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 398\n",
      "Length of fail 3177\n",
      "G021184025 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Rhodopseudomonas; s__Rhodopseudomonas boonkerdii\n",
      "Coverage Percentage: 0.0095164658828484%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 398\n",
      "Length of fail 3178\n",
      "G021282425 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas alloputida\n",
      "Coverage Percentage: 0.0833327112152951%\n",
      "Adj, read counts 35\n",
      "Number of samples per group [3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 7, 1, 1, 1, 1, 1]\n",
      "Number of bins 25\n",
      "total # samples 17\n",
      "Number of total reads 35\n",
      "Number of hits in BIOM tables 16.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 34.285714285714285 %\n",
      "Percentage of reads from samples who only fall into one bin group 37.142857142857146 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 399\n",
      "Length of fail 3178\n",
      "G900000175 d__Bacteria; p__Chlamydiae; c__Chlamydiia; o__Parachlamydiales; f__Criblamydiaceae; g__Estrella; s__Estrella lausannensis\n",
      "Coverage Percentage: 0.0053358568520352%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 399\n",
      "Length of fail 3179\n",
      "G900002575 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Peptostreptococcaceae; g__Romboutsia; s__Romboutsia hominis\n",
      "Coverage Percentage: 0.0770123295390807%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 399\n",
      "Length of fail 3180\n",
      "G900002825 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Peptostreptococcaceae; g__Romboutsia; s__Romboutsia lituseburensis\n",
      "Coverage Percentage: 0.0449879898824717%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 399\n",
      "Length of fail 3181\n",
      "G900007165 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia piptadeniae\n",
      "Coverage Percentage: 0.0015541941318158%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 399\n",
      "Length of fail 3182\n",
      "G900009265 d__Bacteria; p__Cyanobacteria; c__; o__Oscillatoriales; f__Microcoleaceae; g__Planktothrix; s__Planktothrix paucivesiculata\n",
      "Coverage Percentage: 0.0080629416634791%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 399\n",
      "Length of fail 3183\n",
      "G900010755 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Paeniglutamicibacter; s__Paeniglutamicibacter antarcticus\n",
      "Coverage Percentage: 0.0031331265469812%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 399\n",
      "Length of fail 3184\n",
      "G900011295 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter baumannii\n",
      "Coverage Percentage: 0.00955614473938%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 399\n",
      "Length of fail 3185\n",
      "G900013535 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Agrobacterium; s__Agrobacterium tomkonis\n",
      "Coverage Percentage: 0.0277789281999706%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 399\n",
      "Length of fail 3186\n",
      "G900015215 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Peptostreptococcaceae; g__Romboutsia; s__Romboutsia ilealis\n",
      "Coverage Percentage: 0.2997821276687815%\n",
      "Adj, read counts 45\n",
      "Number of samples per group [2, 1, 2, 2, 2, 1, 1, 1, 1, 2, 3, 4, 3, 1, 3, 1, 3]\n",
      "Number of bins 26\n",
      "total # samples 8\n",
      "Number of total reads 43\n",
      "Number of hits in BIOM tables 8.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 35.55555555555555 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.627906976744185 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 400\n",
      "Length of fail 3186\n",
      "G900044055 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Caballeronia; s__Caballeronia calidae\n",
      "Coverage Percentage: 0.0015700949117175%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 400\n",
      "Length of fail 3187\n",
      "G900067095 d__Eukaryota; p__Ascomycota; c__Sordariomycetes; o__Hypocreales; f__Nectriaceae; g__Fusarium; s__Fusarium proliferatum\n",
      "Coverage Percentage: 0.0017473645319151%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 400\n",
      "Length of fail 3188\n",
      "G900067155 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Trichococcus; s__Trichococcus collinsii\n",
      "Coverage Percentage: 0.072519028263265%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 400\n",
      "Length of fail 3189\n",
      "G900067235 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Trichococcus; s__Trichococcus ilyis\n",
      "Coverage Percentage: 0.0420619385908423%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 400\n",
      "Length of fail 3190\n",
      "G900074625 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Peptostreptococcaceae; g__Romboutsia; s__Romboutsia hominis\n",
      "Coverage Percentage: 0.0282339367857728%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 400\n",
      "Length of fail 3191\n",
      "G900078335 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Bordetella; s__Bordetella trematum\n",
      "Coverage Percentage: 0.0289494325168933%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 400\n",
      "Length of fail 3192\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G900078705 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Bordetella; s__Bordetella ansorpii\n",
      "Coverage Percentage: 0.062284843333878%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 400\n",
      "Length of fail 3193\n",
      "G900079135 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Trichococcus; s__Trichococcus pasteurii\n",
      "Coverage Percentage: 0.2034507799667912%\n",
      "Adj, read counts 38\n",
      "Number of samples per group [1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 2, 1, 1, 1, 2]\n",
      "Number of bins 31\n",
      "total # samples 17\n",
      "Number of total reads 38\n",
      "Number of hits in BIOM tables 17.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 18.421052631578945 %\n",
      "Percentage of reads from samples who only fall into one bin group 23.684210526315788 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 401\n",
      "Length of fail 3193\n",
      "G900079775 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella lascolaii\n",
      "Coverage Percentage: 0.0290787841191066%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 401\n",
      "Length of fail 3194\n",
      "G900079805 d__Eukaryota; p__Ascomycota; c__Sordariomycetes; o__Hypocreales; f__Nectriaceae; g__Fusarium; s__Fusarium fujikuroi\n",
      "Coverage Percentage: 0.0015536404990868%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 401\n",
      "Length of fail 3195\n",
      "G900083545 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Rikenellaceae; g__Alistipes; s__Alistipes provencensis\n",
      "Coverage Percentage: 0.0097758774966921%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 401\n",
      "Length of fail 3196\n",
      "G900088125 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Peptoniphilus; s__Peptoniphilus pacaensis\n",
      "Coverage Percentage: 0.0163101124857691%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 401\n",
      "Length of fail 3197\n",
      "G900088145 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter pseudolwoffii\n",
      "Coverage Percentage: 1.7549686019119506%\n",
      "Adj, read counts 323\n",
      "Number of samples per group [9, 4, 4, 2, 3, 1, 1, 1, 1, 2, 2, 1, 2, 3, 5, 4, 3, 1, 2, 4, 4, 8, 3, 2, 2, 4, 1, 4, 3, 3, 3, 4, 3, 4, 3, 1, 1, 3, 1, 3, 1, 1, 5, 1, 4, 4, 4, 7, 13, 1, 2, 2, 11, 1, 3, 2, 4, 6, 11, 7, 2, 14, 4, 2, 3, 2, 2, 3, 1, 4, 1, 10, 7, 9]\n",
      "Number of bins 174\n",
      "total # samples 47\n",
      "Number of total reads 318\n",
      "Number of hits in BIOM tables 268.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.647058823529413 %\n",
      "Percentage of reads from samples who only fall into one bin group 1.5723270440251573 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 402\n",
      "Length of fail 3197\n",
      "G900089075 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Vaginella; s__Vaginella massiliensis\n",
      "Coverage Percentage: 0.0062021591728701%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 402\n",
      "Length of fail 3198\n",
      "G900089585 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora sediminicola\n",
      "Coverage Percentage: 0.0101658780849809%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 402\n",
      "Length of fail 3199\n",
      "G900089945 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Rhodocyclales; f__Rhodocyclaceae; g__Propionivibrio; s__Candidatus Propionivibrio aalborgensis\n",
      "Coverage Percentage: 0.0349538433052606%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 402\n",
      "Length of fail 3200\n",
      "G900089955 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__; f__; g__Candidatus Accumulibacter; s__Candidatus Accumulibacter aalborgensis\n",
      "Coverage Percentage: 0.0068355471377844%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 402\n",
      "Length of fail 3201\n",
      "G900090045 d__Eukaryota; p__Apicomplexa; c__Aconoidasida; o__Haemosporida; f__Plasmodiidae; g__Plasmodium; s__Plasmodium malariae\n",
      "Coverage Percentage: 0.0010591050217464%\n",
      "Adj, read counts 42\n",
      "Number of samples per group [41, 1]\n",
      "Number of bins 2\n",
      "total # samples 42\n",
      "Number of total reads 42\n",
      "Number of hits in BIOM tables 19.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 100.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 100.0 %\n",
      "Status: Fail -- less than 3 bins\n",
      "----------------------------\n",
      "Length of pass 402\n",
      "Length of fail 3202\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G900090235 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora echinaurantiaca\n",
      "Coverage Percentage: 0.0020962809338528%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 402\n",
      "Length of fail 3203\n",
      "G900090245 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora halophytica\n",
      "Coverage Percentage: 0.0048111044113207%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 402\n",
      "Length of fail 3204\n",
      "G900090265 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora rifamycinica\n",
      "Coverage Percentage: 0.0148473144117789%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 402\n",
      "Length of fail 3205\n",
      "G900090275 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora zamorensis\n",
      "Coverage Percentage: 0.0170235985407055%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 402\n",
      "Length of fail 3206\n",
      "G900090295 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora coxensis\n",
      "Coverage Percentage: 0.0067801186830971%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 402\n",
      "Length of fail 3207\n",
      "G900091515 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora purpureochromogenes\n",
      "Coverage Percentage: 0.0052290644888986%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 402\n",
      "Length of fail 3208\n",
      "G900091525 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora matsumotoense\n",
      "Coverage Percentage: 0.0598192761687094%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 402\n",
      "Length of fail 3209\n",
      "G900091535 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora carbonacea\n",
      "Coverage Percentage: 0.0076042487102992%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 402\n",
      "Length of fail 3210\n",
      "G900091565 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora marina\n",
      "Coverage Percentage: 0.0143020760114924%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 402\n",
      "Length of fail 3211\n",
      "G900091605 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora tulbaghiae\n",
      "Coverage Percentage: 0.0245818352412895%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 402\n",
      "Length of fail 3212\n",
      "G900091705 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Clostridium; s__Clostridium mediterraneense\n",
      "Coverage Percentage: 0.0391773025908261%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 402\n",
      "Length of fail 3213\n",
      "G900092335 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium phoceense\n",
      "Coverage Percentage: 0.0378380816486378%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 402\n",
      "Length of fail 3214\n",
      "G900094575 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium yuanmingense\n",
      "Coverage Percentage: 0.0110194691102583%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 402\n",
      "Length of fail 3215\n",
      "G900094765 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Rhodococcus; s__Rhodococcus qingshengii\n",
      "Coverage Percentage: 0.1455166435747283%\n",
      "Adj, read counts 63\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 62\n",
      "total # samples 32\n",
      "Number of total reads 63\n",
      "Number of hits in BIOM tables 29.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 9.523809523809524 %\n",
      "Percentage of reads from samples who only fall into one bin group 25.396825396825395 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 403\n",
      "Length of fail 3215\n",
      "G900095025 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter albensis\n",
      "Coverage Percentage: 0.1164621248626285%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 403\n",
      "Length of fail 3216\n",
      "G900095795 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Dysgonomonadaceae; g__Petrimonas; s__Petrimonas mucosa\n",
      "Coverage Percentage: 0.0145522741357939%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 403\n",
      "Length of fail 3217\n",
      "G900096585 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Sanguibacteraceae; g__Sanguibacter; s__Sanguibacter gelidistatuariae\n",
      "Coverage Percentage: 0.0075238101366512%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 403\n",
      "Length of fail 3218\n",
      "G900096715 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Tannerellaceae; g__Tannerella; s__Tannerella forsythia\n",
      "Coverage Percentage: 0.0051613959346524%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 403\n",
      "Length of fail 3219\n",
      "G900096895 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter kookii\n",
      "Coverage Percentage: 0.0996169252850376%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 403\n",
      "Length of fail 3220\n",
      "G900097015 d__Eukaryota; p__Apicomplexa; c__Aconoidasida; o__Haemosporida; f__Plasmodiidae; g__Plasmodium; s__Plasmodium sp. gorilla clade G2\n",
      "Coverage Percentage: 0.0009823220778148%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 403\n",
      "Length of fail 3221\n",
      "G900099645 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas peli\n",
      "Coverage Percentage: 0.0144846796657381%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 403\n",
      "Length of fail 3222\n",
      "G900099655 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Planococcaceae; g__Planococcus; s__Planococcus glaciei\n",
      "Coverage Percentage: 0.0038537553442143%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 403\n",
      "Length of fail 3223\n",
      "G900099945 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas delhiensis\n",
      "Coverage Percentage: 0.4024389462696185%\n",
      "Adj, read counts 163\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 3, 1, 2, 1, 2, 1, 2, 2, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 3, 2, 1, 1, 1, 3, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2]\n",
      "Number of bins 113\n",
      "total # samples 15\n",
      "Number of total reads 161\n",
      "Number of hits in BIOM tables 48.0\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Percentage of reads in the top 3 bin groups with the largest samples 6.748466257668712 %\n",
      "Percentage of reads from samples who only fall into one bin group 5.590062111801243 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 404\n",
      "Length of fail 3223\n",
      "G900100085 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Roseobacteraceae; g__Salipiger; s__Salipiger marinus\n",
      "Coverage Percentage: 0.0047917614479456%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 404\n",
      "Length of fail 3224\n",
      "G900100115 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium soldanellicola\n",
      "Coverage Percentage: 0.0036261830724455%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 404\n",
      "Length of fail 3225\n",
      "G900100155 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Xanthobacteraceae; g__Ancylobacter; s__Ancylobacter rudongensis\n",
      "Coverage Percentage: 0.0053491080972623%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 404\n",
      "Length of fail 3226\n",
      "G900100305 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax citrulli\n",
      "Coverage Percentage: 0.0458588282162111%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 404\n",
      "Length of fail 3227\n",
      "G900100475 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingobium; s__Sphingobium faniae\n",
      "Coverage Percentage: 0.0233363408989199%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 404\n",
      "Length of fail 3228\n",
      "G900100495 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas benzenivorans\n",
      "Coverage Percentage: 0.0207520746851847%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 404\n",
      "Length of fail 3229\n",
      "G900100535 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas flavescens\n",
      "Coverage Percentage: 0.0051565226596729%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 404\n",
      "Length of fail 3230\n",
      "G900100555 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Selenomonadales; f__Sporomusaceae; g__Pelosinus; s__Pelosinus fermentans\n",
      "Coverage Percentage: 0.0929555303368618%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 404\n",
      "Length of fail 3231\n",
      "G900100625 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium saliperosum\n",
      "Coverage Percentage: 0.0185773358578214%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 404\n",
      "Length of fail 3232\n",
      "G900100705 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter terrae\n",
      "Coverage Percentage: 0.0078695824309226%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 404\n",
      "Length of fail 3233\n",
      "G900100795 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas abietaniphila\n",
      "Coverage Percentage: 0.0083615848304238%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 404\n",
      "Length of fail 3234\n",
      "G900100885 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium pygmaeum\n",
      "Coverage Percentage: 0.0179629129902662%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 404\n",
      "Length of fail 3235\n",
      "G900101025 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Klenkia; s__Klenkia marina\n",
      "Coverage Percentage: 0.0182486768524303%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 404\n",
      "Length of fail 3236\n",
      "G900101065 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Jeotgalicoccus; s__Jeotgalicoccus aerolatus\n",
      "Coverage Percentage: 0.0132582267955787%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 404\n",
      "Length of fail 3237\n",
      "G900101085 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas grimontii\n",
      "Coverage Percentage: 0.064014514833992%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 404\n",
      "Length of fail 3238\n",
      "G900101175 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas pavanii\n",
      "Coverage Percentage: 0.3288414199602605%\n",
      "Adj, read counts 90\n",
      "Number of samples per group [1, 1, 2, 3, 1, 1, 1, 2, 1, 1, 1, 2, 2, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 2, 2, 3, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 3, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 79\n",
      "total # samples 29\n",
      "Number of total reads 89\n",
      "Number of hits in BIOM tables 21.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.11111111111111 %\n",
      "Percentage of reads from samples who only fall into one bin group 23.595505617977526 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 405\n",
      "Length of fail 3238\n",
      "G900101185 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas gessardii\n",
      "Coverage Percentage: 0.0829342195580784%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 405\n",
      "Length of fail 3239\n",
      "G900101535 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia nematodiphila\n",
      "Coverage Percentage: 0.0449842555105713%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 405\n",
      "Length of fail 3240\n",
      "G900101755 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas alcaliphila\n",
      "Coverage Percentage: 0.0643047386044547%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 405\n",
      "Length of fail 3241\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G900101855 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium anhuiense\n",
      "Coverage Percentage: 0.0102426918767963%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 405\n",
      "Length of fail 3242\n",
      "G900101875 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Mucilaginibacter; s__Mucilaginibacter pineti\n",
      "Coverage Percentage: 0.0046062342605877%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 405\n",
      "Length of fail 3243\n",
      "G900102035 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas extremaustralis\n",
      "Coverage Percentage: 0.0413069043059713%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 405\n",
      "Length of fail 3244\n",
      "G900102105 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Rhizobiaceae; g__Rhizobium; s__Rhizobium pusense\n",
      "Coverage Percentage: 0.0711182368127523%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 405\n",
      "Length of fail 3245\n",
      "G900102145 d__Bacteria; p__Deinococcus-Thermus; c__Deinococci; o__Thermales; f__Thermaceae; g__Thermus; s__Thermus arciformis\n",
      "Coverage Percentage: 0.0497128316797167%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 405\n",
      "Length of fail 3246\n",
      "G900102455 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Novosphingobium; s__Novosphingobium aromaticivorans\n",
      "Coverage Percentage: 0.0385143765319696%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 405\n",
      "Length of fail 3247\n",
      "G900102525 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Boseaceae; g__Bosea; s__Bosea robiniae\n",
      "Coverage Percentage: 0.0061748236529082%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 405\n",
      "Length of fail 3248\n",
      "G900102625 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax valerianellae\n",
      "Coverage Percentage: 0.047145984812869%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 405\n",
      "Length of fail 3249\n",
      "G900102635 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas chengduensis\n",
      "Coverage Percentage: 0.1814032569836375%\n",
      "Adj, read counts 64\n",
      "Number of samples per group [4, 2, 1, 1, 1, 1, 1, 1, 1, 4, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 2, 1, 2, 1, 2, 1, 2]\n",
      "Number of bins 59\n",
      "total # samples 45\n",
      "Number of total reads 63\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.625 %\n",
      "Percentage of reads from samples who only fall into one bin group 53.96825396825397 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3249\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G900102665 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas psychrotolerans\n",
      "Coverage Percentage: 0.0549276120704692%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3250\n",
      "G900102715 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus equinus\n",
      "Coverage Percentage: 0.0624321495522635%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3251\n",
      "G900102945 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Oceanospirillales; f__Halomonadaceae; g__Halomonas; s__Halomonas muralis\n",
      "Coverage Percentage: 0.00364835642751%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3252\n",
      "G900103225 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas congelans\n",
      "Coverage Percentage: 0.029437913627523%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3253\n",
      "G900103325 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Phyllobacteriaceae; g__Mesorhizobium; s__Mesorhizobium qingshengii\n",
      "Coverage Percentage: 0.0153693179164639%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3254\n",
      "G900103375 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas balearica\n",
      "Coverage Percentage: 0.0168818974977912%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3255\n",
      "G900103535 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Veillonellales; f__Veillonellaceae; g__Megasphaera; s__Megasphaera paucivorans\n",
      "Coverage Percentage: 0.0051822789397675%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3256\n",
      "G900103645 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Oryzisolibacter; s__Oryzisolibacter propanilivorax\n",
      "Coverage Percentage: 0.0843563956812754%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3257\n",
      "G900103655 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Lactobacillaceae; g__Lactobacillus; s__Lactobacillus kefiranofaciens\n",
      "Coverage Percentage: 0.012786760623083%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3258\n",
      "G900103665 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter steynii\n",
      "Coverage Percentage: 0.005606896283955%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3259\n",
      "G900103785 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Geodermatophilus; s__Geodermatophilus siccatus\n",
      "Coverage Percentage: 0.0050696861000153%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3260\n",
      "G900103795 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas antarctica\n",
      "Coverage Percentage: 0.0205420143781555%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3261\n",
      "G900103815 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Lachnospira; s__Lachnospira pectinoschiza\n",
      "Coverage Percentage: 0.0079496311371152%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3262\n",
      "G900103875 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas arsenicoxydans\n",
      "Coverage Percentage: 0.0164386224280629%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3263\n",
      "G900103935 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides szechwanensis\n",
      "Coverage Percentage: 0.0190560147410701%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3264\n",
      "G900103965 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas poae\n",
      "Coverage Percentage: 0.0335280706418802%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3265\n",
      "G900103975 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Klenkia; s__Klenkia soli\n",
      "Coverage Percentage: 0.0167486763397102%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3266\n",
      "G900104195 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Cloacibacterium; s__Cloacibacterium normanense\n",
      "Coverage Percentage: 0.0376309868890718%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3267\n",
      "G900104265 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas guguanensis\n",
      "Coverage Percentage: 0.0387045690861807%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3268\n",
      "G900104315 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Pseudarthrobacter; s__Pseudarthrobacter chlorophenolicus\n",
      "Coverage Percentage: 0.0073690279754193%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3269\n",
      "G900104515 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax cattleyae\n",
      "Coverage Percentage: 0.0361605643115368%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3270\n",
      "G900104575 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__Bacillus tuaregi\n",
      "Coverage Percentage: 0.0201487263846132%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3271\n",
      "G900104595 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Enterococcaceae; g__Enterococcus; s__Enterococcus timonensis\n",
      "Coverage Percentage: 0.0459511872214268%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3272\n",
      "G900104635 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Coprococcus; s__Coprococcus phoceensis\n",
      "Coverage Percentage: 0.076637310319186%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3273\n",
      "G900104705 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Agrococcus; s__Agrococcus carbonis\n",
      "Coverage Percentage: 0.0193631436802953%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 406\n",
      "Length of fail 3274\n",
      "G900104725 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes derwentensis\n",
      "Coverage Percentage: 0.1915679863144806%\n",
      "Adj, read counts 112\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 2, 1, 1, 1, 2, 1, 2, 4, 1, 1, 2, 1, 1, 1, 4, 1, 2, 1, 1, 1, 1, 1, 2, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 97\n",
      "total # samples 38\n",
      "Number of total reads 112\n",
      "Number of hits in BIOM tables 99.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.5 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.071428571428573 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 407\n",
      "Length of fail 3274\n",
      "G900104735 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas lini\n",
      "Coverage Percentage: 0.0153296346186948%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 407\n",
      "Length of fail 3275\n",
      "G900104825 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas taetrolens\n",
      "Coverage Percentage: 0.0484379912677909%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 407\n",
      "Length of fail 3276\n",
      "G900104835 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium longum\n",
      "Coverage Percentage: 0.3974898728889858%\n",
      "Adj, read counts 56\n",
      "Number of samples per group [1, 1, 2, 1, 2, 1, 5, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 2, 1, 1, 4, 2, 1, 1, 1, 2, 1, 1, 2, 1, 3]\n",
      "Number of bins 46\n",
      "total # samples 23\n",
      "Number of total reads 55\n",
      "Number of hits in BIOM tables 8.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 28.57142857142857 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.181818181818183 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 408\n",
      "Length of fail 3276\n",
      "G900104965 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Marmoricola; s__Marmoricola scoriae\n",
      "Coverage Percentage: 0.0248116213988038%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 408\n",
      "Length of fail 3277\n",
      "G900105005 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Halopseudomonas; s__Halopseudomonas litoralis\n",
      "Coverage Percentage: 0.0256272533552393%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 408\n",
      "Length of fail 3278\n",
      "G900105065 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Friedmanniella; s__Friedmanniella luteola\n",
      "Coverage Percentage: 0.0128030822360625%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 408\n",
      "Length of fail 3279\n",
      "G900105165 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Mucilaginibacter; s__Mucilaginibacter mallensis\n",
      "Coverage Percentage: 0.0025107847349608%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 408\n",
      "Length of fail 3280\n",
      "G900105185 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas saponiphila\n",
      "Coverage Percentage: 0.0564540357856764%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 408\n",
      "Length of fail 3281\n",
      "G900105205 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium hydrocarbonoxydans\n",
      "Coverage Percentage: 0.0258976051530392%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 408\n",
      "Length of fail 3282\n",
      "G900105335 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium paraoxydans\n",
      "Coverage Percentage: 0.0398613523076373%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 408\n",
      "Length of fail 3283\n",
      "G900105355 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas anguilliseptica\n",
      "Coverage Percentage: 0.0275352666532432%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 408\n",
      "Length of fail 3284\n",
      "G900105505 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium coyleae\n",
      "Coverage Percentage: 0.0432082137646577%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 408\n",
      "Length of fail 3285\n",
      "G900105535 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Pseudarthrobacter; s__Pseudarthrobacter equi\n",
      "Coverage Percentage: 0.0049112190632443%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 408\n",
      "Length of fail 3286\n",
      "G900105555 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas coleopterorum\n",
      "Coverage Percentage: 0.0131019934327527%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 408\n",
      "Length of fail 3287\n",
      "G900105585 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides exalbidus\n",
      "Coverage Percentage: 0.0281747154103736%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 408\n",
      "Length of fail 3288\n",
      "G900105675 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas synxantha\n",
      "Coverage Percentage: 0.0388481991272747%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 408\n",
      "Length of fail 3289\n",
      "G900105735 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas yamanorum\n",
      "Coverage Percentage: 0.0668357430837986%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 408\n",
      "Length of fail 3290\n",
      "G900105785 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Microlunatus; s__Microlunatus sagamiharensis\n",
      "Coverage Percentage: 0.0265590081004974%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 408\n",
      "Length of fail 3291\n",
      "G900105815 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas brenneri\n",
      "Coverage Percentage: 0.0234554775253426%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 408\n",
      "Length of fail 3292\n",
      "G900105835 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas fragi\n",
      "Coverage Percentage: 0.1828163296206221%\n",
      "Adj, read counts 57\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 3, 1, 1, 1, 2, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 3, 1, 1, 1, 1, 2, 1, 1, 2]\n",
      "Number of bins 44\n",
      "total # samples 18\n",
      "Number of total reads 57\n",
      "Number of hits in BIOM tables 25.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.789473684210526 %\n",
      "Percentage of reads from samples who only fall into one bin group 22.807017543859647 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 409\n",
      "Length of fail 3292\n",
      "G900105885 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas guangdongensis\n",
      "Coverage Percentage: 0.0094636611081759%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 409\n",
      "Length of fail 3293\n",
      "G900105935 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas costantinii\n",
      "Coverage Percentage: 0.0421988271050576%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 409\n",
      "Length of fail 3294\n",
      "G900105975 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas palleroniana\n",
      "Coverage Percentage: 0.0665837281849717%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 409\n",
      "Length of fail 3295\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G900106015 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas sihuiensis\n",
      "Coverage Percentage: 0.1997126927395479%\n",
      "Adj, read counts 74\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 2, 1, 3, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 2, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 2, 2, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 3, 1]\n",
      "Number of bins 68\n",
      "total # samples 44\n",
      "Number of total reads 71\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.81081081081081 %\n",
      "Percentage of reads from samples who only fall into one bin group 45.07042253521127 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 410\n",
      "Length of fail 3295\n",
      "G900106065 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas mandelii\n",
      "Coverage Percentage: 0.0610926987266532%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 410\n",
      "Length of fail 3296\n",
      "G900106095 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas deceptionensis\n",
      "Coverage Percentage: 0.0575461143913944%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 410\n",
      "Length of fail 3297\n",
      "G900106515 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Peptoniphilus; s__Peptoniphilus lacydonensis\n",
      "Coverage Percentage: 0.0398867995287287%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 410\n",
      "Length of fail 3298\n",
      "G900106525 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Lysobacter; s__Lysobacter enzymogenes\n",
      "Coverage Percentage: 0.1134028468520413%\n",
      "Adj, read counts 42\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 38\n",
      "total # samples 24\n",
      "Number of total reads 41\n",
      "Number of hits in BIOM tables 54.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.666666666666664 %\n",
      "Percentage of reads from samples who only fall into one bin group 36.58536585365854 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 411\n",
      "Length of fail 3298\n",
      "G900106605 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides stercoris\n",
      "Coverage Percentage: 0.0557646302856172%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 411\n",
      "Length of fail 3299\n",
      "G900106645 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium degerlachei\n",
      "Coverage Percentage: 0.0291917892777987%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 411\n",
      "Length of fail 3300\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G900106665 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus sanguinis\n",
      "Coverage Percentage: 0.0109511048494446%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 411\n",
      "Length of fail 3301\n",
      "G900106755 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides faecis\n",
      "Coverage Percentage: 0.033367590054747%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 411\n",
      "Length of fail 3302\n",
      "G900106785 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotellamassilia; s__Prevotellamassilia timonensis\n",
      "Coverage Percentage: 0.0214346976830098%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 411\n",
      "Length of fail 3303\n",
      "G900106845 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Peptostreptococcaceae; g__Romboutsia; s__Romboutsia timonensis\n",
      "Coverage Percentage: 0.1869971900628388%\n",
      "Adj, read counts 31\n",
      "Number of samples per group [1, 1, 2, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 26\n",
      "total # samples 13\n",
      "Number of total reads 31\n",
      "Number of hits in BIOM tables 42.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 25.806451612903224 %\n",
      "Percentage of reads from samples who only fall into one bin group 32.25806451612903 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 412\n",
      "Length of fail 3303\n",
      "G900107155 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas salomonii\n",
      "Coverage Percentage: 0.0427166218700477%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 412\n",
      "Length of fail 3304\n",
      "G900107225 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Delftia; s__Delftia lacustris\n",
      "Coverage Percentage: 0.0848888627539942%\n",
      "Adj, read counts 41\n",
      "Number of samples per group [1, 1, 3, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 1, 1, 1, 2, 1, 1, 5, 1, 1, 1]\n",
      "Number of bins 29\n",
      "total # samples 25\n",
      "Number of total reads 40\n",
      "Number of hits in BIOM tables 16.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 31.70731707317073 %\n",
      "Percentage of reads from samples who only fall into one bin group 55.00000000000001 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 413\n",
      "Length of fail 3304\n",
      "G900107285 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter kyonggiensis\n",
      "Coverage Percentage: 0.6322494310266497%\n",
      "Adj, read counts 155\n",
      "Number of samples per group [8, 7, 1, 1, 1, 2, 3, 1, 1, 2, 1, 1, 2, 1, 1, 1, 2, 1, 5, 2, 1, 3, 1, 1, 3, 1, 1, 2, 2, 3, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 4, 5, 2, 3, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 2, 3, 1, 1, 1, 1, 3, 3, 8, 2, 1, 1, 1, 2, 1, 4, 1]\n",
      "Number of bins 108\n",
      "total # samples 36\n",
      "Number of total reads 151\n",
      "Number of hits in BIOM tables 93.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.7741935483871 %\n",
      "Percentage of reads from samples who only fall into one bin group 7.9470198675496695 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 414\n",
      "Length of fail 3304\n",
      "G900107315 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides uniformis\n",
      "Coverage Percentage: 0.2316214884689685%\n",
      "Adj, read counts 68\n",
      "Number of samples per group [1, 3, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 2, 2, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 2]\n",
      "Number of bins 61\n",
      "total # samples 37\n",
      "Number of total reads 67\n",
      "Number of hits in BIOM tables 23.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.76470588235294 %\n",
      "Percentage of reads from samples who only fall into one bin group 38.80597014925373 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 415\n",
      "Length of fail 3304\n",
      "G900107475 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides ovatus\n",
      "Coverage Percentage: 0.1281389654453021%\n",
      "Adj, read counts 50\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 3, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2]\n",
      "Number of bins 46\n",
      "total # samples 31\n",
      "Number of total reads 49\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 40.816326530612244 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 416\n",
      "Length of fail 3304\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G900107505 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Trichococcus; s__Trichococcus collinsii\n",
      "Coverage Percentage: 0.0626476621775999%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 416\n",
      "Length of fail 3305\n",
      "G900107535 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter hartonius\n",
      "Coverage Percentage: 0.0099623415780541%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 416\n",
      "Length of fail 3306\n",
      "G900107605 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax soli\n",
      "Coverage Percentage: 0.2176426299787294%\n",
      "Adj, read counts 60\n",
      "Number of samples per group [3, 6, 2, 2, 1, 2, 1, 1, 1, 2, 1, 5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 1, 1, 1, 2, 1, 2, 1, 1, 1, 2, 1]\n",
      "Number of bins 41\n",
      "total # samples 26\n",
      "Number of total reads 59\n",
      "Number of hits in BIOM tables 27.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 30.000000000000004 %\n",
      "Percentage of reads from samples who only fall into one bin group 15.254237288135593 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3306\n",
      "G900107795 d__Bacteria; p__Bacteroidetes; c__Chitinophagia; o__Chitinophagales; f__Chitinophagaceae; g__Chitinophaga; s__Chitinophaga terrae (ex Kim and Jung 2007)\n",
      "Coverage Percentage: 0.0034378147560659%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3307\n",
      "G900107815 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Eubacteriaceae; g__Eubacterium; s__Eubacterium aggregans\n",
      "Coverage Percentage: 0.0176381090394956%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3308\n",
      "G900108115 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Epilithonimonas; s__Epilithonimonas bovis\n",
      "Coverage Percentage: 0.0140384624000692%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3309\n",
      "G900108245 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Boseaceae; g__Bosea; s__Bosea lathyri\n",
      "Coverage Percentage: 0.0102042660927949%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3310\n",
      "G900108345 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides ndongoniae\n",
      "Coverage Percentage: 0.0505906621436782%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3311\n",
      "G900108395 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium marinum\n",
      "Coverage Percentage: 0.0156733025930757%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3312\n",
      "G900108525 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Epilithonimonas; s__Epilithonimonas hominis\n",
      "Coverage Percentage: 0.028561783402056%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3313\n",
      "G900108855 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Spirosomaceae; g__Dyadobacter; s__Dyadobacter koreensis\n",
      "Coverage Percentage: 0.0034466631600615%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3314\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G900108945 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia diazotrophica\n",
      "Coverage Percentage: 0.0028420868649906%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3315\n",
      "G900108955 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium terrigena\n",
      "Coverage Percentage: 0.0689621480385343%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3316\n",
      "G900109135 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Trichococcus; s__Trichococcus ilyis\n",
      "Coverage Percentage: 0.0371641159598045%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3317\n",
      "G900109225 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Selenomonadales; f__Selenomonadaceae; g__Propionispira; s__Propionispira arboris\n",
      "Coverage Percentage: 0.0033314666171795%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3318\n",
      "G900109525 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Boseaceae; g__Bosea; s__Bosea lupini\n",
      "Coverage Percentage: 0.0113240747871922%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3319\n",
      "G900109565 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas palmae\n",
      "Coverage Percentage: 0.0170786261452074%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3320\n",
      "G900110065 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Roseobacteraceae; g__Loktanella; s__Loktanella fryxellensis\n",
      "Coverage Percentage: 0.0029837193074506%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3321\n",
      "G900110225 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Brachymonas; s__Brachymonas denitrificans\n",
      "Coverage Percentage: 0.0207130141858348%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3322\n",
      "G900110285 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus alcaliphilus\n",
      "Coverage Percentage: 0.0209208830737264%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3323\n",
      "G900110605 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Epilithonimonas; s__Epilithonimonas lactis\n",
      "Coverage Percentage: 0.0189519312996158%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3324\n",
      "G900110615 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium urocaniciphilum\n",
      "Coverage Percentage: 0.0106018702822554%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3325\n",
      "G900110655 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas soli\n",
      "Coverage Percentage: 0.015268611542964%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3326\n",
      "G900110835 d__Bacteria; p__Bacteroidetes; c__Chitinophagia; o__Chitinophagales; f__Chitinophagaceae; g__Chitinophaga; s__Chitinophaga arvensicola\n",
      "Coverage Percentage: 0.0018156875160901%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3327\n",
      "G900110885 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Azotobacter; s__Azotobacter beijerinckii\n",
      "Coverage Percentage: 0.0254461006884808%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3328\n",
      "G900110925 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas cuatrocienegasensis\n",
      "Coverage Percentage: 0.0112692475109236%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 417\n",
      "Length of fail 3329\n",
      "G900111115 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Giesbergeria; s__Giesbergeria anulus\n",
      "Coverage Percentage: 3.5525818118895054%\n",
      "Adj, read counts 725\n",
      "Number of samples per group [10, 5, 26, 16, 2, 10, 30, 3, 10, 18, 10, 9, 1, 10, 2, 12, 3, 1, 13, 12, 9, 14, 1, 1, 34, 5, 10, 7, 7, 2, 1, 16, 11, 16, 4, 10, 9, 3, 7, 4, 4, 14, 20, 14, 6, 3]\n",
      "Number of bins 271\n",
      "total # samples 50\n",
      "Number of total reads 714\n",
      "Number of hits in BIOM tables 1619.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 32.41379310344828 %\n",
      "Percentage of reads from samples who only fall into one bin group 0.9803921568627451 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 418\n",
      "Length of fail 3329\n",
      "G900111455 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Geodermatophilus; s__Geodermatophilus poikilotrophus\n",
      "Coverage Percentage: 0.0023402732445051%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 418\n",
      "Length of fail 3330\n",
      "G900111495 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium wanjuense\n",
      "Coverage Percentage: 0.0032343992713819%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 418\n",
      "Length of fail 3331\n",
      "G900111735 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas graminis\n",
      "Coverage Percentage: 0.0353946926751606%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 418\n",
      "Length of fail 3332\n",
      "G900111805 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Rhodococcus; s__Rhodococcus kroppenstedtii\n",
      "Coverage Percentage: 0.0099426802037318%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 418\n",
      "Length of fail 3333\n",
      "G900111835 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas otitidis\n",
      "Coverage Percentage: 0.0396330062415287%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 418\n",
      "Length of fail 3334\n",
      "G900112035 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Nocardioides; s__Nocardioides alpinus\n",
      "Coverage Percentage: 0.0179022753291636%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 418\n",
      "Length of fail 3335\n",
      "G900112225 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia yuzhufengensis\n",
      "Coverage Percentage: 0.0220191823089755%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 418\n",
      "Length of fail 3336\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G900112375 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas citronellolis\n",
      "Coverage Percentage: 0.5280553355324991%\n",
      "Adj, read counts 228\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 2, 1, 1, 3, 1, 1, 2, 1, 1, 1, 2, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 3, 2, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 2, 2, 1, 2, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 3, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 143\n",
      "total # samples 14\n",
      "Number of total reads 222\n",
      "Number of hits in BIOM tables 78.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 4.824561403508771 %\n",
      "Percentage of reads from samples who only fall into one bin group 3.153153153153153 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 419\n",
      "Length of fail 3336\n",
      "G900112675 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax konjaci\n",
      "Coverage Percentage: 0.0427511328754833%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 419\n",
      "Length of fail 3337\n",
      "G900112935 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Trichococcus; s__Trichococcus pasteurii\n",
      "Coverage Percentage: 0.2234363675362529%\n",
      "Adj, read counts 45\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 2, 3, 1, 3, 2, 1, 1, 2, 3]\n",
      "Number of bins 34\n",
      "total # samples 17\n",
      "Number of total reads 43\n",
      "Number of hits in BIOM tables 18.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 24.444444444444443 %\n",
      "Percentage of reads from samples who only fall into one bin group 20.930232558139537 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 420\n",
      "Length of fail 3337\n",
      "G900112985 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter antarcticus\n",
      "Coverage Percentage: 0.0132148951997549%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 420\n",
      "Length of fail 3338\n",
      "G900113015 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes philippinensis\n",
      "Coverage Percentage: 0.026333050482596%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 420\n",
      "Length of fail 3339\n",
      "G900113035 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax wautersii\n",
      "Coverage Percentage: 0.3459275884981372%\n",
      "Adj, read counts 117\n",
      "Number of samples per group [1, 1, 2, 2, 2, 2, 7, 2, 1, 2, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 3, 1, 1, 3, 1, 2, 1, 1, 1, 6, 1, 1, 1, 1, 2, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 3, 9, 5]\n",
      "Number of bins 87\n",
      "total # samples 33\n",
      "Number of total reads 116\n",
      "Number of hits in BIOM tables 166.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 22.222222222222225 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.206896551724139 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 421\n",
      "Length of fail 3339\n",
      "G900113085 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Nevskiales; f__Sinobacteraceae; g__Fontimonas; s__Fontimonas thermophila\n",
      "Coverage Percentage: 0.1223812031835206%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 421\n",
      "Length of fail 3340\n",
      "G900113175 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Thermoactinomycetaceae; g__Planifilum; s__Planifilum fulgidum\n",
      "Coverage Percentage: 0.0166097148507992%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 421\n",
      "Length of fail 3341\n",
      "G900113315 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas jatrophae\n",
      "Coverage Percentage: 0.0111558082235298%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 421\n",
      "Length of fail 3342\n",
      "G900113355 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Priestia; s__Priestia megaterium\n",
      "Coverage Percentage: 0.0038382969646204%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 421\n",
      "Length of fail 3343\n",
      "G900113465 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium phyllosphaerae\n",
      "Coverage Percentage: 0.0531799258994135%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 421\n",
      "Length of fail 3344\n",
      "G900113485 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium gossipiicola\n",
      "Coverage Percentage: 0.0126155086800665%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 421\n",
      "Length of fail 3345\n",
      "G900113645 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Trichococcus; s__Trichococcus flocculiformis\n",
      "Coverage Percentage: 0.6059367917654589%\n",
      "Adj, read counts 111\n",
      "Number of samples per group [3, 2, 1, 2, 1, 3, 1, 4, 1, 1, 1, 1, 1, 2, 1, 3, 3, 2, 1, 2, 1, 1, 2, 3, 1, 1, 1, 1, 2, 1, 2, 1, 3, 1, 3, 1, 1, 1, 3, 1, 1, 1, 1, 2, 1, 3, 1, 5, 2, 4, 1, 2, 1, 1, 1, 1, 1, 5, 1, 1]\n",
      "Number of bins 88\n",
      "total # samples 34\n",
      "Number of total reads 108\n",
      "Number of hits in BIOM tables 161.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.612612612612612 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.11111111111111 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 422\n",
      "Length of fail 3345\n",
      "G900113825 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia megapolitana\n",
      "Coverage Percentage: 0.0023396612407118%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 422\n",
      "Length of fail 3346\n",
      "G900113845 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium brachiatum\n",
      "Coverage Percentage: 0.02823602493241%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 422\n",
      "Length of fail 3347\n",
      "G900114065 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas kunmingensis\n",
      "Coverage Percentage: 0.051576919747301%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 422\n",
      "Length of fail 3348\n",
      "G900114085 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Myroides; s__Myroides guanonis\n",
      "Coverage Percentage: 0.0135953835553806%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 422\n",
      "Length of fail 3349\n",
      "G900114255 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Phyllobacteriaceae; g__Mesorhizobium; s__Mesorhizobium albiziae\n",
      "Coverage Percentage: 0.0048127681783592%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 422\n",
      "Length of fail 3350\n",
      "G900114315 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Roseomonas; s__Roseomonas stagni\n",
      "Coverage Percentage: 0.0099603481047697%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 422\n",
      "Length of fail 3351\n",
      "G900114375 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylorubrum; s__Methylorubrum salsuginis\n",
      "Coverage Percentage: 0.0553611592641773%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 422\n",
      "Length of fail 3352\n",
      "G900114535 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium pseudosasicola\n",
      "Coverage Percentage: 0.0129529952304647%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 422\n",
      "Length of fail 3353\n",
      "G900114705 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Rugamonas; s__Rugamonas rubra\n",
      "Coverage Percentage: 0.1072580221404093%\n",
      "Adj, read counts 47\n",
      "Number of samples per group [1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1]\n",
      "Number of bins 41\n",
      "total # samples 23\n",
      "Number of total reads 47\n",
      "Number of hits in BIOM tables 67.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.893617021276595 %\n",
      "Percentage of reads from samples who only fall into one bin group 27.659574468085108 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 423\n",
      "Length of fail 3353\n",
      "G900114825 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas yangmingensis\n",
      "Coverage Percentage: 0.0189980909623503%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 423\n",
      "Length of fail 3354\n",
      "G900114845 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Pimelobacter; s__Pimelobacter simplex\n",
      "Coverage Percentage: 0.0080432453163188%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 423\n",
      "Length of fail 3355\n",
      "G900114945 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium succinicans\n",
      "Coverage Percentage: 0.0525106357905946%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 423\n",
      "Length of fail 3356\n",
      "G900115005 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Pseudonocardia; s__Pseudonocardia ammonioxydans\n",
      "Coverage Percentage: 0.0274548051408104%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 423\n",
      "Length of fail 3357\n",
      "G900115015 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Algoriella; s__Algoriella xinjiangensis\n",
      "Coverage Percentage: 0.0235400644856955%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 423\n",
      "Length of fail 3358\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G900115515 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Pseudonocardiales; f__Pseudonocardiaceae; g__Saccharomonospora; s__Saccharomonospora viridis\n",
      "Coverage Percentage: 0.0028548655961771%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 423\n",
      "Length of fail 3359\n",
      "G900115555 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas borbori\n",
      "Coverage Percentage: 0.0080798532273797%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 423\n",
      "Length of fail 3360\n",
      "G900115695 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas toyotomiensis\n",
      "Coverage Percentage: 0.1244305281704377%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 423\n",
      "Length of fail 3361\n",
      "G900115745 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas rubra\n",
      "Coverage Percentage: 0.1002911254538876%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 423\n",
      "Length of fail 3362\n",
      "G900115805 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Psychrobacillus; s__Psychrobacillus psychrotolerans\n",
      "Coverage Percentage: 0.0041850907665811%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 423\n",
      "Length of fail 3363\n",
      "G900115825 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Desemzia; s__Desemzia incerta\n",
      "Coverage Percentage: 0.0129549904167388%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 423\n",
      "Length of fail 3364\n",
      "G900115905 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Halopseudomonas; s__Halopseudomonas formosensis\n",
      "Coverage Percentage: 0.0110336465243577%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 423\n",
      "Length of fail 3365\n",
      "G900116065 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas viscosa\n",
      "Coverage Percentage: 0.0268918553711976%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 423\n",
      "Length of fail 3366\n",
      "G900116175 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Hyphomicrobiaceae; g__Hyphomicrobium; s__Hyphomicrobium facile\n",
      "Coverage Percentage: 0.0114791001289365%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 423\n",
      "Length of fail 3367\n",
      "G900116415 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium formosense\n",
      "Coverage Percentage: 0.0060457500858361%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 423\n",
      "Length of fail 3368\n",
      "G900116515 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Geodermatophilus; s__Geodermatophilus amargosae\n",
      "Coverage Percentage: 0.005143863821826%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 423\n",
      "Length of fail 3369\n",
      "G900116645 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Massilia; s__Massilia namucuonensis\n",
      "Coverage Percentage: 0.0188667001682104%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 423\n",
      "Length of fail 3370\n",
      "G900116825 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax caeni\n",
      "Coverage Percentage: 0.1261970122381977%\n",
      "Adj, read counts 35\n",
      "Number of samples per group [5, 1, 1, 1, 2, 2, 3, 1, 1, 1, 1, 1, 1, 1, 2, 6, 1, 1]\n",
      "Number of bins 22\n",
      "total # samples 20\n",
      "Number of total reads 34\n",
      "Number of hits in BIOM tables 12.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 42.85714285714286 %\n",
      "Percentage of reads from samples who only fall into one bin group 38.23529411764706 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3370\n",
      "G900119105 d__Bacteria; p__Bacteroidetes; c__Chitinophagia; o__Chitinophagales; f__Chitinophagaceae; g__Chitinophaga; s__Chitinophaga sancti\n",
      "Coverage Percentage: 0.0108296913998788%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3371\n",
      "G900119565 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Cytophagaceae; g__Cytophaga; s__Cytophaga hutchinsonii\n",
      "Coverage Percentage: 0.0103507841918619%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3372\n",
      "G900119835 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Lactococcus; s__Lactococcus chungangensis\n",
      "Coverage Percentage: 0.0335770224853224%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3373\n",
      "G900120405 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Urinicoccus; s__Urinicoccus massiliensis\n",
      "Coverage Percentage: 0.0228993005652965%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3374\n",
      "G900128425 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Anaerococcus; s__Anaerococcus urinomassiliensis\n",
      "Coverage Percentage: 0.0783914816720533%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3375\n",
      "G900128455 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides mediterraneensis\n",
      "Coverage Percentage: 0.0481595353917384%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3376\n",
      "G900128465 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces bouchesdurhonensis\n",
      "Coverage Percentage: 0.0103881201631673%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3377\n",
      "G900129025 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Marinilabiliales; f__Prolixibacteraceae; g__Mariniphaga; s__Mariniphaga anaerophila\n",
      "Coverage Percentage: 0.0029905834251963%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3378\n",
      "G900129065 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides faecichinchillae\n",
      "Coverage Percentage: 0.0137263734796481%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3379\n",
      "G900129205 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Thermomonas; s__Thermomonas hydrothermalis\n",
      "Coverage Percentage: 0.0639487691596348%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3380\n",
      "G900129325 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Kaistiaceae; g__Kaistia; s__Kaistia soli\n",
      "Coverage Percentage: 0.0057463902264784%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3381\n",
      "G900129405 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium fontis\n",
      "Coverage Percentage: 0.0090103755446739%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3382\n",
      "G900129535 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella scopos\n",
      "Coverage Percentage: 0.0274325528185624%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3383\n",
      "G900129545 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium fluvii\n",
      "Coverage Percentage: 0.057462761200698%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3384\n",
      "G900129625 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Erwiniaceae; g__Pantoea; s__Pantoea sesami\n",
      "Coverage Percentage: 0.0200114368517022%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3385\n",
      "G900129655 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides clarus\n",
      "Coverage Percentage: 0.035928162878606%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3386\n",
      "G900129705 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium granuli\n",
      "Coverage Percentage: 0.0629146542971367%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3387\n",
      "G900129725 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Fulvivirgaceae; g__Chryseolinea; s__Chryseolinea serpens\n",
      "Coverage Percentage: 0.0056115400063916%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3388\n",
      "G900130075 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Herminiimonas; s__Herminiimonas arsenitoxidans\n",
      "Coverage Percentage: 0.0407470611644569%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3389\n",
      "G900130125 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides congonensis\n",
      "Coverage Percentage: 0.0112339506305525%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3390\n",
      "G900130135 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides togonis\n",
      "Coverage Percentage: 0.0568441371972597%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3391\n",
      "G900141905 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Roseomonas; s__Roseomonas rosea\n",
      "Coverage Percentage: 0.0049025276459521%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3392\n",
      "G900141915 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Tessaracoccus; s__Tessaracoccus bendigoensis\n",
      "Coverage Percentage: 0.0175592625109745%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3393\n",
      "G900141975 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Aurantimonadaceae; g__Aureimonas; s__Aureimonas altamirensis\n",
      "Coverage Percentage: 0.0273909521576936%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3394\n",
      "G900142015 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides stercorirosoris\n",
      "Coverage Percentage: 0.0107815318738605%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3395\n",
      "G900142395 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Hymenobacteraceae; g__Hymenobacter; s__Hymenobacter psychrotolerans\n",
      "Coverage Percentage: 0.0054308999756133%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3396\n",
      "G900142655 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas punonensis\n",
      "Coverage Percentage: 0.0062785627829251%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3397\n",
      "G900142675 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Alkalihalobacillus; s__Alkalihalobacillus rhizosphaerae\n",
      "Coverage Percentage: 0.0113827146761134%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3398\n",
      "G900142685 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium chilense\n",
      "Coverage Percentage: 0.0326771036048994%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 424\n",
      "Length of fail 3399\n",
      "G900142715 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium pectinovorum\n",
      "Coverage Percentage: 0.1789979004636943%\n",
      "Adj, read counts 63\n",
      "Number of samples per group [4, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 2, 1, 1, 3, 2, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 2, 1]\n",
      "Number of bins 57\n",
      "total # samples 29\n",
      "Number of total reads 61\n",
      "Number of hits in BIOM tables 42.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.46031746031746 %\n",
      "Percentage of reads from samples who only fall into one bin group 22.950819672131146 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 425\n",
      "Length of fail 3399\n",
      "G900142735 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium saccharophilum\n",
      "Coverage Percentage: 0.0706776318731933%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 425\n",
      "Length of fail 3400\n",
      "G900143035 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus salivarius\n",
      "Coverage Percentage: 0.053621211441842%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 425\n",
      "Length of fail 3401\n",
      "G900143065 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Duganella; s__Duganella sacchari\n",
      "Coverage Percentage: 0.0272901727219296%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 425\n",
      "Length of fail 3402\n",
      "G900143095 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas asturiensis\n",
      "Coverage Percentage: 0.0307048711269918%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 425\n",
      "Length of fail 3403\n",
      "G900143175 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Xanthomonas; s__Xanthomonas retroflexus\n",
      "Coverage Percentage: 0.121580142382567%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 425\n",
      "Length of fail 3404\n",
      "G900143185 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium scophthalmum\n",
      "Coverage Percentage: 0.018641381225645%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 425\n",
      "Length of fail 3405\n",
      "G900143235 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Erythrobacteraceae; g__Erythrobacter; s__Erythrobacter sanguineus\n",
      "Coverage Percentage: 0.041855778480558%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 425\n",
      "Length of fail 3406\n",
      "G900148495 d__Bacteria; p__Firmicutes; c__; o__; f__; g__Negativibacillus; s__Negativibacillus massiliensis\n",
      "Coverage Percentage: 0.0376083455769005%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 425\n",
      "Length of fail 3407\n",
      "G900148835 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium cucumis\n",
      "Coverage Percentage: 0.0274282399477076%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 425\n",
      "Length of fail 3408\n",
      "G900155435 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces provencensis\n",
      "Coverage Percentage: 0.0178836078671155%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 425\n",
      "Length of fail 3409\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G900155655 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella ihumii\n",
      "Coverage Percentage: 0.0364647266950682%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 425\n",
      "Length of fail 3410\n",
      "G900155925 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Promicromonosporaceae; g__Cellulosimicrobium; s__Cellulosimicrobium aquatile\n",
      "Coverage Percentage: 0.0053960516649945%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 425\n",
      "Length of fail 3411\n",
      "G900155965 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Mucilaginibacter; s__Mucilaginibacter lappiensis\n",
      "Coverage Percentage: 0.0060074165090578%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 425\n",
      "Length of fail 3412\n",
      "G900156335 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__; g__Sphaerotilus; s__Sphaerotilus natans\n",
      "Coverage Percentage: 0.1904040915502948%\n",
      "Adj, read counts 61\n",
      "Number of samples per group [1, 2, 2, 2, 2, 2, 1, 2, 2, 1, 1, 1, 1, 1, 3, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 5, 4]\n",
      "Number of bins 47\n",
      "total # samples 30\n",
      "Number of total reads 61\n",
      "Number of hits in BIOM tables 20.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 21.311475409836063 %\n",
      "Percentage of reads from samples who only fall into one bin group 26.229508196721312 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 426\n",
      "Length of fail 3412\n",
      "G900156575 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium shigense\n",
      "Coverage Percentage: 0.0055960657615336%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 426\n",
      "Length of fail 3413\n",
      "G900156725 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Kaistella; s__Kaistella chaponensis\n",
      "Coverage Percentage: 0.1967928352952385%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1]\n",
      "Number of bins 32\n",
      "total # samples 22\n",
      "Number of total reads 37\n",
      "Number of hits in BIOM tables 69.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.216216216216218 %\n",
      "Percentage of reads from samples who only fall into one bin group 43.24324324324324 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 427\n",
      "Length of fail 3413\n",
      "G900157305 d__Bacteria; p__Thermotogae; c__Thermotogae; o__Kosmotogales; f__Kosmotogaceae; g__Mesotoga; s__Mesotoga infera\n",
      "Coverage Percentage: 0.0089356163552126%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 427\n",
      "Length of fail 3414\n",
      "G900162815 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter pasteurii\n",
      "Coverage Percentage: 0.1225644931579754%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 427\n",
      "Length of fail 3415\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G900162825 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter piechaudii\n",
      "Coverage Percentage: 0.0609153001405084%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 427\n",
      "Length of fail 3416\n",
      "G900163675 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium casei\n",
      "Coverage Percentage: 0.0219312947448103%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 427\n",
      "Length of fail 3417\n",
      "G900163825 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Micrococcus; s__Micrococcus lylae\n",
      "Coverage Percentage: 0.0489458127194001%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 427\n",
      "Length of fail 3418\n",
      "G900163835 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Mycetocola; s__Mycetocola reblochoni\n",
      "Coverage Percentage: 0.0259329062475634%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 427\n",
      "Length of fail 3419\n",
      "G900166985 d__Bacteria; p__Tenericutes; c__Mollicutes; o__Acholeplasmatales; f__Acholeplasmataceae; g__Acholeplasma; s__Acholeplasma oculi\n",
      "Coverage Percentage: 0.0154971717661526%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 427\n",
      "Length of fail 3420\n",
      "G900167005 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Butyricicoccus; s__Butyricicoccus pullicaecorum\n",
      "Coverage Percentage: 0.0069386685887084%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 427\n",
      "Length of fail 3421\n",
      "G900167175 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Plantibacter; s__Plantibacter cousiniae\n",
      "Coverage Percentage: 0.260477314737374%\n",
      "Adj, read counts 68\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 5, 2, 1, 1, 1, 1, 1, 2, 1, 1, 3, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 2, 2, 1, 1, 1]\n",
      "Number of bins 54\n",
      "total # samples 29\n",
      "Number of total reads 65\n",
      "Number of hits in BIOM tables 64.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 17.647058823529413 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.923076923076923 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 428\n",
      "Length of fail 3421\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G900167195 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas aeruginosa\n",
      "Coverage Percentage: 0.5489905447297226%\n",
      "Adj, read counts 206\n",
      "Number of samples per group [1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 3, 1, 1, 1, 2, 1, 1, 5, 1, 3, 2, 2, 1, 1, 2, 2, 1, 1, 1, 2, 1, 2, 1, 1, 3, 2, 1, 1, 1, 3, 1, 4, 1, 1, 2, 1, 1, 1, 1, 1, 1, 4, 1, 1, 1, 2, 1, 1, 3, 1, 1, 1, 2, 2, 1, 2, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 2, 2, 1, 3, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 3, 2, 1, 1, 2, 2, 1, 1, 3, 3, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2]\n",
      "Number of bins 169\n",
      "total # samples 51\n",
      "Number of total reads 201\n",
      "Number of hits in BIOM tables 54.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 7.766990291262135 %\n",
      "Percentage of reads from samples who only fall into one bin group 12.437810945273633 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 429\n",
      "Length of fail 3421\n",
      "G900167255 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Thiotrichales; f__Thiotrichaceae; g__Thiothrix; s__Thiothrix eikelboomii\n",
      "Coverage Percentage: 0.0036300633674439%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 429\n",
      "Length of fail 3422\n",
      "G900167355 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Porphyromonadaceae; g__Porphyromonas; s__Porphyromonas cangingivalis\n",
      "Coverage Percentage: 0.0233451240985425%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 429\n",
      "Length of fail 3423\n",
      "G900167395 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Nitrosomonadaceae; g__Nitrosomonas; s__Nitrosomonas europaea\n",
      "Coverage Percentage: 0.0131853554165513%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 429\n",
      "Length of fail 3424\n",
      "G900167455 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__; g__Enhydrobacter; s__Enhydrobacter aerosaccus\n",
      "Coverage Percentage: 0.0085224620787363%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 429\n",
      "Length of fail 3425\n",
      "G900167535 d__Bacteria; p__Verrucomicrobia; c__Verrucomicrobiae; o__Verrucomicrobiales; f__Verrucomicrobiaceae; g__Prosthecobacter; s__Prosthecobacter debontii\n",
      "Coverage Percentage: 0.008214882252295%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 429\n",
      "Length of fail 3426\n",
      "G900167545 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Agreia; s__Agreia bicolorata\n",
      "Coverage Percentage: 0.0311873399520659%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 429\n",
      "Length of fail 3427\n",
      "G900167555 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__; g__Gemmiger; s__Gemmiger formicilis\n",
      "Coverage Percentage: 0.2593275120560623%\n",
      "Adj, read counts 55\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 2, 1, 2, 1, 1, 2, 1, 2, 1, 1, 1, 1, 2, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1]\n",
      "Number of bins 46\n",
      "total # samples 21\n",
      "Number of total reads 52\n",
      "Number of hits in BIOM tables 37.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.727272727272727 %\n",
      "Percentage of reads from samples who only fall into one bin group 26.923076923076923 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 430\n",
      "Length of fail 3427\n",
      "G900167575 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Okibacterium; s__Okibacterium fritillariae\n",
      "Coverage Percentage: 0.0334895648618945%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 430\n",
      "Length of fail 3428\n",
      "G900167905 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Soonwooa; s__Soonwooa buanensis\n",
      "Coverage Percentage: 0.0045340470904329%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 430\n",
      "Length of fail 3429\n",
      "G900167915 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Rhizorhabdus; s__Rhizorhabdus histidinilytica\n",
      "Coverage Percentage: 0.0210612879826994%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 430\n",
      "Length of fail 3430\n",
      "G900167945 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Spirosomaceae; g__Dyadobacter; s__Dyadobacter psychrophilus\n",
      "Coverage Percentage: 0.0088162611038137%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 430\n",
      "Length of fail 3431\n",
      "G900167955 d__Archaea; p__Thaumarchaeota; c__; o__; f__; g__Candidatus Nitrosotalea; s__Candidatus Nitrosotalea bavarica\n",
      "Coverage Percentage: 0.1001597019412057%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 430\n",
      "Length of fail 3432\n",
      "G900168155 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Tannerellaceae; g__Parabacteroides; s__Parabacteroides chartae\n",
      "Coverage Percentage: 0.0656428995625338%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 430\n",
      "Length of fail 3433\n",
      "G900168195 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Boseaceae; g__Bosea; s__Bosea thiooxidans\n",
      "Coverage Percentage: 0.0052513643770904%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 430\n",
      "Length of fail 3434\n",
      "G900168205 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium balustinum\n",
      "Coverage Percentage: 0.0311366011878829%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 430\n",
      "Length of fail 3435\n",
      "G900168345 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Clavibacter; s__Clavibacter michiganensis\n",
      "Coverage Percentage: 0.0834050624961506%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 430\n",
      "Length of fail 3436\n",
      "G900169065 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Brevibacteriaceae; g__Brevibacterium; s__Brevibacterium aurantiacum\n",
      "Coverage Percentage: 0.0320809935418622%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 430\n",
      "Length of fail 3437\n",
      "G900169275 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Brevibacteriaceae; g__Brevibacterium; s__Brevibacterium casei\n",
      "Coverage Percentage: 0.3310412622262811%\n",
      "Adj, read counts 67\n",
      "Number of samples per group [1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 3, 1, 1, 1]\n",
      "Number of bins 56\n",
      "total # samples 5\n",
      "Number of total reads 65\n",
      "Number of hits in BIOM tables 60.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.940298507462686 %\n",
      "Percentage of reads from samples who only fall into one bin group 1.5384615384615385 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 431\n",
      "Length of fail 3437\n",
      "G900172365 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Roseobacteraceae; g__Roseovarius; s__Roseovarius gaetbuli\n",
      "Coverage Percentage: 0.0036532879712711%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 431\n",
      "Length of fail 3438\n",
      "G900176155 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium glucuronolyticum\n",
      "Coverage Percentage: 0.2373250626889705%\n",
      "Adj, read counts 38\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 30\n",
      "total # samples 18\n",
      "Number of total reads 38\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 18.421052631578945 %\n",
      "Percentage of reads from samples who only fall into one bin group 26.31578947368421 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 432\n",
      "Length of fail 3438\n",
      "G900176325 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Aerococcaceae; g__Aerococcus; s__Aerococcus suis\n",
      "Coverage Percentage: 0.0184457135416855%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 432\n",
      "Length of fail 3439\n",
      "G900176385 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Intrasporangiaceae; g__Janibacter; s__Janibacter indicus\n",
      "Coverage Percentage: 0.0307885824357715%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 432\n",
      "Length of fail 3440\n",
      "G900177335 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Kocuria; s__Kocuria indica\n",
      "Coverage Percentage: 0.0147709666051035%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 432\n",
      "Length of fail 3441\n",
      "G900177405 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Allosphingosinicella; s__Allosphingosinicella indica\n",
      "Coverage Percentage: 0.0302776316337845%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 432\n",
      "Length of fail 3442\n",
      "G900177615 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Plantibacter; s__Plantibacter flavus\n",
      "Coverage Percentage: 0.2312883407049806%\n",
      "Adj, read counts 62\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 3, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1]\n",
      "Number of bins 49\n",
      "total # samples 29\n",
      "Number of total reads 60\n",
      "Number of hits in BIOM tables 50.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.516129032258062 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.333333333333332 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 433\n",
      "Length of fail 3442\n",
      "G900177685 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Agreia; s__Agreia pratensis\n",
      "Coverage Percentage: 0.0981337286965239%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 433\n",
      "Length of fail 3443\n",
      "G900180435 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Enterobacter; s__Enterobacter asburiae\n",
      "Coverage Percentage: 0.0116457858648688%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 433\n",
      "Length of fail 3444\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G900182595 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter westerhofensis\n",
      "Coverage Percentage: 0.0527567946433889%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 433\n",
      "Length of fail 3445\n",
      "G900182685 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Dietziaceae; g__Dietzia; s__Dietzia kunjamensis\n",
      "Coverage Percentage: 0.0241749410564504%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 433\n",
      "Length of fail 3446\n",
      "G900184245 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Dermabacteraceae; g__Brachybacterium; s__Brachybacterium massiliense\n",
      "Coverage Percentage: 0.0250670783634638%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 433\n",
      "Length of fail 3447\n",
      "G900185755 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Cupriavidus; s__Cupriavidus metallidurans\n",
      "Coverage Percentage: 0.0199795857604521%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 433\n",
      "Length of fail 3448\n",
      "G900186465 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Selenomonadales; f__Selenomonadaceae; g__Selenomonas; s__Selenomonas felix\n",
      "Coverage Percentage: 0.0052424159715611%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 433\n",
      "Length of fail 3449\n",
      "G900186865 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas maltophilia\n",
      "Coverage Percentage: 0.3199473408761054%\n",
      "Adj, read counts 100\n",
      "Number of samples per group [1, 1, 2, 3, 1, 2, 2, 2, 1, 1, 1, 1, 2, 1, 1, 1, 2, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 4, 1, 2, 2, 1, 2, 1, 1, 1, 1, 1, 2, 2, 2, 1, 1, 2, 1, 1, 2, 1, 1, 1, 3, 1]\n",
      "Number of bins 82\n",
      "total # samples 27\n",
      "Number of total reads 97\n",
      "Number of hits in BIOM tables 21.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 14.432989690721648 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 434\n",
      "Length of fail 3449\n",
      "G900186885 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Veillonellales; f__Veillonellaceae; g__Veillonella; s__Veillonella parvula\n",
      "Coverage Percentage: 0.0503249783551001%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 434\n",
      "Length of fail 3450\n",
      "G900186955 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Bacillaceae; g__Bacillus; s__Bacillus pumilus\n",
      "Coverage Percentage: 0.0324457480378881%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 434\n",
      "Length of fail 3451\n",
      "G900186975 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Propionibacteriaceae; g__Cutibacterium; s__Cutibacterium granulosum\n",
      "Coverage Percentage: 0.6480192566902839%\n",
      "Adj, read counts 84\n",
      "Number of samples per group [1, 1, 2, 1, 7, 2, 1, 1, 1, 2, 1, 2, 4, 1, 1, 1, 3, 1, 8, 2, 1, 1, 3, 2, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 3, 1]\n",
      "Number of bins 61\n",
      "total # samples 31\n",
      "Number of total reads 81\n",
      "Number of hits in BIOM tables 59.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 25.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 28.39506172839506 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 435\n",
      "Length of fail 3451\n",
      "G900187035 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Selenomonadales; f__Selenomonadaceae; g__Megamonas; s__Megamonas hypermegale\n",
      "Coverage Percentage: 0.0263188336138583%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 435\n",
      "Length of fail 3452\n",
      "G900187175 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Odoribacteraceae; g__Odoribacter; s__Odoribacter splanchnicus\n",
      "Coverage Percentage: 0.022403468712668%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 435\n",
      "Length of fail 3453\n",
      "G900187185 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium taklimakanense\n",
      "Coverage Percentage: 0.0222071120627296%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 435\n",
      "Length of fail 3454\n",
      "G900187975 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas delhiensis\n",
      "Coverage Percentage: 0.4037283233987662%\n",
      "Adj, read counts 163\n",
      "Number of samples per group [2, 3, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 2, 2, 1, 1, 2, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1]\n",
      "Number of bins 117\n",
      "total # samples 13\n",
      "Number of total reads 161\n",
      "Number of hits in BIOM tables 48.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 7.9754601226993875 %\n",
      "Percentage of reads from samples who only fall into one bin group 4.3478260869565215 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3454\n",
      "G900187995 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella jejuni\n",
      "Coverage Percentage: 0.0195134892475741%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3455\n",
      "G900188005 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Actinoplanes; s__Actinoplanes regularis\n",
      "Coverage Percentage: 0.0165849280877916%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3456\n",
      "G900188015 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas lactitubi\n",
      "Coverage Percentage: 0.107217749541044%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3457\n",
      "G900188095 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Noviherbaspirillum; s__Noviherbaspirillum humi\n",
      "Coverage Percentage: 0.0084166091469775%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3458\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G900188165 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas laterariae\n",
      "Coverage Percentage: 0.0091606824414372%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3459\n",
      "G900188265 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Cereibacter; s__Cereibacter sphaeroides\n",
      "Coverage Percentage: 0.0110927877739779%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3460\n",
      "G900188295 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus sediminis\n",
      "Coverage Percentage: 0.0068009842779342%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3461\n",
      "G900188345 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptosporangiales; f__Streptosporangiaceae; g__Streptosporangium; s__Streptosporangium subroseum\n",
      "Coverage Percentage: 0.0047746397835595%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3462\n",
      "G900188445 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptosporangiales; f__Thermomonosporaceae; g__Actinomadura; s__Actinomadura meyerae\n",
      "Coverage Percentage: 0.0017608590193298%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3463\n",
      "G900197855 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Helicobacteraceae; g__Helicobacter; s__Helicobacter cynogastricus\n",
      "Coverage Percentage: 0.0071090675554206%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3464\n",
      "G900199185 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Sphingomonadales; f__Sphingomonadaceae; g__Sphingomonas; s__Sphingomonas guangdongensis\n",
      "Coverage Percentage: 0.0212980588582435%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3465\n",
      "G900199455 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Phyllobacteriaceae; g__Mesorhizobium; s__Mesorhizobium carmichaelinearum\n",
      "Coverage Percentage: 0.0070508769134827%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3466\n",
      "G900199515 d__Bacteria; p__Firmicutes; c__Erysipelotrichia; o__Erysipelotrichales; f__Erysipelotrichaceae; g__Massilicoli; s__Massilicoli timonensis\n",
      "Coverage Percentage: 0.0111681128043211%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3467\n",
      "G900200035 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Klebsiella; s__Klebsiella grimontii\n",
      "Coverage Percentage: 0.0337928950964807%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3468\n",
      "G900207575 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Cereibacter; s__Cereibacter ovatus\n",
      "Coverage Percentage: 0.0105849706549417%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3469\n",
      "G900215245 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas fluorescens\n",
      "Coverage Percentage: 0.0288410726360417%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3470\n",
      "G900215295 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Salinimicrobium; s__Salinimicrobium sediminis\n",
      "Coverage Percentage: 0.0031012836155453%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3471\n",
      "G900215335 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Pedobacter; s__Pedobacter xixiisoli\n",
      "Coverage Percentage: 0.0060705445519676%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3472\n",
      "G900221005 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Blastococcus; s__Blastococcus aggregatus\n",
      "Coverage Percentage: 0.0190886407654155%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3473\n",
      "G900221015 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Ornithinimicrobiaceae; g__Ornithinimicrobium; s__Ornithinimicrobium cerasi\n",
      "Coverage Percentage: 0.0094462187342066%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3474\n",
      "G900230225 d__Bacteria; p__Bacteroidetes; c__Cytophagia; o__Cytophagales; f__Cytophagaceae; g__Spirosoma; s__Spirosoma fluviale\n",
      "Coverage Percentage: 0.0018699647037758%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3475\n",
      "G900231165 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Pleomorphomonadaceae; g__Hartmannibacter; s__Hartmannibacter diazotrophicus\n",
      "Coverage Percentage: 0.0027707464225718%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3476\n",
      "G900232865 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium dentalis\n",
      "Coverage Percentage: 0.0340387546852131%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3477\n",
      "G900232945 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Urinicoccus; s__Urinicoccus timonensis\n",
      "Coverage Percentage: 0.0076295456428725%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3478\n",
      "G900234795 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylorubrum; s__Methylorubrum extorquens\n",
      "Coverage Percentage: 0.0692151464295762%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3479\n",
      "G900240005 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia stabilis\n",
      "Coverage Percentage: 0.0139555625261519%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3480\n",
      "G900240315 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Dorea; s__Dorea phocaeensis\n",
      "Coverage Percentage: 0.0321333375475415%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3481\n",
      "G900241005 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides cutis\n",
      "Coverage Percentage: 0.0499233859172506%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3482\n",
      "G900243045 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Eisenbergiella; s__Eisenbergiella massiliensis\n",
      "Coverage Percentage: 0.0335915903523828%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3483\n",
      "G900258545 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Odoribacteraceae; g__Butyricimonas; s__Butyricimonas vaginalis\n",
      "Coverage Percentage: 0.0298978533867567%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3484\n",
      "G900277125 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas inefficax\n",
      "Coverage Percentage: 0.0416334666458285%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3485\n",
      "G900289045 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Propionibacteriales; f__Nocardioidaceae; g__Micropruina; s__Micropruina glycogenica\n",
      "Coverage Percentage: 0.0307177950189313%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3486\n",
      "G900289145 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Oscillospiraceae; g__Neobittarella; s__Neobittarella massiliensis\n",
      "Coverage Percentage: 0.0197594297162651%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3487\n",
      "G900291065 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas shirazica\n",
      "Coverage Percentage: 0.0588395444726651%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3488\n",
      "G900291955 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Oscillospiraceae; g__Pseudoruminococcus; s__Pseudoruminococcus massiliensis\n",
      "Coverage Percentage: 0.0272977680060277%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3489\n",
      "G900292075 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Microbacteriaceae; g__Microbacterium; s__Microbacterium timonense\n",
      "Coverage Percentage: 0.007623478901391%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3490\n",
      "G900323515 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter stercoris\n",
      "Coverage Percentage: 0.0170821158632242%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3491\n",
      "G900324035 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium vignae\n",
      "Coverage Percentage: 0.0147007958905451%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3492\n",
      "G900403705 d__Bacteria; p__Nitrospirae; c__Nitrospira; o__Nitrospirales; f__Nitrospiraceae; g__Nitrospira; s__Nitrospira lenta\n",
      "Coverage Percentage: 0.0040195814843702%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 436\n",
      "Length of fail 3493\n",
      "G900416805 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Carnobacteriaceae; g__Trichococcus; s__Trichococcus shcherbakoviae\n",
      "Coverage Percentage: 0.2357066403502874%\n",
      "Adj, read counts 46\n",
      "Number of samples per group [1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 3, 1, 1, 1, 2, 1, 1, 3, 1, 2, 1, 1, 1, 1, 1]\n",
      "Number of bins 39\n",
      "total # samples 22\n",
      "Number of total reads 44\n",
      "Number of hits in BIOM tables 31.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 23.913043478260867 %\n",
      "Percentage of reads from samples who only fall into one bin group 31.818181818181817 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 437\n",
      "Length of fail 3493\n",
      "G900444665 d__Bacteria; p__Tenericutes; c__Mollicutes; o__Acholeplasmatales; f__Acholeplasmataceae; g__Acholeplasma; s__Acholeplasma oculi\n",
      "Coverage Percentage: 0.015494228399921%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 437\n",
      "Length of fail 3494\n",
      "G900444725 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter baumannii\n",
      "Coverage Percentage: 0.0322412992992907%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 437\n",
      "Length of fail 3495\n",
      "G900444845 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter haemolyticus\n",
      "Coverage Percentage: 0.1452995122978238%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 437\n",
      "Length of fail 3496\n",
      "G900444865 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter junii\n",
      "Coverage Percentage: 0.2252214543093148%\n",
      "Adj, read counts 51\n",
      "Number of samples per group [1, 4, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 4, 2, 1, 2, 1, 5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 5]\n",
      "Number of bins 33\n",
      "total # samples 24\n",
      "Number of total reads 51\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 27.450980392156865 %\n",
      "Percentage of reads from samples who only fall into one bin group 25.49019607843137 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 438\n",
      "Length of fail 3496\n",
      "G900445095 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Aerococcaceae; g__Aerococcus; s__Aerococcus viridans\n",
      "Coverage Percentage: 0.1675199368588886%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 438\n",
      "Length of fail 3497\n",
      "G900445115 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas salmonicida\n",
      "Coverage Percentage: 0.0451493159615193%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 438\n",
      "Length of fail 3498\n",
      "G900445155 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Afipia; s__Afipia felis\n",
      "Coverage Percentage: 0.0128716172972607%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 438\n",
      "Length of fail 3499\n",
      "G900445265 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Anaerococcus; s__Anaerococcus octavius\n",
      "Coverage Percentage: 0.0618499616049647%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 438\n",
      "Length of fail 3500\n",
      "G900445285 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Anaerococcus; s__Anaerococcus prevotii\n",
      "Coverage Percentage: 0.1199503041650436%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 438\n",
      "Length of fail 3501\n",
      "G900445295 d__Bacteria; p__Proteobacteria; c__Epsilonproteobacteria; o__Campylobacterales; f__Arcobacteraceae; g__Aliarcobacter; s__Aliarcobacter skirrowii\n",
      "Coverage Percentage: 0.1990679081044153%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 438\n",
      "Length of fail 3502\n",
      "G900445495 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Tannerellaceae; g__Parabacteroides; s__Parabacteroides merdae\n",
      "Coverage Percentage: 0.0790794982880492%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 438\n",
      "Length of fail 3503\n",
      "G900445505 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides ovatus\n",
      "Coverage Percentage: 0.1264007251865148%\n",
      "Adj, read counts 53\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 3, 2]\n",
      "Number of bins 46\n",
      "total # samples 33\n",
      "Number of total reads 52\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.9811320754717 %\n",
      "Percentage of reads from samples who only fall into one bin group 40.38461538461539 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 439\n",
      "Length of fail 3503\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G900445525 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Tannerellaceae; g__Parabacteroides; s__Parabacteroides distasonis\n",
      "Coverage Percentage: 0.1185730482397548%\n",
      "Adj, read counts 36\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1]\n",
      "Number of bins 32\n",
      "total # samples 21\n",
      "Number of total reads 36\n",
      "Number of hits in BIOM tables 13.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.666666666666664 %\n",
      "Percentage of reads from samples who only fall into one bin group 38.88888888888889 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 440\n",
      "Length of fail 3503\n",
      "G900445565 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides eggerthii\n",
      "Coverage Percentage: 0.0506896232186596%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 440\n",
      "Length of fail 3504\n",
      "G900445595 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides thetaiotaomicron\n",
      "Coverage Percentage: 0.1061225879047729%\n",
      "Adj, read counts 43\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 40\n",
      "total # samples 23\n",
      "Number of total reads 42\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 13.953488372093023 %\n",
      "Percentage of reads from samples who only fall into one bin group 30.952380952380953 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 441\n",
      "Length of fail 3504\n",
      "G900445615 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium adolescentis\n",
      "Coverage Percentage: 0.5635704064116389%\n",
      "Adj, read counts 71\n",
      "Number of samples per group [1, 1, 1, 2, 1, 1, 1, 2, 2, 1, 1, 2, 1, 2, 2, 1, 1, 1, 1, 2, 3, 1, 1, 1, 1, 3, 2, 1, 3, 2, 1, 3, 1, 2, 4, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1]\n",
      "Number of bins 64\n",
      "total # samples 32\n",
      "Number of total reads 71\n",
      "Number of hits in BIOM tables 23.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.084507042253522 %\n",
      "Percentage of reads from samples who only fall into one bin group 23.943661971830984 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 442\n",
      "Length of fail 3504\n",
      "G900445655 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Bergeyella; s__Bergeyella zoohelcum\n",
      "Coverage Percentage: 0.0129531011954597%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 442\n",
      "Length of fail 3505\n",
      "G900445945 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Bordetella; s__Bordetella trematum\n",
      "Coverage Percentage: 0.0286110311420069%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 442\n",
      "Length of fail 3506\n",
      "G900445995 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas diminuta\n",
      "Coverage Percentage: 0.0574904000918915%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 442\n",
      "Length of fail 3507\n",
      "G900446175 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia cepacia\n",
      "Coverage Percentage: 0.0079367370753042%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 442\n",
      "Length of fail 3508\n",
      "G900446205 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia multivorans\n",
      "Coverage Percentage: 0.0181826727199616%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 442\n",
      "Length of fail 3509\n",
      "G900446785 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Chryseobacterium; s__Chryseobacterium balustinum\n",
      "Coverage Percentage: 0.0277385249175983%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 442\n",
      "Length of fail 3510\n",
      "G900447045 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Clostridium; s__Clostridium paraputrificum\n",
      "Coverage Percentage: 0.0499474063734213%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 442\n",
      "Length of fail 3511\n",
      "G900447555 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium matruchotii\n",
      "Coverage Percentage: 0.0393995991397228%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 442\n",
      "Length of fail 3512\n",
      "G900447675 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium striatum\n",
      "Coverage Percentage: 0.0817246569787044%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 442\n",
      "Length of fail 3513\n",
      "G900450525 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium hibernum\n",
      "Coverage Percentage: 0.1143537653821474%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 442\n",
      "Length of fail 3514\n",
      "G900450845 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pasteurellales; f__Pasteurellaceae; g__Haemophilus; s__Haemophilus parainfluenzae\n",
      "Coverage Percentage: 0.0410543234345246%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 442\n",
      "Length of fail 3515\n",
      "G900451015 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Ewingella; s__Ewingella americana\n",
      "Coverage Percentage: 0.0082763449313457%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 442\n",
      "Length of fail 3516\n",
      "G900451145 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Oxalobacteraceae; g__Janthinobacterium; s__Janthinobacterium lividum\n",
      "Coverage Percentage: 0.1382169393556907%\n",
      "Adj, read counts 64\n",
      "Number of samples per group [7, 1, 1, 1, 1, 1, 1, 4, 1, 1, 6, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 2, 3]\n",
      "Number of bins 39\n",
      "total # samples 25\n",
      "Number of total reads 63\n",
      "Number of hits in BIOM tables 10.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 35.9375 %\n",
      "Percentage of reads from samples who only fall into one bin group 22.22222222222222 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 443\n",
      "Length of fail 3516\n",
      "G900451195 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Chromobacteriaceae; g__Iodobacter; s__Iodobacter fluviatilis\n",
      "Coverage Percentage: 0.0451691292635352%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 443\n",
      "Length of fail 3517\n",
      "G900453115 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Moraxella; s__Moraxella osloensis\n",
      "Coverage Percentage: 0.1296295470656685%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 443\n",
      "Length of fail 3518\n",
      "G900453245 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Moraxella; s__Moraxella lacunata\n",
      "Coverage Percentage: 0.0284070890000704%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 443\n",
      "Length of fail 3519\n",
      "G900453285 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Moraxella; s__Moraxella caprae\n",
      "Coverage Percentage: 0.01824762681286%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 443\n",
      "Length of fail 3520\n",
      "G900453335 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Moraxella; s__Moraxella equi\n",
      "Coverage Percentage: 0.0136311865392537%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 443\n",
      "Length of fail 3521\n",
      "G900454195 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Odoribacteraceae; g__Odoribacter; s__Odoribacter splanchnicus\n",
      "Coverage Percentage: 0.0221481421623916%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 443\n",
      "Length of fail 3522\n",
      "G900454225 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Brucellaceae; g__Brucella; s__Brucella intermedia\n",
      "Coverage Percentage: 0.0093487080804564%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 443\n",
      "Length of fail 3523\n",
      "G900454955 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella disiens\n",
      "Coverage Percentage: 0.078552767666016%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 443\n",
      "Length of fail 3524\n",
      "G900455475 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas alcaligenes\n",
      "Coverage Percentage: 0.1057310241815862%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 443\n",
      "Length of fail 3525\n",
      "G900455615 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas oleovorans\n",
      "Coverage Percentage: 0.5041486552272818%\n",
      "Adj, read counts 156\n",
      "Number of samples per group [2, 1, 7, 2, 1, 2, 2, 1, 2, 1, 3, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 4, 1, 2, 1, 1, 4, 1, 1, 2, 1, 2, 2, 2, 1, 1, 2, 1, 2, 5, 1, 1, 3, 2, 2, 2, 3, 2, 2, 2, 1, 1, 2, 1, 1, 2, 2, 1, 2, 1, 1, 1, 2, 2, 1, 5, 1, 2, 2, 1, 1, 1, 3, 1, 2, 1, 4, 3, 1, 1, 1, 3]\n",
      "Number of bins 127\n",
      "total # samples 68\n",
      "Number of total reads 155\n",
      "Number of hits in BIOM tables 23.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.897435897435898 %\n",
      "Percentage of reads from samples who only fall into one bin group 25.806451612903224 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 444\n",
      "Length of fail 3525\n",
      "G900455635 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas oleovorans\n",
      "Coverage Percentage: 0.5662105878391035%\n",
      "Adj, read counts 170\n",
      "Number of samples per group [1, 6, 1, 1, 2, 2, 1, 3, 2, 2, 1, 1, 1, 2, 1, 1, 4, 1, 3, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 4, 3, 3, 1, 1, 4, 1, 1, 1, 4, 2, 1, 1, 2, 7, 2, 5, 2, 3, 1, 1, 3, 2, 1, 1, 3, 2, 1, 4, 1, 2, 3, 3, 1, 2, 1, 2, 1, 5, 1, 2, 1, 1, 1, 1, 2, 1, 1, 3, 5, 1, 2, 1, 1, 1]\n",
      "Number of bins 137\n",
      "total # samples 69\n",
      "Number of total reads 168\n",
      "Number of hits in BIOM tables 36.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.76470588235294 %\n",
      "Percentage of reads from samples who only fall into one bin group 21.428571428571427 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 445\n",
      "Length of fail 3525\n",
      "G900455645 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas putida\n",
      "Coverage Percentage: 0.0312242782275289%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 445\n",
      "Length of fail 3526\n",
      "G900455865 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Roseomonas; s__Roseomonas gilardii\n",
      "Coverage Percentage: 0.010391880477387%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 445\n",
      "Length of fail 3527\n",
      "G900455895 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Rothia; s__Rothia dentocariosa\n",
      "Coverage Percentage: 0.0530045338953955%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 445\n",
      "Length of fail 3528\n",
      "G900455935 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodospirillales; f__Acetobacteraceae; g__Roseomonas; s__Roseomonas mucosa\n",
      "Coverage Percentage: 0.0097787698307552%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 445\n",
      "Length of fail 3529\n",
      "G900456775 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Sarcina; s__Sarcina ventriculi\n",
      "Coverage Percentage: 0.0950626260344531%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 445\n",
      "Length of fail 3530\n",
      "G900456835 d__Bacteria; p__Fusobacteria; c__Fusobacteriia; o__Fusobacteriales; f__Leptotrichiaceae; g__Sebaldella; s__Sebaldella termitidis\n",
      "Coverage Percentage: 0.0127267155155601%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 445\n",
      "Length of fail 3531\n",
      "G900456975 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Alteromonadales; f__Shewanellaceae; g__Shewanella; s__Shewanella baltica\n",
      "Coverage Percentage: 0.0069234022692912%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 445\n",
      "Length of fail 3532\n",
      "G900457075 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia quinivorans\n",
      "Coverage Percentage: 0.0253772504438465%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 445\n",
      "Length of fail 3533\n",
      "G900457095 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Shigella; s__Shigella boydii\n",
      "Coverage Percentage: 0.879536623094797%\n",
      "Adj, read counts 259\n",
      "Number of samples per group [1, 1, 3, 2, 2, 1, 1, 5, 2, 2, 1, 3, 2, 1, 3, 1, 1, 1, 2, 5, 2, 2, 2, 4, 4, 1, 1, 2, 1, 2, 2, 2, 1, 1, 1, 3, 1, 1, 1, 2, 1, 2, 1, 1, 2, 3, 1, 1, 1, 1, 2, 1, 1, 1, 2, 1, 1, 2, 2, 2, 3, 1, 1, 2, 2, 1, 4, 1, 1, 2, 1, 4, 3, 2, 2, 1, 2, 5, 1, 2, 1, 2, 2, 1, 3, 2, 2, 1, 1, 4, 4, 1, 2, 3, 1, 1, 1, 1, 1, 2, 2, 2, 2, 1, 1, 1]\n",
      "Number of bins 175\n",
      "total # samples 57\n",
      "Number of total reads 253\n",
      "Number of hits in BIOM tables 34.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 6.94980694980695 %\n",
      "Percentage of reads from samples who only fall into one bin group 12.648221343873518 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 446\n",
      "Length of fail 3533\n",
      "G900457215 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Shigella; s__Shigella dysenteriae\n",
      "Coverage Percentage: 0.787556369162228%\n",
      "Adj, read counts 216\n",
      "Number of samples per group [2, 2, 1, 1, 1, 1, 1, 1, 2, 1, 3, 1, 1, 2, 1, 1, 1, 1, 3, 1, 1, 3, 6, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 4, 4, 1, 1, 1, 1, 4, 2, 2, 5, 1, 2, 1, 2, 1, 1, 2, 3, 2, 2, 1, 2, 2, 2, 1, 1, 3, 2, 1, 1, 1, 1, 1, 2, 1, 2, 2, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 2, 2, 1, 2, 1, 2, 2, 1, 2, 4, 2, 1, 2, 1, 1, 1, 1, 2, 3]\n",
      "Number of bins 148\n",
      "total # samples 50\n",
      "Number of total reads 213\n",
      "Number of hits in BIOM tables 28.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 8.333333333333332 %\n",
      "Percentage of reads from samples who only fall into one bin group 11.267605633802818 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 447\n",
      "Length of fail 3533\n",
      "G900458255 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus cohnii\n",
      "Coverage Percentage: 0.0204137382231288%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 447\n",
      "Length of fail 3534\n",
      "G900458355 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus devriesei\n",
      "Coverage Percentage: 0.0371757116463636%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 447\n",
      "Length of fail 3535\n",
      "G900458595 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus haemolyticus\n",
      "Coverage Percentage: 0.0422256538944117%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 447\n",
      "Length of fail 3536\n",
      "G900458605 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus epidermidis\n",
      "Coverage Percentage: 0.3513274559432139%\n",
      "Adj, read counts 50\n",
      "Number of samples per group [2, 1, 1, 2, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 4, 2, 2, 1, 1, 1, 1, 1, 2, 3, 1, 2, 1, 1, 1, 1, 1, 2]\n",
      "Number of bins 43\n",
      "total # samples 18\n",
      "Number of total reads 50\n",
      "Number of hits in BIOM tables 19.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 24.000000000000004 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.0 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 448\n",
      "Length of fail 3536\n",
      "G900458735 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Mammaliicoccus; s__Mammaliicoccus lentus\n",
      "Coverage Percentage: 0.0072883767743726%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 448\n",
      "Length of fail 3537\n",
      "G900458885 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus saprophyticus\n",
      "Coverage Percentage: 0.2930567486627079%\n",
      "Adj, read counts 43\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 4, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 38\n",
      "total # samples 22\n",
      "Number of total reads 43\n",
      "Number of hits in BIOM tables 9.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 25.581395348837212 %\n",
      "Percentage of reads from samples who only fall into one bin group 34.883720930232556 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 449\n",
      "Length of fail 3537\n",
      "G900459295 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus equinus\n",
      "Coverage Percentage: 0.0621211206827343%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 449\n",
      "Length of fail 3538\n",
      "G900459405 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus hyointestinalis\n",
      "Coverage Percentage: 0.0060309898890846%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 449\n",
      "Length of fail 3539\n",
      "G900459425 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus mitis\n",
      "Coverage Percentage: 0.0766001858618398%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 449\n",
      "Length of fail 3540\n",
      "G900459445 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus infantarius\n",
      "Coverage Percentage: 0.0547928001652899%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 449\n",
      "Length of fail 3541\n",
      "G900460235 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Veillonellales; f__Veillonellaceae; g__Veillonella; s__Veillonella atypica\n",
      "Coverage Percentage: 0.0284144544661924%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 449\n",
      "Length of fail 3542\n",
      "G900461065 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Clostridium; s__Clostridium fallax\n",
      "Coverage Percentage: 0.0741438748745864%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 449\n",
      "Length of fail 3543\n",
      "G900461185 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium jeikeium\n",
      "Coverage Percentage: 0.0561351392115823%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 449\n",
      "Length of fail 3544\n",
      "G900461225 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Comamonas; s__Comamonas testosteroni\n",
      "Coverage Percentage: 0.0777354121189761%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 449\n",
      "Length of fail 3545\n",
      "G900461335 d__Bacteria; p__Actinobacteria; c__Coriobacteriia; o__Coriobacteriales; f__Coriobacteriaceae; g__Collinsella; s__Collinsella aerofaciens\n",
      "Coverage Percentage: 0.1298864183860719%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 449\n",
      "Length of fail 3546\n",
      "G900461435 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Comamonas; s__Comamonas terrigena\n",
      "Coverage Percentage: 0.1189446366782007%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 449\n",
      "Length of fail 3547\n",
      "G900474605 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Enterococcaceae; g__Enterococcus; s__Enterococcus cecorum\n",
      "Coverage Percentage: 0.0265114193189786%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 449\n",
      "Length of fail 3548\n",
      "G900475215 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas fluorescens\n",
      "Coverage Percentage: 0.0389398835425808%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 449\n",
      "Length of fail 3549\n",
      "G900475285 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas taetrolens\n",
      "Coverage Percentage: 0.049361347566853%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 449\n",
      "Length of fail 3550\n",
      "G900475315 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Neisseriales; f__Neisseriaceae; g__Neisseria; s__Neisseria cinerea\n",
      "Coverage Percentage: 0.0290249789405228%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 449\n",
      "Length of fail 3551\n",
      "G900475405 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Stenotrophomonas; s__Stenotrophomonas maltophilia\n",
      "Coverage Percentage: 0.3319930428076207%\n",
      "Adj, read counts 91\n",
      "Number of samples per group [1, 1, 3, 2, 1, 1, 3, 1, 1, 2, 1, 2, 1, 1, 3, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 1, 1, 2, 2, 2, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 3, 1]\n",
      "Number of bins 73\n",
      "total # samples 26\n",
      "Number of total reads 91\n",
      "Number of hits in BIOM tables 21.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.087912087912086 %\n",
      "Percentage of reads from samples who only fall into one bin group 17.582417582417584 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 450\n",
      "Length of fail 3551\n",
      "G900475555 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Micrococcus; s__Micrococcus luteus\n",
      "Coverage Percentage: 0.767682563520433%\n",
      "Adj, read counts 107\n",
      "Number of samples per group [1, 4, 2, 1, 1, 1, 1, 1, 1, 3, 3, 1, 1, 1, 2, 1, 2, 1, 2, 3, 4, 2, 1, 1, 5, 1, 1, 1, 1, 2, 1, 3, 2, 4, 1, 2, 2, 1, 1, 3, 1, 2, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 75\n",
      "total # samples 23\n",
      "Number of total reads 105\n",
      "Number of hits in BIOM tables 43.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.018691588785046 %\n",
      "Percentage of reads from samples who only fall into one bin group 15.238095238095239 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 451\n",
      "Length of fail 3551\n",
      "G900475675 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus lutetiensis\n",
      "Coverage Percentage: 0.0317253045824386%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 451\n",
      "Length of fail 3552\n",
      "G900475835 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Jonesiaceae; g__Jonesia; s__Jonesia denitrificans\n",
      "Coverage Percentage: 0.0196752528597307%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 451\n",
      "Length of fail 3553\n",
      "G900476005 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas caviae\n",
      "Coverage Percentage: 0.0476871617525849%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 451\n",
      "Length of fail 3554\n",
      "G900476035 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pasteurellales; f__Pasteurellaceae; g__Aggregatibacter; s__Aggregatibacter segnis\n",
      "Coverage Percentage: 0.0138914155995159%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 451\n",
      "Length of fail 3555\n",
      "G900476055 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus australis\n",
      "Coverage Percentage: 0.0291931764766843%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 451\n",
      "Length of fail 3556\n",
      "G900478045 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium minutissimum\n",
      "Coverage Percentage: 0.0543403672889535%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 451\n",
      "Length of fail 3557\n",
      "G900478115 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Nocardiaceae; g__Rhodococcus; s__Rhodococcus coprophilus\n",
      "Coverage Percentage: 0.0162669651345441%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 451\n",
      "Length of fail 3558\n",
      "G900478255 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus lugdunensis\n",
      "Coverage Percentage: 0.0231072068110755%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 451\n",
      "Length of fail 3559\n",
      "G900478415 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus auricularis\n",
      "Coverage Percentage: 0.0753834813183402%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 451\n",
      "Length of fail 3560\n",
      "G900491935 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Klenkia; s__Klenkia terrae\n",
      "Coverage Percentage: 0.0287475690934926%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 451\n",
      "Length of fail 3561\n",
      "G900537995 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Roseburia; s__Roseburia intestinalis\n",
      "Coverage Percentage: 0.1368244792079313%\n",
      "Adj, read counts 38\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 3, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1]\n",
      "Number of bins 34\n",
      "total # samples 15\n",
      "Number of total reads 37\n",
      "Number of hits in BIOM tables 10.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 18.421052631578945 %\n",
      "Percentage of reads from samples who only fall into one bin group 29.72972972972973 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 452\n",
      "Length of fail 3561\n",
      "G900585495 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas viridiflava\n",
      "Coverage Percentage: 0.0284088217259298%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 452\n",
      "Length of fail 3562\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G900603025 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Mycobacteriaceae; g__Mycolicibacterium; s__Mycolicibacterium hassiacum\n",
      "Coverage Percentage: 0.0449223083196228%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 452\n",
      "Length of fail 3563\n",
      "G900604285 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Varibaculum; s__Varibaculum vaginae\n",
      "Coverage Percentage: 0.0141930835019795%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 452\n",
      "Length of fail 3564\n",
      "G900607235 d__Bacteria; p__Bacteroidetes; c__Sphingobacteriia; o__Sphingobacteriales; f__Sphingobacteriaceae; g__Olivibacter; s__Olivibacter jilunii\n",
      "Coverage Percentage: 0.003191907946568%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 452\n",
      "Length of fail 3565\n",
      "G900609065 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Arthrobacter; s__Arthrobacter ulcerisalmonis\n",
      "Coverage Percentage: 0.0171692168449346%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 452\n",
      "Length of fail 3566\n",
      "G900618535 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas bubulae\n",
      "Coverage Percentage: 0.3647807795961364%\n",
      "Adj, read counts 112\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 2, 1, 1, 2, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 2]\n",
      "Number of bins 76\n",
      "total # samples 16\n",
      "Number of total reads 110\n",
      "Number of hits in BIOM tables 174.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 10.714285714285714 %\n",
      "Percentage of reads from samples who only fall into one bin group 6.363636363636363 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3566\n",
      "G900624755 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Rikenellaceae; g__Alistipes; s__Alistipes senegalensis\n",
      "Coverage Percentage: 0.0385902108772687%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3567\n",
      "G900624885 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Tannerellaceae; g__Parabacteroides; s__Parabacteroides johnsonii\n",
      "Coverage Percentage: 0.0422593548008942%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3568\n",
      "G900624995 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas antarctica\n",
      "Coverage Percentage: 0.0317840923033226%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3569\n",
      "G900625045 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas extremaustralis\n",
      "Coverage Percentage: 0.0373100145742244%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3570\n",
      "G900625065 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Prevotella; s__Prevotella marseillensis\n",
      "Coverage Percentage: 0.0230465689690805%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3571\n",
      "G900625075 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas tolaasii\n",
      "Coverage Percentage: 0.0412406001788228%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3572\n",
      "G900626145 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__; g__Intestinimonas; s__Intestinimonas timonensis\n",
      "Coverage Percentage: 0.0166322507760265%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3573\n",
      "G900626155 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Anaerococcus; s__Anaerococcus vaginimassiliensis\n",
      "Coverage Percentage: 0.0342508271384169%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3574\n",
      "G900631945 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rhodobacterales; f__Rhodobacteraceae; g__Paracoccus; s__Paracoccus haematequi\n",
      "Coverage Percentage: 0.0036538471844184%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3575\n",
      "G900634715 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Frankiales; f__Frankiaceae; g__Frankia; s__Candidatus Frankia datiscae\n",
      "Coverage Percentage: 0.0406568710841621%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3576\n",
      "G900634735 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Frankiales; f__Frankiaceae; g__Frankia; s__Candidatus Frankia meridionalis\n",
      "Coverage Percentage: 0.0524975118171072%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3577\n",
      "G900634745 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Frankiales; f__Frankiaceae; g__Frankia; s__Frankia symbiont of Coriaria ruscifolia\n",
      "Coverage Percentage: 0.0161372412533885%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3578\n",
      "G900635455 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Yersinia; s__Yersinia intermedia\n",
      "Coverage Percentage: 0.05856285164512%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3579\n",
      "G900636365 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus viridans\n",
      "Coverage Percentage: 0.0391278786135121%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3580\n",
      "G900636435 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus salivarius\n",
      "Coverage Percentage: 0.0731137955261428%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3581\n",
      "G900636445 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus vestibularis\n",
      "Coverage Percentage: 0.1176739022870781%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3582\n",
      "G900636545 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas mendocina\n",
      "Coverage Percentage: 0.0395159717125725%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3583\n",
      "G900637015 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Escherichia; s__Escherichia marmotae\n",
      "Coverage Percentage: 0.1275407024715392%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3584\n",
      "G900637025 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus oralis\n",
      "Coverage Percentage: 0.0670446709064439%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3585\n",
      "G900637145 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium breve\n",
      "Coverage Percentage: 0.0818222725323246%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3586\n",
      "G900637215 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium longum\n",
      "Coverage Percentage: 0.2515578512455044%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3587\n",
      "G900637295 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces slackii\n",
      "Coverage Percentage: 0.0095947595845127%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 453\n",
      "Length of fail 3588\n",
      "G900637335 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Bifidobacteriales; f__Bifidobacteriaceae; g__Bifidobacterium; s__Bifidobacterium longum\n",
      "Coverage Percentage: 0.3944389474919921%\n",
      "Adj, read counts 54\n",
      "Number of samples per group [4, 2, 1, 1, 1, 1, 1, 4, 1, 1, 1, 2, 2, 1, 1, 2, 1, 2, 1, 1, 1, 2, 1, 1, 2, 1, 3, 1, 1, 1, 1, 2, 1, 1]\n",
      "Number of bins 43\n",
      "total # samples 23\n",
      "Number of total reads 54\n",
      "Number of hits in BIOM tables 7.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 25.925925925925924 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.666666666666664 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 454\n",
      "Length of fail 3588\n",
      "G900637545 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Aeromonadales; f__Aeromonadaceae; g__Aeromonas; s__Aeromonas encheleia\n",
      "Coverage Percentage: 0.0249641113381754%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 454\n",
      "Length of fail 3589\n",
      "G900637555 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Lautropia; s__Lautropia mirabilis\n",
      "Coverage Percentage: 0.0257250134772588%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 454\n",
      "Length of fail 3590\n",
      "G900637825 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Corynebacteriales; f__Corynebacteriaceae; g__Corynebacterium; s__Corynebacterium segmentosum\n",
      "Coverage Percentage: 0.2809002966158824%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [4, 1, 1, 1, 1, 1, 1, 1, 1, 3, 2, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 32\n",
      "total # samples 19\n",
      "Number of total reads 37\n",
      "Number of hits in BIOM tables 49.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 24.324324324324326 %\n",
      "Percentage of reads from samples who only fall into one bin group 24.324324324324326 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 455\n",
      "Length of fail 3590\n",
      "G900637975 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces viscosus\n",
      "Coverage Percentage: 0.0289295411001417%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 455\n",
      "Length of fail 3591\n",
      "G900637985 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Micrococcaceae; g__Rothia; s__Rothia aeria\n",
      "Coverage Percentage: 0.0179111268351519%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 455\n",
      "Length of fail 3592\n",
      "G900638135 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Yersiniaceae; g__Serratia; s__Serratia quinivorans\n",
      "Coverage Percentage: 0.0289580459847074%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 455\n",
      "Length of fail 3593\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G900638215 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptosporangiales; f__Nocardiopsaceae; g__Nocardiopsis; s__Nocardiopsis dassonvillei\n",
      "Coverage Percentage: 0.0117255535584914%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 455\n",
      "Length of fail 3594\n",
      "G900638305 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Porphyromonadaceae; g__Porphyromonas; s__Porphyromonas cangingivalis\n",
      "Coverage Percentage: 0.0229951302717791%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 455\n",
      "Length of fail 3595\n",
      "G900639935 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Legionellales; f__Legionellaceae; g__Legionella; s__Legionella nautarum\n",
      "Coverage Percentage: 0.007343033078172%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 455\n",
      "Length of fail 3596\n",
      "G900639975 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Legionellales; f__Legionellaceae; g__Legionella; s__Legionella santicrucis\n",
      "Coverage Percentage: 0.0106533428858307%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 455\n",
      "Length of fail 3597\n",
      "G900683745 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Paraprevotella; s__Paraprevotella xylaniphila\n",
      "Coverage Percentage: 0.0235617971154736%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 455\n",
      "Length of fail 3598\n",
      "G900696085 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Bradyrhizobiaceae; g__Bradyrhizobium; s__Bradyrhizobium ivorense\n",
      "Coverage Percentage: 0.0082225292627493%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 455\n",
      "Length of fail 3599\n",
      "G900699155 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Acinetobacter; s__Acinetobacter lwoffii\n",
      "Coverage Percentage: 0.6512286523443087%\n",
      "Adj, read counts 148\n",
      "Number of samples per group [8, 2, 11, 5, 2, 5, 5, 3, 1, 1, 2, 1, 2, 1, 5, 1, 1, 4, 1, 1, 1, 2, 1, 1, 1, 2, 4, 3, 3, 5, 1, 2, 1, 1, 2, 1, 2, 3, 1, 1, 1, 3, 1, 1, 1, 1, 2, 3, 1, 3, 1, 1, 1, 6, 1, 2]\n",
      "Number of bins 94\n",
      "total # samples 36\n",
      "Number of total reads 145\n",
      "Number of hits in BIOM tables 32.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 22.2972972972973 %\n",
      "Percentage of reads from samples who only fall into one bin group 6.896551724137931 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 456\n",
      "Length of fail 3599\n",
      "G900706715 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Budviciaceae; g__Budvicia; s__Budvicia aquatica\n",
      "Coverage Percentage: 0.0033931037276873%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 456\n",
      "Length of fail 3600\n",
      "G901421975 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas vancanneytii\n",
      "Coverage Percentage: 0.0693986934247693%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 456\n",
      "Length of fail 3601\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G901482615 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas synxantha\n",
      "Coverage Percentage: 0.0474145617076257%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 456\n",
      "Length of fail 3602\n",
      "G901482635 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus capitis\n",
      "Coverage Percentage: 0.1472399056763217%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 456\n",
      "Length of fail 3603\n",
      "G901563875 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Klebsiella; s__Klebsiella spallanzanii\n",
      "Coverage Percentage: 0.0097327658211838%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 456\n",
      "Length of fail 3604\n",
      "G902141855 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Hyphomicrobiales; f__Methylobacteriaceae; g__Methylobacterium; s__Methylobacterium dankookense\n",
      "Coverage Percentage: 0.0244794301266688%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 456\n",
      "Length of fail 3605\n",
      "G902158555 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Enterobacterales; f__Enterobacteriaceae; g__Klebsiella; s__Klebsiella spallanzanii\n",
      "Coverage Percentage: 0.004590047355066%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 456\n",
      "Length of fail 3606\n",
      "G902329575 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas carnis\n",
      "Coverage Percentage: 0.1224127496469638%\n",
      "Adj, read counts 45\n",
      "Number of samples per group [1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 42\n",
      "total # samples 22\n",
      "Number of total reads 45\n",
      "Number of hits in BIOM tables 12.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.555555555555555 %\n",
      "Percentage of reads from samples who only fall into one bin group 28.888888888888886 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 457\n",
      "Length of fail 3606\n",
      "G902363055 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__; g__Gemmiger; s__Gemmiger formicilis\n",
      "Coverage Percentage: 0.1842184226388841%\n",
      "Adj, read counts 35\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1]\n",
      "Number of bins 30\n",
      "total # samples 15\n",
      "Number of total reads 35\n",
      "Number of hits in BIOM tables 22.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 20.0 %\n",
      "Percentage of reads from samples who only fall into one bin group 28.57142857142857 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 458\n",
      "Length of fail 3606\n",
      "G902364025 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Enterocloster; s__Enterocloster lavalensis\n",
      "Coverage Percentage: 0.0085181809347734%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 458\n",
      "Length of fail 3607\n",
      "G902374375 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola plebeius\n",
      "Coverage Percentage: 0.0729487296609625%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 458\n",
      "Length of fail 3608\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G902374385 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Acidaminococcales; f__Acidaminococcaceae; g__Phascolarctobacterium; s__Phascolarctobacterium succinatutens\n",
      "Coverage Percentage: 0.0484025738115708%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 458\n",
      "Length of fail 3609\n",
      "G902374465 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Ralstonia; s__Ralstonia pickettii\n",
      "Coverage Percentage: 0.0531104028455453%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 458\n",
      "Length of fail 3610\n",
      "G902375225 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Bacteroides; s__Bacteroides neonati\n",
      "Coverage Percentage: 0.022564137665913%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 458\n",
      "Length of fail 3611\n",
      "G902375265 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Geodermatophilales; f__Geodermatophilaceae; g__Blastococcus; s__Candidatus Blastococcus massiliensis\n",
      "Coverage Percentage: 0.0188399737788858%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 458\n",
      "Length of fail 3612\n",
      "G902375405 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Actinomycetales; f__Actinomycetaceae; g__Actinomyces; s__Actinomyces polynesiensis\n",
      "Coverage Percentage: 0.0052346782527545%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 458\n",
      "Length of fail 3613\n",
      "G902375965 d__Bacteria; p__Firmicutes; c__Tissierellia; o__Tissierellales; f__Peptoniphilaceae; g__Anaerococcus; s__Anaerococcus rubeinfantis\n",
      "Coverage Percentage: 0.033951844362672%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 458\n",
      "Length of fail 3614\n",
      "G902376185 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Clostridiaceae; g__Clostridium; s__Clostridium mediterraneense\n",
      "Coverage Percentage: 0.0391773025908261%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 458\n",
      "Length of fail 3615\n",
      "G902387545 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Bacteroidaceae; g__Phocaeicola; s__Phocaeicola dorei\n",
      "Coverage Percentage: 0.4050937829665566%\n",
      "Adj, read counts 136\n",
      "Number of samples per group [1, 1, 1, 3, 4, 1, 1, 3, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 2, 1, 3, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 3, 1, 2, 2, 1, 1, 1, 2, 2, 1, 1, 4, 1, 1, 2, 1, 1, 3, 1, 2, 1, 1, 1, 2]\n",
      "Number of bins 114\n",
      "total # samples 44\n",
      "Number of total reads 135\n",
      "Number of hits in BIOM tables 45.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 8.823529411764707 %\n",
      "Percentage of reads from samples who only fall into one bin group 14.814814814814813 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 459\n",
      "Length of fail 3615\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G902387955 d__Bacteria; p__Firmicutes; c__Clostridia; o__Eubacteriales; f__Lachnospiraceae; g__Roseburia; s__Roseburia hominis\n",
      "Coverage Percentage: 0.0660054981382886%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 459\n",
      "Length of fail 3616\n",
      "G902388125 d__Bacteria; p__Bacteroidetes; c__Bacteroidia; o__Bacteroidales; f__Prevotellaceae; g__Paraprevotella; s__Paraprevotella clara\n",
      "Coverage Percentage: 0.0398141528810957%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 459\n",
      "Length of fail 3617\n",
      "G902459475 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Legionellales; f__Coxiellaceae; g__Aquicella; s__Aquicella lusitana\n",
      "Coverage Percentage: 0.0174478051575407%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 459\n",
      "Length of fail 3618\n",
      "G902459485 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Legionellales; f__Coxiellaceae; g__Aquicella; s__Aquicella siphonis\n",
      "Coverage Percentage: 0.0161654209645038%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 459\n",
      "Length of fail 3619\n",
      "G902459665 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Pandoraea; s__Pandoraea cepalis\n",
      "Coverage Percentage: 0.0351826397936507%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 459\n",
      "Length of fail 3620\n",
      "G902498895 d__Eukaryota; p__Ascomycota; c__Saccharomycetes; o__Saccharomycetales; f__Dipodascaceae; g__Saprochaete; s__Saprochaete ingens\n",
      "Coverage Percentage: 0.0112132024320945%\n",
      "Adj, read counts 67\n",
      "Number of samples per group [25, 9, 6, 3, 2, 1, 1, 2, 3, 2, 7, 1, 2, 2]\n",
      "Number of bins 16\n",
      "total # samples 42\n",
      "Number of total reads 66\n",
      "Number of hits in BIOM tables 20.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 61.19402985074627 %\n",
      "Percentage of reads from samples who only fall into one bin group 34.84848484848485 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 460\n",
      "Length of fail 3620\n",
      "G902499175 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Burkholderia; s__Burkholderia aenigmatica\n",
      "Coverage Percentage: 0.0332453905305353%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 460\n",
      "Length of fail 3621\n",
      "G902703175 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Ruaniaceae; g__Occultella; s__Occultella aeris\n",
      "Coverage Percentage: 0.002250307883033%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 460\n",
      "Length of fail 3622\n",
      "G902804475 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium bizetiae\n",
      "Coverage Percentage: 0.0599874160189037%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 460\n",
      "Length of fail 3623\n",
      "G902804485 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium collinsii\n",
      "Coverage Percentage: 0.04517726982242%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 460\n",
      "Length of fail 3624\n",
      "G902810435 d__Bacteria; p__Firmicutes; c__Negativicutes; o__Veillonellales; f__Veillonellaceae; g__Veillonella; s__Veillonella parvula\n",
      "Coverage Percentage: 0.0617755005632472%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 460\n",
      "Length of fail 3625\n",
      "G902812705 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Frankiales; f__Frankiaceae; g__Frankia; s__Candidatus Frankia nodulisporulans\n",
      "Coverage Percentage: 0.0784588605976145%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 460\n",
      "Length of fail 3626\n",
      "G902813185 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Sterolibacteriaceae; g__Denitratisoma; s__Denitratisoma oestradiolicum\n",
      "Coverage Percentage: 0.0149530725070293%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 460\n",
      "Length of fail 3627\n",
      "G902825375 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Streptosporangiales; f__Nocardiopsaceae; g__Nocardiopsis; s__Nocardiopsis dassonvillei\n",
      "Coverage Percentage: 0.0077670345462102%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 460\n",
      "Length of fail 3628\n",
      "G902825405 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micromonosporales; f__Micromonosporaceae; g__Micromonospora; s__Micromonospora maritima\n",
      "Coverage Percentage: 0.0064603480944291%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 460\n",
      "Length of fail 3629\n",
      "G902829245 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Comamonas; s__Comamonas jiangduensis\n",
      "Coverage Percentage: 0.0453163573605265%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 460\n",
      "Length of fail 3630\n",
      "G902833635 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia caledonica\n",
      "Coverage Percentage: 0.0020666218670491%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 460\n",
      "Length of fail 3631\n",
      "G902833655 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia phenoliruptrix\n",
      "Coverage Percentage: 0.0185042912170868%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 460\n",
      "Length of fail 3632\n",
      "G902833665 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia phymatum\n",
      "Coverage Percentage: 0.0028722272831823%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 460\n",
      "Length of fail 3633\n",
      "G902858935 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus salivarius\n",
      "Coverage Percentage: 0.0497008111849707%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 460\n",
      "Length of fail 3634\n",
      "G902859635 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Achromobacter; s__Achromobacter marplatensis\n",
      "Coverage Percentage: 0.0098590761943965%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 460\n",
      "Length of fail 3635\n",
      "G902859645 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Achromobacter; s__Achromobacter insuavis\n",
      "Coverage Percentage: 0.0525635498151705%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 460\n",
      "Length of fail 3636\n",
      "G902859695 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Achromobacter; s__Achromobacter ruhlandii\n",
      "Coverage Percentage: 0.0758789321780705%\n",
      "Adj, read counts 31\n",
      "Number of samples per group [1, 1, 1, 3, 1, 6, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 2, 1, 1]\n",
      "Number of bins 26\n",
      "total # samples 17\n",
      "Number of total reads 30\n",
      "Number of hits in BIOM tables 5.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 38.70967741935484 %\n",
      "Percentage of reads from samples who only fall into one bin group 46.666666666666664 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 461\n",
      "Length of fail 3636\n",
      "G902859705 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Achromobacter; s__Achromobacter deleyi\n",
      "Coverage Percentage: 0.0224786474872366%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 461\n",
      "Length of fail 3637\n",
      "G902859745 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Achromobacter; s__Achromobacter dolens\n",
      "Coverage Percentage: 0.053904142135357%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 461\n",
      "Length of fail 3638\n",
      "G902859765 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Achromobacter; s__Achromobacter pulmonis\n",
      "Coverage Percentage: 0.0601032909679431%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 461\n",
      "Length of fail 3639\n",
      "G902859895 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Pararobbsia; s__Pararobbsia alpina\n",
      "Coverage Percentage: 0.0014608154152397%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 461\n",
      "Length of fail 3640\n",
      "G902860065 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Achromobacter; s__Achromobacter pulmonis\n",
      "Coverage Percentage: 0.0392632397137081%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 461\n",
      "Length of fail 3641\n",
      "G902860075 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Achromobacter; s__Achromobacter insuavis\n",
      "Coverage Percentage: 0.059740830349563%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 461\n",
      "Length of fail 3642\n",
      "G902860085 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Achromobacter; s__Achromobacter ruhlandii\n",
      "Coverage Percentage: 0.0733520752505266%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 461\n",
      "Length of fail 3643\n",
      "G902860115 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Achromobacter; s__Achromobacter dolens\n",
      "Coverage Percentage: 0.0658213587091936%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 461\n",
      "Length of fail 3644\n",
      "G903652925 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Alcaligenaceae; g__Achromobacter; s__Achromobacter anxifer\n",
      "Coverage Percentage: 0.0270602491178733%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 461\n",
      "Length of fail 3645\n",
      "G903797705 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Comamonadaceae; g__Acidovorax; s__Acidovorax facilis\n",
      "Coverage Percentage: 0.1658131556205824%\n",
      "Adj, read counts 62\n",
      "Number of samples per group [1, 2, 1, 1, 2, 2, 1, 4, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 14]\n",
      "Number of bins 42\n",
      "total # samples 27\n",
      "Number of total reads 60\n",
      "Number of hits in BIOM tables 32.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 41.93548387096774 %\n",
      "Percentage of reads from samples who only fall into one bin group 21.666666666666668 %\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 462\n",
      "Length of fail 3645\n",
      "G903819335 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium panici\n",
      "Coverage Percentage: 0.0222402490907898%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 462\n",
      "Length of fail 3646\n",
      "G903819435 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium salmonis\n",
      "Coverage Percentage: 0.1675087274314078%\n",
      "Adj, read counts 58\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 2, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 52\n",
      "total # samples 27\n",
      "Number of total reads 57\n",
      "Number of hits in BIOM tables 65.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 12.068965517241379 %\n",
      "Percentage of reads from samples who only fall into one bin group 22.807017543859647 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 463\n",
      "Length of fail 3646\n",
      "G903886475 d__Bacteria; p__Firmicutes; c__Bacilli; o__Lactobacillales; f__Streptococcaceae; g__Streptococcus; s__Streptococcus thermophilus\n",
      "Coverage Percentage: 0.0912564591154318%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 463\n",
      "Length of fail 3647\n",
      "G903994055 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Jeotgalicoccus; s__Jeotgalicoccus aerolatus\n",
      "Coverage Percentage: 0.0143655607087644%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 463\n",
      "Length of fail 3648\n",
      "G904061905 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas carbonaria\n",
      "Coverage Percentage: 0.0245092473750776%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 463\n",
      "Length of fail 3649\n",
      "G904063055 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas paracarnis\n",
      "Coverage Percentage: 0.1426273383297445%\n",
      "Adj, read counts 51\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 3, 1]\n",
      "Number of bins 46\n",
      "total # samples 26\n",
      "Number of total reads 51\n",
      "Number of hits in BIOM tables 15.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.686274509803921 %\n",
      "Percentage of reads from samples who only fall into one bin group 29.411764705882355 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 464\n",
      "Length of fail 3649\n",
      "G904063065 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas paraversuta\n",
      "Coverage Percentage: 0.0559868237148356%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 464\n",
      "Length of fail 3650\n",
      "G904063075 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas paracarnis\n",
      "Coverage Percentage: 0.1517891901938796%\n",
      "Adj, read counts 54\n",
      "Number of samples per group [1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 3, 1, 1]\n",
      "Number of bins 46\n",
      "total # samples 25\n",
      "Number of total reads 52\n",
      "Number of hits in BIOM tables 14.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 14.814814814814813 %\n",
      "Percentage of reads from samples who only fall into one bin group 28.846153846153843 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 465\n",
      "Length of fail 3650\n",
      "G904063085 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Pseudomonadales; f__Pseudomonadaceae; g__Pseudomonas; s__Pseudomonas paraversuta\n",
      "Coverage Percentage: 0.0503255910216274%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 465\n",
      "Length of fail 3651\n",
      "G904425415 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Flavobacteriaceae; g__Flavobacterium; s__Flavobacterium bizetiae\n",
      "Coverage Percentage: 0.0520893178597418%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 465\n",
      "Length of fail 3652\n",
      "G904830995 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Caulobacterales; f__Caulobacteraceae; g__Brevundimonas; s__Brevundimonas aurantiaca\n",
      "Coverage Percentage: 0.031699977526817%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 465\n",
      "Length of fail 3653\n",
      "G904845885 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter aquimaris\n",
      "Coverage Percentage: 0.0257388068871934%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 465\n",
      "Length of fail 3654\n",
      "G904846155 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter fulvigenes\n",
      "Coverage Percentage: 0.2609747153153928%\n",
      "Adj, read counts 63\n",
      "Number of samples per group [2, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 3, 1, 3, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 4, 1, 1, 2, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 51\n",
      "total # samples 25\n",
      "Number of total reads 63\n",
      "Number of hits in BIOM tables 56.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 15.873015873015872 %\n",
      "Percentage of reads from samples who only fall into one bin group 26.984126984126984 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 466\n",
      "Length of fail 3654\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n",
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G904846325 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter marincola\n",
      "Coverage Percentage: 0.017156901992913%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 466\n",
      "Length of fail 3655\n",
      "G904846345 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter maritimus\n",
      "Coverage Percentage: 0.1887531338182907%\n",
      "Adj, read counts 37\n",
      "Number of samples per group [1, 1, 1, 1, 1, 2, 1, 2, 1, 1, 1, 1, 2, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 2, 1]\n",
      "Number of bins 31\n",
      "total # samples 16\n",
      "Number of total reads 37\n",
      "Number of hits in BIOM tables 37.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 16.216216216216218 %\n",
      "Percentage of reads from samples who only fall into one bin group 18.91891891891892 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 467\n",
      "Length of fail 3655\n",
      "G904846355 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter namhaensis\n",
      "Coverage Percentage: 0.030232945910286%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 467\n",
      "Length of fail 3656\n",
      "G904846445 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter salsus\n",
      "Coverage Percentage: 0.0137435067973381%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 467\n",
      "Length of fail 3657\n",
      "G904846465 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter pulmonis\n",
      "Coverage Percentage: 0.4946103427809227%\n",
      "Adj, read counts 92\n",
      "Number of samples per group [2, 1, 2, 1, 1, 1, 2, 1, 2, 1, 2, 1, 2, 2, 1, 1, 1, 1, 2, 3, 3, 1, 3, 1, 2, 1, 2, 1, 3, 1, 1, 1, 3, 1, 1, 1, 1, 1, 3, 1, 1, 2, 1, 3, 4, 2, 1, 1, 2, 1, 2, 1, 1, 1]\n",
      "Number of bins 77\n",
      "total # samples 30\n",
      "Number of total reads 90\n",
      "Number of hits in BIOM tables 99.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 11.956521739130435 %\n",
      "Percentage of reads from samples who only fall into one bin group 16.666666666666664 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 468\n",
      "Length of fail 3657\n",
      "G904846495 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter sanguinis\n",
      "Coverage Percentage: 0.0539619399396346%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 468\n",
      "Length of fail 3658\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "G904846715 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Moraxellales; f__Moraxellaceae; g__Psychrobacter; s__Psychrobacter vallis\n",
      "Coverage Percentage: 0.0220241837946055%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 468\n",
      "Length of fail 3659\n",
      "G904848625 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia metrosideri\n",
      "Coverage Percentage: 0.0019251103826208%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 468\n",
      "Length of fail 3660\n",
      "G905142465 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Xanthomonas; s__Xanthomonas hydrangea\n",
      "Coverage Percentage: 0.0233870354280554%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 468\n",
      "Length of fail 3661\n",
      "G905220975 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Paraburkholderia; s__Paraburkholderia haematera\n",
      "Coverage Percentage: 0.0064965649412872%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 468\n",
      "Length of fail 3662\n",
      "G905220995 d__Bacteria; p__Nitrospirae; c__Nitrospira; o__Nitrospirales; f__Nitrospiraceae; g__Nitrospira; s__Nitrospira defluvii\n",
      "Coverage Percentage: 0.0022724333588259%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 468\n",
      "Length of fail 3663\n",
      "G905367715 d__Bacteria; p__Proteobacteria; c__Gammaproteobacteria; o__Xanthomonadales; f__Xanthomonadaceae; g__Xanthomonas; s__Xanthomonas arboricola\n",
      "Coverage Percentage: 0.0060425825998031%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 468\n",
      "Length of fail 3664\n",
      "G905397275 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Cupriavidus; s__Cupriavidus yeoncheonensis\n",
      "Coverage Percentage: 0.0021534183662628%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 468\n",
      "Length of fail 3665\n",
      "G905397375 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Ralstonia; s__Ralstonia mannitolilytica\n",
      "Coverage Percentage: 0.0729635600910883%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 468\n",
      "Length of fail 3666\n",
      "G905397435 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Burkholderiales; f__Burkholderiaceae; g__Cupriavidus; s__Cupriavidus numazuensis\n",
      "Coverage Percentage: 0.0045326821306303%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 468\n",
      "Length of fail 3667\n",
      "G907163105 d__Bacteria; p__Bacteroidetes; c__Flavobacteriia; o__Flavobacteriales; f__Weeksellaceae; g__Cloacibacterium; s__Cloacibacterium caeni\n",
      "Coverage Percentage: 0.0517853248186686%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 468\n",
      "Length of fail 3668\n",
      "G907163265 d__Bacteria; p__Proteobacteria; c__Betaproteobacteria; o__Nitrosomonadales; f__Sterolibacteriaceae; g__Georgfuchsia; s__Georgfuchsia toluolica\n",
      "Coverage Percentage: 0.0113077756662949%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 468\n",
      "Length of fail 3669\n",
      "G907166805 d__Eukaryota; p__Ascomycota; c__Dothideomycetes; o__Pleosporales; f__Pleosporaceae; g__Alternaria; s__Alternaria atra\n",
      "Coverage Percentage: 0.0755899902381128%\n",
      "Adj, read counts 165\n",
      "Number of samples per group [2, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1]\n",
      "Number of bins 133\n",
      "total # samples 8\n",
      "Number of total reads 162\n",
      "Number of hits in BIOM tables 406.0\n",
      "\n",
      "Percentage of reads in the top 3 bin groups with the largest samples 4.848484848484849 %\n",
      "Percentage of reads from samples who only fall into one bin group 3.7037037037037033 %\n",
      "Status: Pass\n",
      "----------------------------\n",
      "Length of pass 469\n",
      "Length of fail 3669\n",
      "G910573815 d__Bacteria; p__Actinobacteria; c__Actinomycetia; o__Micrococcales; f__Dermacoccaceae; g__Barrientosiimonas; s__Barrientosiimonas humi\n",
      "Coverage Percentage: 0.0322131155686235%\n",
      "Status: Fail -- less than 30 reads\n",
      "----------------------------\n",
      "Length of pass 469\n",
      "Length of fail 3670\n",
      "G910576415 d__Bacteria; p__Firmicutes; c__Bacilli; o__Bacillales; f__Staphylococcaceae; g__Staphylococcus; s__Staphylococcus pasteuri\n",
      "Coverage Percentage: 0.0205333100331233%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 469\n",
      "Length of fail 3671\n",
      "G913698045 d__Bacteria; p__Proteobacteria; c__Alphaproteobacteria; o__Rickettsiales; f__Rickettsiaceae; g__Candidatus Megaira; s__Candidatus Megaira endosymbiont of Carteria cerasiformis\n",
      "Coverage Percentage: 0.0114604188667722%\n",
      "Status: Fail -- less than 5 BIOM hits\n",
      "----------------------------\n",
      "Length of pass 469\n",
      "Length of fail 3672\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cguccion/Dropbox/current_Fall23/HCC_Amir_v1/HCC-microbialDNA/python_scripts/micov_filter.py:68: UserWarning: Matplotlib is currently using agg, which is a non-GUI backend, so cannot show the figure.\n",
      "  plt.show()\n"
     ]
    }
   ],
   "source": [
    "pass_, fail_ = loop_micov(gen_len, pos, micovRef_df, cov, taxa, plot=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "da0a830f-4045-454b-833d-28cdbd1759b0",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "469\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "3672"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "print(len(pass_))\n",
    "len(fail_)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "fcaacffe-fe8c-4eae-9935-ef972b36c26a",
   "metadata": {},
   "outputs": [],
   "source": [
    "save_pass_fail(pass_, fail_)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "aa9abdfd-7928-4686-9c01-dc51286308d0",
   "metadata": {},
   "source": [
    "### Combine Tech Reps + Micov"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "b5a5bff4-5cf4-41af-84f1-affc2578abbb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>15336.CRC.9.Background</th>\n",
       "      <th>15336.CRC.1.tumor</th>\n",
       "      <th>15336.CRC.10.tumor</th>\n",
       "      <th>15336.HCC.3.Plasma</th>\n",
       "      <th>15336.HCC.2.tumor</th>\n",
       "      <th>15336.HCC.1.Plasma</th>\n",
       "      <th>15336.HCC.12.Background</th>\n",
       "      <th>15336.HCC.14.tumor</th>\n",
       "      <th>15336.HCC.5.tumor</th>\n",
       "      <th>15336.CRC.5.tumor</th>\n",
       "      <th>...</th>\n",
       "      <th>15336.CRC.11.Background</th>\n",
       "      <th>15336.CRC.3.Plasma</th>\n",
       "      <th>15336.HCC.9.Background</th>\n",
       "      <th>15336.HCC.8.tumor</th>\n",
       "      <th>15336.CRC.9.Plasma</th>\n",
       "      <th>15336.HCC.5.Plasma</th>\n",
       "      <th>15336.HCC.16.tumor</th>\n",
       "      <th>15336.CRC.6.tumor</th>\n",
       "      <th>15336.HCC.10.Plasma</th>\n",
       "      <th>15336.CRC.5.Background</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>G000005845</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000006765</th>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000006925</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000008865</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>17.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G000010125</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G904063075</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G904846155</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G904846345</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G904846465</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>10.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>G907166805</th>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>469 rows × 74 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            15336.CRC.9.Background  15336.CRC.1.tumor  15336.CRC.10.tumor  \\\n",
       "G000005845                     0.0                1.0                 0.0   \n",
       "G000006765                     0.0                3.0                 0.0   \n",
       "G000006925                     0.0                1.0                 0.0   \n",
       "G000008865                     0.0                1.0                 0.0   \n",
       "G000010125                     0.0                0.0                 0.0   \n",
       "...                            ...                ...                 ...   \n",
       "G904063075                     0.0                0.0                 0.0   \n",
       "G904846155                     0.0                0.0                 0.0   \n",
       "G904846345                     0.0                0.0                 0.0   \n",
       "G904846465                     0.0                0.0                 0.0   \n",
       "G907166805                     0.0                0.0                 0.0   \n",
       "\n",
       "            15336.HCC.3.Plasma  15336.HCC.2.tumor  15336.HCC.1.Plasma  \\\n",
       "G000005845                 0.0                0.0                 0.0   \n",
       "G000006765                 1.0                0.0                 1.0   \n",
       "G000006925                 0.0                0.0                 0.0   \n",
       "G000008865                 0.0                0.0                 0.0   \n",
       "G000010125                 0.0                0.0                 0.0   \n",
       "...                        ...                ...                 ...   \n",
       "G904063075                 0.0                0.0                 0.0   \n",
       "G904846155                 0.0                0.0                 4.0   \n",
       "G904846345                 0.0                0.0                 0.0   \n",
       "G904846465                 5.0                0.0                 1.0   \n",
       "G907166805                 0.0                0.0                 0.0   \n",
       "\n",
       "            15336.HCC.12.Background  15336.HCC.14.tumor  15336.HCC.5.tumor  \\\n",
       "G000005845                      0.0                20.0                0.0   \n",
       "G000006765                      0.0                 1.0                0.0   \n",
       "G000006925                      0.0                15.0                0.0   \n",
       "G000008865                      0.0                17.0                0.0   \n",
       "G000010125                      0.0                 0.0                0.0   \n",
       "...                             ...                 ...                ...   \n",
       "G904063075                      0.0                 0.0                0.0   \n",
       "G904846155                      0.0                 0.0                0.0   \n",
       "G904846345                      0.0                 0.0                0.0   \n",
       "G904846465                      0.0                 0.0                0.0   \n",
       "G907166805                      0.0                 0.0                0.0   \n",
       "\n",
       "            15336.CRC.5.tumor  ...  15336.CRC.11.Background  \\\n",
       "G000005845                2.0  ...                      0.0   \n",
       "G000006765               10.0  ...                      0.0   \n",
       "G000006925                1.0  ...                      0.0   \n",
       "G000008865                1.0  ...                      0.0   \n",
       "G000010125                0.0  ...                      0.0   \n",
       "...                       ...  ...                      ...   \n",
       "G904063075                0.0  ...                      0.0   \n",
       "G904846155                0.0  ...                      0.0   \n",
       "G904846345                0.0  ...                      0.0   \n",
       "G904846465                0.0  ...                      0.0   \n",
       "G907166805                0.0  ...                      0.0   \n",
       "\n",
       "            15336.CRC.3.Plasma  15336.HCC.9.Background  15336.HCC.8.tumor  \\\n",
       "G000005845                 0.0                     0.0                0.0   \n",
       "G000006765                 0.0                     0.0                0.0   \n",
       "G000006925                 0.0                     0.0                0.0   \n",
       "G000008865                 1.0                     0.0                0.0   \n",
       "G000010125                 0.0                     0.0                0.0   \n",
       "...                        ...                     ...                ...   \n",
       "G904063075                 0.0                     0.0                0.0   \n",
       "G904846155                 2.0                     0.0                0.0   \n",
       "G904846345                 0.0                     0.0                0.0   \n",
       "G904846465                 3.0                     0.0                0.0   \n",
       "G907166805                 0.0                     0.0                0.0   \n",
       "\n",
       "            15336.CRC.9.Plasma  15336.HCC.5.Plasma  15336.HCC.16.tumor  \\\n",
       "G000005845                 0.0                 0.0                 0.0   \n",
       "G000006765                 0.0                 2.0                 0.0   \n",
       "G000006925                 0.0                 0.0                 0.0   \n",
       "G000008865                 0.0                 0.0                 0.0   \n",
       "G000010125                 0.0                 0.0                 0.0   \n",
       "...                        ...                 ...                 ...   \n",
       "G904063075                 0.0                 1.0                 0.0   \n",
       "G904846155                 2.0                 2.0                 0.0   \n",
       "G904846345                 0.0                 3.0                 0.0   \n",
       "G904846465                 3.0                10.0                 0.0   \n",
       "G907166805                 0.0                 0.0                 0.0   \n",
       "\n",
       "            15336.CRC.6.tumor  15336.HCC.10.Plasma  15336.CRC.5.Background  \n",
       "G000005845                0.0                  0.0                     0.0  \n",
       "G000006765                0.0                  3.0                     0.0  \n",
       "G000006925                0.0                  0.0                     0.0  \n",
       "G000008865                0.0                  0.0                     0.0  \n",
       "G000010125                0.0                  1.0                     0.0  \n",
       "...                       ...                  ...                     ...  \n",
       "G904063075                0.0                  2.0                     0.0  \n",
       "G904846155                0.0                  4.0                     0.0  \n",
       "G904846345                0.0                  3.0                     0.0  \n",
       "G904846465                0.0                  3.0                     0.0  \n",
       "G907166805                0.0                  0.0                     0.0  \n",
       "\n",
       "[469 rows x 74 columns]"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Input scrubed df, output with combined tech reps + micov\n",
    "combo_tech_rep(df_rs210c, 'pese_pangenome_align-RS210_masked_none_scrubbed_micovFilter', micov_filter=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e97a917c-fc00-48ad-9200-e2d7d453b471",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "comad_EAC_figures",
   "language": "python",
   "name": "comad_eac_figures"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
