{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "e6ccb780-136b-44a0-9c30-a70ac3f07579",
   "metadata": {},
   "source": [
    "\n",
    "## Step 3: Split Data into Disease Groups\n",
    "**Goal: To create dataframe of each disease groups for future analysis**\n",
    "\n",
    "https: comad"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8bf8de15-87f4-4efa-9e07-52122ddb6d9e",
   "metadata": {},
   "source": [
    "### Imports"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "76ea8838-2f27-4e0e-85c8-dd98d55f1ab9",
   "metadata": {},
   "outputs": [],
   "source": [
    "from biom import load_table\n",
    "import pandas as pd\n",
    "from qiime2 import Artifact\n",
    "from qiime2.plugins import taxa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "ecaeae9b-0682-4f7c-b0ba-684f3aab0757",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Metadata\n",
    "meta = pd.read_csv('qiita_downloads/pese_pangenome_align-RS210-masked/sample_information_from_prep_16181.tsv', sep = '\\t')\n",
    "\n",
    "#Micov pass/fail list\n",
    "qza_list = ['processed_data/TechRep_merged/pese_pangenome_align-RS210_masked_none_scrubbed_micovFilter_T.qza']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "3d40c823-16ce-4b4c-9ed7-6457d281741d",
   "metadata": {},
   "outputs": [],
   "source": [
    "#Import taxonomys to split between genome and species\n",
    "\n",
    "#Path to taxonomic tress\n",
    "taxonomy_path='/Users/cguccion/Dropbox/Storage/HelpfulLabDocs/taxonomy_trees'\n",
    "\n",
    "#RS210\n",
    "rs210_tax = pd.read_csv(taxonomy_path + '/RS210/RS210.txt', sep='\\t', index_col=0, header=None)\n",
    "rs210_tax.columns = ['Taxon']\n",
    "rs210_tax.index.name = 'Feature ID'\n",
    "rs210_tax = Artifact.import_data('FeatureData[Taxonomy]', rs210_tax)\n",
    "\n",
    "#RS210 - no space\n",
    "rs210_tax_ns = pd.read_csv(taxonomy_path + '/RS210/RS210_noSpace.txt', sep='\\t', index_col=0, header=None)\n",
    "rs210_tax_ns.columns = ['Taxon']\n",
    "rs210_tax_ns.index.name = 'Feature ID'\n",
    "rs210_tax_ns = Artifact.import_data('FeatureData[Taxonomy]', rs210_tax_ns)\n",
    "\n",
    "#WOL2 (shouldn't need)\n",
    "wol2_tax = pd.read_csv(taxonomy_path + '/WOL2/lineages.txt', sep='\\t', index_col=0, header=None)\n",
    "wol2_tax.columns = ['Taxon']\n",
    "wol2_tax.index.name = 'Feature ID'\n",
    "wol2_tax = Artifact.import_data('FeatureData[Taxonomy]', wol2_tax)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0c2b9c93-bbcb-4205-a97f-714150167130",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "09883c78-8fa7-4ecc-97e7-7ffa4d84afac",
   "metadata": {},
   "outputs": [],
   "source": [
    "def data_split(disease_type, tissue_type, fn, species=False):\n",
    "    #Create metadata\n",
    "    meta_custom =  meta[(meta.tumor_type.isin(disease_type)) & \n",
    "                        (meta.host_sample_type.isin(tissue_type))]\n",
    "    meta_custom = meta_custom.reset_index(drop = True)\n",
    "    \n",
    "    ##Adjust the metadata for the technical replicates combined##\n",
    "\n",
    "    #Find sample_names ending with '.2' (technical replicates that have been combined into thier non-.2 duplicate)\n",
    "    reps_to_drop = meta_custom[meta_custom['sample_name'].str.endswith('.2')].index\n",
    "    \n",
    "    # Identify rows without a corresponding '.2' row (samples without a technical replicate -- we can't use these becuase not equal to other samples)\n",
    "    singleRows_to_drop = meta_custom[~meta_custom['sample_name'].str.endswith('.2') & ~meta_custom['sample_name'].apply(lambda x: f\"{x}.2\").isin(meta_custom['sample_name'])].index\n",
    "    \n",
    "    #Drop rows\n",
    "    rows_to_drop = reps_to_drop.union(singleRows_to_drop)\n",
    "    meta_custom = meta_custom.drop(rows_to_drop)\n",
    "    \n",
    "    #Export metadata\n",
    "    meta_filename = 'processed_data/metadata/metadata_' + fn + '.tsv'\n",
    "    meta_custom.to_csv(meta_filename, sep = '\\t', index = False)\n",
    "    \n",
    "    #Export taxonomy tables in pandas df format and biom format\n",
    "    for qza_path in qza_list:\n",
    "        \n",
    "        #Import full qza table\n",
    "        qza = Artifact.load(qza_path).view(pd.DataFrame).T\n",
    "        \n",
    "        #Subset qza table to just those features meta\n",
    "        df_custom = qza[qza.columns.intersection(meta_custom['sample_name'].tolist())]\n",
    "        \n",
    "        #save as pandas df\n",
    "        filename = 'processed_data/pandas_df/' + fn + '_' + qza_path.split('/')[-1].split('.qza')[0] + '.tsv'\n",
    "        df_custom.to_csv(filename, sep = '\\t')\n",
    "        \n",
    "        #save as qza\n",
    "        custom_qza = Artifact.import_data(\"FeatureTable[Frequency]\", df_custom.T)\n",
    "        filename = 'processed_data/qza/' + fn + '_' + qza_path.split('/')[-1].split('.qza')[0] + '.qza'\n",
    "        custom_qza.save(filename)\n",
    "        \n",
    "        #Save qza as biom table\n",
    "        filename = 'processed_data/biom/' + fn + '_' + qza_path.split('/')[-1].split('.qza')[0]\n",
    "        Artifact.export_data(custom_qza, filename)\n",
    "        \n",
    "        #Convert qza from genome to species\n",
    "        if 'rs210' in qza_path or 'RS210' in qza_path:\n",
    "            current_taxa = rs210_tax\n",
    "        else:\n",
    "            current_taxa = wol2_tax\n",
    "        qza_custom_species = taxa.methods.collapse(table=custom_qza,\n",
    "                                             taxonomy=current_taxa, level=7).collapsed_table\n",
    "        df_custom_species = qza_custom_species.view(pd.DataFrame).T\n",
    "        \n",
    "        #Save species as pandas df \n",
    "        filename ='processed_data/pandas_df/' + fn + '_' + qza_path.split('/')[-1].split('.qza')[0] + '_species.tsv'\n",
    "        df_custom_species.to_csv(filename, sep = '\\t')\n",
    "        \n",
    "        #Save species as qza\n",
    "        filename ='processed_data/qza/' + fn + '_' + qza_path.split('/')[-1].split('.qza')[0] + '_species.qza'\n",
    "        qza_custom_species.save(filename)\n",
    "        \n",
    "        #Save sepcies as biom\n",
    "        filename ='processed_data/biom/' + fn + '_' + qza_path.split('/')[-1].split('.qza')[0] + '_species'\n",
    "        Artifact.export_data(qza_custom_species, filename)\n",
    "    \n",
    "    print(len(meta_custom), 'samples')\n",
    "    display(meta_custom[:3])\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ba9770f8-e8bb-4e6a-a5fc-2e35e8529672",
   "metadata": {
    "tags": []
   },
   "source": [
    "### Data split based on Aims"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2ce6ab8d-f6e2-4298-8f26-5a750bb1466e",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "#### HCC Tissue Tumor + HCC Tissue Adj."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "6dd36d26-791b-4670-ae0d-bed7781c0d26",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "32 samples\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>empo_1</th>\n",
       "      <th>empo_2</th>\n",
       "      <th>empo_3</th>\n",
       "      <th>empo_4</th>\n",
       "      <th>country</th>\n",
       "      <th>latitude</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>elevation</th>\n",
       "      <th>env_biome</th>\n",
       "      <th>...</th>\n",
       "      <th>scientific_name</th>\n",
       "      <th>host_common_name</th>\n",
       "      <th>host_sample_type</th>\n",
       "      <th>host_body_habitat</th>\n",
       "      <th>host_body_product</th>\n",
       "      <th>qiita_sample_type</th>\n",
       "      <th>collection_timestamp</th>\n",
       "      <th>host_scientific_name</th>\n",
       "      <th>physical_specimen_location</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15336.HCC.1.Background</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>Host-associated (non-saline)</td>\n",
       "      <td>Animal (non-saline)</td>\n",
       "      <td>Animal corpus (non-saline)</td>\n",
       "      <td>USA</td>\n",
       "      <td>29.643946</td>\n",
       "      <td>1685930</td>\n",
       "      <td>54</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>...</td>\n",
       "      <td>liver metagenome</td>\n",
       "      <td>human</td>\n",
       "      <td>Background</td>\n",
       "      <td>UBERON:hepatobiliary system</td>\n",
       "      <td>UBERON:parenchyma</td>\n",
       "      <td>tumor</td>\n",
       "      <td>2018</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>University of Florida</td>\n",
       "      <td>15336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15336.HCC.1.tumor</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>Host-associated (non-saline)</td>\n",
       "      <td>Animal (non-saline)</td>\n",
       "      <td>Animal corpus (non-saline)</td>\n",
       "      <td>USA</td>\n",
       "      <td>29.643946</td>\n",
       "      <td>1685930</td>\n",
       "      <td>54</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>...</td>\n",
       "      <td>liver metagenome</td>\n",
       "      <td>human</td>\n",
       "      <td>tumor</td>\n",
       "      <td>UBERON:hepatobiliary system</td>\n",
       "      <td>UBERON:parenchyma</td>\n",
       "      <td>tumor</td>\n",
       "      <td>2018</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>University of Florida</td>\n",
       "      <td>15336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15336.HCC.10.Background</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>Host-associated (non-saline)</td>\n",
       "      <td>Animal (non-saline)</td>\n",
       "      <td>Animal corpus (non-saline)</td>\n",
       "      <td>USA</td>\n",
       "      <td>29.643946</td>\n",
       "      <td>1685930</td>\n",
       "      <td>54</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>...</td>\n",
       "      <td>liver metagenome</td>\n",
       "      <td>human</td>\n",
       "      <td>Background</td>\n",
       "      <td>UBERON:hepatobiliary system</td>\n",
       "      <td>UBERON:parenchyma</td>\n",
       "      <td>tumor</td>\n",
       "      <td>2019</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>University of Florida</td>\n",
       "      <td>15336</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               sample_name           empo_1                        empo_2  \\\n",
       "0   15336.HCC.1.Background  Host-associated  Host-associated (non-saline)   \n",
       "2        15336.HCC.1.tumor  Host-associated  Host-associated (non-saline)   \n",
       "4  15336.HCC.10.Background  Host-associated  Host-associated (non-saline)   \n",
       "\n",
       "                empo_3                      empo_4 country   latitude  \\\n",
       "0  Animal (non-saline)  Animal corpus (non-saline)     USA  29.643946   \n",
       "2  Animal (non-saline)  Animal corpus (non-saline)     USA  29.643946   \n",
       "4  Animal (non-saline)  Animal corpus (non-saline)     USA  29.643946   \n",
       "\n",
       "   taxon_id  elevation    env_biome  ...   scientific_name host_common_name  \\\n",
       "0   1685930         54  urban biome  ...  liver metagenome            human   \n",
       "2   1685930         54  urban biome  ...  liver metagenome            human   \n",
       "4   1685930         54  urban biome  ...  liver metagenome            human   \n",
       "\n",
       "  host_sample_type            host_body_habitat  host_body_product  \\\n",
       "0       Background  UBERON:hepatobiliary system  UBERON:parenchyma   \n",
       "2            tumor  UBERON:hepatobiliary system  UBERON:parenchyma   \n",
       "4       Background  UBERON:hepatobiliary system  UBERON:parenchyma   \n",
       "\n",
       "  qiita_sample_type collection_timestamp host_scientific_name  \\\n",
       "0             tumor                 2018         Homo sapiens   \n",
       "2             tumor                 2018         Homo sapiens   \n",
       "4             tumor                 2019         Homo sapiens   \n",
       "\n",
       "  physical_specimen_location qiita_study_id  \n",
       "0      University of Florida          15336  \n",
       "2      University of Florida          15336  \n",
       "4      University of Florida          15336  \n",
       "\n",
       "[3 rows x 38 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "data_split(['HCC'], ['Background', 'tumor'], 'HCC_adj_v_tumor')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5a6d2fcb-1950-4ed6-993f-ed4b783f44a6",
   "metadata": {
    "jp-MarkdownHeadingCollapsed": true,
    "tags": []
   },
   "source": [
    "#### CRC Tissue Tumor + CRC Tissue Adj."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "6ff263fa-35d1-4baf-8aae-c50d013acaec",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22 samples\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>empo_1</th>\n",
       "      <th>empo_2</th>\n",
       "      <th>empo_3</th>\n",
       "      <th>empo_4</th>\n",
       "      <th>country</th>\n",
       "      <th>latitude</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>elevation</th>\n",
       "      <th>env_biome</th>\n",
       "      <th>...</th>\n",
       "      <th>scientific_name</th>\n",
       "      <th>host_common_name</th>\n",
       "      <th>host_sample_type</th>\n",
       "      <th>host_body_habitat</th>\n",
       "      <th>host_body_product</th>\n",
       "      <th>qiita_sample_type</th>\n",
       "      <th>collection_timestamp</th>\n",
       "      <th>host_scientific_name</th>\n",
       "      <th>physical_specimen_location</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15336.CRC.1.Background</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>Host-associated (non-saline)</td>\n",
       "      <td>Animal (non-saline)</td>\n",
       "      <td>Animal corpus (non-saline)</td>\n",
       "      <td>USA</td>\n",
       "      <td>29.643946</td>\n",
       "      <td>1685930</td>\n",
       "      <td>54</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>...</td>\n",
       "      <td>liver metagenome</td>\n",
       "      <td>human</td>\n",
       "      <td>Background</td>\n",
       "      <td>UBERON:hepatobiliary system</td>\n",
       "      <td>UBERON:parenchyma</td>\n",
       "      <td>tumor</td>\n",
       "      <td>2018</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>University of Florida</td>\n",
       "      <td>15336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15336.CRC.1.tumor</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>Host-associated (non-saline)</td>\n",
       "      <td>Animal (non-saline)</td>\n",
       "      <td>Animal corpus (non-saline)</td>\n",
       "      <td>USA</td>\n",
       "      <td>29.643946</td>\n",
       "      <td>1685930</td>\n",
       "      <td>54</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>...</td>\n",
       "      <td>liver metagenome</td>\n",
       "      <td>human</td>\n",
       "      <td>tumor</td>\n",
       "      <td>UBERON:hepatobiliary system</td>\n",
       "      <td>UBERON:parenchyma</td>\n",
       "      <td>tumor</td>\n",
       "      <td>2018</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>University of Florida</td>\n",
       "      <td>15336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15336.CRC.10.Background</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>Host-associated (non-saline)</td>\n",
       "      <td>Animal (non-saline)</td>\n",
       "      <td>Animal corpus (non-saline)</td>\n",
       "      <td>USA</td>\n",
       "      <td>29.643946</td>\n",
       "      <td>1685930</td>\n",
       "      <td>54</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>...</td>\n",
       "      <td>liver metagenome</td>\n",
       "      <td>human</td>\n",
       "      <td>Background</td>\n",
       "      <td>UBERON:hepatobiliary system</td>\n",
       "      <td>UBERON:parenchyma</td>\n",
       "      <td>tumor</td>\n",
       "      <td>2019</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>University of Florida</td>\n",
       "      <td>15336</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               sample_name           empo_1                        empo_2  \\\n",
       "0   15336.CRC.1.Background  Host-associated  Host-associated (non-saline)   \n",
       "2        15336.CRC.1.tumor  Host-associated  Host-associated (non-saline)   \n",
       "4  15336.CRC.10.Background  Host-associated  Host-associated (non-saline)   \n",
       "\n",
       "                empo_3                      empo_4 country   latitude  \\\n",
       "0  Animal (non-saline)  Animal corpus (non-saline)     USA  29.643946   \n",
       "2  Animal (non-saline)  Animal corpus (non-saline)     USA  29.643946   \n",
       "4  Animal (non-saline)  Animal corpus (non-saline)     USA  29.643946   \n",
       "\n",
       "   taxon_id  elevation    env_biome  ...   scientific_name host_common_name  \\\n",
       "0   1685930         54  urban biome  ...  liver metagenome            human   \n",
       "2   1685930         54  urban biome  ...  liver metagenome            human   \n",
       "4   1685930         54  urban biome  ...  liver metagenome            human   \n",
       "\n",
       "  host_sample_type            host_body_habitat  host_body_product  \\\n",
       "0       Background  UBERON:hepatobiliary system  UBERON:parenchyma   \n",
       "2            tumor  UBERON:hepatobiliary system  UBERON:parenchyma   \n",
       "4       Background  UBERON:hepatobiliary system  UBERON:parenchyma   \n",
       "\n",
       "  qiita_sample_type collection_timestamp host_scientific_name  \\\n",
       "0             tumor                 2018         Homo sapiens   \n",
       "2             tumor                 2018         Homo sapiens   \n",
       "4             tumor                 2019         Homo sapiens   \n",
       "\n",
       "  physical_specimen_location qiita_study_id  \n",
       "0      University of Florida          15336  \n",
       "2      University of Florida          15336  \n",
       "4      University of Florida          15336  \n",
       "\n",
       "[3 rows x 38 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "data_split(['CRC'], ['Background', 'tumor'], 'CRC_adj_v_tumor')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c67a417a-d055-4444-84b3-149a0fc85e9c",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### HCC Tissue Tumor + CRC Tissue Tumor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "f8efeedd-85a5-42cb-b226-1a4a829a738e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 samples\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>empo_1</th>\n",
       "      <th>empo_2</th>\n",
       "      <th>empo_3</th>\n",
       "      <th>empo_4</th>\n",
       "      <th>country</th>\n",
       "      <th>latitude</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>elevation</th>\n",
       "      <th>env_biome</th>\n",
       "      <th>...</th>\n",
       "      <th>scientific_name</th>\n",
       "      <th>host_common_name</th>\n",
       "      <th>host_sample_type</th>\n",
       "      <th>host_body_habitat</th>\n",
       "      <th>host_body_product</th>\n",
       "      <th>qiita_sample_type</th>\n",
       "      <th>collection_timestamp</th>\n",
       "      <th>host_scientific_name</th>\n",
       "      <th>physical_specimen_location</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15336.CRC.1.tumor</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>Host-associated (non-saline)</td>\n",
       "      <td>Animal (non-saline)</td>\n",
       "      <td>Animal corpus (non-saline)</td>\n",
       "      <td>USA</td>\n",
       "      <td>29.643946</td>\n",
       "      <td>1685930</td>\n",
       "      <td>54</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>...</td>\n",
       "      <td>liver metagenome</td>\n",
       "      <td>human</td>\n",
       "      <td>tumor</td>\n",
       "      <td>UBERON:hepatobiliary system</td>\n",
       "      <td>UBERON:parenchyma</td>\n",
       "      <td>tumor</td>\n",
       "      <td>2018</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>University of Florida</td>\n",
       "      <td>15336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15336.CRC.10.tumor</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>Host-associated (non-saline)</td>\n",
       "      <td>Animal (non-saline)</td>\n",
       "      <td>Animal corpus (non-saline)</td>\n",
       "      <td>USA</td>\n",
       "      <td>29.643946</td>\n",
       "      <td>1685930</td>\n",
       "      <td>54</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>...</td>\n",
       "      <td>liver metagenome</td>\n",
       "      <td>human</td>\n",
       "      <td>tumor</td>\n",
       "      <td>UBERON:hepatobiliary system</td>\n",
       "      <td>UBERON:parenchyma</td>\n",
       "      <td>tumor</td>\n",
       "      <td>2019</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>University of Florida</td>\n",
       "      <td>15336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15336.CRC.11.tumor</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>Host-associated (non-saline)</td>\n",
       "      <td>Animal (non-saline)</td>\n",
       "      <td>Animal corpus (non-saline)</td>\n",
       "      <td>USA</td>\n",
       "      <td>29.643946</td>\n",
       "      <td>1685930</td>\n",
       "      <td>54</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>...</td>\n",
       "      <td>liver metagenome</td>\n",
       "      <td>human</td>\n",
       "      <td>tumor</td>\n",
       "      <td>UBERON:hepatobiliary system</td>\n",
       "      <td>UBERON:parenchyma</td>\n",
       "      <td>tumor</td>\n",
       "      <td>2019</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>University of Florida</td>\n",
       "      <td>15336</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          sample_name           empo_1                        empo_2  \\\n",
       "0   15336.CRC.1.tumor  Host-associated  Host-associated (non-saline)   \n",
       "2  15336.CRC.10.tumor  Host-associated  Host-associated (non-saline)   \n",
       "4  15336.CRC.11.tumor  Host-associated  Host-associated (non-saline)   \n",
       "\n",
       "                empo_3                      empo_4 country   latitude  \\\n",
       "0  Animal (non-saline)  Animal corpus (non-saline)     USA  29.643946   \n",
       "2  Animal (non-saline)  Animal corpus (non-saline)     USA  29.643946   \n",
       "4  Animal (non-saline)  Animal corpus (non-saline)     USA  29.643946   \n",
       "\n",
       "   taxon_id  elevation    env_biome  ...   scientific_name host_common_name  \\\n",
       "0   1685930         54  urban biome  ...  liver metagenome            human   \n",
       "2   1685930         54  urban biome  ...  liver metagenome            human   \n",
       "4   1685930         54  urban biome  ...  liver metagenome            human   \n",
       "\n",
       "  host_sample_type            host_body_habitat  host_body_product  \\\n",
       "0            tumor  UBERON:hepatobiliary system  UBERON:parenchyma   \n",
       "2            tumor  UBERON:hepatobiliary system  UBERON:parenchyma   \n",
       "4            tumor  UBERON:hepatobiliary system  UBERON:parenchyma   \n",
       "\n",
       "  qiita_sample_type collection_timestamp host_scientific_name  \\\n",
       "0             tumor                 2018         Homo sapiens   \n",
       "2             tumor                 2019         Homo sapiens   \n",
       "4             tumor                 2019         Homo sapiens   \n",
       "\n",
       "  physical_specimen_location qiita_study_id  \n",
       "0      University of Florida          15336  \n",
       "2      University of Florida          15336  \n",
       "4      University of Florida          15336  \n",
       "\n",
       "[3 rows x 38 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "data_split(['HCC', 'CRC'], ['tumor'], 'tumor_HCC_v_CRC_T')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "7224c9c6-d16b-49bb-be41-d2b89ae62bd8",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### HCC Blood + CRC Blood"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "ec98e293-1fc5-4185-bea2-d6ef213e6c8d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "25 samples\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>empo_1</th>\n",
       "      <th>empo_2</th>\n",
       "      <th>empo_3</th>\n",
       "      <th>empo_4</th>\n",
       "      <th>country</th>\n",
       "      <th>latitude</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>elevation</th>\n",
       "      <th>env_biome</th>\n",
       "      <th>...</th>\n",
       "      <th>scientific_name</th>\n",
       "      <th>host_common_name</th>\n",
       "      <th>host_sample_type</th>\n",
       "      <th>host_body_habitat</th>\n",
       "      <th>host_body_product</th>\n",
       "      <th>qiita_sample_type</th>\n",
       "      <th>collection_timestamp</th>\n",
       "      <th>host_scientific_name</th>\n",
       "      <th>physical_specimen_location</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15336.CRC.1.Plasma</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>Host-associated (non-saline)</td>\n",
       "      <td>Animal (non-saline)</td>\n",
       "      <td>Animal secretion (non-saline)</td>\n",
       "      <td>USA</td>\n",
       "      <td>29.643946</td>\n",
       "      <td>1504969</td>\n",
       "      <td>54</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>...</td>\n",
       "      <td>human blood metagenome</td>\n",
       "      <td>human</td>\n",
       "      <td>Plasma</td>\n",
       "      <td>UBERON:blood</td>\n",
       "      <td>UBERON:plasma</td>\n",
       "      <td>plasma</td>\n",
       "      <td>2018</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>University of Florida</td>\n",
       "      <td>15336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15336.CRC.10.Plasma</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>Host-associated (non-saline)</td>\n",
       "      <td>Animal (non-saline)</td>\n",
       "      <td>Animal secretion (non-saline)</td>\n",
       "      <td>USA</td>\n",
       "      <td>29.643946</td>\n",
       "      <td>1504969</td>\n",
       "      <td>54</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>...</td>\n",
       "      <td>human blood metagenome</td>\n",
       "      <td>human</td>\n",
       "      <td>Plasma</td>\n",
       "      <td>UBERON:blood</td>\n",
       "      <td>UBERON:plasma</td>\n",
       "      <td>plasma</td>\n",
       "      <td>2019</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>University of Florida</td>\n",
       "      <td>15336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15336.CRC.11.Plasma</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>Host-associated (non-saline)</td>\n",
       "      <td>Animal (non-saline)</td>\n",
       "      <td>Animal secretion (non-saline)</td>\n",
       "      <td>USA</td>\n",
       "      <td>29.643946</td>\n",
       "      <td>1504969</td>\n",
       "      <td>54</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>...</td>\n",
       "      <td>human blood metagenome</td>\n",
       "      <td>human</td>\n",
       "      <td>Plasma</td>\n",
       "      <td>UBERON:blood</td>\n",
       "      <td>UBERON:plasma</td>\n",
       "      <td>plasma</td>\n",
       "      <td>2019</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>University of Florida</td>\n",
       "      <td>15336</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           sample_name           empo_1                        empo_2  \\\n",
       "0   15336.CRC.1.Plasma  Host-associated  Host-associated (non-saline)   \n",
       "2  15336.CRC.10.Plasma  Host-associated  Host-associated (non-saline)   \n",
       "4  15336.CRC.11.Plasma  Host-associated  Host-associated (non-saline)   \n",
       "\n",
       "                empo_3                         empo_4 country   latitude  \\\n",
       "0  Animal (non-saline)  Animal secretion (non-saline)     USA  29.643946   \n",
       "2  Animal (non-saline)  Animal secretion (non-saline)     USA  29.643946   \n",
       "4  Animal (non-saline)  Animal secretion (non-saline)     USA  29.643946   \n",
       "\n",
       "   taxon_id  elevation    env_biome  ...         scientific_name  \\\n",
       "0   1504969         54  urban biome  ...  human blood metagenome   \n",
       "2   1504969         54  urban biome  ...  human blood metagenome   \n",
       "4   1504969         54  urban biome  ...  human blood metagenome   \n",
       "\n",
       "  host_common_name host_sample_type host_body_habitat host_body_product  \\\n",
       "0            human           Plasma      UBERON:blood     UBERON:plasma   \n",
       "2            human           Plasma      UBERON:blood     UBERON:plasma   \n",
       "4            human           Plasma      UBERON:blood     UBERON:plasma   \n",
       "\n",
       "  qiita_sample_type collection_timestamp host_scientific_name  \\\n",
       "0            plasma                 2018         Homo sapiens   \n",
       "2            plasma                 2019         Homo sapiens   \n",
       "4            plasma                 2019         Homo sapiens   \n",
       "\n",
       "  physical_specimen_location qiita_study_id  \n",
       "0      University of Florida          15336  \n",
       "2      University of Florida          15336  \n",
       "4      University of Florida          15336  \n",
       "\n",
       "[3 rows x 38 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "data_split(['HCC', 'CRC'], ['Plasma'], 'blood_HCC_v_CRC_T')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "220ba095-aa30-435b-aec7-dfc454573f62",
   "metadata": {
    "tags": []
   },
   "source": [
    "#### HCC Tissue Adj + CRC Tissue Adj"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "0624b0f1-bb09-411c-809d-872b5e3fd58a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27 samples\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_name</th>\n",
       "      <th>empo_1</th>\n",
       "      <th>empo_2</th>\n",
       "      <th>empo_3</th>\n",
       "      <th>empo_4</th>\n",
       "      <th>country</th>\n",
       "      <th>latitude</th>\n",
       "      <th>taxon_id</th>\n",
       "      <th>elevation</th>\n",
       "      <th>env_biome</th>\n",
       "      <th>...</th>\n",
       "      <th>scientific_name</th>\n",
       "      <th>host_common_name</th>\n",
       "      <th>host_sample_type</th>\n",
       "      <th>host_body_habitat</th>\n",
       "      <th>host_body_product</th>\n",
       "      <th>qiita_sample_type</th>\n",
       "      <th>collection_timestamp</th>\n",
       "      <th>host_scientific_name</th>\n",
       "      <th>physical_specimen_location</th>\n",
       "      <th>qiita_study_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>15336.CRC.1.Background</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>Host-associated (non-saline)</td>\n",
       "      <td>Animal (non-saline)</td>\n",
       "      <td>Animal corpus (non-saline)</td>\n",
       "      <td>USA</td>\n",
       "      <td>29.643946</td>\n",
       "      <td>1685930</td>\n",
       "      <td>54</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>...</td>\n",
       "      <td>liver metagenome</td>\n",
       "      <td>human</td>\n",
       "      <td>Background</td>\n",
       "      <td>UBERON:hepatobiliary system</td>\n",
       "      <td>UBERON:parenchyma</td>\n",
       "      <td>tumor</td>\n",
       "      <td>2018</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>University of Florida</td>\n",
       "      <td>15336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>15336.CRC.10.Background</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>Host-associated (non-saline)</td>\n",
       "      <td>Animal (non-saline)</td>\n",
       "      <td>Animal corpus (non-saline)</td>\n",
       "      <td>USA</td>\n",
       "      <td>29.643946</td>\n",
       "      <td>1685930</td>\n",
       "      <td>54</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>...</td>\n",
       "      <td>liver metagenome</td>\n",
       "      <td>human</td>\n",
       "      <td>Background</td>\n",
       "      <td>UBERON:hepatobiliary system</td>\n",
       "      <td>UBERON:parenchyma</td>\n",
       "      <td>tumor</td>\n",
       "      <td>2019</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>University of Florida</td>\n",
       "      <td>15336</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>15336.CRC.11.Background</td>\n",
       "      <td>Host-associated</td>\n",
       "      <td>Host-associated (non-saline)</td>\n",
       "      <td>Animal (non-saline)</td>\n",
       "      <td>Animal corpus (non-saline)</td>\n",
       "      <td>USA</td>\n",
       "      <td>29.643946</td>\n",
       "      <td>1685930</td>\n",
       "      <td>54</td>\n",
       "      <td>urban biome</td>\n",
       "      <td>...</td>\n",
       "      <td>liver metagenome</td>\n",
       "      <td>human</td>\n",
       "      <td>Background</td>\n",
       "      <td>UBERON:hepatobiliary system</td>\n",
       "      <td>UBERON:parenchyma</td>\n",
       "      <td>tumor</td>\n",
       "      <td>2019</td>\n",
       "      <td>Homo sapiens</td>\n",
       "      <td>University of Florida</td>\n",
       "      <td>15336</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3 rows × 38 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               sample_name           empo_1                        empo_2  \\\n",
       "0   15336.CRC.1.Background  Host-associated  Host-associated (non-saline)   \n",
       "2  15336.CRC.10.Background  Host-associated  Host-associated (non-saline)   \n",
       "4  15336.CRC.11.Background  Host-associated  Host-associated (non-saline)   \n",
       "\n",
       "                empo_3                      empo_4 country   latitude  \\\n",
       "0  Animal (non-saline)  Animal corpus (non-saline)     USA  29.643946   \n",
       "2  Animal (non-saline)  Animal corpus (non-saline)     USA  29.643946   \n",
       "4  Animal (non-saline)  Animal corpus (non-saline)     USA  29.643946   \n",
       "\n",
       "   taxon_id  elevation    env_biome  ...   scientific_name host_common_name  \\\n",
       "0   1685930         54  urban biome  ...  liver metagenome            human   \n",
       "2   1685930         54  urban biome  ...  liver metagenome            human   \n",
       "4   1685930         54  urban biome  ...  liver metagenome            human   \n",
       "\n",
       "  host_sample_type            host_body_habitat  host_body_product  \\\n",
       "0       Background  UBERON:hepatobiliary system  UBERON:parenchyma   \n",
       "2       Background  UBERON:hepatobiliary system  UBERON:parenchyma   \n",
       "4       Background  UBERON:hepatobiliary system  UBERON:parenchyma   \n",
       "\n",
       "  qiita_sample_type collection_timestamp host_scientific_name  \\\n",
       "0             tumor                 2018         Homo sapiens   \n",
       "2             tumor                 2019         Homo sapiens   \n",
       "4             tumor                 2019         Homo sapiens   \n",
       "\n",
       "  physical_specimen_location qiita_study_id  \n",
       "0      University of Florida          15336  \n",
       "2      University of Florida          15336  \n",
       "4      University of Florida          15336  \n",
       "\n",
       "[3 rows x 38 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "data_split(['HCC', 'CRC'], ['Background'], 'adj_HCC_v_CRC_T')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "be4970bd-9a50-4de8-8bb5-b2ccec5e5907",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "comad",
   "language": "python",
   "name": "comad"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
